// oci-rover/lib/model/cluster-type.js
"use strict";
/**
 * RoverCloudService API
 * A description of the RoverCloudService API.
 * OpenAPI spec version: 20201210
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterType = void 0;
/**
 * Possible rover node types.
 **/
var ClusterType;
(function (ClusterType) {
    ClusterType["Standalone"] = "STANDALONE";
    ClusterType["Station"] = "STATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ClusterType["UnknownValue"] = "UNKNOWN_VALUE";
})(ClusterType = exports.ClusterType || (exports.ClusterType = {}));
(function (ClusterType) {
    function getJsonObj(obj) {
        return obj;
    }
    ClusterType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ClusterType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClusterType = exports.ClusterType || (exports.ClusterType = {}));
