// oci-stackmonitoring/lib/model/discovery-job.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryJob = void 0;
const model = __importStar(require("../model"));
var DiscoveryJob;
(function (DiscoveryJob) {
    let DiscoveryType;
    (function (DiscoveryType) {
        DiscoveryType["Add"] = "ADD";
        DiscoveryType["AddWithRetry"] = "ADD_WITH_RETRY";
        DiscoveryType["Refresh"] = "REFRESH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DiscoveryType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DiscoveryType = DiscoveryJob.DiscoveryType || (DiscoveryJob.DiscoveryType = {}));
    let Status;
    (function (Status) {
        Status["Success"] = "SUCCESS";
        Status["Failure"] = "FAILURE";
        Status["Inprogress"] = "INPROGRESS";
        Status["Inactive"] = "INACTIVE";
        Status["Created"] = "CREATED";
        Status["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = DiscoveryJob.Status || (DiscoveryJob.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "discoveryDetails": obj.discoveryDetails
                ? model.DiscoveryDetails.getJsonObj(obj.discoveryDetails)
                : undefined
        });
        return jsonObj;
    }
    DiscoveryJob.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "discoveryDetails": obj.discoveryDetails
                ? model.DiscoveryDetails.getDeserializedJsonObj(obj.discoveryDetails)
                : undefined
        });
        return jsonObj;
    }
    DiscoveryJob.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryJob = exports.DiscoveryJob || (exports.DiscoveryJob = {}));
