// oci-stackmonitoring/lib/model/discovery-job-log-summary.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryJobLogSummary = void 0;
var DiscoveryJobLogSummary;
(function (DiscoveryJobLogSummary) {
    let LogType;
    (function (LogType) {
        LogType["Info"] = "INFO";
        LogType["Warning"] = "WARNING";
        LogType["Error"] = "ERROR";
        LogType["Success"] = "SUCCESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LogType["UnknownValue"] = "UNKNOWN_VALUE";
    })(LogType = DiscoveryJobLogSummary.LogType || (DiscoveryJobLogSummary.LogType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DiscoveryJobLogSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DiscoveryJobLogSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryJobLogSummary = exports.DiscoveryJobLogSummary || (exports.DiscoveryJobLogSummary = {}));
