// oci-stackmonitoring/lib/model/discovery-details.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryDetails = void 0;
const model = __importStar(require("../model"));
var DiscoveryDetails;
(function (DiscoveryDetails) {
    let ResourceType;
    (function (ResourceType) {
        ResourceType["WeblogicDomain"] = "WEBLOGIC_DOMAIN";
        ResourceType["EbsInstance"] = "EBS_INSTANCE";
        ResourceType["SqlServer"] = "SQL_SERVER";
        ResourceType["ApacheTomcat"] = "APACHE_TOMCAT";
        ResourceType["OracleDatabase"] = "ORACLE_DATABASE";
        ResourceType["OciOracleDb"] = "OCI_ORACLE_DB";
        ResourceType["OciOracleCdb"] = "OCI_ORACLE_CDB";
        ResourceType["OciOraclePdb"] = "OCI_ORACLE_PDB";
        ResourceType["Host"] = "HOST";
        ResourceType["OraclePsft"] = "ORACLE_PSFT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResourceType = DiscoveryDetails.ResourceType || (DiscoveryDetails.ResourceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "properties": obj.properties ? model.PropertyDetails.getJsonObj(obj.properties) : undefined,
            "credentials": obj.credentials
                ? model.CredentialCollection.getJsonObj(obj.credentials)
                : undefined,
            "tags": obj.tags ? model.PropertyDetails.getJsonObj(obj.tags) : undefined
        });
        return jsonObj;
    }
    DiscoveryDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "properties": obj.properties
                ? model.PropertyDetails.getDeserializedJsonObj(obj.properties)
                : undefined,
            "credentials": obj.credentials
                ? model.CredentialCollection.getDeserializedJsonObj(obj.credentials)
                : undefined,
            "tags": obj.tags ? model.PropertyDetails.getDeserializedJsonObj(obj.tags) : undefined
        });
        return jsonObj;
    }
    DiscoveryDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryDetails = exports.DiscoveryDetails || (exports.DiscoveryDetails = {}));
