// oci-stackmonitoring/lib/model/index.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateMonitoredResourceDetails = exports.UpdateAndPropagateTagsDetails = exports.SortOrder = exports.SearchMonitoredResourcesDetails = exports.SearchMonitoredResourceMembersDetails = exports.SearchMonitoredResourceAssociationsDetails = exports.SearchAssociatedResourcesDetails = exports.ResourceLifecycleState = exports.PropertyDetails = exports.OperationType = exports.OperationStatus = exports.MonitoredResourceSummary = exports.MonitoredResourceProperty = exports.MonitoredResourceMembersCollection = exports.MonitoredResourceMemberSummary = exports.MonitoredResourceCredential = exports.MonitoredResourceCollection = exports.MonitoredResourceAssociationsCollection = exports.MonitoredResourceAssociationSummary = exports.MonitoredResourceAssociation = exports.MonitoredResourceAliasSourceCredential = exports.MonitoredResourceAliasCredential = exports.MonitoredResource = exports.LifecycleState = exports.DiscoveryJobSummary = exports.DiscoveryJobLogSummary = exports.DiscoveryJobLogCollection = exports.DiscoveryJobCollection = exports.DiscoveryJob = exports.DiscoveryDetails = exports.DisassociateMonitoredResourcesDetails = exports.CredentialProperty = exports.CredentialDetails = exports.CredentialCollection = exports.CreateMonitoredResourceDetails = exports.CreateDiscoveryJobDetails = exports.ConnectionDetails = exports.ChangeMonitoredResourceCompartmentDetails = exports.AssociationResourceDetails = exports.AssociationDetails = exports.AssociatedResourcesSummary = exports.AssociatedResourcesCollection = exports.AssociatedMonitoredResource = exports.AssociateMonitoredResourcesDetails = exports.ActionType = void 0;
exports.PreExistingCredentials = exports.PlainTextCredentials = exports.EncryptedCredentials = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AssociateMonitoredResourcesDetails = __importStar(require("./associate-monitored-resources-details"));
exports.AssociateMonitoredResourcesDetails = AssociateMonitoredResourcesDetails.AssociateMonitoredResourcesDetails;
const AssociatedMonitoredResource = __importStar(require("./associated-monitored-resource"));
exports.AssociatedMonitoredResource = AssociatedMonitoredResource.AssociatedMonitoredResource;
const AssociatedResourcesCollection = __importStar(require("./associated-resources-collection"));
exports.AssociatedResourcesCollection = AssociatedResourcesCollection.AssociatedResourcesCollection;
const AssociatedResourcesSummary = __importStar(require("./associated-resources-summary"));
exports.AssociatedResourcesSummary = AssociatedResourcesSummary.AssociatedResourcesSummary;
const AssociationDetails = __importStar(require("./association-details"));
exports.AssociationDetails = AssociationDetails.AssociationDetails;
const AssociationResourceDetails = __importStar(require("./association-resource-details"));
exports.AssociationResourceDetails = AssociationResourceDetails.AssociationResourceDetails;
const ChangeMonitoredResourceCompartmentDetails = __importStar(require("./change-monitored-resource-compartment-details"));
exports.ChangeMonitoredResourceCompartmentDetails = ChangeMonitoredResourceCompartmentDetails.ChangeMonitoredResourceCompartmentDetails;
const ConnectionDetails = __importStar(require("./connection-details"));
exports.ConnectionDetails = ConnectionDetails.ConnectionDetails;
const CreateDiscoveryJobDetails = __importStar(require("./create-discovery-job-details"));
exports.CreateDiscoveryJobDetails = CreateDiscoveryJobDetails.CreateDiscoveryJobDetails;
const CreateMonitoredResourceDetails = __importStar(require("./create-monitored-resource-details"));
exports.CreateMonitoredResourceDetails = CreateMonitoredResourceDetails.CreateMonitoredResourceDetails;
const CredentialCollection = __importStar(require("./credential-collection"));
exports.CredentialCollection = CredentialCollection.CredentialCollection;
const CredentialDetails = __importStar(require("./credential-details"));
exports.CredentialDetails = CredentialDetails.CredentialDetails;
const CredentialProperty = __importStar(require("./credential-property"));
exports.CredentialProperty = CredentialProperty.CredentialProperty;
const DisassociateMonitoredResourcesDetails = __importStar(require("./disassociate-monitored-resources-details"));
exports.DisassociateMonitoredResourcesDetails = DisassociateMonitoredResourcesDetails.DisassociateMonitoredResourcesDetails;
const DiscoveryDetails = __importStar(require("./discovery-details"));
exports.DiscoveryDetails = DiscoveryDetails.DiscoveryDetails;
const DiscoveryJob = __importStar(require("./discovery-job"));
exports.DiscoveryJob = DiscoveryJob.DiscoveryJob;
const DiscoveryJobCollection = __importStar(require("./discovery-job-collection"));
exports.DiscoveryJobCollection = DiscoveryJobCollection.DiscoveryJobCollection;
const DiscoveryJobLogCollection = __importStar(require("./discovery-job-log-collection"));
exports.DiscoveryJobLogCollection = DiscoveryJobLogCollection.DiscoveryJobLogCollection;
const DiscoveryJobLogSummary = __importStar(require("./discovery-job-log-summary"));
exports.DiscoveryJobLogSummary = DiscoveryJobLogSummary.DiscoveryJobLogSummary;
const DiscoveryJobSummary = __importStar(require("./discovery-job-summary"));
exports.DiscoveryJobSummary = DiscoveryJobSummary.DiscoveryJobSummary;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const MonitoredResource = __importStar(require("./monitored-resource"));
exports.MonitoredResource = MonitoredResource.MonitoredResource;
const MonitoredResourceAliasCredential = __importStar(require("./monitored-resource-alias-credential"));
exports.MonitoredResourceAliasCredential = MonitoredResourceAliasCredential.MonitoredResourceAliasCredential;
const MonitoredResourceAliasSourceCredential = __importStar(require("./monitored-resource-alias-source-credential"));
exports.MonitoredResourceAliasSourceCredential = MonitoredResourceAliasSourceCredential.MonitoredResourceAliasSourceCredential;
const MonitoredResourceAssociation = __importStar(require("./monitored-resource-association"));
exports.MonitoredResourceAssociation = MonitoredResourceAssociation.MonitoredResourceAssociation;
const MonitoredResourceAssociationSummary = __importStar(require("./monitored-resource-association-summary"));
exports.MonitoredResourceAssociationSummary = MonitoredResourceAssociationSummary.MonitoredResourceAssociationSummary;
const MonitoredResourceAssociationsCollection = __importStar(require("./monitored-resource-associations-collection"));
exports.MonitoredResourceAssociationsCollection = MonitoredResourceAssociationsCollection.MonitoredResourceAssociationsCollection;
const MonitoredResourceCollection = __importStar(require("./monitored-resource-collection"));
exports.MonitoredResourceCollection = MonitoredResourceCollection.MonitoredResourceCollection;
const MonitoredResourceCredential = __importStar(require("./monitored-resource-credential"));
exports.MonitoredResourceCredential = MonitoredResourceCredential.MonitoredResourceCredential;
const MonitoredResourceMemberSummary = __importStar(require("./monitored-resource-member-summary"));
exports.MonitoredResourceMemberSummary = MonitoredResourceMemberSummary.MonitoredResourceMemberSummary;
const MonitoredResourceMembersCollection = __importStar(require("./monitored-resource-members-collection"));
exports.MonitoredResourceMembersCollection = MonitoredResourceMembersCollection.MonitoredResourceMembersCollection;
const MonitoredResourceProperty = __importStar(require("./monitored-resource-property"));
exports.MonitoredResourceProperty = MonitoredResourceProperty.MonitoredResourceProperty;
const MonitoredResourceSummary = __importStar(require("./monitored-resource-summary"));
exports.MonitoredResourceSummary = MonitoredResourceSummary.MonitoredResourceSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PropertyDetails = __importStar(require("./property-details"));
exports.PropertyDetails = PropertyDetails.PropertyDetails;
const ResourceLifecycleState = __importStar(require("./resource-lifecycle-state"));
exports.ResourceLifecycleState = ResourceLifecycleState.ResourceLifecycleState;
const SearchAssociatedResourcesDetails = __importStar(require("./search-associated-resources-details"));
exports.SearchAssociatedResourcesDetails = SearchAssociatedResourcesDetails.SearchAssociatedResourcesDetails;
const SearchMonitoredResourceAssociationsDetails = __importStar(require("./search-monitored-resource-associations-details"));
exports.SearchMonitoredResourceAssociationsDetails = SearchMonitoredResourceAssociationsDetails.SearchMonitoredResourceAssociationsDetails;
const SearchMonitoredResourceMembersDetails = __importStar(require("./search-monitored-resource-members-details"));
exports.SearchMonitoredResourceMembersDetails = SearchMonitoredResourceMembersDetails.SearchMonitoredResourceMembersDetails;
const SearchMonitoredResourcesDetails = __importStar(require("./search-monitored-resources-details"));
exports.SearchMonitoredResourcesDetails = SearchMonitoredResourcesDetails.SearchMonitoredResourcesDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateAndPropagateTagsDetails = __importStar(require("./update-and-propagate-tags-details"));
exports.UpdateAndPropagateTagsDetails = UpdateAndPropagateTagsDetails.UpdateAndPropagateTagsDetails;
const UpdateMonitoredResourceDetails = __importStar(require("./update-monitored-resource-details"));
exports.UpdateMonitoredResourceDetails = UpdateMonitoredResourceDetails.UpdateMonitoredResourceDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const EncryptedCredentials = __importStar(require("./encrypted-credentials"));
exports.EncryptedCredentials = EncryptedCredentials.EncryptedCredentials;
const PlainTextCredentials = __importStar(require("./plain-text-credentials"));
exports.PlainTextCredentials = PlainTextCredentials.PlainTextCredentials;
const PreExistingCredentials = __importStar(require("./pre-existing-credentials"));
exports.PreExistingCredentials = PreExistingCredentials.PreExistingCredentials;
