// oci-stackmonitoring/lib/model/resource-lifecycle-state.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceLifecycleState = void 0;
/**
 * Possible lifecycle states for monitored resource.
 *
 **/
var ResourceLifecycleState;
(function (ResourceLifecycleState) {
    ResourceLifecycleState["Creating"] = "CREATING";
    ResourceLifecycleState["Updating"] = "UPDATING";
    ResourceLifecycleState["Active"] = "ACTIVE";
    ResourceLifecycleState["Deleting"] = "DELETING";
    ResourceLifecycleState["Deleted"] = "DELETED";
    ResourceLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ResourceLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ResourceLifecycleState = exports.ResourceLifecycleState || (exports.ResourceLifecycleState = {}));
(function (ResourceLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    ResourceLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ResourceLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourceLifecycleState = exports.ResourceLifecycleState || (exports.ResourceLifecycleState = {}));
