// oci-stackmonitoring/lib/model/search-monitored-resources-details.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchMonitoredResourcesDetails = void 0;
var SearchMonitoredResourcesDetails;
(function (SearchMonitoredResourcesDetails) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["ResourceName"] = "RESOURCE_NAME";
    })(SortBy = SearchMonitoredResourcesDetails.SortBy || (SearchMonitoredResourcesDetails.SortBy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchMonitoredResourcesDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchMonitoredResourcesDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchMonitoredResourcesDetails = exports.SearchMonitoredResourcesDetails || (exports.SearchMonitoredResourcesDetails = {}));
