// oci-stackmonitoring/lib/model/search-monitored-resource-associations-details.js
"use strict";
/**
 * Stack Monitoring API
 * Stack Monitoring API.
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchMonitoredResourceAssociationsDetails = void 0;
var SearchMonitoredResourceAssociationsDetails;
(function (SearchMonitoredResourceAssociationsDetails) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["AssocType"] = "ASSOC_TYPE";
    })(SortBy = SearchMonitoredResourceAssociationsDetails.SortBy || (SearchMonitoredResourceAssociationsDetails.SortBy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchMonitoredResourceAssociationsDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SearchMonitoredResourceAssociationsDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchMonitoredResourceAssociationsDetails = exports.SearchMonitoredResourceAssociationsDetails || (exports.SearchMonitoredResourceAssociationsDetails = {}));
