// oci-datalabelingservicedataplane/lib/model/index.js
"use strict";
/**
 * Data Labeling API
 * Use Data Labeling API to create Annotations on Images, Texts & Documents, and generate snapshots.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextSelectionEntity = exports.TextMetadata = exports.TextDatasetFormatDetails = exports.ObjectStorageSourceDetails = exports.ObjectStorageDatasetSourceDetails = exports.KeyValueSelectionEntity = exports.ImageObjectSelectionEntity = exports.ImageMetadata = exports.ImageDatasetFormatDetails = exports.GenericEntity = exports.DocumentMetadata = exports.DocumentDatasetFormatDetails = exports.DelimitedFileTypeMetadata = exports.CreateObjectStorageSourceDetails = exports.UpdateRecordDetails = exports.UpdateAnnotationDetails = exports.TextSpan = exports.TextFileTypeMetadata = exports.SourceDetails = exports.SortOrders = exports.RecordSummary = exports.RecordMetadata = exports.RecordCollection = exports.RecordAnalyticsAggregationCollection = exports.RecordAnalyticsAggregation = exports.RecordAggregationDimensions = exports.Record = exports.NormalizedVertex = exports.LabelSet = exports.LabelName = exports.Label = exports.InitialRecordGenerationConfiguration = exports.Entity = exports.DatasetSourceDetails = exports.DatasetFormatDetails = exports.Dataset = exports.CreateSourceDetails = exports.CreateRecordDetails = exports.CreateAnnotationDetails = exports.BoundingPolygon = exports.AnnotationSummary = exports.AnnotationCollection = exports.AnnotationAnalyticsAggregationCollection = exports.AnnotationAnalyticsAggregation = exports.AnnotationAggregationDimensions = exports.Annotation = void 0;
const Annotation = __importStar(require("./annotation"));
exports.Annotation = Annotation.Annotation;
const AnnotationAggregationDimensions = __importStar(require("./annotation-aggregation-dimensions"));
exports.AnnotationAggregationDimensions = AnnotationAggregationDimensions.AnnotationAggregationDimensions;
const AnnotationAnalyticsAggregation = __importStar(require("./annotation-analytics-aggregation"));
exports.AnnotationAnalyticsAggregation = AnnotationAnalyticsAggregation.AnnotationAnalyticsAggregation;
const AnnotationAnalyticsAggregationCollection = __importStar(require("./annotation-analytics-aggregation-collection"));
exports.AnnotationAnalyticsAggregationCollection = AnnotationAnalyticsAggregationCollection.AnnotationAnalyticsAggregationCollection;
const AnnotationCollection = __importStar(require("./annotation-collection"));
exports.AnnotationCollection = AnnotationCollection.AnnotationCollection;
const AnnotationSummary = __importStar(require("./annotation-summary"));
exports.AnnotationSummary = AnnotationSummary.AnnotationSummary;
const BoundingPolygon = __importStar(require("./bounding-polygon"));
exports.BoundingPolygon = BoundingPolygon.BoundingPolygon;
const CreateAnnotationDetails = __importStar(require("./create-annotation-details"));
exports.CreateAnnotationDetails = CreateAnnotationDetails.CreateAnnotationDetails;
const CreateRecordDetails = __importStar(require("./create-record-details"));
exports.CreateRecordDetails = CreateRecordDetails.CreateRecordDetails;
const CreateSourceDetails = __importStar(require("./create-source-details"));
exports.CreateSourceDetails = CreateSourceDetails.CreateSourceDetails;
const Dataset = __importStar(require("./dataset"));
exports.Dataset = Dataset.Dataset;
const DatasetFormatDetails = __importStar(require("./dataset-format-details"));
exports.DatasetFormatDetails = DatasetFormatDetails.DatasetFormatDetails;
const DatasetSourceDetails = __importStar(require("./dataset-source-details"));
exports.DatasetSourceDetails = DatasetSourceDetails.DatasetSourceDetails;
const Entity = __importStar(require("./entity"));
exports.Entity = Entity.Entity;
const InitialRecordGenerationConfiguration = __importStar(require("./initial-record-generation-configuration"));
exports.InitialRecordGenerationConfiguration = InitialRecordGenerationConfiguration.InitialRecordGenerationConfiguration;
const Label = __importStar(require("./label"));
exports.Label = Label.Label;
const LabelName = __importStar(require("./label-name"));
exports.LabelName = LabelName.LabelName;
const LabelSet = __importStar(require("./label-set"));
exports.LabelSet = LabelSet.LabelSet;
const NormalizedVertex = __importStar(require("./normalized-vertex"));
exports.NormalizedVertex = NormalizedVertex.NormalizedVertex;
const Record = __importStar(require("./record"));
exports.Record = Record.Record;
const RecordAggregationDimensions = __importStar(require("./record-aggregation-dimensions"));
exports.RecordAggregationDimensions = RecordAggregationDimensions.RecordAggregationDimensions;
const RecordAnalyticsAggregation = __importStar(require("./record-analytics-aggregation"));
exports.RecordAnalyticsAggregation = RecordAnalyticsAggregation.RecordAnalyticsAggregation;
const RecordAnalyticsAggregationCollection = __importStar(require("./record-analytics-aggregation-collection"));
exports.RecordAnalyticsAggregationCollection = RecordAnalyticsAggregationCollection.RecordAnalyticsAggregationCollection;
const RecordCollection = __importStar(require("./record-collection"));
exports.RecordCollection = RecordCollection.RecordCollection;
const RecordMetadata = __importStar(require("./record-metadata"));
exports.RecordMetadata = RecordMetadata.RecordMetadata;
const RecordSummary = __importStar(require("./record-summary"));
exports.RecordSummary = RecordSummary.RecordSummary;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const SourceDetails = __importStar(require("./source-details"));
exports.SourceDetails = SourceDetails.SourceDetails;
const TextFileTypeMetadata = __importStar(require("./text-file-type-metadata"));
exports.TextFileTypeMetadata = TextFileTypeMetadata.TextFileTypeMetadata;
const TextSpan = __importStar(require("./text-span"));
exports.TextSpan = TextSpan.TextSpan;
const UpdateAnnotationDetails = __importStar(require("./update-annotation-details"));
exports.UpdateAnnotationDetails = UpdateAnnotationDetails.UpdateAnnotationDetails;
const UpdateRecordDetails = __importStar(require("./update-record-details"));
exports.UpdateRecordDetails = UpdateRecordDetails.UpdateRecordDetails;
const CreateObjectStorageSourceDetails = __importStar(require("./create-object-storage-source-details"));
exports.CreateObjectStorageSourceDetails = CreateObjectStorageSourceDetails.CreateObjectStorageSourceDetails;
const DelimitedFileTypeMetadata = __importStar(require("./delimited-file-type-metadata"));
exports.DelimitedFileTypeMetadata = DelimitedFileTypeMetadata.DelimitedFileTypeMetadata;
const DocumentDatasetFormatDetails = __importStar(require("./document-dataset-format-details"));
exports.DocumentDatasetFormatDetails = DocumentDatasetFormatDetails.DocumentDatasetFormatDetails;
const DocumentMetadata = __importStar(require("./document-metadata"));
exports.DocumentMetadata = DocumentMetadata.DocumentMetadata;
const GenericEntity = __importStar(require("./generic-entity"));
exports.GenericEntity = GenericEntity.GenericEntity;
const ImageDatasetFormatDetails = __importStar(require("./image-dataset-format-details"));
exports.ImageDatasetFormatDetails = ImageDatasetFormatDetails.ImageDatasetFormatDetails;
const ImageMetadata = __importStar(require("./image-metadata"));
exports.ImageMetadata = ImageMetadata.ImageMetadata;
const ImageObjectSelectionEntity = __importStar(require("./image-object-selection-entity"));
exports.ImageObjectSelectionEntity = ImageObjectSelectionEntity.ImageObjectSelectionEntity;
const KeyValueSelectionEntity = __importStar(require("./key-value-selection-entity"));
exports.KeyValueSelectionEntity = KeyValueSelectionEntity.KeyValueSelectionEntity;
const ObjectStorageDatasetSourceDetails = __importStar(require("./object-storage-dataset-source-details"));
exports.ObjectStorageDatasetSourceDetails = ObjectStorageDatasetSourceDetails.ObjectStorageDatasetSourceDetails;
const ObjectStorageSourceDetails = __importStar(require("./object-storage-source-details"));
exports.ObjectStorageSourceDetails = ObjectStorageSourceDetails.ObjectStorageSourceDetails;
const TextDatasetFormatDetails = __importStar(require("./text-dataset-format-details"));
exports.TextDatasetFormatDetails = TextDatasetFormatDetails.TextDatasetFormatDetails;
const TextMetadata = __importStar(require("./text-metadata"));
exports.TextMetadata = TextMetadata.TextMetadata;
const TextSelectionEntity = __importStar(require("./text-selection-entity"));
exports.TextSelectionEntity = TextSelectionEntity.TextSelectionEntity;
