// oci-databasemigration/lib/model/database-manual-connection-sub-types.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20210929
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseManualConnectionSubTypes = void 0;
/**
 * Supported database manual connection subtypes
 *
 **/
var DatabaseManualConnectionSubTypes;
(function (DatabaseManualConnectionSubTypes) {
    DatabaseManualConnectionSubTypes["Oracle"] = "ORACLE";
    DatabaseManualConnectionSubTypes["RdsOracle"] = "RDS_ORACLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseManualConnectionSubTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseManualConnectionSubTypes = exports.DatabaseManualConnectionSubTypes || (exports.DatabaseManualConnectionSubTypes = {}));
(function (DatabaseManualConnectionSubTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseManualConnectionSubTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseManualConnectionSubTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseManualConnectionSubTypes = exports.DatabaseManualConnectionSubTypes || (exports.DatabaseManualConnectionSubTypes = {}));
