// oci-databasemigration/lib/model/migration-database-target-types-update.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20210929
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationDatabaseTargetTypesUpdate = void 0;
/**
 * Possible Migration Database Target types to specify.
 *
 **/
var MigrationDatabaseTargetTypesUpdate;
(function (MigrationDatabaseTargetTypesUpdate) {
    MigrationDatabaseTargetTypesUpdate["AdbSRemap"] = "ADB_S_REMAP";
    MigrationDatabaseTargetTypesUpdate["AdbDRemap"] = "ADB_D_REMAP";
    MigrationDatabaseTargetTypesUpdate["AdbDAutocreate"] = "ADB_D_AUTOCREATE";
    MigrationDatabaseTargetTypesUpdate["NonAdbRemap"] = "NON_ADB_REMAP";
    MigrationDatabaseTargetTypesUpdate["NonAdbAutocreate"] = "NON_ADB_AUTOCREATE";
    MigrationDatabaseTargetTypesUpdate["TargetDefaultsRemap"] = "TARGET_DEFAULTS_REMAP";
    MigrationDatabaseTargetTypesUpdate["TargetDefaultsAutocreate"] = "TARGET_DEFAULTS_AUTOCREATE";
})(MigrationDatabaseTargetTypesUpdate = exports.MigrationDatabaseTargetTypesUpdate || (exports.MigrationDatabaseTargetTypesUpdate = {}));
(function (MigrationDatabaseTargetTypesUpdate) {
    function getJsonObj(obj) {
        return obj;
    }
    MigrationDatabaseTargetTypesUpdate.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MigrationDatabaseTargetTypesUpdate.getDeserializedJsonObj = getDeserializedJsonObj;
})(MigrationDatabaseTargetTypesUpdate = exports.MigrationDatabaseTargetTypesUpdate || (exports.MigrationDatabaseTargetTypesUpdate = {}));
