// oci-databasemigration/lib/model/migration-database-target-types.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20210929
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationDatabaseTargetTypes = void 0;
/**
 * Possible Migration Database Target types to specify.
 *
 **/
var MigrationDatabaseTargetTypes;
(function (MigrationDatabaseTargetTypes) {
    MigrationDatabaseTargetTypes["AdbSRemap"] = "ADB_S_REMAP";
    MigrationDatabaseTargetTypes["AdbDRemap"] = "ADB_D_REMAP";
    MigrationDatabaseTargetTypes["AdbDAutocreate"] = "ADB_D_AUTOCREATE";
    MigrationDatabaseTargetTypes["NonAdbRemap"] = "NON_ADB_REMAP";
    MigrationDatabaseTargetTypes["NonAdbAutocreate"] = "NON_ADB_AUTOCREATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MigrationDatabaseTargetTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(MigrationDatabaseTargetTypes = exports.MigrationDatabaseTargetTypes || (exports.MigrationDatabaseTargetTypes = {}));
(function (MigrationDatabaseTargetTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    MigrationDatabaseTargetTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MigrationDatabaseTargetTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(MigrationDatabaseTargetTypes = exports.MigrationDatabaseTargetTypes || (exports.MigrationDatabaseTargetTypes = {}));
