// oci-databasemigration/lib/model/migration-lifecycle-states.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20210929
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationLifecycleStates = void 0;
/**
 * Possible Migration lifecycle states.
 *
 **/
var MigrationLifecycleStates;
(function (MigrationLifecycleStates) {
    MigrationLifecycleStates["Creating"] = "CREATING";
    MigrationLifecycleStates["Updating"] = "UPDATING";
    MigrationLifecycleStates["Active"] = "ACTIVE";
    MigrationLifecycleStates["InProgress"] = "IN_PROGRESS";
    MigrationLifecycleStates["Accepted"] = "ACCEPTED";
    MigrationLifecycleStates["Succeeded"] = "SUCCEEDED";
    MigrationLifecycleStates["Canceled"] = "CANCELED";
    MigrationLifecycleStates["Waiting"] = "WAITING";
    MigrationLifecycleStates["NeedsAttention"] = "NEEDS_ATTENTION";
    MigrationLifecycleStates["Inactive"] = "INACTIVE";
    MigrationLifecycleStates["Deleting"] = "DELETING";
    MigrationLifecycleStates["Deleted"] = "DELETED";
    MigrationLifecycleStates["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MigrationLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(MigrationLifecycleStates = exports.MigrationLifecycleStates || (exports.MigrationLifecycleStates = {}));
(function (MigrationLifecycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    MigrationLifecycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MigrationLifecycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(MigrationLifecycleStates = exports.MigrationLifecycleStates || (exports.MigrationLifecycleStates = {}));
