// oci-databasemigration/lib/model/reason-keywords.js
"use strict";
/**
 * Database Migration API
 * Use the Oracle Cloud Infrastructure Database Migration APIs to perform database migration operations.
 * OpenAPI spec version: 20210929
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReasonKeywords = void 0;
/**
 * Object exclusion reason category.
 *
 **/
var ReasonKeywords;
(function (ReasonKeywords) {
    ReasonKeywords["OracleMaintained"] = "ORACLE_MAINTAINED";
    ReasonKeywords["GgUnsupported"] = "GG_UNSUPPORTED";
    ReasonKeywords["UserExcluded"] = "USER_EXCLUDED";
    ReasonKeywords["MandatoryExcluded"] = "MANDATORY_EXCLUDED";
    ReasonKeywords["UserExcludedType"] = "USER_EXCLUDED_TYPE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReasonKeywords["UnknownValue"] = "UNKNOWN_VALUE";
})(ReasonKeywords = exports.ReasonKeywords || (exports.ReasonKeywords = {}));
(function (ReasonKeywords) {
    function getJsonObj(obj) {
        return obj;
    }
    ReasonKeywords.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReasonKeywords.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReasonKeywords = exports.ReasonKeywords || (exports.ReasonKeywords = {}));
