// oci-aivision/lib/model/document-language.js
"use strict";
/**
 * Vision API
 * Using Vision, you can upload images to detect and classify objects in them. If you have lots of images, you can process them in batch using asynchronous API endpoints. Vision's features are thematically split between Document AI for document-centric images, and Image Analysis for object and scene-based images. Pretrained models and custom models are supported.
 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentLanguage = void 0;
/**
 * The language of the document, abbreviated according to ISO 639-2.
 **/
var DocumentLanguage;
(function (DocumentLanguage) {
    DocumentLanguage["Eng"] = "ENG";
    DocumentLanguage["Ces"] = "CES";
    DocumentLanguage["Dan"] = "DAN";
    DocumentLanguage["Nld"] = "NLD";
    DocumentLanguage["Fin"] = "FIN";
    DocumentLanguage["Fra"] = "FRA";
    DocumentLanguage["Deu"] = "DEU";
    DocumentLanguage["Ell"] = "ELL";
    DocumentLanguage["Hun"] = "HUN";
    DocumentLanguage["Ita"] = "ITA";
    DocumentLanguage["Nor"] = "NOR";
    DocumentLanguage["Pol"] = "POL";
    DocumentLanguage["Por"] = "POR";
    DocumentLanguage["Ron"] = "RON";
    DocumentLanguage["Rus"] = "RUS";
    DocumentLanguage["Slk"] = "SLK";
    DocumentLanguage["Spa"] = "SPA";
    DocumentLanguage["Swe"] = "SWE";
    DocumentLanguage["Tur"] = "TUR";
    DocumentLanguage["Ara"] = "ARA";
    DocumentLanguage["ChiSim"] = "CHI_SIM";
    DocumentLanguage["Hin"] = "HIN";
    DocumentLanguage["Jpn"] = "JPN";
    DocumentLanguage["Kor"] = "KOR";
    DocumentLanguage["Others"] = "OTHERS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DocumentLanguage["UnknownValue"] = "UNKNOWN_VALUE";
})(DocumentLanguage = exports.DocumentLanguage || (exports.DocumentLanguage = {}));
(function (DocumentLanguage) {
    function getJsonObj(obj) {
        return obj;
    }
    DocumentLanguage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DocumentLanguage.getDeserializedJsonObj = getDeserializedJsonObj;
})(DocumentLanguage = exports.DocumentLanguage || (exports.DocumentLanguage = {}));
