// oci-aivision/lib/model/index.js
"use strict";
/**
 * Vision API
 * Using Vision, you can upload images to detect and classify objects in them. If you have lots of images, you can process them in batch using asynchronous API endpoints. Vision's features are thematically split between Document AI for document-centric images, and Image Analysis for object and scene-based images. Pretrained models and custom models are supported.
 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrder = exports.ProjectSummary = exports.ProjectCollection = exports.Project = exports.ProcessingError = exports.Page = exports.OutputLocation = exports.OperationType = exports.OperationStatus = exports.OntologyClass = exports.ObjectLocation = exports.NormalizedVertex = exports.ModelSummary = exports.ModelCollection = exports.Model = exports.Line = exports.Label = exports.InputLocation = exports.ImageText = exports.ImageObject = exports.ImageJob = exports.ImageFeature = exports.ImageDetails = exports.FieldValue = exports.FieldName = exports.FieldLabel = exports.DocumentType = exports.DocumentMetadata = exports.DocumentLanguage = exports.DocumentJob = exports.DocumentField = exports.DocumentFeature = exports.DocumentDetails = exports.Dimensions = exports.DetectedLanguage = exports.DetectedDocumentType = exports.Dataset = exports.CreateProjectDetails = exports.CreateModelDetails = exports.CreateImageJobDetails = exports.CreateDocumentJobDetails = exports.ChangeProjectCompartmentDetails = exports.ChangeModelCompartmentDetails = exports.Cell = exports.BoundingPolygon = exports.AnalyzeImageResult = exports.AnalyzeImageDetails = exports.AnalyzeDocumentResult = exports.AnalyzeDocumentDetails = exports.ActionType = void 0;
exports.ValueTime = exports.ValueString = exports.ValuePhoneNumber = exports.ValueNumber = exports.ValueInteger = exports.ValueDate = exports.ValueArray = exports.ObjectStorageImageDetails = exports.ObjectStorageDocumentDetails = exports.ObjectStorageDataset = exports.ObjectListInlineInputLocation = exports.InlineImageDetails = exports.InlineDocumentDetails = exports.ImageTextDetectionFeature = exports.ImageObjectDetectionFeature = exports.ImageClassificationFeature = exports.DocumentTextDetectionFeature = exports.DocumentTableDetectionFeature = exports.DocumentLanguageClassificationFeature = exports.DocumentKeyValueDetectionFeature = exports.DocumentClassificationFeature = exports.DataScienceLabelingDataset = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.Word = exports.UpdateProjectDetails = exports.UpdateModelDetails = exports.TableRow = exports.Table = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AnalyzeDocumentDetails = __importStar(require("./analyze-document-details"));
exports.AnalyzeDocumentDetails = AnalyzeDocumentDetails.AnalyzeDocumentDetails;
const AnalyzeDocumentResult = __importStar(require("./analyze-document-result"));
exports.AnalyzeDocumentResult = AnalyzeDocumentResult.AnalyzeDocumentResult;
const AnalyzeImageDetails = __importStar(require("./analyze-image-details"));
exports.AnalyzeImageDetails = AnalyzeImageDetails.AnalyzeImageDetails;
const AnalyzeImageResult = __importStar(require("./analyze-image-result"));
exports.AnalyzeImageResult = AnalyzeImageResult.AnalyzeImageResult;
const BoundingPolygon = __importStar(require("./bounding-polygon"));
exports.BoundingPolygon = BoundingPolygon.BoundingPolygon;
const Cell = __importStar(require("./cell"));
exports.Cell = Cell.Cell;
const ChangeModelCompartmentDetails = __importStar(require("./change-model-compartment-details"));
exports.ChangeModelCompartmentDetails = ChangeModelCompartmentDetails.ChangeModelCompartmentDetails;
const ChangeProjectCompartmentDetails = __importStar(require("./change-project-compartment-details"));
exports.ChangeProjectCompartmentDetails = ChangeProjectCompartmentDetails.ChangeProjectCompartmentDetails;
const CreateDocumentJobDetails = __importStar(require("./create-document-job-details"));
exports.CreateDocumentJobDetails = CreateDocumentJobDetails.CreateDocumentJobDetails;
const CreateImageJobDetails = __importStar(require("./create-image-job-details"));
exports.CreateImageJobDetails = CreateImageJobDetails.CreateImageJobDetails;
const CreateModelDetails = __importStar(require("./create-model-details"));
exports.CreateModelDetails = CreateModelDetails.CreateModelDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const Dataset = __importStar(require("./dataset"));
exports.Dataset = Dataset.Dataset;
const DetectedDocumentType = __importStar(require("./detected-document-type"));
exports.DetectedDocumentType = DetectedDocumentType.DetectedDocumentType;
const DetectedLanguage = __importStar(require("./detected-language"));
exports.DetectedLanguage = DetectedLanguage.DetectedLanguage;
const Dimensions = __importStar(require("./dimensions"));
exports.Dimensions = Dimensions.Dimensions;
const DocumentDetails = __importStar(require("./document-details"));
exports.DocumentDetails = DocumentDetails.DocumentDetails;
const DocumentFeature = __importStar(require("./document-feature"));
exports.DocumentFeature = DocumentFeature.DocumentFeature;
const DocumentField = __importStar(require("./document-field"));
exports.DocumentField = DocumentField.DocumentField;
const DocumentJob = __importStar(require("./document-job"));
exports.DocumentJob = DocumentJob.DocumentJob;
const DocumentLanguage = __importStar(require("./document-language"));
exports.DocumentLanguage = DocumentLanguage.DocumentLanguage;
const DocumentMetadata = __importStar(require("./document-metadata"));
exports.DocumentMetadata = DocumentMetadata.DocumentMetadata;
const DocumentType = __importStar(require("./document-type"));
exports.DocumentType = DocumentType.DocumentType;
const FieldLabel = __importStar(require("./field-label"));
exports.FieldLabel = FieldLabel.FieldLabel;
const FieldName = __importStar(require("./field-name"));
exports.FieldName = FieldName.FieldName;
const FieldValue = __importStar(require("./field-value"));
exports.FieldValue = FieldValue.FieldValue;
const ImageDetails = __importStar(require("./image-details"));
exports.ImageDetails = ImageDetails.ImageDetails;
const ImageFeature = __importStar(require("./image-feature"));
exports.ImageFeature = ImageFeature.ImageFeature;
const ImageJob = __importStar(require("./image-job"));
exports.ImageJob = ImageJob.ImageJob;
const ImageObject = __importStar(require("./image-object"));
exports.ImageObject = ImageObject.ImageObject;
const ImageText = __importStar(require("./image-text"));
exports.ImageText = ImageText.ImageText;
const InputLocation = __importStar(require("./input-location"));
exports.InputLocation = InputLocation.InputLocation;
const Label = __importStar(require("./label"));
exports.Label = Label.Label;
const Line = __importStar(require("./line"));
exports.Line = Line.Line;
const Model = __importStar(require("./model"));
exports.Model = Model.Model;
const ModelCollection = __importStar(require("./model-collection"));
exports.ModelCollection = ModelCollection.ModelCollection;
const ModelSummary = __importStar(require("./model-summary"));
exports.ModelSummary = ModelSummary.ModelSummary;
const NormalizedVertex = __importStar(require("./normalized-vertex"));
exports.NormalizedVertex = NormalizedVertex.NormalizedVertex;
const ObjectLocation = __importStar(require("./object-location"));
exports.ObjectLocation = ObjectLocation.ObjectLocation;
const OntologyClass = __importStar(require("./ontology-class"));
exports.OntologyClass = OntologyClass.OntologyClass;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OutputLocation = __importStar(require("./output-location"));
exports.OutputLocation = OutputLocation.OutputLocation;
const Page = __importStar(require("./page"));
exports.Page = Page.Page;
const ProcessingError = __importStar(require("./processing-error"));
exports.ProcessingError = ProcessingError.ProcessingError;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectCollection = __importStar(require("./project-collection"));
exports.ProjectCollection = ProjectCollection.ProjectCollection;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Table = __importStar(require("./table"));
exports.Table = Table.Table;
const TableRow = __importStar(require("./table-row"));
exports.TableRow = TableRow.TableRow;
const UpdateModelDetails = __importStar(require("./update-model-details"));
exports.UpdateModelDetails = UpdateModelDetails.UpdateModelDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const Word = __importStar(require("./word"));
exports.Word = Word.Word;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const DataScienceLabelingDataset = __importStar(require("./data-science-labeling-dataset"));
exports.DataScienceLabelingDataset = DataScienceLabelingDataset.DataScienceLabelingDataset;
const DocumentClassificationFeature = __importStar(require("./document-classification-feature"));
exports.DocumentClassificationFeature = DocumentClassificationFeature.DocumentClassificationFeature;
const DocumentKeyValueDetectionFeature = __importStar(require("./document-key-value-detection-feature"));
exports.DocumentKeyValueDetectionFeature = DocumentKeyValueDetectionFeature.DocumentKeyValueDetectionFeature;
const DocumentLanguageClassificationFeature = __importStar(require("./document-language-classification-feature"));
exports.DocumentLanguageClassificationFeature = DocumentLanguageClassificationFeature.DocumentLanguageClassificationFeature;
const DocumentTableDetectionFeature = __importStar(require("./document-table-detection-feature"));
exports.DocumentTableDetectionFeature = DocumentTableDetectionFeature.DocumentTableDetectionFeature;
const DocumentTextDetectionFeature = __importStar(require("./document-text-detection-feature"));
exports.DocumentTextDetectionFeature = DocumentTextDetectionFeature.DocumentTextDetectionFeature;
const ImageClassificationFeature = __importStar(require("./image-classification-feature"));
exports.ImageClassificationFeature = ImageClassificationFeature.ImageClassificationFeature;
const ImageObjectDetectionFeature = __importStar(require("./image-object-detection-feature"));
exports.ImageObjectDetectionFeature = ImageObjectDetectionFeature.ImageObjectDetectionFeature;
const ImageTextDetectionFeature = __importStar(require("./image-text-detection-feature"));
exports.ImageTextDetectionFeature = ImageTextDetectionFeature.ImageTextDetectionFeature;
const InlineDocumentDetails = __importStar(require("./inline-document-details"));
exports.InlineDocumentDetails = InlineDocumentDetails.InlineDocumentDetails;
const InlineImageDetails = __importStar(require("./inline-image-details"));
exports.InlineImageDetails = InlineImageDetails.InlineImageDetails;
const ObjectListInlineInputLocation = __importStar(require("./object-list-inline-input-location"));
exports.ObjectListInlineInputLocation = ObjectListInlineInputLocation.ObjectListInlineInputLocation;
const ObjectStorageDataset = __importStar(require("./object-storage-dataset"));
exports.ObjectStorageDataset = ObjectStorageDataset.ObjectStorageDataset;
const ObjectStorageDocumentDetails = __importStar(require("./object-storage-document-details"));
exports.ObjectStorageDocumentDetails = ObjectStorageDocumentDetails.ObjectStorageDocumentDetails;
const ObjectStorageImageDetails = __importStar(require("./object-storage-image-details"));
exports.ObjectStorageImageDetails = ObjectStorageImageDetails.ObjectStorageImageDetails;
const ValueArray = __importStar(require("./value-array"));
exports.ValueArray = ValueArray.ValueArray;
const ValueDate = __importStar(require("./value-date"));
exports.ValueDate = ValueDate.ValueDate;
const ValueInteger = __importStar(require("./value-integer"));
exports.ValueInteger = ValueInteger.ValueInteger;
const ValueNumber = __importStar(require("./value-number"));
exports.ValueNumber = ValueNumber.ValueNumber;
const ValuePhoneNumber = __importStar(require("./value-phone-number"));
exports.ValuePhoneNumber = ValuePhoneNumber.ValuePhoneNumber;
const ValueString = __importStar(require("./value-string"));
exports.ValueString = ValueString.ValueString;
const ValueTime = __importStar(require("./value-time"));
exports.ValueTime = ValueTime.ValueTime;
