// oci-aivision/lib/model/model.js
"use strict";
/**
 * Vision API
 * Using Vision, you can upload images to detect and classify objects in them. If you have lots of images, you can process them in batch using asynchronous API endpoints. Vision's features are thematically split between Document AI for document-centric images, and Image Analysis for object and scene-based images. Pretrained models and custom models are supported.
 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Model = void 0;
const model = __importStar(require("../model"));
var Model;
(function (Model) {
    let ModelType;
    (function (ModelType) {
        ModelType["ImageClassification"] = "IMAGE_CLASSIFICATION";
        ModelType["ObjectDetection"] = "OBJECT_DETECTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ModelType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ModelType = Model.ModelType || (Model.ModelType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Model.LifecycleState || (Model.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "trainingDataset": obj.trainingDataset
                ? model.Dataset.getJsonObj(obj.trainingDataset)
                : undefined,
            "testingDataset": obj.testingDataset
                ? model.Dataset.getJsonObj(obj.testingDataset)
                : undefined,
            "validationDataset": obj.validationDataset
                ? model.Dataset.getJsonObj(obj.validationDataset)
                : undefined
        });
        return jsonObj;
    }
    Model.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "trainingDataset": obj.trainingDataset
                ? model.Dataset.getDeserializedJsonObj(obj.trainingDataset)
                : undefined,
            "testingDataset": obj.testingDataset
                ? model.Dataset.getDeserializedJsonObj(obj.testingDataset)
                : undefined,
            "validationDataset": obj.validationDataset
                ? model.Dataset.getDeserializedJsonObj(obj.validationDataset)
                : undefined
        });
        return jsonObj;
    }
    Model.getDeserializedJsonObj = getDeserializedJsonObj;
})(Model = exports.Model || (exports.Model = {}));
