// oci-datasafe/lib/request/index.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUserAnalyticsRequest = exports.ListTargetDatabasesRequest = exports.ListTargetAlertPolicyAssociationsRequest = exports.ListTablesRequest = exports.ListSensitiveTypesRequest = exports.ListSensitiveSchemasRequest = exports.ListSensitiveObjectsRequest = exports.ListSensitiveDataModelsRequest = exports.ListSensitiveColumnsRequest = exports.ListSecurityAssessmentsRequest = exports.ListSdmMaskingPolicyDifferencesRequest = exports.ListSchemasRequest = exports.ListRolesRequest = exports.ListReportsRequest = exports.ListReportDefinitionsRequest = exports.ListProfileSummariesRequest = exports.ListProfileAnalyticsRequest = exports.ListOnPremConnectorsRequest = exports.ListMaskingSchemasRequest = exports.ListMaskingReportsRequest = exports.ListMaskingPoliciesRequest = exports.ListMaskingObjectsRequest = exports.ListMaskingColumnsRequest = exports.ListMaskingAnalyticsRequest = exports.ListMaskedColumnsRequest = exports.ListLibraryMaskingFormatsRequest = exports.ListGrantsRequest = exports.ListFindingsRequest = exports.ListDiscoveryJobsRequest = exports.ListDiscoveryJobResultsRequest = exports.ListDiscoveryAnalyticsRequest = exports.ListDifferenceColumnsRequest = exports.ListDataSafePrivateEndpointsRequest = exports.ListColumnsRequest = exports.ListCollectedAuditVolumesRequest = exports.ListAvailableAuditVolumesRequest = exports.ListAuditTrailsRequest = exports.ListAuditTrailAnalyticsRequest = exports.ListAuditProfilesRequest = exports.ListAuditProfileAnalyticsRequest = exports.ListAuditPolicyAnalyticsRequest = exports.ListAuditPoliciesRequest = exports.ListAuditEventsRequest = exports.ListAuditEventAnalyticsRequest = exports.ListAuditArchiveRetrievalsRequest = exports.ListAlertsRequest = exports.ListAlertPoliciesRequest = exports.ListAlertAnalyticsRequest = exports.GetCompatibleFormatsForSensitiveTypesRequest = exports.AlertsUpdateRequest = void 0;
exports.PatchAlertsRequest = exports.ListWorkRequestsRequest = exports.ListUsersRequest = exports.ListUserAssessmentsRequest = void 0;
const AlertsUpdateRequest = __importStar(require("./alerts-update-request"));
exports.AlertsUpdateRequest = AlertsUpdateRequest.AlertsUpdateRequest;
const GetCompatibleFormatsForSensitiveTypesRequest = __importStar(require("./get-compatible-formats-for-sensitive-types-request"));
exports.GetCompatibleFormatsForSensitiveTypesRequest = GetCompatibleFormatsForSensitiveTypesRequest.GetCompatibleFormatsForSensitiveTypesRequest;
const ListAlertAnalyticsRequest = __importStar(require("./list-alert-analytics-request"));
exports.ListAlertAnalyticsRequest = ListAlertAnalyticsRequest.ListAlertAnalyticsRequest;
const ListAlertPoliciesRequest = __importStar(require("./list-alert-policies-request"));
exports.ListAlertPoliciesRequest = ListAlertPoliciesRequest.ListAlertPoliciesRequest;
const ListAlertsRequest = __importStar(require("./list-alerts-request"));
exports.ListAlertsRequest = ListAlertsRequest.ListAlertsRequest;
const ListAuditArchiveRetrievalsRequest = __importStar(require("./list-audit-archive-retrievals-request"));
exports.ListAuditArchiveRetrievalsRequest = ListAuditArchiveRetrievalsRequest.ListAuditArchiveRetrievalsRequest;
const ListAuditEventAnalyticsRequest = __importStar(require("./list-audit-event-analytics-request"));
exports.ListAuditEventAnalyticsRequest = ListAuditEventAnalyticsRequest.ListAuditEventAnalyticsRequest;
const ListAuditEventsRequest = __importStar(require("./list-audit-events-request"));
exports.ListAuditEventsRequest = ListAuditEventsRequest.ListAuditEventsRequest;
const ListAuditPoliciesRequest = __importStar(require("./list-audit-policies-request"));
exports.ListAuditPoliciesRequest = ListAuditPoliciesRequest.ListAuditPoliciesRequest;
const ListAuditPolicyAnalyticsRequest = __importStar(require("./list-audit-policy-analytics-request"));
exports.ListAuditPolicyAnalyticsRequest = ListAuditPolicyAnalyticsRequest.ListAuditPolicyAnalyticsRequest;
const ListAuditProfileAnalyticsRequest = __importStar(require("./list-audit-profile-analytics-request"));
exports.ListAuditProfileAnalyticsRequest = ListAuditProfileAnalyticsRequest.ListAuditProfileAnalyticsRequest;
const ListAuditProfilesRequest = __importStar(require("./list-audit-profiles-request"));
exports.ListAuditProfilesRequest = ListAuditProfilesRequest.ListAuditProfilesRequest;
const ListAuditTrailAnalyticsRequest = __importStar(require("./list-audit-trail-analytics-request"));
exports.ListAuditTrailAnalyticsRequest = ListAuditTrailAnalyticsRequest.ListAuditTrailAnalyticsRequest;
const ListAuditTrailsRequest = __importStar(require("./list-audit-trails-request"));
exports.ListAuditTrailsRequest = ListAuditTrailsRequest.ListAuditTrailsRequest;
const ListAvailableAuditVolumesRequest = __importStar(require("./list-available-audit-volumes-request"));
exports.ListAvailableAuditVolumesRequest = ListAvailableAuditVolumesRequest.ListAvailableAuditVolumesRequest;
const ListCollectedAuditVolumesRequest = __importStar(require("./list-collected-audit-volumes-request"));
exports.ListCollectedAuditVolumesRequest = ListCollectedAuditVolumesRequest.ListCollectedAuditVolumesRequest;
const ListColumnsRequest = __importStar(require("./list-columns-request"));
exports.ListColumnsRequest = ListColumnsRequest.ListColumnsRequest;
const ListDataSafePrivateEndpointsRequest = __importStar(require("./list-data-safe-private-endpoints-request"));
exports.ListDataSafePrivateEndpointsRequest = ListDataSafePrivateEndpointsRequest.ListDataSafePrivateEndpointsRequest;
const ListDifferenceColumnsRequest = __importStar(require("./list-difference-columns-request"));
exports.ListDifferenceColumnsRequest = ListDifferenceColumnsRequest.ListDifferenceColumnsRequest;
const ListDiscoveryAnalyticsRequest = __importStar(require("./list-discovery-analytics-request"));
exports.ListDiscoveryAnalyticsRequest = ListDiscoveryAnalyticsRequest.ListDiscoveryAnalyticsRequest;
const ListDiscoveryJobResultsRequest = __importStar(require("./list-discovery-job-results-request"));
exports.ListDiscoveryJobResultsRequest = ListDiscoveryJobResultsRequest.ListDiscoveryJobResultsRequest;
const ListDiscoveryJobsRequest = __importStar(require("./list-discovery-jobs-request"));
exports.ListDiscoveryJobsRequest = ListDiscoveryJobsRequest.ListDiscoveryJobsRequest;
const ListFindingsRequest = __importStar(require("./list-findings-request"));
exports.ListFindingsRequest = ListFindingsRequest.ListFindingsRequest;
const ListGrantsRequest = __importStar(require("./list-grants-request"));
exports.ListGrantsRequest = ListGrantsRequest.ListGrantsRequest;
const ListLibraryMaskingFormatsRequest = __importStar(require("./list-library-masking-formats-request"));
exports.ListLibraryMaskingFormatsRequest = ListLibraryMaskingFormatsRequest.ListLibraryMaskingFormatsRequest;
const ListMaskedColumnsRequest = __importStar(require("./list-masked-columns-request"));
exports.ListMaskedColumnsRequest = ListMaskedColumnsRequest.ListMaskedColumnsRequest;
const ListMaskingAnalyticsRequest = __importStar(require("./list-masking-analytics-request"));
exports.ListMaskingAnalyticsRequest = ListMaskingAnalyticsRequest.ListMaskingAnalyticsRequest;
const ListMaskingColumnsRequest = __importStar(require("./list-masking-columns-request"));
exports.ListMaskingColumnsRequest = ListMaskingColumnsRequest.ListMaskingColumnsRequest;
const ListMaskingObjectsRequest = __importStar(require("./list-masking-objects-request"));
exports.ListMaskingObjectsRequest = ListMaskingObjectsRequest.ListMaskingObjectsRequest;
const ListMaskingPoliciesRequest = __importStar(require("./list-masking-policies-request"));
exports.ListMaskingPoliciesRequest = ListMaskingPoliciesRequest.ListMaskingPoliciesRequest;
const ListMaskingReportsRequest = __importStar(require("./list-masking-reports-request"));
exports.ListMaskingReportsRequest = ListMaskingReportsRequest.ListMaskingReportsRequest;
const ListMaskingSchemasRequest = __importStar(require("./list-masking-schemas-request"));
exports.ListMaskingSchemasRequest = ListMaskingSchemasRequest.ListMaskingSchemasRequest;
const ListOnPremConnectorsRequest = __importStar(require("./list-on-prem-connectors-request"));
exports.ListOnPremConnectorsRequest = ListOnPremConnectorsRequest.ListOnPremConnectorsRequest;
const ListProfileAnalyticsRequest = __importStar(require("./list-profile-analytics-request"));
exports.ListProfileAnalyticsRequest = ListProfileAnalyticsRequest.ListProfileAnalyticsRequest;
const ListProfileSummariesRequest = __importStar(require("./list-profile-summaries-request"));
exports.ListProfileSummariesRequest = ListProfileSummariesRequest.ListProfileSummariesRequest;
const ListReportDefinitionsRequest = __importStar(require("./list-report-definitions-request"));
exports.ListReportDefinitionsRequest = ListReportDefinitionsRequest.ListReportDefinitionsRequest;
const ListReportsRequest = __importStar(require("./list-reports-request"));
exports.ListReportsRequest = ListReportsRequest.ListReportsRequest;
const ListRolesRequest = __importStar(require("./list-roles-request"));
exports.ListRolesRequest = ListRolesRequest.ListRolesRequest;
const ListSchemasRequest = __importStar(require("./list-schemas-request"));
exports.ListSchemasRequest = ListSchemasRequest.ListSchemasRequest;
const ListSdmMaskingPolicyDifferencesRequest = __importStar(require("./list-sdm-masking-policy-differences-request"));
exports.ListSdmMaskingPolicyDifferencesRequest = ListSdmMaskingPolicyDifferencesRequest.ListSdmMaskingPolicyDifferencesRequest;
const ListSecurityAssessmentsRequest = __importStar(require("./list-security-assessments-request"));
exports.ListSecurityAssessmentsRequest = ListSecurityAssessmentsRequest.ListSecurityAssessmentsRequest;
const ListSensitiveColumnsRequest = __importStar(require("./list-sensitive-columns-request"));
exports.ListSensitiveColumnsRequest = ListSensitiveColumnsRequest.ListSensitiveColumnsRequest;
const ListSensitiveDataModelsRequest = __importStar(require("./list-sensitive-data-models-request"));
exports.ListSensitiveDataModelsRequest = ListSensitiveDataModelsRequest.ListSensitiveDataModelsRequest;
const ListSensitiveObjectsRequest = __importStar(require("./list-sensitive-objects-request"));
exports.ListSensitiveObjectsRequest = ListSensitiveObjectsRequest.ListSensitiveObjectsRequest;
const ListSensitiveSchemasRequest = __importStar(require("./list-sensitive-schemas-request"));
exports.ListSensitiveSchemasRequest = ListSensitiveSchemasRequest.ListSensitiveSchemasRequest;
const ListSensitiveTypesRequest = __importStar(require("./list-sensitive-types-request"));
exports.ListSensitiveTypesRequest = ListSensitiveTypesRequest.ListSensitiveTypesRequest;
const ListTablesRequest = __importStar(require("./list-tables-request"));
exports.ListTablesRequest = ListTablesRequest.ListTablesRequest;
const ListTargetAlertPolicyAssociationsRequest = __importStar(require("./list-target-alert-policy-associations-request"));
exports.ListTargetAlertPolicyAssociationsRequest = ListTargetAlertPolicyAssociationsRequest.ListTargetAlertPolicyAssociationsRequest;
const ListTargetDatabasesRequest = __importStar(require("./list-target-databases-request"));
exports.ListTargetDatabasesRequest = ListTargetDatabasesRequest.ListTargetDatabasesRequest;
const ListUserAnalyticsRequest = __importStar(require("./list-user-analytics-request"));
exports.ListUserAnalyticsRequest = ListUserAnalyticsRequest.ListUserAnalyticsRequest;
const ListUserAssessmentsRequest = __importStar(require("./list-user-assessments-request"));
exports.ListUserAssessmentsRequest = ListUserAssessmentsRequest.ListUserAssessmentsRequest;
const ListUsersRequest = __importStar(require("./list-users-request"));
exports.ListUsersRequest = ListUsersRequest.ListUsersRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const PatchAlertsRequest = __importStar(require("./patch-alerts-request"));
exports.PatchAlertsRequest = PatchAlertsRequest.PatchAlertsRequest;
