// oci-datasafe/lib/request/list-alert-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAlertAnalyticsRequest = void 0;
var ListAlertAnalyticsRequest;
(function (ListAlertAnalyticsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAlertAnalyticsRequest.SortOrder || (ListAlertAnalyticsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListAlertAnalyticsRequest.SortBy || (ListAlertAnalyticsRequest.SortBy = {}));
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAlertAnalyticsRequest.AccessLevel || (ListAlertAnalyticsRequest.AccessLevel = {}));
    let SummaryField;
    (function (SummaryField) {
        SummaryField["AlertType"] = "alertType";
        SummaryField["TargetIds"] = "targetIds";
        SummaryField["TargetNames"] = "targetNames";
        SummaryField["AlertSeverity"] = "alertSeverity";
        SummaryField["AlertStatus"] = "alertStatus";
        SummaryField["TimeCreated"] = "timeCreated";
        SummaryField["PolicyId"] = "policyId";
        SummaryField["Open"] = "open";
        SummaryField["Closed"] = "closed";
        SummaryField["Critical"] = "critical";
        SummaryField["High"] = "high";
        SummaryField["Medium"] = "medium";
        SummaryField["Low"] = "low";
        SummaryField["Alertcount"] = "alertcount";
    })(SummaryField = ListAlertAnalyticsRequest.SummaryField || (ListAlertAnalyticsRequest.SummaryField = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["AlertType"] = "alertType";
        GroupBy["TargetIds"] = "targetIds";
        GroupBy["TargetNames"] = "targetNames";
        GroupBy["AlertSeverity"] = "alertSeverity";
        GroupBy["AlertStatus"] = "alertStatus";
        GroupBy["TimeCreated"] = "timeCreated";
        GroupBy["PolicyId"] = "policyId";
    })(GroupBy = ListAlertAnalyticsRequest.GroupBy || (ListAlertAnalyticsRequest.GroupBy = {}));
})(ListAlertAnalyticsRequest = exports.ListAlertAnalyticsRequest || (exports.ListAlertAnalyticsRequest = {}));
