// oci-datasafe/lib/request/list-security-assessments-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityAssessmentsRequest = void 0;
var ListSecurityAssessmentsRequest;
(function (ListSecurityAssessmentsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListSecurityAssessmentsRequest.AccessLevel || (ListSecurityAssessmentsRequest.AccessLevel = {}));
    let Type;
    (function (Type) {
        Type["Latest"] = "LATEST";
        Type["Saved"] = "SAVED";
        Type["SaveSchedule"] = "SAVE_SCHEDULE";
        Type["Compartment"] = "COMPARTMENT";
    })(Type = ListSecurityAssessmentsRequest.Type || (ListSecurityAssessmentsRequest.Type = {}));
    let TriggeredBy;
    (function (TriggeredBy) {
        TriggeredBy["User"] = "USER";
        TriggeredBy["System"] = "SYSTEM";
    })(TriggeredBy = ListSecurityAssessmentsRequest.TriggeredBy || (ListSecurityAssessmentsRequest.TriggeredBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSecurityAssessmentsRequest.SortOrder || (ListSecurityAssessmentsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListSecurityAssessmentsRequest.SortBy || (ListSecurityAssessmentsRequest.SortBy = {}));
})(ListSecurityAssessmentsRequest = exports.ListSecurityAssessmentsRequest || (exports.ListSecurityAssessmentsRequest = {}));
