// oci-datasafe/lib/request/list-reports-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListReportsRequest = void 0;
var ListReportsRequest;
(function (ListReportsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListReportsRequest.AccessLevel || (ListReportsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListReportsRequest.SortOrder || (ListReportsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeGenerated"] = "timeGenerated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListReportsRequest.SortBy || (ListReportsRequest.SortBy = {}));
})(ListReportsRequest = exports.ListReportsRequest || (exports.ListReportsRequest = {}));
