// oci-datasafe/lib/request/list-profile-summaries-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListProfileSummariesRequest = void 0;
var ListProfileSummariesRequest;
(function (ListProfileSummariesRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListProfileSummariesRequest.AccessLevel || (ListProfileSummariesRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["ProfileName"] = "profileName";
        SortBy["TargetId"] = "targetId";
        SortBy["IsUserCreated"] = "isUserCreated";
        SortBy["PasswordVerificationFunction"] = "passwordVerificationFunction";
        SortBy["UserCount"] = "userCount";
        SortBy["SessionsPerUser"] = "sessionsPerUser";
        SortBy["InactiveAccountTime"] = "inactiveAccountTime";
        SortBy["FailedLoginAttempts"] = "failedLoginAttempts";
    })(SortBy = ListProfileSummariesRequest.SortBy || (ListProfileSummariesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListProfileSummariesRequest.SortOrder || (ListProfileSummariesRequest.SortOrder = {}));
})(ListProfileSummariesRequest = exports.ListProfileSummariesRequest || (exports.ListProfileSummariesRequest = {}));
