// oci-datasafe/lib/request/list-audit-trail-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAuditTrailAnalyticsRequest = void 0;
var ListAuditTrailAnalyticsRequest;
(function (ListAuditTrailAnalyticsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAuditTrailAnalyticsRequest.AccessLevel || (ListAuditTrailAnalyticsRequest.AccessLevel = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["Location"] = "location";
        GroupBy["LifecycleState"] = "lifecycleState";
        GroupBy["Status"] = "status";
        GroupBy["TargetId"] = "targetId";
    })(GroupBy = ListAuditTrailAnalyticsRequest.GroupBy || (ListAuditTrailAnalyticsRequest.GroupBy = {}));
})(ListAuditTrailAnalyticsRequest = exports.ListAuditTrailAnalyticsRequest || (exports.ListAuditTrailAnalyticsRequest = {}));
