// oci-datasafe/lib/request/list-sensitive-objects-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSensitiveObjectsRequest = void 0;
var ListSensitiveObjectsRequest;
(function (ListSensitiveObjectsRequest) {
    let ObjectType;
    (function (ObjectType) {
        ObjectType["All"] = "ALL";
        ObjectType["Table"] = "TABLE";
        ObjectType["EditioningView"] = "EDITIONING_VIEW";
    })(ObjectType = ListSensitiveObjectsRequest.ObjectType || (ListSensitiveObjectsRequest.ObjectType = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSensitiveObjectsRequest.SortOrder || (ListSensitiveObjectsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["SchemaName"] = "schemaName";
        SortBy["ObjectName"] = "objectName";
        SortBy["ObjectType"] = "objectType";
    })(SortBy = ListSensitiveObjectsRequest.SortBy || (ListSensitiveObjectsRequest.SortBy = {}));
})(ListSensitiveObjectsRequest = exports.ListSensitiveObjectsRequest || (exports.ListSensitiveObjectsRequest = {}));
