// oci-datasafe/lib/request/list-user-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUserAnalyticsRequest = void 0;
var ListUserAnalyticsRequest;
(function (ListUserAnalyticsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListUserAnalyticsRequest.AccessLevel || (ListUserAnalyticsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListUserAnalyticsRequest.SortOrder || (ListUserAnalyticsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["UserName"] = "userName";
        SortBy["UserCategory"] = "userCategory";
        SortBy["AccountStatus"] = "accountStatus";
        SortBy["TimeLastLogin"] = "timeLastLogin";
        SortBy["TargetId"] = "targetId";
        SortBy["TimeUserCreated"] = "timeUserCreated";
        SortBy["AuthenticationType"] = "authenticationType";
        SortBy["TimePasswordChanged"] = "timePasswordChanged";
    })(SortBy = ListUserAnalyticsRequest.SortBy || (ListUserAnalyticsRequest.SortBy = {}));
})(ListUserAnalyticsRequest = exports.ListUserAnalyticsRequest || (exports.ListUserAnalyticsRequest = {}));
