// oci-datasafe/lib/request/list-audit-events-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAuditEventsRequest = void 0;
var ListAuditEventsRequest;
(function (ListAuditEventsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAuditEventsRequest.AccessLevel || (ListAuditEventsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAuditEventsRequest.SortOrder || (ListAuditEventsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DbUserName"] = "dbUserName";
        SortBy["TargetName"] = "targetName";
        SortBy["DatabaseType"] = "databaseType";
        SortBy["TargetClass"] = "targetClass";
        SortBy["AuditEventTime"] = "auditEventTime";
        SortBy["TimeCollected"] = "timeCollected";
        SortBy["OsUserName"] = "osUserName";
        SortBy["Operation"] = "operation";
        SortBy["OperationStatus"] = "operationStatus";
        SortBy["EventName"] = "eventName";
        SortBy["ErrorCode"] = "errorCode";
        SortBy["ErrorMessage"] = "errorMessage";
        SortBy["ObjectType"] = "objectType";
        SortBy["ObjectName"] = "objectName";
        SortBy["ObjectOwner"] = "objectOwner";
        SortBy["ClientHostname"] = "clientHostname";
        SortBy["ClientIp"] = "clientIp";
        SortBy["IsAlerted"] = "isAlerted";
        SortBy["ActionTaken"] = "actionTaken";
        SortBy["ClientProgram"] = "clientProgram";
        SortBy["CommandText"] = "commandText";
        SortBy["CommandParam"] = "commandParam";
        SortBy["ExtendedEventAttributes"] = "extendedEventAttributes";
        SortBy["AuditLocation"] = "auditLocation";
        SortBy["OsTerminal"] = "osTerminal";
        SortBy["ClientId"] = "clientId";
        SortBy["AuditPolicies"] = "auditPolicies";
        SortBy["AuditType"] = "auditType";
    })(SortBy = ListAuditEventsRequest.SortBy || (ListAuditEventsRequest.SortBy = {}));
})(ListAuditEventsRequest = exports.ListAuditEventsRequest || (exports.ListAuditEventsRequest = {}));
