// oci-datasafe/lib/request/list-alerts-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAlertsRequest = void 0;
var ListAlertsRequest;
(function (ListAlertsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAlertsRequest.AccessLevel || (ListAlertsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAlertsRequest.SortOrder || (ListAlertsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DisplayName"] = "displayName";
        SortBy["TimeCreated"] = "timeCreated";
    })(SortBy = ListAlertsRequest.SortBy || (ListAlertsRequest.SortBy = {}));
    let Field;
    (function (Field) {
        Field["Id"] = "id";
        Field["DisplayName"] = "displayName";
        Field["AlertType"] = "alertType";
        Field["TargetIds"] = "targetIds";
        Field["TargetNames"] = "targetNames";
        Field["Severity"] = "severity";
        Field["Status"] = "status";
        Field["OperationTime"] = "operationTime";
        Field["Operation"] = "operation";
        Field["OperationStatus"] = "operationStatus";
        Field["TimeCreated"] = "timeCreated";
        Field["TimeUpdated"] = "timeUpdated";
        Field["PolicyId"] = "policyId";
        Field["LifecycleState"] = "lifecycleState";
    })(Field = ListAlertsRequest.Field || (ListAlertsRequest.Field = {}));
})(ListAlertsRequest = exports.ListAlertsRequest || (exports.ListAlertsRequest = {}));
