// oci-servicecatalog/lib/request/index.js
"use strict";
/**
 * Service Catalog API
 * Manage solutions in Oracle Cloud Infrastructure Service Catalog.
 * OpenAPI spec version: 20210527
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListServiceCatalogsRequest = exports.ListServiceCatalogAssociationsRequest = exports.ListPrivateApplicationsRequest = exports.ListPrivateApplicationPackagesRequest = exports.ListApplicationsRequest = void 0;
const ListApplicationsRequest = __importStar(require("./list-applications-request"));
exports.ListApplicationsRequest = ListApplicationsRequest.ListApplicationsRequest;
const ListPrivateApplicationPackagesRequest = __importStar(require("./list-private-application-packages-request"));
exports.ListPrivateApplicationPackagesRequest = ListPrivateApplicationPackagesRequest.ListPrivateApplicationPackagesRequest;
const ListPrivateApplicationsRequest = __importStar(require("./list-private-applications-request"));
exports.ListPrivateApplicationsRequest = ListPrivateApplicationsRequest.ListPrivateApplicationsRequest;
const ListServiceCatalogAssociationsRequest = __importStar(require("./list-service-catalog-associations-request"));
exports.ListServiceCatalogAssociationsRequest = ListServiceCatalogAssociationsRequest.ListServiceCatalogAssociationsRequest;
const ListServiceCatalogsRequest = __importStar(require("./list-service-catalogs-request"));
exports.ListServiceCatalogsRequest = ListServiceCatalogsRequest.ListServiceCatalogsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
