// oci-servicecatalog/lib/request/list-private-application-packages-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210527
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPrivateApplicationPackagesRequest = void 0;
var ListPrivateApplicationPackagesRequest;
(function (ListPrivateApplicationPackagesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Version"] = "VERSION";
    })(SortBy = ListPrivateApplicationPackagesRequest.SortBy || (ListPrivateApplicationPackagesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListPrivateApplicationPackagesRequest.SortOrder || (ListPrivateApplicationPackagesRequest.SortOrder = {}));
})(ListPrivateApplicationPackagesRequest = exports.ListPrivateApplicationPackagesRequest || (exports.ListPrivateApplicationPackagesRequest = {}));
