// oci-waa/lib/model/backend-type.js
"use strict";
/**
 * Web Application Acceleration (WAA) API
 * API for the Web Application Acceleration service.
Use this API to manage regional Web App Acceleration policies such as Caching and Compression
for accelerating HTTP services.

 * OpenAPI spec version: 20211230
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendType = void 0;
/**
 * Type of the WebAppFirewall, as example LOAD_BALANCER.
 **/
var BackendType;
(function (BackendType) {
    BackendType["LoadBalancer"] = "LOAD_BALANCER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    BackendType["UnknownValue"] = "UNKNOWN_VALUE";
})(BackendType = exports.BackendType || (exports.BackendType = {}));
(function (BackendType) {
    function getJsonObj(obj) {
        return obj;
    }
    BackendType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BackendType.getDeserializedJsonObj = getDeserializedJsonObj;
})(BackendType = exports.BackendType || (exports.BackendType = {}));
