// oci-waa/lib/model/index.js
"use strict";
/**
 * Web Application Acceleration (WAA) API
 * API for the Web Application Acceleration service.
Use this API to manage regional Web App Acceleration policies such as Caching and Compression
for accelerating HTTP services.

 * OpenAPI spec version: 20211230
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAccelerationLoadBalancerSummary = exports.WebAppAccelerationLoadBalancer = exports.PurgeEntireWebAppAccelerationCacheDetails = exports.CreateWebAppAccelerationLoadBalancerDetails = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.WebAppAccelerationSummary = exports.WebAppAccelerationPolicySummary = exports.WebAppAccelerationPolicyCollection = exports.WebAppAccelerationPolicy = exports.WebAppAccelerationCollection = exports.WebAppAcceleration = exports.UpdateWebAppAccelerationPolicyDetails = exports.UpdateWebAppAccelerationDetails = exports.SortOrder = exports.ResponseCompressionPolicy = exports.ResponseCachingPolicy = exports.PurgeWebAppAccelerationCacheDetails = exports.GzipCompressionPolicy = exports.CreateWebAppAccelerationPolicyDetails = exports.CreateWebAppAccelerationDetails = exports.ChangeWebAppAccelerationPolicyCompartmentDetails = exports.ChangeWebAppAccelerationCompartmentDetails = exports.ChangeResourceCompartmentDetails = exports.BackendType = void 0;
const BackendType = __importStar(require("./backend-type"));
exports.BackendType = BackendType.BackendType;
const ChangeResourceCompartmentDetails = __importStar(require("./change-resource-compartment-details"));
exports.ChangeResourceCompartmentDetails = ChangeResourceCompartmentDetails.ChangeResourceCompartmentDetails;
const ChangeWebAppAccelerationCompartmentDetails = __importStar(require("./change-web-app-acceleration-compartment-details"));
exports.ChangeWebAppAccelerationCompartmentDetails = ChangeWebAppAccelerationCompartmentDetails.ChangeWebAppAccelerationCompartmentDetails;
const ChangeWebAppAccelerationPolicyCompartmentDetails = __importStar(require("./change-web-app-acceleration-policy-compartment-details"));
exports.ChangeWebAppAccelerationPolicyCompartmentDetails = ChangeWebAppAccelerationPolicyCompartmentDetails.ChangeWebAppAccelerationPolicyCompartmentDetails;
const CreateWebAppAccelerationDetails = __importStar(require("./create-web-app-acceleration-details"));
exports.CreateWebAppAccelerationDetails = CreateWebAppAccelerationDetails.CreateWebAppAccelerationDetails;
const CreateWebAppAccelerationPolicyDetails = __importStar(require("./create-web-app-acceleration-policy-details"));
exports.CreateWebAppAccelerationPolicyDetails = CreateWebAppAccelerationPolicyDetails.CreateWebAppAccelerationPolicyDetails;
const GzipCompressionPolicy = __importStar(require("./gzip-compression-policy"));
exports.GzipCompressionPolicy = GzipCompressionPolicy.GzipCompressionPolicy;
const PurgeWebAppAccelerationCacheDetails = __importStar(require("./purge-web-app-acceleration-cache-details"));
exports.PurgeWebAppAccelerationCacheDetails = PurgeWebAppAccelerationCacheDetails.PurgeWebAppAccelerationCacheDetails;
const ResponseCachingPolicy = __importStar(require("./response-caching-policy"));
exports.ResponseCachingPolicy = ResponseCachingPolicy.ResponseCachingPolicy;
const ResponseCompressionPolicy = __importStar(require("./response-compression-policy"));
exports.ResponseCompressionPolicy = ResponseCompressionPolicy.ResponseCompressionPolicy;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateWebAppAccelerationDetails = __importStar(require("./update-web-app-acceleration-details"));
exports.UpdateWebAppAccelerationDetails = UpdateWebAppAccelerationDetails.UpdateWebAppAccelerationDetails;
const UpdateWebAppAccelerationPolicyDetails = __importStar(require("./update-web-app-acceleration-policy-details"));
exports.UpdateWebAppAccelerationPolicyDetails = UpdateWebAppAccelerationPolicyDetails.UpdateWebAppAccelerationPolicyDetails;
const WebAppAcceleration = __importStar(require("./web-app-acceleration"));
exports.WebAppAcceleration = WebAppAcceleration.WebAppAcceleration;
const WebAppAccelerationCollection = __importStar(require("./web-app-acceleration-collection"));
exports.WebAppAccelerationCollection = WebAppAccelerationCollection.WebAppAccelerationCollection;
const WebAppAccelerationPolicy = __importStar(require("./web-app-acceleration-policy"));
exports.WebAppAccelerationPolicy = WebAppAccelerationPolicy.WebAppAccelerationPolicy;
const WebAppAccelerationPolicyCollection = __importStar(require("./web-app-acceleration-policy-collection"));
exports.WebAppAccelerationPolicyCollection = WebAppAccelerationPolicyCollection.WebAppAccelerationPolicyCollection;
const WebAppAccelerationPolicySummary = __importStar(require("./web-app-acceleration-policy-summary"));
exports.WebAppAccelerationPolicySummary = WebAppAccelerationPolicySummary.WebAppAccelerationPolicySummary;
const WebAppAccelerationSummary = __importStar(require("./web-app-acceleration-summary"));
exports.WebAppAccelerationSummary = WebAppAccelerationSummary.WebAppAccelerationSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const CreateWebAppAccelerationLoadBalancerDetails = __importStar(require("./create-web-app-acceleration-load-balancer-details"));
exports.CreateWebAppAccelerationLoadBalancerDetails = CreateWebAppAccelerationLoadBalancerDetails.CreateWebAppAccelerationLoadBalancerDetails;
const PurgeEntireWebAppAccelerationCacheDetails = __importStar(require("./purge-entire-web-app-acceleration-cache-details"));
exports.PurgeEntireWebAppAccelerationCacheDetails = PurgeEntireWebAppAccelerationCacheDetails.PurgeEntireWebAppAccelerationCacheDetails;
const WebAppAccelerationLoadBalancer = __importStar(require("./web-app-acceleration-load-balancer"));
exports.WebAppAccelerationLoadBalancer = WebAppAccelerationLoadBalancer.WebAppAccelerationLoadBalancer;
const WebAppAccelerationLoadBalancerSummary = __importStar(require("./web-app-acceleration-load-balancer-summary"));
exports.WebAppAccelerationLoadBalancerSummary = WebAppAccelerationLoadBalancerSummary.WebAppAccelerationLoadBalancerSummary;
