// oci-waa/lib/model/web-app-acceleration-policy.js
"use strict";
/**
 * Web Application Acceleration (WAA) API
 * API for the Web Application Acceleration service.
Use this API to manage regional Web App Acceleration policies such as Caching and Compression
for accelerating HTTP services.

 * OpenAPI spec version: 20211230
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAccelerationPolicy = void 0;
const model = __importStar(require("../model"));
var WebAppAccelerationPolicy;
(function (WebAppAccelerationPolicy) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = WebAppAccelerationPolicy.LifecycleState || (WebAppAccelerationPolicy.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "responseCachingPolicy": obj.responseCachingPolicy
                ? model.ResponseCachingPolicy.getJsonObj(obj.responseCachingPolicy)
                : undefined,
            "responseCompressionPolicy": obj.responseCompressionPolicy
                ? model.ResponseCompressionPolicy.getJsonObj(obj.responseCompressionPolicy)
                : undefined
        });
        return jsonObj;
    }
    WebAppAccelerationPolicy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "responseCachingPolicy": obj.responseCachingPolicy
                ? model.ResponseCachingPolicy.getDeserializedJsonObj(obj.responseCachingPolicy)
                : undefined,
            "responseCompressionPolicy": obj.responseCompressionPolicy
                ? model.ResponseCompressionPolicy.getDeserializedJsonObj(obj.responseCompressionPolicy)
                : undefined
        });
        return jsonObj;
    }
    WebAppAccelerationPolicy.getDeserializedJsonObj = getDeserializedJsonObj;
})(WebAppAccelerationPolicy = exports.WebAppAccelerationPolicy || (exports.WebAppAccelerationPolicy = {}));
