// oci-onesubscription/lib/request/index.js
"use strict";
/**
 * OneSubscription APIs
 * OneSubscription APIs

 * OpenAPI spec version: 20190111
 * Contact: kuaskum_org_ww@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSubscriptionsRequest = exports.ListSubscribedServicesRequest = exports.ListRateCardsRequest = exports.ListOrganizationSubscriptionsRequest = exports.ListInvoicesRequest = exports.ListInvoicelineComputedUsagesRequest = exports.ListComputedUsagesRequest = exports.ListAggregatedComputedUsagesRequest = exports.ListCommitmentsRequest = exports.ListBillingSchedulesRequest = void 0;
const ListBillingSchedulesRequest = __importStar(require("./list-billing-schedules-request"));
exports.ListBillingSchedulesRequest = ListBillingSchedulesRequest.ListBillingSchedulesRequest;
const ListCommitmentsRequest = __importStar(require("./list-commitments-request"));
exports.ListCommitmentsRequest = ListCommitmentsRequest.ListCommitmentsRequest;
const ListAggregatedComputedUsagesRequest = __importStar(require("./list-aggregated-computed-usages-request"));
exports.ListAggregatedComputedUsagesRequest = ListAggregatedComputedUsagesRequest.ListAggregatedComputedUsagesRequest;
const ListComputedUsagesRequest = __importStar(require("./list-computed-usages-request"));
exports.ListComputedUsagesRequest = ListComputedUsagesRequest.ListComputedUsagesRequest;
const ListInvoicelineComputedUsagesRequest = __importStar(require("./list-invoiceline-computed-usages-request"));
exports.ListInvoicelineComputedUsagesRequest = ListInvoicelineComputedUsagesRequest.ListInvoicelineComputedUsagesRequest;
const ListInvoicesRequest = __importStar(require("./list-invoices-request"));
exports.ListInvoicesRequest = ListInvoicesRequest.ListInvoicesRequest;
const ListOrganizationSubscriptionsRequest = __importStar(require("./list-organization-subscriptions-request"));
exports.ListOrganizationSubscriptionsRequest = ListOrganizationSubscriptionsRequest.ListOrganizationSubscriptionsRequest;
const ListRateCardsRequest = __importStar(require("./list-rate-cards-request"));
exports.ListRateCardsRequest = ListRateCardsRequest.ListRateCardsRequest;
const ListSubscribedServicesRequest = __importStar(require("./list-subscribed-services-request"));
exports.ListSubscribedServicesRequest = ListSubscribedServicesRequest.ListSubscribedServicesRequest;
const ListSubscriptionsRequest = __importStar(require("./list-subscriptions-request"));
exports.ListSubscriptionsRequest = ListSubscriptionsRequest.ListSubscriptionsRequest;
