// oci-onesubscription/lib/request/list-billing-schedules-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListBillingSchedulesRequest = void 0;
var ListBillingSchedulesRequest;
(function (ListBillingSchedulesRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListBillingSchedulesRequest.SortOrder || (ListBillingSchedulesRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Ordernumber"] = "ORDERNUMBER";
        SortBy["Timeinvoicing"] = "TIMEINVOICING";
    })(SortBy = ListBillingSchedulesRequest.SortBy || (ListBillingSchedulesRequest.SortBy = {}));
})(ListBillingSchedulesRequest = exports.ListBillingSchedulesRequest || (exports.ListBillingSchedulesRequest = {}));
