// oci-onesubscription/lib/model/computed-usage.js
"use strict";
/**
 * OneSubscription APIs
 * OneSubscription APIs

 * OpenAPI spec version: 20190111
 * Contact: kuaskum_org_ww@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputedUsage = void 0;
const model = __importStar(require("../model"));
var ComputedUsage;
(function (ComputedUsage) {
    let Type;
    (function (Type) {
        Type["Promotion"] = "PROMOTION";
        Type["DoNotBill"] = "DO_NOT_BILL";
        Type["Usage"] = "USAGE";
        Type["Commit"] = "COMMIT";
        Type["Overage"] = "OVERAGE";
        Type["PayAsYouGo"] = "PAY_AS_YOU_GO";
        Type["MonthlyMinimum"] = "MONTHLY_MINIMUM";
        Type["DelayedUsageInvoiceTiming"] = "DELAYED_USAGE_INVOICE_TIMING";
        Type["DelayedUsageCommitmentExp"] = "DELAYED_USAGE_COMMITMENT_EXP";
        Type["OnAccountCredit"] = "ON_ACCOUNT_CREDIT";
        Type["ServiceCredit"] = "SERVICE_CREDIT";
        Type["CommitmentExpiration"] = "COMMITMENT_EXPIRATION";
        Type["FundedAllocation"] = "FUNDED_ALLOCATION";
        Type["DonotBillUsagePostTermination"] = "DONOT_BILL_USAGE_POST_TERMINATION";
        Type["DelayedUsagePostTermination"] = "DELAYED_USAGE_POST_TERMINATION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = ComputedUsage.Type || (ComputedUsage.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "parentProduct": obj.parentProduct
                ? model.ComputedUsageProduct.getJsonObj(obj.parentProduct)
                : undefined,
            "product": obj.product ? model.ComputedUsageProduct.getJsonObj(obj.product) : undefined
        });
        return jsonObj;
    }
    ComputedUsage.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "parentProduct": obj.parentProduct
                ? model.ComputedUsageProduct.getDeserializedJsonObj(obj.parentProduct)
                : undefined,
            "product": obj.product
                ? model.ComputedUsageProduct.getDeserializedJsonObj(obj.product)
                : undefined
        });
        return jsonObj;
    }
    ComputedUsage.getDeserializedJsonObj = getDeserializedJsonObj;
})(ComputedUsage = exports.ComputedUsage || (exports.ComputedUsage = {}));
