// oci-onesubscription/lib/model/index.js
"use strict";
/**
 * OneSubscription APIs
 * OneSubscription APIs

 * OpenAPI spec version: 20190111
 * Contact: kuaskum_org_ww@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionSummary = exports.SubscriptionSubscribedService = exports.SubscriptionProduct = exports.SubscriptionCurrency = exports.SubscribedServiceUser = exports.SubscribedServiceSummary = exports.SubscribedServicePaymentTerm = exports.SubscribedServiceLocation = exports.SubscribedServiceBusinessPartner = exports.SubscribedServiceAddress = exports.SubscribedService = exports.RateCardTier = exports.RateCardSummary = exports.RateCardProduct = exports.OrgnizationSubsCurrency = exports.OrganizationSubscriptionSummary = exports.InvoicingUser = exports.InvoicingProduct = exports.InvoicingPaymentTerm = exports.InvoicingOrganization = exports.InvoicingLocation = exports.InvoicingCurrency = exports.InvoicingBusinessPartner = exports.InvoicingAddress = exports.InvoicelineComputedUsageSummary = exports.InvoiceSummary = exports.InvoiceLineSummary = exports.ComputedUsageSummary = exports.ComputedUsageProduct = exports.ComputedUsageAggregation = exports.ComputedUsage = exports.CommitmentSummary = exports.CommitmentService = exports.Commitment = exports.BillingScheduleSummary = exports.BillingScheduleProduct = exports.AggregatedComputedUsageSummary = void 0;
const AggregatedComputedUsageSummary = __importStar(require("./aggregated-computed-usage-summary"));
exports.AggregatedComputedUsageSummary = AggregatedComputedUsageSummary.AggregatedComputedUsageSummary;
const BillingScheduleProduct = __importStar(require("./billing-schedule-product"));
exports.BillingScheduleProduct = BillingScheduleProduct.BillingScheduleProduct;
const BillingScheduleSummary = __importStar(require("./billing-schedule-summary"));
exports.BillingScheduleSummary = BillingScheduleSummary.BillingScheduleSummary;
const Commitment = __importStar(require("./commitment"));
exports.Commitment = Commitment.Commitment;
const CommitmentService = __importStar(require("./commitment-service"));
exports.CommitmentService = CommitmentService.CommitmentService;
const CommitmentSummary = __importStar(require("./commitment-summary"));
exports.CommitmentSummary = CommitmentSummary.CommitmentSummary;
const ComputedUsage = __importStar(require("./computed-usage"));
exports.ComputedUsage = ComputedUsage.ComputedUsage;
const ComputedUsageAggregation = __importStar(require("./computed-usage-aggregation"));
exports.ComputedUsageAggregation = ComputedUsageAggregation.ComputedUsageAggregation;
const ComputedUsageProduct = __importStar(require("./computed-usage-product"));
exports.ComputedUsageProduct = ComputedUsageProduct.ComputedUsageProduct;
const ComputedUsageSummary = __importStar(require("./computed-usage-summary"));
exports.ComputedUsageSummary = ComputedUsageSummary.ComputedUsageSummary;
const InvoiceLineSummary = __importStar(require("./invoice-line-summary"));
exports.InvoiceLineSummary = InvoiceLineSummary.InvoiceLineSummary;
const InvoiceSummary = __importStar(require("./invoice-summary"));
exports.InvoiceSummary = InvoiceSummary.InvoiceSummary;
const InvoicelineComputedUsageSummary = __importStar(require("./invoiceline-computed-usage-summary"));
exports.InvoicelineComputedUsageSummary = InvoicelineComputedUsageSummary.InvoicelineComputedUsageSummary;
const InvoicingAddress = __importStar(require("./invoicing-address"));
exports.InvoicingAddress = InvoicingAddress.InvoicingAddress;
const InvoicingBusinessPartner = __importStar(require("./invoicing-business-partner"));
exports.InvoicingBusinessPartner = InvoicingBusinessPartner.InvoicingBusinessPartner;
const InvoicingCurrency = __importStar(require("./invoicing-currency"));
exports.InvoicingCurrency = InvoicingCurrency.InvoicingCurrency;
const InvoicingLocation = __importStar(require("./invoicing-location"));
exports.InvoicingLocation = InvoicingLocation.InvoicingLocation;
const InvoicingOrganization = __importStar(require("./invoicing-organization"));
exports.InvoicingOrganization = InvoicingOrganization.InvoicingOrganization;
const InvoicingPaymentTerm = __importStar(require("./invoicing-payment-term"));
exports.InvoicingPaymentTerm = InvoicingPaymentTerm.InvoicingPaymentTerm;
const InvoicingProduct = __importStar(require("./invoicing-product"));
exports.InvoicingProduct = InvoicingProduct.InvoicingProduct;
const InvoicingUser = __importStar(require("./invoicing-user"));
exports.InvoicingUser = InvoicingUser.InvoicingUser;
const OrganizationSubscriptionSummary = __importStar(require("./organization-subscription-summary"));
exports.OrganizationSubscriptionSummary = OrganizationSubscriptionSummary.OrganizationSubscriptionSummary;
const OrgnizationSubsCurrency = __importStar(require("./orgnization-subs-currency"));
exports.OrgnizationSubsCurrency = OrgnizationSubsCurrency.OrgnizationSubsCurrency;
const RateCardProduct = __importStar(require("./rate-card-product"));
exports.RateCardProduct = RateCardProduct.RateCardProduct;
const RateCardSummary = __importStar(require("./rate-card-summary"));
exports.RateCardSummary = RateCardSummary.RateCardSummary;
const RateCardTier = __importStar(require("./rate-card-tier"));
exports.RateCardTier = RateCardTier.RateCardTier;
const SubscribedService = __importStar(require("./subscribed-service"));
exports.SubscribedService = SubscribedService.SubscribedService;
const SubscribedServiceAddress = __importStar(require("./subscribed-service-address"));
exports.SubscribedServiceAddress = SubscribedServiceAddress.SubscribedServiceAddress;
const SubscribedServiceBusinessPartner = __importStar(require("./subscribed-service-business-partner"));
exports.SubscribedServiceBusinessPartner = SubscribedServiceBusinessPartner.SubscribedServiceBusinessPartner;
const SubscribedServiceLocation = __importStar(require("./subscribed-service-location"));
exports.SubscribedServiceLocation = SubscribedServiceLocation.SubscribedServiceLocation;
const SubscribedServicePaymentTerm = __importStar(require("./subscribed-service-payment-term"));
exports.SubscribedServicePaymentTerm = SubscribedServicePaymentTerm.SubscribedServicePaymentTerm;
const SubscribedServiceSummary = __importStar(require("./subscribed-service-summary"));
exports.SubscribedServiceSummary = SubscribedServiceSummary.SubscribedServiceSummary;
const SubscribedServiceUser = __importStar(require("./subscribed-service-user"));
exports.SubscribedServiceUser = SubscribedServiceUser.SubscribedServiceUser;
const SubscriptionCurrency = __importStar(require("./subscription-currency"));
exports.SubscriptionCurrency = SubscriptionCurrency.SubscriptionCurrency;
const SubscriptionProduct = __importStar(require("./subscription-product"));
exports.SubscriptionProduct = SubscriptionProduct.SubscriptionProduct;
const SubscriptionSubscribedService = __importStar(require("./subscription-subscribed-service"));
exports.SubscriptionSubscribedService = SubscriptionSubscribedService.SubscriptionSubscribedService;
const SubscriptionSummary = __importStar(require("./subscription-summary"));
exports.SubscriptionSummary = SubscriptionSummary.SubscriptionSummary;
