// oci-servicemesh/lib/model/external-service-access-policy-target-details.js
"use strict";
/**
 * Service Mesh API
 * Use the Service Mesh API to manage mesh, virtual service, access policy and other mesh related items.
 * OpenAPI spec version: 20220615
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalServiceAccessPolicyTargetDetails = void 0;
const model = __importStar(require("../model"));
var ExternalServiceAccessPolicyTargetDetails;
(function (ExternalServiceAccessPolicyTargetDetails) {
    let Protocol;
    (function (Protocol) {
        Protocol["Http"] = "HTTP";
        Protocol["Https"] = "HTTPS";
        Protocol["Tcp"] = "TCP";
    })(Protocol = ExternalServiceAccessPolicyTargetDetails.Protocol || (ExternalServiceAccessPolicyTargetDetails.Protocol = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.AccessPolicyTargetDetails.getJsonObj(obj))), {});
        return jsonObj;
    }
    ExternalServiceAccessPolicyTargetDetails.getJsonObj = getJsonObj;
    ExternalServiceAccessPolicyTargetDetails.type = "EXTERNAL_SERVICE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.AccessPolicyTargetDetails.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    ExternalServiceAccessPolicyTargetDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalServiceAccessPolicyTargetDetails = exports.ExternalServiceAccessPolicyTargetDetails || (exports.ExternalServiceAccessPolicyTargetDetails = {}));
