// oci-fusionapps/lib/model/action.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Action = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var Action;
(function (Action) {
    let State;
    (function (State) {
        State["Accepted"] = "ACCEPTED";
        State["InProgress"] = "IN_PROGRESS";
        State["Succeeded"] = "SUCCEEDED";
        State["Failed"] = "FAILED";
        State["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        State["UnknownValue"] = "UNKNOWN_VALUE";
    })(State = Action.State || (Action.State = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "actionType" in obj && obj.actionType) {
            switch (obj.actionType) {
                case "PATCH":
                    return model.PatchAction.getJsonObj(jsonObj, true);
                case "QUARTERLY_UPGRADE":
                    return model.UpgradeAction.getJsonObj(jsonObj, true);
                case "VERTEX":
                    return model.VertexAction.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.actionType}`);
            }
        }
        return jsonObj;
    }
    Action.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "actionType" in obj && obj.actionType) {
            switch (obj.actionType) {
                case "PATCH":
                    return model.PatchAction.getDeserializedJsonObj(jsonObj, true);
                case "QUARTERLY_UPGRADE":
                    return model.UpgradeAction.getDeserializedJsonObj(jsonObj, true);
                case "VERTEX":
                    return model.VertexAction.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.actionType}`);
            }
        }
        return jsonObj;
    }
    Action.getDeserializedJsonObj = getDeserializedJsonObj;
})(Action = exports.Action || (exports.Action = {}));
