// oci-fusionapps/lib/model/fusion-environment-status.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FusionEnvironmentStatus = void 0;
var FusionEnvironmentStatus;
(function (FusionEnvironmentStatus) {
    let Status;
    (function (Status) {
        Status["Available"] = "AVAILABLE";
        Status["Unavailable"] = "UNAVAILABLE";
        Status["NotApplicable"] = "NOT_APPLICABLE";
        Status["MaintenanceInProgress"] = "MAINTENANCE_IN_PROGRESS";
        Status["RefreshInProgress"] = "REFRESH_IN_PROGRESS";
        Status["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = FusionEnvironmentStatus.Status || (FusionEnvironmentStatus.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FusionEnvironmentStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FusionEnvironmentStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(FusionEnvironmentStatus = exports.FusionEnvironmentStatus || (exports.FusionEnvironmentStatus = {}));
