// oci-fusionapps/lib/model/fusion-environment.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FusionEnvironment = void 0;
const model = __importStar(require("../model"));
var FusionEnvironment;
(function (FusionEnvironment) {
    let FusionEnvironmentType;
    (function (FusionEnvironmentType) {
        FusionEnvironmentType["Production"] = "PRODUCTION";
        FusionEnvironmentType["Test"] = "TEST";
        FusionEnvironmentType["Development"] = "DEVELOPMENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FusionEnvironmentType["UnknownValue"] = "UNKNOWN_VALUE";
    })(FusionEnvironmentType = FusionEnvironment.FusionEnvironmentType || (FusionEnvironment.FusionEnvironmentType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = FusionEnvironment.LifecycleState || (FusionEnvironment.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "maintenancePolicy": obj.maintenancePolicy
                ? model.GetMaintenancePolicyDetails.getJsonObj(obj.maintenancePolicy)
                : undefined,
            "refresh": obj.refresh ? model.RefreshDetails.getJsonObj(obj.refresh) : undefined,
            "rules": obj.rules
                ? obj.rules.map(item => {
                    return model.Rule.getJsonObj(item);
                })
                : undefined,
            "environmentRole": obj.environmentRole
                ? model.EnvironmentRole.getJsonObj(obj.environmentRole)
                : undefined
        });
        return jsonObj;
    }
    FusionEnvironment.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "maintenancePolicy": obj.maintenancePolicy
                ? model.GetMaintenancePolicyDetails.getDeserializedJsonObj(obj.maintenancePolicy)
                : undefined,
            "refresh": obj.refresh
                ? model.RefreshDetails.getDeserializedJsonObj(obj.refresh)
                : undefined,
            "rules": obj.rules
                ? obj.rules.map(item => {
                    return model.Rule.getDeserializedJsonObj(item);
                })
                : undefined,
            "environmentRole": obj.environmentRole
                ? model.EnvironmentRole.getDeserializedJsonObj(obj.environmentRole)
                : undefined
        });
        return jsonObj;
    }
    FusionEnvironment.getDeserializedJsonObj = getDeserializedJsonObj;
})(FusionEnvironment = exports.FusionEnvironment || (exports.FusionEnvironment = {}));
