// oci-fusionapps/lib/model/index.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeAvailableForRefreshSummary = exports.TimeAvailableForRefreshCollection = exports.TimeAvailableForRefresh = exports.SubscriptionSku = exports.SubscriptionDetail = exports.Subscription = exports.ServiceAttachmentSummary = exports.ServiceAttachmentCollection = exports.ServiceAttachment = exports.ScheduledActivitySummary = exports.ScheduledActivityCollection = exports.ScheduledActivity = exports.RuleCondition = exports.Rule = exports.ResetFusionEnvironmentPasswordDetails = exports.RefreshIssueDetails = exports.RefreshDetails = exports.RefreshActivitySummary = exports.RefreshActivityCollection = exports.RefreshActivity = exports.QuarterlyUpgradeBeginTimes = exports.MaintenancePolicy = exports.LimitAndUsage = exports.KmsKeyInfo = exports.GetMaintenancePolicyDetails = exports.FusionEnvironmentSummary = exports.FusionEnvironmentStatus = exports.FusionEnvironmentFamilySummary = exports.FusionEnvironmentFamilyLimitsAndUsage = exports.FusionEnvironmentFamilyCollection = exports.FusionEnvironmentFamily = exports.FusionEnvironmentCollection = exports.FusionEnvironment = exports.FamilyMaintenancePolicy = exports.EnvironmentRole = exports.DataMaskingActivitySummary = exports.DataMaskingActivityCollection = exports.DataMaskingActivity = exports.CreateServiceAttachmentDetails = exports.CreateRefreshActivityDetails = exports.CreateFusionEnvironmentFamilyDetails = exports.CreateFusionEnvironmentDetails = exports.CreateFusionEnvironmentAdminUserDetails = exports.CreateDataMaskingActivityDetails = exports.ChangeFusionEnvironmentFamilyCompartmentDetails = exports.ChangeFusionEnvironmentCompartmentDetails = exports.Capabilities = exports.AdminUserSummary = exports.AdminUserCollection = exports.Action = void 0;
exports.VertexAction = exports.UpgradeAction = exports.SourceVcnIpAddressCondition = exports.SourceVcnIdCondition = exports.SourceIpAddressCondition = exports.PatchAction = exports.AllowRule = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VerifyServiceAttachmentDetails = exports.UpdateRefreshActivityDetails = exports.UpdateFusionEnvironmentFamilyDetails = exports.UpdateFusionEnvironmentDetails = exports.UpdateFamilyMaintenancePolicyDetails = void 0;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const AdminUserCollection = __importStar(require("./admin-user-collection"));
exports.AdminUserCollection = AdminUserCollection.AdminUserCollection;
const AdminUserSummary = __importStar(require("./admin-user-summary"));
exports.AdminUserSummary = AdminUserSummary.AdminUserSummary;
const Capabilities = __importStar(require("./capabilities"));
exports.Capabilities = Capabilities.Capabilities;
const ChangeFusionEnvironmentCompartmentDetails = __importStar(require("./change-fusion-environment-compartment-details"));
exports.ChangeFusionEnvironmentCompartmentDetails = ChangeFusionEnvironmentCompartmentDetails.ChangeFusionEnvironmentCompartmentDetails;
const ChangeFusionEnvironmentFamilyCompartmentDetails = __importStar(require("./change-fusion-environment-family-compartment-details"));
exports.ChangeFusionEnvironmentFamilyCompartmentDetails = ChangeFusionEnvironmentFamilyCompartmentDetails.ChangeFusionEnvironmentFamilyCompartmentDetails;
const CreateDataMaskingActivityDetails = __importStar(require("./create-data-masking-activity-details"));
exports.CreateDataMaskingActivityDetails = CreateDataMaskingActivityDetails.CreateDataMaskingActivityDetails;
const CreateFusionEnvironmentAdminUserDetails = __importStar(require("./create-fusion-environment-admin-user-details"));
exports.CreateFusionEnvironmentAdminUserDetails = CreateFusionEnvironmentAdminUserDetails.CreateFusionEnvironmentAdminUserDetails;
const CreateFusionEnvironmentDetails = __importStar(require("./create-fusion-environment-details"));
exports.CreateFusionEnvironmentDetails = CreateFusionEnvironmentDetails.CreateFusionEnvironmentDetails;
const CreateFusionEnvironmentFamilyDetails = __importStar(require("./create-fusion-environment-family-details"));
exports.CreateFusionEnvironmentFamilyDetails = CreateFusionEnvironmentFamilyDetails.CreateFusionEnvironmentFamilyDetails;
const CreateRefreshActivityDetails = __importStar(require("./create-refresh-activity-details"));
exports.CreateRefreshActivityDetails = CreateRefreshActivityDetails.CreateRefreshActivityDetails;
const CreateServiceAttachmentDetails = __importStar(require("./create-service-attachment-details"));
exports.CreateServiceAttachmentDetails = CreateServiceAttachmentDetails.CreateServiceAttachmentDetails;
const DataMaskingActivity = __importStar(require("./data-masking-activity"));
exports.DataMaskingActivity = DataMaskingActivity.DataMaskingActivity;
const DataMaskingActivityCollection = __importStar(require("./data-masking-activity-collection"));
exports.DataMaskingActivityCollection = DataMaskingActivityCollection.DataMaskingActivityCollection;
const DataMaskingActivitySummary = __importStar(require("./data-masking-activity-summary"));
exports.DataMaskingActivitySummary = DataMaskingActivitySummary.DataMaskingActivitySummary;
const EnvironmentRole = __importStar(require("./environment-role"));
exports.EnvironmentRole = EnvironmentRole.EnvironmentRole;
const FamilyMaintenancePolicy = __importStar(require("./family-maintenance-policy"));
exports.FamilyMaintenancePolicy = FamilyMaintenancePolicy.FamilyMaintenancePolicy;
const FusionEnvironment = __importStar(require("./fusion-environment"));
exports.FusionEnvironment = FusionEnvironment.FusionEnvironment;
const FusionEnvironmentCollection = __importStar(require("./fusion-environment-collection"));
exports.FusionEnvironmentCollection = FusionEnvironmentCollection.FusionEnvironmentCollection;
const FusionEnvironmentFamily = __importStar(require("./fusion-environment-family"));
exports.FusionEnvironmentFamily = FusionEnvironmentFamily.FusionEnvironmentFamily;
const FusionEnvironmentFamilyCollection = __importStar(require("./fusion-environment-family-collection"));
exports.FusionEnvironmentFamilyCollection = FusionEnvironmentFamilyCollection.FusionEnvironmentFamilyCollection;
const FusionEnvironmentFamilyLimitsAndUsage = __importStar(require("./fusion-environment-family-limits-and-usage"));
exports.FusionEnvironmentFamilyLimitsAndUsage = FusionEnvironmentFamilyLimitsAndUsage.FusionEnvironmentFamilyLimitsAndUsage;
const FusionEnvironmentFamilySummary = __importStar(require("./fusion-environment-family-summary"));
exports.FusionEnvironmentFamilySummary = FusionEnvironmentFamilySummary.FusionEnvironmentFamilySummary;
const FusionEnvironmentStatus = __importStar(require("./fusion-environment-status"));
exports.FusionEnvironmentStatus = FusionEnvironmentStatus.FusionEnvironmentStatus;
const FusionEnvironmentSummary = __importStar(require("./fusion-environment-summary"));
exports.FusionEnvironmentSummary = FusionEnvironmentSummary.FusionEnvironmentSummary;
const GetMaintenancePolicyDetails = __importStar(require("./get-maintenance-policy-details"));
exports.GetMaintenancePolicyDetails = GetMaintenancePolicyDetails.GetMaintenancePolicyDetails;
const KmsKeyInfo = __importStar(require("./kms-key-info"));
exports.KmsKeyInfo = KmsKeyInfo.KmsKeyInfo;
const LimitAndUsage = __importStar(require("./limit-and-usage"));
exports.LimitAndUsage = LimitAndUsage.LimitAndUsage;
const MaintenancePolicy = __importStar(require("./maintenance-policy"));
exports.MaintenancePolicy = MaintenancePolicy.MaintenancePolicy;
const QuarterlyUpgradeBeginTimes = __importStar(require("./quarterly-upgrade-begin-times"));
exports.QuarterlyUpgradeBeginTimes = QuarterlyUpgradeBeginTimes.QuarterlyUpgradeBeginTimes;
const RefreshActivity = __importStar(require("./refresh-activity"));
exports.RefreshActivity = RefreshActivity.RefreshActivity;
const RefreshActivityCollection = __importStar(require("./refresh-activity-collection"));
exports.RefreshActivityCollection = RefreshActivityCollection.RefreshActivityCollection;
const RefreshActivitySummary = __importStar(require("./refresh-activity-summary"));
exports.RefreshActivitySummary = RefreshActivitySummary.RefreshActivitySummary;
const RefreshDetails = __importStar(require("./refresh-details"));
exports.RefreshDetails = RefreshDetails.RefreshDetails;
const RefreshIssueDetails = __importStar(require("./refresh-issue-details"));
exports.RefreshIssueDetails = RefreshIssueDetails.RefreshIssueDetails;
const ResetFusionEnvironmentPasswordDetails = __importStar(require("./reset-fusion-environment-password-details"));
exports.ResetFusionEnvironmentPasswordDetails = ResetFusionEnvironmentPasswordDetails.ResetFusionEnvironmentPasswordDetails;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RuleCondition = __importStar(require("./rule-condition"));
exports.RuleCondition = RuleCondition.RuleCondition;
const ScheduledActivity = __importStar(require("./scheduled-activity"));
exports.ScheduledActivity = ScheduledActivity.ScheduledActivity;
const ScheduledActivityCollection = __importStar(require("./scheduled-activity-collection"));
exports.ScheduledActivityCollection = ScheduledActivityCollection.ScheduledActivityCollection;
const ScheduledActivitySummary = __importStar(require("./scheduled-activity-summary"));
exports.ScheduledActivitySummary = ScheduledActivitySummary.ScheduledActivitySummary;
const ServiceAttachment = __importStar(require("./service-attachment"));
exports.ServiceAttachment = ServiceAttachment.ServiceAttachment;
const ServiceAttachmentCollection = __importStar(require("./service-attachment-collection"));
exports.ServiceAttachmentCollection = ServiceAttachmentCollection.ServiceAttachmentCollection;
const ServiceAttachmentSummary = __importStar(require("./service-attachment-summary"));
exports.ServiceAttachmentSummary = ServiceAttachmentSummary.ServiceAttachmentSummary;
const Subscription = __importStar(require("./subscription"));
exports.Subscription = Subscription.Subscription;
const SubscriptionDetail = __importStar(require("./subscription-detail"));
exports.SubscriptionDetail = SubscriptionDetail.SubscriptionDetail;
const SubscriptionSku = __importStar(require("./subscription-sku"));
exports.SubscriptionSku = SubscriptionSku.SubscriptionSku;
const TimeAvailableForRefresh = __importStar(require("./time-available-for-refresh"));
exports.TimeAvailableForRefresh = TimeAvailableForRefresh.TimeAvailableForRefresh;
const TimeAvailableForRefreshCollection = __importStar(require("./time-available-for-refresh-collection"));
exports.TimeAvailableForRefreshCollection = TimeAvailableForRefreshCollection.TimeAvailableForRefreshCollection;
const TimeAvailableForRefreshSummary = __importStar(require("./time-available-for-refresh-summary"));
exports.TimeAvailableForRefreshSummary = TimeAvailableForRefreshSummary.TimeAvailableForRefreshSummary;
const UpdateFamilyMaintenancePolicyDetails = __importStar(require("./update-family-maintenance-policy-details"));
exports.UpdateFamilyMaintenancePolicyDetails = UpdateFamilyMaintenancePolicyDetails.UpdateFamilyMaintenancePolicyDetails;
const UpdateFusionEnvironmentDetails = __importStar(require("./update-fusion-environment-details"));
exports.UpdateFusionEnvironmentDetails = UpdateFusionEnvironmentDetails.UpdateFusionEnvironmentDetails;
const UpdateFusionEnvironmentFamilyDetails = __importStar(require("./update-fusion-environment-family-details"));
exports.UpdateFusionEnvironmentFamilyDetails = UpdateFusionEnvironmentFamilyDetails.UpdateFusionEnvironmentFamilyDetails;
const UpdateRefreshActivityDetails = __importStar(require("./update-refresh-activity-details"));
exports.UpdateRefreshActivityDetails = UpdateRefreshActivityDetails.UpdateRefreshActivityDetails;
const VerifyServiceAttachmentDetails = __importStar(require("./verify-service-attachment-details"));
exports.VerifyServiceAttachmentDetails = VerifyServiceAttachmentDetails.VerifyServiceAttachmentDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const AllowRule = __importStar(require("./allow-rule"));
exports.AllowRule = AllowRule.AllowRule;
const PatchAction = __importStar(require("./patch-action"));
exports.PatchAction = PatchAction.PatchAction;
const SourceIpAddressCondition = __importStar(require("./source-ip-address-condition"));
exports.SourceIpAddressCondition = SourceIpAddressCondition.SourceIpAddressCondition;
const SourceVcnIdCondition = __importStar(require("./source-vcn-id-condition"));
exports.SourceVcnIdCondition = SourceVcnIdCondition.SourceVcnIdCondition;
const SourceVcnIpAddressCondition = __importStar(require("./source-vcn-ip-address-condition"));
exports.SourceVcnIpAddressCondition = SourceVcnIpAddressCondition.SourceVcnIpAddressCondition;
const UpgradeAction = __importStar(require("./upgrade-action"));
exports.UpgradeAction = UpgradeAction.UpgradeAction;
const VertexAction = __importStar(require("./vertex-action"));
exports.VertexAction = VertexAction.VertexAction;
