// oci-fusionapps/lib/model/maintenance-policy.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenancePolicy = void 0;
var MaintenancePolicy;
(function (MaintenancePolicy) {
    let MonthlyPatchingOverride;
    (function (MonthlyPatchingOverride) {
        MonthlyPatchingOverride["Enabled"] = "ENABLED";
        MonthlyPatchingOverride["Disabled"] = "DISABLED";
        MonthlyPatchingOverride["None"] = "NONE";
    })(MonthlyPatchingOverride = MaintenancePolicy.MonthlyPatchingOverride || (MaintenancePolicy.MonthlyPatchingOverride = {}));
    let EnvironmentMaintenanceOverride;
    (function (EnvironmentMaintenanceOverride) {
        EnvironmentMaintenanceOverride["Prod"] = "PROD";
        EnvironmentMaintenanceOverride["NonProd"] = "NON_PROD";
        EnvironmentMaintenanceOverride["None"] = "NONE";
    })(EnvironmentMaintenanceOverride = MaintenancePolicy.EnvironmentMaintenanceOverride || (MaintenancePolicy.EnvironmentMaintenanceOverride = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaintenancePolicy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    MaintenancePolicy.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenancePolicy = exports.MaintenancePolicy || (exports.MaintenancePolicy = {}));
