// oci-fusionapps/lib/model/service-attachment.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAttachment = void 0;
var ServiceAttachment;
(function (ServiceAttachment) {
    let ServiceInstanceType;
    (function (ServiceInstanceType) {
        ServiceInstanceType["DigitalAssistant"] = "DIGITAL_ASSISTANT";
        ServiceInstanceType["IntegrationCloud"] = "INTEGRATION_CLOUD";
        ServiceInstanceType["AnalyticsWarehouse"] = "ANALYTICS_WAREHOUSE";
        ServiceInstanceType["Vbcs"] = "VBCS";
        ServiceInstanceType["VisualBuilderStudio"] = "VISUAL_BUILDER_STUDIO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ServiceInstanceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ServiceInstanceType = ServiceAttachment.ServiceInstanceType || (ServiceAttachment.ServiceInstanceType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ServiceAttachment.LifecycleState || (ServiceAttachment.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ServiceAttachment.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ServiceAttachment.getDeserializedJsonObj = getDeserializedJsonObj;
})(ServiceAttachment = exports.ServiceAttachment || (exports.ServiceAttachment = {}));
