// oci-fusionapps/lib/model/work-request.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequest = void 0;
const model = __importStar(require("../model"));
var WorkRequest;
(function (WorkRequest) {
    let OperationType;
    (function (OperationType) {
        OperationType["CreateFusionEnvironment"] = "CREATE_FUSION_ENVIRONMENT";
        OperationType["UpdateFusionEnvironment"] = "UPDATE_FUSION_ENVIRONMENT";
        OperationType["ResetFusionEnvironmentAdminPassword"] = "RESET_FUSION_ENVIRONMENT_ADMIN_PASSWORD";
        OperationType["ScaleFusionEnvironment"] = "SCALE_FUSION_ENVIRONMENT";
        OperationType["ArchiveFusionEnvironment"] = "ARCHIVE_FUSION_ENVIRONMENT";
        OperationType["RestoreFusionEnvironment"] = "RESTORE_FUSION_ENVIRONMENT";
        OperationType["CreateServiceInstance"] = "CREATE_SERVICE_INSTANCE";
        OperationType["UpdateServiceInstance"] = "UPDATE_SERVICE_INSTANCE";
        OperationType["DetachServiceInstance"] = "DETACH_SERVICE_INSTANCE";
        OperationType["AddUser"] = "ADD_USER";
        OperationType["RemoveUser"] = "REMOVE_USER";
        OperationType["DeleteFusionEnvironment"] = "DELETE_FUSION_ENVIRONMENT";
        OperationType["ChangeFusionEnvironmentCompartment"] = "CHANGE_FUSION_ENVIRONMENT_COMPARTMENT";
        OperationType["UpgradeFusionEnvironment"] = "UPGRADE_FUSION_ENVIRONMENT";
        OperationType["CreateFusionEnvironmentFamily"] = "CREATE_FUSION_ENVIRONMENT_FAMILY";
        OperationType["DeleteFusionEnvironmentFamily"] = "DELETE_FUSION_ENVIRONMENT_FAMILY";
        OperationType["UpdateFusionEnvironmentFamily"] = "UPDATE_FUSION_ENVIRONMENT_FAMILY";
        OperationType["ChangeFusionEnvironmentFamilyCompartment"] = "CHANGE_FUSION_ENVIRONMENT_FAMILY_COMPARTMENT";
        OperationType["RefreshFusionEnvironment"] = "REFRESH_FUSION_ENVIRONMENT";
        OperationType["ExecuteColdPatch"] = "EXECUTE_COLD_PATCH";
        OperationType["DataMaskFusionEnvironment"] = "DATA_MASK_FUSION_ENVIRONMENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationType = WorkRequest.OperationType || (WorkRequest.OperationType = {}));
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Failed"] = "FAILED";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = WorkRequest.Status || (WorkRequest.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.WorkRequestResource.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    WorkRequest.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "resources": obj.resources
                ? obj.resources.map(item => {
                    return model.WorkRequestResource.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    WorkRequest.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequest = exports.WorkRequest || (exports.WorkRequest = {}));
