// oci-threatintelligence/lib/model/data-visibility.js
"use strict";
/**
 * Threat Intelligence API
 * Use the Threat Intelligence API to search for information about known threat indicators, including suspicious IP addresses, domain names, and other digital fingerprints. Threat Intelligence is a managed database of curated threat intelligence that comes from first party Oracle security insights, open source feeds, and vendor-procured data. For more information, see the [Threat Intelligence documentation](/iaas/Content/threat-intel/home.htm).
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataVisibility = void 0;
var DataVisibility;
(function (DataVisibility) {
    let TlpName;
    (function (TlpName) {
        TlpName["TlpInternalAudit"] = "TLP_INTERNAL_AUDIT";
        TlpName["TlpWhite"] = "TLP_WHITE";
        TlpName["TlpGreen"] = "TLP_GREEN";
        TlpName["TlpAmber"] = "TLP_AMBER";
        TlpName["TlpRed"] = "TLP_RED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TlpName["UnknownValue"] = "UNKNOWN_VALUE";
    })(TlpName = DataVisibility.TlpName || (DataVisibility.TlpName = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DataVisibility.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DataVisibility.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataVisibility = exports.DataVisibility || (exports.DataVisibility = {}));
