// oci-secrets/lib/request/get-secret-bundle-by-name-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSecretBundleByNameRequest = void 0;
var GetSecretBundleByNameRequest;
(function (GetSecretBundleByNameRequest) {
    let Stage;
    (function (Stage) {
        Stage["Current"] = "CURRENT";
        Stage["Pending"] = "PENDING";
        Stage["Latest"] = "LATEST";
        Stage["Previous"] = "PREVIOUS";
        Stage["Deprecated"] = "DEPRECATED";
    })(Stage = GetSecretBundleByNameRequest.Stage || (GetSecretBundleByNameRequest.Stage = {}));
})(GetSecretBundleByNameRequest = exports.GetSecretBundleByNameRequest || (exports.GetSecretBundleByNameRequest = {}));
