// oci-devops/lib/model/deploy-stage-execution-progress.js
"use strict";
/**
 * DevOps API
 * Use the DevOps API to create DevOps projects, configure code repositories,  add artifacts to deploy, build and test software applications, configure  target deployment environments, and deploy software applications.  For more information, see [DevOps](/Content/devops/using/home.htm).
 * OpenAPI spec version: 20210630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeployStageExecutionProgress = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var DeployStageExecutionProgress;
(function (DeployStageExecutionProgress) {
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Failed"] = "FAILED";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        Status["RollbackInProgress"] = "ROLLBACK_IN_PROGRESS";
        Status["RollbackSucceeded"] = "ROLLBACK_SUCCEEDED";
        Status["RollbackFailed"] = "ROLLBACK_FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = DeployStageExecutionProgress.Status || (DeployStageExecutionProgress.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "deployStagePredecessors": obj.deployStagePredecessors
                ? model.DeployStagePredecessorCollection.getJsonObj(obj.deployStagePredecessors)
                : undefined,
            "deployStageExecutionProgressDetails": obj.deployStageExecutionProgressDetails
                ? obj.deployStageExecutionProgressDetails.map(item => {
                    return model.DeployStageExecutionProgressDetails.getJsonObj(item);
                })
                : undefined
        });
        if (obj && "deployStageType" in obj && obj.deployStageType) {
            switch (obj.deployStageType) {
                case "COMPUTE_INSTANCE_GROUP_BLUE_GREEN_TRAFFIC_SHIFT":
                    return model.ComputeInstanceGroupBlueGreenTrafficShiftDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_DEPLOYMENT":
                    return model.ComputeInstanceGroupCanaryDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_BLUE_GREEN_DEPLOYMENT":
                    return model.OkeBlueGreenDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_ROLLING_DEPLOYMENT":
                    return model.ComputeInstanceGroupDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "SHELL":
                    return model.ShellDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_CANARY_DEPLOYMENT":
                    return model.OkeCanaryDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "LOAD_BALANCER_TRAFFIC_SHIFT":
                    return model.LoadBalancerTrafficShiftDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "WAIT":
                    return model.WaitDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_TRAFFIC_SHIFT":
                    return model.ComputeInstanceGroupCanaryTrafficShiftDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_CANARY_APPROVAL":
                    return model.OkeCanaryApprovalDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "MANUAL_APPROVAL":
                    return model.ManualApprovalDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_DEPLOYMENT":
                    return model.OkeDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "DEPLOY_FUNCTION":
                    return model.FunctionDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_BLUE_GREEN_TRAFFIC_SHIFT":
                    return model.OkeBlueGreenTrafficShiftDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_HELM_CHART_DEPLOYMENT":
                    return model.OkeHelmChartDeploymentStageExecutionProgress.getJsonObj(jsonObj, true);
                case "INVOKE_FUNCTION":
                    return model.InvokeFunctionDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "OKE_CANARY_TRAFFIC_SHIFT":
                    return model.OkeCanaryTrafficShiftDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_APPROVAL":
                    return model.ComputeInstanceGroupCanaryApprovalDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_BLUE_GREEN_DEPLOYMENT":
                    return model.ComputeInstanceGroupBlueGreenDeployStageExecutionProgress.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.deployStageType}`);
            }
        }
        return jsonObj;
    }
    DeployStageExecutionProgress.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "deployStagePredecessors": obj.deployStagePredecessors
                ? model.DeployStagePredecessorCollection.getDeserializedJsonObj(obj.deployStagePredecessors)
                : undefined,
            "deployStageExecutionProgressDetails": obj.deployStageExecutionProgressDetails
                ? obj.deployStageExecutionProgressDetails.map(item => {
                    return model.DeployStageExecutionProgressDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        if (obj && "deployStageType" in obj && obj.deployStageType) {
            switch (obj.deployStageType) {
                case "COMPUTE_INSTANCE_GROUP_BLUE_GREEN_TRAFFIC_SHIFT":
                    return model.ComputeInstanceGroupBlueGreenTrafficShiftDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_DEPLOYMENT":
                    return model.ComputeInstanceGroupCanaryDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_BLUE_GREEN_DEPLOYMENT":
                    return model.OkeBlueGreenDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_ROLLING_DEPLOYMENT":
                    return model.ComputeInstanceGroupDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "SHELL":
                    return model.ShellDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_CANARY_DEPLOYMENT":
                    return model.OkeCanaryDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "LOAD_BALANCER_TRAFFIC_SHIFT":
                    return model.LoadBalancerTrafficShiftDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "WAIT":
                    return model.WaitDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_TRAFFIC_SHIFT":
                    return model.ComputeInstanceGroupCanaryTrafficShiftDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_CANARY_APPROVAL":
                    return model.OkeCanaryApprovalDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "MANUAL_APPROVAL":
                    return model.ManualApprovalDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_DEPLOYMENT":
                    return model.OkeDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "DEPLOY_FUNCTION":
                    return model.FunctionDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_BLUE_GREEN_TRAFFIC_SHIFT":
                    return model.OkeBlueGreenTrafficShiftDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_HELM_CHART_DEPLOYMENT":
                    return model.OkeHelmChartDeploymentStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "INVOKE_FUNCTION":
                    return model.InvokeFunctionDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "OKE_CANARY_TRAFFIC_SHIFT":
                    return model.OkeCanaryTrafficShiftDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_CANARY_APPROVAL":
                    return model.ComputeInstanceGroupCanaryApprovalDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                case "COMPUTE_INSTANCE_GROUP_BLUE_GREEN_DEPLOYMENT":
                    return model.ComputeInstanceGroupBlueGreenDeployStageExecutionProgress.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.deployStageType}`);
            }
        }
        return jsonObj;
    }
    DeployStageExecutionProgress.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeployStageExecutionProgress = exports.DeployStageExecutionProgress || (exports.DeployStageExecutionProgress = {}));
