// oci-cloudguard/lib/model/index.js
"use strict";
/**
 * Cloud Guard and Security Zones API
 * Use the Cloud Guard and Security Zones API to automate processes that you would otherwise perform through the Cloud Guard Console or the Security Zones Console. For more information on these services, see the [Cloud Guard](/iaas/cloud-guard/home.htm) and [Security Zones](/iaas/security-zone/home.htm) documentation.

**Note:** For Cloud Guard, you can perform Create, Update, and Delete operations only from the reporting region of your Cloud Guard tenancy. You can perform Read operations from any region.

 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataMaskRuleSummary = exports.DataMaskRuleStatus = exports.DataMaskRuleCollection = exports.DataMaskRule = exports.DataMaskCategory = exports.CreateTargetResponderRecipeDetails = exports.CreateTargetDetectorRecipeDetails = exports.CreateTargetDetails = exports.CreateSecurityZoneDetails = exports.CreateSecurityRecipeDetails = exports.CreateSecurityPolicyDetails = exports.CreateResponderRecipeDetails = exports.CreateManagedListDetails = exports.CreateDetectorRuleDetails = exports.CreateDetectorRecipeDetectorRuleDetails = exports.CreateDetectorRecipeDetails = exports.CreateDataSourceDetails = exports.CreateDataMaskRuleDetails = exports.ContinuousQueryStartPolicy = exports.ConfigurationListItemType = exports.Configuration = exports.ConfigValue = exports.Confidence = exports.ConditionValueType = exports.ConditionTypeEnum = exports.ConditionOperatorName = exports.ConditionOperator = exports.ConditionMetadataTypeSummary = exports.ConditionMetadataTypeCollection = exports.ConditionMetadataType = exports.ConditionGroup = exports.ConditionFilterType = exports.Condition = exports.CloudGuardStatus = exports.ClassificationStatus = exports.ChangeSecurityZoneCompartmentDetails = exports.ChangeSecurityRecipeCompartmentDetails = exports.ChangeSecurityPolicyCompartmentDetails = exports.ChangeResponderRecipeCompartmentDetails = exports.ChangeManagedListCompartmentDetails = exports.ChangeDetectorRecipeCompartmentDetails = exports.ChangeDataSourceCompartmentDetails = exports.CandidateResponderRule = exports.AttachTargetResponderRecipeDetails = exports.AttachTargetDetectorRecipeDetails = exports.AddCompartmentDetails = exports.ActorType = exports.ActivityProblemAggregationCollection = exports.ActivityProblemAggregation = exports.ActionType = void 0;
exports.OperatorType = exports.OperatorSummary = exports.OperationType = exports.OperationStatus = exports.ManagedListTypeSummary = exports.ManagedListTypeCollection = exports.ManagedListType = exports.ManagedListSummary = exports.ManagedListCollection = exports.ManagedList = exports.LoggingQueryType = exports.LoggingQueryOperatorType = exports.LoggingQueryDetails = exports.LifecycleState = exports.ImpactedResourceSummary = exports.ImpactedResourceCollection = exports.GeographicalLocation = exports.FeedProviderType = exports.ExecuteResponderExecutionDetails = exports.EventStatus = exports.EntityType = exports.EntityDetails = exports.EntitiesMapping = exports.DetectorSummary = exports.DetectorRuleSummary = exports.DetectorRuleCollection = exports.DetectorRule = exports.DetectorRecipeSummary = exports.DetectorRecipeDetectorRuleSummary = exports.DetectorRecipeDetectorRuleCollection = exports.DetectorRecipeDetectorRule = exports.DetectorRecipeCollection = exports.DetectorRecipe = exports.DetectorEnum = exports.DetectorDetails = exports.DetectorConfiguration = exports.DetectorCollection = exports.Detector = exports.DataSourceSummaryDetails = exports.DataSourceSummary = exports.DataSourceStatus = exports.DataSourceMappingInfo = exports.DataSourceFeedProvider = exports.DataSourceEventSummary = exports.DataSourceEventInfoStatus = exports.DataSourceEventInfo = exports.DataSourceEventCollection = exports.DataSourceDetails = exports.DataSourceCollection = exports.DataSource = void 0;
exports.ResponderExecutionAggregation = exports.ResponderExecution = exports.ResponderDimension = exports.ResponderConfiguration = exports.ResponderActivityType = exports.ResponderActivitySummary = exports.ResponderActivityCollection = exports.ResourceTypeSummary = exports.ResourceTypeCollection = exports.ResourceRiskScoreDimension = exports.ResourceRiskScoreAggregationCollection = exports.ResourceRiskScoreAggregation = exports.ResourceProfileSummary = exports.ResourceProfileRiskScoreAggregationSummaryCollection = exports.ResourceProfileRiskScoreAggregationSummary = exports.ResourceProfileImpactedResourceSummary = exports.ResourceProfileImpactedResourceCollection = exports.ResourceProfileEndpointSummary = exports.ResourceProfileEndpointCollection = exports.ResourceProfileCollection = exports.ResourceProfile = exports.RequestSummarizedTrendResourceRiskScoresDetails = exports.RemoveCompartmentDetails = exports.RegionStatusDetail = exports.RecommendationType = exports.RecommendationSummaryCollection = exports.RecommendationSummary = exports.RecommendationLifecycleDetail = exports.QueryReplicationStatus = exports.ProblemTrendAggregationCollection = exports.ProblemTrendAggregation = exports.ProblemSummary = exports.ProblemLifecycleState = exports.ProblemLifecycleDetail = exports.ProblemHistorySummary = exports.ProblemHistoryCollection = exports.ProblemEntitySummary = exports.ProblemEntityCollection = exports.ProblemEndpointSummary = exports.ProblemEndpointCollection = exports.ProblemDimension = exports.ProblemCollection = exports.ProblemCategoryEnum = exports.ProblemAggregationCollection = exports.ProblemAggregation = exports.Problem = exports.PoliticalLocation = exports.PolicySummary = exports.PolicyCollection = exports.OwnerType = void 0;
exports.SkipBulkResponderExecutionDetails = exports.SightingType = exports.SightingSummary = exports.SightingImpactedResourceSummary = exports.SightingImpactedResourceCollection = exports.SightingEndpointSummary = exports.SightingEndpointCollection = exports.SightingCollection = exports.Sighting = exports.Severity = exports.ServiceTypeSummary = exports.SecurityZoneSummary = exports.SecurityZoneCollection = exports.SecurityZone = exports.SecurityScoreTrendAggregationCollection = exports.SecurityScoreTrendAggregation = exports.SecurityScoreAggregationCollection = exports.SecurityScoreAggregation = exports.SecurityRecipeSummary = exports.SecurityRecipeCollection = exports.SecurityRecipe = exports.SecurityRating = exports.SecurityPolicySummary = exports.SecurityPolicyCollection = exports.SecurityPolicy = exports.RuleSummary = exports.RiskScoreAggregationCollection = exports.RiskScoreAggregation = exports.RiskLevel = exports.ResponderType = exports.ResponderRuleSummary = exports.ResponderRuleExecutionDetails = exports.ResponderRuleDetails = exports.ResponderRuleCollection = exports.ResponderRule = exports.ResponderRecipeSummary = exports.ResponderRecipeResponderRuleSummary = exports.ResponderRecipeResponderRuleCollection = exports.ResponderRecipeResponderRule = exports.ResponderRecipeCollection = exports.ResponderRecipe = exports.ResponderModeTypes = exports.ResponderExecutionTrendAggregationCollection = exports.ResponderExecutionTrendAggregation = exports.ResponderExecutionSummary = exports.ResponderExecutionStatus = exports.ResponderExecutionStates = exports.ResponderExecutionModes = exports.ResponderExecutionCollection = exports.ResponderExecutionAggregationCollection = void 0;
exports.UpdateTargetResponderRecipe = exports.UpdateTargetRecipeResponderRuleDetails = exports.UpdateTargetRecipeDetectorRuleDetails = exports.UpdateTargetDetectorRuleDetails = exports.UpdateTargetDetectorRecipeDetectorRuleDetails = exports.UpdateTargetDetectorRecipeDetails = exports.UpdateTargetDetectorRecipe = exports.UpdateTargetDetails = exports.UpdateSecurityZoneDetails = exports.UpdateSecurityRecipeDetails = exports.UpdateSecurityPolicyDetails = exports.UpdateResponderRuleDetails = exports.UpdateResponderRecipeResponderRuleDetails = exports.UpdateResponderRecipeResponderRule = exports.UpdateResponderRecipeDetails = exports.UpdateProblemStatusDetails = exports.UpdateManagedListDetails = exports.UpdateDetectorRuleDetails = exports.UpdateDetectorRecipeDetectorRuleDetails = exports.UpdateDetectorRecipeDetectorRule = exports.UpdateDetectorRecipeDetails = exports.UpdateDataSourceDetails = exports.UpdateDataMaskRuleDetails = exports.UpdateConfigurationDetails = exports.UpdateBulkProblemStatusDetails = exports.TriggerResponderDetails = exports.TechniqueSummary = exports.TechniqueCollection = exports.TargetSummary = exports.TargetSelected = exports.TargetResponderRecipeSummary = exports.TargetResponderRecipeResponderRuleSummary = exports.TargetResponderRecipeResponderRuleCollection = exports.TargetResponderRecipeResponderRule = exports.TargetResponderRecipeCollection = exports.TargetResponderRecipe = exports.TargetResourceType = exports.TargetDetectorRecipeSummary = exports.TargetDetectorRecipeDetectorRuleSummary = exports.TargetDetectorRecipeDetectorRuleCollection = exports.TargetDetectorRecipeDetectorRule = exports.TargetDetectorRecipeCollection = exports.TargetDetectorRecipe = exports.TargetDetectorDetails = exports.TargetDetails = exports.TargetCollection = exports.Target = exports.TacticSummary = exports.TacticCollection = exports.SortOrders = void 0;
exports.TargetResourceTypesSelected = exports.TargetIdsSelected = exports.SimpleCondition = exports.SecurityZoneTargetDetails = exports.NoDelayStartPolicy = exports.LoggingQueryDataSourceSummaryDetails = exports.LoggingQueryDataSourceDetails = exports.LoggingEventInfo = exports.InsightTypeLoggingQueryDetails = exports.CompositeCondition = exports.AllTargetsSelected = exports.AbsoluteTimeStartPolicy = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateTargetResponderRuleDetails = exports.UpdateTargetResponderRecipeResponderRuleDetails = exports.UpdateTargetResponderRecipeDetails = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ActivityProblemAggregation = __importStar(require("./activity-problem-aggregation"));
exports.ActivityProblemAggregation = ActivityProblemAggregation.ActivityProblemAggregation;
const ActivityProblemAggregationCollection = __importStar(require("./activity-problem-aggregation-collection"));
exports.ActivityProblemAggregationCollection = ActivityProblemAggregationCollection.ActivityProblemAggregationCollection;
const ActorType = __importStar(require("./actor-type"));
exports.ActorType = ActorType.ActorType;
const AddCompartmentDetails = __importStar(require("./add-compartment-details"));
exports.AddCompartmentDetails = AddCompartmentDetails.AddCompartmentDetails;
const AttachTargetDetectorRecipeDetails = __importStar(require("./attach-target-detector-recipe-details"));
exports.AttachTargetDetectorRecipeDetails = AttachTargetDetectorRecipeDetails.AttachTargetDetectorRecipeDetails;
const AttachTargetResponderRecipeDetails = __importStar(require("./attach-target-responder-recipe-details"));
exports.AttachTargetResponderRecipeDetails = AttachTargetResponderRecipeDetails.AttachTargetResponderRecipeDetails;
const CandidateResponderRule = __importStar(require("./candidate-responder-rule"));
exports.CandidateResponderRule = CandidateResponderRule.CandidateResponderRule;
const ChangeDataSourceCompartmentDetails = __importStar(require("./change-data-source-compartment-details"));
exports.ChangeDataSourceCompartmentDetails = ChangeDataSourceCompartmentDetails.ChangeDataSourceCompartmentDetails;
const ChangeDetectorRecipeCompartmentDetails = __importStar(require("./change-detector-recipe-compartment-details"));
exports.ChangeDetectorRecipeCompartmentDetails = ChangeDetectorRecipeCompartmentDetails.ChangeDetectorRecipeCompartmentDetails;
const ChangeManagedListCompartmentDetails = __importStar(require("./change-managed-list-compartment-details"));
exports.ChangeManagedListCompartmentDetails = ChangeManagedListCompartmentDetails.ChangeManagedListCompartmentDetails;
const ChangeResponderRecipeCompartmentDetails = __importStar(require("./change-responder-recipe-compartment-details"));
exports.ChangeResponderRecipeCompartmentDetails = ChangeResponderRecipeCompartmentDetails.ChangeResponderRecipeCompartmentDetails;
const ChangeSecurityPolicyCompartmentDetails = __importStar(require("./change-security-policy-compartment-details"));
exports.ChangeSecurityPolicyCompartmentDetails = ChangeSecurityPolicyCompartmentDetails.ChangeSecurityPolicyCompartmentDetails;
const ChangeSecurityRecipeCompartmentDetails = __importStar(require("./change-security-recipe-compartment-details"));
exports.ChangeSecurityRecipeCompartmentDetails = ChangeSecurityRecipeCompartmentDetails.ChangeSecurityRecipeCompartmentDetails;
const ChangeSecurityZoneCompartmentDetails = __importStar(require("./change-security-zone-compartment-details"));
exports.ChangeSecurityZoneCompartmentDetails = ChangeSecurityZoneCompartmentDetails.ChangeSecurityZoneCompartmentDetails;
const ClassificationStatus = __importStar(require("./classification-status"));
exports.ClassificationStatus = ClassificationStatus.ClassificationStatus;
const CloudGuardStatus = __importStar(require("./cloud-guard-status"));
exports.CloudGuardStatus = CloudGuardStatus.CloudGuardStatus;
const Condition = __importStar(require("./condition"));
exports.Condition = Condition.Condition;
const ConditionFilterType = __importStar(require("./condition-filter-type"));
exports.ConditionFilterType = ConditionFilterType.ConditionFilterType;
const ConditionGroup = __importStar(require("./condition-group"));
exports.ConditionGroup = ConditionGroup.ConditionGroup;
const ConditionMetadataType = __importStar(require("./condition-metadata-type"));
exports.ConditionMetadataType = ConditionMetadataType.ConditionMetadataType;
const ConditionMetadataTypeCollection = __importStar(require("./condition-metadata-type-collection"));
exports.ConditionMetadataTypeCollection = ConditionMetadataTypeCollection.ConditionMetadataTypeCollection;
const ConditionMetadataTypeSummary = __importStar(require("./condition-metadata-type-summary"));
exports.ConditionMetadataTypeSummary = ConditionMetadataTypeSummary.ConditionMetadataTypeSummary;
const ConditionOperator = __importStar(require("./condition-operator"));
exports.ConditionOperator = ConditionOperator.ConditionOperator;
const ConditionOperatorName = __importStar(require("./condition-operator-name"));
exports.ConditionOperatorName = ConditionOperatorName.ConditionOperatorName;
const ConditionTypeEnum = __importStar(require("./condition-type-enum"));
exports.ConditionTypeEnum = ConditionTypeEnum.ConditionTypeEnum;
const ConditionValueType = __importStar(require("./condition-value-type"));
exports.ConditionValueType = ConditionValueType.ConditionValueType;
const Confidence = __importStar(require("./confidence"));
exports.Confidence = Confidence.Confidence;
const ConfigValue = __importStar(require("./config-value"));
exports.ConfigValue = ConfigValue.ConfigValue;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const ConfigurationListItemType = __importStar(require("./configuration-list-item-type"));
exports.ConfigurationListItemType = ConfigurationListItemType.ConfigurationListItemType;
const ContinuousQueryStartPolicy = __importStar(require("./continuous-query-start-policy"));
exports.ContinuousQueryStartPolicy = ContinuousQueryStartPolicy.ContinuousQueryStartPolicy;
const CreateDataMaskRuleDetails = __importStar(require("./create-data-mask-rule-details"));
exports.CreateDataMaskRuleDetails = CreateDataMaskRuleDetails.CreateDataMaskRuleDetails;
const CreateDataSourceDetails = __importStar(require("./create-data-source-details"));
exports.CreateDataSourceDetails = CreateDataSourceDetails.CreateDataSourceDetails;
const CreateDetectorRecipeDetails = __importStar(require("./create-detector-recipe-details"));
exports.CreateDetectorRecipeDetails = CreateDetectorRecipeDetails.CreateDetectorRecipeDetails;
const CreateDetectorRecipeDetectorRuleDetails = __importStar(require("./create-detector-recipe-detector-rule-details"));
exports.CreateDetectorRecipeDetectorRuleDetails = CreateDetectorRecipeDetectorRuleDetails.CreateDetectorRecipeDetectorRuleDetails;
const CreateDetectorRuleDetails = __importStar(require("./create-detector-rule-details"));
exports.CreateDetectorRuleDetails = CreateDetectorRuleDetails.CreateDetectorRuleDetails;
const CreateManagedListDetails = __importStar(require("./create-managed-list-details"));
exports.CreateManagedListDetails = CreateManagedListDetails.CreateManagedListDetails;
const CreateResponderRecipeDetails = __importStar(require("./create-responder-recipe-details"));
exports.CreateResponderRecipeDetails = CreateResponderRecipeDetails.CreateResponderRecipeDetails;
const CreateSecurityPolicyDetails = __importStar(require("./create-security-policy-details"));
exports.CreateSecurityPolicyDetails = CreateSecurityPolicyDetails.CreateSecurityPolicyDetails;
const CreateSecurityRecipeDetails = __importStar(require("./create-security-recipe-details"));
exports.CreateSecurityRecipeDetails = CreateSecurityRecipeDetails.CreateSecurityRecipeDetails;
const CreateSecurityZoneDetails = __importStar(require("./create-security-zone-details"));
exports.CreateSecurityZoneDetails = CreateSecurityZoneDetails.CreateSecurityZoneDetails;
const CreateTargetDetails = __importStar(require("./create-target-details"));
exports.CreateTargetDetails = CreateTargetDetails.CreateTargetDetails;
const CreateTargetDetectorRecipeDetails = __importStar(require("./create-target-detector-recipe-details"));
exports.CreateTargetDetectorRecipeDetails = CreateTargetDetectorRecipeDetails.CreateTargetDetectorRecipeDetails;
const CreateTargetResponderRecipeDetails = __importStar(require("./create-target-responder-recipe-details"));
exports.CreateTargetResponderRecipeDetails = CreateTargetResponderRecipeDetails.CreateTargetResponderRecipeDetails;
const DataMaskCategory = __importStar(require("./data-mask-category"));
exports.DataMaskCategory = DataMaskCategory.DataMaskCategory;
const DataMaskRule = __importStar(require("./data-mask-rule"));
exports.DataMaskRule = DataMaskRule.DataMaskRule;
const DataMaskRuleCollection = __importStar(require("./data-mask-rule-collection"));
exports.DataMaskRuleCollection = DataMaskRuleCollection.DataMaskRuleCollection;
const DataMaskRuleStatus = __importStar(require("./data-mask-rule-status"));
exports.DataMaskRuleStatus = DataMaskRuleStatus.DataMaskRuleStatus;
const DataMaskRuleSummary = __importStar(require("./data-mask-rule-summary"));
exports.DataMaskRuleSummary = DataMaskRuleSummary.DataMaskRuleSummary;
const DataSource = __importStar(require("./data-source"));
exports.DataSource = DataSource.DataSource;
const DataSourceCollection = __importStar(require("./data-source-collection"));
exports.DataSourceCollection = DataSourceCollection.DataSourceCollection;
const DataSourceDetails = __importStar(require("./data-source-details"));
exports.DataSourceDetails = DataSourceDetails.DataSourceDetails;
const DataSourceEventCollection = __importStar(require("./data-source-event-collection"));
exports.DataSourceEventCollection = DataSourceEventCollection.DataSourceEventCollection;
const DataSourceEventInfo = __importStar(require("./data-source-event-info"));
exports.DataSourceEventInfo = DataSourceEventInfo.DataSourceEventInfo;
const DataSourceEventInfoStatus = __importStar(require("./data-source-event-info-status"));
exports.DataSourceEventInfoStatus = DataSourceEventInfoStatus.DataSourceEventInfoStatus;
const DataSourceEventSummary = __importStar(require("./data-source-event-summary"));
exports.DataSourceEventSummary = DataSourceEventSummary.DataSourceEventSummary;
const DataSourceFeedProvider = __importStar(require("./data-source-feed-provider"));
exports.DataSourceFeedProvider = DataSourceFeedProvider.DataSourceFeedProvider;
const DataSourceMappingInfo = __importStar(require("./data-source-mapping-info"));
exports.DataSourceMappingInfo = DataSourceMappingInfo.DataSourceMappingInfo;
const DataSourceStatus = __importStar(require("./data-source-status"));
exports.DataSourceStatus = DataSourceStatus.DataSourceStatus;
const DataSourceSummary = __importStar(require("./data-source-summary"));
exports.DataSourceSummary = DataSourceSummary.DataSourceSummary;
const DataSourceSummaryDetails = __importStar(require("./data-source-summary-details"));
exports.DataSourceSummaryDetails = DataSourceSummaryDetails.DataSourceSummaryDetails;
const Detector = __importStar(require("./detector"));
exports.Detector = Detector.Detector;
const DetectorCollection = __importStar(require("./detector-collection"));
exports.DetectorCollection = DetectorCollection.DetectorCollection;
const DetectorConfiguration = __importStar(require("./detector-configuration"));
exports.DetectorConfiguration = DetectorConfiguration.DetectorConfiguration;
const DetectorDetails = __importStar(require("./detector-details"));
exports.DetectorDetails = DetectorDetails.DetectorDetails;
const DetectorEnum = __importStar(require("./detector-enum"));
exports.DetectorEnum = DetectorEnum.DetectorEnum;
const DetectorRecipe = __importStar(require("./detector-recipe"));
exports.DetectorRecipe = DetectorRecipe.DetectorRecipe;
const DetectorRecipeCollection = __importStar(require("./detector-recipe-collection"));
exports.DetectorRecipeCollection = DetectorRecipeCollection.DetectorRecipeCollection;
const DetectorRecipeDetectorRule = __importStar(require("./detector-recipe-detector-rule"));
exports.DetectorRecipeDetectorRule = DetectorRecipeDetectorRule.DetectorRecipeDetectorRule;
const DetectorRecipeDetectorRuleCollection = __importStar(require("./detector-recipe-detector-rule-collection"));
exports.DetectorRecipeDetectorRuleCollection = DetectorRecipeDetectorRuleCollection.DetectorRecipeDetectorRuleCollection;
const DetectorRecipeDetectorRuleSummary = __importStar(require("./detector-recipe-detector-rule-summary"));
exports.DetectorRecipeDetectorRuleSummary = DetectorRecipeDetectorRuleSummary.DetectorRecipeDetectorRuleSummary;
const DetectorRecipeSummary = __importStar(require("./detector-recipe-summary"));
exports.DetectorRecipeSummary = DetectorRecipeSummary.DetectorRecipeSummary;
const DetectorRule = __importStar(require("./detector-rule"));
exports.DetectorRule = DetectorRule.DetectorRule;
const DetectorRuleCollection = __importStar(require("./detector-rule-collection"));
exports.DetectorRuleCollection = DetectorRuleCollection.DetectorRuleCollection;
const DetectorRuleSummary = __importStar(require("./detector-rule-summary"));
exports.DetectorRuleSummary = DetectorRuleSummary.DetectorRuleSummary;
const DetectorSummary = __importStar(require("./detector-summary"));
exports.DetectorSummary = DetectorSummary.DetectorSummary;
const EntitiesMapping = __importStar(require("./entities-mapping"));
exports.EntitiesMapping = EntitiesMapping.EntitiesMapping;
const EntityDetails = __importStar(require("./entity-details"));
exports.EntityDetails = EntityDetails.EntityDetails;
const EntityType = __importStar(require("./entity-type"));
exports.EntityType = EntityType.EntityType;
const EventStatus = __importStar(require("./event-status"));
exports.EventStatus = EventStatus.EventStatus;
const ExecuteResponderExecutionDetails = __importStar(require("./execute-responder-execution-details"));
exports.ExecuteResponderExecutionDetails = ExecuteResponderExecutionDetails.ExecuteResponderExecutionDetails;
const FeedProviderType = __importStar(require("./feed-provider-type"));
exports.FeedProviderType = FeedProviderType.FeedProviderType;
const GeographicalLocation = __importStar(require("./geographical-location"));
exports.GeographicalLocation = GeographicalLocation.GeographicalLocation;
const ImpactedResourceCollection = __importStar(require("./impacted-resource-collection"));
exports.ImpactedResourceCollection = ImpactedResourceCollection.ImpactedResourceCollection;
const ImpactedResourceSummary = __importStar(require("./impacted-resource-summary"));
exports.ImpactedResourceSummary = ImpactedResourceSummary.ImpactedResourceSummary;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const LoggingQueryDetails = __importStar(require("./logging-query-details"));
exports.LoggingQueryDetails = LoggingQueryDetails.LoggingQueryDetails;
const LoggingQueryOperatorType = __importStar(require("./logging-query-operator-type"));
exports.LoggingQueryOperatorType = LoggingQueryOperatorType.LoggingQueryOperatorType;
const LoggingQueryType = __importStar(require("./logging-query-type"));
exports.LoggingQueryType = LoggingQueryType.LoggingQueryType;
const ManagedList = __importStar(require("./managed-list"));
exports.ManagedList = ManagedList.ManagedList;
const ManagedListCollection = __importStar(require("./managed-list-collection"));
exports.ManagedListCollection = ManagedListCollection.ManagedListCollection;
const ManagedListSummary = __importStar(require("./managed-list-summary"));
exports.ManagedListSummary = ManagedListSummary.ManagedListSummary;
const ManagedListType = __importStar(require("./managed-list-type"));
exports.ManagedListType = ManagedListType.ManagedListType;
const ManagedListTypeCollection = __importStar(require("./managed-list-type-collection"));
exports.ManagedListTypeCollection = ManagedListTypeCollection.ManagedListTypeCollection;
const ManagedListTypeSummary = __importStar(require("./managed-list-type-summary"));
exports.ManagedListTypeSummary = ManagedListTypeSummary.ManagedListTypeSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OperatorSummary = __importStar(require("./operator-summary"));
exports.OperatorSummary = OperatorSummary.OperatorSummary;
const OperatorType = __importStar(require("./operator-type"));
exports.OperatorType = OperatorType.OperatorType;
const OwnerType = __importStar(require("./owner-type"));
exports.OwnerType = OwnerType.OwnerType;
const PolicyCollection = __importStar(require("./policy-collection"));
exports.PolicyCollection = PolicyCollection.PolicyCollection;
const PolicySummary = __importStar(require("./policy-summary"));
exports.PolicySummary = PolicySummary.PolicySummary;
const PoliticalLocation = __importStar(require("./political-location"));
exports.PoliticalLocation = PoliticalLocation.PoliticalLocation;
const Problem = __importStar(require("./problem"));
exports.Problem = Problem.Problem;
const ProblemAggregation = __importStar(require("./problem-aggregation"));
exports.ProblemAggregation = ProblemAggregation.ProblemAggregation;
const ProblemAggregationCollection = __importStar(require("./problem-aggregation-collection"));
exports.ProblemAggregationCollection = ProblemAggregationCollection.ProblemAggregationCollection;
const ProblemCategoryEnum = __importStar(require("./problem-category-enum"));
exports.ProblemCategoryEnum = ProblemCategoryEnum.ProblemCategoryEnum;
const ProblemCollection = __importStar(require("./problem-collection"));
exports.ProblemCollection = ProblemCollection.ProblemCollection;
const ProblemDimension = __importStar(require("./problem-dimension"));
exports.ProblemDimension = ProblemDimension.ProblemDimension;
const ProblemEndpointCollection = __importStar(require("./problem-endpoint-collection"));
exports.ProblemEndpointCollection = ProblemEndpointCollection.ProblemEndpointCollection;
const ProblemEndpointSummary = __importStar(require("./problem-endpoint-summary"));
exports.ProblemEndpointSummary = ProblemEndpointSummary.ProblemEndpointSummary;
const ProblemEntityCollection = __importStar(require("./problem-entity-collection"));
exports.ProblemEntityCollection = ProblemEntityCollection.ProblemEntityCollection;
const ProblemEntitySummary = __importStar(require("./problem-entity-summary"));
exports.ProblemEntitySummary = ProblemEntitySummary.ProblemEntitySummary;
const ProblemHistoryCollection = __importStar(require("./problem-history-collection"));
exports.ProblemHistoryCollection = ProblemHistoryCollection.ProblemHistoryCollection;
const ProblemHistorySummary = __importStar(require("./problem-history-summary"));
exports.ProblemHistorySummary = ProblemHistorySummary.ProblemHistorySummary;
const ProblemLifecycleDetail = __importStar(require("./problem-lifecycle-detail"));
exports.ProblemLifecycleDetail = ProblemLifecycleDetail.ProblemLifecycleDetail;
const ProblemLifecycleState = __importStar(require("./problem-lifecycle-state"));
exports.ProblemLifecycleState = ProblemLifecycleState.ProblemLifecycleState;
const ProblemSummary = __importStar(require("./problem-summary"));
exports.ProblemSummary = ProblemSummary.ProblemSummary;
const ProblemTrendAggregation = __importStar(require("./problem-trend-aggregation"));
exports.ProblemTrendAggregation = ProblemTrendAggregation.ProblemTrendAggregation;
const ProblemTrendAggregationCollection = __importStar(require("./problem-trend-aggregation-collection"));
exports.ProblemTrendAggregationCollection = ProblemTrendAggregationCollection.ProblemTrendAggregationCollection;
const QueryReplicationStatus = __importStar(require("./query-replication-status"));
exports.QueryReplicationStatus = QueryReplicationStatus.QueryReplicationStatus;
const RecommendationLifecycleDetail = __importStar(require("./recommendation-lifecycle-detail"));
exports.RecommendationLifecycleDetail = RecommendationLifecycleDetail.RecommendationLifecycleDetail;
const RecommendationSummary = __importStar(require("./recommendation-summary"));
exports.RecommendationSummary = RecommendationSummary.RecommendationSummary;
const RecommendationSummaryCollection = __importStar(require("./recommendation-summary-collection"));
exports.RecommendationSummaryCollection = RecommendationSummaryCollection.RecommendationSummaryCollection;
const RecommendationType = __importStar(require("./recommendation-type"));
exports.RecommendationType = RecommendationType.RecommendationType;
const RegionStatusDetail = __importStar(require("./region-status-detail"));
exports.RegionStatusDetail = RegionStatusDetail.RegionStatusDetail;
const RemoveCompartmentDetails = __importStar(require("./remove-compartment-details"));
exports.RemoveCompartmentDetails = RemoveCompartmentDetails.RemoveCompartmentDetails;
const RequestSummarizedTrendResourceRiskScoresDetails = __importStar(require("./request-summarized-trend-resource-risk-scores-details"));
exports.RequestSummarizedTrendResourceRiskScoresDetails = RequestSummarizedTrendResourceRiskScoresDetails.RequestSummarizedTrendResourceRiskScoresDetails;
const ResourceProfile = __importStar(require("./resource-profile"));
exports.ResourceProfile = ResourceProfile.ResourceProfile;
const ResourceProfileCollection = __importStar(require("./resource-profile-collection"));
exports.ResourceProfileCollection = ResourceProfileCollection.ResourceProfileCollection;
const ResourceProfileEndpointCollection = __importStar(require("./resource-profile-endpoint-collection"));
exports.ResourceProfileEndpointCollection = ResourceProfileEndpointCollection.ResourceProfileEndpointCollection;
const ResourceProfileEndpointSummary = __importStar(require("./resource-profile-endpoint-summary"));
exports.ResourceProfileEndpointSummary = ResourceProfileEndpointSummary.ResourceProfileEndpointSummary;
const ResourceProfileImpactedResourceCollection = __importStar(require("./resource-profile-impacted-resource-collection"));
exports.ResourceProfileImpactedResourceCollection = ResourceProfileImpactedResourceCollection.ResourceProfileImpactedResourceCollection;
const ResourceProfileImpactedResourceSummary = __importStar(require("./resource-profile-impacted-resource-summary"));
exports.ResourceProfileImpactedResourceSummary = ResourceProfileImpactedResourceSummary.ResourceProfileImpactedResourceSummary;
const ResourceProfileRiskScoreAggregationSummary = __importStar(require("./resource-profile-risk-score-aggregation-summary"));
exports.ResourceProfileRiskScoreAggregationSummary = ResourceProfileRiskScoreAggregationSummary.ResourceProfileRiskScoreAggregationSummary;
const ResourceProfileRiskScoreAggregationSummaryCollection = __importStar(require("./resource-profile-risk-score-aggregation-summary-collection"));
exports.ResourceProfileRiskScoreAggregationSummaryCollection = ResourceProfileRiskScoreAggregationSummaryCollection.ResourceProfileRiskScoreAggregationSummaryCollection;
const ResourceProfileSummary = __importStar(require("./resource-profile-summary"));
exports.ResourceProfileSummary = ResourceProfileSummary.ResourceProfileSummary;
const ResourceRiskScoreAggregation = __importStar(require("./resource-risk-score-aggregation"));
exports.ResourceRiskScoreAggregation = ResourceRiskScoreAggregation.ResourceRiskScoreAggregation;
const ResourceRiskScoreAggregationCollection = __importStar(require("./resource-risk-score-aggregation-collection"));
exports.ResourceRiskScoreAggregationCollection = ResourceRiskScoreAggregationCollection.ResourceRiskScoreAggregationCollection;
const ResourceRiskScoreDimension = __importStar(require("./resource-risk-score-dimension"));
exports.ResourceRiskScoreDimension = ResourceRiskScoreDimension.ResourceRiskScoreDimension;
const ResourceTypeCollection = __importStar(require("./resource-type-collection"));
exports.ResourceTypeCollection = ResourceTypeCollection.ResourceTypeCollection;
const ResourceTypeSummary = __importStar(require("./resource-type-summary"));
exports.ResourceTypeSummary = ResourceTypeSummary.ResourceTypeSummary;
const ResponderActivityCollection = __importStar(require("./responder-activity-collection"));
exports.ResponderActivityCollection = ResponderActivityCollection.ResponderActivityCollection;
const ResponderActivitySummary = __importStar(require("./responder-activity-summary"));
exports.ResponderActivitySummary = ResponderActivitySummary.ResponderActivitySummary;
const ResponderActivityType = __importStar(require("./responder-activity-type"));
exports.ResponderActivityType = ResponderActivityType.ResponderActivityType;
const ResponderConfiguration = __importStar(require("./responder-configuration"));
exports.ResponderConfiguration = ResponderConfiguration.ResponderConfiguration;
const ResponderDimension = __importStar(require("./responder-dimension"));
exports.ResponderDimension = ResponderDimension.ResponderDimension;
const ResponderExecution = __importStar(require("./responder-execution"));
exports.ResponderExecution = ResponderExecution.ResponderExecution;
const ResponderExecutionAggregation = __importStar(require("./responder-execution-aggregation"));
exports.ResponderExecutionAggregation = ResponderExecutionAggregation.ResponderExecutionAggregation;
const ResponderExecutionAggregationCollection = __importStar(require("./responder-execution-aggregation-collection"));
exports.ResponderExecutionAggregationCollection = ResponderExecutionAggregationCollection.ResponderExecutionAggregationCollection;
const ResponderExecutionCollection = __importStar(require("./responder-execution-collection"));
exports.ResponderExecutionCollection = ResponderExecutionCollection.ResponderExecutionCollection;
const ResponderExecutionModes = __importStar(require("./responder-execution-modes"));
exports.ResponderExecutionModes = ResponderExecutionModes.ResponderExecutionModes;
const ResponderExecutionStates = __importStar(require("./responder-execution-states"));
exports.ResponderExecutionStates = ResponderExecutionStates.ResponderExecutionStates;
const ResponderExecutionStatus = __importStar(require("./responder-execution-status"));
exports.ResponderExecutionStatus = ResponderExecutionStatus.ResponderExecutionStatus;
const ResponderExecutionSummary = __importStar(require("./responder-execution-summary"));
exports.ResponderExecutionSummary = ResponderExecutionSummary.ResponderExecutionSummary;
const ResponderExecutionTrendAggregation = __importStar(require("./responder-execution-trend-aggregation"));
exports.ResponderExecutionTrendAggregation = ResponderExecutionTrendAggregation.ResponderExecutionTrendAggregation;
const ResponderExecutionTrendAggregationCollection = __importStar(require("./responder-execution-trend-aggregation-collection"));
exports.ResponderExecutionTrendAggregationCollection = ResponderExecutionTrendAggregationCollection.ResponderExecutionTrendAggregationCollection;
const ResponderModeTypes = __importStar(require("./responder-mode-types"));
exports.ResponderModeTypes = ResponderModeTypes.ResponderModeTypes;
const ResponderRecipe = __importStar(require("./responder-recipe"));
exports.ResponderRecipe = ResponderRecipe.ResponderRecipe;
const ResponderRecipeCollection = __importStar(require("./responder-recipe-collection"));
exports.ResponderRecipeCollection = ResponderRecipeCollection.ResponderRecipeCollection;
const ResponderRecipeResponderRule = __importStar(require("./responder-recipe-responder-rule"));
exports.ResponderRecipeResponderRule = ResponderRecipeResponderRule.ResponderRecipeResponderRule;
const ResponderRecipeResponderRuleCollection = __importStar(require("./responder-recipe-responder-rule-collection"));
exports.ResponderRecipeResponderRuleCollection = ResponderRecipeResponderRuleCollection.ResponderRecipeResponderRuleCollection;
const ResponderRecipeResponderRuleSummary = __importStar(require("./responder-recipe-responder-rule-summary"));
exports.ResponderRecipeResponderRuleSummary = ResponderRecipeResponderRuleSummary.ResponderRecipeResponderRuleSummary;
const ResponderRecipeSummary = __importStar(require("./responder-recipe-summary"));
exports.ResponderRecipeSummary = ResponderRecipeSummary.ResponderRecipeSummary;
const ResponderRule = __importStar(require("./responder-rule"));
exports.ResponderRule = ResponderRule.ResponderRule;
const ResponderRuleCollection = __importStar(require("./responder-rule-collection"));
exports.ResponderRuleCollection = ResponderRuleCollection.ResponderRuleCollection;
const ResponderRuleDetails = __importStar(require("./responder-rule-details"));
exports.ResponderRuleDetails = ResponderRuleDetails.ResponderRuleDetails;
const ResponderRuleExecutionDetails = __importStar(require("./responder-rule-execution-details"));
exports.ResponderRuleExecutionDetails = ResponderRuleExecutionDetails.ResponderRuleExecutionDetails;
const ResponderRuleSummary = __importStar(require("./responder-rule-summary"));
exports.ResponderRuleSummary = ResponderRuleSummary.ResponderRuleSummary;
const ResponderType = __importStar(require("./responder-type"));
exports.ResponderType = ResponderType.ResponderType;
const RiskLevel = __importStar(require("./risk-level"));
exports.RiskLevel = RiskLevel.RiskLevel;
const RiskScoreAggregation = __importStar(require("./risk-score-aggregation"));
exports.RiskScoreAggregation = RiskScoreAggregation.RiskScoreAggregation;
const RiskScoreAggregationCollection = __importStar(require("./risk-score-aggregation-collection"));
exports.RiskScoreAggregationCollection = RiskScoreAggregationCollection.RiskScoreAggregationCollection;
const RuleSummary = __importStar(require("./rule-summary"));
exports.RuleSummary = RuleSummary.RuleSummary;
const SecurityPolicy = __importStar(require("./security-policy"));
exports.SecurityPolicy = SecurityPolicy.SecurityPolicy;
const SecurityPolicyCollection = __importStar(require("./security-policy-collection"));
exports.SecurityPolicyCollection = SecurityPolicyCollection.SecurityPolicyCollection;
const SecurityPolicySummary = __importStar(require("./security-policy-summary"));
exports.SecurityPolicySummary = SecurityPolicySummary.SecurityPolicySummary;
const SecurityRating = __importStar(require("./security-rating"));
exports.SecurityRating = SecurityRating.SecurityRating;
const SecurityRecipe = __importStar(require("./security-recipe"));
exports.SecurityRecipe = SecurityRecipe.SecurityRecipe;
const SecurityRecipeCollection = __importStar(require("./security-recipe-collection"));
exports.SecurityRecipeCollection = SecurityRecipeCollection.SecurityRecipeCollection;
const SecurityRecipeSummary = __importStar(require("./security-recipe-summary"));
exports.SecurityRecipeSummary = SecurityRecipeSummary.SecurityRecipeSummary;
const SecurityScoreAggregation = __importStar(require("./security-score-aggregation"));
exports.SecurityScoreAggregation = SecurityScoreAggregation.SecurityScoreAggregation;
const SecurityScoreAggregationCollection = __importStar(require("./security-score-aggregation-collection"));
exports.SecurityScoreAggregationCollection = SecurityScoreAggregationCollection.SecurityScoreAggregationCollection;
const SecurityScoreTrendAggregation = __importStar(require("./security-score-trend-aggregation"));
exports.SecurityScoreTrendAggregation = SecurityScoreTrendAggregation.SecurityScoreTrendAggregation;
const SecurityScoreTrendAggregationCollection = __importStar(require("./security-score-trend-aggregation-collection"));
exports.SecurityScoreTrendAggregationCollection = SecurityScoreTrendAggregationCollection.SecurityScoreTrendAggregationCollection;
const SecurityZone = __importStar(require("./security-zone"));
exports.SecurityZone = SecurityZone.SecurityZone;
const SecurityZoneCollection = __importStar(require("./security-zone-collection"));
exports.SecurityZoneCollection = SecurityZoneCollection.SecurityZoneCollection;
const SecurityZoneSummary = __importStar(require("./security-zone-summary"));
exports.SecurityZoneSummary = SecurityZoneSummary.SecurityZoneSummary;
const ServiceTypeSummary = __importStar(require("./service-type-summary"));
exports.ServiceTypeSummary = ServiceTypeSummary.ServiceTypeSummary;
const Severity = __importStar(require("./severity"));
exports.Severity = Severity.Severity;
const Sighting = __importStar(require("./sighting"));
exports.Sighting = Sighting.Sighting;
const SightingCollection = __importStar(require("./sighting-collection"));
exports.SightingCollection = SightingCollection.SightingCollection;
const SightingEndpointCollection = __importStar(require("./sighting-endpoint-collection"));
exports.SightingEndpointCollection = SightingEndpointCollection.SightingEndpointCollection;
const SightingEndpointSummary = __importStar(require("./sighting-endpoint-summary"));
exports.SightingEndpointSummary = SightingEndpointSummary.SightingEndpointSummary;
const SightingImpactedResourceCollection = __importStar(require("./sighting-impacted-resource-collection"));
exports.SightingImpactedResourceCollection = SightingImpactedResourceCollection.SightingImpactedResourceCollection;
const SightingImpactedResourceSummary = __importStar(require("./sighting-impacted-resource-summary"));
exports.SightingImpactedResourceSummary = SightingImpactedResourceSummary.SightingImpactedResourceSummary;
const SightingSummary = __importStar(require("./sighting-summary"));
exports.SightingSummary = SightingSummary.SightingSummary;
const SightingType = __importStar(require("./sighting-type"));
exports.SightingType = SightingType.SightingType;
const SkipBulkResponderExecutionDetails = __importStar(require("./skip-bulk-responder-execution-details"));
exports.SkipBulkResponderExecutionDetails = SkipBulkResponderExecutionDetails.SkipBulkResponderExecutionDetails;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const TacticCollection = __importStar(require("./tactic-collection"));
exports.TacticCollection = TacticCollection.TacticCollection;
const TacticSummary = __importStar(require("./tactic-summary"));
exports.TacticSummary = TacticSummary.TacticSummary;
const Target = __importStar(require("./target"));
exports.Target = Target.Target;
const TargetCollection = __importStar(require("./target-collection"));
exports.TargetCollection = TargetCollection.TargetCollection;
const TargetDetails = __importStar(require("./target-details"));
exports.TargetDetails = TargetDetails.TargetDetails;
const TargetDetectorDetails = __importStar(require("./target-detector-details"));
exports.TargetDetectorDetails = TargetDetectorDetails.TargetDetectorDetails;
const TargetDetectorRecipe = __importStar(require("./target-detector-recipe"));
exports.TargetDetectorRecipe = TargetDetectorRecipe.TargetDetectorRecipe;
const TargetDetectorRecipeCollection = __importStar(require("./target-detector-recipe-collection"));
exports.TargetDetectorRecipeCollection = TargetDetectorRecipeCollection.TargetDetectorRecipeCollection;
const TargetDetectorRecipeDetectorRule = __importStar(require("./target-detector-recipe-detector-rule"));
exports.TargetDetectorRecipeDetectorRule = TargetDetectorRecipeDetectorRule.TargetDetectorRecipeDetectorRule;
const TargetDetectorRecipeDetectorRuleCollection = __importStar(require("./target-detector-recipe-detector-rule-collection"));
exports.TargetDetectorRecipeDetectorRuleCollection = TargetDetectorRecipeDetectorRuleCollection.TargetDetectorRecipeDetectorRuleCollection;
const TargetDetectorRecipeDetectorRuleSummary = __importStar(require("./target-detector-recipe-detector-rule-summary"));
exports.TargetDetectorRecipeDetectorRuleSummary = TargetDetectorRecipeDetectorRuleSummary.TargetDetectorRecipeDetectorRuleSummary;
const TargetDetectorRecipeSummary = __importStar(require("./target-detector-recipe-summary"));
exports.TargetDetectorRecipeSummary = TargetDetectorRecipeSummary.TargetDetectorRecipeSummary;
const TargetResourceType = __importStar(require("./target-resource-type"));
exports.TargetResourceType = TargetResourceType.TargetResourceType;
const TargetResponderRecipe = __importStar(require("./target-responder-recipe"));
exports.TargetResponderRecipe = TargetResponderRecipe.TargetResponderRecipe;
const TargetResponderRecipeCollection = __importStar(require("./target-responder-recipe-collection"));
exports.TargetResponderRecipeCollection = TargetResponderRecipeCollection.TargetResponderRecipeCollection;
const TargetResponderRecipeResponderRule = __importStar(require("./target-responder-recipe-responder-rule"));
exports.TargetResponderRecipeResponderRule = TargetResponderRecipeResponderRule.TargetResponderRecipeResponderRule;
const TargetResponderRecipeResponderRuleCollection = __importStar(require("./target-responder-recipe-responder-rule-collection"));
exports.TargetResponderRecipeResponderRuleCollection = TargetResponderRecipeResponderRuleCollection.TargetResponderRecipeResponderRuleCollection;
const TargetResponderRecipeResponderRuleSummary = __importStar(require("./target-responder-recipe-responder-rule-summary"));
exports.TargetResponderRecipeResponderRuleSummary = TargetResponderRecipeResponderRuleSummary.TargetResponderRecipeResponderRuleSummary;
const TargetResponderRecipeSummary = __importStar(require("./target-responder-recipe-summary"));
exports.TargetResponderRecipeSummary = TargetResponderRecipeSummary.TargetResponderRecipeSummary;
const TargetSelected = __importStar(require("./target-selected"));
exports.TargetSelected = TargetSelected.TargetSelected;
const TargetSummary = __importStar(require("./target-summary"));
exports.TargetSummary = TargetSummary.TargetSummary;
const TechniqueCollection = __importStar(require("./technique-collection"));
exports.TechniqueCollection = TechniqueCollection.TechniqueCollection;
const TechniqueSummary = __importStar(require("./technique-summary"));
exports.TechniqueSummary = TechniqueSummary.TechniqueSummary;
const TriggerResponderDetails = __importStar(require("./trigger-responder-details"));
exports.TriggerResponderDetails = TriggerResponderDetails.TriggerResponderDetails;
const UpdateBulkProblemStatusDetails = __importStar(require("./update-bulk-problem-status-details"));
exports.UpdateBulkProblemStatusDetails = UpdateBulkProblemStatusDetails.UpdateBulkProblemStatusDetails;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateDataMaskRuleDetails = __importStar(require("./update-data-mask-rule-details"));
exports.UpdateDataMaskRuleDetails = UpdateDataMaskRuleDetails.UpdateDataMaskRuleDetails;
const UpdateDataSourceDetails = __importStar(require("./update-data-source-details"));
exports.UpdateDataSourceDetails = UpdateDataSourceDetails.UpdateDataSourceDetails;
const UpdateDetectorRecipeDetails = __importStar(require("./update-detector-recipe-details"));
exports.UpdateDetectorRecipeDetails = UpdateDetectorRecipeDetails.UpdateDetectorRecipeDetails;
const UpdateDetectorRecipeDetectorRule = __importStar(require("./update-detector-recipe-detector-rule"));
exports.UpdateDetectorRecipeDetectorRule = UpdateDetectorRecipeDetectorRule.UpdateDetectorRecipeDetectorRule;
const UpdateDetectorRecipeDetectorRuleDetails = __importStar(require("./update-detector-recipe-detector-rule-details"));
exports.UpdateDetectorRecipeDetectorRuleDetails = UpdateDetectorRecipeDetectorRuleDetails.UpdateDetectorRecipeDetectorRuleDetails;
const UpdateDetectorRuleDetails = __importStar(require("./update-detector-rule-details"));
exports.UpdateDetectorRuleDetails = UpdateDetectorRuleDetails.UpdateDetectorRuleDetails;
const UpdateManagedListDetails = __importStar(require("./update-managed-list-details"));
exports.UpdateManagedListDetails = UpdateManagedListDetails.UpdateManagedListDetails;
const UpdateProblemStatusDetails = __importStar(require("./update-problem-status-details"));
exports.UpdateProblemStatusDetails = UpdateProblemStatusDetails.UpdateProblemStatusDetails;
const UpdateResponderRecipeDetails = __importStar(require("./update-responder-recipe-details"));
exports.UpdateResponderRecipeDetails = UpdateResponderRecipeDetails.UpdateResponderRecipeDetails;
const UpdateResponderRecipeResponderRule = __importStar(require("./update-responder-recipe-responder-rule"));
exports.UpdateResponderRecipeResponderRule = UpdateResponderRecipeResponderRule.UpdateResponderRecipeResponderRule;
const UpdateResponderRecipeResponderRuleDetails = __importStar(require("./update-responder-recipe-responder-rule-details"));
exports.UpdateResponderRecipeResponderRuleDetails = UpdateResponderRecipeResponderRuleDetails.UpdateResponderRecipeResponderRuleDetails;
const UpdateResponderRuleDetails = __importStar(require("./update-responder-rule-details"));
exports.UpdateResponderRuleDetails = UpdateResponderRuleDetails.UpdateResponderRuleDetails;
const UpdateSecurityPolicyDetails = __importStar(require("./update-security-policy-details"));
exports.UpdateSecurityPolicyDetails = UpdateSecurityPolicyDetails.UpdateSecurityPolicyDetails;
const UpdateSecurityRecipeDetails = __importStar(require("./update-security-recipe-details"));
exports.UpdateSecurityRecipeDetails = UpdateSecurityRecipeDetails.UpdateSecurityRecipeDetails;
const UpdateSecurityZoneDetails = __importStar(require("./update-security-zone-details"));
exports.UpdateSecurityZoneDetails = UpdateSecurityZoneDetails.UpdateSecurityZoneDetails;
const UpdateTargetDetails = __importStar(require("./update-target-details"));
exports.UpdateTargetDetails = UpdateTargetDetails.UpdateTargetDetails;
const UpdateTargetDetectorRecipe = __importStar(require("./update-target-detector-recipe"));
exports.UpdateTargetDetectorRecipe = UpdateTargetDetectorRecipe.UpdateTargetDetectorRecipe;
const UpdateTargetDetectorRecipeDetails = __importStar(require("./update-target-detector-recipe-details"));
exports.UpdateTargetDetectorRecipeDetails = UpdateTargetDetectorRecipeDetails.UpdateTargetDetectorRecipeDetails;
const UpdateTargetDetectorRecipeDetectorRuleDetails = __importStar(require("./update-target-detector-recipe-detector-rule-details"));
exports.UpdateTargetDetectorRecipeDetectorRuleDetails = UpdateTargetDetectorRecipeDetectorRuleDetails.UpdateTargetDetectorRecipeDetectorRuleDetails;
const UpdateTargetDetectorRuleDetails = __importStar(require("./update-target-detector-rule-details"));
exports.UpdateTargetDetectorRuleDetails = UpdateTargetDetectorRuleDetails.UpdateTargetDetectorRuleDetails;
const UpdateTargetRecipeDetectorRuleDetails = __importStar(require("./update-target-recipe-detector-rule-details"));
exports.UpdateTargetRecipeDetectorRuleDetails = UpdateTargetRecipeDetectorRuleDetails.UpdateTargetRecipeDetectorRuleDetails;
const UpdateTargetRecipeResponderRuleDetails = __importStar(require("./update-target-recipe-responder-rule-details"));
exports.UpdateTargetRecipeResponderRuleDetails = UpdateTargetRecipeResponderRuleDetails.UpdateTargetRecipeResponderRuleDetails;
const UpdateTargetResponderRecipe = __importStar(require("./update-target-responder-recipe"));
exports.UpdateTargetResponderRecipe = UpdateTargetResponderRecipe.UpdateTargetResponderRecipe;
const UpdateTargetResponderRecipeDetails = __importStar(require("./update-target-responder-recipe-details"));
exports.UpdateTargetResponderRecipeDetails = UpdateTargetResponderRecipeDetails.UpdateTargetResponderRecipeDetails;
const UpdateTargetResponderRecipeResponderRuleDetails = __importStar(require("./update-target-responder-recipe-responder-rule-details"));
exports.UpdateTargetResponderRecipeResponderRuleDetails = UpdateTargetResponderRecipeResponderRuleDetails.UpdateTargetResponderRecipeResponderRuleDetails;
const UpdateTargetResponderRuleDetails = __importStar(require("./update-target-responder-rule-details"));
exports.UpdateTargetResponderRuleDetails = UpdateTargetResponderRuleDetails.UpdateTargetResponderRuleDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const AbsoluteTimeStartPolicy = __importStar(require("./absolute-time-start-policy"));
exports.AbsoluteTimeStartPolicy = AbsoluteTimeStartPolicy.AbsoluteTimeStartPolicy;
const AllTargetsSelected = __importStar(require("./all-targets-selected"));
exports.AllTargetsSelected = AllTargetsSelected.AllTargetsSelected;
const CompositeCondition = __importStar(require("./composite-condition"));
exports.CompositeCondition = CompositeCondition.CompositeCondition;
const InsightTypeLoggingQueryDetails = __importStar(require("./insight-type-logging-query-details"));
exports.InsightTypeLoggingQueryDetails = InsightTypeLoggingQueryDetails.InsightTypeLoggingQueryDetails;
const LoggingEventInfo = __importStar(require("./logging-event-info"));
exports.LoggingEventInfo = LoggingEventInfo.LoggingEventInfo;
const LoggingQueryDataSourceDetails = __importStar(require("./logging-query-data-source-details"));
exports.LoggingQueryDataSourceDetails = LoggingQueryDataSourceDetails.LoggingQueryDataSourceDetails;
const LoggingQueryDataSourceSummaryDetails = __importStar(require("./logging-query-data-source-summary-details"));
exports.LoggingQueryDataSourceSummaryDetails = LoggingQueryDataSourceSummaryDetails.LoggingQueryDataSourceSummaryDetails;
const NoDelayStartPolicy = __importStar(require("./no-delay-start-policy"));
exports.NoDelayStartPolicy = NoDelayStartPolicy.NoDelayStartPolicy;
const SecurityZoneTargetDetails = __importStar(require("./security-zone-target-details"));
exports.SecurityZoneTargetDetails = SecurityZoneTargetDetails.SecurityZoneTargetDetails;
const SimpleCondition = __importStar(require("./simple-condition"));
exports.SimpleCondition = SimpleCondition.SimpleCondition;
const TargetIdsSelected = __importStar(require("./target-ids-selected"));
exports.TargetIdsSelected = TargetIdsSelected.TargetIdsSelected;
const TargetResourceTypesSelected = __importStar(require("./target-resource-types-selected"));
exports.TargetResourceTypesSelected = TargetResourceTypesSelected.TargetResourceTypesSelected;
