// oci-cloudguard/lib/model/problem-dimension.js
"use strict";
/**
 * Cloud Guard and Security Zones API
 * Use the Cloud Guard and Security Zones API to automate processes that you would otherwise perform through the Cloud Guard Console or the Security Zones Console. For more information on these services, see the [Cloud Guard](/iaas/cloud-guard/home.htm) and [Security Zones](/iaas/security-zone/home.htm) documentation.

**Note:** For Cloud Guard, you can perform Create, Update, and Delete operations only from the reporting region of your Cloud Guard tenancy. You can perform Read operations from any region.

 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProblemDimension = void 0;
/**
 * Possible dimensions for distinguishing Problems
 **/
var ProblemDimension;
(function (ProblemDimension) {
    ProblemDimension["ResourceType"] = "RESOURCE_TYPE";
    ProblemDimension["Region"] = "REGION";
    ProblemDimension["CompartmentId"] = "COMPARTMENT_ID";
    ProblemDimension["RiskLevel"] = "RISK_LEVEL";
})(ProblemDimension = exports.ProblemDimension || (exports.ProblemDimension = {}));
(function (ProblemDimension) {
    function getJsonObj(obj) {
        return obj;
    }
    ProblemDimension.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ProblemDimension.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProblemDimension = exports.ProblemDimension || (exports.ProblemDimension = {}));
