// oci-databasemanagement/lib/request/addm-tasks-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddmTasksRequest = void 0;
var AddmTasksRequest;
(function (AddmTasksRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TaskName"] = "TASK_NAME";
        SortBy["TaskId"] = "TASK_ID";
        SortBy["Description"] = "DESCRIPTION";
        SortBy["DbUser"] = "DB_USER";
        SortBy["Status"] = "STATUS";
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["BeginTime"] = "BEGIN_TIME";
        SortBy["EndTime"] = "END_TIME";
    })(SortBy = AddmTasksRequest.SortBy || (AddmTasksRequest.SortBy = {}));
})(AddmTasksRequest = exports.AddmTasksRequest || (exports.AddmTasksRequest = {}));
