// oci-databasemanagement/lib/request/get-sql-tuning-advisor-task-summary-report-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSqlTuningAdvisorTaskSummaryReportRequest = void 0;
var GetSqlTuningAdvisorTaskSummaryReportRequest;
(function (GetSqlTuningAdvisorTaskSummaryReportRequest) {
    let SearchPeriod;
    (function (SearchPeriod) {
        SearchPeriod["Last24Hr"] = "LAST_24HR";
        SearchPeriod["Last7Day"] = "LAST_7DAY";
        SearchPeriod["Last31Day"] = "LAST_31DAY";
        SearchPeriod["SinceLast"] = "SINCE_LAST";
        SearchPeriod["All"] = "ALL";
    })(SearchPeriod = GetSqlTuningAdvisorTaskSummaryReportRequest.SearchPeriod || (GetSqlTuningAdvisorTaskSummaryReportRequest.SearchPeriod = {}));
})(GetSqlTuningAdvisorTaskSummaryReportRequest = exports.GetSqlTuningAdvisorTaskSummaryReportRequest || (exports.GetSqlTuningAdvisorTaskSummaryReportRequest = {}));
