// oci-databasemanagement/lib/request/get-sql-execution-plan-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSqlExecutionPlanRequest = void 0;
var GetSqlExecutionPlanRequest;
(function (GetSqlExecutionPlanRequest) {
    let Attribute;
    (function (Attribute) {
        Attribute["Original"] = "ORIGINAL";
        Attribute["OriginalWithAdjustedCost"] = "ORIGINAL_WITH_ADJUSTED_COST";
        Attribute["UsingSqlProfile"] = "USING_SQL_PROFILE";
        Attribute["UsingNewIndices"] = "USING_NEW_INDICES";
        Attribute["UsingParallelExecution"] = "USING_PARALLEL_EXECUTION";
    })(Attribute = GetSqlExecutionPlanRequest.Attribute || (GetSqlExecutionPlanRequest.Attribute = {}));
})(GetSqlExecutionPlanRequest = exports.GetSqlExecutionPlanRequest || (exports.GetSqlExecutionPlanRequest = {}));
