// oci-databasemanagement/lib/request/list-sql-tuning-advisor-tasks-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSqlTuningAdvisorTasksRequest = void 0;
var ListSqlTuningAdvisorTasksRequest;
(function (ListSqlTuningAdvisorTasksRequest) {
    let Status;
    (function (Status) {
        Status["Initial"] = "INITIAL";
        Status["Executing"] = "EXECUTING";
        Status["Interrupted"] = "INTERRUPTED";
        Status["Completed"] = "COMPLETED";
        Status["Error"] = "ERROR";
    })(Status = ListSqlTuningAdvisorTasksRequest.Status || (ListSqlTuningAdvisorTasksRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
        SortBy["StartTime"] = "START_TIME";
    })(SortBy = ListSqlTuningAdvisorTasksRequest.SortBy || (ListSqlTuningAdvisorTasksRequest.SortBy = {}));
})(ListSqlTuningAdvisorTasksRequest = exports.ListSqlTuningAdvisorTasksRequest || (exports.ListSqlTuningAdvisorTasksRequest = {}));
