// oci-databasemanagement/lib/request/list-alert-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAlertLogsRequest = void 0;
var ListAlertLogsRequest;
(function (ListAlertLogsRequest) {
    let LevelFilter;
    (function (LevelFilter) {
        LevelFilter["Critical"] = "CRITICAL";
        LevelFilter["Severe"] = "SEVERE";
        LevelFilter["Important"] = "IMPORTANT";
        LevelFilter["Normal"] = "NORMAL";
        LevelFilter["All"] = "ALL";
    })(LevelFilter = ListAlertLogsRequest.LevelFilter || (ListAlertLogsRequest.LevelFilter = {}));
    let TypeFilter;
    (function (TypeFilter) {
        TypeFilter["Unknown"] = "UNKNOWN";
        TypeFilter["IncidentError"] = "INCIDENT_ERROR";
        TypeFilter["Error"] = "ERROR";
        TypeFilter["Warning"] = "WARNING";
        TypeFilter["Notification"] = "NOTIFICATION";
        TypeFilter["Trace"] = "TRACE";
        TypeFilter["All"] = "ALL";
    })(TypeFilter = ListAlertLogsRequest.TypeFilter || (ListAlertLogsRequest.TypeFilter = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Level"] = "LEVEL";
        SortBy["Type"] = "TYPE";
        SortBy["Message"] = "MESSAGE";
        SortBy["Timestamp"] = "TIMESTAMP";
    })(SortBy = ListAlertLogsRequest.SortBy || (ListAlertLogsRequest.SortBy = {}));
})(ListAlertLogsRequest = exports.ListAlertLogsRequest || (exports.ListAlertLogsRequest = {}));
