// oci-databasemanagement/lib/request/list-sql-tuning-advisor-task-findings-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSqlTuningAdvisorTaskFindingsRequest = void 0;
var ListSqlTuningAdvisorTaskFindingsRequest;
(function (ListSqlTuningAdvisorTaskFindingsRequest) {
    let SearchPeriod;
    (function (SearchPeriod) {
        SearchPeriod["Last24Hr"] = "LAST_24HR";
        SearchPeriod["Last7Day"] = "LAST_7DAY";
        SearchPeriod["Last31Day"] = "LAST_31DAY";
        SearchPeriod["SinceLast"] = "SINCE_LAST";
        SearchPeriod["All"] = "ALL";
    })(SearchPeriod = ListSqlTuningAdvisorTaskFindingsRequest.SearchPeriod || (ListSqlTuningAdvisorTaskFindingsRequest.SearchPeriod = {}));
    let FindingFilter;
    (function (FindingFilter) {
        FindingFilter["None"] = "none";
        FindingFilter["Findings"] = "FINDINGS";
        FindingFilter["Nofindings"] = "NOFINDINGS";
        FindingFilter["Errors"] = "ERRORS";
        FindingFilter["Profiles"] = "PROFILES";
        FindingFilter["Indices"] = "INDICES";
        FindingFilter["Stats"] = "STATS";
        FindingFilter["Restructure"] = "RESTRUCTURE";
        FindingFilter["Alternative"] = "ALTERNATIVE";
        FindingFilter["AutoProfiles"] = "AUTO_PROFILES";
        FindingFilter["OtherProfiles"] = "OTHER_PROFILES";
    })(FindingFilter = ListSqlTuningAdvisorTaskFindingsRequest.FindingFilter || (ListSqlTuningAdvisorTaskFindingsRequest.FindingFilter = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["DbtimeBenefit"] = "DBTIME_BENEFIT";
        SortBy["ParsingSchema"] = "PARSING_SCHEMA";
        SortBy["SqlId"] = "SQL_ID";
        SortBy["Stats"] = "STATS";
        SortBy["Profiles"] = "PROFILES";
        SortBy["SqlBenefit"] = "SQL_BENEFIT";
        SortBy["Date"] = "DATE";
        SortBy["Indices"] = "INDICES";
        SortBy["Restructure"] = "RESTRUCTURE";
        SortBy["Alternative"] = "ALTERNATIVE";
        SortBy["Misc"] = "MISC";
        SortBy["Error"] = "ERROR";
        SortBy["Timeouts"] = "TIMEOUTS";
    })(SortBy = ListSqlTuningAdvisorTaskFindingsRequest.SortBy || (ListSqlTuningAdvisorTaskFindingsRequest.SortBy = {}));
})(ListSqlTuningAdvisorTaskFindingsRequest = exports.ListSqlTuningAdvisorTaskFindingsRequest || (exports.ListSqlTuningAdvisorTaskFindingsRequest = {}));
