// oci-databasemanagement/lib/request/list-attention-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAttentionLogsRequest = void 0;
var ListAttentionLogsRequest;
(function (ListAttentionLogsRequest) {
    let UrgencyFilter;
    (function (UrgencyFilter) {
        UrgencyFilter["Immediate"] = "IMMEDIATE";
        UrgencyFilter["Soon"] = "SOON";
        UrgencyFilter["Deferrable"] = "DEFERRABLE";
        UrgencyFilter["Info"] = "INFO";
        UrgencyFilter["All"] = "ALL";
    })(UrgencyFilter = ListAttentionLogsRequest.UrgencyFilter || (ListAttentionLogsRequest.UrgencyFilter = {}));
    let TypeFilter;
    (function (TypeFilter) {
        TypeFilter["Unknown"] = "UNKNOWN";
        TypeFilter["IncidentError"] = "INCIDENT_ERROR";
        TypeFilter["Error"] = "ERROR";
        TypeFilter["Warning"] = "WARNING";
        TypeFilter["Notification"] = "NOTIFICATION";
        TypeFilter["Trace"] = "TRACE";
        TypeFilter["All"] = "ALL";
    })(TypeFilter = ListAttentionLogsRequest.TypeFilter || (ListAttentionLogsRequest.TypeFilter = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Urgency"] = "URGENCY";
        SortBy["Type"] = "TYPE";
        SortBy["Message"] = "MESSAGE";
        SortBy["Timestamp"] = "TIMESTAMP";
        SortBy["Scope"] = "SCOPE";
        SortBy["TargetUser"] = "TARGET_USER";
    })(SortBy = ListAttentionLogsRequest.SortBy || (ListAttentionLogsRequest.SortBy = {}));
})(ListAttentionLogsRequest = exports.ListAttentionLogsRequest || (exports.ListAttentionLogsRequest = {}));
