// oci-databasemanagement/lib/request/summarize-attention-log-counts-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAttentionLogCountsRequest = void 0;
var SummarizeAttentionLogCountsRequest;
(function (SummarizeAttentionLogCountsRequest) {
    let UrgencyFilter;
    (function (UrgencyFilter) {
        UrgencyFilter["Immediate"] = "IMMEDIATE";
        UrgencyFilter["Soon"] = "SOON";
        UrgencyFilter["Deferrable"] = "DEFERRABLE";
        UrgencyFilter["Info"] = "INFO";
        UrgencyFilter["All"] = "ALL";
    })(UrgencyFilter = SummarizeAttentionLogCountsRequest.UrgencyFilter || (SummarizeAttentionLogCountsRequest.UrgencyFilter = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["Urgency"] = "URGENCY";
        GroupBy["Type"] = "TYPE";
    })(GroupBy = SummarizeAttentionLogCountsRequest.GroupBy || (SummarizeAttentionLogCountsRequest.GroupBy = {}));
    let TypeFilter;
    (function (TypeFilter) {
        TypeFilter["Unknown"] = "UNKNOWN";
        TypeFilter["IncidentError"] = "INCIDENT_ERROR";
        TypeFilter["Error"] = "ERROR";
        TypeFilter["Warning"] = "WARNING";
        TypeFilter["Notification"] = "NOTIFICATION";
        TypeFilter["Trace"] = "TRACE";
        TypeFilter["All"] = "ALL";
    })(TypeFilter = SummarizeAttentionLogCountsRequest.TypeFilter || (SummarizeAttentionLogCountsRequest.TypeFilter = {}));
})(SummarizeAttentionLogCountsRequest = exports.SummarizeAttentionLogCountsRequest || (exports.SummarizeAttentionLogCountsRequest = {}));
