// oci-databasemanagement/lib/request/summarize-awr-db-wait-events-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAwrDbWaitEventsRequest = void 0;
var SummarizeAwrDbWaitEventsRequest;
(function (SummarizeAwrDbWaitEventsRequest) {
    let SessionType;
    (function (SessionType) {
        SessionType["Foreground"] = "FOREGROUND";
        SessionType["Background"] = "BACKGROUND";
        SessionType["All"] = "ALL";
    })(SessionType = SummarizeAwrDbWaitEventsRequest.SessionType || (SummarizeAwrDbWaitEventsRequest.SessionType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeBegin"] = "TIME_BEGIN";
        SortBy["Name"] = "NAME";
    })(SortBy = SummarizeAwrDbWaitEventsRequest.SortBy || (SummarizeAwrDbWaitEventsRequest.SortBy = {}));
})(SummarizeAwrDbWaitEventsRequest = exports.SummarizeAwrDbWaitEventsRequest || (exports.SummarizeAwrDbWaitEventsRequest = {}));
