// oci-databasemanagement/lib/request/summarize-awr-db-parameters-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAwrDbParametersRequest = void 0;
var SummarizeAwrDbParametersRequest;
(function (SummarizeAwrDbParametersRequest) {
    let ValueChanged;
    (function (ValueChanged) {
        ValueChanged["Y"] = "Y";
        ValueChanged["N"] = "N";
    })(ValueChanged = SummarizeAwrDbParametersRequest.ValueChanged || (SummarizeAwrDbParametersRequest.ValueChanged = {}));
    let ValueDefault;
    (function (ValueDefault) {
        ValueDefault["True"] = "TRUE";
        ValueDefault["False"] = "FALSE";
    })(ValueDefault = SummarizeAwrDbParametersRequest.ValueDefault || (SummarizeAwrDbParametersRequest.ValueDefault = {}));
    let ValueModified;
    (function (ValueModified) {
        ValueModified["Modified"] = "MODIFIED";
        ValueModified["SystemMod"] = "SYSTEM_MOD";
        ValueModified["False"] = "FALSE";
    })(ValueModified = SummarizeAwrDbParametersRequest.ValueModified || (SummarizeAwrDbParametersRequest.ValueModified = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["IsChanged"] = "IS_CHANGED";
        SortBy["Name"] = "NAME";
    })(SortBy = SummarizeAwrDbParametersRequest.SortBy || (SummarizeAwrDbParametersRequest.SortBy = {}));
})(SummarizeAwrDbParametersRequest = exports.SummarizeAwrDbParametersRequest || (exports.SummarizeAwrDbParametersRequest = {}));
