// oci-databasemanagement/lib/request/summarize-alert-log-counts-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAlertLogCountsRequest = void 0;
var SummarizeAlertLogCountsRequest;
(function (SummarizeAlertLogCountsRequest) {
    let LevelFilter;
    (function (LevelFilter) {
        LevelFilter["Critical"] = "CRITICAL";
        LevelFilter["Severe"] = "SEVERE";
        LevelFilter["Important"] = "IMPORTANT";
        LevelFilter["Normal"] = "NORMAL";
        LevelFilter["All"] = "ALL";
    })(LevelFilter = SummarizeAlertLogCountsRequest.LevelFilter || (SummarizeAlertLogCountsRequest.LevelFilter = {}));
    let GroupBy;
    (function (GroupBy) {
        GroupBy["Level"] = "LEVEL";
        GroupBy["Type"] = "TYPE";
    })(GroupBy = SummarizeAlertLogCountsRequest.GroupBy || (SummarizeAlertLogCountsRequest.GroupBy = {}));
    let TypeFilter;
    (function (TypeFilter) {
        TypeFilter["Unknown"] = "UNKNOWN";
        TypeFilter["IncidentError"] = "INCIDENT_ERROR";
        TypeFilter["Error"] = "ERROR";
        TypeFilter["Warning"] = "WARNING";
        TypeFilter["Notification"] = "NOTIFICATION";
        TypeFilter["Trace"] = "TRACE";
        TypeFilter["All"] = "ALL";
    })(TypeFilter = SummarizeAlertLogCountsRequest.TypeFilter || (SummarizeAlertLogCountsRequest.TypeFilter = {}));
})(SummarizeAlertLogCountsRequest = exports.SummarizeAlertLogCountsRequest || (exports.SummarizeAlertLogCountsRequest = {}));
