// oci-recovery/lib/model/fetch-protected-database-configuration-details.js
"use strict";
/**
 * Oracle Database Autonomous Recovery Service API
 * Use Oracle Database Autonomous Recovery Service API to manage Protected Databases.
 * OpenAPI spec version: 20210216
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FetchProtectedDatabaseConfigurationDetails = void 0;
var FetchProtectedDatabaseConfigurationDetails;
(function (FetchProtectedDatabaseConfigurationDetails) {
    let ConfigurationType;
    (function (ConfigurationType) {
        ConfigurationType["Cabundle"] = "CABUNDLE";
        ConfigurationType["Tnsnames"] = "TNSNAMES";
        ConfigurationType["Hosts"] = "HOSTS";
        ConfigurationType["All"] = "ALL";
    })(ConfigurationType = FetchProtectedDatabaseConfigurationDetails.ConfigurationType || (FetchProtectedDatabaseConfigurationDetails.ConfigurationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FetchProtectedDatabaseConfigurationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FetchProtectedDatabaseConfigurationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(FetchProtectedDatabaseConfigurationDetails = exports.FetchProtectedDatabaseConfigurationDetails || (exports.FetchProtectedDatabaseConfigurationDetails = {}));
