// oci-recovery/lib/model/health.js
"use strict";
/**
 * Oracle Database Autonomous Recovery Service API
 * Use Oracle Database Autonomous Recovery Service API to manage Protected Databases.
 * OpenAPI spec version: 20210216
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Health = void 0;
/**
 * The current health of the resource
 **/
var Health;
(function (Health) {
    Health["Protected"] = "PROTECTED";
    Health["Warning"] = "WARNING";
    Health["Alert"] = "ALERT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Health["UnknownValue"] = "UNKNOWN_VALUE";
})(Health = exports.Health || (exports.Health = {}));
(function (Health) {
    function getJsonObj(obj) {
        return obj;
    }
    Health.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    Health.getDeserializedJsonObj = getDeserializedJsonObj;
})(Health = exports.Health || (exports.Health = {}));
