// oci-recovery/lib/model/index.js
"use strict";
/**
 * Oracle Database Autonomous Recovery Service API
 * Use Oracle Database Autonomous Recovery Service API to manage Protected Databases.
 * OpenAPI spec version: 20210216
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateRecoveryServiceSubnetDetails = exports.UpdateProtectionPolicyDetails = exports.UpdateProtectedDatabaseDetails = exports.SortOrder = exports.RecoveryServiceSubnetSummary = exports.RecoveryServiceSubnetInput = exports.RecoveryServiceSubnetDetails = exports.RecoveryServiceSubnetCollection = exports.RecoveryServiceSubnet = exports.ProtectionPolicySummary = exports.ProtectionPolicyCollection = exports.ProtectionPolicy = exports.ProtectedDatabaseSummary = exports.ProtectedDatabaseCollection = exports.ProtectedDatabase = exports.OperationType = exports.OperationStatus = exports.MetricsSummary = exports.Metrics = exports.LifecycleState = exports.Health = exports.FetchProtectedDatabaseConfigurationDetails = exports.DatabaseSizes = exports.CreateRecoveryServiceSubnetDetails = exports.CreateProtectionPolicyDetails = exports.CreateProtectedDatabaseDetails = exports.ChangeRecoveryServiceSubnetCompartmentDetails = exports.ChangeProtectionPolicyCompartmentDetails = exports.ChangeProtectedDatabaseCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeProtectedDatabaseCompartmentDetails = __importStar(require("./change-protected-database-compartment-details"));
exports.ChangeProtectedDatabaseCompartmentDetails = ChangeProtectedDatabaseCompartmentDetails.ChangeProtectedDatabaseCompartmentDetails;
const ChangeProtectionPolicyCompartmentDetails = __importStar(require("./change-protection-policy-compartment-details"));
exports.ChangeProtectionPolicyCompartmentDetails = ChangeProtectionPolicyCompartmentDetails.ChangeProtectionPolicyCompartmentDetails;
const ChangeRecoveryServiceSubnetCompartmentDetails = __importStar(require("./change-recovery-service-subnet-compartment-details"));
exports.ChangeRecoveryServiceSubnetCompartmentDetails = ChangeRecoveryServiceSubnetCompartmentDetails.ChangeRecoveryServiceSubnetCompartmentDetails;
const CreateProtectedDatabaseDetails = __importStar(require("./create-protected-database-details"));
exports.CreateProtectedDatabaseDetails = CreateProtectedDatabaseDetails.CreateProtectedDatabaseDetails;
const CreateProtectionPolicyDetails = __importStar(require("./create-protection-policy-details"));
exports.CreateProtectionPolicyDetails = CreateProtectionPolicyDetails.CreateProtectionPolicyDetails;
const CreateRecoveryServiceSubnetDetails = __importStar(require("./create-recovery-service-subnet-details"));
exports.CreateRecoveryServiceSubnetDetails = CreateRecoveryServiceSubnetDetails.CreateRecoveryServiceSubnetDetails;
const DatabaseSizes = __importStar(require("./database-sizes"));
exports.DatabaseSizes = DatabaseSizes.DatabaseSizes;
const FetchProtectedDatabaseConfigurationDetails = __importStar(require("./fetch-protected-database-configuration-details"));
exports.FetchProtectedDatabaseConfigurationDetails = FetchProtectedDatabaseConfigurationDetails.FetchProtectedDatabaseConfigurationDetails;
const Health = __importStar(require("./health"));
exports.Health = Health.Health;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const Metrics = __importStar(require("./metrics"));
exports.Metrics = Metrics.Metrics;
const MetricsSummary = __importStar(require("./metrics-summary"));
exports.MetricsSummary = MetricsSummary.MetricsSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ProtectedDatabase = __importStar(require("./protected-database"));
exports.ProtectedDatabase = ProtectedDatabase.ProtectedDatabase;
const ProtectedDatabaseCollection = __importStar(require("./protected-database-collection"));
exports.ProtectedDatabaseCollection = ProtectedDatabaseCollection.ProtectedDatabaseCollection;
const ProtectedDatabaseSummary = __importStar(require("./protected-database-summary"));
exports.ProtectedDatabaseSummary = ProtectedDatabaseSummary.ProtectedDatabaseSummary;
const ProtectionPolicy = __importStar(require("./protection-policy"));
exports.ProtectionPolicy = ProtectionPolicy.ProtectionPolicy;
const ProtectionPolicyCollection = __importStar(require("./protection-policy-collection"));
exports.ProtectionPolicyCollection = ProtectionPolicyCollection.ProtectionPolicyCollection;
const ProtectionPolicySummary = __importStar(require("./protection-policy-summary"));
exports.ProtectionPolicySummary = ProtectionPolicySummary.ProtectionPolicySummary;
const RecoveryServiceSubnet = __importStar(require("./recovery-service-subnet"));
exports.RecoveryServiceSubnet = RecoveryServiceSubnet.RecoveryServiceSubnet;
const RecoveryServiceSubnetCollection = __importStar(require("./recovery-service-subnet-collection"));
exports.RecoveryServiceSubnetCollection = RecoveryServiceSubnetCollection.RecoveryServiceSubnetCollection;
const RecoveryServiceSubnetDetails = __importStar(require("./recovery-service-subnet-details"));
exports.RecoveryServiceSubnetDetails = RecoveryServiceSubnetDetails.RecoveryServiceSubnetDetails;
const RecoveryServiceSubnetInput = __importStar(require("./recovery-service-subnet-input"));
exports.RecoveryServiceSubnetInput = RecoveryServiceSubnetInput.RecoveryServiceSubnetInput;
const RecoveryServiceSubnetSummary = __importStar(require("./recovery-service-subnet-summary"));
exports.RecoveryServiceSubnetSummary = RecoveryServiceSubnetSummary.RecoveryServiceSubnetSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateProtectedDatabaseDetails = __importStar(require("./update-protected-database-details"));
exports.UpdateProtectedDatabaseDetails = UpdateProtectedDatabaseDetails.UpdateProtectedDatabaseDetails;
const UpdateProtectionPolicyDetails = __importStar(require("./update-protection-policy-details"));
exports.UpdateProtectionPolicyDetails = UpdateProtectionPolicyDetails.UpdateProtectionPolicyDetails;
const UpdateRecoveryServiceSubnetDetails = __importStar(require("./update-recovery-service-subnet-details"));
exports.UpdateRecoveryServiceSubnetDetails = UpdateRecoveryServiceSubnetDetails.UpdateRecoveryServiceSubnetDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
