// oci-databasemanagement/lib/model/alert-log-count-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertLogCountSummary = void 0;
var AlertLogCountSummary;
(function (AlertLogCountSummary) {
    let Category;
    (function (Category) {
        Category["Unknown"] = "UNKNOWN";
        Category["IncidentError"] = "INCIDENT_ERROR";
        Category["Error"] = "ERROR";
        Category["Warning"] = "WARNING";
        Category["Notification"] = "NOTIFICATION";
        Category["Trace"] = "TRACE";
        Category["Critical"] = "CRITICAL";
        Category["Severe"] = "SEVERE";
        Category["Important"] = "IMPORTANT";
        Category["Normal"] = "NORMAL";
        Category["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Category["UnknownValue"] = "UNKNOWN_VALUE";
    })(Category = AlertLogCountSummary.Category || (AlertLogCountSummary.Category = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AlertLogCountSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AlertLogCountSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlertLogCountSummary = exports.AlertLogCountSummary || (exports.AlertLogCountSummary = {}));
