// oci-databasemanagement/lib/model/alert-type-enum.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertTypeEnum = void 0;
/**
 * Stateful alerts are automatically cleared on severity transition to normal.
 * Stateless alerts are never cleared. You can change the alert by setting the examinedBy attribute.
 *
 **/
var AlertTypeEnum;
(function (AlertTypeEnum) {
    AlertTypeEnum["Stateful"] = "STATEFUL";
    AlertTypeEnum["Stateless"] = "STATELESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AlertTypeEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(AlertTypeEnum = exports.AlertTypeEnum || (exports.AlertTypeEnum = {}));
(function (AlertTypeEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    AlertTypeEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlertTypeEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlertTypeEnum = exports.AlertTypeEnum || (exports.AlertTypeEnum = {}));
