// oci-databasemanagement/lib/model/dbm-resource.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbmResource = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var DbmResource;
(function (DbmResource) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = DbmResource.LifecycleState || (DbmResource.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "resourceType" in obj && obj.resourceType) {
            switch (obj.resourceType) {
                case "STORAGE_CONNECTOR":
                    return model.ExternalExadataStorageConnector.getJsonObj(jsonObj, true);
                case "STORAGE_GRID_SUMMARY":
                    return model.ExternalExadataStorageGridSummary.getJsonObj(jsonObj, true);
                case "STORAGE_SERVER":
                    return model.ExternalExadataStorageServer.getJsonObj(jsonObj, true);
                case "INFRASTRUCTURE":
                    return model.ExternalExadataInfrastructure.getJsonObj(jsonObj, true);
                case "STORAGE_GRID":
                    return model.ExternalExadataStorageGrid.getJsonObj(jsonObj, true);
                case "INFRASTRUCTURE_SUMMARY":
                    return model.ExternalExadataInfrastructureSummary.getJsonObj(jsonObj, true);
                case "DATABASE_SYSTEM_SUMMARY":
                    return model.ExternalExadataDatabaseSystemSummary.getJsonObj(jsonObj, true);
                case "STORAGE_CONNECTOR_SUMMARY":
                    return model.ExternalExadataStorageConnectorSummary.getJsonObj(jsonObj, true);
                case "STORAGE_SERVER_SUMMARY":
                    return model.ExternalExadataStorageServerSummary.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.resourceType}`);
            }
        }
        return jsonObj;
    }
    DbmResource.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "resourceType" in obj && obj.resourceType) {
            switch (obj.resourceType) {
                case "STORAGE_CONNECTOR":
                    return model.ExternalExadataStorageConnector.getDeserializedJsonObj(jsonObj, true);
                case "STORAGE_GRID_SUMMARY":
                    return model.ExternalExadataStorageGridSummary.getDeserializedJsonObj(jsonObj, true);
                case "STORAGE_SERVER":
                    return model.ExternalExadataStorageServer.getDeserializedJsonObj(jsonObj, true);
                case "INFRASTRUCTURE":
                    return model.ExternalExadataInfrastructure.getDeserializedJsonObj(jsonObj, true);
                case "STORAGE_GRID":
                    return model.ExternalExadataStorageGrid.getDeserializedJsonObj(jsonObj, true);
                case "INFRASTRUCTURE_SUMMARY":
                    return model.ExternalExadataInfrastructureSummary.getDeserializedJsonObj(jsonObj, true);
                case "DATABASE_SYSTEM_SUMMARY":
                    return model.ExternalExadataDatabaseSystemSummary.getDeserializedJsonObj(jsonObj, true);
                case "STORAGE_CONNECTOR_SUMMARY":
                    return model.ExternalExadataStorageConnectorSummary.getDeserializedJsonObj(jsonObj, true);
                case "STORAGE_SERVER_SUMMARY":
                    return model.ExternalExadataStorageServerSummary.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.resourceType}`);
            }
        }
        return jsonObj;
    }
    DbmResource.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbmResource = exports.DbmResource || (exports.DbmResource = {}));
