// oci-databasemanagement/lib/model/external-asm-disk-group-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalAsmDiskGroupSummary = void 0;
var ExternalAsmDiskGroupSummary;
(function (ExternalAsmDiskGroupSummary) {
    let RedundancyType;
    (function (RedundancyType) {
        RedundancyType["Extend"] = "EXTEND";
        RedundancyType["Extern"] = "EXTERN";
        RedundancyType["Flex"] = "FLEX";
        RedundancyType["High"] = "HIGH";
        RedundancyType["Normal"] = "NORMAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RedundancyType["UnknownValue"] = "UNKNOWN_VALUE";
    })(RedundancyType = ExternalAsmDiskGroupSummary.RedundancyType || (ExternalAsmDiskGroupSummary.RedundancyType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExternalAsmDiskGroupSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExternalAsmDiskGroupSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalAsmDiskGroupSummary = exports.ExternalAsmDiskGroupSummary || (exports.ExternalAsmDiskGroupSummary = {}));
