// oci-databasemanagement/lib/model/external-db-system-component-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalDbSystemComponentType = void 0;
/**
 * The external DB system component type.
 **/
var ExternalDbSystemComponentType;
(function (ExternalDbSystemComponentType) {
    ExternalDbSystemComponentType["Asm"] = "ASM";
    ExternalDbSystemComponentType["AsmInstance"] = "ASM_INSTANCE";
    ExternalDbSystemComponentType["Cluster"] = "CLUSTER";
    ExternalDbSystemComponentType["ClusterInstance"] = "CLUSTER_INSTANCE";
    ExternalDbSystemComponentType["Database"] = "DATABASE";
    ExternalDbSystemComponentType["DatabaseInstance"] = "DATABASE_INSTANCE";
    ExternalDbSystemComponentType["DatabaseHome"] = "DATABASE_HOME";
    ExternalDbSystemComponentType["DatabaseNode"] = "DATABASE_NODE";
    ExternalDbSystemComponentType["Dbsystem"] = "DBSYSTEM";
    ExternalDbSystemComponentType["Listener"] = "LISTENER";
    ExternalDbSystemComponentType["PluggableDatabase"] = "PLUGGABLE_DATABASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ExternalDbSystemComponentType["UnknownValue"] = "UNKNOWN_VALUE";
})(ExternalDbSystemComponentType = exports.ExternalDbSystemComponentType || (exports.ExternalDbSystemComponentType = {}));
(function (ExternalDbSystemComponentType) {
    function getJsonObj(obj) {
        return obj;
    }
    ExternalDbSystemComponentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExternalDbSystemComponentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalDbSystemComponentType = exports.ExternalDbSystemComponentType || (exports.ExternalDbSystemComponentType = {}));
