// oci-databasemanagement/lib/model/external-listener.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalListener = void 0;
const model = __importStar(require("../model"));
var ExternalListener;
(function (ExternalListener) {
    let ListenerType;
    (function (ListenerType) {
        ListenerType["Asm"] = "ASM";
        ListenerType["Local"] = "LOCAL";
        ListenerType["Scan"] = "SCAN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ListenerType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ListenerType = ExternalListener.ListenerType || (ExternalListener.ListenerType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["NotConnected"] = "NOT_CONNECTED";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ExternalListener.LifecycleState || (ExternalListener.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "endpoints": obj.endpoints
                ? obj.endpoints.map(item => {
                    return model.ExternalListenerEndpoint.getJsonObj(item);
                })
                : undefined,
            "servicedDatabases": obj.servicedDatabases
                ? obj.servicedDatabases.map(item => {
                    return model.ExternalListenerServicedDatabase.getJsonObj(item);
                })
                : undefined,
            "servicedAsms": obj.servicedAsms
                ? obj.servicedAsms.map(item => {
                    return model.ExternalServicedAsm.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ExternalListener.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "endpoints": obj.endpoints
                ? obj.endpoints.map(item => {
                    return model.ExternalListenerEndpoint.getDeserializedJsonObj(item);
                })
                : undefined,
            "servicedDatabases": obj.servicedDatabases
                ? obj.servicedDatabases.map(item => {
                    return model.ExternalListenerServicedDatabase.getDeserializedJsonObj(item);
                })
                : undefined,
            "servicedAsms": obj.servicedAsms
                ? obj.servicedAsms.map(item => {
                    return model.ExternalServicedAsm.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ExternalListener.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExternalListener = exports.ExternalListener || (exports.ExternalListener = {}));
