// oci-databasemanagement/lib/model/object-privilege-summary.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectPrivilegeSummary = void 0;
var ObjectPrivilegeSummary;
(function (ObjectPrivilegeSummary) {
    let Hierarchy;
    (function (Hierarchy) {
        Hierarchy["Yes"] = "YES";
        Hierarchy["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Hierarchy["UnknownValue"] = "UNKNOWN_VALUE";
    })(Hierarchy = ObjectPrivilegeSummary.Hierarchy || (ObjectPrivilegeSummary.Hierarchy = {}));
    let GrantOption;
    (function (GrantOption) {
        GrantOption["Yes"] = "YES";
        GrantOption["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        GrantOption["UnknownValue"] = "UNKNOWN_VALUE";
    })(GrantOption = ObjectPrivilegeSummary.GrantOption || (ObjectPrivilegeSummary.GrantOption = {}));
    let Common;
    (function (Common) {
        Common["Yes"] = "YES";
        Common["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Common["UnknownValue"] = "UNKNOWN_VALUE";
    })(Common = ObjectPrivilegeSummary.Common || (ObjectPrivilegeSummary.Common = {}));
    let Inherited;
    (function (Inherited) {
        Inherited["Yes"] = "YES";
        Inherited["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Inherited["UnknownValue"] = "UNKNOWN_VALUE";
    })(Inherited = ObjectPrivilegeSummary.Inherited || (ObjectPrivilegeSummary.Inherited = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ObjectPrivilegeSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ObjectPrivilegeSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ObjectPrivilegeSummary = exports.ObjectPrivilegeSummary || (exports.ObjectPrivilegeSummary = {}));
