// oci-databasemanagement/lib/model/optimizer-statistics-task-filter-types.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptimizerStatisticsTaskFilterTypes = void 0;
/**
 * Indicates which Gather Optimizer Statistics Collection or Optimizer Statistics Advisor tasks should be filtered.
 **/
var OptimizerStatisticsTaskFilterTypes;
(function (OptimizerStatisticsTaskFilterTypes) {
    OptimizerStatisticsTaskFilterTypes["All"] = "ALL";
    OptimizerStatisticsTaskFilterTypes["Manual"] = "MANUAL";
    OptimizerStatisticsTaskFilterTypes["Auto"] = "AUTO";
})(OptimizerStatisticsTaskFilterTypes = exports.OptimizerStatisticsTaskFilterTypes || (exports.OptimizerStatisticsTaskFilterTypes = {}));
(function (OptimizerStatisticsTaskFilterTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    OptimizerStatisticsTaskFilterTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OptimizerStatisticsTaskFilterTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(OptimizerStatisticsTaskFilterTypes = exports.OptimizerStatisticsTaskFilterTypes || (exports.OptimizerStatisticsTaskFilterTypes = {}));
