// oci-databasemanagement/lib/model/sql-tuning-task-status-types.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlTuningTaskStatusTypes = void 0;
/**
 * The status types of a SQL Tuning Advisor task.
 **/
var SqlTuningTaskStatusTypes;
(function (SqlTuningTaskStatusTypes) {
    SqlTuningTaskStatusTypes["Completed"] = "COMPLETED";
    SqlTuningTaskStatusTypes["Initial"] = "INITIAL";
    SqlTuningTaskStatusTypes["Executing"] = "EXECUTING";
    SqlTuningTaskStatusTypes["Interrupted"] = "INTERRUPTED";
    SqlTuningTaskStatusTypes["Error"] = "ERROR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SqlTuningTaskStatusTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(SqlTuningTaskStatusTypes = exports.SqlTuningTaskStatusTypes || (exports.SqlTuningTaskStatusTypes = {}));
(function (SqlTuningTaskStatusTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    SqlTuningTaskStatusTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SqlTuningTaskStatusTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(SqlTuningTaskStatusTypes = exports.SqlTuningTaskStatusTypes || (exports.SqlTuningTaskStatusTypes = {}));
