// oci-databasemanagement/lib/model/start-sql-tuning-task-details.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartSqlTuningTaskDetails = void 0;
const model = __importStar(require("../model"));
var StartSqlTuningTaskDetails;
(function (StartSqlTuningTaskDetails) {
    let Scope;
    (function (Scope) {
        Scope["Limited"] = "LIMITED";
        Scope["Comprehensive"] = "COMPREHENSIVE";
    })(Scope = StartSqlTuningTaskDetails.Scope || (StartSqlTuningTaskDetails.Scope = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "credentialDetails": obj.credentialDetails
                ? model.SqlTuningTaskCredentialDetails.getJsonObj(obj.credentialDetails)
                : undefined,
            "sqlTuningSet": obj.sqlTuningSet
                ? model.SqlTuningSetInput.getJsonObj(obj.sqlTuningSet)
                : undefined,
            "sqlDetails": obj.sqlDetails
                ? obj.sqlDetails.map(item => {
                    return model.SqlTuningTaskSqlDetail.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    StartSqlTuningTaskDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "credentialDetails": obj.credentialDetails
                ? model.SqlTuningTaskCredentialDetails.getDeserializedJsonObj(obj.credentialDetails)
                : undefined,
            "sqlTuningSet": obj.sqlTuningSet
                ? model.SqlTuningSetInput.getDeserializedJsonObj(obj.sqlTuningSet)
                : undefined,
            "sqlDetails": obj.sqlDetails
                ? obj.sqlDetails.map(item => {
                    return model.SqlTuningTaskSqlDetail.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    StartSqlTuningTaskDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(StartSqlTuningTaskDetails = exports.StartSqlTuningTaskDetails || (exports.StartSqlTuningTaskDetails = {}));
