// oci-databasemanagement/lib/model/table-statistics-status-category.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to perform tasks such as obtaining performance and resource usage metrics
for a fleet of Managed Databases or a specific Managed Database, creating Managed Database Groups, and
running a SQL job on a Managed Database or Managed Database Group.

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableStatisticsStatusCategory = void 0;
/**
 * The table statistics collection status, which can be divided into the following categories:
 * - NO_STATS: Table objects with no statistics.
 * - STALE: Table objects with statistics that aren't up-to-date.
 * - NOT_STALE: Table objects with fresh or latest stats.
 *
 **/
var TableStatisticsStatusCategory;
(function (TableStatisticsStatusCategory) {
    TableStatisticsStatusCategory["NoStats"] = "NO_STATS";
    TableStatisticsStatusCategory["Stale"] = "STALE";
    TableStatisticsStatusCategory["NotStale"] = "NOT_STALE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TableStatisticsStatusCategory["UnknownValue"] = "UNKNOWN_VALUE";
})(TableStatisticsStatusCategory = exports.TableStatisticsStatusCategory || (exports.TableStatisticsStatusCategory = {}));
(function (TableStatisticsStatusCategory) {
    function getJsonObj(obj) {
        return obj;
    }
    TableStatisticsStatusCategory.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TableStatisticsStatusCategory.getDeserializedJsonObj = getDeserializedJsonObj;
})(TableStatisticsStatusCategory = exports.TableStatisticsStatusCategory || (exports.TableStatisticsStatusCategory = {}));
