// oci-networkfirewall/lib/model/index.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VaultMappedSecret = exports.UdpApplication = exports.TcpApplication = exports.SslInboundInspectionProfile = exports.SslForwardProxyProfile = exports.SimpleUrlPattern = exports.IcmpApplication = exports.Icmp6Application = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UrlPattern = exports.UpdateNetworkFirewallPolicyDetails = exports.UpdateNetworkFirewallDetails = exports.SortOrder = exports.SecurityRuleMatchCriteria = exports.SecurityRule = exports.OperationType = exports.OperationStatus = exports.NetworkFirewallSummary = exports.NetworkFirewallPolicySummaryCollection = exports.NetworkFirewallPolicySummary = exports.NetworkFirewallPolicy = exports.NetworkFirewallCollection = exports.NetworkFirewall = exports.MappedSecret = exports.LifecycleState = exports.DecryptionRuleMatchCriteria = exports.DecryptionRule = exports.DecryptionProfile = exports.CreateNetworkFirewallPolicyDetails = exports.CreateNetworkFirewallDetails = exports.ChangeNetworkFirewallPolicyCompartmentDetails = exports.ChangeNetworkFirewallCompartmentDetails = exports.Application = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Application = __importStar(require("./application"));
exports.Application = Application.Application;
const ChangeNetworkFirewallCompartmentDetails = __importStar(require("./change-network-firewall-compartment-details"));
exports.ChangeNetworkFirewallCompartmentDetails = ChangeNetworkFirewallCompartmentDetails.ChangeNetworkFirewallCompartmentDetails;
const ChangeNetworkFirewallPolicyCompartmentDetails = __importStar(require("./change-network-firewall-policy-compartment-details"));
exports.ChangeNetworkFirewallPolicyCompartmentDetails = ChangeNetworkFirewallPolicyCompartmentDetails.ChangeNetworkFirewallPolicyCompartmentDetails;
const CreateNetworkFirewallDetails = __importStar(require("./create-network-firewall-details"));
exports.CreateNetworkFirewallDetails = CreateNetworkFirewallDetails.CreateNetworkFirewallDetails;
const CreateNetworkFirewallPolicyDetails = __importStar(require("./create-network-firewall-policy-details"));
exports.CreateNetworkFirewallPolicyDetails = CreateNetworkFirewallPolicyDetails.CreateNetworkFirewallPolicyDetails;
const DecryptionProfile = __importStar(require("./decryption-profile"));
exports.DecryptionProfile = DecryptionProfile.DecryptionProfile;
const DecryptionRule = __importStar(require("./decryption-rule"));
exports.DecryptionRule = DecryptionRule.DecryptionRule;
const DecryptionRuleMatchCriteria = __importStar(require("./decryption-rule-match-criteria"));
exports.DecryptionRuleMatchCriteria = DecryptionRuleMatchCriteria.DecryptionRuleMatchCriteria;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const MappedSecret = __importStar(require("./mapped-secret"));
exports.MappedSecret = MappedSecret.MappedSecret;
const NetworkFirewall = __importStar(require("./network-firewall"));
exports.NetworkFirewall = NetworkFirewall.NetworkFirewall;
const NetworkFirewallCollection = __importStar(require("./network-firewall-collection"));
exports.NetworkFirewallCollection = NetworkFirewallCollection.NetworkFirewallCollection;
const NetworkFirewallPolicy = __importStar(require("./network-firewall-policy"));
exports.NetworkFirewallPolicy = NetworkFirewallPolicy.NetworkFirewallPolicy;
const NetworkFirewallPolicySummary = __importStar(require("./network-firewall-policy-summary"));
exports.NetworkFirewallPolicySummary = NetworkFirewallPolicySummary.NetworkFirewallPolicySummary;
const NetworkFirewallPolicySummaryCollection = __importStar(require("./network-firewall-policy-summary-collection"));
exports.NetworkFirewallPolicySummaryCollection = NetworkFirewallPolicySummaryCollection.NetworkFirewallPolicySummaryCollection;
const NetworkFirewallSummary = __importStar(require("./network-firewall-summary"));
exports.NetworkFirewallSummary = NetworkFirewallSummary.NetworkFirewallSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const SecurityRule = __importStar(require("./security-rule"));
exports.SecurityRule = SecurityRule.SecurityRule;
const SecurityRuleMatchCriteria = __importStar(require("./security-rule-match-criteria"));
exports.SecurityRuleMatchCriteria = SecurityRuleMatchCriteria.SecurityRuleMatchCriteria;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateNetworkFirewallDetails = __importStar(require("./update-network-firewall-details"));
exports.UpdateNetworkFirewallDetails = UpdateNetworkFirewallDetails.UpdateNetworkFirewallDetails;
const UpdateNetworkFirewallPolicyDetails = __importStar(require("./update-network-firewall-policy-details"));
exports.UpdateNetworkFirewallPolicyDetails = UpdateNetworkFirewallPolicyDetails.UpdateNetworkFirewallPolicyDetails;
const UrlPattern = __importStar(require("./url-pattern"));
exports.UrlPattern = UrlPattern.UrlPattern;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const Icmp6Application = __importStar(require("./icmp6-application"));
exports.Icmp6Application = Icmp6Application.Icmp6Application;
const IcmpApplication = __importStar(require("./icmp-application"));
exports.IcmpApplication = IcmpApplication.IcmpApplication;
const SimpleUrlPattern = __importStar(require("./simple-url-pattern"));
exports.SimpleUrlPattern = SimpleUrlPattern.SimpleUrlPattern;
const SslForwardProxyProfile = __importStar(require("./ssl-forward-proxy-profile"));
exports.SslForwardProxyProfile = SslForwardProxyProfile.SslForwardProxyProfile;
const SslInboundInspectionProfile = __importStar(require("./ssl-inbound-inspection-profile"));
exports.SslInboundInspectionProfile = SslInboundInspectionProfile.SslInboundInspectionProfile;
const TcpApplication = __importStar(require("./tcp-application"));
exports.TcpApplication = TcpApplication.TcpApplication;
const UdpApplication = __importStar(require("./udp-application"));
exports.UdpApplication = UdpApplication.UdpApplication;
const VaultMappedSecret = __importStar(require("./vault-mapped-secret"));
exports.VaultMappedSecret = VaultMappedSecret.VaultMappedSecret;
