// oci-networkfirewall/lib/model/operation-type.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateNetworkFirewall"] = "CREATE_NETWORK_FIREWALL";
    OperationType["UpdateNetworkFirewall"] = "UPDATE_NETWORK_FIREWALL";
    OperationType["DeleteNetworkFirewall"] = "DELETE_NETWORK_FIREWALL";
    OperationType["MoveNetworkFirewall"] = "MOVE_NETWORK_FIREWALL";
    OperationType["CreateNetworkFirewallPolicy"] = "CREATE_NETWORK_FIREWALL_POLICY";
    OperationType["UpdateNetworkFirewallPolicy"] = "UPDATE_NETWORK_FIREWALL_POLICY";
    OperationType["DeleteNetworkFirewallPolicy"] = "DELETE_NETWORK_FIREWALL_POLICY";
    OperationType["MoveNetworkFirewallPolicy"] = "MOVE_NETWORK_FIREWALL_POLICY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
