// oci-networkfirewall/lib/model/security-rule.js
"use strict";
/**
 * Network Firewall API
 * Use the Network Firewall API to create network firewalls and configure policies that regulates network traffic in and across VCNs.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityRule = void 0;
const model = __importStar(require("../model"));
var SecurityRule;
(function (SecurityRule) {
    let Action;
    (function (Action) {
        Action["Allow"] = "ALLOW";
        Action["Drop"] = "DROP";
        Action["Reject"] = "REJECT";
        Action["Inspect"] = "INSPECT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Action["UnknownValue"] = "UNKNOWN_VALUE";
    })(Action = SecurityRule.Action || (SecurityRule.Action = {}));
    let Inspection;
    (function (Inspection) {
        Inspection["IntrusionDetection"] = "INTRUSION_DETECTION";
        Inspection["IntrusionPrevention"] = "INTRUSION_PREVENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Inspection["UnknownValue"] = "UNKNOWN_VALUE";
    })(Inspection = SecurityRule.Inspection || (SecurityRule.Inspection = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "condition": obj.condition
                ? model.SecurityRuleMatchCriteria.getJsonObj(obj.condition)
                : undefined
        });
        return jsonObj;
    }
    SecurityRule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "condition": obj.condition
                ? model.SecurityRuleMatchCriteria.getDeserializedJsonObj(obj.condition)
                : undefined
        });
        return jsonObj;
    }
    SecurityRule.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityRule = exports.SecurityRule || (exports.SecurityRule = {}));
