// oci-adm/lib/model/index.js
"use strict";
/**
 * Application Dependency Management API
 * Use the Application Dependency Management API to create knowledge bases and vulnerability audits.  For more information, see [ADM](/Content/application-dependency-management/home.htm).
 * OpenAPI spec version: 20220421
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnknownSourceVulnerabilityAuditSource = exports.OciResourceVulnerabilityAuditSource = exports.ExternalResourceVulnerabilityAuditSource = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VulnerabilityAuditSummary = exports.VulnerabilityAuditSource = exports.VulnerabilityAuditConfiguration = exports.VulnerabilityAuditCollection = exports.VulnerabilityAudit = exports.Vulnerability = exports.UpdateVulnerabilityAuditDetails = exports.UpdateKnowledgeBaseDetails = exports.SortOrder = exports.OperationType = exports.OperationStatus = exports.KnowledgeBaseSummary = exports.KnowledgeBaseCollection = exports.KnowledgeBase = exports.CreateVulnerabilityAuditDetails = exports.CreateKnowledgeBaseDetails = exports.ChangeVulnerabilityAuditCompartmentDetails = exports.ChangeKnowledgeBaseCompartmentDetails = exports.ApplicationDependencyVulnerabilitySummary = exports.ApplicationDependencyVulnerabilityCollection = exports.ApplicationDependency = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ApplicationDependency = __importStar(require("./application-dependency"));
exports.ApplicationDependency = ApplicationDependency.ApplicationDependency;
const ApplicationDependencyVulnerabilityCollection = __importStar(require("./application-dependency-vulnerability-collection"));
exports.ApplicationDependencyVulnerabilityCollection = ApplicationDependencyVulnerabilityCollection.ApplicationDependencyVulnerabilityCollection;
const ApplicationDependencyVulnerabilitySummary = __importStar(require("./application-dependency-vulnerability-summary"));
exports.ApplicationDependencyVulnerabilitySummary = ApplicationDependencyVulnerabilitySummary.ApplicationDependencyVulnerabilitySummary;
const ChangeKnowledgeBaseCompartmentDetails = __importStar(require("./change-knowledge-base-compartment-details"));
exports.ChangeKnowledgeBaseCompartmentDetails = ChangeKnowledgeBaseCompartmentDetails.ChangeKnowledgeBaseCompartmentDetails;
const ChangeVulnerabilityAuditCompartmentDetails = __importStar(require("./change-vulnerability-audit-compartment-details"));
exports.ChangeVulnerabilityAuditCompartmentDetails = ChangeVulnerabilityAuditCompartmentDetails.ChangeVulnerabilityAuditCompartmentDetails;
const CreateKnowledgeBaseDetails = __importStar(require("./create-knowledge-base-details"));
exports.CreateKnowledgeBaseDetails = CreateKnowledgeBaseDetails.CreateKnowledgeBaseDetails;
const CreateVulnerabilityAuditDetails = __importStar(require("./create-vulnerability-audit-details"));
exports.CreateVulnerabilityAuditDetails = CreateVulnerabilityAuditDetails.CreateVulnerabilityAuditDetails;
const KnowledgeBase = __importStar(require("./knowledge-base"));
exports.KnowledgeBase = KnowledgeBase.KnowledgeBase;
const KnowledgeBaseCollection = __importStar(require("./knowledge-base-collection"));
exports.KnowledgeBaseCollection = KnowledgeBaseCollection.KnowledgeBaseCollection;
const KnowledgeBaseSummary = __importStar(require("./knowledge-base-summary"));
exports.KnowledgeBaseSummary = KnowledgeBaseSummary.KnowledgeBaseSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateKnowledgeBaseDetails = __importStar(require("./update-knowledge-base-details"));
exports.UpdateKnowledgeBaseDetails = UpdateKnowledgeBaseDetails.UpdateKnowledgeBaseDetails;
const UpdateVulnerabilityAuditDetails = __importStar(require("./update-vulnerability-audit-details"));
exports.UpdateVulnerabilityAuditDetails = UpdateVulnerabilityAuditDetails.UpdateVulnerabilityAuditDetails;
const Vulnerability = __importStar(require("./vulnerability"));
exports.Vulnerability = Vulnerability.Vulnerability;
const VulnerabilityAudit = __importStar(require("./vulnerability-audit"));
exports.VulnerabilityAudit = VulnerabilityAudit.VulnerabilityAudit;
const VulnerabilityAuditCollection = __importStar(require("./vulnerability-audit-collection"));
exports.VulnerabilityAuditCollection = VulnerabilityAuditCollection.VulnerabilityAuditCollection;
const VulnerabilityAuditConfiguration = __importStar(require("./vulnerability-audit-configuration"));
exports.VulnerabilityAuditConfiguration = VulnerabilityAuditConfiguration.VulnerabilityAuditConfiguration;
const VulnerabilityAuditSource = __importStar(require("./vulnerability-audit-source"));
exports.VulnerabilityAuditSource = VulnerabilityAuditSource.VulnerabilityAuditSource;
const VulnerabilityAuditSummary = __importStar(require("./vulnerability-audit-summary"));
exports.VulnerabilityAuditSummary = VulnerabilityAuditSummary.VulnerabilityAuditSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ExternalResourceVulnerabilityAuditSource = __importStar(require("./external-resource-vulnerability-audit-source"));
exports.ExternalResourceVulnerabilityAuditSource = ExternalResourceVulnerabilityAuditSource.ExternalResourceVulnerabilityAuditSource;
const OciResourceVulnerabilityAuditSource = __importStar(require("./oci-resource-vulnerability-audit-source"));
exports.OciResourceVulnerabilityAuditSource = OciResourceVulnerabilityAuditSource.OciResourceVulnerabilityAuditSource;
const UnknownSourceVulnerabilityAuditSource = __importStar(require("./unknown-source-vulnerability-audit-source"));
exports.UnknownSourceVulnerabilityAuditSource = UnknownSourceVulnerabilityAuditSource.UnknownSourceVulnerabilityAuditSource;
