// oci-functions/lib/request/list-pbf-listings-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPbfListingsRequest = void 0;
var ListPbfListingsRequest;
(function (ListPbfListingsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListPbfListingsRequest.SortOrder || (ListPbfListingsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListPbfListingsRequest.SortBy || (ListPbfListingsRequest.SortBy = {}));
})(ListPbfListingsRequest = exports.ListPbfListingsRequest || (exports.ListPbfListingsRequest = {}));
