// oci-bds/lib/request/index.js
"use strict";
/**
 * Big Data Service API
 * REST API for Oracle Big Data Service. Use this API to build, deploy, and manage fully elastic Big Data Service clusters. Build on Hadoop, Spark and Data Science distributions, which can be fully integrated with existing enterprise data in Oracle Database and Oracle applications.

 * OpenAPI spec version: 20190531
 * Contact: rahul.ra.arora@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListPatchHistoriesRequest = exports.ListBdsMetastoreConfigurationsRequest = exports.ListBdsInstancesRequest = exports.ListBdsApiKeysRequest = exports.ListAutoScalingConfigurationsRequest = void 0;
const ListAutoScalingConfigurationsRequest = __importStar(require("./list-auto-scaling-configurations-request"));
exports.ListAutoScalingConfigurationsRequest = ListAutoScalingConfigurationsRequest.ListAutoScalingConfigurationsRequest;
const ListBdsApiKeysRequest = __importStar(require("./list-bds-api-keys-request"));
exports.ListBdsApiKeysRequest = ListBdsApiKeysRequest.ListBdsApiKeysRequest;
const ListBdsInstancesRequest = __importStar(require("./list-bds-instances-request"));
exports.ListBdsInstancesRequest = ListBdsInstancesRequest.ListBdsInstancesRequest;
const ListBdsMetastoreConfigurationsRequest = __importStar(require("./list-bds-metastore-configurations-request"));
exports.ListBdsMetastoreConfigurationsRequest = ListBdsMetastoreConfigurationsRequest.ListBdsMetastoreConfigurationsRequest;
const ListPatchHistoriesRequest = __importStar(require("./list-patch-histories-request"));
exports.ListPatchHistoriesRequest = ListPatchHistoriesRequest.ListPatchHistoriesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
