// oci-cloudguard/lib/request/list-resource-profiles-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceProfilesRequest = void 0;
var ListResourceProfilesRequest;
(function (ListResourceProfilesRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListResourceProfilesRequest.AccessLevel || (ListResourceProfilesRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["RiskScore"] = "riskScore";
        SortBy["RiskScoreGrowth"] = "riskScoreGrowth";
        SortBy["TimeFirstDetected"] = "timeFirstDetected";
        SortBy["TimeLastDetected"] = "timeLastDetected";
        SortBy["SightingsCount"] = "sightingsCount";
        SortBy["DisplayName"] = "displayName";
        SortBy["Type"] = "type";
    })(SortBy = ListResourceProfilesRequest.SortBy || (ListResourceProfilesRequest.SortBy = {}));
})(ListResourceProfilesRequest = exports.ListResourceProfilesRequest || (exports.ListResourceProfilesRequest = {}));
