// oci-cloudguard/lib/request/list-data-sources-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200131
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDataSourcesRequest = void 0;
var ListDataSourcesRequest;
(function (ListDataSourcesRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListDataSourcesRequest.AccessLevel || (ListDataSourcesRequest.AccessLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListDataSourcesRequest.SortBy || (ListDataSourcesRequest.SortBy = {}));
})(ListDataSourcesRequest = exports.ListDataSourcesRequest || (exports.ListDataSourcesRequest = {}));
