// oci-opsi/lib/request/create-opsi-configuration-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOpsiConfigurationRequest = void 0;
var CreateOpsiConfigurationRequest;
(function (CreateOpsiConfigurationRequest) {
    let OpsiConfigField;
    (function (OpsiConfigField) {
        OpsiConfigField["ConfigItems"] = "configItems";
    })(OpsiConfigField = CreateOpsiConfigurationRequest.OpsiConfigField || (CreateOpsiConfigurationRequest.OpsiConfigField = {}));
    let ConfigItemCustomStatus;
    (function (ConfigItemCustomStatus) {
        ConfigItemCustomStatus["Customized"] = "customized";
        ConfigItemCustomStatus["NonCustomized"] = "nonCustomized";
    })(ConfigItemCustomStatus = CreateOpsiConfigurationRequest.ConfigItemCustomStatus || (CreateOpsiConfigurationRequest.ConfigItemCustomStatus = {}));
    let ConfigItemField;
    (function (ConfigItemField) {
        ConfigItemField["Name"] = "name";
        ConfigItemField["Value"] = "value";
        ConfigItemField["DefaultValue"] = "defaultValue";
        ConfigItemField["Metadata"] = "metadata";
        ConfigItemField["ApplicableContexts"] = "applicableContexts";
    })(ConfigItemField = CreateOpsiConfigurationRequest.ConfigItemField || (CreateOpsiConfigurationRequest.ConfigItemField = {}));
})(CreateOpsiConfigurationRequest = exports.CreateOpsiConfigurationRequest || (exports.CreateOpsiConfigurationRequest = {}));
