// oci-opsi/lib/request/get-awr-database-report-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAwrDatabaseReportRequest = void 0;
var GetAwrDatabaseReportRequest;
(function (GetAwrDatabaseReportRequest) {
    let ReportType;
    (function (ReportType) {
        ReportType["Awr"] = "AWR";
        ReportType["Ash"] = "ASH";
    })(ReportType = GetAwrDatabaseReportRequest.ReportType || (GetAwrDatabaseReportRequest.ReportType = {}));
    let ReportFormat;
    (function (ReportFormat) {
        ReportFormat["Html"] = "HTML";
        ReportFormat["Text"] = "TEXT";
    })(ReportFormat = GetAwrDatabaseReportRequest.ReportFormat || (GetAwrDatabaseReportRequest.ReportFormat = {}));
})(GetAwrDatabaseReportRequest = exports.GetAwrDatabaseReportRequest || (exports.GetAwrDatabaseReportRequest = {}));
