// oci-opsi/lib/request/list-host-configurations-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListHostConfigurationsRequest = void 0;
var ListHostConfigurationsRequest;
(function (ListHostConfigurationsRequest) {
    let PlatformType;
    (function (PlatformType) {
        PlatformType["Linux"] = "LINUX";
        PlatformType["Solaris"] = "SOLARIS";
        PlatformType["Sunos"] = "SUNOS";
        PlatformType["Zlinux"] = "ZLINUX";
    })(PlatformType = ListHostConfigurationsRequest.PlatformType || (ListHostConfigurationsRequest.PlatformType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["HostName"] = "hostName";
        SortBy["PlatformType"] = "platformType";
    })(SortBy = ListHostConfigurationsRequest.SortBy || (ListHostConfigurationsRequest.SortBy = {}));
})(ListHostConfigurationsRequest = exports.ListHostConfigurationsRequest || (exports.ListHostConfigurationsRequest = {}));
