// oci-opsi/lib/request/list-importable-compute-entities-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListImportableComputeEntitiesRequest = void 0;
var ListImportableComputeEntitiesRequest;
(function (ListImportableComputeEntitiesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["ComputeId"] = "computeId";
        SortBy["ComputeDisplayName"] = "computeDisplayName";
        SortBy["PlatformType"] = "platformType";
        SortBy["HostName"] = "hostName";
    })(SortBy = ListImportableComputeEntitiesRequest.SortBy || (ListImportableComputeEntitiesRequest.SortBy = {}));
})(ListImportableComputeEntitiesRequest = exports.ListImportableComputeEntitiesRequest || (exports.ListImportableComputeEntitiesRequest = {}));
