// oci-opsi/lib/request/summarize-exadata-members-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeExadataMembersRequest = void 0;
var SummarizeExadataMembersRequest;
(function (SummarizeExadataMembersRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "name";
        SortBy["DisplayName"] = "displayName";
        SortBy["EntityType"] = "entityType";
    })(SortBy = SummarizeExadataMembersRequest.SortBy || (SummarizeExadataMembersRequest.SortBy = {}));
})(SummarizeExadataMembersRequest = exports.SummarizeExadataMembersRequest || (exports.SummarizeExadataMembersRequest = {}));
