// oci-opsi/lib/request/summarize-configuration-items-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeConfigurationItemsRequest = void 0;
var SummarizeConfigurationItemsRequest;
(function (SummarizeConfigurationItemsRequest) {
    let ConfigItemField;
    (function (ConfigItemField) {
        ConfigItemField["Name"] = "name";
        ConfigItemField["Value"] = "value";
        ConfigItemField["DefaultValue"] = "defaultValue";
        ConfigItemField["ValueSourceConfig"] = "valueSourceConfig";
        ConfigItemField["Metadata"] = "metadata";
        ConfigItemField["ApplicableContexts"] = "applicableContexts";
    })(ConfigItemField = SummarizeConfigurationItemsRequest.ConfigItemField || (SummarizeConfigurationItemsRequest.ConfigItemField = {}));
})(SummarizeConfigurationItemsRequest = exports.SummarizeConfigurationItemsRequest || (exports.SummarizeConfigurationItemsRequest = {}));
