// oci-opsi/lib/request/summarize-exadata-insight-resource-forecast-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeExadataInsightResourceForecastTrendRequest = void 0;
var SummarizeExadataInsightResourceForecastTrendRequest;
(function (SummarizeExadataInsightResourceForecastTrendRequest) {
    let Statistic;
    (function (Statistic) {
        Statistic["Avg"] = "AVG";
        Statistic["Max"] = "MAX";
    })(Statistic = SummarizeExadataInsightResourceForecastTrendRequest.Statistic || (SummarizeExadataInsightResourceForecastTrendRequest.Statistic = {}));
    let ForecastModel;
    (function (ForecastModel) {
        ForecastModel["Linear"] = "LINEAR";
        ForecastModel["MlAuto"] = "ML_AUTO";
        ForecastModel["MlNoAuto"] = "ML_NO_AUTO";
    })(ForecastModel = SummarizeExadataInsightResourceForecastTrendRequest.ForecastModel || (SummarizeExadataInsightResourceForecastTrendRequest.ForecastModel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Id"] = "id";
        SortBy["Name"] = "name";
        SortBy["DaysToReachCapacity"] = "daysToReachCapacity";
    })(SortBy = SummarizeExadataInsightResourceForecastTrendRequest.SortBy || (SummarizeExadataInsightResourceForecastTrendRequest.SortBy = {}));
})(SummarizeExadataInsightResourceForecastTrendRequest = exports.SummarizeExadataInsightResourceForecastTrendRequest || (exports.SummarizeExadataInsightResourceForecastTrendRequest = {}));
