// oci-opsi/lib/request/summarize-addm-db-findings-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAddmDbFindingsRequest = void 0;
var SummarizeAddmDbFindingsRequest;
(function (SummarizeAddmDbFindingsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["ImpactOverallPercent"] = "impactOverallPercent";
        SortBy["ImpactMaxPercent"] = "impactMaxPercent";
        SortBy["ImpactAvgActiveSessions"] = "impactAvgActiveSessions";
        SortBy["FrequencyCount"] = "frequencyCount";
    })(SortBy = SummarizeAddmDbFindingsRequest.SortBy || (SummarizeAddmDbFindingsRequest.SortBy = {}));
})(SummarizeAddmDbFindingsRequest = exports.SummarizeAddmDbFindingsRequest || (exports.SummarizeAddmDbFindingsRequest = {}));
