// oci-opsi/lib/request/summarize-addm-db-parameters-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAddmDbParametersRequest = void 0;
var SummarizeAddmDbParametersRequest;
(function (SummarizeAddmDbParametersRequest) {
    let IsChanged;
    (function (IsChanged) {
        IsChanged["True"] = "true";
        IsChanged["False"] = "false";
    })(IsChanged = SummarizeAddmDbParametersRequest.IsChanged || (SummarizeAddmDbParametersRequest.IsChanged = {}));
    let IsDefault;
    (function (IsDefault) {
        IsDefault["True"] = "true";
        IsDefault["False"] = "false";
    })(IsDefault = SummarizeAddmDbParametersRequest.IsDefault || (SummarizeAddmDbParametersRequest.IsDefault = {}));
    let HasRecommendations;
    (function (HasRecommendations) {
        HasRecommendations["True"] = "true";
        HasRecommendations["False"] = "false";
    })(HasRecommendations = SummarizeAddmDbParametersRequest.HasRecommendations || (SummarizeAddmDbParametersRequest.HasRecommendations = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["IsChanged"] = "isChanged";
        SortBy["Name"] = "name";
    })(SortBy = SummarizeAddmDbParametersRequest.SortBy || (SummarizeAddmDbParametersRequest.SortBy = {}));
})(SummarizeAddmDbParametersRequest = exports.SummarizeAddmDbParametersRequest || (exports.SummarizeAddmDbParametersRequest = {}));
