// oci-opsi/lib/request/summarize-database-insight-resource-capacity-trend-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeDatabaseInsightResourceCapacityTrendRequest = void 0;
var SummarizeDatabaseInsightResourceCapacityTrendRequest;
(function (SummarizeDatabaseInsightResourceCapacityTrendRequest) {
    let DatabaseType;
    (function (DatabaseType) {
        DatabaseType["AdwS"] = "ADW-S";
        DatabaseType["AtpS"] = "ATP-S";
        DatabaseType["AdwD"] = "ADW-D";
        DatabaseType["AtpD"] = "ATP-D";
        DatabaseType["ExternalPdb"] = "EXTERNAL-PDB";
        DatabaseType["ExternalNoncdb"] = "EXTERNAL-NONCDB";
        DatabaseType["ComanagedVmCdb"] = "COMANAGED-VM-CDB";
        DatabaseType["ComanagedVmPdb"] = "COMANAGED-VM-PDB";
        DatabaseType["ComanagedVmNoncdb"] = "COMANAGED-VM-NONCDB";
        DatabaseType["ComanagedBmCdb"] = "COMANAGED-BM-CDB";
        DatabaseType["ComanagedBmPdb"] = "COMANAGED-BM-PDB";
        DatabaseType["ComanagedBmNoncdb"] = "COMANAGED-BM-NONCDB";
        DatabaseType["ComanagedExacsCdb"] = "COMANAGED-EXACS-CDB";
        DatabaseType["ComanagedExacsPdb"] = "COMANAGED-EXACS-PDB";
        DatabaseType["ComanagedExacsNoncdb"] = "COMANAGED-EXACS-NONCDB";
    })(DatabaseType = SummarizeDatabaseInsightResourceCapacityTrendRequest.DatabaseType || (SummarizeDatabaseInsightResourceCapacityTrendRequest.DatabaseType = {}));
    let UtilizationLevel;
    (function (UtilizationLevel) {
        UtilizationLevel["HighUtilization"] = "HIGH_UTILIZATION";
        UtilizationLevel["LowUtilization"] = "LOW_UTILIZATION";
        UtilizationLevel["MediumHighUtilization"] = "MEDIUM_HIGH_UTILIZATION";
        UtilizationLevel["MediumLowUtilization"] = "MEDIUM_LOW_UTILIZATION";
    })(UtilizationLevel = SummarizeDatabaseInsightResourceCapacityTrendRequest.UtilizationLevel || (SummarizeDatabaseInsightResourceCapacityTrendRequest.UtilizationLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["EndTimestamp"] = "endTimestamp";
        SortBy["Capacity"] = "capacity";
        SortBy["BaseCapacity"] = "baseCapacity";
    })(SortBy = SummarizeDatabaseInsightResourceCapacityTrendRequest.SortBy || (SummarizeDatabaseInsightResourceCapacityTrendRequest.SortBy = {}));
})(SummarizeDatabaseInsightResourceCapacityTrendRequest = exports.SummarizeDatabaseInsightResourceCapacityTrendRequest || (exports.SummarizeDatabaseInsightResourceCapacityTrendRequest = {}));
