// oci-opsi/lib/request/summarize-awr-database-parameters-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAwrDatabaseParametersRequest = void 0;
var SummarizeAwrDatabaseParametersRequest;
(function (SummarizeAwrDatabaseParametersRequest) {
    let ValueChanged;
    (function (ValueChanged) {
        ValueChanged["Y"] = "Y";
        ValueChanged["N"] = "N";
    })(ValueChanged = SummarizeAwrDatabaseParametersRequest.ValueChanged || (SummarizeAwrDatabaseParametersRequest.ValueChanged = {}));
    let ValueDefault;
    (function (ValueDefault) {
        ValueDefault["True"] = "TRUE";
        ValueDefault["False"] = "FALSE";
    })(ValueDefault = SummarizeAwrDatabaseParametersRequest.ValueDefault || (SummarizeAwrDatabaseParametersRequest.ValueDefault = {}));
    let ValueModified;
    (function (ValueModified) {
        ValueModified["Modified"] = "MODIFIED";
        ValueModified["SystemMod"] = "SYSTEM_MOD";
        ValueModified["False"] = "FALSE";
    })(ValueModified = SummarizeAwrDatabaseParametersRequest.ValueModified || (SummarizeAwrDatabaseParametersRequest.ValueModified = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["IsChanged"] = "IS_CHANGED";
        SortBy["Name"] = "NAME";
    })(SortBy = SummarizeAwrDatabaseParametersRequest.SortBy || (SummarizeAwrDatabaseParametersRequest.SortBy = {}));
})(SummarizeAwrDatabaseParametersRequest = exports.SummarizeAwrDatabaseParametersRequest || (exports.SummarizeAwrDatabaseParametersRequest = {}));
