// oci-datascience/lib/model/fast-launch-job-config-summary.js
"use strict";
/**
 * Data Science API
 * Use the Data Science API to organize your data science work, access data and computing resources, and build, train, deploy and manage models and model deployments. For more information, see [Data Science](https://docs.oracle.com/iaas/data-science/using/data-science.htm).

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FastLaunchJobConfigSummary = void 0;
var FastLaunchJobConfigSummary;
(function (FastLaunchJobConfigSummary) {
    let ShapeSeries;
    (function (ShapeSeries) {
        ShapeSeries["AmdRome"] = "AMD_ROME";
        ShapeSeries["IntelSkylake"] = "INTEL_SKYLAKE";
        ShapeSeries["NvidiaGpu"] = "NVIDIA_GPU";
        ShapeSeries["Legacy"] = "LEGACY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ShapeSeries["UnknownValue"] = "UNKNOWN_VALUE";
    })(ShapeSeries = FastLaunchJobConfigSummary.ShapeSeries || (FastLaunchJobConfigSummary.ShapeSeries = {}));
    let ManagedEgressSupport;
    (function (ManagedEgressSupport) {
        ManagedEgressSupport["Required"] = "REQUIRED";
        ManagedEgressSupport["Supported"] = "SUPPORTED";
        ManagedEgressSupport["Unsupported"] = "UNSUPPORTED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ManagedEgressSupport["UnknownValue"] = "UNKNOWN_VALUE";
    })(ManagedEgressSupport = FastLaunchJobConfigSummary.ManagedEgressSupport || (FastLaunchJobConfigSummary.ManagedEgressSupport = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FastLaunchJobConfigSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FastLaunchJobConfigSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(FastLaunchJobConfigSummary = exports.FastLaunchJobConfigSummary || (exports.FastLaunchJobConfigSummary = {}));
