// oci-datascience/lib/model/job-lifecycle-state.js
"use strict";
/**
 * Data Science API
 * Use the Data Science API to organize your data science work, access data and computing resources, and build, train, deploy and manage models and model deployments. For more information, see [Data Science](https://docs.oracle.com/iaas/data-science/using/data-science.htm).

 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobLifecycleState = void 0;
/**
 * The status of a job.
 **/
var JobLifecycleState;
(function (JobLifecycleState) {
    JobLifecycleState["Creating"] = "CREATING";
    JobLifecycleState["Active"] = "ACTIVE";
    JobLifecycleState["Deleting"] = "DELETING";
    JobLifecycleState["Failed"] = "FAILED";
    JobLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    JobLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(JobLifecycleState = exports.JobLifecycleState || (exports.JobLifecycleState = {}));
(function (JobLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    JobLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JobLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(JobLifecycleState = exports.JobLifecycleState || (exports.JobLifecycleState = {}));
