// oci-ospgateway/lib/request/list-invoices-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20191001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListInvoicesRequest = void 0;
var ListInvoicesRequest;
(function (ListInvoicesRequest) {
    let Type;
    (function (Type) {
        Type["Hardware"] = "HARDWARE";
        Type["Subscription"] = "SUBSCRIPTION";
        Type["Support"] = "SUPPORT";
        Type["License"] = "LICENSE";
        Type["Education"] = "EDUCATION";
        Type["Consulting"] = "CONSULTING";
        Type["Service"] = "SERVICE";
        Type["Usage"] = "USAGE";
    })(Type = ListInvoicesRequest.Type || (ListInvoicesRequest.Type = {}));
    let Status;
    (function (Status) {
        Status["Open"] = "OPEN";
        Status["PastDue"] = "PAST_DUE";
        Status["PaymentSubmitted"] = "PAYMENT_SUBMITTED";
        Status["Closed"] = "CLOSED";
    })(Status = ListInvoicesRequest.Status || (ListInvoicesRequest.Status = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["InvoiceNo"] = "INVOICE_NO";
        SortBy["RefNo"] = "REF_NO";
        SortBy["Status"] = "STATUS";
        SortBy["Type"] = "TYPE";
        SortBy["InvoiceDate"] = "INVOICE_DATE";
        SortBy["DueDate"] = "DUE_DATE";
        SortBy["PaymRef"] = "PAYM_REF";
        SortBy["TotalAmount"] = "TOTAL_AMOUNT";
        SortBy["BalanceDue"] = "BALANCE_DUE";
    })(SortBy = ListInvoicesRequest.SortBy || (ListInvoicesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListInvoicesRequest.SortOrder || (ListInvoicesRequest.SortOrder = {}));
})(ListInvoicesRequest = exports.ListInvoicesRequest || (exports.ListInvoicesRequest = {}));
