// oci-ospgateway/lib/request/list-subscriptions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20191001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSubscriptionsRequest = void 0;
var ListSubscriptionsRequest;
(function (ListSubscriptionsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["InvoiceNo"] = "INVOICE_NO";
        SortBy["RefNo"] = "REF_NO";
        SortBy["Status"] = "STATUS";
        SortBy["Type"] = "TYPE";
        SortBy["InvoiceDate"] = "INVOICE_DATE";
        SortBy["DueDate"] = "DUE_DATE";
        SortBy["PaymRef"] = "PAYM_REF";
        SortBy["TotalAmount"] = "TOTAL_AMOUNT";
        SortBy["BalanceDue"] = "BALANCE_DUE";
    })(SortBy = ListSubscriptionsRequest.SortBy || (ListSubscriptionsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSubscriptionsRequest.SortOrder || (ListSubscriptionsRequest.SortOrder = {}));
})(ListSubscriptionsRequest = exports.ListSubscriptionsRequest || (exports.ListSubscriptionsRequest = {}));
