// oci-apigateway/lib/model/index.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntitlementTarget = exports.EntitlementSummary = exports.Entitlement = exports.DynamicSelectionKey = exports.DynamicRoutingTypeRoutingBackend = exports.DynamicAuthenticationPolicy = exports.DeploymentSummary = exports.DeploymentCollection = exports.Deployment = exports.CreateUsagePlanDetails = exports.CreateSubscriberDetails = exports.CreateSdkDetails = exports.CreateGatewayDetails = exports.CreateDeploymentDetails = exports.CreateCertificateDetails = exports.CreateApiDetails = exports.CorsPolicy = exports.ContentValidation = exports.ClientSummary = exports.ClientAppDetails = exports.Client = exports.ChangeUsagePlanCompartmentDetails = exports.ChangeSubscriberCompartmentDetails = exports.ChangeGatewayCompartmentDetails = exports.ChangeDeploymentCompartmentDetails = exports.ChangeCertificateCompartmentDetails = exports.ChangeApiCompartmentDetails = exports.CertificateSummary = exports.CertificateCollection = exports.Certificate = exports.CaBundle = exports.BodyValidationRequestPolicy = exports.AuthenticationServerPolicy = exports.AuthenticationPolicy = exports.ApiValidations = exports.ApiValidationResult = exports.ApiValidationDetails = exports.ApiValidationDetail = exports.ApiSummary = exports.ApiSpecificationRouteResponsePolicies = exports.ApiSpecificationRouteRequestPolicies = exports.ApiSpecificationRouteBackend = exports.ApiSpecificationRoute = exports.ApiSpecificationRequestPolicies = exports.ApiSpecificationLoggingPolicies = exports.ApiSpecification = exports.ApiCollection = exports.Api = exports.AdditionalValidationPolicy = exports.AccessLogPolicy = void 0;
exports.SubscriberSummary = exports.SubscriberCollection = exports.Subscriber = exports.StaticPublicKey = exports.SourceUriDetails = exports.SetQueryParameterPolicyItem = exports.SetQueryParameterPolicy = exports.SetHeaderPolicyItem = exports.SetHeaderPolicy = exports.SelectionSourcePolicy = exports.SdkSummary = exports.SdkLanguageTypes = exports.SdkLanguageTypeSummary = exports.SdkLanguageTypeCollection = exports.SdkLanguageOptionalParametersAllowedValue = exports.SdkLanguageOptionalParameters = exports.SdkCollection = exports.Sdk = exports.RouteAuthorizationPolicy = exports.ResponseCacheStorePolicy = exports.ResponseCacheRespServer = exports.ResponseCacheLookupPolicy = exports.ResponseCacheDetails = exports.RequestParameterValidation = exports.RenameQueryParameterPolicyItem = exports.RenameQueryParameterPolicy = exports.RenameHeaderPolicyItem = exports.RenameHeaderPolicy = exports.RateLimitingPolicy = exports.RateLimit = exports.Quota = exports.QueryParameterValidationRequestPolicy = exports.QueryParameterValidationItem = exports.QueryParameterTransformationPolicy = exports.PublicKeySet = exports.MutualTlsDetails = exports.JsonWebTokenClaim = exports.IpAddress = exports.HeaderValidationRequestPolicy = exports.HeaderValidationItem = exports.HeaderTransformationPolicy = exports.HeaderFieldSpecification = exports.GatewaySummary = exports.GatewayCollection = exports.Gateway = exports.FilterQueryParameterPolicyItem = exports.FilterQueryParameterPolicy = exports.FilterHeaderPolicyItem = exports.FilterHeaderPolicy = exports.ExecutionLogPolicy = void 0;
exports.TokenAuthenticationPolicy = exports.StockResponseBackend = exports.StaticPublicKeySet = exports.SingleSelectionSourcePolicy = exports.SimpleLookupPolicy = exports.RemoteJsonWebKeySet = exports.PemEncodedPublicKey = exports.OracleFunctionBackend = exports.OAuth2ResponseValidationFailurePolicy = exports.OAuth2LogoutBackend = exports.NoContentValidation = exports.NoCache = exports.ModifyResponseValidationFailurePolicy = exports.JwtAuthenticationPolicy = exports.JsonWebKey = exports.HTTPBackend = exports.FixedTTLResponseCacheStorePolicy = exports.ExternalRespCache = exports.DynamicRoutingBackend = exports.DiscoveryUriSourceUriDetails = exports.CustomClientAppDetails = exports.CustomAuthenticationPolicy = exports.CertificatesCertificateAuthority = exports.CertificatesCaBundle = exports.AuthenticationOnlyRouteAuthorizationPolicy = exports.AnyOfSelectionKey = exports.AnyOfRouteAuthorizationPolicy = exports.AnonymousRouteAuthorizationPolicy = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogCollection = exports.WorkRequestLog = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.ValidationRequestPolicy = exports.ValidationFailurePolicy = exports.UsagePlansPolicy = exports.UsagePlanSummary = exports.UsagePlanCollection = exports.UsagePlan = exports.UpdateUsagePlanDetails = exports.UpdateSubscriberDetails = exports.UpdateSdkDetails = exports.UpdateGatewayDetails = exports.UpdateDeploymentDetails = exports.UpdateCertificateDetails = exports.UpdateApiDetails = exports.TokenAuthenticationValidationPolicy = void 0;
exports.WildcardSelectionKey = exports.ValidationBlockSourceUriDetails = exports.ValidationBlockClientAppDetails = exports.TokenAuthenticationStaticKeysValidationPolicy = exports.TokenAuthenticationRemoteJWKSValidationPolicy = exports.TokenAuthenticationRemoteDiscoveryValidationPolicy = void 0;
const AccessLogPolicy = __importStar(require("./access-log-policy"));
exports.AccessLogPolicy = AccessLogPolicy.AccessLogPolicy;
const AdditionalValidationPolicy = __importStar(require("./additional-validation-policy"));
exports.AdditionalValidationPolicy = AdditionalValidationPolicy.AdditionalValidationPolicy;
const Api = __importStar(require("./api"));
exports.Api = Api.Api;
const ApiCollection = __importStar(require("./api-collection"));
exports.ApiCollection = ApiCollection.ApiCollection;
const ApiSpecification = __importStar(require("./api-specification"));
exports.ApiSpecification = ApiSpecification.ApiSpecification;
const ApiSpecificationLoggingPolicies = __importStar(require("./api-specification-logging-policies"));
exports.ApiSpecificationLoggingPolicies = ApiSpecificationLoggingPolicies.ApiSpecificationLoggingPolicies;
const ApiSpecificationRequestPolicies = __importStar(require("./api-specification-request-policies"));
exports.ApiSpecificationRequestPolicies = ApiSpecificationRequestPolicies.ApiSpecificationRequestPolicies;
const ApiSpecificationRoute = __importStar(require("./api-specification-route"));
exports.ApiSpecificationRoute = ApiSpecificationRoute.ApiSpecificationRoute;
const ApiSpecificationRouteBackend = __importStar(require("./api-specification-route-backend"));
exports.ApiSpecificationRouteBackend = ApiSpecificationRouteBackend.ApiSpecificationRouteBackend;
const ApiSpecificationRouteRequestPolicies = __importStar(require("./api-specification-route-request-policies"));
exports.ApiSpecificationRouteRequestPolicies = ApiSpecificationRouteRequestPolicies.ApiSpecificationRouteRequestPolicies;
const ApiSpecificationRouteResponsePolicies = __importStar(require("./api-specification-route-response-policies"));
exports.ApiSpecificationRouteResponsePolicies = ApiSpecificationRouteResponsePolicies.ApiSpecificationRouteResponsePolicies;
const ApiSummary = __importStar(require("./api-summary"));
exports.ApiSummary = ApiSummary.ApiSummary;
const ApiValidationDetail = __importStar(require("./api-validation-detail"));
exports.ApiValidationDetail = ApiValidationDetail.ApiValidationDetail;
const ApiValidationDetails = __importStar(require("./api-validation-details"));
exports.ApiValidationDetails = ApiValidationDetails.ApiValidationDetails;
const ApiValidationResult = __importStar(require("./api-validation-result"));
exports.ApiValidationResult = ApiValidationResult.ApiValidationResult;
const ApiValidations = __importStar(require("./api-validations"));
exports.ApiValidations = ApiValidations.ApiValidations;
const AuthenticationPolicy = __importStar(require("./authentication-policy"));
exports.AuthenticationPolicy = AuthenticationPolicy.AuthenticationPolicy;
const AuthenticationServerPolicy = __importStar(require("./authentication-server-policy"));
exports.AuthenticationServerPolicy = AuthenticationServerPolicy.AuthenticationServerPolicy;
const BodyValidationRequestPolicy = __importStar(require("./body-validation-request-policy"));
exports.BodyValidationRequestPolicy = BodyValidationRequestPolicy.BodyValidationRequestPolicy;
const CaBundle = __importStar(require("./ca-bundle"));
exports.CaBundle = CaBundle.CaBundle;
const Certificate = __importStar(require("./certificate"));
exports.Certificate = Certificate.Certificate;
const CertificateCollection = __importStar(require("./certificate-collection"));
exports.CertificateCollection = CertificateCollection.CertificateCollection;
const CertificateSummary = __importStar(require("./certificate-summary"));
exports.CertificateSummary = CertificateSummary.CertificateSummary;
const ChangeApiCompartmentDetails = __importStar(require("./change-api-compartment-details"));
exports.ChangeApiCompartmentDetails = ChangeApiCompartmentDetails.ChangeApiCompartmentDetails;
const ChangeCertificateCompartmentDetails = __importStar(require("./change-certificate-compartment-details"));
exports.ChangeCertificateCompartmentDetails = ChangeCertificateCompartmentDetails.ChangeCertificateCompartmentDetails;
const ChangeDeploymentCompartmentDetails = __importStar(require("./change-deployment-compartment-details"));
exports.ChangeDeploymentCompartmentDetails = ChangeDeploymentCompartmentDetails.ChangeDeploymentCompartmentDetails;
const ChangeGatewayCompartmentDetails = __importStar(require("./change-gateway-compartment-details"));
exports.ChangeGatewayCompartmentDetails = ChangeGatewayCompartmentDetails.ChangeGatewayCompartmentDetails;
const ChangeSubscriberCompartmentDetails = __importStar(require("./change-subscriber-compartment-details"));
exports.ChangeSubscriberCompartmentDetails = ChangeSubscriberCompartmentDetails.ChangeSubscriberCompartmentDetails;
const ChangeUsagePlanCompartmentDetails = __importStar(require("./change-usage-plan-compartment-details"));
exports.ChangeUsagePlanCompartmentDetails = ChangeUsagePlanCompartmentDetails.ChangeUsagePlanCompartmentDetails;
const Client = __importStar(require("./client"));
exports.Client = Client.Client;
const ClientAppDetails = __importStar(require("./client-app-details"));
exports.ClientAppDetails = ClientAppDetails.ClientAppDetails;
const ClientSummary = __importStar(require("./client-summary"));
exports.ClientSummary = ClientSummary.ClientSummary;
const ContentValidation = __importStar(require("./content-validation"));
exports.ContentValidation = ContentValidation.ContentValidation;
const CorsPolicy = __importStar(require("./cors-policy"));
exports.CorsPolicy = CorsPolicy.CorsPolicy;
const CreateApiDetails = __importStar(require("./create-api-details"));
exports.CreateApiDetails = CreateApiDetails.CreateApiDetails;
const CreateCertificateDetails = __importStar(require("./create-certificate-details"));
exports.CreateCertificateDetails = CreateCertificateDetails.CreateCertificateDetails;
const CreateDeploymentDetails = __importStar(require("./create-deployment-details"));
exports.CreateDeploymentDetails = CreateDeploymentDetails.CreateDeploymentDetails;
const CreateGatewayDetails = __importStar(require("./create-gateway-details"));
exports.CreateGatewayDetails = CreateGatewayDetails.CreateGatewayDetails;
const CreateSdkDetails = __importStar(require("./create-sdk-details"));
exports.CreateSdkDetails = CreateSdkDetails.CreateSdkDetails;
const CreateSubscriberDetails = __importStar(require("./create-subscriber-details"));
exports.CreateSubscriberDetails = CreateSubscriberDetails.CreateSubscriberDetails;
const CreateUsagePlanDetails = __importStar(require("./create-usage-plan-details"));
exports.CreateUsagePlanDetails = CreateUsagePlanDetails.CreateUsagePlanDetails;
const Deployment = __importStar(require("./deployment"));
exports.Deployment = Deployment.Deployment;
const DeploymentCollection = __importStar(require("./deployment-collection"));
exports.DeploymentCollection = DeploymentCollection.DeploymentCollection;
const DeploymentSummary = __importStar(require("./deployment-summary"));
exports.DeploymentSummary = DeploymentSummary.DeploymentSummary;
const DynamicAuthenticationPolicy = __importStar(require("./dynamic-authentication-policy"));
exports.DynamicAuthenticationPolicy = DynamicAuthenticationPolicy.DynamicAuthenticationPolicy;
const DynamicRoutingTypeRoutingBackend = __importStar(require("./dynamic-routing-type-routing-backend"));
exports.DynamicRoutingTypeRoutingBackend = DynamicRoutingTypeRoutingBackend.DynamicRoutingTypeRoutingBackend;
const DynamicSelectionKey = __importStar(require("./dynamic-selection-key"));
exports.DynamicSelectionKey = DynamicSelectionKey.DynamicSelectionKey;
const Entitlement = __importStar(require("./entitlement"));
exports.Entitlement = Entitlement.Entitlement;
const EntitlementSummary = __importStar(require("./entitlement-summary"));
exports.EntitlementSummary = EntitlementSummary.EntitlementSummary;
const EntitlementTarget = __importStar(require("./entitlement-target"));
exports.EntitlementTarget = EntitlementTarget.EntitlementTarget;
const ExecutionLogPolicy = __importStar(require("./execution-log-policy"));
exports.ExecutionLogPolicy = ExecutionLogPolicy.ExecutionLogPolicy;
const FilterHeaderPolicy = __importStar(require("./filter-header-policy"));
exports.FilterHeaderPolicy = FilterHeaderPolicy.FilterHeaderPolicy;
const FilterHeaderPolicyItem = __importStar(require("./filter-header-policy-item"));
exports.FilterHeaderPolicyItem = FilterHeaderPolicyItem.FilterHeaderPolicyItem;
const FilterQueryParameterPolicy = __importStar(require("./filter-query-parameter-policy"));
exports.FilterQueryParameterPolicy = FilterQueryParameterPolicy.FilterQueryParameterPolicy;
const FilterQueryParameterPolicyItem = __importStar(require("./filter-query-parameter-policy-item"));
exports.FilterQueryParameterPolicyItem = FilterQueryParameterPolicyItem.FilterQueryParameterPolicyItem;
const Gateway = __importStar(require("./gateway"));
exports.Gateway = Gateway.Gateway;
const GatewayCollection = __importStar(require("./gateway-collection"));
exports.GatewayCollection = GatewayCollection.GatewayCollection;
const GatewaySummary = __importStar(require("./gateway-summary"));
exports.GatewaySummary = GatewaySummary.GatewaySummary;
const HeaderFieldSpecification = __importStar(require("./header-field-specification"));
exports.HeaderFieldSpecification = HeaderFieldSpecification.HeaderFieldSpecification;
const HeaderTransformationPolicy = __importStar(require("./header-transformation-policy"));
exports.HeaderTransformationPolicy = HeaderTransformationPolicy.HeaderTransformationPolicy;
const HeaderValidationItem = __importStar(require("./header-validation-item"));
exports.HeaderValidationItem = HeaderValidationItem.HeaderValidationItem;
const HeaderValidationRequestPolicy = __importStar(require("./header-validation-request-policy"));
exports.HeaderValidationRequestPolicy = HeaderValidationRequestPolicy.HeaderValidationRequestPolicy;
const IpAddress = __importStar(require("./ip-address"));
exports.IpAddress = IpAddress.IpAddress;
const JsonWebTokenClaim = __importStar(require("./json-web-token-claim"));
exports.JsonWebTokenClaim = JsonWebTokenClaim.JsonWebTokenClaim;
const MutualTlsDetails = __importStar(require("./mutual-tls-details"));
exports.MutualTlsDetails = MutualTlsDetails.MutualTlsDetails;
const PublicKeySet = __importStar(require("./public-key-set"));
exports.PublicKeySet = PublicKeySet.PublicKeySet;
const QueryParameterTransformationPolicy = __importStar(require("./query-parameter-transformation-policy"));
exports.QueryParameterTransformationPolicy = QueryParameterTransformationPolicy.QueryParameterTransformationPolicy;
const QueryParameterValidationItem = __importStar(require("./query-parameter-validation-item"));
exports.QueryParameterValidationItem = QueryParameterValidationItem.QueryParameterValidationItem;
const QueryParameterValidationRequestPolicy = __importStar(require("./query-parameter-validation-request-policy"));
exports.QueryParameterValidationRequestPolicy = QueryParameterValidationRequestPolicy.QueryParameterValidationRequestPolicy;
const Quota = __importStar(require("./quota"));
exports.Quota = Quota.Quota;
const RateLimit = __importStar(require("./rate-limit"));
exports.RateLimit = RateLimit.RateLimit;
const RateLimitingPolicy = __importStar(require("./rate-limiting-policy"));
exports.RateLimitingPolicy = RateLimitingPolicy.RateLimitingPolicy;
const RenameHeaderPolicy = __importStar(require("./rename-header-policy"));
exports.RenameHeaderPolicy = RenameHeaderPolicy.RenameHeaderPolicy;
const RenameHeaderPolicyItem = __importStar(require("./rename-header-policy-item"));
exports.RenameHeaderPolicyItem = RenameHeaderPolicyItem.RenameHeaderPolicyItem;
const RenameQueryParameterPolicy = __importStar(require("./rename-query-parameter-policy"));
exports.RenameQueryParameterPolicy = RenameQueryParameterPolicy.RenameQueryParameterPolicy;
const RenameQueryParameterPolicyItem = __importStar(require("./rename-query-parameter-policy-item"));
exports.RenameQueryParameterPolicyItem = RenameQueryParameterPolicyItem.RenameQueryParameterPolicyItem;
const RequestParameterValidation = __importStar(require("./request-parameter-validation"));
exports.RequestParameterValidation = RequestParameterValidation.RequestParameterValidation;
const ResponseCacheDetails = __importStar(require("./response-cache-details"));
exports.ResponseCacheDetails = ResponseCacheDetails.ResponseCacheDetails;
const ResponseCacheLookupPolicy = __importStar(require("./response-cache-lookup-policy"));
exports.ResponseCacheLookupPolicy = ResponseCacheLookupPolicy.ResponseCacheLookupPolicy;
const ResponseCacheRespServer = __importStar(require("./response-cache-resp-server"));
exports.ResponseCacheRespServer = ResponseCacheRespServer.ResponseCacheRespServer;
const ResponseCacheStorePolicy = __importStar(require("./response-cache-store-policy"));
exports.ResponseCacheStorePolicy = ResponseCacheStorePolicy.ResponseCacheStorePolicy;
const RouteAuthorizationPolicy = __importStar(require("./route-authorization-policy"));
exports.RouteAuthorizationPolicy = RouteAuthorizationPolicy.RouteAuthorizationPolicy;
const Sdk = __importStar(require("./sdk"));
exports.Sdk = Sdk.Sdk;
const SdkCollection = __importStar(require("./sdk-collection"));
exports.SdkCollection = SdkCollection.SdkCollection;
const SdkLanguageOptionalParameters = __importStar(require("./sdk-language-optional-parameters"));
exports.SdkLanguageOptionalParameters = SdkLanguageOptionalParameters.SdkLanguageOptionalParameters;
const SdkLanguageOptionalParametersAllowedValue = __importStar(require("./sdk-language-optional-parameters-allowed-value"));
exports.SdkLanguageOptionalParametersAllowedValue = SdkLanguageOptionalParametersAllowedValue.SdkLanguageOptionalParametersAllowedValue;
const SdkLanguageTypeCollection = __importStar(require("./sdk-language-type-collection"));
exports.SdkLanguageTypeCollection = SdkLanguageTypeCollection.SdkLanguageTypeCollection;
const SdkLanguageTypeSummary = __importStar(require("./sdk-language-type-summary"));
exports.SdkLanguageTypeSummary = SdkLanguageTypeSummary.SdkLanguageTypeSummary;
const SdkLanguageTypes = __importStar(require("./sdk-language-types"));
exports.SdkLanguageTypes = SdkLanguageTypes.SdkLanguageTypes;
const SdkSummary = __importStar(require("./sdk-summary"));
exports.SdkSummary = SdkSummary.SdkSummary;
const SelectionSourcePolicy = __importStar(require("./selection-source-policy"));
exports.SelectionSourcePolicy = SelectionSourcePolicy.SelectionSourcePolicy;
const SetHeaderPolicy = __importStar(require("./set-header-policy"));
exports.SetHeaderPolicy = SetHeaderPolicy.SetHeaderPolicy;
const SetHeaderPolicyItem = __importStar(require("./set-header-policy-item"));
exports.SetHeaderPolicyItem = SetHeaderPolicyItem.SetHeaderPolicyItem;
const SetQueryParameterPolicy = __importStar(require("./set-query-parameter-policy"));
exports.SetQueryParameterPolicy = SetQueryParameterPolicy.SetQueryParameterPolicy;
const SetQueryParameterPolicyItem = __importStar(require("./set-query-parameter-policy-item"));
exports.SetQueryParameterPolicyItem = SetQueryParameterPolicyItem.SetQueryParameterPolicyItem;
const SourceUriDetails = __importStar(require("./source-uri-details"));
exports.SourceUriDetails = SourceUriDetails.SourceUriDetails;
const StaticPublicKey = __importStar(require("./static-public-key"));
exports.StaticPublicKey = StaticPublicKey.StaticPublicKey;
const Subscriber = __importStar(require("./subscriber"));
exports.Subscriber = Subscriber.Subscriber;
const SubscriberCollection = __importStar(require("./subscriber-collection"));
exports.SubscriberCollection = SubscriberCollection.SubscriberCollection;
const SubscriberSummary = __importStar(require("./subscriber-summary"));
exports.SubscriberSummary = SubscriberSummary.SubscriberSummary;
const TokenAuthenticationValidationPolicy = __importStar(require("./token-authentication-validation-policy"));
exports.TokenAuthenticationValidationPolicy = TokenAuthenticationValidationPolicy.TokenAuthenticationValidationPolicy;
const UpdateApiDetails = __importStar(require("./update-api-details"));
exports.UpdateApiDetails = UpdateApiDetails.UpdateApiDetails;
const UpdateCertificateDetails = __importStar(require("./update-certificate-details"));
exports.UpdateCertificateDetails = UpdateCertificateDetails.UpdateCertificateDetails;
const UpdateDeploymentDetails = __importStar(require("./update-deployment-details"));
exports.UpdateDeploymentDetails = UpdateDeploymentDetails.UpdateDeploymentDetails;
const UpdateGatewayDetails = __importStar(require("./update-gateway-details"));
exports.UpdateGatewayDetails = UpdateGatewayDetails.UpdateGatewayDetails;
const UpdateSdkDetails = __importStar(require("./update-sdk-details"));
exports.UpdateSdkDetails = UpdateSdkDetails.UpdateSdkDetails;
const UpdateSubscriberDetails = __importStar(require("./update-subscriber-details"));
exports.UpdateSubscriberDetails = UpdateSubscriberDetails.UpdateSubscriberDetails;
const UpdateUsagePlanDetails = __importStar(require("./update-usage-plan-details"));
exports.UpdateUsagePlanDetails = UpdateUsagePlanDetails.UpdateUsagePlanDetails;
const UsagePlan = __importStar(require("./usage-plan"));
exports.UsagePlan = UsagePlan.UsagePlan;
const UsagePlanCollection = __importStar(require("./usage-plan-collection"));
exports.UsagePlanCollection = UsagePlanCollection.UsagePlanCollection;
const UsagePlanSummary = __importStar(require("./usage-plan-summary"));
exports.UsagePlanSummary = UsagePlanSummary.UsagePlanSummary;
const UsagePlansPolicy = __importStar(require("./usage-plans-policy"));
exports.UsagePlansPolicy = UsagePlansPolicy.UsagePlansPolicy;
const ValidationFailurePolicy = __importStar(require("./validation-failure-policy"));
exports.ValidationFailurePolicy = ValidationFailurePolicy.ValidationFailurePolicy;
const ValidationRequestPolicy = __importStar(require("./validation-request-policy"));
exports.ValidationRequestPolicy = ValidationRequestPolicy.ValidationRequestPolicy;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestLogCollection = __importStar(require("./work-request-log-collection"));
exports.WorkRequestLogCollection = WorkRequestLogCollection.WorkRequestLogCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const AnonymousRouteAuthorizationPolicy = __importStar(require("./anonymous-route-authorization-policy"));
exports.AnonymousRouteAuthorizationPolicy = AnonymousRouteAuthorizationPolicy.AnonymousRouteAuthorizationPolicy;
const AnyOfRouteAuthorizationPolicy = __importStar(require("./any-of-route-authorization-policy"));
exports.AnyOfRouteAuthorizationPolicy = AnyOfRouteAuthorizationPolicy.AnyOfRouteAuthorizationPolicy;
const AnyOfSelectionKey = __importStar(require("./any-of-selection-key"));
exports.AnyOfSelectionKey = AnyOfSelectionKey.AnyOfSelectionKey;
const AuthenticationOnlyRouteAuthorizationPolicy = __importStar(require("./authentication-only-route-authorization-policy"));
exports.AuthenticationOnlyRouteAuthorizationPolicy = AuthenticationOnlyRouteAuthorizationPolicy.AuthenticationOnlyRouteAuthorizationPolicy;
const CertificatesCaBundle = __importStar(require("./certificates-ca-bundle"));
exports.CertificatesCaBundle = CertificatesCaBundle.CertificatesCaBundle;
const CertificatesCertificateAuthority = __importStar(require("./certificates-certificate-authority"));
exports.CertificatesCertificateAuthority = CertificatesCertificateAuthority.CertificatesCertificateAuthority;
const CustomAuthenticationPolicy = __importStar(require("./custom-authentication-policy"));
exports.CustomAuthenticationPolicy = CustomAuthenticationPolicy.CustomAuthenticationPolicy;
const CustomClientAppDetails = __importStar(require("./custom-client-app-details"));
exports.CustomClientAppDetails = CustomClientAppDetails.CustomClientAppDetails;
const DiscoveryUriSourceUriDetails = __importStar(require("./discovery-uri-source-uri-details"));
exports.DiscoveryUriSourceUriDetails = DiscoveryUriSourceUriDetails.DiscoveryUriSourceUriDetails;
const DynamicRoutingBackend = __importStar(require("./dynamic-routing-backend"));
exports.DynamicRoutingBackend = DynamicRoutingBackend.DynamicRoutingBackend;
const ExternalRespCache = __importStar(require("./external-resp-cache"));
exports.ExternalRespCache = ExternalRespCache.ExternalRespCache;
const FixedTTLResponseCacheStorePolicy = __importStar(require("./fixed-tt-lresponse-cache-store-policy"));
exports.FixedTTLResponseCacheStorePolicy = FixedTTLResponseCacheStorePolicy.FixedTTLResponseCacheStorePolicy;
const HTTPBackend = __importStar(require("./h-tt-pbackend"));
exports.HTTPBackend = HTTPBackend.HTTPBackend;
const JsonWebKey = __importStar(require("./json-web-key"));
exports.JsonWebKey = JsonWebKey.JsonWebKey;
const JwtAuthenticationPolicy = __importStar(require("./jwt-authentication-policy"));
exports.JwtAuthenticationPolicy = JwtAuthenticationPolicy.JwtAuthenticationPolicy;
const ModifyResponseValidationFailurePolicy = __importStar(require("./modify-response-validation-failure-policy"));
exports.ModifyResponseValidationFailurePolicy = ModifyResponseValidationFailurePolicy.ModifyResponseValidationFailurePolicy;
const NoCache = __importStar(require("./no-cache"));
exports.NoCache = NoCache.NoCache;
const NoContentValidation = __importStar(require("./no-content-validation"));
exports.NoContentValidation = NoContentValidation.NoContentValidation;
const OAuth2LogoutBackend = __importStar(require("./o-auth2-logout-backend"));
exports.OAuth2LogoutBackend = OAuth2LogoutBackend.OAuth2LogoutBackend;
const OAuth2ResponseValidationFailurePolicy = __importStar(require("./o-auth2-response-validation-failure-policy"));
exports.OAuth2ResponseValidationFailurePolicy = OAuth2ResponseValidationFailurePolicy.OAuth2ResponseValidationFailurePolicy;
const OracleFunctionBackend = __importStar(require("./oracle-function-backend"));
exports.OracleFunctionBackend = OracleFunctionBackend.OracleFunctionBackend;
const PemEncodedPublicKey = __importStar(require("./pem-encoded-public-key"));
exports.PemEncodedPublicKey = PemEncodedPublicKey.PemEncodedPublicKey;
const RemoteJsonWebKeySet = __importStar(require("./remote-json-web-key-set"));
exports.RemoteJsonWebKeySet = RemoteJsonWebKeySet.RemoteJsonWebKeySet;
const SimpleLookupPolicy = __importStar(require("./simple-lookup-policy"));
exports.SimpleLookupPolicy = SimpleLookupPolicy.SimpleLookupPolicy;
const SingleSelectionSourcePolicy = __importStar(require("./single-selection-source-policy"));
exports.SingleSelectionSourcePolicy = SingleSelectionSourcePolicy.SingleSelectionSourcePolicy;
const StaticPublicKeySet = __importStar(require("./static-public-key-set"));
exports.StaticPublicKeySet = StaticPublicKeySet.StaticPublicKeySet;
const StockResponseBackend = __importStar(require("./stock-response-backend"));
exports.StockResponseBackend = StockResponseBackend.StockResponseBackend;
const TokenAuthenticationPolicy = __importStar(require("./token-authentication-policy"));
exports.TokenAuthenticationPolicy = TokenAuthenticationPolicy.TokenAuthenticationPolicy;
const TokenAuthenticationRemoteDiscoveryValidationPolicy = __importStar(require("./token-authentication-remote-discovery-validation-policy"));
exports.TokenAuthenticationRemoteDiscoveryValidationPolicy = TokenAuthenticationRemoteDiscoveryValidationPolicy.TokenAuthenticationRemoteDiscoveryValidationPolicy;
const TokenAuthenticationRemoteJWKSValidationPolicy = __importStar(require("./token-authentication-remote-jw-ks-validation-policy"));
exports.TokenAuthenticationRemoteJWKSValidationPolicy = TokenAuthenticationRemoteJWKSValidationPolicy.TokenAuthenticationRemoteJWKSValidationPolicy;
const TokenAuthenticationStaticKeysValidationPolicy = __importStar(require("./token-authentication-static-keys-validation-policy"));
exports.TokenAuthenticationStaticKeysValidationPolicy = TokenAuthenticationStaticKeysValidationPolicy.TokenAuthenticationStaticKeysValidationPolicy;
const ValidationBlockClientAppDetails = __importStar(require("./validation-block-client-app-details"));
exports.ValidationBlockClientAppDetails = ValidationBlockClientAppDetails.ValidationBlockClientAppDetails;
const ValidationBlockSourceUriDetails = __importStar(require("./validation-block-source-uri-details"));
exports.ValidationBlockSourceUriDetails = ValidationBlockSourceUriDetails.ValidationBlockSourceUriDetails;
const WildcardSelectionKey = __importStar(require("./wildcard-selection-key"));
exports.WildcardSelectionKey = WildcardSelectionKey.WildcardSelectionKey;
