// oci-datascience/lib/request/list-model-version-sets-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListModelVersionSetsRequest = void 0;
var ListModelVersionSetsRequest;
(function (ListModelVersionSetsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListModelVersionSetsRequest.SortOrder || (ListModelVersionSetsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
        SortBy["LifecycleState"] = "lifecycleState";
    })(SortBy = ListModelVersionSetsRequest.SortBy || (ListModelVersionSetsRequest.SortBy = {}));
})(ListModelVersionSetsRequest = exports.ListModelVersionSetsRequest || (exports.ListModelVersionSetsRequest = {}));
