// oci-managementagent/lib/model/management-agent-plugin-summary.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.cloud.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementAgentPluginSummary = void 0;
var ManagementAgentPluginSummary;
(function (ManagementAgentPluginSummary) {
    let SupportedPlatformTypes;
    (function (SupportedPlatformTypes) {
        SupportedPlatformTypes["Linux"] = "LINUX";
        SupportedPlatformTypes["Windows"] = "WINDOWS";
        SupportedPlatformTypes["Solaris"] = "SOLARIS";
        SupportedPlatformTypes["Macosx"] = "MACOSX";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SupportedPlatformTypes["UnknownValue"] = "UNKNOWN_VALUE";
    })(SupportedPlatformTypes = ManagementAgentPluginSummary.SupportedPlatformTypes || (ManagementAgentPluginSummary.SupportedPlatformTypes = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ManagementAgentPluginSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ManagementAgentPluginSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ManagementAgentPluginSummary = exports.ManagementAgentPluginSummary || (exports.ManagementAgentPluginSummary = {}));
