// oci-managementagent/lib/model/operation-status.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.cloud.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationStatus = void 0;
/**
 * Possible operation status.
 **/
var OperationStatus;
(function (OperationStatus) {
    OperationStatus["Created"] = "CREATED";
    OperationStatus["Accepted"] = "ACCEPTED";
    OperationStatus["InProgress"] = "IN_PROGRESS";
    OperationStatus["Failed"] = "FAILED";
    OperationStatus["Succeeded"] = "SUCCEEDED";
    OperationStatus["Canceling"] = "CANCELING";
    OperationStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
(function (OperationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
