// oci-managementagent/lib/model/package-types.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.cloud.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageTypes = void 0;
/**
 * The agent image installation package type.
 * RPM - Package is in rpm format.
 * ZIP - Package is in zip format.
 *
 **/
var PackageTypes;
(function (PackageTypes) {
    PackageTypes["Rpm"] = "RPM";
    PackageTypes["Zip"] = "ZIP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PackageTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(PackageTypes = exports.PackageTypes || (exports.PackageTypes = {}));
(function (PackageTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    PackageTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PackageTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(PackageTypes = exports.PackageTypes || (exports.PackageTypes = {}));
