// oci-managementagent/lib/model/plugin-status.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.cloud.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PluginStatus = void 0;
/**
 * Status of Management Agent Plugin.
 * RUNNING - The plugin is running.
 * STOPPED - The plugin is stopped.
 * INVALID - The plugin status is not recognized.
 * FAILED - The plugin deployment has failed.
 *
 **/
var PluginStatus;
(function (PluginStatus) {
    PluginStatus["Running"] = "RUNNING";
    PluginStatus["Stopped"] = "STOPPED";
    PluginStatus["Invalid"] = "INVALID";
    PluginStatus["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PluginStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(PluginStatus = exports.PluginStatus || (exports.PluginStatus = {}));
(function (PluginStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    PluginStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PluginStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(PluginStatus = exports.PluginStatus || (exports.PluginStatus = {}));
