// oci-managementagent/lib/model/work-delivery-status.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.cloud.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkDeliveryStatus = void 0;
/**
 * Possible Work Delivery states.
 **/
var WorkDeliveryStatus;
(function (WorkDeliveryStatus) {
    WorkDeliveryStatus["Accepted"] = "ACCEPTED";
    WorkDeliveryStatus["InProgress"] = "IN_PROGRESS";
    WorkDeliveryStatus["Failed"] = "FAILED";
    WorkDeliveryStatus["Succeeded"] = "SUCCEEDED";
    WorkDeliveryStatus["Canceling"] = "CANCELING";
    WorkDeliveryStatus["Canceled"] = "CANCELED";
})(WorkDeliveryStatus = exports.WorkDeliveryStatus || (exports.WorkDeliveryStatus = {}));
(function (WorkDeliveryStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkDeliveryStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkDeliveryStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkDeliveryStatus = exports.WorkDeliveryStatus || (exports.WorkDeliveryStatus = {}));
