// oci-oda/lib/request/index.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPackagesRequest = exports.ListImportedPackagesRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListOdaInstancesRequest = exports.ListOdaInstanceAttachmentsRequest = exports.ListTranslatorsRequest = exports.ListSkillsRequest = exports.ListSkillParametersRequest = exports.ListOdaPrivateEndpointsRequest = exports.ListOdaPrivateEndpointScanProxiesRequest = exports.ListOdaPrivateEndpointAttachmentsRequest = exports.ListDigitalAssistantsRequest = exports.ListDigitalAssistantParametersRequest = exports.ListChannelsRequest = exports.ListAuthenticationProvidersRequest = void 0;
const ListAuthenticationProvidersRequest = __importStar(require("./list-authentication-providers-request"));
exports.ListAuthenticationProvidersRequest = ListAuthenticationProvidersRequest.ListAuthenticationProvidersRequest;
const ListChannelsRequest = __importStar(require("./list-channels-request"));
exports.ListChannelsRequest = ListChannelsRequest.ListChannelsRequest;
const ListDigitalAssistantParametersRequest = __importStar(require("./list-digital-assistant-parameters-request"));
exports.ListDigitalAssistantParametersRequest = ListDigitalAssistantParametersRequest.ListDigitalAssistantParametersRequest;
const ListDigitalAssistantsRequest = __importStar(require("./list-digital-assistants-request"));
exports.ListDigitalAssistantsRequest = ListDigitalAssistantsRequest.ListDigitalAssistantsRequest;
const ListOdaPrivateEndpointAttachmentsRequest = __importStar(require("./list-oda-private-endpoint-attachments-request"));
exports.ListOdaPrivateEndpointAttachmentsRequest = ListOdaPrivateEndpointAttachmentsRequest.ListOdaPrivateEndpointAttachmentsRequest;
const ListOdaPrivateEndpointScanProxiesRequest = __importStar(require("./list-oda-private-endpoint-scan-proxies-request"));
exports.ListOdaPrivateEndpointScanProxiesRequest = ListOdaPrivateEndpointScanProxiesRequest.ListOdaPrivateEndpointScanProxiesRequest;
const ListOdaPrivateEndpointsRequest = __importStar(require("./list-oda-private-endpoints-request"));
exports.ListOdaPrivateEndpointsRequest = ListOdaPrivateEndpointsRequest.ListOdaPrivateEndpointsRequest;
const ListSkillParametersRequest = __importStar(require("./list-skill-parameters-request"));
exports.ListSkillParametersRequest = ListSkillParametersRequest.ListSkillParametersRequest;
const ListSkillsRequest = __importStar(require("./list-skills-request"));
exports.ListSkillsRequest = ListSkillsRequest.ListSkillsRequest;
const ListTranslatorsRequest = __importStar(require("./list-translators-request"));
exports.ListTranslatorsRequest = ListTranslatorsRequest.ListTranslatorsRequest;
const ListOdaInstanceAttachmentsRequest = __importStar(require("./list-oda-instance-attachments-request"));
exports.ListOdaInstanceAttachmentsRequest = ListOdaInstanceAttachmentsRequest.ListOdaInstanceAttachmentsRequest;
const ListOdaInstancesRequest = __importStar(require("./list-oda-instances-request"));
exports.ListOdaInstancesRequest = ListOdaInstancesRequest.ListOdaInstancesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const ListImportedPackagesRequest = __importStar(require("./list-imported-packages-request"));
exports.ListImportedPackagesRequest = ListImportedPackagesRequest.ListImportedPackagesRequest;
const ListPackagesRequest = __importStar(require("./list-packages-request"));
exports.ListPackagesRequest = ListPackagesRequest.ListPackagesRequest;
