// oci-oda/lib/request/list-oda-instance-attachments-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListOdaInstanceAttachmentsRequest = void 0;
var ListOdaInstanceAttachmentsRequest;
(function (ListOdaInstanceAttachmentsRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Attaching"] = "ATTACHING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Detaching"] = "DETACHING";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = ListOdaInstanceAttachmentsRequest.LifecycleState || (ListOdaInstanceAttachmentsRequest.LifecycleState = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListOdaInstanceAttachmentsRequest.SortOrder || (ListOdaInstanceAttachmentsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
    })(SortBy = ListOdaInstanceAttachmentsRequest.SortBy || (ListOdaInstanceAttachmentsRequest.SortBy = {}));
})(ListOdaInstanceAttachmentsRequest = exports.ListOdaInstanceAttachmentsRequest || (exports.ListOdaInstanceAttachmentsRequest = {}));
