// oci-oda/lib/request/list-channels-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListChannelsRequest = void 0;
var ListChannelsRequest;
(function (ListChannelsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListChannelsRequest.SortOrder || (ListChannelsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["Name"] = "name";
    })(SortBy = ListChannelsRequest.SortBy || (ListChannelsRequest.SortBy = {}));
})(ListChannelsRequest = exports.ListChannelsRequest || (exports.ListChannelsRequest = {}));
