// oci-licensemanager/lib/model/index.js
"use strict";
/**
 * License Manager API
 * Use the License Manager API to manage product licenses and license records. For more information, see [License Manager Overview](/iaas/Content/LicenseManager/Concepts/licensemanageroverview.htm).
 * OpenAPI spec version: 20220430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateProductLicenseDetails = exports.UpdateLicenseRecordDetails = exports.UpdateConfigurationDetails = exports.TopUtilizedResourceSummary = exports.TopUtilizedResourceCollection = exports.TopUtilizedProductLicenseSummary = exports.TopUtilizedProductLicenseCollection = exports.Status = exports.ResourceUnit = exports.ProductLicenseSummary = exports.ProductLicenseConsumerSummary = exports.ProductLicenseConsumerCollection = exports.ProductLicenseCollection = exports.ProductLicense = exports.ProductCategory = exports.Product = exports.LifeCycleState = exports.LicenseUnit = exports.LicenseRecordSummary = exports.LicenseRecordCollection = exports.LicenseRecord = exports.LicenseMetric = exports.ImageResponse = exports.ImageDetails = exports.CreateProductLicenseDetails = exports.CreateLicenseRecordDetails = exports.Configuration = exports.BulkUploadValidationErrorInfo = exports.BulkUploadTemplate = exports.BulkUploadResponse = exports.BulkUploadLicenseRecordsDetails = exports.BulkUploadFailedRecordInfo = exports.BulkUploadCellInfo = void 0;
const BulkUploadCellInfo = __importStar(require("./bulk-upload-cell-info"));
exports.BulkUploadCellInfo = BulkUploadCellInfo.BulkUploadCellInfo;
const BulkUploadFailedRecordInfo = __importStar(require("./bulk-upload-failed-record-info"));
exports.BulkUploadFailedRecordInfo = BulkUploadFailedRecordInfo.BulkUploadFailedRecordInfo;
const BulkUploadLicenseRecordsDetails = __importStar(require("./bulk-upload-license-records-details"));
exports.BulkUploadLicenseRecordsDetails = BulkUploadLicenseRecordsDetails.BulkUploadLicenseRecordsDetails;
const BulkUploadResponse = __importStar(require("./bulk-upload-response"));
exports.BulkUploadResponse = BulkUploadResponse.BulkUploadResponse;
const BulkUploadTemplate = __importStar(require("./bulk-upload-template"));
exports.BulkUploadTemplate = BulkUploadTemplate.BulkUploadTemplate;
const BulkUploadValidationErrorInfo = __importStar(require("./bulk-upload-validation-error-info"));
exports.BulkUploadValidationErrorInfo = BulkUploadValidationErrorInfo.BulkUploadValidationErrorInfo;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const CreateLicenseRecordDetails = __importStar(require("./create-license-record-details"));
exports.CreateLicenseRecordDetails = CreateLicenseRecordDetails.CreateLicenseRecordDetails;
const CreateProductLicenseDetails = __importStar(require("./create-product-license-details"));
exports.CreateProductLicenseDetails = CreateProductLicenseDetails.CreateProductLicenseDetails;
const ImageDetails = __importStar(require("./image-details"));
exports.ImageDetails = ImageDetails.ImageDetails;
const ImageResponse = __importStar(require("./image-response"));
exports.ImageResponse = ImageResponse.ImageResponse;
const LicenseMetric = __importStar(require("./license-metric"));
exports.LicenseMetric = LicenseMetric.LicenseMetric;
const LicenseRecord = __importStar(require("./license-record"));
exports.LicenseRecord = LicenseRecord.LicenseRecord;
const LicenseRecordCollection = __importStar(require("./license-record-collection"));
exports.LicenseRecordCollection = LicenseRecordCollection.LicenseRecordCollection;
const LicenseRecordSummary = __importStar(require("./license-record-summary"));
exports.LicenseRecordSummary = LicenseRecordSummary.LicenseRecordSummary;
const LicenseUnit = __importStar(require("./license-unit"));
exports.LicenseUnit = LicenseUnit.LicenseUnit;
const LifeCycleState = __importStar(require("./life-cycle-state"));
exports.LifeCycleState = LifeCycleState.LifeCycleState;
const Product = __importStar(require("./product"));
exports.Product = Product.Product;
const ProductCategory = __importStar(require("./product-category"));
exports.ProductCategory = ProductCategory.ProductCategory;
const ProductLicense = __importStar(require("./product-license"));
exports.ProductLicense = ProductLicense.ProductLicense;
const ProductLicenseCollection = __importStar(require("./product-license-collection"));
exports.ProductLicenseCollection = ProductLicenseCollection.ProductLicenseCollection;
const ProductLicenseConsumerCollection = __importStar(require("./product-license-consumer-collection"));
exports.ProductLicenseConsumerCollection = ProductLicenseConsumerCollection.ProductLicenseConsumerCollection;
const ProductLicenseConsumerSummary = __importStar(require("./product-license-consumer-summary"));
exports.ProductLicenseConsumerSummary = ProductLicenseConsumerSummary.ProductLicenseConsumerSummary;
const ProductLicenseSummary = __importStar(require("./product-license-summary"));
exports.ProductLicenseSummary = ProductLicenseSummary.ProductLicenseSummary;
const ResourceUnit = __importStar(require("./resource-unit"));
exports.ResourceUnit = ResourceUnit.ResourceUnit;
const Status = __importStar(require("./status"));
exports.Status = Status.Status;
const TopUtilizedProductLicenseCollection = __importStar(require("./top-utilized-product-license-collection"));
exports.TopUtilizedProductLicenseCollection = TopUtilizedProductLicenseCollection.TopUtilizedProductLicenseCollection;
const TopUtilizedProductLicenseSummary = __importStar(require("./top-utilized-product-license-summary"));
exports.TopUtilizedProductLicenseSummary = TopUtilizedProductLicenseSummary.TopUtilizedProductLicenseSummary;
const TopUtilizedResourceCollection = __importStar(require("./top-utilized-resource-collection"));
exports.TopUtilizedResourceCollection = TopUtilizedResourceCollection.TopUtilizedResourceCollection;
const TopUtilizedResourceSummary = __importStar(require("./top-utilized-resource-summary"));
exports.TopUtilizedResourceSummary = TopUtilizedResourceSummary.TopUtilizedResourceSummary;
const UpdateConfigurationDetails = __importStar(require("./update-configuration-details"));
exports.UpdateConfigurationDetails = UpdateConfigurationDetails.UpdateConfigurationDetails;
const UpdateLicenseRecordDetails = __importStar(require("./update-license-record-details"));
exports.UpdateLicenseRecordDetails = UpdateLicenseRecordDetails.UpdateLicenseRecordDetails;
const UpdateProductLicenseDetails = __importStar(require("./update-product-license-details"));
exports.UpdateProductLicenseDetails = UpdateProductLicenseDetails.UpdateProductLicenseDetails;
