// oci-licensemanager/lib/model/product-category.js
"use strict";
/**
 * License Manager API
 * Use the License Manager API to manage product licenses and license records. For more information, see [License Manager Overview](/iaas/Content/LicenseManager/Concepts/licensemanageroverview.htm).
 * OpenAPI spec version: 20220430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCategory = void 0;
/**
 * Possible category of oracle product licenses.
 **/
var ProductCategory;
(function (ProductCategory) {
    ProductCategory["Base"] = "BASE";
    ProductCategory["Option"] = "OPTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ProductCategory["UnknownValue"] = "UNKNOWN_VALUE";
})(ProductCategory = exports.ProductCategory || (exports.ProductCategory = {}));
(function (ProductCategory) {
    function getJsonObj(obj) {
        return obj;
    }
    ProductCategory.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ProductCategory.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProductCategory = exports.ProductCategory || (exports.ProductCategory = {}));
