// oci-visualbuilder/lib/model/attachment-details.js
"use strict";
/**
 * Visual Builder API
 * Oracle Visual Builder enables developers to quickly build web and mobile applications. With a visual development environment that makes it easy to connect to Oracle data and third-party REST services, developers can build modern, consumer-grade applications in a fraction of the time it would take in other tools.
The Visual Builder Instance Management API allows users to create and manage a Visual Builder instance.

 * OpenAPI spec version: 20210601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachmentDetails = void 0;
var AttachmentDetails;
(function (AttachmentDetails) {
    let TargetRole;
    (function (TargetRole) {
        TargetRole["Parent"] = "PARENT";
        TargetRole["Child"] = "CHILD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TargetRole["UnknownValue"] = "UNKNOWN_VALUE";
    })(TargetRole = AttachmentDetails.TargetRole || (AttachmentDetails.TargetRole = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttachmentDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttachmentDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttachmentDetails = exports.AttachmentDetails || (exports.AttachmentDetails = {}));
