// oci-visualbuilder/lib/model/index.js
"use strict";
/**
 * Visual Builder API
 * Oracle Visual Builder enables developers to quickly build web and mobile applications. With a visual development environment that makes it easy to connect to Oracle data and third-party REST services, developers can build modern, consumer-grade applications in a fraction of the time it would take in other tools.
The Visual Builder Instance Management API allows users to create and manage a Visual Builder instance.

 * OpenAPI spec version: 20210601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VbInstanceSummaryCollection = exports.VbInstanceSummary = exports.VbInstance = exports.UpdateVbInstanceDetails = exports.UpdateCustomEndpointDetails = exports.RequestSummarizedApplicationsDetails = exports.IdcsInfoDetails = exports.CustomEndpointDetails = exports.CreateVbInstanceDetails = exports.CreateCustomEndpointDetails = exports.ChangeVbInstanceCompartmentDetails = exports.AttachmentDetails = exports.ApplicationSummaryCollection = exports.ApplicationSummary = void 0;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const ApplicationSummaryCollection = __importStar(require("./application-summary-collection"));
exports.ApplicationSummaryCollection = ApplicationSummaryCollection.ApplicationSummaryCollection;
const AttachmentDetails = __importStar(require("./attachment-details"));
exports.AttachmentDetails = AttachmentDetails.AttachmentDetails;
const ChangeVbInstanceCompartmentDetails = __importStar(require("./change-vb-instance-compartment-details"));
exports.ChangeVbInstanceCompartmentDetails = ChangeVbInstanceCompartmentDetails.ChangeVbInstanceCompartmentDetails;
const CreateCustomEndpointDetails = __importStar(require("./create-custom-endpoint-details"));
exports.CreateCustomEndpointDetails = CreateCustomEndpointDetails.CreateCustomEndpointDetails;
const CreateVbInstanceDetails = __importStar(require("./create-vb-instance-details"));
exports.CreateVbInstanceDetails = CreateVbInstanceDetails.CreateVbInstanceDetails;
const CustomEndpointDetails = __importStar(require("./custom-endpoint-details"));
exports.CustomEndpointDetails = CustomEndpointDetails.CustomEndpointDetails;
const IdcsInfoDetails = __importStar(require("./idcs-info-details"));
exports.IdcsInfoDetails = IdcsInfoDetails.IdcsInfoDetails;
const RequestSummarizedApplicationsDetails = __importStar(require("./request-summarized-applications-details"));
exports.RequestSummarizedApplicationsDetails = RequestSummarizedApplicationsDetails.RequestSummarizedApplicationsDetails;
const UpdateCustomEndpointDetails = __importStar(require("./update-custom-endpoint-details"));
exports.UpdateCustomEndpointDetails = UpdateCustomEndpointDetails.UpdateCustomEndpointDetails;
const UpdateVbInstanceDetails = __importStar(require("./update-vb-instance-details"));
exports.UpdateVbInstanceDetails = UpdateVbInstanceDetails.UpdateVbInstanceDetails;
const VbInstance = __importStar(require("./vb-instance"));
exports.VbInstance = VbInstance.VbInstance;
const VbInstanceSummary = __importStar(require("./vb-instance-summary"));
exports.VbInstanceSummary = VbInstanceSummary.VbInstanceSummary;
const VbInstanceSummaryCollection = __importStar(require("./vb-instance-summary-collection"));
exports.VbInstanceSummaryCollection = VbInstanceSummaryCollection.VbInstanceSummaryCollection;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
