// oci-visualbuilder/lib/model/vb-instance-summary.js
"use strict";
/**
 * Visual Builder API
 * Oracle Visual Builder enables developers to quickly build web and mobile applications. With a visual development environment that makes it easy to connect to Oracle data and third-party REST services, developers can build modern, consumer-grade applications in a fraction of the time it would take in other tools.
The Visual Builder Instance Management API allows users to create and manage a Visual Builder instance.

 * OpenAPI spec version: 20210601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VbInstanceSummary = void 0;
const model = __importStar(require("../model"));
var VbInstanceSummary;
(function (VbInstanceSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = VbInstanceSummary.LifecycleState || (VbInstanceSummary.LifecycleState = {}));
    let ConsumptionModel;
    (function (ConsumptionModel) {
        ConsumptionModel["Ucm"] = "UCM";
        ConsumptionModel["Gov"] = "GOV";
        ConsumptionModel["Vb4Saas"] = "VB4SAAS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ConsumptionModel["UnknownValue"] = "UNKNOWN_VALUE";
    })(ConsumptionModel = VbInstanceSummary.ConsumptionModel || (VbInstanceSummary.ConsumptionModel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "customEndpoint": obj.customEndpoint
                ? model.CustomEndpointDetails.getJsonObj(obj.customEndpoint)
                : undefined,
            "alternateCustomEndpoints": obj.alternateCustomEndpoints
                ? obj.alternateCustomEndpoints.map(item => {
                    return model.CustomEndpointDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    VbInstanceSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "customEndpoint": obj.customEndpoint
                ? model.CustomEndpointDetails.getDeserializedJsonObj(obj.customEndpoint)
                : undefined,
            "alternateCustomEndpoints": obj.alternateCustomEndpoints
                ? obj.alternateCustomEndpoints.map(item => {
                    return model.CustomEndpointDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    VbInstanceSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(VbInstanceSummary = exports.VbInstanceSummary || (exports.VbInstanceSummary = {}));
