// oci-dataintegration/lib/request/index.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration API to organize your data integration projects, create data flows, pipelines and tasks, and then publish, schedule, and run tasks that extract, transform, and load data. For more information, see [Data Integration](https://docs.oracle.com/iaas/data-integration/home.htm).
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkspacesRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListUserDefinedFunctionsRequest = exports.ListUserDefinedFunctionValidationsRequest = exports.ListTemplatesRequest = exports.ListTasksRequest = exports.ListTaskValidationsRequest = exports.ListTaskSchedulesRequest = exports.ListTaskRunsRequest = exports.ListTaskRunLogsRequest = exports.ListTaskRunLineagesRequest = exports.ListSchemasRequest = exports.ListSchedulesRequest = exports.ListRuntimePipelinesRequest = exports.ListRuntimeOperatorsRequest = exports.ListReferencesRequest = exports.ListPublishedObjectsRequest = exports.ListProjectsRequest = exports.ListPipelinesRequest = exports.ListPipelineValidationsRequest = exports.ListPatchesRequest = exports.ListPatchChangesRequest = exports.ListFunctionLibrariesRequest = exports.ListFoldersRequest = exports.ListExternalPublicationsRequest = exports.ListExternalPublicationValidationsRequest = exports.ListDisApplicationsRequest = exports.ListDisApplicationTaskRunLineagesRequest = exports.ListDependentObjectsRequest = exports.ListDataFlowsRequest = exports.ListDataFlowValidationsRequest = exports.ListDataEntitiesRequest = exports.ListDataAssetsRequest = exports.ListCopyObjectRequestsRequest = exports.ListConnectionsRequest = exports.ListConnectionValidationsRequest = exports.ListApplicationsRequest = exports.GetProjectRequest = exports.GetFunctionLibraryRequest = exports.GetFolderRequest = void 0;
const GetFolderRequest = __importStar(require("./get-folder-request"));
exports.GetFolderRequest = GetFolderRequest.GetFolderRequest;
const GetFunctionLibraryRequest = __importStar(require("./get-function-library-request"));
exports.GetFunctionLibraryRequest = GetFunctionLibraryRequest.GetFunctionLibraryRequest;
const GetProjectRequest = __importStar(require("./get-project-request"));
exports.GetProjectRequest = GetProjectRequest.GetProjectRequest;
const ListApplicationsRequest = __importStar(require("./list-applications-request"));
exports.ListApplicationsRequest = ListApplicationsRequest.ListApplicationsRequest;
const ListConnectionValidationsRequest = __importStar(require("./list-connection-validations-request"));
exports.ListConnectionValidationsRequest = ListConnectionValidationsRequest.ListConnectionValidationsRequest;
const ListConnectionsRequest = __importStar(require("./list-connections-request"));
exports.ListConnectionsRequest = ListConnectionsRequest.ListConnectionsRequest;
const ListCopyObjectRequestsRequest = __importStar(require("./list-copy-object-requests-request"));
exports.ListCopyObjectRequestsRequest = ListCopyObjectRequestsRequest.ListCopyObjectRequestsRequest;
const ListDataAssetsRequest = __importStar(require("./list-data-assets-request"));
exports.ListDataAssetsRequest = ListDataAssetsRequest.ListDataAssetsRequest;
const ListDataEntitiesRequest = __importStar(require("./list-data-entities-request"));
exports.ListDataEntitiesRequest = ListDataEntitiesRequest.ListDataEntitiesRequest;
const ListDataFlowValidationsRequest = __importStar(require("./list-data-flow-validations-request"));
exports.ListDataFlowValidationsRequest = ListDataFlowValidationsRequest.ListDataFlowValidationsRequest;
const ListDataFlowsRequest = __importStar(require("./list-data-flows-request"));
exports.ListDataFlowsRequest = ListDataFlowsRequest.ListDataFlowsRequest;
const ListDependentObjectsRequest = __importStar(require("./list-dependent-objects-request"));
exports.ListDependentObjectsRequest = ListDependentObjectsRequest.ListDependentObjectsRequest;
const ListDisApplicationTaskRunLineagesRequest = __importStar(require("./list-dis-application-task-run-lineages-request"));
exports.ListDisApplicationTaskRunLineagesRequest = ListDisApplicationTaskRunLineagesRequest.ListDisApplicationTaskRunLineagesRequest;
const ListDisApplicationsRequest = __importStar(require("./list-dis-applications-request"));
exports.ListDisApplicationsRequest = ListDisApplicationsRequest.ListDisApplicationsRequest;
const ListExternalPublicationValidationsRequest = __importStar(require("./list-external-publication-validations-request"));
exports.ListExternalPublicationValidationsRequest = ListExternalPublicationValidationsRequest.ListExternalPublicationValidationsRequest;
const ListExternalPublicationsRequest = __importStar(require("./list-external-publications-request"));
exports.ListExternalPublicationsRequest = ListExternalPublicationsRequest.ListExternalPublicationsRequest;
const ListFoldersRequest = __importStar(require("./list-folders-request"));
exports.ListFoldersRequest = ListFoldersRequest.ListFoldersRequest;
const ListFunctionLibrariesRequest = __importStar(require("./list-function-libraries-request"));
exports.ListFunctionLibrariesRequest = ListFunctionLibrariesRequest.ListFunctionLibrariesRequest;
const ListPatchChangesRequest = __importStar(require("./list-patch-changes-request"));
exports.ListPatchChangesRequest = ListPatchChangesRequest.ListPatchChangesRequest;
const ListPatchesRequest = __importStar(require("./list-patches-request"));
exports.ListPatchesRequest = ListPatchesRequest.ListPatchesRequest;
const ListPipelineValidationsRequest = __importStar(require("./list-pipeline-validations-request"));
exports.ListPipelineValidationsRequest = ListPipelineValidationsRequest.ListPipelineValidationsRequest;
const ListPipelinesRequest = __importStar(require("./list-pipelines-request"));
exports.ListPipelinesRequest = ListPipelinesRequest.ListPipelinesRequest;
const ListProjectsRequest = __importStar(require("./list-projects-request"));
exports.ListProjectsRequest = ListProjectsRequest.ListProjectsRequest;
const ListPublishedObjectsRequest = __importStar(require("./list-published-objects-request"));
exports.ListPublishedObjectsRequest = ListPublishedObjectsRequest.ListPublishedObjectsRequest;
const ListReferencesRequest = __importStar(require("./list-references-request"));
exports.ListReferencesRequest = ListReferencesRequest.ListReferencesRequest;
const ListRuntimeOperatorsRequest = __importStar(require("./list-runtime-operators-request"));
exports.ListRuntimeOperatorsRequest = ListRuntimeOperatorsRequest.ListRuntimeOperatorsRequest;
const ListRuntimePipelinesRequest = __importStar(require("./list-runtime-pipelines-request"));
exports.ListRuntimePipelinesRequest = ListRuntimePipelinesRequest.ListRuntimePipelinesRequest;
const ListSchedulesRequest = __importStar(require("./list-schedules-request"));
exports.ListSchedulesRequest = ListSchedulesRequest.ListSchedulesRequest;
const ListSchemasRequest = __importStar(require("./list-schemas-request"));
exports.ListSchemasRequest = ListSchemasRequest.ListSchemasRequest;
const ListTaskRunLineagesRequest = __importStar(require("./list-task-run-lineages-request"));
exports.ListTaskRunLineagesRequest = ListTaskRunLineagesRequest.ListTaskRunLineagesRequest;
const ListTaskRunLogsRequest = __importStar(require("./list-task-run-logs-request"));
exports.ListTaskRunLogsRequest = ListTaskRunLogsRequest.ListTaskRunLogsRequest;
const ListTaskRunsRequest = __importStar(require("./list-task-runs-request"));
exports.ListTaskRunsRequest = ListTaskRunsRequest.ListTaskRunsRequest;
const ListTaskSchedulesRequest = __importStar(require("./list-task-schedules-request"));
exports.ListTaskSchedulesRequest = ListTaskSchedulesRequest.ListTaskSchedulesRequest;
const ListTaskValidationsRequest = __importStar(require("./list-task-validations-request"));
exports.ListTaskValidationsRequest = ListTaskValidationsRequest.ListTaskValidationsRequest;
const ListTasksRequest = __importStar(require("./list-tasks-request"));
exports.ListTasksRequest = ListTasksRequest.ListTasksRequest;
const ListTemplatesRequest = __importStar(require("./list-templates-request"));
exports.ListTemplatesRequest = ListTemplatesRequest.ListTemplatesRequest;
const ListUserDefinedFunctionValidationsRequest = __importStar(require("./list-user-defined-function-validations-request"));
exports.ListUserDefinedFunctionValidationsRequest = ListUserDefinedFunctionValidationsRequest.ListUserDefinedFunctionValidationsRequest;
const ListUserDefinedFunctionsRequest = __importStar(require("./list-user-defined-functions-request"));
exports.ListUserDefinedFunctionsRequest = ListUserDefinedFunctionsRequest.ListUserDefinedFunctionsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const ListWorkspacesRequest = __importStar(require("./list-workspaces-request"));
exports.ListWorkspacesRequest = ListWorkspacesRequest.ListWorkspacesRequest;
