// oci-dataintegration/lib/request/list-copy-object-requests-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20200430
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCopyObjectRequestsRequest = void 0;
var ListCopyObjectRequestsRequest;
(function (ListCopyObjectRequestsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListCopyObjectRequestsRequest.SortOrder || (ListCopyObjectRequestsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "TIME_CREATED";
        SortBy["DisplayName"] = "DISPLAY_NAME";
        SortBy["TimeUpdated"] = "TIME_UPDATED";
    })(SortBy = ListCopyObjectRequestsRequest.SortBy || (ListCopyObjectRequestsRequest.SortBy = {}));
    let CopyStatus;
    (function (CopyStatus) {
        CopyStatus["InProgress"] = "IN_PROGRESS";
        CopyStatus["Successful"] = "SUCCESSFUL";
        CopyStatus["Queued"] = "QUEUED";
        CopyStatus["Terminating"] = "TERMINATING";
        CopyStatus["Terminated"] = "TERMINATED";
        CopyStatus["Failed"] = "FAILED";
        CopyStatus["All"] = "ALL";
    })(CopyStatus = ListCopyObjectRequestsRequest.CopyStatus || (ListCopyObjectRequestsRequest.CopyStatus = {}));
    let Projection;
    (function (Projection) {
        Projection["Summary"] = "SUMMARY";
        Projection["Details"] = "DETAILS";
    })(Projection = ListCopyObjectRequestsRequest.Projection || (ListCopyObjectRequestsRequest.Projection = {}));
})(ListCopyObjectRequestsRequest = exports.ListCopyObjectRequestsRequest || (exports.ListCopyObjectRequestsRequest = {}));
