// oci-ospgateway/lib/model/credit-card-type.js
"use strict";
/**
 * OSP Gateway API
 * This site describes all the Rest endpoints of OSP Gateway.
 * OpenAPI spec version: 20191001
 * Contact: osp_team_oci_cam_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreditCardType = void 0;
/**
 * Credit card type.
 **/
var CreditCardType;
(function (CreditCardType) {
    CreditCardType["Visa"] = "VISA";
    CreditCardType["Amex"] = "AMEX";
    CreditCardType["Mastercard"] = "MASTERCARD";
    CreditCardType["Discover"] = "DISCOVER";
    CreditCardType["Jcb"] = "JCB";
    CreditCardType["Diner"] = "DINER";
    CreditCardType["Elo"] = "ELO";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CreditCardType["UnknownValue"] = "UNKNOWN_VALUE";
})(CreditCardType = exports.CreditCardType || (exports.CreditCardType = {}));
(function (CreditCardType) {
    function getJsonObj(obj) {
        return obj;
    }
    CreditCardType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CreditCardType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreditCardType = exports.CreditCardType || (exports.CreditCardType = {}));
