// oci-ospgateway/lib/model/index.js
"use strict";
/**
 * OSP Gateway API
 * This site describes all the Rest endpoints of OSP Gateway.
 * OpenAPI spec version: 20191001
 * Contact: osp_team_oci_cam_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaypalPaymentOption = exports.PaypalPaymentDetail = exports.OtherPaymentDetail = exports.CreditCardPaymentOption = exports.CreditCardPaymentDetail = exports.UpdateSubscriptionDetails = exports.TaxInfo = exports.SubscriptionSummary = exports.SubscriptionCollection = exports.Subscription = exports.PaymentOption = exports.PaymentMethod = exports.PaymentGateway = exports.PaymentDetail = exports.PaySubscriptionReceipt = exports.PaySubscriptionDetails = exports.PayInvoiceReceipt = exports.PayInvoiceDetails = exports.MerchantDefinedData = exports.InvoiceSummary = exports.InvoiceLineSummary = exports.InvoiceLineCollection = exports.InvoiceCollection = exports.Invoice = exports.Currency = exports.CreditCardType = exports.Country = exports.BillingAddress = exports.BillToAddress = exports.AuthorizeSubscriptionPaymentReceipt = exports.AuthorizeSubscriptionPaymentDetails = void 0;
const AuthorizeSubscriptionPaymentDetails = __importStar(require("./authorize-subscription-payment-details"));
exports.AuthorizeSubscriptionPaymentDetails = AuthorizeSubscriptionPaymentDetails.AuthorizeSubscriptionPaymentDetails;
const AuthorizeSubscriptionPaymentReceipt = __importStar(require("./authorize-subscription-payment-receipt"));
exports.AuthorizeSubscriptionPaymentReceipt = AuthorizeSubscriptionPaymentReceipt.AuthorizeSubscriptionPaymentReceipt;
const BillToAddress = __importStar(require("./bill-to-address"));
exports.BillToAddress = BillToAddress.BillToAddress;
const BillingAddress = __importStar(require("./billing-address"));
exports.BillingAddress = BillingAddress.BillingAddress;
const Country = __importStar(require("./country"));
exports.Country = Country.Country;
const CreditCardType = __importStar(require("./credit-card-type"));
exports.CreditCardType = CreditCardType.CreditCardType;
const Currency = __importStar(require("./currency"));
exports.Currency = Currency.Currency;
const Invoice = __importStar(require("./invoice"));
exports.Invoice = Invoice.Invoice;
const InvoiceCollection = __importStar(require("./invoice-collection"));
exports.InvoiceCollection = InvoiceCollection.InvoiceCollection;
const InvoiceLineCollection = __importStar(require("./invoice-line-collection"));
exports.InvoiceLineCollection = InvoiceLineCollection.InvoiceLineCollection;
const InvoiceLineSummary = __importStar(require("./invoice-line-summary"));
exports.InvoiceLineSummary = InvoiceLineSummary.InvoiceLineSummary;
const InvoiceSummary = __importStar(require("./invoice-summary"));
exports.InvoiceSummary = InvoiceSummary.InvoiceSummary;
const MerchantDefinedData = __importStar(require("./merchant-defined-data"));
exports.MerchantDefinedData = MerchantDefinedData.MerchantDefinedData;
const PayInvoiceDetails = __importStar(require("./pay-invoice-details"));
exports.PayInvoiceDetails = PayInvoiceDetails.PayInvoiceDetails;
const PayInvoiceReceipt = __importStar(require("./pay-invoice-receipt"));
exports.PayInvoiceReceipt = PayInvoiceReceipt.PayInvoiceReceipt;
const PaySubscriptionDetails = __importStar(require("./pay-subscription-details"));
exports.PaySubscriptionDetails = PaySubscriptionDetails.PaySubscriptionDetails;
const PaySubscriptionReceipt = __importStar(require("./pay-subscription-receipt"));
exports.PaySubscriptionReceipt = PaySubscriptionReceipt.PaySubscriptionReceipt;
const PaymentDetail = __importStar(require("./payment-detail"));
exports.PaymentDetail = PaymentDetail.PaymentDetail;
const PaymentGateway = __importStar(require("./payment-gateway"));
exports.PaymentGateway = PaymentGateway.PaymentGateway;
const PaymentMethod = __importStar(require("./payment-method"));
exports.PaymentMethod = PaymentMethod.PaymentMethod;
const PaymentOption = __importStar(require("./payment-option"));
exports.PaymentOption = PaymentOption.PaymentOption;
const Subscription = __importStar(require("./subscription"));
exports.Subscription = Subscription.Subscription;
const SubscriptionCollection = __importStar(require("./subscription-collection"));
exports.SubscriptionCollection = SubscriptionCollection.SubscriptionCollection;
const SubscriptionSummary = __importStar(require("./subscription-summary"));
exports.SubscriptionSummary = SubscriptionSummary.SubscriptionSummary;
const TaxInfo = __importStar(require("./tax-info"));
exports.TaxInfo = TaxInfo.TaxInfo;
const UpdateSubscriptionDetails = __importStar(require("./update-subscription-details"));
exports.UpdateSubscriptionDetails = UpdateSubscriptionDetails.UpdateSubscriptionDetails;
const CreditCardPaymentDetail = __importStar(require("./credit-card-payment-detail"));
exports.CreditCardPaymentDetail = CreditCardPaymentDetail.CreditCardPaymentDetail;
const CreditCardPaymentOption = __importStar(require("./credit-card-payment-option"));
exports.CreditCardPaymentOption = CreditCardPaymentOption.CreditCardPaymentOption;
const OtherPaymentDetail = __importStar(require("./other-payment-detail"));
exports.OtherPaymentDetail = OtherPaymentDetail.OtherPaymentDetail;
const PaypalPaymentDetail = __importStar(require("./paypal-payment-detail"));
exports.PaypalPaymentDetail = PaypalPaymentDetail.PaypalPaymentDetail;
const PaypalPaymentOption = __importStar(require("./paypal-payment-option"));
exports.PaypalPaymentOption = PaypalPaymentOption.PaypalPaymentOption;
