// oci-cloudmigrations/lib/request/index.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTargetAssetsRequest = exports.ListReplicationSchedulesRequest = exports.ListMigrationsRequest = exports.ListMigrationPlansRequest = exports.ListMigrationAssetsRequest = exports.ListAvailableShapesRequest = void 0;
const ListAvailableShapesRequest = __importStar(require("./list-available-shapes-request"));
exports.ListAvailableShapesRequest = ListAvailableShapesRequest.ListAvailableShapesRequest;
const ListMigrationAssetsRequest = __importStar(require("./list-migration-assets-request"));
exports.ListMigrationAssetsRequest = ListMigrationAssetsRequest.ListMigrationAssetsRequest;
const ListMigrationPlansRequest = __importStar(require("./list-migration-plans-request"));
exports.ListMigrationPlansRequest = ListMigrationPlansRequest.ListMigrationPlansRequest;
const ListMigrationsRequest = __importStar(require("./list-migrations-request"));
exports.ListMigrationsRequest = ListMigrationsRequest.ListMigrationsRequest;
const ListReplicationSchedulesRequest = __importStar(require("./list-replication-schedules-request"));
exports.ListReplicationSchedulesRequest = ListReplicationSchedulesRequest.ListReplicationSchedulesRequest;
const ListTargetAssetsRequest = __importStar(require("./list-target-assets-request"));
exports.ListTargetAssetsRequest = ListTargetAssetsRequest.ListTargetAssetsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
