// oci-servicecatalog/lib/model/index.js
"use strict";
/**
 * Service Catalog API
 * Manage solutions in Oracle Cloud Infrastructure Service Catalog.
 * OpenAPI spec version: 20210527
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateApplicationStackPackage = exports.CreatePrivateApplicationStackPackage = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UploadData = exports.UpdateServiceCatalogDetails = exports.UpdatePrivateApplicationDetails = exports.ServiceCatalogSummary = exports.ServiceCatalogCollection = exports.ServiceCatalogAssociationSummary = exports.ServiceCatalogAssociationCollection = exports.ServiceCatalogAssociation = exports.ServiceCatalog = exports.PublisherSummary = exports.PrivateApplicationSummary = exports.PrivateApplicationPackageSummary = exports.PrivateApplicationPackageCollection = exports.PrivateApplicationPackage = exports.PrivateApplicationCollection = exports.PrivateApplication = exports.PricingTypeEnum = exports.PackageTypeEnum = exports.OperationType = exports.OperationStatus = exports.ErrorEntity = exports.CreateServiceCatalogDetails = exports.CreateServiceCatalogAssociationDetails = exports.CreatePrivateApplicationPackage = exports.CreatePrivateApplicationDetails = exports.ChangeServiceCatalogCompartmentDetails = exports.ChangePrivateApplicationCompartmentDetails = exports.BulkReplaceServiceCatalogAssociationsDetails = exports.ApplicationSummary = exports.ApplicationCollection = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ApplicationCollection = __importStar(require("./application-collection"));
exports.ApplicationCollection = ApplicationCollection.ApplicationCollection;
const ApplicationSummary = __importStar(require("./application-summary"));
exports.ApplicationSummary = ApplicationSummary.ApplicationSummary;
const BulkReplaceServiceCatalogAssociationsDetails = __importStar(require("./bulk-replace-service-catalog-associations-details"));
exports.BulkReplaceServiceCatalogAssociationsDetails = BulkReplaceServiceCatalogAssociationsDetails.BulkReplaceServiceCatalogAssociationsDetails;
const ChangePrivateApplicationCompartmentDetails = __importStar(require("./change-private-application-compartment-details"));
exports.ChangePrivateApplicationCompartmentDetails = ChangePrivateApplicationCompartmentDetails.ChangePrivateApplicationCompartmentDetails;
const ChangeServiceCatalogCompartmentDetails = __importStar(require("./change-service-catalog-compartment-details"));
exports.ChangeServiceCatalogCompartmentDetails = ChangeServiceCatalogCompartmentDetails.ChangeServiceCatalogCompartmentDetails;
const CreatePrivateApplicationDetails = __importStar(require("./create-private-application-details"));
exports.CreatePrivateApplicationDetails = CreatePrivateApplicationDetails.CreatePrivateApplicationDetails;
const CreatePrivateApplicationPackage = __importStar(require("./create-private-application-package"));
exports.CreatePrivateApplicationPackage = CreatePrivateApplicationPackage.CreatePrivateApplicationPackage;
const CreateServiceCatalogAssociationDetails = __importStar(require("./create-service-catalog-association-details"));
exports.CreateServiceCatalogAssociationDetails = CreateServiceCatalogAssociationDetails.CreateServiceCatalogAssociationDetails;
const CreateServiceCatalogDetails = __importStar(require("./create-service-catalog-details"));
exports.CreateServiceCatalogDetails = CreateServiceCatalogDetails.CreateServiceCatalogDetails;
const ErrorEntity = __importStar(require("./error-entity"));
exports.ErrorEntity = ErrorEntity.ErrorEntity;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PackageTypeEnum = __importStar(require("./package-type-enum"));
exports.PackageTypeEnum = PackageTypeEnum.PackageTypeEnum;
const PricingTypeEnum = __importStar(require("./pricing-type-enum"));
exports.PricingTypeEnum = PricingTypeEnum.PricingTypeEnum;
const PrivateApplication = __importStar(require("./private-application"));
exports.PrivateApplication = PrivateApplication.PrivateApplication;
const PrivateApplicationCollection = __importStar(require("./private-application-collection"));
exports.PrivateApplicationCollection = PrivateApplicationCollection.PrivateApplicationCollection;
const PrivateApplicationPackage = __importStar(require("./private-application-package"));
exports.PrivateApplicationPackage = PrivateApplicationPackage.PrivateApplicationPackage;
const PrivateApplicationPackageCollection = __importStar(require("./private-application-package-collection"));
exports.PrivateApplicationPackageCollection = PrivateApplicationPackageCollection.PrivateApplicationPackageCollection;
const PrivateApplicationPackageSummary = __importStar(require("./private-application-package-summary"));
exports.PrivateApplicationPackageSummary = PrivateApplicationPackageSummary.PrivateApplicationPackageSummary;
const PrivateApplicationSummary = __importStar(require("./private-application-summary"));
exports.PrivateApplicationSummary = PrivateApplicationSummary.PrivateApplicationSummary;
const PublisherSummary = __importStar(require("./publisher-summary"));
exports.PublisherSummary = PublisherSummary.PublisherSummary;
const ServiceCatalog = __importStar(require("./service-catalog"));
exports.ServiceCatalog = ServiceCatalog.ServiceCatalog;
const ServiceCatalogAssociation = __importStar(require("./service-catalog-association"));
exports.ServiceCatalogAssociation = ServiceCatalogAssociation.ServiceCatalogAssociation;
const ServiceCatalogAssociationCollection = __importStar(require("./service-catalog-association-collection"));
exports.ServiceCatalogAssociationCollection = ServiceCatalogAssociationCollection.ServiceCatalogAssociationCollection;
const ServiceCatalogAssociationSummary = __importStar(require("./service-catalog-association-summary"));
exports.ServiceCatalogAssociationSummary = ServiceCatalogAssociationSummary.ServiceCatalogAssociationSummary;
const ServiceCatalogCollection = __importStar(require("./service-catalog-collection"));
exports.ServiceCatalogCollection = ServiceCatalogCollection.ServiceCatalogCollection;
const ServiceCatalogSummary = __importStar(require("./service-catalog-summary"));
exports.ServiceCatalogSummary = ServiceCatalogSummary.ServiceCatalogSummary;
const UpdatePrivateApplicationDetails = __importStar(require("./update-private-application-details"));
exports.UpdatePrivateApplicationDetails = UpdatePrivateApplicationDetails.UpdatePrivateApplicationDetails;
const UpdateServiceCatalogDetails = __importStar(require("./update-service-catalog-details"));
exports.UpdateServiceCatalogDetails = UpdateServiceCatalogDetails.UpdateServiceCatalogDetails;
const UploadData = __importStar(require("./upload-data"));
exports.UploadData = UploadData.UploadData;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const CreatePrivateApplicationStackPackage = __importStar(require("./create-private-application-stack-package"));
exports.CreatePrivateApplicationStackPackage = CreatePrivateApplicationStackPackage.CreatePrivateApplicationStackPackage;
const PrivateApplicationStackPackage = __importStar(require("./private-application-stack-package"));
exports.PrivateApplicationStackPackage = PrivateApplicationStackPackage.PrivateApplicationStackPackage;
