// oci-servicecatalog/lib/model/package-type-enum.js
"use strict";
/**
 * Service Catalog API
 * Manage solutions in Oracle Cloud Infrastructure Service Catalog.
 * OpenAPI spec version: 20210527
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageTypeEnum = void 0;
/**
 * Type of a package.
 **/
var PackageTypeEnum;
(function (PackageTypeEnum) {
    PackageTypeEnum["Stack"] = "STACK";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    PackageTypeEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(PackageTypeEnum = exports.PackageTypeEnum || (exports.PackageTypeEnum = {}));
(function (PackageTypeEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    PackageTypeEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PackageTypeEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(PackageTypeEnum = exports.PackageTypeEnum || (exports.PackageTypeEnum = {}));
