// oci-devops/lib/request/index.js
"use strict";
/**
 * DevOps API
 * Use the DevOps API to create DevOps projects, configure code repositories,  add artifacts to deploy, build and test software applications, configure  target deployment environments, and deploy software applications.  For more information, see [DevOps](/Content/devops/using/home.htm).
 * OpenAPI spec version: 20210630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTriggersRequest = exports.ListRepositoriesRequest = exports.ListRefsRequest = exports.ListProjectsRequest = exports.ListPathsRequest = exports.ListDeploymentsRequest = exports.ListDeployStagesRequest = exports.ListDeployPipelinesRequest = exports.ListDeployEnvironmentsRequest = exports.ListDeployArtifactsRequest = exports.ListConnectionsRequest = exports.ListBuildRunsRequest = exports.ListBuildPipelinesRequest = exports.ListBuildPipelineStagesRequest = exports.GetRepositoryRequest = exports.GetMirrorRecordRequest = void 0;
const GetMirrorRecordRequest = __importStar(require("./get-mirror-record-request"));
exports.GetMirrorRecordRequest = GetMirrorRecordRequest.GetMirrorRecordRequest;
const GetRepositoryRequest = __importStar(require("./get-repository-request"));
exports.GetRepositoryRequest = GetRepositoryRequest.GetRepositoryRequest;
const ListBuildPipelineStagesRequest = __importStar(require("./list-build-pipeline-stages-request"));
exports.ListBuildPipelineStagesRequest = ListBuildPipelineStagesRequest.ListBuildPipelineStagesRequest;
const ListBuildPipelinesRequest = __importStar(require("./list-build-pipelines-request"));
exports.ListBuildPipelinesRequest = ListBuildPipelinesRequest.ListBuildPipelinesRequest;
const ListBuildRunsRequest = __importStar(require("./list-build-runs-request"));
exports.ListBuildRunsRequest = ListBuildRunsRequest.ListBuildRunsRequest;
const ListConnectionsRequest = __importStar(require("./list-connections-request"));
exports.ListConnectionsRequest = ListConnectionsRequest.ListConnectionsRequest;
const ListDeployArtifactsRequest = __importStar(require("./list-deploy-artifacts-request"));
exports.ListDeployArtifactsRequest = ListDeployArtifactsRequest.ListDeployArtifactsRequest;
const ListDeployEnvironmentsRequest = __importStar(require("./list-deploy-environments-request"));
exports.ListDeployEnvironmentsRequest = ListDeployEnvironmentsRequest.ListDeployEnvironmentsRequest;
const ListDeployPipelinesRequest = __importStar(require("./list-deploy-pipelines-request"));
exports.ListDeployPipelinesRequest = ListDeployPipelinesRequest.ListDeployPipelinesRequest;
const ListDeployStagesRequest = __importStar(require("./list-deploy-stages-request"));
exports.ListDeployStagesRequest = ListDeployStagesRequest.ListDeployStagesRequest;
const ListDeploymentsRequest = __importStar(require("./list-deployments-request"));
exports.ListDeploymentsRequest = ListDeploymentsRequest.ListDeploymentsRequest;
const ListPathsRequest = __importStar(require("./list-paths-request"));
exports.ListPathsRequest = ListPathsRequest.ListPathsRequest;
const ListProjectsRequest = __importStar(require("./list-projects-request"));
exports.ListProjectsRequest = ListProjectsRequest.ListProjectsRequest;
const ListRefsRequest = __importStar(require("./list-refs-request"));
exports.ListRefsRequest = ListRefsRequest.ListRefsRequest;
const ListRepositoriesRequest = __importStar(require("./list-repositories-request"));
exports.ListRepositoriesRequest = ListRepositoriesRequest.ListRepositoriesRequest;
const ListTriggersRequest = __importStar(require("./list-triggers-request"));
exports.ListTriggersRequest = ListTriggersRequest.ListTriggersRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
