// oci-devops/lib/request/list-refs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRefsRequest = void 0;
var ListRefsRequest;
(function (ListRefsRequest) {
    let RefType;
    (function (RefType) {
        RefType["Branch"] = "BRANCH";
        RefType["Tag"] = "TAG";
    })(RefType = ListRefsRequest.RefType || (ListRefsRequest.RefType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["RefType"] = "refType";
        SortBy["RefName"] = "refName";
    })(SortBy = ListRefsRequest.SortBy || (ListRefsRequest.SortBy = {}));
})(ListRefsRequest = exports.ListRefsRequest || (exports.ListRefsRequest = {}));
