// oci-databasetools/lib/model/key-store-type-my-sql.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyStoreTypeMySql = void 0;
/**
 * The key store type.
 **/
var KeyStoreTypeMySql;
(function (KeyStoreTypeMySql) {
    KeyStoreTypeMySql["ClientCertificatePem"] = "CLIENT_CERTIFICATE_PEM";
    KeyStoreTypeMySql["ClientPrivateKeyPem"] = "CLIENT_PRIVATE_KEY_PEM";
    KeyStoreTypeMySql["CaCertificatePem"] = "CA_CERTIFICATE_PEM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    KeyStoreTypeMySql["UnknownValue"] = "UNKNOWN_VALUE";
})(KeyStoreTypeMySql = exports.KeyStoreTypeMySql || (exports.KeyStoreTypeMySql = {}));
(function (KeyStoreTypeMySql) {
    function getJsonObj(obj) {
        return obj;
    }
    KeyStoreTypeMySql.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KeyStoreTypeMySql.getDeserializedJsonObj = getDeserializedJsonObj;
})(KeyStoreTypeMySql = exports.KeyStoreTypeMySql || (exports.KeyStoreTypeMySql = {}));
