// oci-databasetools/lib/model/operation-type.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateDatabaseToolsConnection"] = "CREATE_DATABASE_TOOLS_CONNECTION";
    OperationType["UpdateDatabaseToolsConnection"] = "UPDATE_DATABASE_TOOLS_CONNECTION";
    OperationType["DeleteDatabaseToolsConnection"] = "DELETE_DATABASE_TOOLS_CONNECTION";
    OperationType["CreateDatabaseToolsServiceInstance"] = "CREATE_DATABASE_TOOLS_SERVICE_INSTANCE";
    OperationType["UpdateDatabaseToolsServiceInstance"] = "UPDATE_DATABASE_TOOLS_SERVICE_INSTANCE";
    OperationType["DeleteDatabaseToolsServiceInstance"] = "DELETE_DATABASE_TOOLS_SERVICE_INSTANCE";
    OperationType["CreateDatabaseToolsPrivateEndpoint"] = "CREATE_DATABASE_TOOLS_PRIVATE_ENDPOINT";
    OperationType["UpdateDatabaseToolsPrivateEndpoint"] = "UPDATE_DATABASE_TOOLS_PRIVATE_ENDPOINT";
    OperationType["DeleteDatabaseToolsPrivateEndpoint"] = "DELETE_DATABASE_TOOLS_PRIVATE_ENDPOINT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
