// oci-aianomalydetection/lib/request/index.js
"use strict";
/**
 * Anomaly Detection API
 * OCI AI Service solutions can help Enterprise customers integrate AI into their products immediately by using our proven,
pre-trained/custom models or containers, and without a need to set up in house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI/ML operations, shortening the time to market.

 * OpenAPI spec version: 20210101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListProjectsRequest = exports.ListModelsRequest = exports.ListDetectAnomalyJobsRequest = exports.ListDataAssetsRequest = exports.ListAiPrivateEndpointsRequest = void 0;
const ListAiPrivateEndpointsRequest = __importStar(require("./list-ai-private-endpoints-request"));
exports.ListAiPrivateEndpointsRequest = ListAiPrivateEndpointsRequest.ListAiPrivateEndpointsRequest;
const ListDataAssetsRequest = __importStar(require("./list-data-assets-request"));
exports.ListDataAssetsRequest = ListDataAssetsRequest.ListDataAssetsRequest;
const ListDetectAnomalyJobsRequest = __importStar(require("./list-detect-anomaly-jobs-request"));
exports.ListDetectAnomalyJobsRequest = ListDetectAnomalyJobsRequest.ListDetectAnomalyJobsRequest;
const ListModelsRequest = __importStar(require("./list-models-request"));
exports.ListModelsRequest = ListModelsRequest.ListModelsRequest;
const ListProjectsRequest = __importStar(require("./list-projects-request"));
exports.ListProjectsRequest = ListProjectsRequest.ListProjectsRequest;
