// oci-stackmonitoring/lib/request/list-discovery-job-logs-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210330
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDiscoveryJobLogsRequest = void 0;
var ListDiscoveryJobLogsRequest;
(function (ListDiscoveryJobLogsRequest) {
    let LogType;
    (function (LogType) {
        LogType["Info"] = "INFO";
        LogType["Warning"] = "WARNING";
        LogType["Error"] = "ERROR";
        LogType["Success"] = "SUCCESS";
    })(LogType = ListDiscoveryJobLogsRequest.LogType || (ListDiscoveryJobLogsRequest.LogType = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["LogType"] = "logType";
    })(SortBy = ListDiscoveryJobLogsRequest.SortBy || (ListDiscoveryJobLogsRequest.SortBy = {}));
})(ListDiscoveryJobLogsRequest = exports.ListDiscoveryJobLogsRequest || (exports.ListDiscoveryJobLogsRequest = {}));
