// oci-containerengine/lib/model/addon-lifecycle-state.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddonLifecycleState = void 0;
/**
 * The lifecycle state of an addon.
 **/
var AddonLifecycleState;
(function (AddonLifecycleState) {
    AddonLifecycleState["Creating"] = "CREATING";
    AddonLifecycleState["Active"] = "ACTIVE";
    AddonLifecycleState["Deleting"] = "DELETING";
    AddonLifecycleState["Deleted"] = "DELETED";
    AddonLifecycleState["Updating"] = "UPDATING";
    AddonLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    AddonLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AddonLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AddonLifecycleState = exports.AddonLifecycleState || (exports.AddonLifecycleState = {}));
(function (AddonLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AddonLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AddonLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AddonLifecycleState = exports.AddonLifecycleState || (exports.AddonLifecycleState = {}));
