// oci-containerengine/lib/model/index.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeSourceDetails = exports.NodeShapeConfig = exports.NodePoolSummary = exports.NodePoolPodNetworkOptionDetails = exports.NodePoolPlacementConfigDetails = exports.NodePoolOptions = exports.NodePoolNodeConfigDetails = exports.NodePoolLifecycleState = exports.NodePoolCyclingDetails = exports.NodePool = exports.NodeEvictionNodePoolSettings = exports.NodeError = exports.Node = exports.KubernetesVersionsFilters = exports.KubernetesNetworkConfig = exports.KeyValue = exports.KeyDetails = exports.InstallAddonDetails = exports.InitialVirtualNodeLabel = exports.ImagePolicyConfig = exports.CreateVirtualNodePoolDetails = exports.CreateNodeShapeConfigDetails = exports.CreateNodePoolNodeConfigDetails = exports.CreateNodePoolDetails = exports.CreateImagePolicyConfigDetails = exports.CreateClusterKubeconfigContentDetails = exports.CreateClusterEndpointConfigDetails = exports.CreateClusterDetails = exports.ClusterType = exports.ClusterSummary = exports.ClusterPodNetworkOptionDetails = exports.ClusterOptions = exports.ClusterMigrateToNativeVcnStatus = exports.ClusterMigrateToNativeVcnDetails = exports.ClusterMetadata = exports.ClusterLifecycleState = exports.ClusterEndpoints = exports.ClusterEndpointConfig = exports.ClusterCreateOptions = exports.Cluster = exports.AdmissionControllerOptions = exports.AddonVersions = exports.AddonVersionConfiguration = exports.AddonSummary = exports.AddonOptionSummary = exports.AddonLifecycleState = exports.AddonError = exports.AddonConfiguration = exports.Addon = exports.AddOnOptions = void 0;
exports.TerminatePreemptionAction = exports.OciVcnIpNativeNodePoolPodNetworkOptionDetails = exports.OciVcnIpNativeClusterPodNetworkOptionDetails = exports.NodeSourceViaImageOption = exports.NodeSourceViaImageDetails = exports.FlannelOverlayNodePoolPodNetworkOptionDetails = exports.FlannelOverlayClusterPodNetworkOptionDetails = exports.WorkRequestSummary = exports.WorkRequestStatus = exports.WorkRequestResource = exports.WorkRequestOperationType = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.VirtualNodeTags = exports.VirtualNodeSummary = exports.VirtualNodePoolSummary = exports.VirtualNodePoolLifecycleState = exports.VirtualNodePool = exports.VirtualNodeLifecycleState = exports.VirtualNode = exports.UpdateVirtualNodePoolDetails = exports.UpdateNodeShapeConfigDetails = exports.UpdateNodePoolNodeConfigDetails = exports.UpdateNodePoolDetails = exports.UpdateImagePolicyConfigDetails = exports.UpdateClusterOptionsDetails = exports.UpdateClusterEndpointConfigDetails = exports.UpdateClusterDetails = exports.UpdateAddonDetails = exports.Taint = exports.SortOrder = exports.ShapeOcpuOptions = exports.ShapeNetworkBandwidthOptions = exports.ShapeMemoryOptions = exports.ServiceLbConfigDetails = exports.PreemptionAction = exports.PreemptibleNodeConfigDetails = exports.PodShapeSummary = exports.PodShape = exports.PodConfiguration = exports.PlacementConfiguration = exports.PersistentVolumeConfigDetails = exports.NodeSourceType = exports.NodeSourceOption = void 0;
const AddOnOptions = __importStar(require("./add-on-options"));
exports.AddOnOptions = AddOnOptions.AddOnOptions;
const Addon = __importStar(require("./addon"));
exports.Addon = Addon.Addon;
const AddonConfiguration = __importStar(require("./addon-configuration"));
exports.AddonConfiguration = AddonConfiguration.AddonConfiguration;
const AddonError = __importStar(require("./addon-error"));
exports.AddonError = AddonError.AddonError;
const AddonLifecycleState = __importStar(require("./addon-lifecycle-state"));
exports.AddonLifecycleState = AddonLifecycleState.AddonLifecycleState;
const AddonOptionSummary = __importStar(require("./addon-option-summary"));
exports.AddonOptionSummary = AddonOptionSummary.AddonOptionSummary;
const AddonSummary = __importStar(require("./addon-summary"));
exports.AddonSummary = AddonSummary.AddonSummary;
const AddonVersionConfiguration = __importStar(require("./addon-version-configuration"));
exports.AddonVersionConfiguration = AddonVersionConfiguration.AddonVersionConfiguration;
const AddonVersions = __importStar(require("./addon-versions"));
exports.AddonVersions = AddonVersions.AddonVersions;
const AdmissionControllerOptions = __importStar(require("./admission-controller-options"));
exports.AdmissionControllerOptions = AdmissionControllerOptions.AdmissionControllerOptions;
const Cluster = __importStar(require("./cluster"));
exports.Cluster = Cluster.Cluster;
const ClusterCreateOptions = __importStar(require("./cluster-create-options"));
exports.ClusterCreateOptions = ClusterCreateOptions.ClusterCreateOptions;
const ClusterEndpointConfig = __importStar(require("./cluster-endpoint-config"));
exports.ClusterEndpointConfig = ClusterEndpointConfig.ClusterEndpointConfig;
const ClusterEndpoints = __importStar(require("./cluster-endpoints"));
exports.ClusterEndpoints = ClusterEndpoints.ClusterEndpoints;
const ClusterLifecycleState = __importStar(require("./cluster-lifecycle-state"));
exports.ClusterLifecycleState = ClusterLifecycleState.ClusterLifecycleState;
const ClusterMetadata = __importStar(require("./cluster-metadata"));
exports.ClusterMetadata = ClusterMetadata.ClusterMetadata;
const ClusterMigrateToNativeVcnDetails = __importStar(require("./cluster-migrate-to-native-vcn-details"));
exports.ClusterMigrateToNativeVcnDetails = ClusterMigrateToNativeVcnDetails.ClusterMigrateToNativeVcnDetails;
const ClusterMigrateToNativeVcnStatus = __importStar(require("./cluster-migrate-to-native-vcn-status"));
exports.ClusterMigrateToNativeVcnStatus = ClusterMigrateToNativeVcnStatus.ClusterMigrateToNativeVcnStatus;
const ClusterOptions = __importStar(require("./cluster-options"));
exports.ClusterOptions = ClusterOptions.ClusterOptions;
const ClusterPodNetworkOptionDetails = __importStar(require("./cluster-pod-network-option-details"));
exports.ClusterPodNetworkOptionDetails = ClusterPodNetworkOptionDetails.ClusterPodNetworkOptionDetails;
const ClusterSummary = __importStar(require("./cluster-summary"));
exports.ClusterSummary = ClusterSummary.ClusterSummary;
const ClusterType = __importStar(require("./cluster-type"));
exports.ClusterType = ClusterType.ClusterType;
const CreateClusterDetails = __importStar(require("./create-cluster-details"));
exports.CreateClusterDetails = CreateClusterDetails.CreateClusterDetails;
const CreateClusterEndpointConfigDetails = __importStar(require("./create-cluster-endpoint-config-details"));
exports.CreateClusterEndpointConfigDetails = CreateClusterEndpointConfigDetails.CreateClusterEndpointConfigDetails;
const CreateClusterKubeconfigContentDetails = __importStar(require("./create-cluster-kubeconfig-content-details"));
exports.CreateClusterKubeconfigContentDetails = CreateClusterKubeconfigContentDetails.CreateClusterKubeconfigContentDetails;
const CreateImagePolicyConfigDetails = __importStar(require("./create-image-policy-config-details"));
exports.CreateImagePolicyConfigDetails = CreateImagePolicyConfigDetails.CreateImagePolicyConfigDetails;
const CreateNodePoolDetails = __importStar(require("./create-node-pool-details"));
exports.CreateNodePoolDetails = CreateNodePoolDetails.CreateNodePoolDetails;
const CreateNodePoolNodeConfigDetails = __importStar(require("./create-node-pool-node-config-details"));
exports.CreateNodePoolNodeConfigDetails = CreateNodePoolNodeConfigDetails.CreateNodePoolNodeConfigDetails;
const CreateNodeShapeConfigDetails = __importStar(require("./create-node-shape-config-details"));
exports.CreateNodeShapeConfigDetails = CreateNodeShapeConfigDetails.CreateNodeShapeConfigDetails;
const CreateVirtualNodePoolDetails = __importStar(require("./create-virtual-node-pool-details"));
exports.CreateVirtualNodePoolDetails = CreateVirtualNodePoolDetails.CreateVirtualNodePoolDetails;
const ImagePolicyConfig = __importStar(require("./image-policy-config"));
exports.ImagePolicyConfig = ImagePolicyConfig.ImagePolicyConfig;
const InitialVirtualNodeLabel = __importStar(require("./initial-virtual-node-label"));
exports.InitialVirtualNodeLabel = InitialVirtualNodeLabel.InitialVirtualNodeLabel;
const InstallAddonDetails = __importStar(require("./install-addon-details"));
exports.InstallAddonDetails = InstallAddonDetails.InstallAddonDetails;
const KeyDetails = __importStar(require("./key-details"));
exports.KeyDetails = KeyDetails.KeyDetails;
const KeyValue = __importStar(require("./key-value"));
exports.KeyValue = KeyValue.KeyValue;
const KubernetesNetworkConfig = __importStar(require("./kubernetes-network-config"));
exports.KubernetesNetworkConfig = KubernetesNetworkConfig.KubernetesNetworkConfig;
const KubernetesVersionsFilters = __importStar(require("./kubernetes-versions-filters"));
exports.KubernetesVersionsFilters = KubernetesVersionsFilters.KubernetesVersionsFilters;
const Node = __importStar(require("./node"));
exports.Node = Node.Node;
const NodeError = __importStar(require("./node-error"));
exports.NodeError = NodeError.NodeError;
const NodeEvictionNodePoolSettings = __importStar(require("./node-eviction-node-pool-settings"));
exports.NodeEvictionNodePoolSettings = NodeEvictionNodePoolSettings.NodeEvictionNodePoolSettings;
const NodePool = __importStar(require("./node-pool"));
exports.NodePool = NodePool.NodePool;
const NodePoolCyclingDetails = __importStar(require("./node-pool-cycling-details"));
exports.NodePoolCyclingDetails = NodePoolCyclingDetails.NodePoolCyclingDetails;
const NodePoolLifecycleState = __importStar(require("./node-pool-lifecycle-state"));
exports.NodePoolLifecycleState = NodePoolLifecycleState.NodePoolLifecycleState;
const NodePoolNodeConfigDetails = __importStar(require("./node-pool-node-config-details"));
exports.NodePoolNodeConfigDetails = NodePoolNodeConfigDetails.NodePoolNodeConfigDetails;
const NodePoolOptions = __importStar(require("./node-pool-options"));
exports.NodePoolOptions = NodePoolOptions.NodePoolOptions;
const NodePoolPlacementConfigDetails = __importStar(require("./node-pool-placement-config-details"));
exports.NodePoolPlacementConfigDetails = NodePoolPlacementConfigDetails.NodePoolPlacementConfigDetails;
const NodePoolPodNetworkOptionDetails = __importStar(require("./node-pool-pod-network-option-details"));
exports.NodePoolPodNetworkOptionDetails = NodePoolPodNetworkOptionDetails.NodePoolPodNetworkOptionDetails;
const NodePoolSummary = __importStar(require("./node-pool-summary"));
exports.NodePoolSummary = NodePoolSummary.NodePoolSummary;
const NodeShapeConfig = __importStar(require("./node-shape-config"));
exports.NodeShapeConfig = NodeShapeConfig.NodeShapeConfig;
const NodeSourceDetails = __importStar(require("./node-source-details"));
exports.NodeSourceDetails = NodeSourceDetails.NodeSourceDetails;
const NodeSourceOption = __importStar(require("./node-source-option"));
exports.NodeSourceOption = NodeSourceOption.NodeSourceOption;
const NodeSourceType = __importStar(require("./node-source-type"));
exports.NodeSourceType = NodeSourceType.NodeSourceType;
const PersistentVolumeConfigDetails = __importStar(require("./persistent-volume-config-details"));
exports.PersistentVolumeConfigDetails = PersistentVolumeConfigDetails.PersistentVolumeConfigDetails;
const PlacementConfiguration = __importStar(require("./placement-configuration"));
exports.PlacementConfiguration = PlacementConfiguration.PlacementConfiguration;
const PodConfiguration = __importStar(require("./pod-configuration"));
exports.PodConfiguration = PodConfiguration.PodConfiguration;
const PodShape = __importStar(require("./pod-shape"));
exports.PodShape = PodShape.PodShape;
const PodShapeSummary = __importStar(require("./pod-shape-summary"));
exports.PodShapeSummary = PodShapeSummary.PodShapeSummary;
const PreemptibleNodeConfigDetails = __importStar(require("./preemptible-node-config-details"));
exports.PreemptibleNodeConfigDetails = PreemptibleNodeConfigDetails.PreemptibleNodeConfigDetails;
const PreemptionAction = __importStar(require("./preemption-action"));
exports.PreemptionAction = PreemptionAction.PreemptionAction;
const ServiceLbConfigDetails = __importStar(require("./service-lb-config-details"));
exports.ServiceLbConfigDetails = ServiceLbConfigDetails.ServiceLbConfigDetails;
const ShapeMemoryOptions = __importStar(require("./shape-memory-options"));
exports.ShapeMemoryOptions = ShapeMemoryOptions.ShapeMemoryOptions;
const ShapeNetworkBandwidthOptions = __importStar(require("./shape-network-bandwidth-options"));
exports.ShapeNetworkBandwidthOptions = ShapeNetworkBandwidthOptions.ShapeNetworkBandwidthOptions;
const ShapeOcpuOptions = __importStar(require("./shape-ocpu-options"));
exports.ShapeOcpuOptions = ShapeOcpuOptions.ShapeOcpuOptions;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Taint = __importStar(require("./taint"));
exports.Taint = Taint.Taint;
const UpdateAddonDetails = __importStar(require("./update-addon-details"));
exports.UpdateAddonDetails = UpdateAddonDetails.UpdateAddonDetails;
const UpdateClusterDetails = __importStar(require("./update-cluster-details"));
exports.UpdateClusterDetails = UpdateClusterDetails.UpdateClusterDetails;
const UpdateClusterEndpointConfigDetails = __importStar(require("./update-cluster-endpoint-config-details"));
exports.UpdateClusterEndpointConfigDetails = UpdateClusterEndpointConfigDetails.UpdateClusterEndpointConfigDetails;
const UpdateClusterOptionsDetails = __importStar(require("./update-cluster-options-details"));
exports.UpdateClusterOptionsDetails = UpdateClusterOptionsDetails.UpdateClusterOptionsDetails;
const UpdateImagePolicyConfigDetails = __importStar(require("./update-image-policy-config-details"));
exports.UpdateImagePolicyConfigDetails = UpdateImagePolicyConfigDetails.UpdateImagePolicyConfigDetails;
const UpdateNodePoolDetails = __importStar(require("./update-node-pool-details"));
exports.UpdateNodePoolDetails = UpdateNodePoolDetails.UpdateNodePoolDetails;
const UpdateNodePoolNodeConfigDetails = __importStar(require("./update-node-pool-node-config-details"));
exports.UpdateNodePoolNodeConfigDetails = UpdateNodePoolNodeConfigDetails.UpdateNodePoolNodeConfigDetails;
const UpdateNodeShapeConfigDetails = __importStar(require("./update-node-shape-config-details"));
exports.UpdateNodeShapeConfigDetails = UpdateNodeShapeConfigDetails.UpdateNodeShapeConfigDetails;
const UpdateVirtualNodePoolDetails = __importStar(require("./update-virtual-node-pool-details"));
exports.UpdateVirtualNodePoolDetails = UpdateVirtualNodePoolDetails.UpdateVirtualNodePoolDetails;
const VirtualNode = __importStar(require("./virtual-node"));
exports.VirtualNode = VirtualNode.VirtualNode;
const VirtualNodeLifecycleState = __importStar(require("./virtual-node-lifecycle-state"));
exports.VirtualNodeLifecycleState = VirtualNodeLifecycleState.VirtualNodeLifecycleState;
const VirtualNodePool = __importStar(require("./virtual-node-pool"));
exports.VirtualNodePool = VirtualNodePool.VirtualNodePool;
const VirtualNodePoolLifecycleState = __importStar(require("./virtual-node-pool-lifecycle-state"));
exports.VirtualNodePoolLifecycleState = VirtualNodePoolLifecycleState.VirtualNodePoolLifecycleState;
const VirtualNodePoolSummary = __importStar(require("./virtual-node-pool-summary"));
exports.VirtualNodePoolSummary = VirtualNodePoolSummary.VirtualNodePoolSummary;
const VirtualNodeSummary = __importStar(require("./virtual-node-summary"));
exports.VirtualNodeSummary = VirtualNodeSummary.VirtualNodeSummary;
const VirtualNodeTags = __importStar(require("./virtual-node-tags"));
exports.VirtualNodeTags = VirtualNodeTags.VirtualNodeTags;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestOperationType = __importStar(require("./work-request-operation-type"));
exports.WorkRequestOperationType = WorkRequestOperationType.WorkRequestOperationType;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestStatus = __importStar(require("./work-request-status"));
exports.WorkRequestStatus = WorkRequestStatus.WorkRequestStatus;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const FlannelOverlayClusterPodNetworkOptionDetails = __importStar(require("./flannel-overlay-cluster-pod-network-option-details"));
exports.FlannelOverlayClusterPodNetworkOptionDetails = FlannelOverlayClusterPodNetworkOptionDetails.FlannelOverlayClusterPodNetworkOptionDetails;
const FlannelOverlayNodePoolPodNetworkOptionDetails = __importStar(require("./flannel-overlay-node-pool-pod-network-option-details"));
exports.FlannelOverlayNodePoolPodNetworkOptionDetails = FlannelOverlayNodePoolPodNetworkOptionDetails.FlannelOverlayNodePoolPodNetworkOptionDetails;
const NodeSourceViaImageDetails = __importStar(require("./node-source-via-image-details"));
exports.NodeSourceViaImageDetails = NodeSourceViaImageDetails.NodeSourceViaImageDetails;
const NodeSourceViaImageOption = __importStar(require("./node-source-via-image-option"));
exports.NodeSourceViaImageOption = NodeSourceViaImageOption.NodeSourceViaImageOption;
const OciVcnIpNativeClusterPodNetworkOptionDetails = __importStar(require("./oci-vcn-ip-native-cluster-pod-network-option-details"));
exports.OciVcnIpNativeClusterPodNetworkOptionDetails = OciVcnIpNativeClusterPodNetworkOptionDetails.OciVcnIpNativeClusterPodNetworkOptionDetails;
const OciVcnIpNativeNodePoolPodNetworkOptionDetails = __importStar(require("./oci-vcn-ip-native-node-pool-pod-network-option-details"));
exports.OciVcnIpNativeNodePoolPodNetworkOptionDetails = OciVcnIpNativeNodePoolPodNetworkOptionDetails.OciVcnIpNativeNodePoolPodNetworkOptionDetails;
const TerminatePreemptionAction = __importStar(require("./terminate-preemption-action"));
exports.TerminatePreemptionAction = TerminatePreemptionAction.TerminatePreemptionAction;
