// oci-containerengine/lib/model/node-pool-lifecycle-state.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodePoolLifecycleState = void 0;
/**
 * The lifecycle state of a cluster.
 **/
var NodePoolLifecycleState;
(function (NodePoolLifecycleState) {
    NodePoolLifecycleState["Deleted"] = "DELETED";
    NodePoolLifecycleState["Creating"] = "CREATING";
    NodePoolLifecycleState["Active"] = "ACTIVE";
    NodePoolLifecycleState["Updating"] = "UPDATING";
    NodePoolLifecycleState["Deleting"] = "DELETING";
    NodePoolLifecycleState["Failed"] = "FAILED";
    NodePoolLifecycleState["Inactive"] = "INACTIVE";
    NodePoolLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NodePoolLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(NodePoolLifecycleState = exports.NodePoolLifecycleState || (exports.NodePoolLifecycleState = {}));
(function (NodePoolLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    NodePoolLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NodePoolLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(NodePoolLifecycleState = exports.NodePoolLifecycleState || (exports.NodePoolLifecycleState = {}));
