// oci-containerengine/lib/model/virtual-node-pool-lifecycle-state.js
"use strict";
/**
 * Container Engine for Kubernetes API
 * API for the Container Engine for Kubernetes service. Use this API to build, deploy,
and manage cloud-native applications. For more information, see
[Overview of Container Engine for Kubernetes](/iaas/Content/ContEng/Concepts/contengoverview.htm).

 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNodePoolLifecycleState = void 0;
/**
 * The state of the Virtual Node Pool.
 **/
var VirtualNodePoolLifecycleState;
(function (VirtualNodePoolLifecycleState) {
    VirtualNodePoolLifecycleState["Creating"] = "CREATING";
    VirtualNodePoolLifecycleState["Active"] = "ACTIVE";
    VirtualNodePoolLifecycleState["Updating"] = "UPDATING";
    VirtualNodePoolLifecycleState["Deleting"] = "DELETING";
    VirtualNodePoolLifecycleState["Deleted"] = "DELETED";
    VirtualNodePoolLifecycleState["Failed"] = "FAILED";
    VirtualNodePoolLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    VirtualNodePoolLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(VirtualNodePoolLifecycleState = exports.VirtualNodePoolLifecycleState || (exports.VirtualNodePoolLifecycleState = {}));
(function (VirtualNodePoolLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    VirtualNodePoolLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    VirtualNodePoolLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(VirtualNodePoolLifecycleState = exports.VirtualNodePoolLifecycleState || (exports.VirtualNodePoolLifecycleState = {}));
