// oci-goldengate/lib/model/cancel-deployment-backup-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CancelDeploymentBackupType = void 0;
/**
 * The deployment backup cancel type.
 *
 **/
var CancelDeploymentBackupType;
(function (CancelDeploymentBackupType) {
    CancelDeploymentBackupType["Default"] = "DEFAULT";
})(CancelDeploymentBackupType = exports.CancelDeploymentBackupType || (exports.CancelDeploymentBackupType = {}));
(function (CancelDeploymentBackupType) {
    function getJsonObj(obj) {
        return obj;
    }
    CancelDeploymentBackupType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CancelDeploymentBackupType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CancelDeploymentBackupType = exports.CancelDeploymentBackupType || (exports.CancelDeploymentBackupType = {}));
