// oci-goldengate/lib/model/connection-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionType = void 0;
/**
 * The connection type.
 *
 **/
var ConnectionType;
(function (ConnectionType) {
    ConnectionType["Goldengate"] = "GOLDENGATE";
    ConnectionType["Kafka"] = "KAFKA";
    ConnectionType["KafkaSchemaRegistry"] = "KAFKA_SCHEMA_REGISTRY";
    ConnectionType["Mysql"] = "MYSQL";
    ConnectionType["JavaMessageService"] = "JAVA_MESSAGE_SERVICE";
    ConnectionType["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
    ConnectionType["OciObjectStorage"] = "OCI_OBJECT_STORAGE";
    ConnectionType["Oracle"] = "ORACLE";
    ConnectionType["AzureDataLakeStorage"] = "AZURE_DATA_LAKE_STORAGE";
    ConnectionType["Postgresql"] = "POSTGRESQL";
    ConnectionType["AzureSynapseAnalytics"] = "AZURE_SYNAPSE_ANALYTICS";
    ConnectionType["Snowflake"] = "SNOWFLAKE";
    ConnectionType["AmazonS3"] = "AMAZON_S3";
    ConnectionType["Hdfs"] = "HDFS";
    ConnectionType["OracleNosql"] = "ORACLE_NOSQL";
    ConnectionType["Mongodb"] = "MONGODB";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConnectionType["UnknownValue"] = "UNKNOWN_VALUE";
})(ConnectionType = exports.ConnectionType || (exports.ConnectionType = {}));
(function (ConnectionType) {
    function getJsonObj(obj) {
        return obj;
    }
    ConnectionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ConnectionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConnectionType = exports.ConnectionType || (exports.ConnectionType = {}));
