// oci-goldengate/lib/model/create-database-registration-details.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDatabaseRegistrationDetails = void 0;
var CreateDatabaseRegistrationDetails;
(function (CreateDatabaseRegistrationDetails) {
    let SessionMode;
    (function (SessionMode) {
        SessionMode["Direct"] = "DIRECT";
        SessionMode["Redirect"] = "REDIRECT";
    })(SessionMode = CreateDatabaseRegistrationDetails.SessionMode || (CreateDatabaseRegistrationDetails.SessionMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDatabaseRegistrationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDatabaseRegistrationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDatabaseRegistrationDetails = exports.CreateDatabaseRegistrationDetails || (exports.CreateDatabaseRegistrationDetails = {}));
