// oci-goldengate/lib/model/database-registration.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseRegistration = void 0;
var DatabaseRegistration;
(function (DatabaseRegistration) {
    let SessionMode;
    (function (SessionMode) {
        SessionMode["Direct"] = "DIRECT";
        SessionMode["Redirect"] = "REDIRECT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SessionMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(SessionMode = DatabaseRegistration.SessionMode || (DatabaseRegistration.SessionMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseRegistration.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DatabaseRegistration.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseRegistration = exports.DatabaseRegistration || (exports.DatabaseRegistration = {}));
