// oci-goldengate/lib/model/deployment-upgrade-lifecycle-state.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentUpgradeLifecycleState = void 0;
/**
 * Possible deploymentUpgrade lifecycle states.
 *
 **/
var DeploymentUpgradeLifecycleState;
(function (DeploymentUpgradeLifecycleState) {
    DeploymentUpgradeLifecycleState["Waiting"] = "WAITING";
    DeploymentUpgradeLifecycleState["InProgress"] = "IN_PROGRESS";
    DeploymentUpgradeLifecycleState["Failed"] = "FAILED";
    DeploymentUpgradeLifecycleState["Succeeded"] = "SUCCEEDED";
    DeploymentUpgradeLifecycleState["Canceling"] = "CANCELING";
    DeploymentUpgradeLifecycleState["Canceled"] = "CANCELED";
    DeploymentUpgradeLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
})(DeploymentUpgradeLifecycleState = exports.DeploymentUpgradeLifecycleState || (exports.DeploymentUpgradeLifecycleState = {}));
(function (DeploymentUpgradeLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentUpgradeLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentUpgradeLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentUpgradeLifecycleState = exports.DeploymentUpgradeLifecycleState || (exports.DeploymentUpgradeLifecycleState = {}));
