// oci-goldengate/lib/model/deployment-diagnostic-state.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentDiagnosticState = void 0;
/**
 * The Possible DeploymentDiagnosticState that applies to a Deployment Diagnostic collection.
 *
 **/
var DeploymentDiagnosticState;
(function (DeploymentDiagnosticState) {
    DeploymentDiagnosticState["InProgress"] = "IN_PROGRESS";
    DeploymentDiagnosticState["Succeeded"] = "SUCCEEDED";
    DeploymentDiagnosticState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DeploymentDiagnosticState["UnknownValue"] = "UNKNOWN_VALUE";
})(DeploymentDiagnosticState = exports.DeploymentDiagnosticState || (exports.DeploymentDiagnosticState = {}));
(function (DeploymentDiagnosticState) {
    function getJsonObj(obj) {
        return obj;
    }
    DeploymentDiagnosticState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeploymentDiagnosticState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentDiagnosticState = exports.DeploymentDiagnosticState || (exports.DeploymentDiagnosticState = {}));
