// oci-goldengate/lib/model/deployment-type-summary.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentTypeSummary = void 0;
var DeploymentTypeSummary;
(function (DeploymentTypeSummary) {
    let Category;
    (function (Category) {
        Category["DataReplication"] = "DATA_REPLICATION";
        Category["StreamAnalytics"] = "STREAM_ANALYTICS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Category["UnknownValue"] = "UNKNOWN_VALUE";
    })(Category = DeploymentTypeSummary.Category || (DeploymentTypeSummary.Category = {}));
    let ConnectionTypes;
    (function (ConnectionTypes) {
        ConnectionTypes["Goldengate"] = "GOLDENGATE";
        ConnectionTypes["Kafka"] = "KAFKA";
        ConnectionTypes["KafkaSchemaRegistry"] = "KAFKA_SCHEMA_REGISTRY";
        ConnectionTypes["Mysql"] = "MYSQL";
        ConnectionTypes["JavaMessageService"] = "JAVA_MESSAGE_SERVICE";
        ConnectionTypes["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
        ConnectionTypes["OciObjectStorage"] = "OCI_OBJECT_STORAGE";
        ConnectionTypes["Oracle"] = "ORACLE";
        ConnectionTypes["AzureDataLakeStorage"] = "AZURE_DATA_LAKE_STORAGE";
        ConnectionTypes["Postgresql"] = "POSTGRESQL";
        ConnectionTypes["AzureSynapseAnalytics"] = "AZURE_SYNAPSE_ANALYTICS";
        ConnectionTypes["Snowflake"] = "SNOWFLAKE";
        ConnectionTypes["AmazonS3"] = "AMAZON_S3";
        ConnectionTypes["Hdfs"] = "HDFS";
        ConnectionTypes["OracleNosql"] = "ORACLE_NOSQL";
        ConnectionTypes["Mongodb"] = "MONGODB";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ConnectionTypes["UnknownValue"] = "UNKNOWN_VALUE";
    })(ConnectionTypes = DeploymentTypeSummary.ConnectionTypes || (DeploymentTypeSummary.ConnectionTypes = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeploymentTypeSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeploymentTypeSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeploymentTypeSummary = exports.DeploymentTypeSummary || (exports.DeploymentTypeSummary = {}));
