// oci-goldengate/lib/model/lifecycle-sub-state.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleSubState = void 0;
/**
 * Possible lifecycle sub-states.
 *
 **/
var LifecycleSubState;
(function (LifecycleSubState) {
    LifecycleSubState["Recovering"] = "RECOVERING";
    LifecycleSubState["Starting"] = "STARTING";
    LifecycleSubState["Stopping"] = "STOPPING";
    LifecycleSubState["Moving"] = "MOVING";
    LifecycleSubState["Upgrading"] = "UPGRADING";
    LifecycleSubState["Restoring"] = "RESTORING";
    LifecycleSubState["BackupInProgress"] = "BACKUP_IN_PROGRESS";
    LifecycleSubState["RollbackInProgress"] = "ROLLBACK_IN_PROGRESS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LifecycleSubState["UnknownValue"] = "UNKNOWN_VALUE";
})(LifecycleSubState = exports.LifecycleSubState || (exports.LifecycleSubState = {}));
(function (LifecycleSubState) {
    function getJsonObj(obj) {
        return obj;
    }
    LifecycleSubState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LifecycleSubState.getDeserializedJsonObj = getDeserializedJsonObj;
})(LifecycleSubState = exports.LifecycleSubState || (exports.LifecycleSubState = {}));
