// oci-goldengate/lib/model/microsoft-sqlserver-connection.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MicrosoftSqlserverConnection = void 0;
const model = __importStar(require("../model"));
var MicrosoftSqlserverConnection;
(function (MicrosoftSqlserverConnection) {
    let TechnologyType;
    (function (TechnologyType) {
        TechnologyType["AmazonRdsSqlserver"] = "AMAZON_RDS_SQLSERVER";
        TechnologyType["AzureSqlserverManagedInstance"] = "AZURE_SQLSERVER_MANAGED_INSTANCE";
        TechnologyType["AzureSqlserverNonManagedInstance"] = "AZURE_SQLSERVER_NON_MANAGED_INSTANCE";
        TechnologyType["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TechnologyType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TechnologyType = MicrosoftSqlserverConnection.TechnologyType || (MicrosoftSqlserverConnection.TechnologyType = {}));
    let SecurityProtocol;
    (function (SecurityProtocol) {
        SecurityProtocol["Plain"] = "PLAIN";
        SecurityProtocol["Tls"] = "TLS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SecurityProtocol["UnknownValue"] = "UNKNOWN_VALUE";
    })(SecurityProtocol = MicrosoftSqlserverConnection.SecurityProtocol || (MicrosoftSqlserverConnection.SecurityProtocol = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.Connection.getJsonObj(obj))), {
            "additionalAttributes": obj.additionalAttributes
                ? obj.additionalAttributes.map(item => {
                    return model.NameValuePair.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    MicrosoftSqlserverConnection.getJsonObj = getJsonObj;
    MicrosoftSqlserverConnection.connectionType = "MICROSOFT_SQLSERVER";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.Connection.getDeserializedJsonObj(obj))), {
            "additionalAttributes": obj.additionalAttributes
                ? obj.additionalAttributes.map(item => {
                    return model.NameValuePair.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    MicrosoftSqlserverConnection.getDeserializedJsonObj = getDeserializedJsonObj;
})(MicrosoftSqlserverConnection = exports.MicrosoftSqlserverConnection || (exports.MicrosoftSqlserverConnection = {}));
