// oci-goldengate/lib/model/oracle-connection.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OracleConnection = void 0;
const model = __importStar(require("../model"));
var OracleConnection;
(function (OracleConnection) {
    let TechnologyType;
    (function (TechnologyType) {
        TechnologyType["AmazonRdsOracle"] = "AMAZON_RDS_ORACLE";
        TechnologyType["OciAutonomousDatabase"] = "OCI_AUTONOMOUS_DATABASE";
        TechnologyType["OracleDatabase"] = "ORACLE_DATABASE";
        TechnologyType["OracleExadata"] = "ORACLE_EXADATA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TechnologyType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TechnologyType = OracleConnection.TechnologyType || (OracleConnection.TechnologyType = {}));
    let SessionMode;
    (function (SessionMode) {
        SessionMode["Direct"] = "DIRECT";
        SessionMode["Redirect"] = "REDIRECT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SessionMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(SessionMode = OracleConnection.SessionMode || (OracleConnection.SessionMode = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.Connection.getJsonObj(obj))), {});
        return jsonObj;
    }
    OracleConnection.getJsonObj = getJsonObj;
    OracleConnection.connectionType = "ORACLE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.Connection.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    OracleConnection.getDeserializedJsonObj = getDeserializedJsonObj;
})(OracleConnection = exports.OracleConnection || (exports.OracleConnection = {}));
