// oci-goldengate/lib/model/postgresql-connection.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostgresqlConnection = void 0;
const model = __importStar(require("../model"));
var PostgresqlConnection;
(function (PostgresqlConnection) {
    let TechnologyType;
    (function (TechnologyType) {
        TechnologyType["PostgresqlServer"] = "POSTGRESQL_SERVER";
        TechnologyType["AmazonAuroraPostgresql"] = "AMAZON_AURORA_POSTGRESQL";
        TechnologyType["AmazonRdsPostgresql"] = "AMAZON_RDS_POSTGRESQL";
        TechnologyType["AzurePostgresql"] = "AZURE_POSTGRESQL";
        TechnologyType["GoogleCloudSqlPostgresql"] = "GOOGLE_CLOUD_SQL_POSTGRESQL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TechnologyType["UnknownValue"] = "UNKNOWN_VALUE";
    })(TechnologyType = PostgresqlConnection.TechnologyType || (PostgresqlConnection.TechnologyType = {}));
    let SecurityProtocol;
    (function (SecurityProtocol) {
        SecurityProtocol["Plain"] = "PLAIN";
        SecurityProtocol["Tls"] = "TLS";
        SecurityProtocol["Mtls"] = "MTLS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SecurityProtocol["UnknownValue"] = "UNKNOWN_VALUE";
    })(SecurityProtocol = PostgresqlConnection.SecurityProtocol || (PostgresqlConnection.SecurityProtocol = {}));
    let SslMode;
    (function (SslMode) {
        SslMode["Prefer"] = "PREFER";
        SslMode["Require"] = "REQUIRE";
        SslMode["VerifyCa"] = "VERIFY_CA";
        SslMode["VerifyFull"] = "VERIFY_FULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SslMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(SslMode = PostgresqlConnection.SslMode || (PostgresqlConnection.SslMode = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.Connection.getJsonObj(obj))), {
            "additionalAttributes": obj.additionalAttributes
                ? obj.additionalAttributes.map(item => {
                    return model.NameValuePair.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    PostgresqlConnection.getJsonObj = getJsonObj;
    PostgresqlConnection.connectionType = "POSTGRESQL";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.Connection.getDeserializedJsonObj(obj))), {
            "additionalAttributes": obj.additionalAttributes
                ? obj.additionalAttributes.map(item => {
                    return model.NameValuePair.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    PostgresqlConnection.getDeserializedJsonObj = getDeserializedJsonObj;
})(PostgresqlConnection = exports.PostgresqlConnection || (exports.PostgresqlConnection = {}));
