// oci-goldengate/lib/model/release-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReleaseType = void 0;
/**
 * The type of release.
 *
 **/
var ReleaseType;
(function (ReleaseType) {
    ReleaseType["Major"] = "MAJOR";
    ReleaseType["Bundle"] = "BUNDLE";
    ReleaseType["Minor"] = "MINOR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReleaseType["UnknownValue"] = "UNKNOWN_VALUE";
})(ReleaseType = exports.ReleaseType || (exports.ReleaseType = {}));
(function (ReleaseType) {
    function getJsonObj(obj) {
        return obj;
    }
    ReleaseType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReleaseType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReleaseType = exports.ReleaseType || (exports.ReleaseType = {}));
