// oci-goldengate/lib/model/technology-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TechnologyType = void 0;
/**
 * The technology type.
 *
 **/
var TechnologyType;
(function (TechnologyType) {
    TechnologyType["Goldengate"] = "GOLDENGATE";
    TechnologyType["OciAutonomousDatabase"] = "OCI_AUTONOMOUS_DATABASE";
    TechnologyType["OciAutonomousJsonDatabase"] = "OCI_AUTONOMOUS_JSON_DATABASE";
    TechnologyType["OciMysql"] = "OCI_MYSQL";
    TechnologyType["OciObjectStorage"] = "OCI_OBJECT_STORAGE";
    TechnologyType["OciStreaming"] = "OCI_STREAMING";
    TechnologyType["OracleDatabase"] = "ORACLE_DATABASE";
    TechnologyType["OracleExadata"] = "ORACLE_EXADATA";
    TechnologyType["OracleNosql"] = "ORACLE_NOSQL";
    TechnologyType["OracleWeblogicJms"] = "ORACLE_WEBLOGIC_JMS";
    TechnologyType["AmazonRdsOracle"] = "AMAZON_RDS_ORACLE";
    TechnologyType["AmazonRdsSqlserver"] = "AMAZON_RDS_SQLSERVER";
    TechnologyType["AmazonS3"] = "AMAZON_S3";
    TechnologyType["AmazonAuroraMysql"] = "AMAZON_AURORA_MYSQL";
    TechnologyType["AmazonAuroraPostgresql"] = "AMAZON_AURORA_POSTGRESQL";
    TechnologyType["AmazonRdsMariadb"] = "AMAZON_RDS_MARIADB";
    TechnologyType["AmazonRdsMysql"] = "AMAZON_RDS_MYSQL";
    TechnologyType["AmazonRdsPostgresql"] = "AMAZON_RDS_POSTGRESQL";
    TechnologyType["ApacheKafka"] = "APACHE_KAFKA";
    TechnologyType["AzureCosmosDbForMongodb"] = "AZURE_COSMOS_DB_FOR_MONGODB";
    TechnologyType["AzureDataLakeStorage"] = "AZURE_DATA_LAKE_STORAGE";
    TechnologyType["AzureEventHubs"] = "AZURE_EVENT_HUBS";
    TechnologyType["AzureMysql"] = "AZURE_MYSQL";
    TechnologyType["AzurePostgresql"] = "AZURE_POSTGRESQL";
    TechnologyType["AzureSqlserverManagedInstance"] = "AZURE_SQLSERVER_MANAGED_INSTANCE";
    TechnologyType["AzureSqlserverNonManagedInstance"] = "AZURE_SQLSERVER_NON_MANAGED_INSTANCE";
    TechnologyType["AzureSynapseAnalytics"] = "AZURE_SYNAPSE_ANALYTICS";
    TechnologyType["ConfluentKafka"] = "CONFLUENT_KAFKA";
    TechnologyType["ConfluentSchemaRegistry"] = "CONFLUENT_SCHEMA_REGISTRY";
    TechnologyType["GoogleCloudSqlMysql"] = "GOOGLE_CLOUD_SQL_MYSQL";
    TechnologyType["GoogleCloudSqlPostgresql"] = "GOOGLE_CLOUD_SQL_POSTGRESQL";
    TechnologyType["Hdfs"] = "HDFS";
    TechnologyType["Mariadb"] = "MARIADB";
    TechnologyType["MicrosoftSqlserver"] = "MICROSOFT_SQLSERVER";
    TechnologyType["Mongodb"] = "MONGODB";
    TechnologyType["MysqlServer"] = "MYSQL_SERVER";
    TechnologyType["PostgresqlServer"] = "POSTGRESQL_SERVER";
    TechnologyType["Snowflake"] = "SNOWFLAKE";
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
(function (TechnologyType) {
    function getJsonObj(obj) {
        return obj;
    }
    TechnologyType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TechnologyType.getDeserializedJsonObj = getDeserializedJsonObj;
})(TechnologyType = exports.TechnologyType || (exports.TechnologyType = {}));
