// oci-goldengate/lib/model/upgrade-deployment-upgrade-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpgradeDeploymentUpgradeType = void 0;
/**
 * The upgrade type of deploymentUpgrade.
 *
 **/
var UpgradeDeploymentUpgradeType;
(function (UpgradeDeploymentUpgradeType) {
    UpgradeDeploymentUpgradeType["Default"] = "DEFAULT";
})(UpgradeDeploymentUpgradeType = exports.UpgradeDeploymentUpgradeType || (exports.UpgradeDeploymentUpgradeType = {}));
(function (UpgradeDeploymentUpgradeType) {
    function getJsonObj(obj) {
        return obj;
    }
    UpgradeDeploymentUpgradeType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UpgradeDeploymentUpgradeType.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpgradeDeploymentUpgradeType = exports.UpgradeDeploymentUpgradeType || (exports.UpgradeDeploymentUpgradeType = {}));
