// oci-announcementsservice/lib/model/filter.js
"use strict";
/**
 * Announcements Service API
 * Manage Oracle Cloud Infrastructure console announcements.
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Filter = void 0;
var Filter;
(function (Filter) {
    let Type;
    (function (Type) {
        Type["CompartmentId"] = "COMPARTMENT_ID";
        Type["PlatformType"] = "PLATFORM_TYPE";
        Type["Region"] = "REGION";
        Type["Service"] = "SERVICE";
        Type["ResourceId"] = "RESOURCE_ID";
        Type["AnnouncementType"] = "ANNOUNCEMENT_TYPE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = Filter.Type || (Filter.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Filter.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Filter.getDeserializedJsonObj = getDeserializedJsonObj;
})(Filter = exports.Filter || (exports.Filter = {}));
