// oci-announcementsservice/lib/model/index.js
"use strict";
/**
 * Announcements Service API
 * Manage Oracle Cloud Infrastructure console announcements.
 * OpenAPI spec version: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAnnouncementsPreferencesDetails = exports.CreateAnnouncementsPreferencesDetails = exports.AnnouncementsPreferencesSummary = exports.AnnouncementsPreferences = exports.AnnouncementSummary = exports.Announcement = exports.UpdateFilterGroupDetails = exports.UpdateAnnouncementSubscriptionDetails = exports.SortOrder = exports.Property = exports.FilterGroupDetails = exports.FilterGroup = exports.Filter = exports.CreateFilterGroupDetails = exports.CreateAnnouncementSubscriptionDetails = exports.ChangeAnnouncementSubscriptionCompartmentDetails = exports.BaseCreateAnnouncementsPreferencesDetails = exports.BaseAnnouncementsPreferences = exports.BaseAnnouncement = exports.AnnouncementsCollection = exports.AnnouncementUserStatusDetails = exports.AnnouncementSubscriptionSummary = exports.AnnouncementSubscriptionCollection = exports.AnnouncementSubscription = exports.AffectedResource = void 0;
const AffectedResource = __importStar(require("./affected-resource"));
exports.AffectedResource = AffectedResource.AffectedResource;
const AnnouncementSubscription = __importStar(require("./announcement-subscription"));
exports.AnnouncementSubscription = AnnouncementSubscription.AnnouncementSubscription;
const AnnouncementSubscriptionCollection = __importStar(require("./announcement-subscription-collection"));
exports.AnnouncementSubscriptionCollection = AnnouncementSubscriptionCollection.AnnouncementSubscriptionCollection;
const AnnouncementSubscriptionSummary = __importStar(require("./announcement-subscription-summary"));
exports.AnnouncementSubscriptionSummary = AnnouncementSubscriptionSummary.AnnouncementSubscriptionSummary;
const AnnouncementUserStatusDetails = __importStar(require("./announcement-user-status-details"));
exports.AnnouncementUserStatusDetails = AnnouncementUserStatusDetails.AnnouncementUserStatusDetails;
const AnnouncementsCollection = __importStar(require("./announcements-collection"));
exports.AnnouncementsCollection = AnnouncementsCollection.AnnouncementsCollection;
const BaseAnnouncement = __importStar(require("./base-announcement"));
exports.BaseAnnouncement = BaseAnnouncement.BaseAnnouncement;
const BaseAnnouncementsPreferences = __importStar(require("./base-announcements-preferences"));
exports.BaseAnnouncementsPreferences = BaseAnnouncementsPreferences.BaseAnnouncementsPreferences;
const BaseCreateAnnouncementsPreferencesDetails = __importStar(require("./base-create-announcements-preferences-details"));
exports.BaseCreateAnnouncementsPreferencesDetails = BaseCreateAnnouncementsPreferencesDetails.BaseCreateAnnouncementsPreferencesDetails;
const ChangeAnnouncementSubscriptionCompartmentDetails = __importStar(require("./change-announcement-subscription-compartment-details"));
exports.ChangeAnnouncementSubscriptionCompartmentDetails = ChangeAnnouncementSubscriptionCompartmentDetails.ChangeAnnouncementSubscriptionCompartmentDetails;
const CreateAnnouncementSubscriptionDetails = __importStar(require("./create-announcement-subscription-details"));
exports.CreateAnnouncementSubscriptionDetails = CreateAnnouncementSubscriptionDetails.CreateAnnouncementSubscriptionDetails;
const CreateFilterGroupDetails = __importStar(require("./create-filter-group-details"));
exports.CreateFilterGroupDetails = CreateFilterGroupDetails.CreateFilterGroupDetails;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const FilterGroup = __importStar(require("./filter-group"));
exports.FilterGroup = FilterGroup.FilterGroup;
const FilterGroupDetails = __importStar(require("./filter-group-details"));
exports.FilterGroupDetails = FilterGroupDetails.FilterGroupDetails;
const Property = __importStar(require("./property"));
exports.Property = Property.Property;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateAnnouncementSubscriptionDetails = __importStar(require("./update-announcement-subscription-details"));
exports.UpdateAnnouncementSubscriptionDetails = UpdateAnnouncementSubscriptionDetails.UpdateAnnouncementSubscriptionDetails;
const UpdateFilterGroupDetails = __importStar(require("./update-filter-group-details"));
exports.UpdateFilterGroupDetails = UpdateFilterGroupDetails.UpdateFilterGroupDetails;
const Announcement = __importStar(require("./announcement"));
exports.Announcement = Announcement.Announcement;
const AnnouncementSummary = __importStar(require("./announcement-summary"));
exports.AnnouncementSummary = AnnouncementSummary.AnnouncementSummary;
const AnnouncementsPreferences = __importStar(require("./announcements-preferences"));
exports.AnnouncementsPreferences = AnnouncementsPreferences.AnnouncementsPreferences;
const AnnouncementsPreferencesSummary = __importStar(require("./announcements-preferences-summary"));
exports.AnnouncementsPreferencesSummary = AnnouncementsPreferencesSummary.AnnouncementsPreferencesSummary;
const CreateAnnouncementsPreferencesDetails = __importStar(require("./create-announcements-preferences-details"));
exports.CreateAnnouncementsPreferencesDetails = CreateAnnouncementsPreferencesDetails.CreateAnnouncementsPreferencesDetails;
const UpdateAnnouncementsPreferencesDetails = __importStar(require("./update-announcements-preferences-details"));
exports.UpdateAnnouncementsPreferencesDetails = UpdateAnnouncementsPreferencesDetails.UpdateAnnouncementsPreferencesDetails;
