// oci-containerengine/lib/request/list-virtual-nodes-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20180222
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVirtualNodesRequest = void 0;
var ListVirtualNodesRequest;
(function (ListVirtualNodesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Id"] = "ID";
        SortBy["Name"] = "NAME";
        SortBy["TimeCreated"] = "TIME_CREATED";
    })(SortBy = ListVirtualNodesRequest.SortBy || (ListVirtualNodesRequest.SortBy = {}));
})(ListVirtualNodesRequest = exports.ListVirtualNodesRequest || (exports.ListVirtualNodesRequest = {}));
