// oci-usage/lib/model/index.js
"use strict";
/**
 * Usage Proxy API
 * Use the Usage Proxy API to list Oracle Support Rewards, view related detailed usage information, and manage users who redeem rewards. For more information, see [Oracle Support Rewards Overview](/iaas/Content/Billing/Concepts/supportrewardsoverview.htm).
 * OpenAPI spec version: 20190111
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RewardDetails = exports.RewardCollection = exports.RedemptionSummary = exports.RedemptionCollection = exports.RedeemableUserSummary = exports.RedeemableUserCollection = exports.RedeemableUser = exports.ProductSummary = exports.ProductCollection = exports.MonthlyRewardSummary = exports.CreateRedeemableUserDetails = void 0;
const CreateRedeemableUserDetails = __importStar(require("./create-redeemable-user-details"));
exports.CreateRedeemableUserDetails = CreateRedeemableUserDetails.CreateRedeemableUserDetails;
const MonthlyRewardSummary = __importStar(require("./monthly-reward-summary"));
exports.MonthlyRewardSummary = MonthlyRewardSummary.MonthlyRewardSummary;
const ProductCollection = __importStar(require("./product-collection"));
exports.ProductCollection = ProductCollection.ProductCollection;
const ProductSummary = __importStar(require("./product-summary"));
exports.ProductSummary = ProductSummary.ProductSummary;
const RedeemableUser = __importStar(require("./redeemable-user"));
exports.RedeemableUser = RedeemableUser.RedeemableUser;
const RedeemableUserCollection = __importStar(require("./redeemable-user-collection"));
exports.RedeemableUserCollection = RedeemableUserCollection.RedeemableUserCollection;
const RedeemableUserSummary = __importStar(require("./redeemable-user-summary"));
exports.RedeemableUserSummary = RedeemableUserSummary.RedeemableUserSummary;
const RedemptionCollection = __importStar(require("./redemption-collection"));
exports.RedemptionCollection = RedemptionCollection.RedemptionCollection;
const RedemptionSummary = __importStar(require("./redemption-summary"));
exports.RedemptionSummary = RedemptionSummary.RedemptionSummary;
const RewardCollection = __importStar(require("./reward-collection"));
exports.RewardCollection = RewardCollection.RewardCollection;
const RewardDetails = __importStar(require("./reward-details"));
exports.RewardDetails = RewardDetails.RewardDetails;
