// oci-resourcemanager/lib/model/configuration-source-provider.js
"use strict";
/**
 * Resource Manager API
 * Use the Resource Manager API to automate deployment and operations for all Oracle Cloud Infrastructure resources.
Using the infrastructure-as-code (IaC) model, the service is based on Terraform, an open source industry standard that lets DevOps engineers develop and deploy their infrastructure anywhere.
For more information, see
[the Resource Manager documentation](/iaas/Content/ResourceManager/home.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationSourceProvider = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ConfigurationSourceProvider;
(function (ConfigurationSourceProvider) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ConfigurationSourceProvider.LifecycleState || (ConfigurationSourceProvider.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "privateServerConfigDetails": obj.privateServerConfigDetails
                ? model.PrivateServerConfigDetails.getJsonObj(obj.privateServerConfigDetails)
                : undefined
        });
        if (obj && "configSourceProviderType" in obj && obj.configSourceProviderType) {
            switch (obj.configSourceProviderType) {
                case "GITHUB_ACCESS_TOKEN":
                    return model.GithubAccessTokenConfigurationSourceProvider.getJsonObj(jsonObj, true);
                case "GITLAB_ACCESS_TOKEN":
                    return model.GitlabAccessTokenConfigurationSourceProvider.getJsonObj(jsonObj, true);
                case "BITBUCKET_SERVER_ACCESS_TOKEN":
                    return model.BitbucketServerAccessTokenConfigurationSourceProvider.getJsonObj(jsonObj, true);
                case "BITBUCKET_CLOUD_USERNAME_APPPASSWORD":
                    return model.BitbucketCloudUsernameAppPasswordConfigurationSourceProvider.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.configSourceProviderType}`);
            }
        }
        return jsonObj;
    }
    ConfigurationSourceProvider.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "privateServerConfigDetails": obj.privateServerConfigDetails
                ? model.PrivateServerConfigDetails.getDeserializedJsonObj(obj.privateServerConfigDetails)
                : undefined
        });
        if (obj && "configSourceProviderType" in obj && obj.configSourceProviderType) {
            switch (obj.configSourceProviderType) {
                case "GITHUB_ACCESS_TOKEN":
                    return model.GithubAccessTokenConfigurationSourceProvider.getDeserializedJsonObj(jsonObj, true);
                case "GITLAB_ACCESS_TOKEN":
                    return model.GitlabAccessTokenConfigurationSourceProvider.getDeserializedJsonObj(jsonObj, true);
                case "BITBUCKET_SERVER_ACCESS_TOKEN":
                    return model.BitbucketServerAccessTokenConfigurationSourceProvider.getDeserializedJsonObj(jsonObj, true);
                case "BITBUCKET_CLOUD_USERNAME_APPPASSWORD":
                    return model.BitbucketCloudUsernameAppPasswordConfigurationSourceProvider.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.configSourceProviderType}`);
            }
        }
        return jsonObj;
    }
    ConfigurationSourceProvider.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConfigurationSourceProvider = exports.ConfigurationSourceProvider || (exports.ConfigurationSourceProvider = {}));
