// oci-resourcemanager/lib/model/failure-details.js
"use strict";
/**
 * Resource Manager API
 * Use the Resource Manager API to automate deployment and operations for all Oracle Cloud Infrastructure resources.
Using the infrastructure-as-code (IaC) model, the service is based on Terraform, an open source industry standard that lets DevOps engineers develop and deploy their infrastructure anywhere.
For more information, see
[the Resource Manager documentation](/iaas/Content/ResourceManager/home.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FailureDetails = void 0;
var FailureDetails;
(function (FailureDetails) {
    let Code;
    (function (Code) {
        Code["InternalServiceError"] = "INTERNAL_SERVICE_ERROR";
        Code["TerraformExecutionError"] = "TERRAFORM_EXECUTION_ERROR";
        Code["TerraformConfigUnzipFailed"] = "TERRAFORM_CONFIG_UNZIP_FAILED";
        Code["InvalidWorkingDirectory"] = "INVALID_WORKING_DIRECTORY";
        Code["JobTimeout"] = "JOB_TIMEOUT";
        Code["TerraformConfigVirusFound"] = "TERRAFORM_CONFIG_VIRUS_FOUND";
        Code["TerraformGitCloneFailure"] = "TERRAFORM_GIT_CLONE_FAILURE";
        Code["TerraformGitCheckoutFailure"] = "TERRAFORM_GIT_CHECKOUT_FAILURE";
        Code["TerraformObjectStorageConfigSourceEmptyBucket"] = "TERRAFORM_OBJECT_STORAGE_CONFIG_SOURCE_EMPTY_BUCKET";
        Code["TerraformObjectStorageConfigSourceNoTfFilePresent"] = "TERRAFORM_OBJECT_STORAGE_CONFIG_SOURCE_NO_TF_FILE_PRESENT";
        Code["TerraformObjectStorageConfigSourceUnsupportedObjectSize"] = "TERRAFORM_OBJECT_STORAGE_CONFIG_SOURCE_UNSUPPORTED_OBJECT_SIZE";
        Code["CustomTerraformProviderBucketNotFound"] = "CUSTOM_TERRAFORM_PROVIDER_BUCKET_NOT_FOUND";
        Code["CustomTerraformProviderUnsupportedObjectSize"] = "CUSTOM_TERRAFORM_PROVIDER_UNSUPPORTED_OBJECT_SIZE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Code["UnknownValue"] = "UNKNOWN_VALUE";
    })(Code = FailureDetails.Code || (FailureDetails.Code = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FailureDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FailureDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(FailureDetails = exports.FailureDetails || (exports.FailureDetails = {}));
