// oci-resourcemanager/lib/model/stack.js
"use strict";
/**
 * Resource Manager API
 * Use the Resource Manager API to automate deployment and operations for all Oracle Cloud Infrastructure resources.
Using the infrastructure-as-code (IaC) model, the service is based on Terraform, an open source industry standard that lets DevOps engineers develop and deploy their infrastructure anywhere.
For more information, see
[the Resource Manager documentation](/iaas/Content/ResourceManager/home.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stack = void 0;
const model = __importStar(require("../model"));
var Stack;
(function (Stack) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Stack.LifecycleState || (Stack.LifecycleState = {}));
    let StackDriftStatus;
    (function (StackDriftStatus) {
        StackDriftStatus["NotChecked"] = "NOT_CHECKED";
        StackDriftStatus["InSync"] = "IN_SYNC";
        StackDriftStatus["Drifted"] = "DRIFTED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StackDriftStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(StackDriftStatus = Stack.StackDriftStatus || (Stack.StackDriftStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "configSource": obj.configSource
                ? model.ConfigSource.getJsonObj(obj.configSource)
                : undefined,
            "customTerraformProvider": obj.customTerraformProvider
                ? model.CustomTerraformProvider.getJsonObj(obj.customTerraformProvider)
                : undefined
        });
        return jsonObj;
    }
    Stack.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "configSource": obj.configSource
                ? model.ConfigSource.getDeserializedJsonObj(obj.configSource)
                : undefined,
            "customTerraformProvider": obj.customTerraformProvider
                ? model.CustomTerraformProvider.getDeserializedJsonObj(obj.customTerraformProvider)
                : undefined
        });
        return jsonObj;
    }
    Stack.getDeserializedJsonObj = getDeserializedJsonObj;
})(Stack = exports.Stack || (exports.Stack = {}));
