// oci-resourcemanager/lib/model/terraform-advanced-options.js
"use strict";
/**
 * Resource Manager API
 * Use the Resource Manager API to automate deployment and operations for all Oracle Cloud Infrastructure resources.
Using the infrastructure-as-code (IaC) model, the service is based on Terraform, an open source industry standard that lets DevOps engineers develop and deploy their infrastructure anywhere.
For more information, see
[the Resource Manager documentation](/iaas/Content/ResourceManager/home.htm).

 * OpenAPI spec version: 20180917
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerraformAdvancedOptions = void 0;
var TerraformAdvancedOptions;
(function (TerraformAdvancedOptions) {
    let DetailedLogLevel;
    (function (DetailedLogLevel) {
        DetailedLogLevel["Error"] = "ERROR";
        DetailedLogLevel["Warn"] = "WARN";
        DetailedLogLevel["Info"] = "INFO";
        DetailedLogLevel["Debug"] = "DEBUG";
        DetailedLogLevel["Trace"] = "TRACE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DetailedLogLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(DetailedLogLevel = TerraformAdvancedOptions.DetailedLogLevel || (TerraformAdvancedOptions.DetailedLogLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TerraformAdvancedOptions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    TerraformAdvancedOptions.getDeserializedJsonObj = getDeserializedJsonObj;
})(TerraformAdvancedOptions = exports.TerraformAdvancedOptions || (exports.TerraformAdvancedOptions = {}));
