// oci-servicemesh/lib/request/index.js
"use strict";
/**
 * Service Mesh API
 * Use the Service Mesh API to manage mesh, virtual service, access policy and other mesh related items.
 * OpenAPI spec version: 20220615
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListVirtualServicesRequest = exports.ListVirtualServiceRouteTablesRequest = exports.ListVirtualDeploymentsRequest = exports.ListMeshesRequest = exports.ListIngressGatewaysRequest = exports.ListIngressGatewayRouteTablesRequest = exports.ListAccessPoliciesRequest = void 0;
const ListAccessPoliciesRequest = __importStar(require("./list-access-policies-request"));
exports.ListAccessPoliciesRequest = ListAccessPoliciesRequest.ListAccessPoliciesRequest;
const ListIngressGatewayRouteTablesRequest = __importStar(require("./list-ingress-gateway-route-tables-request"));
exports.ListIngressGatewayRouteTablesRequest = ListIngressGatewayRouteTablesRequest.ListIngressGatewayRouteTablesRequest;
const ListIngressGatewaysRequest = __importStar(require("./list-ingress-gateways-request"));
exports.ListIngressGatewaysRequest = ListIngressGatewaysRequest.ListIngressGatewaysRequest;
const ListMeshesRequest = __importStar(require("./list-meshes-request"));
exports.ListMeshesRequest = ListMeshesRequest.ListMeshesRequest;
const ListVirtualDeploymentsRequest = __importStar(require("./list-virtual-deployments-request"));
exports.ListVirtualDeploymentsRequest = ListVirtualDeploymentsRequest.ListVirtualDeploymentsRequest;
const ListVirtualServiceRouteTablesRequest = __importStar(require("./list-virtual-service-route-tables-request"));
exports.ListVirtualServiceRouteTablesRequest = ListVirtualServiceRouteTablesRequest.ListVirtualServiceRouteTablesRequest;
const ListVirtualServicesRequest = __importStar(require("./list-virtual-services-request"));
exports.ListVirtualServicesRequest = ListVirtualServicesRequest.ListVirtualServicesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
