// oci-servicemesh/lib/request/list-ingress-gateway-route-tables-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220615
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListIngressGatewayRouteTablesRequest = void 0;
var ListIngressGatewayRouteTablesRequest;
(function (ListIngressGatewayRouteTablesRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Id"] = "id";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["Name"] = "name";
    })(SortBy = ListIngressGatewayRouteTablesRequest.SortBy || (ListIngressGatewayRouteTablesRequest.SortBy = {}));
})(ListIngressGatewayRouteTablesRequest = exports.ListIngressGatewayRouteTablesRequest || (exports.ListIngressGatewayRouteTablesRequest = {}));
