// oci-mediaservices/lib/request/list-media-workflow-job-facts-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20211101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMediaWorkflowJobFactsRequest = void 0;
var ListMediaWorkflowJobFactsRequest;
(function (ListMediaWorkflowJobFactsRequest) {
    let Type;
    (function (Type) {
        Type["RunnableJob"] = "runnableJob";
        Type["TaskDeclaration"] = "taskDeclaration";
        Type["Workflow"] = "workflow";
        Type["Configuration"] = "configuration";
        Type["ParameterResolutionEvent"] = "parameterResolutionEvent";
    })(Type = ListMediaWorkflowJobFactsRequest.Type || (ListMediaWorkflowJobFactsRequest.Type = {}));
})(ListMediaWorkflowJobFactsRequest = exports.ListMediaWorkflowJobFactsRequest || (exports.ListMediaWorkflowJobFactsRequest = {}));
