// oci-jms/lib/model/application-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSortBy = void 0;
/**
 * The field to sort applications.
 **/
var ApplicationSortBy;
(function (ApplicationSortBy) {
    ApplicationSortBy["TimeFirstSeen"] = "timeFirstSeen";
    ApplicationSortBy["TimeLastSeen"] = "timeLastSeen";
    ApplicationSortBy["DisplayName"] = "displayName";
    ApplicationSortBy["ApproximateJreCount"] = "approximateJreCount";
    ApplicationSortBy["ApproximateInstallationCount"] = "approximateInstallationCount";
    ApplicationSortBy["ApproximateManagedInstanceCount"] = "approximateManagedInstanceCount";
    ApplicationSortBy["OsName"] = "osName";
})(ApplicationSortBy = exports.ApplicationSortBy || (exports.ApplicationSortBy = {}));
(function (ApplicationSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    ApplicationSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ApplicationSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(ApplicationSortBy = exports.ApplicationSortBy || (exports.ApplicationSortBy = {}));
