// oci-jms/lib/model/artifact-content-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArtifactContentType = void 0;
/**
 * Contents of the Java Runtime artifact.
 **/
var ArtifactContentType;
(function (ArtifactContentType) {
    ArtifactContentType["Jdk"] = "JDK";
    ArtifactContentType["Jre"] = "JRE";
    ArtifactContentType["ServerJre"] = "SERVER_JRE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ArtifactContentType["UnknownValue"] = "UNKNOWN_VALUE";
})(ArtifactContentType = exports.ArtifactContentType || (exports.ArtifactContentType = {}));
(function (ArtifactContentType) {
    function getJsonObj(obj) {
        return obj;
    }
    ArtifactContentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ArtifactContentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ArtifactContentType = exports.ArtifactContentType || (exports.ArtifactContentType = {}));
