// oci-jms/lib/model/blocklist-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlocklistSortBy = void 0;
/**
 * The field to sort blocklist records.
 **/
var BlocklistSortBy;
(function (BlocklistSortBy) {
    BlocklistSortBy["Operation"] = "operation";
})(BlocklistSortBy = exports.BlocklistSortBy || (exports.BlocklistSortBy = {}));
(function (BlocklistSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    BlocklistSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BlocklistSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(BlocklistSortBy = exports.BlocklistSortBy || (exports.BlocklistSortBy = {}));
