// oci-jms/lib/model/crypto-analysis-result-mode.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CryptoAnalysisResultMode = void 0;
/**
 * The aggregation mode of the crypto analysis result.
 *
 **/
var CryptoAnalysisResultMode;
(function (CryptoAnalysisResultMode) {
    CryptoAnalysisResultMode["Jfr"] = "JFR";
    CryptoAnalysisResultMode["ManagedInstance"] = "MANAGED_INSTANCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CryptoAnalysisResultMode["UnknownValue"] = "UNKNOWN_VALUE";
})(CryptoAnalysisResultMode = exports.CryptoAnalysisResultMode || (exports.CryptoAnalysisResultMode = {}));
(function (CryptoAnalysisResultMode) {
    function getJsonObj(obj) {
        return obj;
    }
    CryptoAnalysisResultMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CryptoAnalysisResultMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(CryptoAnalysisResultMode = exports.CryptoAnalysisResultMode || (exports.CryptoAnalysisResultMode = {}));
