// oci-jms/lib/model/deployed-application-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeployedApplicationSortBy = void 0;
/**
 * The field to sort deployed applications.
 *
 **/
var DeployedApplicationSortBy;
(function (DeployedApplicationSortBy) {
    DeployedApplicationSortBy["ApplicationName"] = "applicationName";
    DeployedApplicationSortBy["ApplicationType"] = "applicationType";
    DeployedApplicationSortBy["IsClustered"] = "isClustered";
    DeployedApplicationSortBy["JavaServerInstanceCount"] = "javaServerInstanceCount";
    DeployedApplicationSortBy["TimeFirstSeen"] = "timeFirstSeen";
    DeployedApplicationSortBy["TimeLastSeen"] = "timeLastSeen";
})(DeployedApplicationSortBy = exports.DeployedApplicationSortBy || (exports.DeployedApplicationSortBy = {}));
(function (DeployedApplicationSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    DeployedApplicationSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DeployedApplicationSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeployedApplicationSortBy = exports.DeployedApplicationSortBy || (exports.DeployedApplicationSortBy = {}));
