// oci-jms/lib/model/installation-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstallationSortBy = void 0;
/**
 * The field to sort installations.
 **/
var InstallationSortBy;
(function (InstallationSortBy) {
    InstallationSortBy["JreDistribution"] = "jreDistribution";
    InstallationSortBy["JreVendor"] = "jreVendor";
    InstallationSortBy["JreVersion"] = "jreVersion";
    InstallationSortBy["Path"] = "path";
    InstallationSortBy["TimeFirstSeen"] = "timeFirstSeen";
    InstallationSortBy["TimeLastSeen"] = "timeLastSeen";
    InstallationSortBy["ApproximateApplicationCount"] = "approximateApplicationCount";
    InstallationSortBy["ApproximateManagedInstanceCount"] = "approximateManagedInstanceCount";
    InstallationSortBy["OsName"] = "osName";
})(InstallationSortBy = exports.InstallationSortBy || (exports.InstallationSortBy = {}));
(function (InstallationSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    InstallationSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InstallationSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(InstallationSortBy = exports.InstallationSortBy || (exports.InstallationSortBy = {}));
