// oci-jms/lib/model/installation-site-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstallationSiteSortBy = void 0;
/**
 * The field to sort installation sites.
 **/
var InstallationSiteSortBy;
(function (InstallationSiteSortBy) {
    InstallationSiteSortBy["ManagedInstanceId"] = "managedInstanceId";
    InstallationSiteSortBy["JreDistribution"] = "jreDistribution";
    InstallationSiteSortBy["JreVendor"] = "jreVendor";
    InstallationSiteSortBy["JreVersion"] = "jreVersion";
    InstallationSiteSortBy["Path"] = "path";
    InstallationSiteSortBy["ApproximateApplicationCount"] = "approximateApplicationCount";
    InstallationSiteSortBy["OsName"] = "osName";
    InstallationSiteSortBy["SecurityStatus"] = "securityStatus";
})(InstallationSiteSortBy = exports.InstallationSiteSortBy || (exports.InstallationSiteSortBy = {}));
(function (InstallationSiteSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    InstallationSiteSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InstallationSiteSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(InstallationSiteSortBy = exports.InstallationSiteSortBy || (exports.InstallationSiteSortBy = {}));
