// oci-jms/lib/model/java-server-instance-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JavaServerInstanceSortBy = void 0;
/**
 * The field to sort Java Server instances.
 *
 **/
var JavaServerInstanceSortBy;
(function (JavaServerInstanceSortBy) {
    JavaServerInstanceSortBy["ServerInstanceName"] = "serverInstanceName";
    JavaServerInstanceSortBy["ManagedInstanceName"] = "managedInstanceName";
    JavaServerInstanceSortBy["ApproximateDeployedApplicationCount"] = "approximateDeployedApplicationCount";
    JavaServerInstanceSortBy["TimeFirstSeen"] = "timeFirstSeen";
    JavaServerInstanceSortBy["TimeLastSeen"] = "timeLastSeen";
})(JavaServerInstanceSortBy = exports.JavaServerInstanceSortBy || (exports.JavaServerInstanceSortBy = {}));
(function (JavaServerInstanceSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    JavaServerInstanceSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    JavaServerInstanceSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(JavaServerInstanceSortBy = exports.JavaServerInstanceSortBy || (exports.JavaServerInstanceSortBy = {}));
