// oci-jms/lib/model/library-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibrarySortBy = void 0;
/**
 * The field to sort libraries.
 *
 **/
var LibrarySortBy;
(function (LibrarySortBy) {
    LibrarySortBy["ApplicationCount"] = "applicationCount";
    LibrarySortBy["JavaServerInstanceCount"] = "javaServerInstanceCount";
    LibrarySortBy["CvssScore"] = "cvssScore";
    LibrarySortBy["DeployedApplicationCount"] = "deployedApplicationCount";
    LibrarySortBy["LibraryName"] = "libraryName";
    LibrarySortBy["LibraryVersion"] = "libraryVersion";
    LibrarySortBy["ManagedInstanceCount"] = "managedInstanceCount";
    LibrarySortBy["TimeFirstSeen"] = "timeFirstSeen";
    LibrarySortBy["TimeLastSeen"] = "timeLastSeen";
})(LibrarySortBy = exports.LibrarySortBy || (exports.LibrarySortBy = {}));
(function (LibrarySortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    LibrarySortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LibrarySortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(LibrarySortBy = exports.LibrarySortBy || (exports.LibrarySortBy = {}));
