// oci-jms/lib/model/license-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseType = void 0;
/**
 * Type of license associated with the Java version.
 **/
var LicenseType;
(function (LicenseType) {
    LicenseType["Otn"] = "OTN";
    LicenseType["Nftc"] = "NFTC";
    LicenseType["Restricted"] = "RESTRICTED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LicenseType["UnknownValue"] = "UNKNOWN_VALUE";
})(LicenseType = exports.LicenseType || (exports.LicenseType = {}));
(function (LicenseType) {
    function getJsonObj(obj) {
        return obj;
    }
    LicenseType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LicenseType.getDeserializedJsonObj = getDeserializedJsonObj;
})(LicenseType = exports.LicenseType || (exports.LicenseType = {}));
