// oci-jms/lib/model/managed-instance-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedInstanceType = void 0;
/**
 * The types of managed instance.
 **/
var ManagedInstanceType;
(function (ManagedInstanceType) {
    ManagedInstanceType["OracleManagementAgent"] = "ORACLE_MANAGEMENT_AGENT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ManagedInstanceType["UnknownValue"] = "UNKNOWN_VALUE";
})(ManagedInstanceType = exports.ManagedInstanceType || (exports.ManagedInstanceType = {}));
(function (ManagedInstanceType) {
    function getJsonObj(obj) {
        return obj;
    }
    ManagedInstanceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ManagedInstanceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ManagedInstanceType = exports.ManagedInstanceType || (exports.ManagedInstanceType = {}));
