// oci-jms/lib/model/operation-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateFleet"] = "CREATE_FLEET";
    OperationType["DeleteFleet"] = "DELETE_FLEET";
    OperationType["MoveFleet"] = "MOVE_FLEET";
    OperationType["UpdateFleet"] = "UPDATE_FLEET";
    OperationType["UpdateFleetAgentConfiguration"] = "UPDATE_FLEET_AGENT_CONFIGURATION";
    OperationType["DeleteJavaInstallation"] = "DELETE_JAVA_INSTALLATION";
    OperationType["CreateJavaInstallation"] = "CREATE_JAVA_INSTALLATION";
    OperationType["CollectJfr"] = "COLLECT_JFR";
    OperationType["RequestCryptoEventAnalysis"] = "REQUEST_CRYPTO_EVENT_ANALYSIS";
    OperationType["ScanJavaServerUsage"] = "SCAN_JAVA_SERVER_USAGE";
    OperationType["ScanLibraryUsage"] = "SCAN_LIBRARY_USAGE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
