// oci-jms/lib/model/os-family.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OsFamily = void 0;
/**
 * Operating System types.
 **/
var OsFamily;
(function (OsFamily) {
    OsFamily["Linux"] = "LINUX";
    OsFamily["Windows"] = "WINDOWS";
    OsFamily["Macos"] = "MACOS";
    OsFamily["Unknown"] = "UNKNOWN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OsFamily["UnknownValue"] = "UNKNOWN_VALUE";
})(OsFamily = exports.OsFamily || (exports.OsFamily = {}));
(function (OsFamily) {
    function getJsonObj(obj) {
        return obj;
    }
    OsFamily.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OsFamily.getDeserializedJsonObj = getDeserializedJsonObj;
})(OsFamily = exports.OsFamily || (exports.OsFamily = {}));
