// oci-jms/lib/model/post-installation-actions.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostInstallationActions = void 0;
/**
 * list of post installation actions
 **/
var PostInstallationActions;
(function (PostInstallationActions) {
    PostInstallationActions["ChangeMinimumKeyLengthForEc"] = "CHANGE_MINIMUM_KEY_LENGTH_FOR_EC";
    PostInstallationActions["ChangeMinimumKeyLengthForDsa"] = "CHANGE_MINIMUM_KEY_LENGTH_FOR_DSA";
    PostInstallationActions["ChangeMinimumKeyLengthForDiffieHellman"] = "CHANGE_MINIMUM_KEY_LENGTH_FOR_DIFFIE_HELLMAN";
    PostInstallationActions["ChangeMinimumKeyLengthForRsaSignedJars"] = "CHANGE_MINIMUM_KEY_LENGTH_FOR_RSA_SIGNED_JARS";
    PostInstallationActions["DisableTls"] = "DISABLE_TLS";
    PostInstallationActions["UseOsCacerts"] = "USE_OS_CACERTS";
})(PostInstallationActions = exports.PostInstallationActions || (exports.PostInstallationActions = {}));
(function (PostInstallationActions) {
    function getJsonObj(obj) {
        return obj;
    }
    PostInstallationActions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PostInstallationActions.getDeserializedJsonObj = getDeserializedJsonObj;
})(PostInstallationActions = exports.PostInstallationActions || (exports.PostInstallationActions = {}));
