// oci-jms/lib/model/summarize-application-usage-fields.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeApplicationUsageFields = void 0;
/**
 * Potential fields to include into response model to summarize applications endpoints.
 **/
var SummarizeApplicationUsageFields;
(function (SummarizeApplicationUsageFields) {
    SummarizeApplicationUsageFields["ApproximateJreCount"] = "approximateJreCount";
    SummarizeApplicationUsageFields["ApproximateInstallationCount"] = "approximateInstallationCount";
    SummarizeApplicationUsageFields["ApproximateManagedInstanceCount"] = "approximateManagedInstanceCount";
})(SummarizeApplicationUsageFields = exports.SummarizeApplicationUsageFields || (exports.SummarizeApplicationUsageFields = {}));
(function (SummarizeApplicationUsageFields) {
    function getJsonObj(obj) {
        return obj;
    }
    SummarizeApplicationUsageFields.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SummarizeApplicationUsageFields.getDeserializedJsonObj = getDeserializedJsonObj;
})(SummarizeApplicationUsageFields = exports.SummarizeApplicationUsageFields || (exports.SummarizeApplicationUsageFields = {}));
