// oci-jms/lib/model/support-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportType = void 0;
/**
 * Support category for the Java release family.
 **/
var SupportType;
(function (SupportType) {
    SupportType["Lts"] = "LTS";
    SupportType["NonLts"] = "NON_LTS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SupportType["UnknownValue"] = "UNKNOWN_VALUE";
})(SupportType = exports.SupportType || (exports.SupportType = {}));
(function (SupportType) {
    function getJsonObj(obj) {
        return obj;
    }
    SupportType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SupportType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SupportType = exports.SupportType || (exports.SupportType = {}));
