// oci-jms/lib/model/tls-versions.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TlsVersions = void 0;
/**
 * List of tls versions to be disabled
 **/
var TlsVersions;
(function (TlsVersions) {
    TlsVersions["Tls10"] = "TLS_1_0";
    TlsVersions["Tls11"] = "TLS_1_1";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TlsVersions["UnknownValue"] = "UNKNOWN_VALUE";
})(TlsVersions = exports.TlsVersions || (exports.TlsVersions = {}));
(function (TlsVersions) {
    function getJsonObj(obj) {
        return obj;
    }
    TlsVersions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TlsVersions.getDeserializedJsonObj = getDeserializedJsonObj;
})(TlsVersions = exports.TlsVersions || (exports.TlsVersions = {}));
