// oci-vnmonitoring/lib/model/create-public-ip-details.js
"use strict";
/**
 * Network Monitoring API
 * Use the Network Monitoring API to troubleshoot routing and security issues for resources such as virtual cloud networks (VCNs) and compute instances. For more information, see the console
documentation for the [Network Path Analyzer](/iaas/Content/Network/Concepts/path_analyzer.htm) tool.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePublicIpDetails = void 0;
var CreatePublicIpDetails;
(function (CreatePublicIpDetails) {
    let Lifetime;
    (function (Lifetime) {
        Lifetime["Ephemeral"] = "EPHEMERAL";
        Lifetime["Reserved"] = "RESERVED";
    })(Lifetime = CreatePublicIpDetails.Lifetime || (CreatePublicIpDetails.Lifetime = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreatePublicIpDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreatePublicIpDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreatePublicIpDetails = exports.CreatePublicIpDetails || (exports.CreatePublicIpDetails = {}));
