// oci-vnmonitoring/lib/model/remote-peering-connection.js
"use strict";
/**
 * Network Monitoring API
 * Use the Network Monitoring API to troubleshoot routing and security issues for resources such as virtual cloud networks (VCNs) and compute instances. For more information, see the console
documentation for the [Network Path Analyzer](/iaas/Content/Network/Concepts/path_analyzer.htm) tool.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemotePeeringConnection = void 0;
var RemotePeeringConnection;
(function (RemotePeeringConnection) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Available"] = "AVAILABLE";
        LifecycleState["Provisioning"] = "PROVISIONING";
        LifecycleState["Terminating"] = "TERMINATING";
        LifecycleState["Terminated"] = "TERMINATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = RemotePeeringConnection.LifecycleState || (RemotePeeringConnection.LifecycleState = {}));
    let PeeringStatus;
    (function (PeeringStatus) {
        PeeringStatus["Invalid"] = "INVALID";
        PeeringStatus["New"] = "NEW";
        PeeringStatus["Pending"] = "PENDING";
        PeeringStatus["Peered"] = "PEERED";
        PeeringStatus["Revoked"] = "REVOKED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PeeringStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(PeeringStatus = RemotePeeringConnection.PeeringStatus || (RemotePeeringConnection.PeeringStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemotePeeringConnection.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RemotePeeringConnection.getDeserializedJsonObj = getDeserializedJsonObj;
})(RemotePeeringConnection = exports.RemotePeeringConnection || (exports.RemotePeeringConnection = {}));
