// oci-logging/lib/model/operation-status.js
"use strict";
/**
 * Logging Management API
 * Use the Logging Management API to create, read, list, update, move and delete
log groups, log objects, log saved searches, agent configurations, log data models,
continuous queries, and managed continuous queries.

For more information, see [Logging Overview](/iaas/Content/Logging/Concepts/loggingoverview.htm).

 * OpenAPI spec version: 20200531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationStatus = void 0;
/**
 * The status of a work request.
 **/
var OperationStatus;
(function (OperationStatus) {
    OperationStatus["Accepted"] = "ACCEPTED";
    OperationStatus["InProgress"] = "IN_PROGRESS";
    OperationStatus["Failed"] = "FAILED";
    OperationStatus["Succeeded"] = "SUCCEEDED";
    OperationStatus["Cancelling"] = "CANCELLING";
    OperationStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
(function (OperationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
