// oci-datasafe/lib/model/alert-policy-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertPolicyLifecycleState = void 0;
/**
 * The current state of the alert policy resource.
 **/
var AlertPolicyLifecycleState;
(function (AlertPolicyLifecycleState) {
    AlertPolicyLifecycleState["Creating"] = "CREATING";
    AlertPolicyLifecycleState["Updating"] = "UPDATING";
    AlertPolicyLifecycleState["Active"] = "ACTIVE";
    AlertPolicyLifecycleState["Deleting"] = "DELETING";
    AlertPolicyLifecycleState["Deleted"] = "DELETED";
    AlertPolicyLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AlertPolicyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AlertPolicyLifecycleState = exports.AlertPolicyLifecycleState || (exports.AlertPolicyLifecycleState = {}));
(function (AlertPolicyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AlertPolicyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlertPolicyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlertPolicyLifecycleState = exports.AlertPolicyLifecycleState || (exports.AlertPolicyLifecycleState = {}));
