// oci-datasafe/lib/model/alert-policy-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertPolicyType = void 0;
/**
 * The Data Safe feature that relates to the alert policy.
 **/
var AlertPolicyType;
(function (AlertPolicyType) {
    AlertPolicyType["Auditing"] = "AUDITING";
    AlertPolicyType["SecurityAssessment"] = "SECURITY_ASSESSMENT";
    AlertPolicyType["UserAssessment"] = "USER_ASSESSMENT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AlertPolicyType["UnknownValue"] = "UNKNOWN_VALUE";
})(AlertPolicyType = exports.AlertPolicyType || (exports.AlertPolicyType = {}));
(function (AlertPolicyType) {
    function getJsonObj(obj) {
        return obj;
    }
    AlertPolicyType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlertPolicyType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlertPolicyType = exports.AlertPolicyType || (exports.AlertPolicyType = {}));
