// oci-datasafe/lib/model/alert-severity.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertSeverity = void 0;
/**
 * The severity level of the alert.
 **/
var AlertSeverity;
(function (AlertSeverity) {
    AlertSeverity["Critical"] = "CRITICAL";
    AlertSeverity["High"] = "HIGH";
    AlertSeverity["Medium"] = "MEDIUM";
    AlertSeverity["Low"] = "LOW";
    AlertSeverity["Evaluate"] = "EVALUATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AlertSeverity["UnknownValue"] = "UNKNOWN_VALUE";
})(AlertSeverity = exports.AlertSeverity || (exports.AlertSeverity = {}));
(function (AlertSeverity) {
    function getJsonObj(obj) {
        return obj;
    }
    AlertSeverity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AlertSeverity.getDeserializedJsonObj = getDeserializedJsonObj;
})(AlertSeverity = exports.AlertSeverity || (exports.AlertSeverity = {}));
