// oci-datasafe/lib/model/audit-archive-retrieval-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditArchiveRetrievalLifecycleState = void 0;
/**
 * The current state of the archive retrieval.
 **/
var AuditArchiveRetrievalLifecycleState;
(function (AuditArchiveRetrievalLifecycleState) {
    AuditArchiveRetrievalLifecycleState["Creating"] = "CREATING";
    AuditArchiveRetrievalLifecycleState["Active"] = "ACTIVE";
    AuditArchiveRetrievalLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    AuditArchiveRetrievalLifecycleState["Failed"] = "FAILED";
    AuditArchiveRetrievalLifecycleState["Deleting"] = "DELETING";
    AuditArchiveRetrievalLifecycleState["Deleted"] = "DELETED";
    AuditArchiveRetrievalLifecycleState["Updating"] = "UPDATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AuditArchiveRetrievalLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AuditArchiveRetrievalLifecycleState = exports.AuditArchiveRetrievalLifecycleState || (exports.AuditArchiveRetrievalLifecycleState = {}));
(function (AuditArchiveRetrievalLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AuditArchiveRetrievalLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AuditArchiveRetrievalLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AuditArchiveRetrievalLifecycleState = exports.AuditArchiveRetrievalLifecycleState || (exports.AuditArchiveRetrievalLifecycleState = {}));
