// oci-datasafe/lib/model/audit-trail-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditTrailLifecycleState = void 0;
/**
 * The current state of the audit trail.
 **/
var AuditTrailLifecycleState;
(function (AuditTrailLifecycleState) {
    AuditTrailLifecycleState["Inactive"] = "INACTIVE";
    AuditTrailLifecycleState["Updating"] = "UPDATING";
    AuditTrailLifecycleState["Active"] = "ACTIVE";
    AuditTrailLifecycleState["Deleting"] = "DELETING";
    AuditTrailLifecycleState["Failed"] = "FAILED";
    AuditTrailLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AuditTrailLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AuditTrailLifecycleState = exports.AuditTrailLifecycleState || (exports.AuditTrailLifecycleState = {}));
(function (AuditTrailLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AuditTrailLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AuditTrailLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AuditTrailLifecycleState = exports.AuditTrailLifecycleState || (exports.AuditTrailLifecycleState = {}));
