// oci-datasafe/lib/model/audit-trail-status.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditTrailStatus = void 0;
/**
 * The current sub-state of the audit trail.
 **/
var AuditTrailStatus;
(function (AuditTrailStatus) {
    AuditTrailStatus["Starting"] = "STARTING";
    AuditTrailStatus["Collecting"] = "COLLECTING";
    AuditTrailStatus["Recovering"] = "RECOVERING";
    AuditTrailStatus["Idle"] = "IDLE";
    AuditTrailStatus["Stopping"] = "STOPPING";
    AuditTrailStatus["Stopped"] = "STOPPED";
    AuditTrailStatus["Resuming"] = "RESUMING";
    AuditTrailStatus["Retrying"] = "RETRYING";
    AuditTrailStatus["NotStarted"] = "NOT_STARTED";
    AuditTrailStatus["StoppedNeedsAttn"] = "STOPPED_NEEDS_ATTN";
    AuditTrailStatus["StoppedFailed"] = "STOPPED_FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AuditTrailStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(AuditTrailStatus = exports.AuditTrailStatus || (exports.AuditTrailStatus = {}));
(function (AuditTrailStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    AuditTrailStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AuditTrailStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(AuditTrailStatus = exports.AuditTrailStatus || (exports.AuditTrailStatus = {}));
