// oci-datasafe/lib/model/audit-policy-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditPolicyLifecycleState = void 0;
/**
 * The current state of audit policies available on the target.
 **/
var AuditPolicyLifecycleState;
(function (AuditPolicyLifecycleState) {
    AuditPolicyLifecycleState["Creating"] = "CREATING";
    AuditPolicyLifecycleState["Updating"] = "UPDATING";
    AuditPolicyLifecycleState["Active"] = "ACTIVE";
    AuditPolicyLifecycleState["Failed"] = "FAILED";
    AuditPolicyLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    AuditPolicyLifecycleState["Deleting"] = "DELETING";
    AuditPolicyLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AuditPolicyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AuditPolicyLifecycleState = exports.AuditPolicyLifecycleState || (exports.AuditPolicyLifecycleState = {}));
(function (AuditPolicyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AuditPolicyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AuditPolicyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AuditPolicyLifecycleState = exports.AuditPolicyLifecycleState || (exports.AuditPolicyLifecycleState = {}));
