// oci-datasafe/lib/model/data-model-format.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataModelFormat = void 0;
/**
 * The format of the sensitive data model file.
 **/
var DataModelFormat;
(function (DataModelFormat) {
    DataModelFormat["Xml"] = "XML";
})(DataModelFormat = exports.DataModelFormat || (exports.DataModelFormat = {}));
(function (DataModelFormat) {
    function getJsonObj(obj) {
        return obj;
    }
    DataModelFormat.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DataModelFormat.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataModelFormat = exports.DataModelFormat || (exports.DataModelFormat = {}));
