// oci-datasafe/lib/model/database-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseType = void 0;
/**
 * The type of the database. Allowed values are
 *   - DATABASE_CLOUD_SERVICE - Represents Oracle Database Cloud Services.
 *   - AUTONOMOUS_DATABASE - Represents Oracle Autonomous Databases.
 *   - INSTALLED_DATABASE - Represents databases running on-premises or on compute instances.
 *
 **/
var DatabaseType;
(function (DatabaseType) {
    DatabaseType["DatabaseCloudService"] = "DATABASE_CLOUD_SERVICE";
    DatabaseType["AutonomousDatabase"] = "AUTONOMOUS_DATABASE";
    DatabaseType["InstalledDatabase"] = "INSTALLED_DATABASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseType["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseType = exports.DatabaseType || (exports.DatabaseType = {}));
(function (DatabaseType) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseType = exports.DatabaseType || (exports.DatabaseType = {}));
