// oci-datasafe/lib/model/discovery-analytics-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryAnalyticsSummary = void 0;
const model = __importStar(require("../model"));
var DiscoveryAnalyticsSummary;
(function (DiscoveryAnalyticsSummary) {
    let MetricName;
    (function (MetricName) {
        MetricName["SensitiveDataModel"] = "SENSITIVE_DATA_MODEL";
        MetricName["SensitiveType"] = "SENSITIVE_TYPE";
        MetricName["SensitiveSchema"] = "SENSITIVE_SCHEMA";
        MetricName["SensitiveTable"] = "SENSITIVE_TABLE";
        MetricName["SensitiveColumn"] = "SENSITIVE_COLUMN";
        MetricName["SensitiveDataValue"] = "SENSITIVE_DATA_VALUE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MetricName["UnknownValue"] = "UNKNOWN_VALUE";
    })(MetricName = DiscoveryAnalyticsSummary.MetricName || (DiscoveryAnalyticsSummary.MetricName = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dimensions": obj.dimensions ? model.Dimensions.getJsonObj(obj.dimensions) : undefined
        });
        return jsonObj;
    }
    DiscoveryAnalyticsSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dimensions": obj.dimensions
                ? model.Dimensions.getDeserializedJsonObj(obj.dimensions)
                : undefined
        });
        return jsonObj;
    }
    DiscoveryAnalyticsSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryAnalyticsSummary = exports.DiscoveryAnalyticsSummary || (exports.DiscoveryAnalyticsSummary = {}));
