// oci-datasafe/lib/model/discovery-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryLifecycleState = void 0;
/**
 * The current state of a discovery resource such as sensitive data model or sensitive type.
 **/
var DiscoveryLifecycleState;
(function (DiscoveryLifecycleState) {
    DiscoveryLifecycleState["Creating"] = "CREATING";
    DiscoveryLifecycleState["Active"] = "ACTIVE";
    DiscoveryLifecycleState["Updating"] = "UPDATING";
    DiscoveryLifecycleState["Deleting"] = "DELETING";
    DiscoveryLifecycleState["Deleted"] = "DELETED";
    DiscoveryLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DiscoveryLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(DiscoveryLifecycleState = exports.DiscoveryLifecycleState || (exports.DiscoveryLifecycleState = {}));
(function (DiscoveryLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DiscoveryLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DiscoveryLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryLifecycleState = exports.DiscoveryLifecycleState || (exports.DiscoveryLifecycleState = {}));
