// oci-datasafe/lib/model/discovery-job.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryJob = void 0;
var DiscoveryJob;
(function (DiscoveryJob) {
    let DiscoveryType;
    (function (DiscoveryType) {
        DiscoveryType["All"] = "ALL";
        DiscoveryType["New"] = "NEW";
        DiscoveryType["Modified"] = "MODIFIED";
        DiscoveryType["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DiscoveryType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DiscoveryType = DiscoveryJob.DiscoveryType || (DiscoveryJob.DiscoveryType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DiscoveryJob.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DiscoveryJob.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryJob = exports.DiscoveryJob || (exports.DiscoveryJob = {}));
