// oci-datasafe/lib/model/discovery-job-result.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryJobResult = void 0;
const model = __importStar(require("../model"));
var DiscoveryJobResult;
(function (DiscoveryJobResult) {
    let DiscoveryType;
    (function (DiscoveryType) {
        DiscoveryType["New"] = "NEW";
        DiscoveryType["Modified"] = "MODIFIED";
        DiscoveryType["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DiscoveryType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DiscoveryType = DiscoveryJobResult.DiscoveryType || (DiscoveryJobResult.DiscoveryType = {}));
    let ObjectType;
    (function (ObjectType) {
        ObjectType["Table"] = "TABLE";
        ObjectType["EditioningView"] = "EDITIONING_VIEW";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ObjectType = DiscoveryJobResult.ObjectType || (DiscoveryJobResult.ObjectType = {}));
    let RelationType;
    (function (RelationType) {
        RelationType["None"] = "NONE";
        RelationType["AppDefined"] = "APP_DEFINED";
        RelationType["DbDefined"] = "DB_DEFINED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RelationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(RelationType = DiscoveryJobResult.RelationType || (DiscoveryJobResult.RelationType = {}));
    let PlannedAction;
    (function (PlannedAction) {
        PlannedAction["None"] = "NONE";
        PlannedAction["Accept"] = "ACCEPT";
        PlannedAction["Invalidate"] = "INVALIDATE";
        PlannedAction["Reject"] = "REJECT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PlannedAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(PlannedAction = DiscoveryJobResult.PlannedAction || (DiscoveryJobResult.PlannedAction = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "modifiedAttributes": obj.modifiedAttributes
                ? model.ModifiedAttributes.getJsonObj(obj.modifiedAttributes)
                : undefined
        });
        return jsonObj;
    }
    DiscoveryJobResult.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "modifiedAttributes": obj.modifiedAttributes
                ? model.ModifiedAttributes.getDeserializedJsonObj(obj.modifiedAttributes)
                : undefined
        });
        return jsonObj;
    }
    DiscoveryJobResult.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryJobResult = exports.DiscoveryJobResult || (exports.DiscoveryJobResult = {}));
