// oci-datasafe/lib/model/enable-conditions.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnableConditions = void 0;
var EnableConditions;
(function (EnableConditions) {
    let EntitySelection;
    (function (EntitySelection) {
        EntitySelection["Include"] = "INCLUDE";
        EntitySelection["Exclude"] = "EXCLUDE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EntitySelection["UnknownValue"] = "UNKNOWN_VALUE";
    })(EntitySelection = EnableConditions.EntitySelection || (EnableConditions.EntitySelection = {}));
    let EntityType;
    (function (EntityType) {
        EntityType["User"] = "USER";
        EntityType["Role"] = "ROLE";
        EntityType["AllUsers"] = "ALL_USERS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EntityType["UnknownValue"] = "UNKNOWN_VALUE";
    })(EntityType = EnableConditions.EntityType || (EnableConditions.EntityType = {}));
    let OperationStatus;
    (function (OperationStatus) {
        OperationStatus["Success"] = "SUCCESS";
        OperationStatus["Failure"] = "FAILURE";
        OperationStatus["Both"] = "BOTH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationStatus = EnableConditions.OperationStatus || (EnableConditions.OperationStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EnableConditions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EnableConditions.getDeserializedJsonObj = getDeserializedJsonObj;
})(EnableConditions = exports.EnableConditions || (exports.EnableConditions = {}));
