// oci-datasafe/lib/model/format-entry-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormatEntryType = void 0;
/**
 * The type of the format entry.
 **/
var FormatEntryType;
(function (FormatEntryType) {
    FormatEntryType["DeleteRows"] = "DELETE_ROWS";
    FormatEntryType["DeterministicSubstitution"] = "DETERMINISTIC_SUBSTITUTION";
    FormatEntryType["DeterministicEncryption"] = "DETERMINISTIC_ENCRYPTION";
    FormatEntryType["DeterministicEncryptionDate"] = "DETERMINISTIC_ENCRYPTION_DATE";
    FormatEntryType["FixedNumber"] = "FIXED_NUMBER";
    FormatEntryType["FixedString"] = "FIXED_STRING";
    FormatEntryType["LibraryMaskingFormat"] = "LIBRARY_MASKING_FORMAT";
    FormatEntryType["NullValue"] = "NULL_VALUE";
    FormatEntryType["PostProcessingFunction"] = "POST_PROCESSING_FUNCTION";
    FormatEntryType["PreserveOriginalData"] = "PRESERVE_ORIGINAL_DATA";
    FormatEntryType["RandomDate"] = "RANDOM_DATE";
    FormatEntryType["RandomDecimalNumber"] = "RANDOM_DECIMAL_NUMBER";
    FormatEntryType["RandomDigits"] = "RANDOM_DIGITS";
    FormatEntryType["RandomList"] = "RANDOM_LIST";
    FormatEntryType["RandomNumber"] = "RANDOM_NUMBER";
    FormatEntryType["RandomString"] = "RANDOM_STRING";
    FormatEntryType["RandomSubstitution"] = "RANDOM_SUBSTITUTION";
    FormatEntryType["RegularExpression"] = "REGULAR_EXPRESSION";
    FormatEntryType["Shuffle"] = "SHUFFLE";
    FormatEntryType["SqlExpression"] = "SQL_EXPRESSION";
    FormatEntryType["Substring"] = "SUBSTRING";
    FormatEntryType["TruncateTable"] = "TRUNCATE_TABLE";
    FormatEntryType["UserDefinedFunction"] = "USER_DEFINED_FUNCTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    FormatEntryType["UnknownValue"] = "UNKNOWN_VALUE";
})(FormatEntryType = exports.FormatEntryType || (exports.FormatEntryType = {}));
(function (FormatEntryType) {
    function getJsonObj(obj) {
        return obj;
    }
    FormatEntryType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    FormatEntryType.getDeserializedJsonObj = getDeserializedJsonObj;
})(FormatEntryType = exports.FormatEntryType || (exports.FormatEntryType = {}));
