// oci-datasafe/lib/model/generate-masking-report-for-download-details.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenerateMaskingReportForDownloadDetails = void 0;
var GenerateMaskingReportForDownloadDetails;
(function (GenerateMaskingReportForDownloadDetails) {
    let ReportFormat;
    (function (ReportFormat) {
        ReportFormat["Pdf"] = "PDF";
        ReportFormat["Xls"] = "XLS";
    })(ReportFormat = GenerateMaskingReportForDownloadDetails.ReportFormat || (GenerateMaskingReportForDownloadDetails.ReportFormat = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GenerateMaskingReportForDownloadDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GenerateMaskingReportForDownloadDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(GenerateMaskingReportForDownloadDetails = exports.GenerateMaskingReportForDownloadDetails || (exports.GenerateMaskingReportForDownloadDetails = {}));
