// oci-datasafe/lib/model/infrastructure-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfrastructureType = void 0;
/**
 * The infrastructure type the database is running on. Allowed values are
 *   - ORACLE_CLOUD - Represents that the database is running on the Oracle Cloud Infrastructure.
 *   - CLOUD_AT_CUSTOMER - Represents that the database is running on Oracle Cloud@Customer infrastructure.
 *   - ON_PREMISES - Represents that the database is running on on-premises infrastructure.
 *   - NON_ORACLE_CLOUD - Represents that the database is running on non-Oracle cloud infrastructure.
 *
 **/
var InfrastructureType;
(function (InfrastructureType) {
    InfrastructureType["OracleCloud"] = "ORACLE_CLOUD";
    InfrastructureType["CloudAtCustomer"] = "CLOUD_AT_CUSTOMER";
    InfrastructureType["OnPremises"] = "ON_PREMISES";
    InfrastructureType["NonOracleCloud"] = "NON_ORACLE_CLOUD";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    InfrastructureType["UnknownValue"] = "UNKNOWN_VALUE";
})(InfrastructureType = exports.InfrastructureType || (exports.InfrastructureType = {}));
(function (InfrastructureType) {
    function getJsonObj(obj) {
        return obj;
    }
    InfrastructureType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InfrastructureType.getDeserializedJsonObj = getDeserializedJsonObj;
})(InfrastructureType = exports.InfrastructureType || (exports.InfrastructureType = {}));
