// oci-datasafe/lib/model/masking-column-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingColumnLifecycleState = void 0;
/**
 * The current state of a masking column.
 **/
var MaskingColumnLifecycleState;
(function (MaskingColumnLifecycleState) {
    MaskingColumnLifecycleState["Creating"] = "CREATING";
    MaskingColumnLifecycleState["Active"] = "ACTIVE";
    MaskingColumnLifecycleState["Updating"] = "UPDATING";
    MaskingColumnLifecycleState["Deleting"] = "DELETING";
    MaskingColumnLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    MaskingColumnLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MaskingColumnLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(MaskingColumnLifecycleState = exports.MaskingColumnLifecycleState || (exports.MaskingColumnLifecycleState = {}));
(function (MaskingColumnLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    MaskingColumnLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaskingColumnLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingColumnLifecycleState = exports.MaskingColumnLifecycleState || (exports.MaskingColumnLifecycleState = {}));
