// oci-datasafe/lib/model/masking-analytics-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingAnalyticsSummary = void 0;
const model = __importStar(require("../model"));
var MaskingAnalyticsSummary;
(function (MaskingAnalyticsSummary) {
    let MetricName;
    (function (MetricName) {
        MetricName["MaskingPolicy"] = "MASKING_POLICY";
        MetricName["MaskingDatabase"] = "MASKING_DATABASE";
        MetricName["MaskingWorkRequest"] = "MASKING_WORK_REQUEST";
        MetricName["MaskedSensitiveType"] = "MASKED_SENSITIVE_TYPE";
        MetricName["MaskedSchema"] = "MASKED_SCHEMA";
        MetricName["MaskedTable"] = "MASKED_TABLE";
        MetricName["MaskedColumn"] = "MASKED_COLUMN";
        MetricName["MaskedDataValue"] = "MASKED_DATA_VALUE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MetricName["UnknownValue"] = "UNKNOWN_VALUE";
    })(MetricName = MaskingAnalyticsSummary.MetricName || (MaskingAnalyticsSummary.MetricName = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dimensions": obj.dimensions
                ? model.MaskingAnalyticsDimensions.getJsonObj(obj.dimensions)
                : undefined
        });
        return jsonObj;
    }
    MaskingAnalyticsSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dimensions": obj.dimensions
                ? model.MaskingAnalyticsDimensions.getDeserializedJsonObj(obj.dimensions)
                : undefined
        });
        return jsonObj;
    }
    MaskingAnalyticsSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingAnalyticsSummary = exports.MaskingAnalyticsSummary || (exports.MaskingAnalyticsSummary = {}));
