// oci-datasafe/lib/model/masking-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaskingLifecycleState = void 0;
/**
 * The current state of a masking resource such as masking policy or library masking format.
 **/
var MaskingLifecycleState;
(function (MaskingLifecycleState) {
    MaskingLifecycleState["Creating"] = "CREATING";
    MaskingLifecycleState["Active"] = "ACTIVE";
    MaskingLifecycleState["Updating"] = "UPDATING";
    MaskingLifecycleState["Deleting"] = "DELETING";
    MaskingLifecycleState["Deleted"] = "DELETED";
    MaskingLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    MaskingLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MaskingLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(MaskingLifecycleState = exports.MaskingLifecycleState || (exports.MaskingLifecycleState = {}));
(function (MaskingLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    MaskingLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaskingLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaskingLifecycleState = exports.MaskingLifecycleState || (exports.MaskingLifecycleState = {}));
