// oci-datasafe/lib/model/object-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectType = void 0;
/**
 * The type of the object to which a column belongs.
 **/
var ObjectType;
(function (ObjectType) {
    ObjectType["Table"] = "TABLE";
    ObjectType["EditioningView"] = "EDITIONING_VIEW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
})(ObjectType = exports.ObjectType || (exports.ObjectType = {}));
(function (ObjectType) {
    function getJsonObj(obj) {
        return obj;
    }
    ObjectType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ObjectType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ObjectType = exports.ObjectType || (exports.ObjectType = {}));
