// oci-datasafe/lib/model/report-definition.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportDefinition = void 0;
const model = __importStar(require("../model"));
var ReportDefinition;
(function (ReportDefinition) {
    let Category;
    (function (Category) {
        Category["CustomReports"] = "CUSTOM_REPORTS";
        Category["Summary"] = "SUMMARY";
        Category["ActivityAuditing"] = "ACTIVITY_AUDITING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Category["UnknownValue"] = "UNKNOWN_VALUE";
    })(Category = ReportDefinition.Category || (ReportDefinition.Category = {}));
    let ScheduledReportMimeType;
    (function (ScheduledReportMimeType) {
        ScheduledReportMimeType["Pdf"] = "PDF";
        ScheduledReportMimeType["Xls"] = "XLS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ScheduledReportMimeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ScheduledReportMimeType = ReportDefinition.ScheduledReportMimeType || (ReportDefinition.ScheduledReportMimeType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "columnInfo": obj.columnInfo
                ? obj.columnInfo.map(item => {
                    return model.Column.getJsonObj(item);
                })
                : undefined,
            "columnFilters": obj.columnFilters
                ? obj.columnFilters.map(item => {
                    return model.ColumnFilter.getJsonObj(item);
                })
                : undefined,
            "columnSortings": obj.columnSortings
                ? obj.columnSortings.map(item => {
                    return model.ColumnSorting.getJsonObj(item);
                })
                : undefined,
            "summary": obj.summary
                ? obj.summary.map(item => {
                    return model.Summary.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ReportDefinition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "columnInfo": obj.columnInfo
                ? obj.columnInfo.map(item => {
                    return model.Column.getDeserializedJsonObj(item);
                })
                : undefined,
            "columnFilters": obj.columnFilters
                ? obj.columnFilters.map(item => {
                    return model.ColumnFilter.getDeserializedJsonObj(item);
                })
                : undefined,
            "columnSortings": obj.columnSortings
                ? obj.columnSortings.map(item => {
                    return model.ColumnSorting.getDeserializedJsonObj(item);
                })
                : undefined,
            "summary": obj.summary
                ? obj.summary.map(item => {
                    return model.Summary.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ReportDefinition.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReportDefinition = exports.ReportDefinition || (exports.ReportDefinition = {}));
