// oci-datasafe/lib/model/report-definition-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportDefinitionLifecycleState = void 0;
/**
 * The current state of the report definition.
 **/
var ReportDefinitionLifecycleState;
(function (ReportDefinitionLifecycleState) {
    ReportDefinitionLifecycleState["Creating"] = "CREATING";
    ReportDefinitionLifecycleState["Updating"] = "UPDATING";
    ReportDefinitionLifecycleState["Active"] = "ACTIVE";
    ReportDefinitionLifecycleState["Deleting"] = "DELETING";
    ReportDefinitionLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ReportDefinitionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(ReportDefinitionLifecycleState = exports.ReportDefinitionLifecycleState || (exports.ReportDefinitionLifecycleState = {}));
(function (ReportDefinitionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    ReportDefinitionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ReportDefinitionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(ReportDefinitionLifecycleState = exports.ReportDefinitionLifecycleState || (exports.ReportDefinitionLifecycleState = {}));
