// oci-datasafe/lib/model/sdm-masking-policy-difference.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SdmMaskingPolicyDifference = void 0;
var SdmMaskingPolicyDifference;
(function (SdmMaskingPolicyDifference) {
    let DifferenceType;
    (function (DifferenceType) {
        DifferenceType["All"] = "ALL";
        DifferenceType["New"] = "NEW";
        DifferenceType["Modified"] = "MODIFIED";
        DifferenceType["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DifferenceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DifferenceType = SdmMaskingPolicyDifference.DifferenceType || (SdmMaskingPolicyDifference.DifferenceType = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = SdmMaskingPolicyDifference.LifecycleState || (SdmMaskingPolicyDifference.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SdmMaskingPolicyDifference.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SdmMaskingPolicyDifference.getDeserializedJsonObj = getDeserializedJsonObj;
})(SdmMaskingPolicyDifference = exports.SdmMaskingPolicyDifference || (exports.SdmMaskingPolicyDifference = {}));
