// oci-datasafe/lib/model/sensitive-column-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensitiveColumnLifecycleState = void 0;
/**
 * The current state of a sensitive column.
 **/
var SensitiveColumnLifecycleState;
(function (SensitiveColumnLifecycleState) {
    SensitiveColumnLifecycleState["Creating"] = "CREATING";
    SensitiveColumnLifecycleState["Active"] = "ACTIVE";
    SensitiveColumnLifecycleState["Updating"] = "UPDATING";
    SensitiveColumnLifecycleState["Deleting"] = "DELETING";
    SensitiveColumnLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SensitiveColumnLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SensitiveColumnLifecycleState = exports.SensitiveColumnLifecycleState || (exports.SensitiveColumnLifecycleState = {}));
(function (SensitiveColumnLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SensitiveColumnLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SensitiveColumnLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SensitiveColumnLifecycleState = exports.SensitiveColumnLifecycleState || (exports.SensitiveColumnLifecycleState = {}));
