// oci-datasafe/lib/model/sensitive-type-entity.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensitiveTypeEntity = void 0;
/**
 * The entity type. It can be either a sensitive type with regular expressions or a sensitive category used for
 * grouping similar sensitive types.
 *
 **/
var SensitiveTypeEntity;
(function (SensitiveTypeEntity) {
    SensitiveTypeEntity["SensitiveType"] = "SENSITIVE_TYPE";
    SensitiveTypeEntity["SensitiveCategory"] = "SENSITIVE_CATEGORY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SensitiveTypeEntity["UnknownValue"] = "UNKNOWN_VALUE";
})(SensitiveTypeEntity = exports.SensitiveTypeEntity || (exports.SensitiveTypeEntity = {}));
(function (SensitiveTypeEntity) {
    function getJsonObj(obj) {
        return obj;
    }
    SensitiveTypeEntity.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SensitiveTypeEntity.getDeserializedJsonObj = getDeserializedJsonObj;
})(SensitiveTypeEntity = exports.SensitiveTypeEntity || (exports.SensitiveTypeEntity = {}));
