// oci-datasafe/lib/model/sensitive-column.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensitiveColumn = void 0;
var SensitiveColumn;
(function (SensitiveColumn) {
    let ObjectType;
    (function (ObjectType) {
        ObjectType["Table"] = "TABLE";
        ObjectType["EditioningView"] = "EDITIONING_VIEW";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ObjectType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ObjectType = SensitiveColumn.ObjectType || (SensitiveColumn.ObjectType = {}));
    let Status;
    (function (Status) {
        Status["Valid"] = "VALID";
        Status["Invalid"] = "INVALID";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = SensitiveColumn.Status || (SensitiveColumn.Status = {}));
    let Source;
    (function (Source) {
        Source["Manual"] = "MANUAL";
        Source["Discovery"] = "DISCOVERY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Source["UnknownValue"] = "UNKNOWN_VALUE";
    })(Source = SensitiveColumn.Source || (SensitiveColumn.Source = {}));
    let RelationType;
    (function (RelationType) {
        RelationType["None"] = "NONE";
        RelationType["AppDefined"] = "APP_DEFINED";
        RelationType["DbDefined"] = "DB_DEFINED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RelationType["UnknownValue"] = "UNKNOWN_VALUE";
    })(RelationType = SensitiveColumn.RelationType || (SensitiveColumn.RelationType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SensitiveColumn.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SensitiveColumn.getDeserializedJsonObj = getDeserializedJsonObj;
})(SensitiveColumn = exports.SensitiveColumn || (exports.SensitiveColumn = {}));
