// oci-datasafe/lib/model/target-database-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetDatabaseLifecycleState = void 0;
/**
 * The lifecycle state of the target database.
 * - CREATING - Represents that the target database is creating.
 * - UPDATING - Represents that the target database is updating.
 * - ACTIVE - Represents that the target database is active.
 * - INACTIVE - Represents that the target database is inactive.
 * - DELETING - Represents that the target database is deleting.
 * - DELETED - Represents that the target database is deleted.
 * - NEEDS_ATTENTION - Represents that the target database needs user attention.
 * - FAILED - Represents that the target database is failed.
 *
 **/
var TargetDatabaseLifecycleState;
(function (TargetDatabaseLifecycleState) {
    TargetDatabaseLifecycleState["Creating"] = "CREATING";
    TargetDatabaseLifecycleState["Updating"] = "UPDATING";
    TargetDatabaseLifecycleState["Active"] = "ACTIVE";
    TargetDatabaseLifecycleState["Inactive"] = "INACTIVE";
    TargetDatabaseLifecycleState["Deleting"] = "DELETING";
    TargetDatabaseLifecycleState["Deleted"] = "DELETED";
    TargetDatabaseLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    TargetDatabaseLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TargetDatabaseLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(TargetDatabaseLifecycleState = exports.TargetDatabaseLifecycleState || (exports.TargetDatabaseLifecycleState = {}));
(function (TargetDatabaseLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    TargetDatabaseLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TargetDatabaseLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(TargetDatabaseLifecycleState = exports.TargetDatabaseLifecycleState || (exports.TargetDatabaseLifecycleState = {}));
