// oci-datasafe/lib/model/user-assessment-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssessmentLifecycleState = void 0;
/**
 * The current state of the user assessment.
 **/
var UserAssessmentLifecycleState;
(function (UserAssessmentLifecycleState) {
    UserAssessmentLifecycleState["Creating"] = "CREATING";
    UserAssessmentLifecycleState["Succeeded"] = "SUCCEEDED";
    UserAssessmentLifecycleState["Updating"] = "UPDATING";
    UserAssessmentLifecycleState["Deleting"] = "DELETING";
    UserAssessmentLifecycleState["Deleted"] = "DELETED";
    UserAssessmentLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UserAssessmentLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(UserAssessmentLifecycleState = exports.UserAssessmentLifecycleState || (exports.UserAssessmentLifecycleState = {}));
(function (UserAssessmentLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    UserAssessmentLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UserAssessmentLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(UserAssessmentLifecycleState = exports.UserAssessmentLifecycleState || (exports.UserAssessmentLifecycleState = {}));
