// oci-datasafe/lib/model/user-assessment-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssessmentSummary = void 0;
var UserAssessmentSummary;
(function (UserAssessmentSummary) {
    let TriggeredBy;
    (function (TriggeredBy) {
        TriggeredBy["User"] = "USER";
        TriggeredBy["System"] = "SYSTEM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TriggeredBy["UnknownValue"] = "UNKNOWN_VALUE";
    })(TriggeredBy = UserAssessmentSummary.TriggeredBy || (UserAssessmentSummary.TriggeredBy = {}));
    let Type;
    (function (Type) {
        Type["Latest"] = "LATEST";
        Type["Saved"] = "SAVED";
        Type["SaveSchedule"] = "SAVE_SCHEDULE";
        Type["Compartment"] = "COMPARTMENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = UserAssessmentSummary.Type || (UserAssessmentSummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UserAssessmentSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UserAssessmentSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(UserAssessmentSummary = exports.UserAssessmentSummary || (exports.UserAssessmentSummary = {}));
