// oci-opensearch/lib/model/index.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.UpdateOpensearchClusterDetails = exports.UpdateOpensearchClusterBackupDetails = exports.UpdateClusterStatusDetails = exports.UpdateClusterHardenedImageDetails = exports.UpdateCheckinDetails = exports.SortOrder = exports.SecurityMode = exports.RestoreOpensearchClusterDetails = exports.RestoreOpensearchClusterBackupDetails = exports.ResizeOpensearchClusterVerticalDetails = exports.ResizeOpensearchClusterHorizontalDetails = exports.OperationType = exports.OperationStatus = exports.OpensearchVersionsSummary = exports.OpensearchVersionsCollection = exports.OpensearchClusterSummary = exports.OpensearchClusterCollection = exports.OpensearchClusterBackupSummary = exports.OpensearchClusterBackupCollection = exports.OpensearchClusterBackup = exports.OpensearchCluster = exports.MasterNodeHostType = exports.GetManifestResponse = exports.ExportOpensearchClusterBackupDetails = exports.DataNodeHostType = exports.CreateOpensearchClusterDetails = exports.ChangeOpensearchClusterCompartmentDetails = exports.ChangeOpensearchClusterBackupCompartmentDetails = exports.BackupState = exports.BackupOpensearchClusterDetails = exports.BackupEventDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const BackupEventDetails = __importStar(require("./backup-event-details"));
exports.BackupEventDetails = BackupEventDetails.BackupEventDetails;
const BackupOpensearchClusterDetails = __importStar(require("./backup-opensearch-cluster-details"));
exports.BackupOpensearchClusterDetails = BackupOpensearchClusterDetails.BackupOpensearchClusterDetails;
const BackupState = __importStar(require("./backup-state"));
exports.BackupState = BackupState.BackupState;
const ChangeOpensearchClusterBackupCompartmentDetails = __importStar(require("./change-opensearch-cluster-backup-compartment-details"));
exports.ChangeOpensearchClusterBackupCompartmentDetails = ChangeOpensearchClusterBackupCompartmentDetails.ChangeOpensearchClusterBackupCompartmentDetails;
const ChangeOpensearchClusterCompartmentDetails = __importStar(require("./change-opensearch-cluster-compartment-details"));
exports.ChangeOpensearchClusterCompartmentDetails = ChangeOpensearchClusterCompartmentDetails.ChangeOpensearchClusterCompartmentDetails;
const CreateOpensearchClusterDetails = __importStar(require("./create-opensearch-cluster-details"));
exports.CreateOpensearchClusterDetails = CreateOpensearchClusterDetails.CreateOpensearchClusterDetails;
const DataNodeHostType = __importStar(require("./data-node-host-type"));
exports.DataNodeHostType = DataNodeHostType.DataNodeHostType;
const ExportOpensearchClusterBackupDetails = __importStar(require("./export-opensearch-cluster-backup-details"));
exports.ExportOpensearchClusterBackupDetails = ExportOpensearchClusterBackupDetails.ExportOpensearchClusterBackupDetails;
const GetManifestResponse = __importStar(require("./get-manifest-response"));
exports.GetManifestResponse = GetManifestResponse.GetManifestResponse;
const MasterNodeHostType = __importStar(require("./master-node-host-type"));
exports.MasterNodeHostType = MasterNodeHostType.MasterNodeHostType;
const OpensearchCluster = __importStar(require("./opensearch-cluster"));
exports.OpensearchCluster = OpensearchCluster.OpensearchCluster;
const OpensearchClusterBackup = __importStar(require("./opensearch-cluster-backup"));
exports.OpensearchClusterBackup = OpensearchClusterBackup.OpensearchClusterBackup;
const OpensearchClusterBackupCollection = __importStar(require("./opensearch-cluster-backup-collection"));
exports.OpensearchClusterBackupCollection = OpensearchClusterBackupCollection.OpensearchClusterBackupCollection;
const OpensearchClusterBackupSummary = __importStar(require("./opensearch-cluster-backup-summary"));
exports.OpensearchClusterBackupSummary = OpensearchClusterBackupSummary.OpensearchClusterBackupSummary;
const OpensearchClusterCollection = __importStar(require("./opensearch-cluster-collection"));
exports.OpensearchClusterCollection = OpensearchClusterCollection.OpensearchClusterCollection;
const OpensearchClusterSummary = __importStar(require("./opensearch-cluster-summary"));
exports.OpensearchClusterSummary = OpensearchClusterSummary.OpensearchClusterSummary;
const OpensearchVersionsCollection = __importStar(require("./opensearch-versions-collection"));
exports.OpensearchVersionsCollection = OpensearchVersionsCollection.OpensearchVersionsCollection;
const OpensearchVersionsSummary = __importStar(require("./opensearch-versions-summary"));
exports.OpensearchVersionsSummary = OpensearchVersionsSummary.OpensearchVersionsSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ResizeOpensearchClusterHorizontalDetails = __importStar(require("./resize-opensearch-cluster-horizontal-details"));
exports.ResizeOpensearchClusterHorizontalDetails = ResizeOpensearchClusterHorizontalDetails.ResizeOpensearchClusterHorizontalDetails;
const ResizeOpensearchClusterVerticalDetails = __importStar(require("./resize-opensearch-cluster-vertical-details"));
exports.ResizeOpensearchClusterVerticalDetails = ResizeOpensearchClusterVerticalDetails.ResizeOpensearchClusterVerticalDetails;
const RestoreOpensearchClusterBackupDetails = __importStar(require("./restore-opensearch-cluster-backup-details"));
exports.RestoreOpensearchClusterBackupDetails = RestoreOpensearchClusterBackupDetails.RestoreOpensearchClusterBackupDetails;
const RestoreOpensearchClusterDetails = __importStar(require("./restore-opensearch-cluster-details"));
exports.RestoreOpensearchClusterDetails = RestoreOpensearchClusterDetails.RestoreOpensearchClusterDetails;
const SecurityMode = __importStar(require("./security-mode"));
exports.SecurityMode = SecurityMode.SecurityMode;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateCheckinDetails = __importStar(require("./update-checkin-details"));
exports.UpdateCheckinDetails = UpdateCheckinDetails.UpdateCheckinDetails;
const UpdateClusterHardenedImageDetails = __importStar(require("./update-cluster-hardened-image-details"));
exports.UpdateClusterHardenedImageDetails = UpdateClusterHardenedImageDetails.UpdateClusterHardenedImageDetails;
const UpdateClusterStatusDetails = __importStar(require("./update-cluster-status-details"));
exports.UpdateClusterStatusDetails = UpdateClusterStatusDetails.UpdateClusterStatusDetails;
const UpdateOpensearchClusterBackupDetails = __importStar(require("./update-opensearch-cluster-backup-details"));
exports.UpdateOpensearchClusterBackupDetails = UpdateOpensearchClusterBackupDetails.UpdateOpensearchClusterBackupDetails;
const UpdateOpensearchClusterDetails = __importStar(require("./update-opensearch-cluster-details"));
exports.UpdateOpensearchClusterDetails = UpdateOpensearchClusterDetails.UpdateOpensearchClusterDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
