// oci-opensearch/lib/model/master-node-host-type.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterNodeHostType = void 0;
/**
 * The instance type for the cluster's master nodes.
 **/
var MasterNodeHostType;
(function (MasterNodeHostType) {
    MasterNodeHostType["Flex"] = "FLEX";
    MasterNodeHostType["Bm"] = "BM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MasterNodeHostType["UnknownValue"] = "UNKNOWN_VALUE";
})(MasterNodeHostType = exports.MasterNodeHostType || (exports.MasterNodeHostType = {}));
(function (MasterNodeHostType) {
    function getJsonObj(obj) {
        return obj;
    }
    MasterNodeHostType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MasterNodeHostType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MasterNodeHostType = exports.MasterNodeHostType || (exports.MasterNodeHostType = {}));
