// oci-opensearch/lib/model/operation-type.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The types of operations that spawn work requests.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateOpensearchCluster"] = "CREATE_OPENSEARCH_CLUSTER";
    OperationType["UpdateOpensearchCluster"] = "UPDATE_OPENSEARCH_CLUSTER";
    OperationType["DeleteOpensearchCluster"] = "DELETE_OPENSEARCH_CLUSTER";
    OperationType["MoveOpensearchCluster"] = "MOVE_OPENSEARCH_CLUSTER";
    OperationType["RestoreOpensearchCluster"] = "RESTORE_OPENSEARCH_CLUSTER";
    OperationType["BackupOpensearchCluster"] = "BACKUP_OPENSEARCH_CLUSTER";
    OperationType["UpdateOpensearchClusterBackup"] = "UPDATE_OPENSEARCH_CLUSTER_BACKUP";
    OperationType["MoveOpensearchClusterBackup"] = "MOVE_OPENSEARCH_CLUSTER_BACKUP";
    OperationType["DeleteOpensearchClusterBackup"] = "DELETE_OPENSEARCH_CLUSTER_BACKUP";
    OperationType["UpdateOpensearchClusterSecurityConfig"] = "UPDATE_OPENSEARCH_CLUSTER_SECURITY_CONFIG";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
