// oci-opensearch/lib/model/security-mode.js
"use strict";
/**
 * OpenSearch Service API
 * The OpenSearch service API provides access to OCI Search Service with OpenSearch.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMode = void 0;
/**
 * The security mode of cluster
 **/
var SecurityMode;
(function (SecurityMode) {
    SecurityMode["Disabled"] = "DISABLED";
    SecurityMode["Permissive"] = "PERMISSIVE";
    SecurityMode["Enforcing"] = "ENFORCING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityMode["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityMode = exports.SecurityMode || (exports.SecurityMode = {}));
(function (SecurityMode) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityMode = exports.SecurityMode || (exports.SecurityMode = {}));
