// oci-certificatesmanagement/lib/request/list-associations-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAssociationsRequest = void 0;
var ListAssociationsRequest;
(function (ListAssociationsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
        SortBy["Timecreated"] = "TIMECREATED";
    })(SortBy = ListAssociationsRequest.SortBy || (ListAssociationsRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAssociationsRequest.SortOrder || (ListAssociationsRequest.SortOrder = {}));
    let AssociationType;
    (function (AssociationType) {
        AssociationType["Certificate"] = "CERTIFICATE";
        AssociationType["CertificateAuthority"] = "CERTIFICATE_AUTHORITY";
        AssociationType["CaBundle"] = "CA_BUNDLE";
    })(AssociationType = ListAssociationsRequest.AssociationType || (ListAssociationsRequest.AssociationType = {}));
})(ListAssociationsRequest = exports.ListAssociationsRequest || (exports.ListAssociationsRequest = {}));
