// oci-certificatesmanagement/lib/request/list-certificates-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCertificatesRequest = void 0;
var ListCertificatesRequest;
(function (ListCertificatesRequest) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["SchedulingDeletion"] = "SCHEDULING_DELETION";
        LifecycleState["PendingDeletion"] = "PENDING_DELETION";
        LifecycleState["CancellingDeletion"] = "CANCELLING_DELETION";
        LifecycleState["Failed"] = "FAILED";
    })(LifecycleState = ListCertificatesRequest.LifecycleState || (ListCertificatesRequest.LifecycleState = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Name"] = "NAME";
        SortBy["Expirationdate"] = "EXPIRATIONDATE";
        SortBy["Timecreated"] = "TIMECREATED";
    })(SortBy = ListCertificatesRequest.SortBy || (ListCertificatesRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListCertificatesRequest.SortOrder || (ListCertificatesRequest.SortOrder = {}));
})(ListCertificatesRequest = exports.ListCertificatesRequest || (exports.ListCertificatesRequest = {}));
