// oci-servicemanagerproxy/lib/model/index.js
"use strict";
/**
 * Service Manager Proxy API
 * Use the Service Manager Proxy API to obtain information about SaaS environments provisioned by Service Manager.
You can get information such as service types and service environment URLs.

 * OpenAPI spec version: 20210914
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEnvironmentSummary = exports.ServiceEnvironmentEndPointOverview = exports.ServiceEnvironmentCollection = exports.ServiceEnvironment = exports.ServiceEntitlementRegistrationStatus = exports.ServiceDefinition = exports.ErrorEntity = void 0;
const ErrorEntity = __importStar(require("./error-entity"));
exports.ErrorEntity = ErrorEntity.ErrorEntity;
const ServiceDefinition = __importStar(require("./service-definition"));
exports.ServiceDefinition = ServiceDefinition.ServiceDefinition;
const ServiceEntitlementRegistrationStatus = __importStar(require("./service-entitlement-registration-status"));
exports.ServiceEntitlementRegistrationStatus = ServiceEntitlementRegistrationStatus.ServiceEntitlementRegistrationStatus;
const ServiceEnvironment = __importStar(require("./service-environment"));
exports.ServiceEnvironment = ServiceEnvironment.ServiceEnvironment;
const ServiceEnvironmentCollection = __importStar(require("./service-environment-collection"));
exports.ServiceEnvironmentCollection = ServiceEnvironmentCollection.ServiceEnvironmentCollection;
const ServiceEnvironmentEndPointOverview = __importStar(require("./service-environment-end-point-overview"));
exports.ServiceEnvironmentEndPointOverview = ServiceEnvironmentEndPointOverview.ServiceEnvironmentEndPointOverview;
const ServiceEnvironmentSummary = __importStar(require("./service-environment-summary"));
exports.ServiceEnvironmentSummary = ServiceEnvironmentSummary.ServiceEnvironmentSummary;
