// oci-servicemanagerproxy/lib/model/service-environment-end-point-overview.js
"use strict";
/**
 * Service Manager Proxy API
 * Use the Service Manager Proxy API to obtain information about SaaS environments provisioned by Service Manager.
You can get information such as service types and service environment URLs.

 * OpenAPI spec version: 20210914
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEnvironmentEndPointOverview = void 0;
var ServiceEnvironmentEndPointOverview;
(function (ServiceEnvironmentEndPointOverview) {
    let EnvironmentType;
    (function (EnvironmentType) {
        EnvironmentType["InstanceUrlProd"] = "INSTANCE_URL_PROD";
        EnvironmentType["InstanceUrlTest"] = "INSTANCE_URL_TEST";
        EnvironmentType["InstanceUrlDev"] = "INSTANCE_URL_DEV";
        EnvironmentType["Other"] = "OTHER";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EnvironmentType["UnknownValue"] = "UNKNOWN_VALUE";
    })(EnvironmentType = ServiceEnvironmentEndPointOverview.EnvironmentType || (ServiceEnvironmentEndPointOverview.EnvironmentType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ServiceEnvironmentEndPointOverview.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ServiceEnvironmentEndPointOverview.getDeserializedJsonObj = getDeserializedJsonObj;
})(ServiceEnvironmentEndPointOverview = exports.ServiceEnvironmentEndPointOverview || (exports.ServiceEnvironmentEndPointOverview = {}));
