// oci-servicemanagerproxy/lib/model/service-entitlement-registration-status.js
"use strict";
/**
 * Service Manager Proxy API
 * Use the Service Manager Proxy API to obtain information about SaaS environments provisioned by Service Manager.
You can get information such as service types and service environment URLs.

 * OpenAPI spec version: 20210914
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEntitlementRegistrationStatus = void 0;
/**
 * Service Entitlement registration status
 **/
var ServiceEntitlementRegistrationStatus;
(function (ServiceEntitlementRegistrationStatus) {
    ServiceEntitlementRegistrationStatus["Initialized"] = "INITIALIZED";
    ServiceEntitlementRegistrationStatus["BeginActivation"] = "BEGIN_ACTIVATION";
    ServiceEntitlementRegistrationStatus["Active"] = "ACTIVE";
    ServiceEntitlementRegistrationStatus["BeginSoftTermination"] = "BEGIN_SOFT_TERMINATION";
    ServiceEntitlementRegistrationStatus["SoftTerminated"] = "SOFT_TERMINATED";
    ServiceEntitlementRegistrationStatus["BeginTermination"] = "BEGIN_TERMINATION";
    ServiceEntitlementRegistrationStatus["Canceled"] = "CANCELED";
    ServiceEntitlementRegistrationStatus["Terminated"] = "TERMINATED";
    ServiceEntitlementRegistrationStatus["BeginDisabling"] = "BEGIN_DISABLING";
    ServiceEntitlementRegistrationStatus["BeginEnabling"] = "BEGIN_ENABLING";
    ServiceEntitlementRegistrationStatus["BeginMigration"] = "BEGIN_MIGRATION";
    ServiceEntitlementRegistrationStatus["Disabled"] = "DISABLED";
    ServiceEntitlementRegistrationStatus["BeginSuspension"] = "BEGIN_SUSPENSION";
    ServiceEntitlementRegistrationStatus["BeginResumption"] = "BEGIN_RESUMPTION";
    ServiceEntitlementRegistrationStatus["Suspended"] = "SUSPENDED";
    ServiceEntitlementRegistrationStatus["BeginLockRelocation"] = "BEGIN_LOCK_RELOCATION";
    ServiceEntitlementRegistrationStatus["LockedRelocation"] = "LOCKED_RELOCATION";
    ServiceEntitlementRegistrationStatus["BeginRelocation"] = "BEGIN_RELOCATION";
    ServiceEntitlementRegistrationStatus["Relocated"] = "RELOCATED";
    ServiceEntitlementRegistrationStatus["BeginUnlockRelocation"] = "BEGIN_UNLOCK_RELOCATION";
    ServiceEntitlementRegistrationStatus["UnlockedRelocation"] = "UNLOCKED_RELOCATION";
    ServiceEntitlementRegistrationStatus["FailedLockRelocation"] = "FAILED_LOCK_RELOCATION";
    ServiceEntitlementRegistrationStatus["FailedActivation"] = "FAILED_ACTIVATION";
    ServiceEntitlementRegistrationStatus["FailedMigration"] = "FAILED_MIGRATION";
    ServiceEntitlementRegistrationStatus["AccessDisabled"] = "ACCESS_DISABLED";
    ServiceEntitlementRegistrationStatus["BeginDisablingAccess"] = "BEGIN_DISABLING_ACCESS";
    ServiceEntitlementRegistrationStatus["BeginEnablingAccess"] = "BEGIN_ENABLING_ACCESS";
    ServiceEntitlementRegistrationStatus["TraUnknown"] = "TRA_UNKNOWN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ServiceEntitlementRegistrationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(ServiceEntitlementRegistrationStatus = exports.ServiceEntitlementRegistrationStatus || (exports.ServiceEntitlementRegistrationStatus = {}));
(function (ServiceEntitlementRegistrationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    ServiceEntitlementRegistrationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ServiceEntitlementRegistrationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(ServiceEntitlementRegistrationStatus = exports.ServiceEntitlementRegistrationStatus || (exports.ServiceEntitlementRegistrationStatus = {}));
