// oci-osmanagement/lib/request/list-module-streams-on-managed-instance-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListModuleStreamsOnManagedInstanceRequest = void 0;
var ListModuleStreamsOnManagedInstanceRequest;
(function (ListModuleStreamsOnManagedInstanceRequest) {
    let StreamStatus;
    (function (StreamStatus) {
        StreamStatus["Enabled"] = "ENABLED";
        StreamStatus["Disabled"] = "DISABLED";
        StreamStatus["Active"] = "ACTIVE";
    })(StreamStatus = ListModuleStreamsOnManagedInstanceRequest.StreamStatus || (ListModuleStreamsOnManagedInstanceRequest.StreamStatus = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListModuleStreamsOnManagedInstanceRequest.SortOrder || (ListModuleStreamsOnManagedInstanceRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListModuleStreamsOnManagedInstanceRequest.SortBy || (ListModuleStreamsOnManagedInstanceRequest.SortBy = {}));
})(ListModuleStreamsOnManagedInstanceRequest = exports.ListModuleStreamsOnManagedInstanceRequest || (exports.ListModuleStreamsOnManagedInstanceRequest = {}));
