// oci-aianomalydetection/lib/model/data-source-type.js
"use strict";
/**
 * Anomaly Detection API
 * OCI AI Service solutions can help Enterprise customers integrate AI into their products immediately by using our proven,
pre-trained/custom models or containers, and without a need to set up in house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI/ML operations, shortening the time to market.

 * OpenAPI spec version: 20210101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSourceType = void 0;
/**
 * Possible data sources
 **/
var DataSourceType;
(function (DataSourceType) {
    DataSourceType["OracleObjectStorage"] = "ORACLE_OBJECT_STORAGE";
    DataSourceType["OracleAtp"] = "ORACLE_ATP";
    DataSourceType["Influx"] = "INFLUX";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DataSourceType["UnknownValue"] = "UNKNOWN_VALUE";
})(DataSourceType = exports.DataSourceType || (exports.DataSourceType = {}));
(function (DataSourceType) {
    function getJsonObj(obj) {
        return obj;
    }
    DataSourceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DataSourceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataSourceType = exports.DataSourceType || (exports.DataSourceType = {}));
