// oci-aianomalydetection/lib/model/detect-anomaly-job.js
"use strict";
/**
 * Anomaly Detection API
 * OCI AI Service solutions can help Enterprise customers integrate AI into their products immediately by using our proven,
pre-trained/custom models or containers, and without a need to set up in house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI/ML operations, shortening the time to market.

 * OpenAPI spec version: 20210101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetectAnomalyJob = void 0;
const model = __importStar(require("../model"));
var DetectAnomalyJob;
(function (DetectAnomalyJob) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["PartiallySucceeded"] = "PARTIALLY_SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = DetectAnomalyJob.LifecycleState || (DetectAnomalyJob.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "inputDetails": obj.inputDetails
                ? model.InputJobDetails.getJsonObj(obj.inputDetails)
                : undefined,
            "outputDetails": obj.outputDetails
                ? model.OutputJobDetails.getJsonObj(obj.outputDetails)
                : undefined
        });
        return jsonObj;
    }
    DetectAnomalyJob.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "inputDetails": obj.inputDetails
                ? model.InputJobDetails.getDeserializedJsonObj(obj.inputDetails)
                : undefined,
            "outputDetails": obj.outputDetails
                ? model.OutputJobDetails.getDeserializedJsonObj(obj.outputDetails)
                : undefined
        });
        return jsonObj;
    }
    DetectAnomalyJob.getDeserializedJsonObj = getDeserializedJsonObj;
})(DetectAnomalyJob = exports.DetectAnomalyJob || (exports.DetectAnomalyJob = {}));
