// oci-aianomalydetection/lib/model/index.js
"use strict";
/**
 * Anomaly Detection API
 * OCI AI Service solutions can help Enterprise customers integrate AI into their products immediately by using our proven,
pre-trained/custom models or containers, and without a need to set up in house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI/ML operations, shortening the time to market.

 * OpenAPI spec version: 20210101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDetectAnomalyJobDetails = exports.UpdateDataAssetDetails = exports.UpdateAiPrivateEndpointDetails = exports.SortOrder = exports.RowReductionDetails = exports.ProjectSummary = exports.ProjectCollection = exports.Project = exports.PerSignalDetails = exports.OutputJobDetails = exports.OutputDetails = exports.OperationType = exports.OperationStatus = exports.ObjectLocation = exports.ModelTrainingResults = exports.ModelTrainingDetails = exports.ModelSummary = exports.ModelCollection = exports.Model = exports.InputJobDetails = exports.InputDetails = exports.InfluxVersion = exports.InfluxDetails = exports.DetectionResultItem = exports.DetectAnomalyJobSummary = exports.DetectAnomalyJobCollection = exports.DetectAnomalyJob = exports.DetectAnomaliesDetails = exports.DataSourceType = exports.DataSourceDetails = exports.DataItem = exports.DataAssetSummary = exports.DataAssetCollection = exports.DataAsset = exports.CreateProjectDetails = exports.CreateModelDetails = exports.CreateDetectAnomalyJobDetails = exports.CreateDataAssetDetails = exports.CreateAiPrivateEndpointDetails = exports.ChangeProjectCompartmentDetails = exports.ChangeModelCompartmentDetails = exports.ChangeDetectAnomalyJobCompartmentDetails = exports.ChangeDataAssetCompartmentDetails = exports.ChangeAiPrivateEndpointCompartmentDetails = exports.AnomalyDetectResult = exports.Anomaly = exports.AiPrivateEndpointSummary = exports.AiPrivateEndpointCollection = exports.AiPrivateEndpoint = exports.ActionType = void 0;
exports.ObjectStoreOutputDetails = exports.ObjectStorageLocation = exports.ObjectListInputJobDetails = exports.ObjectListInputDetails = exports.InlineInputJobDetails = exports.InlineInputDetails = exports.InlineDetectAnomaliesRequest = exports.InfluxDetailsV2v0 = exports.InfluxDetailsV1v8 = exports.EmbeddedInputDetails = exports.EmbeddedDetectAnomaliesRequest = exports.DataSourceDetailsObjectStorage = exports.DataSourceDetailsInflux = exports.DataSourceDetailsATP = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateProjectDetails = exports.UpdateModelDetails = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AiPrivateEndpoint = __importStar(require("./ai-private-endpoint"));
exports.AiPrivateEndpoint = AiPrivateEndpoint.AiPrivateEndpoint;
const AiPrivateEndpointCollection = __importStar(require("./ai-private-endpoint-collection"));
exports.AiPrivateEndpointCollection = AiPrivateEndpointCollection.AiPrivateEndpointCollection;
const AiPrivateEndpointSummary = __importStar(require("./ai-private-endpoint-summary"));
exports.AiPrivateEndpointSummary = AiPrivateEndpointSummary.AiPrivateEndpointSummary;
const Anomaly = __importStar(require("./anomaly"));
exports.Anomaly = Anomaly.Anomaly;
const AnomalyDetectResult = __importStar(require("./anomaly-detect-result"));
exports.AnomalyDetectResult = AnomalyDetectResult.AnomalyDetectResult;
const ChangeAiPrivateEndpointCompartmentDetails = __importStar(require("./change-ai-private-endpoint-compartment-details"));
exports.ChangeAiPrivateEndpointCompartmentDetails = ChangeAiPrivateEndpointCompartmentDetails.ChangeAiPrivateEndpointCompartmentDetails;
const ChangeDataAssetCompartmentDetails = __importStar(require("./change-data-asset-compartment-details"));
exports.ChangeDataAssetCompartmentDetails = ChangeDataAssetCompartmentDetails.ChangeDataAssetCompartmentDetails;
const ChangeDetectAnomalyJobCompartmentDetails = __importStar(require("./change-detect-anomaly-job-compartment-details"));
exports.ChangeDetectAnomalyJobCompartmentDetails = ChangeDetectAnomalyJobCompartmentDetails.ChangeDetectAnomalyJobCompartmentDetails;
const ChangeModelCompartmentDetails = __importStar(require("./change-model-compartment-details"));
exports.ChangeModelCompartmentDetails = ChangeModelCompartmentDetails.ChangeModelCompartmentDetails;
const ChangeProjectCompartmentDetails = __importStar(require("./change-project-compartment-details"));
exports.ChangeProjectCompartmentDetails = ChangeProjectCompartmentDetails.ChangeProjectCompartmentDetails;
const CreateAiPrivateEndpointDetails = __importStar(require("./create-ai-private-endpoint-details"));
exports.CreateAiPrivateEndpointDetails = CreateAiPrivateEndpointDetails.CreateAiPrivateEndpointDetails;
const CreateDataAssetDetails = __importStar(require("./create-data-asset-details"));
exports.CreateDataAssetDetails = CreateDataAssetDetails.CreateDataAssetDetails;
const CreateDetectAnomalyJobDetails = __importStar(require("./create-detect-anomaly-job-details"));
exports.CreateDetectAnomalyJobDetails = CreateDetectAnomalyJobDetails.CreateDetectAnomalyJobDetails;
const CreateModelDetails = __importStar(require("./create-model-details"));
exports.CreateModelDetails = CreateModelDetails.CreateModelDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const DataAsset = __importStar(require("./data-asset"));
exports.DataAsset = DataAsset.DataAsset;
const DataAssetCollection = __importStar(require("./data-asset-collection"));
exports.DataAssetCollection = DataAssetCollection.DataAssetCollection;
const DataAssetSummary = __importStar(require("./data-asset-summary"));
exports.DataAssetSummary = DataAssetSummary.DataAssetSummary;
const DataItem = __importStar(require("./data-item"));
exports.DataItem = DataItem.DataItem;
const DataSourceDetails = __importStar(require("./data-source-details"));
exports.DataSourceDetails = DataSourceDetails.DataSourceDetails;
const DataSourceType = __importStar(require("./data-source-type"));
exports.DataSourceType = DataSourceType.DataSourceType;
const DetectAnomaliesDetails = __importStar(require("./detect-anomalies-details"));
exports.DetectAnomaliesDetails = DetectAnomaliesDetails.DetectAnomaliesDetails;
const DetectAnomalyJob = __importStar(require("./detect-anomaly-job"));
exports.DetectAnomalyJob = DetectAnomalyJob.DetectAnomalyJob;
const DetectAnomalyJobCollection = __importStar(require("./detect-anomaly-job-collection"));
exports.DetectAnomalyJobCollection = DetectAnomalyJobCollection.DetectAnomalyJobCollection;
const DetectAnomalyJobSummary = __importStar(require("./detect-anomaly-job-summary"));
exports.DetectAnomalyJobSummary = DetectAnomalyJobSummary.DetectAnomalyJobSummary;
const DetectionResultItem = __importStar(require("./detection-result-item"));
exports.DetectionResultItem = DetectionResultItem.DetectionResultItem;
const InfluxDetails = __importStar(require("./influx-details"));
exports.InfluxDetails = InfluxDetails.InfluxDetails;
const InfluxVersion = __importStar(require("./influx-version"));
exports.InfluxVersion = InfluxVersion.InfluxVersion;
const InputDetails = __importStar(require("./input-details"));
exports.InputDetails = InputDetails.InputDetails;
const InputJobDetails = __importStar(require("./input-job-details"));
exports.InputJobDetails = InputJobDetails.InputJobDetails;
const Model = __importStar(require("./model"));
exports.Model = Model.Model;
const ModelCollection = __importStar(require("./model-collection"));
exports.ModelCollection = ModelCollection.ModelCollection;
const ModelSummary = __importStar(require("./model-summary"));
exports.ModelSummary = ModelSummary.ModelSummary;
const ModelTrainingDetails = __importStar(require("./model-training-details"));
exports.ModelTrainingDetails = ModelTrainingDetails.ModelTrainingDetails;
const ModelTrainingResults = __importStar(require("./model-training-results"));
exports.ModelTrainingResults = ModelTrainingResults.ModelTrainingResults;
const ObjectLocation = __importStar(require("./object-location"));
exports.ObjectLocation = ObjectLocation.ObjectLocation;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OutputDetails = __importStar(require("./output-details"));
exports.OutputDetails = OutputDetails.OutputDetails;
const OutputJobDetails = __importStar(require("./output-job-details"));
exports.OutputJobDetails = OutputJobDetails.OutputJobDetails;
const PerSignalDetails = __importStar(require("./per-signal-details"));
exports.PerSignalDetails = PerSignalDetails.PerSignalDetails;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectCollection = __importStar(require("./project-collection"));
exports.ProjectCollection = ProjectCollection.ProjectCollection;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const RowReductionDetails = __importStar(require("./row-reduction-details"));
exports.RowReductionDetails = RowReductionDetails.RowReductionDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateAiPrivateEndpointDetails = __importStar(require("./update-ai-private-endpoint-details"));
exports.UpdateAiPrivateEndpointDetails = UpdateAiPrivateEndpointDetails.UpdateAiPrivateEndpointDetails;
const UpdateDataAssetDetails = __importStar(require("./update-data-asset-details"));
exports.UpdateDataAssetDetails = UpdateDataAssetDetails.UpdateDataAssetDetails;
const UpdateDetectAnomalyJobDetails = __importStar(require("./update-detect-anomaly-job-details"));
exports.UpdateDetectAnomalyJobDetails = UpdateDetectAnomalyJobDetails.UpdateDetectAnomalyJobDetails;
const UpdateModelDetails = __importStar(require("./update-model-details"));
exports.UpdateModelDetails = UpdateModelDetails.UpdateModelDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const DataSourceDetailsATP = __importStar(require("./data-source-details-at-p"));
exports.DataSourceDetailsATP = DataSourceDetailsATP.DataSourceDetailsATP;
const DataSourceDetailsInflux = __importStar(require("./data-source-details-influx"));
exports.DataSourceDetailsInflux = DataSourceDetailsInflux.DataSourceDetailsInflux;
const DataSourceDetailsObjectStorage = __importStar(require("./data-source-details-object-storage"));
exports.DataSourceDetailsObjectStorage = DataSourceDetailsObjectStorage.DataSourceDetailsObjectStorage;
const EmbeddedDetectAnomaliesRequest = __importStar(require("./embedded-detect-anomalies-request"));
exports.EmbeddedDetectAnomaliesRequest = EmbeddedDetectAnomaliesRequest.EmbeddedDetectAnomaliesRequest;
const EmbeddedInputDetails = __importStar(require("./embedded-input-details"));
exports.EmbeddedInputDetails = EmbeddedInputDetails.EmbeddedInputDetails;
const InfluxDetailsV1v8 = __importStar(require("./influx-details-v1v8"));
exports.InfluxDetailsV1v8 = InfluxDetailsV1v8.InfluxDetailsV1v8;
const InfluxDetailsV2v0 = __importStar(require("./influx-details-v2v0"));
exports.InfluxDetailsV2v0 = InfluxDetailsV2v0.InfluxDetailsV2v0;
const InlineDetectAnomaliesRequest = __importStar(require("./inline-detect-anomalies-request"));
exports.InlineDetectAnomaliesRequest = InlineDetectAnomaliesRequest.InlineDetectAnomaliesRequest;
const InlineInputDetails = __importStar(require("./inline-input-details"));
exports.InlineInputDetails = InlineInputDetails.InlineInputDetails;
const InlineInputJobDetails = __importStar(require("./inline-input-job-details"));
exports.InlineInputJobDetails = InlineInputJobDetails.InlineInputJobDetails;
const ObjectListInputDetails = __importStar(require("./object-list-input-details"));
exports.ObjectListInputDetails = ObjectListInputDetails.ObjectListInputDetails;
const ObjectListInputJobDetails = __importStar(require("./object-list-input-job-details"));
exports.ObjectListInputJobDetails = ObjectListInputJobDetails.ObjectListInputJobDetails;
const ObjectStorageLocation = __importStar(require("./object-storage-location"));
exports.ObjectStorageLocation = ObjectStorageLocation.ObjectStorageLocation;
const ObjectStoreOutputDetails = __importStar(require("./object-store-output-details"));
exports.ObjectStoreOutputDetails = ObjectStoreOutputDetails.ObjectStoreOutputDetails;
