// oci-waf/lib/model/index.js
"use strict";
/**
 * Web Application Firewall (WAF) API
 * API for the Web Application Firewall service.
Use this API to manage regional Web App Firewalls and corresponding policies for protecting HTTP services.

 * OpenAPI spec version: 20210930
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllowAction = exports.AccessControlRule = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequestCollection = exports.WorkRequest = exports.WebAppFirewallSummary = exports.WebAppFirewallPolicySummary = exports.WebAppFirewallPolicyRule = exports.WebAppFirewallPolicyCollection = exports.WebAppFirewallPolicy = exports.WebAppFirewallCollection = exports.WebAppFirewall = exports.UpdateWebAppFirewallPolicyDetails = exports.UpdateWebAppFirewallDetails = exports.UpdateNetworkAddressListDetails = exports.SortOrder = exports.ResponseProtection = exports.ResponseHeader = exports.ResponseAccessControl = exports.RequestRateLimitingConfiguration = exports.RequestRateLimiting = exports.RequestProtection = exports.RequestAccessControl = exports.ProtectionCapabilitySummary = exports.ProtectionCapabilitySettings = exports.ProtectionCapabilityGroupTagSummary = exports.ProtectionCapabilityGroupTagCollection = exports.ProtectionCapabilityExclusions = exports.ProtectionCapabilityCollection = exports.ProtectionCapability = exports.PrivateAddresses = exports.NetworkAddressListSummary = exports.NetworkAddressListCollection = exports.NetworkAddressList = exports.HttpResponseBody = exports.CreateWebAppFirewallPolicyDetails = exports.CreateWebAppFirewallDetails = exports.CreateNetworkAddressListDetails = exports.CollaborativeCapabilityWeightOverride = exports.CollaborativeCapabilityWeight = exports.ChangeWebAppFirewallPolicyCompartmentDetails = exports.ChangeWebAppFirewallCompartmentDetails = exports.ChangeResourceCompartmentDetails = exports.ChangeNetworkAddressListCompartmentDetails = exports.BackendType = exports.Action = void 0;
exports.WebAppFirewallLoadBalancerSummary = exports.WebAppFirewallLoadBalancer = exports.UpdateNetworkAddressListVcnAddressesDetails = exports.UpdateNetworkAddressListAddressesDetails = exports.StaticTextHttpResponseBody = exports.ReturnHttpResponseAction = exports.RequestRateLimitingRule = exports.ProtectionRule = exports.NetworkAddressListVcnAddressesSummary = exports.NetworkAddressListVcnAddresses = exports.NetworkAddressListAddressesSummary = exports.NetworkAddressListAddresses = exports.CreateWebAppFirewallLoadBalancerDetails = exports.CreateNetworkAddressListVcnAddressesDetails = exports.CreateNetworkAddressListAddressesDetails = exports.CheckAction = void 0;
const Action = __importStar(require("./action"));
exports.Action = Action.Action;
const BackendType = __importStar(require("./backend-type"));
exports.BackendType = BackendType.BackendType;
const ChangeNetworkAddressListCompartmentDetails = __importStar(require("./change-network-address-list-compartment-details"));
exports.ChangeNetworkAddressListCompartmentDetails = ChangeNetworkAddressListCompartmentDetails.ChangeNetworkAddressListCompartmentDetails;
const ChangeResourceCompartmentDetails = __importStar(require("./change-resource-compartment-details"));
exports.ChangeResourceCompartmentDetails = ChangeResourceCompartmentDetails.ChangeResourceCompartmentDetails;
const ChangeWebAppFirewallCompartmentDetails = __importStar(require("./change-web-app-firewall-compartment-details"));
exports.ChangeWebAppFirewallCompartmentDetails = ChangeWebAppFirewallCompartmentDetails.ChangeWebAppFirewallCompartmentDetails;
const ChangeWebAppFirewallPolicyCompartmentDetails = __importStar(require("./change-web-app-firewall-policy-compartment-details"));
exports.ChangeWebAppFirewallPolicyCompartmentDetails = ChangeWebAppFirewallPolicyCompartmentDetails.ChangeWebAppFirewallPolicyCompartmentDetails;
const CollaborativeCapabilityWeight = __importStar(require("./collaborative-capability-weight"));
exports.CollaborativeCapabilityWeight = CollaborativeCapabilityWeight.CollaborativeCapabilityWeight;
const CollaborativeCapabilityWeightOverride = __importStar(require("./collaborative-capability-weight-override"));
exports.CollaborativeCapabilityWeightOverride = CollaborativeCapabilityWeightOverride.CollaborativeCapabilityWeightOverride;
const CreateNetworkAddressListDetails = __importStar(require("./create-network-address-list-details"));
exports.CreateNetworkAddressListDetails = CreateNetworkAddressListDetails.CreateNetworkAddressListDetails;
const CreateWebAppFirewallDetails = __importStar(require("./create-web-app-firewall-details"));
exports.CreateWebAppFirewallDetails = CreateWebAppFirewallDetails.CreateWebAppFirewallDetails;
const CreateWebAppFirewallPolicyDetails = __importStar(require("./create-web-app-firewall-policy-details"));
exports.CreateWebAppFirewallPolicyDetails = CreateWebAppFirewallPolicyDetails.CreateWebAppFirewallPolicyDetails;
const HttpResponseBody = __importStar(require("./http-response-body"));
exports.HttpResponseBody = HttpResponseBody.HttpResponseBody;
const NetworkAddressList = __importStar(require("./network-address-list"));
exports.NetworkAddressList = NetworkAddressList.NetworkAddressList;
const NetworkAddressListCollection = __importStar(require("./network-address-list-collection"));
exports.NetworkAddressListCollection = NetworkAddressListCollection.NetworkAddressListCollection;
const NetworkAddressListSummary = __importStar(require("./network-address-list-summary"));
exports.NetworkAddressListSummary = NetworkAddressListSummary.NetworkAddressListSummary;
const PrivateAddresses = __importStar(require("./private-addresses"));
exports.PrivateAddresses = PrivateAddresses.PrivateAddresses;
const ProtectionCapability = __importStar(require("./protection-capability"));
exports.ProtectionCapability = ProtectionCapability.ProtectionCapability;
const ProtectionCapabilityCollection = __importStar(require("./protection-capability-collection"));
exports.ProtectionCapabilityCollection = ProtectionCapabilityCollection.ProtectionCapabilityCollection;
const ProtectionCapabilityExclusions = __importStar(require("./protection-capability-exclusions"));
exports.ProtectionCapabilityExclusions = ProtectionCapabilityExclusions.ProtectionCapabilityExclusions;
const ProtectionCapabilityGroupTagCollection = __importStar(require("./protection-capability-group-tag-collection"));
exports.ProtectionCapabilityGroupTagCollection = ProtectionCapabilityGroupTagCollection.ProtectionCapabilityGroupTagCollection;
const ProtectionCapabilityGroupTagSummary = __importStar(require("./protection-capability-group-tag-summary"));
exports.ProtectionCapabilityGroupTagSummary = ProtectionCapabilityGroupTagSummary.ProtectionCapabilityGroupTagSummary;
const ProtectionCapabilitySettings = __importStar(require("./protection-capability-settings"));
exports.ProtectionCapabilitySettings = ProtectionCapabilitySettings.ProtectionCapabilitySettings;
const ProtectionCapabilitySummary = __importStar(require("./protection-capability-summary"));
exports.ProtectionCapabilitySummary = ProtectionCapabilitySummary.ProtectionCapabilitySummary;
const RequestAccessControl = __importStar(require("./request-access-control"));
exports.RequestAccessControl = RequestAccessControl.RequestAccessControl;
const RequestProtection = __importStar(require("./request-protection"));
exports.RequestProtection = RequestProtection.RequestProtection;
const RequestRateLimiting = __importStar(require("./request-rate-limiting"));
exports.RequestRateLimiting = RequestRateLimiting.RequestRateLimiting;
const RequestRateLimitingConfiguration = __importStar(require("./request-rate-limiting-configuration"));
exports.RequestRateLimitingConfiguration = RequestRateLimitingConfiguration.RequestRateLimitingConfiguration;
const ResponseAccessControl = __importStar(require("./response-access-control"));
exports.ResponseAccessControl = ResponseAccessControl.ResponseAccessControl;
const ResponseHeader = __importStar(require("./response-header"));
exports.ResponseHeader = ResponseHeader.ResponseHeader;
const ResponseProtection = __importStar(require("./response-protection"));
exports.ResponseProtection = ResponseProtection.ResponseProtection;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateNetworkAddressListDetails = __importStar(require("./update-network-address-list-details"));
exports.UpdateNetworkAddressListDetails = UpdateNetworkAddressListDetails.UpdateNetworkAddressListDetails;
const UpdateWebAppFirewallDetails = __importStar(require("./update-web-app-firewall-details"));
exports.UpdateWebAppFirewallDetails = UpdateWebAppFirewallDetails.UpdateWebAppFirewallDetails;
const UpdateWebAppFirewallPolicyDetails = __importStar(require("./update-web-app-firewall-policy-details"));
exports.UpdateWebAppFirewallPolicyDetails = UpdateWebAppFirewallPolicyDetails.UpdateWebAppFirewallPolicyDetails;
const WebAppFirewall = __importStar(require("./web-app-firewall"));
exports.WebAppFirewall = WebAppFirewall.WebAppFirewall;
const WebAppFirewallCollection = __importStar(require("./web-app-firewall-collection"));
exports.WebAppFirewallCollection = WebAppFirewallCollection.WebAppFirewallCollection;
const WebAppFirewallPolicy = __importStar(require("./web-app-firewall-policy"));
exports.WebAppFirewallPolicy = WebAppFirewallPolicy.WebAppFirewallPolicy;
const WebAppFirewallPolicyCollection = __importStar(require("./web-app-firewall-policy-collection"));
exports.WebAppFirewallPolicyCollection = WebAppFirewallPolicyCollection.WebAppFirewallPolicyCollection;
const WebAppFirewallPolicyRule = __importStar(require("./web-app-firewall-policy-rule"));
exports.WebAppFirewallPolicyRule = WebAppFirewallPolicyRule.WebAppFirewallPolicyRule;
const WebAppFirewallPolicySummary = __importStar(require("./web-app-firewall-policy-summary"));
exports.WebAppFirewallPolicySummary = WebAppFirewallPolicySummary.WebAppFirewallPolicySummary;
const WebAppFirewallSummary = __importStar(require("./web-app-firewall-summary"));
exports.WebAppFirewallSummary = WebAppFirewallSummary.WebAppFirewallSummary;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestCollection = __importStar(require("./work-request-collection"));
exports.WorkRequestCollection = WorkRequestCollection.WorkRequestCollection;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const AccessControlRule = __importStar(require("./access-control-rule"));
exports.AccessControlRule = AccessControlRule.AccessControlRule;
const AllowAction = __importStar(require("./allow-action"));
exports.AllowAction = AllowAction.AllowAction;
const CheckAction = __importStar(require("./check-action"));
exports.CheckAction = CheckAction.CheckAction;
const CreateNetworkAddressListAddressesDetails = __importStar(require("./create-network-address-list-addresses-details"));
exports.CreateNetworkAddressListAddressesDetails = CreateNetworkAddressListAddressesDetails.CreateNetworkAddressListAddressesDetails;
const CreateNetworkAddressListVcnAddressesDetails = __importStar(require("./create-network-address-list-vcn-addresses-details"));
exports.CreateNetworkAddressListVcnAddressesDetails = CreateNetworkAddressListVcnAddressesDetails.CreateNetworkAddressListVcnAddressesDetails;
const CreateWebAppFirewallLoadBalancerDetails = __importStar(require("./create-web-app-firewall-load-balancer-details"));
exports.CreateWebAppFirewallLoadBalancerDetails = CreateWebAppFirewallLoadBalancerDetails.CreateWebAppFirewallLoadBalancerDetails;
const NetworkAddressListAddresses = __importStar(require("./network-address-list-addresses"));
exports.NetworkAddressListAddresses = NetworkAddressListAddresses.NetworkAddressListAddresses;
const NetworkAddressListAddressesSummary = __importStar(require("./network-address-list-addresses-summary"));
exports.NetworkAddressListAddressesSummary = NetworkAddressListAddressesSummary.NetworkAddressListAddressesSummary;
const NetworkAddressListVcnAddresses = __importStar(require("./network-address-list-vcn-addresses"));
exports.NetworkAddressListVcnAddresses = NetworkAddressListVcnAddresses.NetworkAddressListVcnAddresses;
const NetworkAddressListVcnAddressesSummary = __importStar(require("./network-address-list-vcn-addresses-summary"));
exports.NetworkAddressListVcnAddressesSummary = NetworkAddressListVcnAddressesSummary.NetworkAddressListVcnAddressesSummary;
const ProtectionRule = __importStar(require("./protection-rule"));
exports.ProtectionRule = ProtectionRule.ProtectionRule;
const RequestRateLimitingRule = __importStar(require("./request-rate-limiting-rule"));
exports.RequestRateLimitingRule = RequestRateLimitingRule.RequestRateLimitingRule;
const ReturnHttpResponseAction = __importStar(require("./return-http-response-action"));
exports.ReturnHttpResponseAction = ReturnHttpResponseAction.ReturnHttpResponseAction;
const StaticTextHttpResponseBody = __importStar(require("./static-text-http-response-body"));
exports.StaticTextHttpResponseBody = StaticTextHttpResponseBody.StaticTextHttpResponseBody;
const UpdateNetworkAddressListAddressesDetails = __importStar(require("./update-network-address-list-addresses-details"));
exports.UpdateNetworkAddressListAddressesDetails = UpdateNetworkAddressListAddressesDetails.UpdateNetworkAddressListAddressesDetails;
const UpdateNetworkAddressListVcnAddressesDetails = __importStar(require("./update-network-address-list-vcn-addresses-details"));
exports.UpdateNetworkAddressListVcnAddressesDetails = UpdateNetworkAddressListVcnAddressesDetails.UpdateNetworkAddressListVcnAddressesDetails;
const WebAppFirewallLoadBalancer = __importStar(require("./web-app-firewall-load-balancer"));
exports.WebAppFirewallLoadBalancer = WebAppFirewallLoadBalancer.WebAppFirewallLoadBalancer;
const WebAppFirewallLoadBalancerSummary = __importStar(require("./web-app-firewall-load-balancer-summary"));
exports.WebAppFirewallLoadBalancerSummary = WebAppFirewallLoadBalancerSummary.WebAppFirewallLoadBalancerSummary;
