// oci-waf/lib/model/protection-capability-summary.js
"use strict";
/**
 * Web Application Firewall (WAF) API
 * API for the Web Application Firewall service.
Use this API to manage regional Web App Firewalls and corresponding policies for protecting HTTP services.

 * OpenAPI spec version: 20210930
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionCapabilitySummary = void 0;
const model = __importStar(require("../model"));
var ProtectionCapabilitySummary;
(function (ProtectionCapabilitySummary) {
    let Type;
    (function (Type) {
        Type["RequestProtectionCapability"] = "REQUEST_PROTECTION_CAPABILITY";
        Type["ResponseProtectionCapability"] = "RESPONSE_PROTECTION_CAPABILITY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = ProtectionCapabilitySummary.Type || (ProtectionCapabilitySummary.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "collaborativeWeights": obj.collaborativeWeights
                ? obj.collaborativeWeights.map(item => {
                    return model.CollaborativeCapabilityWeight.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ProtectionCapabilitySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "collaborativeWeights": obj.collaborativeWeights
                ? obj.collaborativeWeights.map(item => {
                    return model.CollaborativeCapabilityWeight.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ProtectionCapabilitySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProtectionCapabilitySummary = exports.ProtectionCapabilitySummary || (exports.ProtectionCapabilitySummary = {}));
