// oci-waf/lib/model/web-app-firewall-policy-rule.js
"use strict";
/**
 * Web Application Firewall (WAF) API
 * API for the Web Application Firewall service.
Use this API to manage regional Web App Firewalls and corresponding policies for protecting HTTP services.

 * OpenAPI spec version: 20210930
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppFirewallPolicyRule = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var WebAppFirewallPolicyRule;
(function (WebAppFirewallPolicyRule) {
    let ConditionLanguage;
    (function (ConditionLanguage) {
        ConditionLanguage["Jmespath"] = "JMESPATH";
    })(ConditionLanguage = WebAppFirewallPolicyRule.ConditionLanguage || (WebAppFirewallPolicyRule.ConditionLanguage = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "PROTECTION":
                    return model.ProtectionRule.getJsonObj(jsonObj, true);
                case "REQUEST_RATE_LIMITING":
                    return model.RequestRateLimitingRule.getJsonObj(jsonObj, true);
                case "ACCESS_CONTROL":
                    return model.AccessControlRule.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    WebAppFirewallPolicyRule.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "PROTECTION":
                    return model.ProtectionRule.getDeserializedJsonObj(jsonObj, true);
                case "REQUEST_RATE_LIMITING":
                    return model.RequestRateLimitingRule.getDeserializedJsonObj(jsonObj, true);
                case "ACCESS_CONTROL":
                    return model.AccessControlRule.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    WebAppFirewallPolicyRule.getDeserializedJsonObj = getDeserializedJsonObj;
})(WebAppFirewallPolicyRule = exports.WebAppFirewallPolicyRule || (exports.WebAppFirewallPolicyRule = {}));
