// oci-cloudbridge/lib/model/asset-source-lifecycle-state.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSourceLifecycleState = void 0;
/**
 * Lifecycle state of an asset source.
 **/
var AssetSourceLifecycleState;
(function (AssetSourceLifecycleState) {
    AssetSourceLifecycleState["Creating"] = "CREATING";
    AssetSourceLifecycleState["Active"] = "ACTIVE";
    AssetSourceLifecycleState["Deleting"] = "DELETING";
    AssetSourceLifecycleState["Deleted"] = "DELETED";
    AssetSourceLifecycleState["Failed"] = "FAILED";
    AssetSourceLifecycleState["Updating"] = "UPDATING";
    AssetSourceLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AssetSourceLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AssetSourceLifecycleState = exports.AssetSourceLifecycleState || (exports.AssetSourceLifecycleState = {}));
(function (AssetSourceLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AssetSourceLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AssetSourceLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AssetSourceLifecycleState = exports.AssetSourceLifecycleState || (exports.AssetSourceLifecycleState = {}));
