// oci-cloudbridge/lib/model/asset-type.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetType = void 0;
/**
 * Asset type.
 **/
var AssetType;
(function (AssetType) {
    AssetType["VmwareVm"] = "VMWARE_VM";
    AssetType["Vm"] = "VM";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AssetType["UnknownValue"] = "UNKNOWN_VALUE";
})(AssetType = exports.AssetType || (exports.AssetType = {}));
(function (AssetType) {
    function getJsonObj(obj) {
        return obj;
    }
    AssetType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AssetType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AssetType = exports.AssetType || (exports.AssetType = {}));
