// oci-cloudbridge/lib/model/asset-source-connection-lifecycle-state.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetSourceConnectionLifecycleState = void 0;
/**
 * Lifecycle state of a connection.
 **/
var AssetSourceConnectionLifecycleState;
(function (AssetSourceConnectionLifecycleState) {
    AssetSourceConnectionLifecycleState["Active"] = "ACTIVE";
    AssetSourceConnectionLifecycleState["Updating"] = "UPDATING";
    AssetSourceConnectionLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    AssetSourceConnectionLifecycleState["Deleted"] = "DELETED";
    AssetSourceConnectionLifecycleState["Creating"] = "CREATING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AssetSourceConnectionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AssetSourceConnectionLifecycleState = exports.AssetSourceConnectionLifecycleState || (exports.AssetSourceConnectionLifecycleState = {}));
(function (AssetSourceConnectionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AssetSourceConnectionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AssetSourceConnectionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AssetSourceConnectionLifecycleState = exports.AssetSourceConnectionLifecycleState || (exports.AssetSourceConnectionLifecycleState = {}));
