// oci-cloudbridge/lib/model/index.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentCollection = exports.Environment = exports.Disk = exports.DiscoveryScheduleSummary = exports.DiscoveryScheduleLifecycleState = exports.DiscoveryScheduleCollection = exports.DiscoverySchedule = exports.CustomerTag = exports.CreateInventoryDetails = exports.CreateEnvironmentDetails = exports.CreateDiscoveryScheduleDetails = exports.CreateAssetSourceDetails = exports.CreateAssetDetails = exports.CreateAgentDetails = exports.CreateAgentDependencyDetails = exports.ComputeProperties = exports.ChangeEnvironmentCompartmentDetails = exports.ChangeDiscoveryScheduleCompartmentDetails = exports.ChangeAssetTagsDetails = exports.ChangeAssetSourceCompartmentDetails = exports.ChangeAssetCompartmentDetails = exports.ChangeAgentDependencyCompartmentDetails = exports.ChangeAgentCompartmentDetails = exports.AssetType = exports.AssetSummary = exports.AssetSourceType = exports.AssetSourceSummary = exports.AssetSourceLifecycleState = exports.AssetSourceCredentialsType = exports.AssetSourceCredentials = exports.AssetSourceConnectionType = exports.AssetSourceConnectionLifecycleState = exports.AssetSourceConnectionCollection = exports.AssetSourceConnection = exports.AssetSourceCollection = exports.AssetSource = exports.AssetCollection = exports.AssetAggregationCollection = exports.AssetAggregation = exports.Asset = exports.ApplianceImageSummary = exports.ApplianceImageCollection = exports.AgentSummary = exports.AgentDependencySummary = exports.AgentDependencyCollection = exports.AgentDependency = exports.AgentCollection = exports.Agent = exports.AddAgentDependencyDetails = exports.ActionType = void 0;
exports.VmwareVmAsset = exports.VmWareAssetSourceSummary = exports.VmWareAssetSource = exports.VmAsset = exports.UpdateVmwareVmAssetDetails = exports.UpdateVmWareAssetSourceDetails = exports.UpdateVmAssetDetails = exports.ImportInventoryViaAssetsDetails = exports.CreateVmwareVmAssetDetails = exports.CreateVmWareAssetSourceDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VmwareVmProperties = exports.VmwareVCenterProperties = exports.VmProperties = exports.UpdatePluginDetails = exports.UpdateInventoryDetails = exports.UpdateEnvironmentDetails = exports.UpdateDiscoveryScheduleDetails = exports.UpdateAssetSourceDetails = exports.UpdateAssetDetails = exports.UpdateAgentDetails = exports.UpdateAgentDependencyDetails = exports.SubmitHistoricalMetricsDetails = exports.SortOrders = exports.ScsiController = exports.RemoveAgentDependencyDetails = exports.PluginSummary = exports.Plugin = exports.OperationType = exports.OperationStatus = exports.NvdimmController = exports.Nvdimm = exports.Nic = exports.InventorySummary = exports.InventoryCollection = exports.Inventory = exports.ImportInventoryDetails = exports.HistoricalMetricSummary = exports.HistoricalMetricCollection = exports.HistoricalMetric = exports.GpuDevice = exports.EnvironmentSummary = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AddAgentDependencyDetails = __importStar(require("./add-agent-dependency-details"));
exports.AddAgentDependencyDetails = AddAgentDependencyDetails.AddAgentDependencyDetails;
const Agent = __importStar(require("./agent"));
exports.Agent = Agent.Agent;
const AgentCollection = __importStar(require("./agent-collection"));
exports.AgentCollection = AgentCollection.AgentCollection;
const AgentDependency = __importStar(require("./agent-dependency"));
exports.AgentDependency = AgentDependency.AgentDependency;
const AgentDependencyCollection = __importStar(require("./agent-dependency-collection"));
exports.AgentDependencyCollection = AgentDependencyCollection.AgentDependencyCollection;
const AgentDependencySummary = __importStar(require("./agent-dependency-summary"));
exports.AgentDependencySummary = AgentDependencySummary.AgentDependencySummary;
const AgentSummary = __importStar(require("./agent-summary"));
exports.AgentSummary = AgentSummary.AgentSummary;
const ApplianceImageCollection = __importStar(require("./appliance-image-collection"));
exports.ApplianceImageCollection = ApplianceImageCollection.ApplianceImageCollection;
const ApplianceImageSummary = __importStar(require("./appliance-image-summary"));
exports.ApplianceImageSummary = ApplianceImageSummary.ApplianceImageSummary;
const Asset = __importStar(require("./asset"));
exports.Asset = Asset.Asset;
const AssetAggregation = __importStar(require("./asset-aggregation"));
exports.AssetAggregation = AssetAggregation.AssetAggregation;
const AssetAggregationCollection = __importStar(require("./asset-aggregation-collection"));
exports.AssetAggregationCollection = AssetAggregationCollection.AssetAggregationCollection;
const AssetCollection = __importStar(require("./asset-collection"));
exports.AssetCollection = AssetCollection.AssetCollection;
const AssetSource = __importStar(require("./asset-source"));
exports.AssetSource = AssetSource.AssetSource;
const AssetSourceCollection = __importStar(require("./asset-source-collection"));
exports.AssetSourceCollection = AssetSourceCollection.AssetSourceCollection;
const AssetSourceConnection = __importStar(require("./asset-source-connection"));
exports.AssetSourceConnection = AssetSourceConnection.AssetSourceConnection;
const AssetSourceConnectionCollection = __importStar(require("./asset-source-connection-collection"));
exports.AssetSourceConnectionCollection = AssetSourceConnectionCollection.AssetSourceConnectionCollection;
const AssetSourceConnectionLifecycleState = __importStar(require("./asset-source-connection-lifecycle-state"));
exports.AssetSourceConnectionLifecycleState = AssetSourceConnectionLifecycleState.AssetSourceConnectionLifecycleState;
const AssetSourceConnectionType = __importStar(require("./asset-source-connection-type"));
exports.AssetSourceConnectionType = AssetSourceConnectionType.AssetSourceConnectionType;
const AssetSourceCredentials = __importStar(require("./asset-source-credentials"));
exports.AssetSourceCredentials = AssetSourceCredentials.AssetSourceCredentials;
const AssetSourceCredentialsType = __importStar(require("./asset-source-credentials-type"));
exports.AssetSourceCredentialsType = AssetSourceCredentialsType.AssetSourceCredentialsType;
const AssetSourceLifecycleState = __importStar(require("./asset-source-lifecycle-state"));
exports.AssetSourceLifecycleState = AssetSourceLifecycleState.AssetSourceLifecycleState;
const AssetSourceSummary = __importStar(require("./asset-source-summary"));
exports.AssetSourceSummary = AssetSourceSummary.AssetSourceSummary;
const AssetSourceType = __importStar(require("./asset-source-type"));
exports.AssetSourceType = AssetSourceType.AssetSourceType;
const AssetSummary = __importStar(require("./asset-summary"));
exports.AssetSummary = AssetSummary.AssetSummary;
const AssetType = __importStar(require("./asset-type"));
exports.AssetType = AssetType.AssetType;
const ChangeAgentCompartmentDetails = __importStar(require("./change-agent-compartment-details"));
exports.ChangeAgentCompartmentDetails = ChangeAgentCompartmentDetails.ChangeAgentCompartmentDetails;
const ChangeAgentDependencyCompartmentDetails = __importStar(require("./change-agent-dependency-compartment-details"));
exports.ChangeAgentDependencyCompartmentDetails = ChangeAgentDependencyCompartmentDetails.ChangeAgentDependencyCompartmentDetails;
const ChangeAssetCompartmentDetails = __importStar(require("./change-asset-compartment-details"));
exports.ChangeAssetCompartmentDetails = ChangeAssetCompartmentDetails.ChangeAssetCompartmentDetails;
const ChangeAssetSourceCompartmentDetails = __importStar(require("./change-asset-source-compartment-details"));
exports.ChangeAssetSourceCompartmentDetails = ChangeAssetSourceCompartmentDetails.ChangeAssetSourceCompartmentDetails;
const ChangeAssetTagsDetails = __importStar(require("./change-asset-tags-details"));
exports.ChangeAssetTagsDetails = ChangeAssetTagsDetails.ChangeAssetTagsDetails;
const ChangeDiscoveryScheduleCompartmentDetails = __importStar(require("./change-discovery-schedule-compartment-details"));
exports.ChangeDiscoveryScheduleCompartmentDetails = ChangeDiscoveryScheduleCompartmentDetails.ChangeDiscoveryScheduleCompartmentDetails;
const ChangeEnvironmentCompartmentDetails = __importStar(require("./change-environment-compartment-details"));
exports.ChangeEnvironmentCompartmentDetails = ChangeEnvironmentCompartmentDetails.ChangeEnvironmentCompartmentDetails;
const ComputeProperties = __importStar(require("./compute-properties"));
exports.ComputeProperties = ComputeProperties.ComputeProperties;
const CreateAgentDependencyDetails = __importStar(require("./create-agent-dependency-details"));
exports.CreateAgentDependencyDetails = CreateAgentDependencyDetails.CreateAgentDependencyDetails;
const CreateAgentDetails = __importStar(require("./create-agent-details"));
exports.CreateAgentDetails = CreateAgentDetails.CreateAgentDetails;
const CreateAssetDetails = __importStar(require("./create-asset-details"));
exports.CreateAssetDetails = CreateAssetDetails.CreateAssetDetails;
const CreateAssetSourceDetails = __importStar(require("./create-asset-source-details"));
exports.CreateAssetSourceDetails = CreateAssetSourceDetails.CreateAssetSourceDetails;
const CreateDiscoveryScheduleDetails = __importStar(require("./create-discovery-schedule-details"));
exports.CreateDiscoveryScheduleDetails = CreateDiscoveryScheduleDetails.CreateDiscoveryScheduleDetails;
const CreateEnvironmentDetails = __importStar(require("./create-environment-details"));
exports.CreateEnvironmentDetails = CreateEnvironmentDetails.CreateEnvironmentDetails;
const CreateInventoryDetails = __importStar(require("./create-inventory-details"));
exports.CreateInventoryDetails = CreateInventoryDetails.CreateInventoryDetails;
const CustomerTag = __importStar(require("./customer-tag"));
exports.CustomerTag = CustomerTag.CustomerTag;
const DiscoverySchedule = __importStar(require("./discovery-schedule"));
exports.DiscoverySchedule = DiscoverySchedule.DiscoverySchedule;
const DiscoveryScheduleCollection = __importStar(require("./discovery-schedule-collection"));
exports.DiscoveryScheduleCollection = DiscoveryScheduleCollection.DiscoveryScheduleCollection;
const DiscoveryScheduleLifecycleState = __importStar(require("./discovery-schedule-lifecycle-state"));
exports.DiscoveryScheduleLifecycleState = DiscoveryScheduleLifecycleState.DiscoveryScheduleLifecycleState;
const DiscoveryScheduleSummary = __importStar(require("./discovery-schedule-summary"));
exports.DiscoveryScheduleSummary = DiscoveryScheduleSummary.DiscoveryScheduleSummary;
const Disk = __importStar(require("./disk"));
exports.Disk = Disk.Disk;
const Environment = __importStar(require("./environment"));
exports.Environment = Environment.Environment;
const EnvironmentCollection = __importStar(require("./environment-collection"));
exports.EnvironmentCollection = EnvironmentCollection.EnvironmentCollection;
const EnvironmentSummary = __importStar(require("./environment-summary"));
exports.EnvironmentSummary = EnvironmentSummary.EnvironmentSummary;
const GpuDevice = __importStar(require("./gpu-device"));
exports.GpuDevice = GpuDevice.GpuDevice;
const HistoricalMetric = __importStar(require("./historical-metric"));
exports.HistoricalMetric = HistoricalMetric.HistoricalMetric;
const HistoricalMetricCollection = __importStar(require("./historical-metric-collection"));
exports.HistoricalMetricCollection = HistoricalMetricCollection.HistoricalMetricCollection;
const HistoricalMetricSummary = __importStar(require("./historical-metric-summary"));
exports.HistoricalMetricSummary = HistoricalMetricSummary.HistoricalMetricSummary;
const ImportInventoryDetails = __importStar(require("./import-inventory-details"));
exports.ImportInventoryDetails = ImportInventoryDetails.ImportInventoryDetails;
const Inventory = __importStar(require("./inventory"));
exports.Inventory = Inventory.Inventory;
const InventoryCollection = __importStar(require("./inventory-collection"));
exports.InventoryCollection = InventoryCollection.InventoryCollection;
const InventorySummary = __importStar(require("./inventory-summary"));
exports.InventorySummary = InventorySummary.InventorySummary;
const Nic = __importStar(require("./nic"));
exports.Nic = Nic.Nic;
const Nvdimm = __importStar(require("./nvdimm"));
exports.Nvdimm = Nvdimm.Nvdimm;
const NvdimmController = __importStar(require("./nvdimm-controller"));
exports.NvdimmController = NvdimmController.NvdimmController;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const Plugin = __importStar(require("./plugin"));
exports.Plugin = Plugin.Plugin;
const PluginSummary = __importStar(require("./plugin-summary"));
exports.PluginSummary = PluginSummary.PluginSummary;
const RemoveAgentDependencyDetails = __importStar(require("./remove-agent-dependency-details"));
exports.RemoveAgentDependencyDetails = RemoveAgentDependencyDetails.RemoveAgentDependencyDetails;
const ScsiController = __importStar(require("./scsi-controller"));
exports.ScsiController = ScsiController.ScsiController;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const SubmitHistoricalMetricsDetails = __importStar(require("./submit-historical-metrics-details"));
exports.SubmitHistoricalMetricsDetails = SubmitHistoricalMetricsDetails.SubmitHistoricalMetricsDetails;
const UpdateAgentDependencyDetails = __importStar(require("./update-agent-dependency-details"));
exports.UpdateAgentDependencyDetails = UpdateAgentDependencyDetails.UpdateAgentDependencyDetails;
const UpdateAgentDetails = __importStar(require("./update-agent-details"));
exports.UpdateAgentDetails = UpdateAgentDetails.UpdateAgentDetails;
const UpdateAssetDetails = __importStar(require("./update-asset-details"));
exports.UpdateAssetDetails = UpdateAssetDetails.UpdateAssetDetails;
const UpdateAssetSourceDetails = __importStar(require("./update-asset-source-details"));
exports.UpdateAssetSourceDetails = UpdateAssetSourceDetails.UpdateAssetSourceDetails;
const UpdateDiscoveryScheduleDetails = __importStar(require("./update-discovery-schedule-details"));
exports.UpdateDiscoveryScheduleDetails = UpdateDiscoveryScheduleDetails.UpdateDiscoveryScheduleDetails;
const UpdateEnvironmentDetails = __importStar(require("./update-environment-details"));
exports.UpdateEnvironmentDetails = UpdateEnvironmentDetails.UpdateEnvironmentDetails;
const UpdateInventoryDetails = __importStar(require("./update-inventory-details"));
exports.UpdateInventoryDetails = UpdateInventoryDetails.UpdateInventoryDetails;
const UpdatePluginDetails = __importStar(require("./update-plugin-details"));
exports.UpdatePluginDetails = UpdatePluginDetails.UpdatePluginDetails;
const VmProperties = __importStar(require("./vm-properties"));
exports.VmProperties = VmProperties.VmProperties;
const VmwareVCenterProperties = __importStar(require("./vmware-vcenter-properties"));
exports.VmwareVCenterProperties = VmwareVCenterProperties.VmwareVCenterProperties;
const VmwareVmProperties = __importStar(require("./vmware-vm-properties"));
exports.VmwareVmProperties = VmwareVmProperties.VmwareVmProperties;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const CreateVmWareAssetSourceDetails = __importStar(require("./create-vm-ware-asset-source-details"));
exports.CreateVmWareAssetSourceDetails = CreateVmWareAssetSourceDetails.CreateVmWareAssetSourceDetails;
const CreateVmwareVmAssetDetails = __importStar(require("./create-vmware-vm-asset-details"));
exports.CreateVmwareVmAssetDetails = CreateVmwareVmAssetDetails.CreateVmwareVmAssetDetails;
const ImportInventoryViaAssetsDetails = __importStar(require("./import-inventory-via-assets-details"));
exports.ImportInventoryViaAssetsDetails = ImportInventoryViaAssetsDetails.ImportInventoryViaAssetsDetails;
const UpdateVmAssetDetails = __importStar(require("./update-vm-asset-details"));
exports.UpdateVmAssetDetails = UpdateVmAssetDetails.UpdateVmAssetDetails;
const UpdateVmWareAssetSourceDetails = __importStar(require("./update-vm-ware-asset-source-details"));
exports.UpdateVmWareAssetSourceDetails = UpdateVmWareAssetSourceDetails.UpdateVmWareAssetSourceDetails;
const UpdateVmwareVmAssetDetails = __importStar(require("./update-vmware-vm-asset-details"));
exports.UpdateVmwareVmAssetDetails = UpdateVmwareVmAssetDetails.UpdateVmwareVmAssetDetails;
const VmAsset = __importStar(require("./vm-asset"));
exports.VmAsset = VmAsset.VmAsset;
const VmWareAssetSource = __importStar(require("./vm-ware-asset-source"));
exports.VmWareAssetSource = VmWareAssetSource.VmWareAssetSource;
const VmWareAssetSourceSummary = __importStar(require("./vm-ware-asset-source-summary"));
exports.VmWareAssetSourceSummary = VmWareAssetSourceSummary.VmWareAssetSourceSummary;
const VmwareVmAsset = __importStar(require("./vmware-vm-asset"));
exports.VmwareVmAsset = VmwareVmAsset.VmwareVmAsset;
