// oci-cloudbridge/lib/model/operation-type.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateEnvironment"] = "CREATE_ENVIRONMENT";
    OperationType["UpdateEnvironment"] = "UPDATE_ENVIRONMENT";
    OperationType["DeleteEnvironment"] = "DELETE_ENVIRONMENT";
    OperationType["MoveEnvironment"] = "MOVE_ENVIRONMENT";
    OperationType["CreateOcbAgent"] = "CREATE_OCB_AGENT";
    OperationType["UpdateOcbAgent"] = "UPDATE_OCB_AGENT";
    OperationType["DeleteOcbAgent"] = "DELETE_OCB_AGENT";
    OperationType["MoveOcbAgent"] = "MOVE_OCB_AGENT";
    OperationType["CreateAgentDependency"] = "CREATE_AGENT_DEPENDENCY";
    OperationType["UpdateAgentDependency"] = "UPDATE_AGENT_DEPENDENCY";
    OperationType["DeleteAgentDependency"] = "DELETE_AGENT_DEPENDENCY";
    OperationType["MoveAgentDependency"] = "MOVE_AGENT_DEPENDENCY";
    OperationType["CreateInventory"] = "CREATE_INVENTORY";
    OperationType["DeleteInventory"] = "DELETE_INVENTORY";
    OperationType["ImportInventory"] = "IMPORT_INVENTORY";
    OperationType["DeleteAssetSource"] = "DELETE_ASSET_SOURCE";
    OperationType["RefreshAssetSource"] = "REFRESH_ASSET_SOURCE";
    OperationType["CreateAssetSource"] = "CREATE_ASSET_SOURCE";
    OperationType["UpdateAssetSource"] = "UPDATE_ASSET_SOURCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
