// oci-cloudbridge/lib/model/work-request-resource-metadata-key.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResourceMetadataKey = void 0;
/**
 * Possible metadata keys for work request resource metadata.
 **/
var WorkRequestResourceMetadataKey;
(function (WorkRequestResourceMetadataKey) {
    WorkRequestResourceMetadataKey["IsDryRun"] = "IS_DRY_RUN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestResourceMetadataKey["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestResourceMetadataKey = exports.WorkRequestResourceMetadataKey || (exports.WorkRequestResourceMetadataKey = {}));
(function (WorkRequestResourceMetadataKey) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestResourceMetadataKey.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestResourceMetadataKey.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestResourceMetadataKey = exports.WorkRequestResourceMetadataKey || (exports.WorkRequestResourceMetadataKey = {}));
