// oci-mediaservices/lib/model/media-asset-distribution-channel-attachment-sort-by.js
"use strict";
/**
 * Media Services API
 * Media Services (includes Media Flow and Media Streams) is a fully managed service for processing media (video) source content. Use Media Flow and Media Streams to transcode and package digital video using configurable workflows and stream video outputs.

Use the Media Services API to configure media workflows and run Media Flow jobs, create distribution channels, ingest assets, create Preview URLs and play assets. For more information, see [Media Flow](/iaas/Content/dms-mediaflow/home.htm) and [Media Streams](/iaas/Content/dms-mediastream/home.htm).

 * OpenAPI spec version: 20211101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaAssetDistributionChannelAttachmentSortBy = void 0;
/**
 * The field to sort by.
 **/
var MediaAssetDistributionChannelAttachmentSortBy;
(function (MediaAssetDistributionChannelAttachmentSortBy) {
    MediaAssetDistributionChannelAttachmentSortBy["MediaAssetId"] = "mediaAssetId";
    MediaAssetDistributionChannelAttachmentSortBy["DistributionChannelId"] = "distributionChannelId";
    MediaAssetDistributionChannelAttachmentSortBy["DisplayName"] = "displayName";
    MediaAssetDistributionChannelAttachmentSortBy["Version"] = "version";
})(MediaAssetDistributionChannelAttachmentSortBy = exports.MediaAssetDistributionChannelAttachmentSortBy || (exports.MediaAssetDistributionChannelAttachmentSortBy = {}));
(function (MediaAssetDistributionChannelAttachmentSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    MediaAssetDistributionChannelAttachmentSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MediaAssetDistributionChannelAttachmentSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(MediaAssetDistributionChannelAttachmentSortBy = exports.MediaAssetDistributionChannelAttachmentSortBy || (exports.MediaAssetDistributionChannelAttachmentSortBy = {}));
