// oci-mediaservices/lib/model/media-workflow-job.js
"use strict";
/**
 * Media Services API
 * Media Services (includes Media Flow and Media Streams) is a fully managed service for processing media (video) source content. Use Media Flow and Media Streams to transcode and package digital video using configurable workflows and stream video outputs.

Use the Media Services API to configure media workflows and run Media Flow jobs, create distribution channels, ingest assets, create Preview URLs and play assets. For more information, see [Media Flow](/iaas/Content/dms-mediaflow/home.htm) and [Media Streams](/iaas/Content/dms-mediastream/home.htm).

 * OpenAPI spec version: 20211101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaWorkflowJob = void 0;
const model = __importStar(require("../model"));
var MediaWorkflowJob;
(function (MediaWorkflowJob) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Waiting"] = "WAITING";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Canceling"] = "CANCELING";
        LifecycleState["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = MediaWorkflowJob.LifecycleState || (MediaWorkflowJob.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "taskLifecycleState": obj.taskLifecycleState
                ? obj.taskLifecycleState.map(item => {
                    return model.MediaWorkflowTaskState.getJsonObj(item);
                })
                : undefined,
            "outputs": obj.outputs
                ? obj.outputs.map(item => {
                    return model.JobOutput.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    MediaWorkflowJob.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "taskLifecycleState": obj.taskLifecycleState
                ? obj.taskLifecycleState.map(item => {
                    return model.MediaWorkflowTaskState.getDeserializedJsonObj(item);
                })
                : undefined,
            "outputs": obj.outputs
                ? obj.outputs.map(item => {
                    return model.JobOutput.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    MediaWorkflowJob.getDeserializedJsonObj = getDeserializedJsonObj;
})(MediaWorkflowJob = exports.MediaWorkflowJob || (exports.MediaWorkflowJob = {}));
