// oci-datacatalog/lib/model/event-config-status.js
"use strict";
/**
 * Data Catalog API
 * Use the Data Catalog APIs to collect, organize, find, access, understand, enrich, and activate technical, business, and operational metadata.
For more information, see [Data Catalog](https://docs.oracle.com/iaas/data-catalog/home.htm).

 * OpenAPI spec version: 20190325
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventConfigStatus = void 0;
/**
 * Enum representing whether or not to emit an event when a relevant action (e.g. modifying a custom property value) is performed by the user on this type.
 * ENABLED - Event is emitted.
 * DISABLED - Event is not emitted.
 *
 **/
var EventConfigStatus;
(function (EventConfigStatus) {
    EventConfigStatus["Enabled"] = "ENABLED";
    EventConfigStatus["Disabled"] = "DISABLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    EventConfigStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(EventConfigStatus = exports.EventConfigStatus || (exports.EventConfigStatus = {}));
(function (EventConfigStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    EventConfigStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    EventConfigStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(EventConfigStatus = exports.EventConfigStatus || (exports.EventConfigStatus = {}));
