// oci-cloudbridge/lib/request/index.js
"use strict";
/**
 * Oracle Cloud Bridge API
 * API for Oracle Cloud Bridge service.
 * OpenAPI spec version: 20220509
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListEnvironmentsRequest = exports.ListApplianceImagesRequest = exports.ListAgentsRequest = exports.ListAgentDependenciesRequest = exports.ListInventoriesRequest = exports.ListHistoricalMetricsRequest = exports.ListAssetsRequest = exports.ListDiscoverySchedulesRequest = exports.ListAssetSourcesRequest = exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = void 0;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
const ListAssetSourcesRequest = __importStar(require("./list-asset-sources-request"));
exports.ListAssetSourcesRequest = ListAssetSourcesRequest.ListAssetSourcesRequest;
const ListDiscoverySchedulesRequest = __importStar(require("./list-discovery-schedules-request"));
exports.ListDiscoverySchedulesRequest = ListDiscoverySchedulesRequest.ListDiscoverySchedulesRequest;
const ListAssetsRequest = __importStar(require("./list-assets-request"));
exports.ListAssetsRequest = ListAssetsRequest.ListAssetsRequest;
const ListHistoricalMetricsRequest = __importStar(require("./list-historical-metrics-request"));
exports.ListHistoricalMetricsRequest = ListHistoricalMetricsRequest.ListHistoricalMetricsRequest;
const ListInventoriesRequest = __importStar(require("./list-inventories-request"));
exports.ListInventoriesRequest = ListInventoriesRequest.ListInventoriesRequest;
const ListAgentDependenciesRequest = __importStar(require("./list-agent-dependencies-request"));
exports.ListAgentDependenciesRequest = ListAgentDependenciesRequest.ListAgentDependenciesRequest;
const ListAgentsRequest = __importStar(require("./list-agents-request"));
exports.ListAgentsRequest = ListAgentsRequest.ListAgentsRequest;
const ListApplianceImagesRequest = __importStar(require("./list-appliance-images-request"));
exports.ListApplianceImagesRequest = ListApplianceImagesRequest.ListApplianceImagesRequest;
const ListEnvironmentsRequest = __importStar(require("./list-environments-request"));
exports.ListEnvironmentsRequest = ListEnvironmentsRequest.ListEnvironmentsRequest;
