// oci-analytics/lib/model/access-control-service-type.js
"use strict";
/**
 * Analytics API
 * Analytics API.

 * OpenAPI spec version: 20190331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessControlServiceType = void 0;
/**
 * Various types of Oracle Service categories that can be whitelisted to access an Analytics Instance.
 *
 **/
var AccessControlServiceType;
(function (AccessControlServiceType) {
    AccessControlServiceType["All"] = "ALL";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AccessControlServiceType["UnknownValue"] = "UNKNOWN_VALUE";
})(AccessControlServiceType = exports.AccessControlServiceType || (exports.AccessControlServiceType = {}));
(function (AccessControlServiceType) {
    function getJsonObj(obj) {
        return obj;
    }
    AccessControlServiceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AccessControlServiceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AccessControlServiceType = exports.AccessControlServiceType || (exports.AccessControlServiceType = {}));
