// oci-goldengate/lib/request/index.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTrailSequencesRequest = exports.ListTrailFilesRequest = exports.ListDeploymentsRequest = exports.ListDeploymentWalletsOperationsRequest = exports.ListDeploymentVersionsRequest = exports.ListDeploymentUpgradesRequest = exports.ListDeploymentTypesRequest = exports.ListDeploymentBackupsRequest = exports.ListDatabaseRegistrationsRequest = exports.ListConnectionsRequest = exports.ListConnectionAssignmentsRequest = void 0;
const ListConnectionAssignmentsRequest = __importStar(require("./list-connection-assignments-request"));
exports.ListConnectionAssignmentsRequest = ListConnectionAssignmentsRequest.ListConnectionAssignmentsRequest;
const ListConnectionsRequest = __importStar(require("./list-connections-request"));
exports.ListConnectionsRequest = ListConnectionsRequest.ListConnectionsRequest;
const ListDatabaseRegistrationsRequest = __importStar(require("./list-database-registrations-request"));
exports.ListDatabaseRegistrationsRequest = ListDatabaseRegistrationsRequest.ListDatabaseRegistrationsRequest;
const ListDeploymentBackupsRequest = __importStar(require("./list-deployment-backups-request"));
exports.ListDeploymentBackupsRequest = ListDeploymentBackupsRequest.ListDeploymentBackupsRequest;
const ListDeploymentTypesRequest = __importStar(require("./list-deployment-types-request"));
exports.ListDeploymentTypesRequest = ListDeploymentTypesRequest.ListDeploymentTypesRequest;
const ListDeploymentUpgradesRequest = __importStar(require("./list-deployment-upgrades-request"));
exports.ListDeploymentUpgradesRequest = ListDeploymentUpgradesRequest.ListDeploymentUpgradesRequest;
const ListDeploymentVersionsRequest = __importStar(require("./list-deployment-versions-request"));
exports.ListDeploymentVersionsRequest = ListDeploymentVersionsRequest.ListDeploymentVersionsRequest;
const ListDeploymentWalletsOperationsRequest = __importStar(require("./list-deployment-wallets-operations-request"));
exports.ListDeploymentWalletsOperationsRequest = ListDeploymentWalletsOperationsRequest.ListDeploymentWalletsOperationsRequest;
const ListDeploymentsRequest = __importStar(require("./list-deployments-request"));
exports.ListDeploymentsRequest = ListDeploymentsRequest.ListDeploymentsRequest;
const ListTrailFilesRequest = __importStar(require("./list-trail-files-request"));
exports.ListTrailFilesRequest = ListTrailFilesRequest.ListTrailFilesRequest;
const ListTrailSequencesRequest = __importStar(require("./list-trail-sequences-request"));
exports.ListTrailSequencesRequest = ListTrailSequencesRequest.ListTrailSequencesRequest;
