// oci-servicemesh/lib/servicemesh-waiter.js
"use strict";
/**
 * Service Mesh API
 * Use the Service Mesh API to manage mesh, virtual service, access policy and other mesh related items.
 * OpenAPI spec version: 20220615
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceMeshWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
class ServiceMeshWaiter {
    constructor(client, config) {
        this.client = client;
        this.config = config;
    }
    /**
     * Waits forAccessPolicy till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetAccessPolicyResponse | null (null in case of 404 response)
     */
    forAccessPolicy(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getAccessPolicy(request), response => targetStates.includes(response.accessPolicy.lifecycleState), targetStates.includes(models.AccessPolicy.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forIngressGateway till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetIngressGatewayResponse | null (null in case of 404 response)
     */
    forIngressGateway(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getIngressGateway(request), response => targetStates.includes(response.ingressGateway.lifecycleState), targetStates.includes(models.IngressGateway.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forIngressGatewayRouteTable till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetIngressGatewayRouteTableResponse | null (null in case of 404 response)
     */
    forIngressGatewayRouteTable(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getIngressGatewayRouteTable(request), response => targetStates.includes(response.ingressGatewayRouteTable.lifecycleState), targetStates.includes(models.IngressGatewayRouteTable.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forMesh till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMeshResponse | null (null in case of 404 response)
     */
    forMesh(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getMesh(request), response => targetStates.includes(response.mesh.lifecycleState), targetStates.includes(models.Mesh.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forVirtualDeployment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVirtualDeploymentResponse | null (null in case of 404 response)
     */
    forVirtualDeployment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getVirtualDeployment(request), response => targetStates.includes(response.virtualDeployment.lifecycleState), targetStates.includes(models.VirtualDeployment.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forVirtualService till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVirtualServiceResponse | null (null in case of 404 response)
     */
    forVirtualService(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getVirtualService(request), response => targetStates.includes(response.virtualService.lifecycleState), targetStates.includes(models.VirtualService.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forVirtualServiceRouteTable till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetVirtualServiceRouteTableResponse | null (null in case of 404 response)
     */
    forVirtualServiceRouteTable(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getVirtualServiceRouteTable(request), response => targetStates.includes(response.virtualServiceRouteTable.lifecycleState), targetStates.includes(models.VirtualServiceRouteTable.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forWorkRequest
     *
     * @param request the request to send
     * @return response returns GetWorkRequestResponse
     */
    forWorkRequest(request) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getWorkRequest(request), response => (response.workRequest.timeFinished ? true : false));
        });
    }
}
exports.ServiceMeshWaiter = ServiceMeshWaiter;
