// oci-containerinstances/lib/model/container.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Container = void 0;
const model = __importStar(require("../model"));
var Container;
(function (Container) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Container.LifecycleState || (Container.LifecycleState = {}));
    let AdditionalCapabilities;
    (function (AdditionalCapabilities) {
        AdditionalCapabilities["CapNetAdmin"] = "CAP_NET_ADMIN";
        AdditionalCapabilities["CapNetRaw"] = "CAP_NET_RAW";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AdditionalCapabilities["UnknownValue"] = "UNKNOWN_VALUE";
    })(AdditionalCapabilities = Container.AdditionalCapabilities || (Container.AdditionalCapabilities = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "volumeMounts": obj.volumeMounts
                ? obj.volumeMounts.map(item => {
                    return model.VolumeMount.getJsonObj(item);
                })
                : undefined,
            "healthChecks": obj.healthChecks
                ? obj.healthChecks.map(item => {
                    return model.ContainerHealthCheck.getJsonObj(item);
                })
                : undefined,
            "resourceConfig": obj.resourceConfig
                ? model.ContainerResourceConfig.getJsonObj(obj.resourceConfig)
                : undefined
        });
        return jsonObj;
    }
    Container.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "volumeMounts": obj.volumeMounts
                ? obj.volumeMounts.map(item => {
                    return model.VolumeMount.getDeserializedJsonObj(item);
                })
                : undefined,
            "healthChecks": obj.healthChecks
                ? obj.healthChecks.map(item => {
                    return model.ContainerHealthCheck.getDeserializedJsonObj(item);
                })
                : undefined,
            "resourceConfig": obj.resourceConfig
                ? model.ContainerResourceConfig.getDeserializedJsonObj(obj.resourceConfig)
                : undefined
        });
        return jsonObj;
    }
    Container.getDeserializedJsonObj = getDeserializedJsonObj;
})(Container = exports.Container || (exports.Container = {}));
