// oci-containerinstances/lib/model/container-health-check-type.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerHealthCheckType = void 0;
/**
 * The container health check type.
 **/
var ContainerHealthCheckType;
(function (ContainerHealthCheckType) {
    ContainerHealthCheckType["Http"] = "HTTP";
    ContainerHealthCheckType["Tcp"] = "TCP";
    ContainerHealthCheckType["Command"] = "COMMAND";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ContainerHealthCheckType["UnknownValue"] = "UNKNOWN_VALUE";
})(ContainerHealthCheckType = exports.ContainerHealthCheckType || (exports.ContainerHealthCheckType = {}));
(function (ContainerHealthCheckType) {
    function getJsonObj(obj) {
        return obj;
    }
    ContainerHealthCheckType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ContainerHealthCheckType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerHealthCheckType = exports.ContainerHealthCheckType || (exports.ContainerHealthCheckType = {}));
