// oci-containerinstances/lib/model/container-capability.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerCapability = void 0;
/**
 * additional configurable container capabilities
 **/
var ContainerCapability;
(function (ContainerCapability) {
    ContainerCapability["CapNetAdmin"] = "CAP_NET_ADMIN";
    ContainerCapability["CapNetRaw"] = "CAP_NET_RAW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ContainerCapability["UnknownValue"] = "UNKNOWN_VALUE";
})(ContainerCapability = exports.ContainerCapability || (exports.ContainerCapability = {}));
(function (ContainerCapability) {
    function getJsonObj(obj) {
        return obj;
    }
    ContainerCapability.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ContainerCapability.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerCapability = exports.ContainerCapability || (exports.ContainerCapability = {}));
