// oci-containerinstances/lib/model/container-instance.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerInstance = void 0;
const model = __importStar(require("../model"));
var ContainerInstance;
(function (ContainerInstance) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ContainerInstance.LifecycleState || (ContainerInstance.LifecycleState = {}));
    let ContainerRestartPolicy;
    (function (ContainerRestartPolicy) {
        ContainerRestartPolicy["Always"] = "ALWAYS";
        ContainerRestartPolicy["Never"] = "NEVER";
        ContainerRestartPolicy["OnFailure"] = "ON_FAILURE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ContainerRestartPolicy["UnknownValue"] = "UNKNOWN_VALUE";
    })(ContainerRestartPolicy = ContainerInstance.ContainerRestartPolicy || (ContainerInstance.ContainerRestartPolicy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "volumes": obj.volumes
                ? obj.volumes.map(item => {
                    return model.ContainerVolume.getJsonObj(item);
                })
                : undefined,
            "containers": obj.containers
                ? obj.containers.map(item => {
                    return model.ContainerInstanceContainer.getJsonObj(item);
                })
                : undefined,
            "shapeConfig": obj.shapeConfig
                ? model.ContainerInstanceShapeConfig.getJsonObj(obj.shapeConfig)
                : undefined,
            "vnics": obj.vnics
                ? obj.vnics.map(item => {
                    return model.ContainerVnic.getJsonObj(item);
                })
                : undefined,
            "dnsConfig": obj.dnsConfig ? model.ContainerDnsConfig.getJsonObj(obj.dnsConfig) : undefined,
            "imagePullSecrets": obj.imagePullSecrets
                ? obj.imagePullSecrets.map(item => {
                    return model.ImagePullSecret.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ContainerInstance.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "volumes": obj.volumes
                ? obj.volumes.map(item => {
                    return model.ContainerVolume.getDeserializedJsonObj(item);
                })
                : undefined,
            "containers": obj.containers
                ? obj.containers.map(item => {
                    return model.ContainerInstanceContainer.getDeserializedJsonObj(item);
                })
                : undefined,
            "shapeConfig": obj.shapeConfig
                ? model.ContainerInstanceShapeConfig.getDeserializedJsonObj(obj.shapeConfig)
                : undefined,
            "vnics": obj.vnics
                ? obj.vnics.map(item => {
                    return model.ContainerVnic.getDeserializedJsonObj(item);
                })
                : undefined,
            "dnsConfig": obj.dnsConfig
                ? model.ContainerDnsConfig.getDeserializedJsonObj(obj.dnsConfig)
                : undefined,
            "imagePullSecrets": obj.imagePullSecrets
                ? obj.imagePullSecrets.map(item => {
                    return model.ImagePullSecret.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ContainerInstance.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerInstance = exports.ContainerInstance || (exports.ContainerInstance = {}));
