// oci-containerinstances/lib/model/container-health-check-failure-action.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerHealthCheckFailureAction = void 0;
/**
 * The failure action will be taken on behalf of container instance status.
 **/
var ContainerHealthCheckFailureAction;
(function (ContainerHealthCheckFailureAction) {
    ContainerHealthCheckFailureAction["Kill"] = "KILL";
    ContainerHealthCheckFailureAction["None"] = "NONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ContainerHealthCheckFailureAction["UnknownValue"] = "UNKNOWN_VALUE";
})(ContainerHealthCheckFailureAction = exports.ContainerHealthCheckFailureAction || (exports.ContainerHealthCheckFailureAction = {}));
(function (ContainerHealthCheckFailureAction) {
    function getJsonObj(obj) {
        return obj;
    }
    ContainerHealthCheckFailureAction.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ContainerHealthCheckFailureAction.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerHealthCheckFailureAction = exports.ContainerHealthCheckFailureAction || (exports.ContainerHealthCheckFailureAction = {}));
