// oci-containerinstances/lib/model/container-health-check.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerHealthCheck = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ContainerHealthCheck;
(function (ContainerHealthCheck) {
    let Status;
    (function (Status) {
        Status["Healthy"] = "HEALTHY";
        Status["Unhealthy"] = "UNHEALTHY";
        Status["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = ContainerHealthCheck.Status || (ContainerHealthCheck.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "healthCheckType" in obj && obj.healthCheckType) {
            switch (obj.healthCheckType) {
                case "TCP":
                    return model.ContainerTcpHealthCheck.getJsonObj(jsonObj, true);
                case "HTTP":
                    return model.ContainerHttpHealthCheck.getJsonObj(jsonObj, true);
                case "COMMAND":
                    return model.ContainerCommandHealthCheck.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.healthCheckType}`);
            }
        }
        return jsonObj;
    }
    ContainerHealthCheck.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "healthCheckType" in obj && obj.healthCheckType) {
            switch (obj.healthCheckType) {
                case "TCP":
                    return model.ContainerTcpHealthCheck.getDeserializedJsonObj(jsonObj, true);
                case "HTTP":
                    return model.ContainerHttpHealthCheck.getDeserializedJsonObj(jsonObj, true);
                case "COMMAND":
                    return model.ContainerCommandHealthCheck.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.healthCheckType}`);
            }
        }
        return jsonObj;
    }
    ContainerHealthCheck.getDeserializedJsonObj = getDeserializedJsonObj;
})(ContainerHealthCheck = exports.ContainerHealthCheck || (exports.ContainerHealthCheck = {}));
