// oci-containerinstances/lib/model/index.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VolumeMount = exports.UpdateContainerInstanceDetails = exports.UpdateContainerDetails = exports.SortOrder = exports.ShapeOcpuOptions = exports.ShapeNetworkingBandwidthOptions = exports.ShapeMemoryOptions = exports.OperationType = exports.OperationStatus = exports.ImagePullSecret = exports.HealthCheckHttpHeader = exports.CreateVolumeMountDetails = exports.CreateImagePullSecretDetails = exports.CreateContainerVolumeDetails = exports.CreateContainerVnicDetails = exports.CreateContainerResourceConfigDetails = exports.CreateContainerInstanceShapeConfigDetails = exports.CreateContainerInstanceDetails = exports.CreateContainerHealthCheckDetails = exports.CreateContainerDnsConfigDetails = exports.CreateContainerDetails = exports.ContainerVolumeType = exports.ContainerVolume = exports.ContainerVnic = exports.ContainerSummary = exports.ContainerResourceConfig = exports.ContainerInstanceSummary = exports.ContainerInstanceShapeSummary = exports.ContainerInstanceShapeConfig = exports.ContainerInstanceShapeCollection = exports.ContainerInstanceContainer = exports.ContainerInstanceCollection = exports.ContainerInstance = exports.ContainerHealthCheckType = exports.ContainerHealthCheckFailureAction = exports.ContainerHealthCheck = exports.ContainerDnsConfig = exports.ContainerConfigFile = exports.ContainerCollection = exports.ContainerCapability = exports.Container = exports.ChangeContainerInstanceCompartmentDetails = exports.ActionType = void 0;
exports.VaultImagePullSecret = exports.CreateVaultImagePullSecretDetails = exports.CreateContainerTcpHealthCheckDetails = exports.CreateContainerHttpHealthCheckDetails = exports.CreateContainerEmptyDirVolumeDetails = exports.CreateContainerConfigFileVolumeDetails = exports.CreateContainerCommandHealthCheckDetails = exports.CreateBasicImagePullSecretDetails = exports.ContainerTcpHealthCheck = exports.ContainerHttpHealthCheck = exports.ContainerEmptyDirVolume = exports.ContainerConfigFileVolume = exports.ContainerCommandHealthCheck = exports.BasicImagePullSecret = exports.WorkRequestSummaryCollection = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeContainerInstanceCompartmentDetails = __importStar(require("./change-container-instance-compartment-details"));
exports.ChangeContainerInstanceCompartmentDetails = ChangeContainerInstanceCompartmentDetails.ChangeContainerInstanceCompartmentDetails;
const Container = __importStar(require("./container"));
exports.Container = Container.Container;
const ContainerCapability = __importStar(require("./container-capability"));
exports.ContainerCapability = ContainerCapability.ContainerCapability;
const ContainerCollection = __importStar(require("./container-collection"));
exports.ContainerCollection = ContainerCollection.ContainerCollection;
const ContainerConfigFile = __importStar(require("./container-config-file"));
exports.ContainerConfigFile = ContainerConfigFile.ContainerConfigFile;
const ContainerDnsConfig = __importStar(require("./container-dns-config"));
exports.ContainerDnsConfig = ContainerDnsConfig.ContainerDnsConfig;
const ContainerHealthCheck = __importStar(require("./container-health-check"));
exports.ContainerHealthCheck = ContainerHealthCheck.ContainerHealthCheck;
const ContainerHealthCheckFailureAction = __importStar(require("./container-health-check-failure-action"));
exports.ContainerHealthCheckFailureAction = ContainerHealthCheckFailureAction.ContainerHealthCheckFailureAction;
const ContainerHealthCheckType = __importStar(require("./container-health-check-type"));
exports.ContainerHealthCheckType = ContainerHealthCheckType.ContainerHealthCheckType;
const ContainerInstance = __importStar(require("./container-instance"));
exports.ContainerInstance = ContainerInstance.ContainerInstance;
const ContainerInstanceCollection = __importStar(require("./container-instance-collection"));
exports.ContainerInstanceCollection = ContainerInstanceCollection.ContainerInstanceCollection;
const ContainerInstanceContainer = __importStar(require("./container-instance-container"));
exports.ContainerInstanceContainer = ContainerInstanceContainer.ContainerInstanceContainer;
const ContainerInstanceShapeCollection = __importStar(require("./container-instance-shape-collection"));
exports.ContainerInstanceShapeCollection = ContainerInstanceShapeCollection.ContainerInstanceShapeCollection;
const ContainerInstanceShapeConfig = __importStar(require("./container-instance-shape-config"));
exports.ContainerInstanceShapeConfig = ContainerInstanceShapeConfig.ContainerInstanceShapeConfig;
const ContainerInstanceShapeSummary = __importStar(require("./container-instance-shape-summary"));
exports.ContainerInstanceShapeSummary = ContainerInstanceShapeSummary.ContainerInstanceShapeSummary;
const ContainerInstanceSummary = __importStar(require("./container-instance-summary"));
exports.ContainerInstanceSummary = ContainerInstanceSummary.ContainerInstanceSummary;
const ContainerResourceConfig = __importStar(require("./container-resource-config"));
exports.ContainerResourceConfig = ContainerResourceConfig.ContainerResourceConfig;
const ContainerSummary = __importStar(require("./container-summary"));
exports.ContainerSummary = ContainerSummary.ContainerSummary;
const ContainerVnic = __importStar(require("./container-vnic"));
exports.ContainerVnic = ContainerVnic.ContainerVnic;
const ContainerVolume = __importStar(require("./container-volume"));
exports.ContainerVolume = ContainerVolume.ContainerVolume;
const ContainerVolumeType = __importStar(require("./container-volume-type"));
exports.ContainerVolumeType = ContainerVolumeType.ContainerVolumeType;
const CreateContainerDetails = __importStar(require("./create-container-details"));
exports.CreateContainerDetails = CreateContainerDetails.CreateContainerDetails;
const CreateContainerDnsConfigDetails = __importStar(require("./create-container-dns-config-details"));
exports.CreateContainerDnsConfigDetails = CreateContainerDnsConfigDetails.CreateContainerDnsConfigDetails;
const CreateContainerHealthCheckDetails = __importStar(require("./create-container-health-check-details"));
exports.CreateContainerHealthCheckDetails = CreateContainerHealthCheckDetails.CreateContainerHealthCheckDetails;
const CreateContainerInstanceDetails = __importStar(require("./create-container-instance-details"));
exports.CreateContainerInstanceDetails = CreateContainerInstanceDetails.CreateContainerInstanceDetails;
const CreateContainerInstanceShapeConfigDetails = __importStar(require("./create-container-instance-shape-config-details"));
exports.CreateContainerInstanceShapeConfigDetails = CreateContainerInstanceShapeConfigDetails.CreateContainerInstanceShapeConfigDetails;
const CreateContainerResourceConfigDetails = __importStar(require("./create-container-resource-config-details"));
exports.CreateContainerResourceConfigDetails = CreateContainerResourceConfigDetails.CreateContainerResourceConfigDetails;
const CreateContainerVnicDetails = __importStar(require("./create-container-vnic-details"));
exports.CreateContainerVnicDetails = CreateContainerVnicDetails.CreateContainerVnicDetails;
const CreateContainerVolumeDetails = __importStar(require("./create-container-volume-details"));
exports.CreateContainerVolumeDetails = CreateContainerVolumeDetails.CreateContainerVolumeDetails;
const CreateImagePullSecretDetails = __importStar(require("./create-image-pull-secret-details"));
exports.CreateImagePullSecretDetails = CreateImagePullSecretDetails.CreateImagePullSecretDetails;
const CreateVolumeMountDetails = __importStar(require("./create-volume-mount-details"));
exports.CreateVolumeMountDetails = CreateVolumeMountDetails.CreateVolumeMountDetails;
const HealthCheckHttpHeader = __importStar(require("./health-check-http-header"));
exports.HealthCheckHttpHeader = HealthCheckHttpHeader.HealthCheckHttpHeader;
const ImagePullSecret = __importStar(require("./image-pull-secret"));
exports.ImagePullSecret = ImagePullSecret.ImagePullSecret;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ShapeMemoryOptions = __importStar(require("./shape-memory-options"));
exports.ShapeMemoryOptions = ShapeMemoryOptions.ShapeMemoryOptions;
const ShapeNetworkingBandwidthOptions = __importStar(require("./shape-networking-bandwidth-options"));
exports.ShapeNetworkingBandwidthOptions = ShapeNetworkingBandwidthOptions.ShapeNetworkingBandwidthOptions;
const ShapeOcpuOptions = __importStar(require("./shape-ocpu-options"));
exports.ShapeOcpuOptions = ShapeOcpuOptions.ShapeOcpuOptions;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateContainerDetails = __importStar(require("./update-container-details"));
exports.UpdateContainerDetails = UpdateContainerDetails.UpdateContainerDetails;
const UpdateContainerInstanceDetails = __importStar(require("./update-container-instance-details"));
exports.UpdateContainerInstanceDetails = UpdateContainerInstanceDetails.UpdateContainerInstanceDetails;
const VolumeMount = __importStar(require("./volume-mount"));
exports.VolumeMount = VolumeMount.VolumeMount;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const BasicImagePullSecret = __importStar(require("./basic-image-pull-secret"));
exports.BasicImagePullSecret = BasicImagePullSecret.BasicImagePullSecret;
const ContainerCommandHealthCheck = __importStar(require("./container-command-health-check"));
exports.ContainerCommandHealthCheck = ContainerCommandHealthCheck.ContainerCommandHealthCheck;
const ContainerConfigFileVolume = __importStar(require("./container-config-file-volume"));
exports.ContainerConfigFileVolume = ContainerConfigFileVolume.ContainerConfigFileVolume;
const ContainerEmptyDirVolume = __importStar(require("./container-empty-dir-volume"));
exports.ContainerEmptyDirVolume = ContainerEmptyDirVolume.ContainerEmptyDirVolume;
const ContainerHttpHealthCheck = __importStar(require("./container-http-health-check"));
exports.ContainerHttpHealthCheck = ContainerHttpHealthCheck.ContainerHttpHealthCheck;
const ContainerTcpHealthCheck = __importStar(require("./container-tcp-health-check"));
exports.ContainerTcpHealthCheck = ContainerTcpHealthCheck.ContainerTcpHealthCheck;
const CreateBasicImagePullSecretDetails = __importStar(require("./create-basic-image-pull-secret-details"));
exports.CreateBasicImagePullSecretDetails = CreateBasicImagePullSecretDetails.CreateBasicImagePullSecretDetails;
const CreateContainerCommandHealthCheckDetails = __importStar(require("./create-container-command-health-check-details"));
exports.CreateContainerCommandHealthCheckDetails = CreateContainerCommandHealthCheckDetails.CreateContainerCommandHealthCheckDetails;
const CreateContainerConfigFileVolumeDetails = __importStar(require("./create-container-config-file-volume-details"));
exports.CreateContainerConfigFileVolumeDetails = CreateContainerConfigFileVolumeDetails.CreateContainerConfigFileVolumeDetails;
const CreateContainerEmptyDirVolumeDetails = __importStar(require("./create-container-empty-dir-volume-details"));
exports.CreateContainerEmptyDirVolumeDetails = CreateContainerEmptyDirVolumeDetails.CreateContainerEmptyDirVolumeDetails;
const CreateContainerHttpHealthCheckDetails = __importStar(require("./create-container-http-health-check-details"));
exports.CreateContainerHttpHealthCheckDetails = CreateContainerHttpHealthCheckDetails.CreateContainerHttpHealthCheckDetails;
const CreateContainerTcpHealthCheckDetails = __importStar(require("./create-container-tcp-health-check-details"));
exports.CreateContainerTcpHealthCheckDetails = CreateContainerTcpHealthCheckDetails.CreateContainerTcpHealthCheckDetails;
const CreateVaultImagePullSecretDetails = __importStar(require("./create-vault-image-pull-secret-details"));
exports.CreateVaultImagePullSecretDetails = CreateVaultImagePullSecretDetails.CreateVaultImagePullSecretDetails;
const VaultImagePullSecret = __importStar(require("./vault-image-pull-secret"));
exports.VaultImagePullSecret = VaultImagePullSecret.VaultImagePullSecret;
