// oci-containerinstances/lib/model/operation-type.js
"use strict";
/**
 * Container Instance API
 * A description of the Container Instance API
 * OpenAPI spec version: 20210415
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateContainerInstance"] = "CREATE_CONTAINER_INSTANCE";
    OperationType["UpdateContainerInstance"] = "UPDATE_CONTAINER_INSTANCE";
    OperationType["DeleteContainerInstance"] = "DELETE_CONTAINER_INSTANCE";
    OperationType["MoveContainerInstance"] = "MOVE_CONTAINER_INSTANCE";
    OperationType["StartContainerInstance"] = "START_CONTAINER_INSTANCE";
    OperationType["StopContainerInstance"] = "STOP_CONTAINER_INSTANCE";
    OperationType["RestartContainerInstance"] = "RESTART_CONTAINER_INSTANCE";
    OperationType["UpdateContainer"] = "UPDATE_CONTAINER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
