// oci-governancerulescontrolplane/lib/model/creation-option.js
"use strict";
/**
 * GovernanceRulesControlPlane API
 * A description of the GovernanceRulesControlPlane API
 * OpenAPI spec version: 20220504
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreationOption = void 0;
/**
 * The type of option used to create the governance rule, could be one of TEMPLATE or CLONE.
 **/
var CreationOption;
(function (CreationOption) {
    CreationOption["Template"] = "TEMPLATE";
    CreationOption["Clone"] = "CLONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CreationOption["UnknownValue"] = "UNKNOWN_VALUE";
})(CreationOption = exports.CreationOption || (exports.CreationOption = {}));
(function (CreationOption) {
    function getJsonObj(obj) {
        return obj;
    }
    CreationOption.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CreationOption.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreationOption = exports.CreationOption || (exports.CreationOption = {}));
