// oci-governancerulescontrolplane/lib/model/governance-rule-lifecycle-state.js
"use strict";
/**
 * GovernanceRulesControlPlane API
 * A description of the GovernanceRulesControlPlane API
 * OpenAPI spec version: 20220504
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceRuleLifecycleState = void 0;
/**
 * The current state of governance rule.
 **/
var GovernanceRuleLifecycleState;
(function (GovernanceRuleLifecycleState) {
    GovernanceRuleLifecycleState["Active"] = "ACTIVE";
    GovernanceRuleLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    GovernanceRuleLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(GovernanceRuleLifecycleState = exports.GovernanceRuleLifecycleState || (exports.GovernanceRuleLifecycleState = {}));
(function (GovernanceRuleLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    GovernanceRuleLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    GovernanceRuleLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(GovernanceRuleLifecycleState = exports.GovernanceRuleLifecycleState || (exports.GovernanceRuleLifecycleState = {}));
