// oci-governancerulescontrolplane/lib/model/index.js
"use strict";
/**
 * GovernanceRulesControlPlane API
 * A description of the GovernanceRulesControlPlane API
 * OpenAPI spec version: 20220504
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenancyAssociation = exports.TagTemplate = exports.QuotaTemplate = exports.EnumTagDefinitionValidator = exports.DefaultTagDefinitionValidator = exports.AllowedRegionsTemplate = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateGovernanceRuleDetails = exports.TenancyAttachmentSummary = exports.TenancyAttachmentCollection = exports.TenancyAttachment = exports.Template = exports.TagDefault = exports.Tag = exports.SortOrder = exports.OperationType = exports.OperationStatus = exports.InclusionCriterionSummary = exports.InclusionCriterionCollection = exports.InclusionCriterion = exports.GovernanceRuleType = exports.GovernanceRuleSummary = exports.GovernanceRuleLifecycleState = exports.GovernanceRuleCollection = exports.GovernanceRule = exports.EnforcedGovernanceRuleSummary = exports.EnforcedGovernanceRuleCollection = exports.EnforcedGovernanceRule = exports.CreationOption = exports.CreateInclusionCriterionDetails = exports.CreateGovernanceRuleDetails = exports.BaseTagDefinitionValidator = exports.Association = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Association = __importStar(require("./association"));
exports.Association = Association.Association;
const BaseTagDefinitionValidator = __importStar(require("./base-tag-definition-validator"));
exports.BaseTagDefinitionValidator = BaseTagDefinitionValidator.BaseTagDefinitionValidator;
const CreateGovernanceRuleDetails = __importStar(require("./create-governance-rule-details"));
exports.CreateGovernanceRuleDetails = CreateGovernanceRuleDetails.CreateGovernanceRuleDetails;
const CreateInclusionCriterionDetails = __importStar(require("./create-inclusion-criterion-details"));
exports.CreateInclusionCriterionDetails = CreateInclusionCriterionDetails.CreateInclusionCriterionDetails;
const CreationOption = __importStar(require("./creation-option"));
exports.CreationOption = CreationOption.CreationOption;
const EnforcedGovernanceRule = __importStar(require("./enforced-governance-rule"));
exports.EnforcedGovernanceRule = EnforcedGovernanceRule.EnforcedGovernanceRule;
const EnforcedGovernanceRuleCollection = __importStar(require("./enforced-governance-rule-collection"));
exports.EnforcedGovernanceRuleCollection = EnforcedGovernanceRuleCollection.EnforcedGovernanceRuleCollection;
const EnforcedGovernanceRuleSummary = __importStar(require("./enforced-governance-rule-summary"));
exports.EnforcedGovernanceRuleSummary = EnforcedGovernanceRuleSummary.EnforcedGovernanceRuleSummary;
const GovernanceRule = __importStar(require("./governance-rule"));
exports.GovernanceRule = GovernanceRule.GovernanceRule;
const GovernanceRuleCollection = __importStar(require("./governance-rule-collection"));
exports.GovernanceRuleCollection = GovernanceRuleCollection.GovernanceRuleCollection;
const GovernanceRuleLifecycleState = __importStar(require("./governance-rule-lifecycle-state"));
exports.GovernanceRuleLifecycleState = GovernanceRuleLifecycleState.GovernanceRuleLifecycleState;
const GovernanceRuleSummary = __importStar(require("./governance-rule-summary"));
exports.GovernanceRuleSummary = GovernanceRuleSummary.GovernanceRuleSummary;
const GovernanceRuleType = __importStar(require("./governance-rule-type"));
exports.GovernanceRuleType = GovernanceRuleType.GovernanceRuleType;
const InclusionCriterion = __importStar(require("./inclusion-criterion"));
exports.InclusionCriterion = InclusionCriterion.InclusionCriterion;
const InclusionCriterionCollection = __importStar(require("./inclusion-criterion-collection"));
exports.InclusionCriterionCollection = InclusionCriterionCollection.InclusionCriterionCollection;
const InclusionCriterionSummary = __importStar(require("./inclusion-criterion-summary"));
exports.InclusionCriterionSummary = InclusionCriterionSummary.InclusionCriterionSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Tag = __importStar(require("./tag"));
exports.Tag = Tag.Tag;
const TagDefault = __importStar(require("./tag-default"));
exports.TagDefault = TagDefault.TagDefault;
const Template = __importStar(require("./template"));
exports.Template = Template.Template;
const TenancyAttachment = __importStar(require("./tenancy-attachment"));
exports.TenancyAttachment = TenancyAttachment.TenancyAttachment;
const TenancyAttachmentCollection = __importStar(require("./tenancy-attachment-collection"));
exports.TenancyAttachmentCollection = TenancyAttachmentCollection.TenancyAttachmentCollection;
const TenancyAttachmentSummary = __importStar(require("./tenancy-attachment-summary"));
exports.TenancyAttachmentSummary = TenancyAttachmentSummary.TenancyAttachmentSummary;
const UpdateGovernanceRuleDetails = __importStar(require("./update-governance-rule-details"));
exports.UpdateGovernanceRuleDetails = UpdateGovernanceRuleDetails.UpdateGovernanceRuleDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const AllowedRegionsTemplate = __importStar(require("./allowed-regions-template"));
exports.AllowedRegionsTemplate = AllowedRegionsTemplate.AllowedRegionsTemplate;
const DefaultTagDefinitionValidator = __importStar(require("./default-tag-definition-validator"));
exports.DefaultTagDefinitionValidator = DefaultTagDefinitionValidator.DefaultTagDefinitionValidator;
const EnumTagDefinitionValidator = __importStar(require("./enum-tag-definition-validator"));
exports.EnumTagDefinitionValidator = EnumTagDefinitionValidator.EnumTagDefinitionValidator;
const QuotaTemplate = __importStar(require("./quota-template"));
exports.QuotaTemplate = QuotaTemplate.QuotaTemplate;
const TagTemplate = __importStar(require("./tag-template"));
exports.TagTemplate = TagTemplate.TagTemplate;
const TenancyAssociation = __importStar(require("./tenancy-association"));
exports.TenancyAssociation = TenancyAssociation.TenancyAssociation;
