// oci-governancerulescontrolplane/lib/model/operation-type.js
"use strict";
/**
 * GovernanceRulesControlPlane API
 * A description of the GovernanceRulesControlPlane API
 * OpenAPI spec version: 20220504
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateGovernanceRule"] = "CREATE_GOVERNANCE_RULE";
    OperationType["UpdateGovernanceRule"] = "UPDATE_GOVERNANCE_RULE";
    OperationType["DeleteGovernanceRule"] = "DELETE_GOVERNANCE_RULE";
    OperationType["RetryGovernanceRule"] = "RETRY_GOVERNANCE_RULE";
    OperationType["CreateInclusionCriteria"] = "CREATE_INCLUSION_CRITERIA";
    OperationType["DeleteInclusionCriteria"] = "DELETE_INCLUSION_CRITERIA";
    OperationType["RetryTenancyAttachment"] = "RETRY_TENANCY_ATTACHMENT";
    OperationType["ApplyTenancyAttachment"] = "APPLY_TENANCY_ATTACHMENT";
    OperationType["CreateEnforcedQuotaGovernanceRule"] = "CREATE_ENFORCED_QUOTA_GOVERNANCE_RULE";
    OperationType["CreateEnforcedTagGovernanceRule"] = "CREATE_ENFORCED_TAG_GOVERNANCE_RULE";
    OperationType["UpdateEnforcedQuotaGovernanceRule"] = "UPDATE_ENFORCED_QUOTA_GOVERNANCE_RULE";
    OperationType["UpdateEnforcedTagGovernanceRule"] = "UPDATE_ENFORCED_TAG_GOVERNANCE_RULE";
    OperationType["DeleteEnforcedQuotaGovernanceRule"] = "DELETE_ENFORCED_QUOTA_GOVERNANCE_RULE";
    OperationType["DeleteEnforcedTagGovernanceRule"] = "DELETE_ENFORCED_TAG_GOVERNANCE_RULE";
    OperationType["TenantManagerOptInEventHandler"] = "TENANT_MANAGER_OPT_IN_EVENT_HANDLER";
    OperationType["TenantManagerOptOutEventHandler"] = "TENANT_MANAGER_OPT_OUT_EVENT_HANDLER";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
