// oci-recovery/lib/request/list-protection-policies-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210216
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListProtectionPoliciesRequest = void 0;
var ListProtectionPoliciesRequest;
(function (ListProtectionPoliciesRequest) {
    let Owner;
    (function (Owner) {
        Owner["Oracle"] = "oracle";
        Owner["Customer"] = "customer";
    })(Owner = ListProtectionPoliciesRequest.Owner || (ListProtectionPoliciesRequest.Owner = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListProtectionPoliciesRequest.SortBy || (ListProtectionPoliciesRequest.SortBy = {}));
})(ListProtectionPoliciesRequest = exports.ListProtectionPoliciesRequest || (exports.ListProtectionPoliciesRequest = {}));
