// oci-loganalytics/lib/model/log-set-key-types.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogSetKeyTypes = void 0;
/**
 * Allowed values for logSetKey.
 *
 **/
var LogSetKeyTypes;
(function (LogSetKeyTypes) {
    LogSetKeyTypes["ObjectPath"] = "OBJECT_PATH";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LogSetKeyTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(LogSetKeyTypes = exports.LogSetKeyTypes || (exports.LogSetKeyTypes = {}));
(function (LogSetKeyTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    LogSetKeyTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LogSetKeyTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(LogSetKeyTypes = exports.LogSetKeyTypes || (exports.LogSetKeyTypes = {}));
