// oci-loganalytics/lib/model/scheduled-task.js
"use strict";
/**
 * LogAnalytics API
 * The LogAnalytics API for the LogAnalytics service.

 * OpenAPI spec version: 20200601
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledTask = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ScheduledTask;
(function (ScheduledTask) {
    let TaskStatus;
    (function (TaskStatus) {
        TaskStatus["Ready"] = "READY";
        TaskStatus["Paused"] = "PAUSED";
        TaskStatus["Completed"] = "COMPLETED";
        TaskStatus["Blocked"] = "BLOCKED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        TaskStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(TaskStatus = ScheduledTask.TaskStatus || (ScheduledTask.TaskStatus = {}));
    let PauseReason;
    (function (PauseReason) {
        PauseReason["MetricExtractionNotValid"] = "METRIC_EXTRACTION_NOT_VALID";
        PauseReason["SavedSearchNotValid"] = "SAVED_SEARCH_NOT_VALID";
        PauseReason["SavedSearchNotFound"] = "SAVED_SEARCH_NOT_FOUND";
        PauseReason["QueryStringNotValid"] = "QUERY_STRING_NOT_VALID";
        PauseReason["UserAction"] = "USER_ACTION";
        PauseReason["TenancyLifecycle"] = "TENANCY_LIFECYCLE";
        PauseReason["PurgeResourceNotFound"] = "PURGE_RESOURCE_NOT_FOUND";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PauseReason["UnknownValue"] = "UNKNOWN_VALUE";
    })(PauseReason = ScheduledTask.PauseReason || (ScheduledTask.PauseReason = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleted"] = "DELETED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ScheduledTask.LifecycleState || (ScheduledTask.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "schedules": obj.schedules
                ? obj.schedules.map(item => {
                    return model.Schedule.getJsonObj(item);
                })
                : undefined,
            "action": obj.action ? model.Action.getJsonObj(obj.action) : undefined
        });
        if (obj && "kind" in obj && obj.kind) {
            switch (obj.kind) {
                case "STANDARD":
                    return model.StandardTask.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.kind}`);
            }
        }
        return jsonObj;
    }
    ScheduledTask.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "schedules": obj.schedules
                ? obj.schedules.map(item => {
                    return model.Schedule.getDeserializedJsonObj(item);
                })
                : undefined,
            "action": obj.action ? model.Action.getDeserializedJsonObj(obj.action) : undefined
        });
        if (obj && "kind" in obj && obj.kind) {
            switch (obj.kind) {
                case "STANDARD":
                    return model.StandardTask.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.kind}`);
            }
        }
        return jsonObj;
    }
    ScheduledTask.getDeserializedJsonObj = getDeserializedJsonObj;
})(ScheduledTask = exports.ScheduledTask || (exports.ScheduledTask = {}));
