// oci-oda/lib/model/authentication-grant-type.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationGrantType = void 0;
/**
 * The valid set of Grant Types.
 **/
var AuthenticationGrantType;
(function (AuthenticationGrantType) {
    AuthenticationGrantType["ClientCredentials"] = "CLIENT_CREDENTIALS";
    AuthenticationGrantType["AuthorizationCode"] = "AUTHORIZATION_CODE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AuthenticationGrantType["UnknownValue"] = "UNKNOWN_VALUE";
})(AuthenticationGrantType = exports.AuthenticationGrantType || (exports.AuthenticationGrantType = {}));
(function (AuthenticationGrantType) {
    function getJsonObj(obj) {
        return obj;
    }
    AuthenticationGrantType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AuthenticationGrantType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AuthenticationGrantType = exports.AuthenticationGrantType || (exports.AuthenticationGrantType = {}));
