// oci-oda/lib/model/bot-publish-state.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BotPublishState = void 0;
/**
 * The valid set of Bot publish states.
 **/
var BotPublishState;
(function (BotPublishState) {
    BotPublishState["Published"] = "PUBLISHED";
    BotPublishState["Draft"] = "DRAFT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    BotPublishState["UnknownValue"] = "UNKNOWN_VALUE";
})(BotPublishState = exports.BotPublishState || (exports.BotPublishState = {}));
(function (BotPublishState) {
    function getJsonObj(obj) {
        return obj;
    }
    BotPublishState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BotPublishState.getDeserializedJsonObj = getDeserializedJsonObj;
})(BotPublishState = exports.BotPublishState || (exports.BotPublishState = {}));
