// oci-oda/lib/model/bot-multilingual-mode.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BotMultilingualMode = void 0;
/**
 * The valid set of multilingual modes.
 **/
var BotMultilingualMode;
(function (BotMultilingualMode) {
    BotMultilingualMode["Native"] = "NATIVE";
    BotMultilingualMode["Translation"] = "TRANSLATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    BotMultilingualMode["UnknownValue"] = "UNKNOWN_VALUE";
})(BotMultilingualMode = exports.BotMultilingualMode || (exports.BotMultilingualMode = {}));
(function (BotMultilingualMode) {
    function getJsonObj(obj) {
        return obj;
    }
    BotMultilingualMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    BotMultilingualMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(BotMultilingualMode = exports.BotMultilingualMode || (exports.BotMultilingualMode = {}));
