// oci-oda/lib/model/channel-type.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelType = void 0;
/**
 * The set of supported Channel types.
 **/
var ChannelType;
(function (ChannelType) {
    ChannelType["Android"] = "ANDROID";
    ChannelType["Appevent"] = "APPEVENT";
    ChannelType["Application"] = "APPLICATION";
    ChannelType["Cortana"] = "CORTANA";
    ChannelType["Facebook"] = "FACEBOOK";
    ChannelType["Ios"] = "IOS";
    ChannelType["Msteams"] = "MSTEAMS";
    ChannelType["Oss"] = "OSS";
    ChannelType["Osvc"] = "OSVC";
    ChannelType["Servicecloud"] = "SERVICECLOUD";
    ChannelType["Slack"] = "SLACK";
    ChannelType["Test"] = "TEST";
    ChannelType["Twilio"] = "TWILIO";
    ChannelType["Web"] = "WEB";
    ChannelType["Webhook"] = "WEBHOOK";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ChannelType["UnknownValue"] = "UNKNOWN_VALUE";
})(ChannelType = exports.ChannelType || (exports.ChannelType = {}));
(function (ChannelType) {
    function getJsonObj(obj) {
        return obj;
    }
    ChannelType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ChannelType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ChannelType = exports.ChannelType || (exports.ChannelType = {}));
