// oci-oda/lib/model/channel-category.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelCategory = void 0;
/**
 * The valid set of Categories for a Channel.
 **/
var ChannelCategory;
(function (ChannelCategory) {
    ChannelCategory["Agent"] = "AGENT";
    ChannelCategory["Application"] = "APPLICATION";
    ChannelCategory["Bot"] = "BOT";
    ChannelCategory["BotAsAgent"] = "BOT_AS_AGENT";
    ChannelCategory["System"] = "SYSTEM";
    ChannelCategory["Event"] = "EVENT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ChannelCategory["UnknownValue"] = "UNKNOWN_VALUE";
})(ChannelCategory = exports.ChannelCategory || (exports.ChannelCategory = {}));
(function (ChannelCategory) {
    function getJsonObj(obj) {
        return obj;
    }
    ChannelCategory.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ChannelCategory.getDeserializedJsonObj = getDeserializedJsonObj;
})(ChannelCategory = exports.ChannelCategory || (exports.ChannelCategory = {}));
