// oci-oda/lib/model/index.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdaInstanceAttachment = exports.OdaInstance = exports.MetadataProperty = exports.LifecycleState = exports.ImportedPackageSummary = exports.ImportedPackage = exports.ImportContract = exports.ImportBotDetails = exports.ExportSkillDetails = exports.ExportDigitalAssistantDetails = exports.ExportBotDetails = exports.ErrorBody = exports.DigitalAssistantSummary = exports.DigitalAssistantParameterValue = exports.DigitalAssistantParameterSummary = exports.DigitalAssistantParameterCollection = exports.DigitalAssistantParameter = exports.DigitalAssistantCollection = exports.DigitalAssistant = exports.DefaultParameterValues = exports.CreateTranslatorDetails = exports.CreateSkillParameterDetails = exports.CreateSkillDetails = exports.CreateOdaPrivateEndpointScanProxyDetails = exports.CreateOdaPrivateEndpointDetails = exports.CreateOdaPrivateEndpointAttachmentDetails = exports.CreateOdaInstanceDetails = exports.CreateOdaInstanceAttachmentDetails = exports.CreateImportedPackageDetails = exports.CreateDigitalAssistantDetails = exports.CreateChannelResult = exports.CreateChannelDetails = exports.CreateBotKind = exports.CreateAuthenticationProviderDetails = exports.ConfigureDigitalAssistantParametersDetails = exports.ChannelType = exports.ChannelSummary = exports.ChannelCollection = exports.ChannelCategory = exports.Channel = exports.ChangeOdaPrivateEndpointCompartmentDetails = exports.ChangeOdaInstanceCompartmentDetails = exports.BotPublishState = exports.BotMultilingualMode = exports.Bot = exports.AuthenticationProviderSummary = exports.AuthenticationProviderCollection = exports.AuthenticationProvider = exports.AuthenticationIdentityProvider = exports.AuthenticationGrantType = void 0;
exports.WorkRequest = exports.WebhookPayloadVersion = exports.UpdateTranslatorDetails = exports.UpdateSkillParameterDetails = exports.UpdateSkillDetails = exports.UpdateOdaPrivateEndpointDetails = exports.UpdateOdaInstanceDetails = exports.UpdateOdaInstanceAttachmentDetails = exports.UpdateImportedPackageDetails = exports.UpdateDigitalAssistantParameterDetails = exports.UpdateDigitalAssistantDetails = exports.UpdateChannelDetails = exports.UpdateAuthenticationProviderDetails = exports.TranslatorSummary = exports.TranslatorCollection = exports.Translator = exports.TranslationService = exports.StorageLocation = exports.SkillSummary = exports.SkillParameterSummary = exports.SkillParameterCollection = exports.SkillParameter = exports.SkillCollection = exports.Skill = exports.ServiceCloudClientType = exports.ScanListenerInfo = exports.RestrictedOperation = exports.ResourceTypeMetadata = exports.ResourceTypeImportContract = exports.ResourceTypeDefaultParameterValues = exports.ParameterType = exports.ParameterDefinition = exports.Parameter = exports.PackageSummary = exports.Package = exports.OsvcServiceType = exports.OdaPrivateEndpointSummary = exports.OdaPrivateEndpointScanProxySummary = exports.OdaPrivateEndpointScanProxyCollection = exports.OdaPrivateEndpointScanProxy = exports.OdaPrivateEndpointCollection = exports.OdaPrivateEndpointAttachmentSummary = exports.OdaPrivateEndpointAttachmentCollection = exports.OdaPrivateEndpointAttachment = exports.OdaPrivateEndpoint = exports.OdaInstanceSummary = exports.OdaInstanceOwner = exports.OdaInstanceAttachmentSummary = exports.OdaInstanceAttachmentOwner = exports.OdaInstanceAttachmentCollection = void 0;
exports.OsvcChannel = exports.OSSChannel = exports.MSTeamsChannel = exports.IosChannel = exports.FacebookChannel = exports.ExtendSkillDetails = exports.ExtendDigitalAssistantDetails = exports.CreateWebhookChannelResult = exports.CreateWebhookChannelDetails = exports.CreateWebChannelResult = exports.CreateWebChannelDetails = exports.CreateTwilioChannelResult = exports.CreateTwilioChannelDetails = exports.CreateTestChannelResult = exports.CreateSlackChannelResult = exports.CreateSlackChannelDetails = exports.CreateSkillVersionDetails = exports.CreateServiceCloudChannelResult = exports.CreateServiceCloudChannelDetails = exports.CreateOsvcChannelResult = exports.CreateOsvcChannelDetails = exports.CreateOSSChannelResult = exports.CreateOSSChannelDetails = exports.CreateNewSkillDetails = exports.CreateNewDigitalAssistantDetails = exports.CreateMSTeamsChannelResult = exports.CreateMSTeamsChannelDetails = exports.CreateIosChannelResult = exports.CreateIosChannelDetails = exports.CreateFacebookChannelResult = exports.CreateFacebookChannelDetails = exports.CreateDigitalAssistantVersionDetails = exports.CreateCortanaChannelResult = exports.CreateCortanaChannelDetails = exports.CreateApplicationChannelResult = exports.CreateApplicationChannelDetails = exports.CreateAppEventChannelResult = exports.CreateAppEventChannelDetails = exports.CreateAndroidChannelResult = exports.CreateAndroidChannelDetails = exports.CortanaChannel = exports.CloneSkillDetails = exports.CloneDigitalAssistantDetails = exports.ApplicationChannel = exports.AppEventChannel = exports.AndroidChannel = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = void 0;
exports.WebhookChannel = exports.WebChannel = exports.UpdateWebhookChannelDetails = exports.UpdateWebChannelDetails = exports.UpdateTwilioChannelDetails = exports.UpdateSlackChannelDetails = exports.UpdateServiceCloudChannelDetails = exports.UpdateOsvcChannelDetails = exports.UpdateOSSChannelDetails = exports.UpdateMSTeamsChannelDetails = exports.UpdateIosChannelDetails = exports.UpdateFacebookChannelDetails = exports.UpdateCortanaChannelDetails = exports.UpdateApplicationChannelDetails = exports.UpdateAppEventChannelDetails = exports.UpdateAndroidChannelDetails = exports.TwilioChannel = exports.TestChannel = exports.SlackChannel = exports.ServiceCloudChannel = void 0;
const AuthenticationGrantType = __importStar(require("./authentication-grant-type"));
exports.AuthenticationGrantType = AuthenticationGrantType.AuthenticationGrantType;
const AuthenticationIdentityProvider = __importStar(require("./authentication-identity-provider"));
exports.AuthenticationIdentityProvider = AuthenticationIdentityProvider.AuthenticationIdentityProvider;
const AuthenticationProvider = __importStar(require("./authentication-provider"));
exports.AuthenticationProvider = AuthenticationProvider.AuthenticationProvider;
const AuthenticationProviderCollection = __importStar(require("./authentication-provider-collection"));
exports.AuthenticationProviderCollection = AuthenticationProviderCollection.AuthenticationProviderCollection;
const AuthenticationProviderSummary = __importStar(require("./authentication-provider-summary"));
exports.AuthenticationProviderSummary = AuthenticationProviderSummary.AuthenticationProviderSummary;
const Bot = __importStar(require("./bot"));
exports.Bot = Bot.Bot;
const BotMultilingualMode = __importStar(require("./bot-multilingual-mode"));
exports.BotMultilingualMode = BotMultilingualMode.BotMultilingualMode;
const BotPublishState = __importStar(require("./bot-publish-state"));
exports.BotPublishState = BotPublishState.BotPublishState;
const ChangeOdaInstanceCompartmentDetails = __importStar(require("./change-oda-instance-compartment-details"));
exports.ChangeOdaInstanceCompartmentDetails = ChangeOdaInstanceCompartmentDetails.ChangeOdaInstanceCompartmentDetails;
const ChangeOdaPrivateEndpointCompartmentDetails = __importStar(require("./change-oda-private-endpoint-compartment-details"));
exports.ChangeOdaPrivateEndpointCompartmentDetails = ChangeOdaPrivateEndpointCompartmentDetails.ChangeOdaPrivateEndpointCompartmentDetails;
const Channel = __importStar(require("./channel"));
exports.Channel = Channel.Channel;
const ChannelCategory = __importStar(require("./channel-category"));
exports.ChannelCategory = ChannelCategory.ChannelCategory;
const ChannelCollection = __importStar(require("./channel-collection"));
exports.ChannelCollection = ChannelCollection.ChannelCollection;
const ChannelSummary = __importStar(require("./channel-summary"));
exports.ChannelSummary = ChannelSummary.ChannelSummary;
const ChannelType = __importStar(require("./channel-type"));
exports.ChannelType = ChannelType.ChannelType;
const ConfigureDigitalAssistantParametersDetails = __importStar(require("./configure-digital-assistant-parameters-details"));
exports.ConfigureDigitalAssistantParametersDetails = ConfigureDigitalAssistantParametersDetails.ConfigureDigitalAssistantParametersDetails;
const CreateAuthenticationProviderDetails = __importStar(require("./create-authentication-provider-details"));
exports.CreateAuthenticationProviderDetails = CreateAuthenticationProviderDetails.CreateAuthenticationProviderDetails;
const CreateBotKind = __importStar(require("./create-bot-kind"));
exports.CreateBotKind = CreateBotKind.CreateBotKind;
const CreateChannelDetails = __importStar(require("./create-channel-details"));
exports.CreateChannelDetails = CreateChannelDetails.CreateChannelDetails;
const CreateChannelResult = __importStar(require("./create-channel-result"));
exports.CreateChannelResult = CreateChannelResult.CreateChannelResult;
const CreateDigitalAssistantDetails = __importStar(require("./create-digital-assistant-details"));
exports.CreateDigitalAssistantDetails = CreateDigitalAssistantDetails.CreateDigitalAssistantDetails;
const CreateImportedPackageDetails = __importStar(require("./create-imported-package-details"));
exports.CreateImportedPackageDetails = CreateImportedPackageDetails.CreateImportedPackageDetails;
const CreateOdaInstanceAttachmentDetails = __importStar(require("./create-oda-instance-attachment-details"));
exports.CreateOdaInstanceAttachmentDetails = CreateOdaInstanceAttachmentDetails.CreateOdaInstanceAttachmentDetails;
const CreateOdaInstanceDetails = __importStar(require("./create-oda-instance-details"));
exports.CreateOdaInstanceDetails = CreateOdaInstanceDetails.CreateOdaInstanceDetails;
const CreateOdaPrivateEndpointAttachmentDetails = __importStar(require("./create-oda-private-endpoint-attachment-details"));
exports.CreateOdaPrivateEndpointAttachmentDetails = CreateOdaPrivateEndpointAttachmentDetails.CreateOdaPrivateEndpointAttachmentDetails;
const CreateOdaPrivateEndpointDetails = __importStar(require("./create-oda-private-endpoint-details"));
exports.CreateOdaPrivateEndpointDetails = CreateOdaPrivateEndpointDetails.CreateOdaPrivateEndpointDetails;
const CreateOdaPrivateEndpointScanProxyDetails = __importStar(require("./create-oda-private-endpoint-scan-proxy-details"));
exports.CreateOdaPrivateEndpointScanProxyDetails = CreateOdaPrivateEndpointScanProxyDetails.CreateOdaPrivateEndpointScanProxyDetails;
const CreateSkillDetails = __importStar(require("./create-skill-details"));
exports.CreateSkillDetails = CreateSkillDetails.CreateSkillDetails;
const CreateSkillParameterDetails = __importStar(require("./create-skill-parameter-details"));
exports.CreateSkillParameterDetails = CreateSkillParameterDetails.CreateSkillParameterDetails;
const CreateTranslatorDetails = __importStar(require("./create-translator-details"));
exports.CreateTranslatorDetails = CreateTranslatorDetails.CreateTranslatorDetails;
const DefaultParameterValues = __importStar(require("./default-parameter-values"));
exports.DefaultParameterValues = DefaultParameterValues.DefaultParameterValues;
const DigitalAssistant = __importStar(require("./digital-assistant"));
exports.DigitalAssistant = DigitalAssistant.DigitalAssistant;
const DigitalAssistantCollection = __importStar(require("./digital-assistant-collection"));
exports.DigitalAssistantCollection = DigitalAssistantCollection.DigitalAssistantCollection;
const DigitalAssistantParameter = __importStar(require("./digital-assistant-parameter"));
exports.DigitalAssistantParameter = DigitalAssistantParameter.DigitalAssistantParameter;
const DigitalAssistantParameterCollection = __importStar(require("./digital-assistant-parameter-collection"));
exports.DigitalAssistantParameterCollection = DigitalAssistantParameterCollection.DigitalAssistantParameterCollection;
const DigitalAssistantParameterSummary = __importStar(require("./digital-assistant-parameter-summary"));
exports.DigitalAssistantParameterSummary = DigitalAssistantParameterSummary.DigitalAssistantParameterSummary;
const DigitalAssistantParameterValue = __importStar(require("./digital-assistant-parameter-value"));
exports.DigitalAssistantParameterValue = DigitalAssistantParameterValue.DigitalAssistantParameterValue;
const DigitalAssistantSummary = __importStar(require("./digital-assistant-summary"));
exports.DigitalAssistantSummary = DigitalAssistantSummary.DigitalAssistantSummary;
const ErrorBody = __importStar(require("./error-body"));
exports.ErrorBody = ErrorBody.ErrorBody;
const ExportBotDetails = __importStar(require("./export-bot-details"));
exports.ExportBotDetails = ExportBotDetails.ExportBotDetails;
const ExportDigitalAssistantDetails = __importStar(require("./export-digital-assistant-details"));
exports.ExportDigitalAssistantDetails = ExportDigitalAssistantDetails.ExportDigitalAssistantDetails;
const ExportSkillDetails = __importStar(require("./export-skill-details"));
exports.ExportSkillDetails = ExportSkillDetails.ExportSkillDetails;
const ImportBotDetails = __importStar(require("./import-bot-details"));
exports.ImportBotDetails = ImportBotDetails.ImportBotDetails;
const ImportContract = __importStar(require("./import-contract"));
exports.ImportContract = ImportContract.ImportContract;
const ImportedPackage = __importStar(require("./imported-package"));
exports.ImportedPackage = ImportedPackage.ImportedPackage;
const ImportedPackageSummary = __importStar(require("./imported-package-summary"));
exports.ImportedPackageSummary = ImportedPackageSummary.ImportedPackageSummary;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const MetadataProperty = __importStar(require("./metadata-property"));
exports.MetadataProperty = MetadataProperty.MetadataProperty;
const OdaInstance = __importStar(require("./oda-instance"));
exports.OdaInstance = OdaInstance.OdaInstance;
const OdaInstanceAttachment = __importStar(require("./oda-instance-attachment"));
exports.OdaInstanceAttachment = OdaInstanceAttachment.OdaInstanceAttachment;
const OdaInstanceAttachmentCollection = __importStar(require("./oda-instance-attachment-collection"));
exports.OdaInstanceAttachmentCollection = OdaInstanceAttachmentCollection.OdaInstanceAttachmentCollection;
const OdaInstanceAttachmentOwner = __importStar(require("./oda-instance-attachment-owner"));
exports.OdaInstanceAttachmentOwner = OdaInstanceAttachmentOwner.OdaInstanceAttachmentOwner;
const OdaInstanceAttachmentSummary = __importStar(require("./oda-instance-attachment-summary"));
exports.OdaInstanceAttachmentSummary = OdaInstanceAttachmentSummary.OdaInstanceAttachmentSummary;
const OdaInstanceOwner = __importStar(require("./oda-instance-owner"));
exports.OdaInstanceOwner = OdaInstanceOwner.OdaInstanceOwner;
const OdaInstanceSummary = __importStar(require("./oda-instance-summary"));
exports.OdaInstanceSummary = OdaInstanceSummary.OdaInstanceSummary;
const OdaPrivateEndpoint = __importStar(require("./oda-private-endpoint"));
exports.OdaPrivateEndpoint = OdaPrivateEndpoint.OdaPrivateEndpoint;
const OdaPrivateEndpointAttachment = __importStar(require("./oda-private-endpoint-attachment"));
exports.OdaPrivateEndpointAttachment = OdaPrivateEndpointAttachment.OdaPrivateEndpointAttachment;
const OdaPrivateEndpointAttachmentCollection = __importStar(require("./oda-private-endpoint-attachment-collection"));
exports.OdaPrivateEndpointAttachmentCollection = OdaPrivateEndpointAttachmentCollection.OdaPrivateEndpointAttachmentCollection;
const OdaPrivateEndpointAttachmentSummary = __importStar(require("./oda-private-endpoint-attachment-summary"));
exports.OdaPrivateEndpointAttachmentSummary = OdaPrivateEndpointAttachmentSummary.OdaPrivateEndpointAttachmentSummary;
const OdaPrivateEndpointCollection = __importStar(require("./oda-private-endpoint-collection"));
exports.OdaPrivateEndpointCollection = OdaPrivateEndpointCollection.OdaPrivateEndpointCollection;
const OdaPrivateEndpointScanProxy = __importStar(require("./oda-private-endpoint-scan-proxy"));
exports.OdaPrivateEndpointScanProxy = OdaPrivateEndpointScanProxy.OdaPrivateEndpointScanProxy;
const OdaPrivateEndpointScanProxyCollection = __importStar(require("./oda-private-endpoint-scan-proxy-collection"));
exports.OdaPrivateEndpointScanProxyCollection = OdaPrivateEndpointScanProxyCollection.OdaPrivateEndpointScanProxyCollection;
const OdaPrivateEndpointScanProxySummary = __importStar(require("./oda-private-endpoint-scan-proxy-summary"));
exports.OdaPrivateEndpointScanProxySummary = OdaPrivateEndpointScanProxySummary.OdaPrivateEndpointScanProxySummary;
const OdaPrivateEndpointSummary = __importStar(require("./oda-private-endpoint-summary"));
exports.OdaPrivateEndpointSummary = OdaPrivateEndpointSummary.OdaPrivateEndpointSummary;
const OsvcServiceType = __importStar(require("./osvc-service-type"));
exports.OsvcServiceType = OsvcServiceType.OsvcServiceType;
const Package = __importStar(require("./package"));
exports.Package = Package.Package;
const PackageSummary = __importStar(require("./package-summary"));
exports.PackageSummary = PackageSummary.PackageSummary;
const Parameter = __importStar(require("./parameter"));
exports.Parameter = Parameter.Parameter;
const ParameterDefinition = __importStar(require("./parameter-definition"));
exports.ParameterDefinition = ParameterDefinition.ParameterDefinition;
const ParameterType = __importStar(require("./parameter-type"));
exports.ParameterType = ParameterType.ParameterType;
const ResourceTypeDefaultParameterValues = __importStar(require("./resource-type-default-parameter-values"));
exports.ResourceTypeDefaultParameterValues = ResourceTypeDefaultParameterValues.ResourceTypeDefaultParameterValues;
const ResourceTypeImportContract = __importStar(require("./resource-type-import-contract"));
exports.ResourceTypeImportContract = ResourceTypeImportContract.ResourceTypeImportContract;
const ResourceTypeMetadata = __importStar(require("./resource-type-metadata"));
exports.ResourceTypeMetadata = ResourceTypeMetadata.ResourceTypeMetadata;
const RestrictedOperation = __importStar(require("./restricted-operation"));
exports.RestrictedOperation = RestrictedOperation.RestrictedOperation;
const ScanListenerInfo = __importStar(require("./scan-listener-info"));
exports.ScanListenerInfo = ScanListenerInfo.ScanListenerInfo;
const ServiceCloudClientType = __importStar(require("./service-cloud-client-type"));
exports.ServiceCloudClientType = ServiceCloudClientType.ServiceCloudClientType;
const Skill = __importStar(require("./skill"));
exports.Skill = Skill.Skill;
const SkillCollection = __importStar(require("./skill-collection"));
exports.SkillCollection = SkillCollection.SkillCollection;
const SkillParameter = __importStar(require("./skill-parameter"));
exports.SkillParameter = SkillParameter.SkillParameter;
const SkillParameterCollection = __importStar(require("./skill-parameter-collection"));
exports.SkillParameterCollection = SkillParameterCollection.SkillParameterCollection;
const SkillParameterSummary = __importStar(require("./skill-parameter-summary"));
exports.SkillParameterSummary = SkillParameterSummary.SkillParameterSummary;
const SkillSummary = __importStar(require("./skill-summary"));
exports.SkillSummary = SkillSummary.SkillSummary;
const StorageLocation = __importStar(require("./storage-location"));
exports.StorageLocation = StorageLocation.StorageLocation;
const TranslationService = __importStar(require("./translation-service"));
exports.TranslationService = TranslationService.TranslationService;
const Translator = __importStar(require("./translator"));
exports.Translator = Translator.Translator;
const TranslatorCollection = __importStar(require("./translator-collection"));
exports.TranslatorCollection = TranslatorCollection.TranslatorCollection;
const TranslatorSummary = __importStar(require("./translator-summary"));
exports.TranslatorSummary = TranslatorSummary.TranslatorSummary;
const UpdateAuthenticationProviderDetails = __importStar(require("./update-authentication-provider-details"));
exports.UpdateAuthenticationProviderDetails = UpdateAuthenticationProviderDetails.UpdateAuthenticationProviderDetails;
const UpdateChannelDetails = __importStar(require("./update-channel-details"));
exports.UpdateChannelDetails = UpdateChannelDetails.UpdateChannelDetails;
const UpdateDigitalAssistantDetails = __importStar(require("./update-digital-assistant-details"));
exports.UpdateDigitalAssistantDetails = UpdateDigitalAssistantDetails.UpdateDigitalAssistantDetails;
const UpdateDigitalAssistantParameterDetails = __importStar(require("./update-digital-assistant-parameter-details"));
exports.UpdateDigitalAssistantParameterDetails = UpdateDigitalAssistantParameterDetails.UpdateDigitalAssistantParameterDetails;
const UpdateImportedPackageDetails = __importStar(require("./update-imported-package-details"));
exports.UpdateImportedPackageDetails = UpdateImportedPackageDetails.UpdateImportedPackageDetails;
const UpdateOdaInstanceAttachmentDetails = __importStar(require("./update-oda-instance-attachment-details"));
exports.UpdateOdaInstanceAttachmentDetails = UpdateOdaInstanceAttachmentDetails.UpdateOdaInstanceAttachmentDetails;
const UpdateOdaInstanceDetails = __importStar(require("./update-oda-instance-details"));
exports.UpdateOdaInstanceDetails = UpdateOdaInstanceDetails.UpdateOdaInstanceDetails;
const UpdateOdaPrivateEndpointDetails = __importStar(require("./update-oda-private-endpoint-details"));
exports.UpdateOdaPrivateEndpointDetails = UpdateOdaPrivateEndpointDetails.UpdateOdaPrivateEndpointDetails;
const UpdateSkillDetails = __importStar(require("./update-skill-details"));
exports.UpdateSkillDetails = UpdateSkillDetails.UpdateSkillDetails;
const UpdateSkillParameterDetails = __importStar(require("./update-skill-parameter-details"));
exports.UpdateSkillParameterDetails = UpdateSkillParameterDetails.UpdateSkillParameterDetails;
const UpdateTranslatorDetails = __importStar(require("./update-translator-details"));
exports.UpdateTranslatorDetails = UpdateTranslatorDetails.UpdateTranslatorDetails;
const WebhookPayloadVersion = __importStar(require("./webhook-payload-version"));
exports.WebhookPayloadVersion = WebhookPayloadVersion.WebhookPayloadVersion;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const AndroidChannel = __importStar(require("./android-channel"));
exports.AndroidChannel = AndroidChannel.AndroidChannel;
const AppEventChannel = __importStar(require("./app-event-channel"));
exports.AppEventChannel = AppEventChannel.AppEventChannel;
const ApplicationChannel = __importStar(require("./application-channel"));
exports.ApplicationChannel = ApplicationChannel.ApplicationChannel;
const CloneDigitalAssistantDetails = __importStar(require("./clone-digital-assistant-details"));
exports.CloneDigitalAssistantDetails = CloneDigitalAssistantDetails.CloneDigitalAssistantDetails;
const CloneSkillDetails = __importStar(require("./clone-skill-details"));
exports.CloneSkillDetails = CloneSkillDetails.CloneSkillDetails;
const CortanaChannel = __importStar(require("./cortana-channel"));
exports.CortanaChannel = CortanaChannel.CortanaChannel;
const CreateAndroidChannelDetails = __importStar(require("./create-android-channel-details"));
exports.CreateAndroidChannelDetails = CreateAndroidChannelDetails.CreateAndroidChannelDetails;
const CreateAndroidChannelResult = __importStar(require("./create-android-channel-result"));
exports.CreateAndroidChannelResult = CreateAndroidChannelResult.CreateAndroidChannelResult;
const CreateAppEventChannelDetails = __importStar(require("./create-app-event-channel-details"));
exports.CreateAppEventChannelDetails = CreateAppEventChannelDetails.CreateAppEventChannelDetails;
const CreateAppEventChannelResult = __importStar(require("./create-app-event-channel-result"));
exports.CreateAppEventChannelResult = CreateAppEventChannelResult.CreateAppEventChannelResult;
const CreateApplicationChannelDetails = __importStar(require("./create-application-channel-details"));
exports.CreateApplicationChannelDetails = CreateApplicationChannelDetails.CreateApplicationChannelDetails;
const CreateApplicationChannelResult = __importStar(require("./create-application-channel-result"));
exports.CreateApplicationChannelResult = CreateApplicationChannelResult.CreateApplicationChannelResult;
const CreateCortanaChannelDetails = __importStar(require("./create-cortana-channel-details"));
exports.CreateCortanaChannelDetails = CreateCortanaChannelDetails.CreateCortanaChannelDetails;
const CreateCortanaChannelResult = __importStar(require("./create-cortana-channel-result"));
exports.CreateCortanaChannelResult = CreateCortanaChannelResult.CreateCortanaChannelResult;
const CreateDigitalAssistantVersionDetails = __importStar(require("./create-digital-assistant-version-details"));
exports.CreateDigitalAssistantVersionDetails = CreateDigitalAssistantVersionDetails.CreateDigitalAssistantVersionDetails;
const CreateFacebookChannelDetails = __importStar(require("./create-facebook-channel-details"));
exports.CreateFacebookChannelDetails = CreateFacebookChannelDetails.CreateFacebookChannelDetails;
const CreateFacebookChannelResult = __importStar(require("./create-facebook-channel-result"));
exports.CreateFacebookChannelResult = CreateFacebookChannelResult.CreateFacebookChannelResult;
const CreateIosChannelDetails = __importStar(require("./create-ios-channel-details"));
exports.CreateIosChannelDetails = CreateIosChannelDetails.CreateIosChannelDetails;
const CreateIosChannelResult = __importStar(require("./create-ios-channel-result"));
exports.CreateIosChannelResult = CreateIosChannelResult.CreateIosChannelResult;
const CreateMSTeamsChannelDetails = __importStar(require("./create-ms-teams-channel-details"));
exports.CreateMSTeamsChannelDetails = CreateMSTeamsChannelDetails.CreateMSTeamsChannelDetails;
const CreateMSTeamsChannelResult = __importStar(require("./create-ms-teams-channel-result"));
exports.CreateMSTeamsChannelResult = CreateMSTeamsChannelResult.CreateMSTeamsChannelResult;
const CreateNewDigitalAssistantDetails = __importStar(require("./create-new-digital-assistant-details"));
exports.CreateNewDigitalAssistantDetails = CreateNewDigitalAssistantDetails.CreateNewDigitalAssistantDetails;
const CreateNewSkillDetails = __importStar(require("./create-new-skill-details"));
exports.CreateNewSkillDetails = CreateNewSkillDetails.CreateNewSkillDetails;
const CreateOSSChannelDetails = __importStar(require("./create-os-schannel-details"));
exports.CreateOSSChannelDetails = CreateOSSChannelDetails.CreateOSSChannelDetails;
const CreateOSSChannelResult = __importStar(require("./create-os-schannel-result"));
exports.CreateOSSChannelResult = CreateOSSChannelResult.CreateOSSChannelResult;
const CreateOsvcChannelDetails = __importStar(require("./create-osvc-channel-details"));
exports.CreateOsvcChannelDetails = CreateOsvcChannelDetails.CreateOsvcChannelDetails;
const CreateOsvcChannelResult = __importStar(require("./create-osvc-channel-result"));
exports.CreateOsvcChannelResult = CreateOsvcChannelResult.CreateOsvcChannelResult;
const CreateServiceCloudChannelDetails = __importStar(require("./create-service-cloud-channel-details"));
exports.CreateServiceCloudChannelDetails = CreateServiceCloudChannelDetails.CreateServiceCloudChannelDetails;
const CreateServiceCloudChannelResult = __importStar(require("./create-service-cloud-channel-result"));
exports.CreateServiceCloudChannelResult = CreateServiceCloudChannelResult.CreateServiceCloudChannelResult;
const CreateSkillVersionDetails = __importStar(require("./create-skill-version-details"));
exports.CreateSkillVersionDetails = CreateSkillVersionDetails.CreateSkillVersionDetails;
const CreateSlackChannelDetails = __importStar(require("./create-slack-channel-details"));
exports.CreateSlackChannelDetails = CreateSlackChannelDetails.CreateSlackChannelDetails;
const CreateSlackChannelResult = __importStar(require("./create-slack-channel-result"));
exports.CreateSlackChannelResult = CreateSlackChannelResult.CreateSlackChannelResult;
const CreateTestChannelResult = __importStar(require("./create-test-channel-result"));
exports.CreateTestChannelResult = CreateTestChannelResult.CreateTestChannelResult;
const CreateTwilioChannelDetails = __importStar(require("./create-twilio-channel-details"));
exports.CreateTwilioChannelDetails = CreateTwilioChannelDetails.CreateTwilioChannelDetails;
const CreateTwilioChannelResult = __importStar(require("./create-twilio-channel-result"));
exports.CreateTwilioChannelResult = CreateTwilioChannelResult.CreateTwilioChannelResult;
const CreateWebChannelDetails = __importStar(require("./create-web-channel-details"));
exports.CreateWebChannelDetails = CreateWebChannelDetails.CreateWebChannelDetails;
const CreateWebChannelResult = __importStar(require("./create-web-channel-result"));
exports.CreateWebChannelResult = CreateWebChannelResult.CreateWebChannelResult;
const CreateWebhookChannelDetails = __importStar(require("./create-webhook-channel-details"));
exports.CreateWebhookChannelDetails = CreateWebhookChannelDetails.CreateWebhookChannelDetails;
const CreateWebhookChannelResult = __importStar(require("./create-webhook-channel-result"));
exports.CreateWebhookChannelResult = CreateWebhookChannelResult.CreateWebhookChannelResult;
const ExtendDigitalAssistantDetails = __importStar(require("./extend-digital-assistant-details"));
exports.ExtendDigitalAssistantDetails = ExtendDigitalAssistantDetails.ExtendDigitalAssistantDetails;
const ExtendSkillDetails = __importStar(require("./extend-skill-details"));
exports.ExtendSkillDetails = ExtendSkillDetails.ExtendSkillDetails;
const FacebookChannel = __importStar(require("./facebook-channel"));
exports.FacebookChannel = FacebookChannel.FacebookChannel;
const IosChannel = __importStar(require("./ios-channel"));
exports.IosChannel = IosChannel.IosChannel;
const MSTeamsChannel = __importStar(require("./m-steams-channel"));
exports.MSTeamsChannel = MSTeamsChannel.MSTeamsChannel;
const OSSChannel = __importStar(require("./o-ss-channel"));
exports.OSSChannel = OSSChannel.OSSChannel;
const OsvcChannel = __importStar(require("./osvc-channel"));
exports.OsvcChannel = OsvcChannel.OsvcChannel;
const ServiceCloudChannel = __importStar(require("./service-cloud-channel"));
exports.ServiceCloudChannel = ServiceCloudChannel.ServiceCloudChannel;
const SlackChannel = __importStar(require("./slack-channel"));
exports.SlackChannel = SlackChannel.SlackChannel;
const TestChannel = __importStar(require("./test-channel"));
exports.TestChannel = TestChannel.TestChannel;
const TwilioChannel = __importStar(require("./twilio-channel"));
exports.TwilioChannel = TwilioChannel.TwilioChannel;
const UpdateAndroidChannelDetails = __importStar(require("./update-android-channel-details"));
exports.UpdateAndroidChannelDetails = UpdateAndroidChannelDetails.UpdateAndroidChannelDetails;
const UpdateAppEventChannelDetails = __importStar(require("./update-app-event-channel-details"));
exports.UpdateAppEventChannelDetails = UpdateAppEventChannelDetails.UpdateAppEventChannelDetails;
const UpdateApplicationChannelDetails = __importStar(require("./update-application-channel-details"));
exports.UpdateApplicationChannelDetails = UpdateApplicationChannelDetails.UpdateApplicationChannelDetails;
const UpdateCortanaChannelDetails = __importStar(require("./update-cortana-channel-details"));
exports.UpdateCortanaChannelDetails = UpdateCortanaChannelDetails.UpdateCortanaChannelDetails;
const UpdateFacebookChannelDetails = __importStar(require("./update-facebook-channel-details"));
exports.UpdateFacebookChannelDetails = UpdateFacebookChannelDetails.UpdateFacebookChannelDetails;
const UpdateIosChannelDetails = __importStar(require("./update-ios-channel-details"));
exports.UpdateIosChannelDetails = UpdateIosChannelDetails.UpdateIosChannelDetails;
const UpdateMSTeamsChannelDetails = __importStar(require("./update-ms-teams-channel-details"));
exports.UpdateMSTeamsChannelDetails = UpdateMSTeamsChannelDetails.UpdateMSTeamsChannelDetails;
const UpdateOSSChannelDetails = __importStar(require("./update-os-schannel-details"));
exports.UpdateOSSChannelDetails = UpdateOSSChannelDetails.UpdateOSSChannelDetails;
const UpdateOsvcChannelDetails = __importStar(require("./update-osvc-channel-details"));
exports.UpdateOsvcChannelDetails = UpdateOsvcChannelDetails.UpdateOsvcChannelDetails;
const UpdateServiceCloudChannelDetails = __importStar(require("./update-service-cloud-channel-details"));
exports.UpdateServiceCloudChannelDetails = UpdateServiceCloudChannelDetails.UpdateServiceCloudChannelDetails;
const UpdateSlackChannelDetails = __importStar(require("./update-slack-channel-details"));
exports.UpdateSlackChannelDetails = UpdateSlackChannelDetails.UpdateSlackChannelDetails;
const UpdateTwilioChannelDetails = __importStar(require("./update-twilio-channel-details"));
exports.UpdateTwilioChannelDetails = UpdateTwilioChannelDetails.UpdateTwilioChannelDetails;
const UpdateWebChannelDetails = __importStar(require("./update-web-channel-details"));
exports.UpdateWebChannelDetails = UpdateWebChannelDetails.UpdateWebChannelDetails;
const UpdateWebhookChannelDetails = __importStar(require("./update-webhook-channel-details"));
exports.UpdateWebhookChannelDetails = UpdateWebhookChannelDetails.UpdateWebhookChannelDetails;
const WebChannel = __importStar(require("./web-channel"));
exports.WebChannel = WebChannel.WebChannel;
const WebhookChannel = __importStar(require("./webhook-channel"));
exports.WebhookChannel = WebhookChannel.WebhookChannel;
