// oci-oda/lib/model/parameter-definition.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterDefinition = void 0;
var ParameterDefinition;
(function (ParameterDefinition) {
    let Type;
    (function (Type) {
        Type["String"] = "STRING";
        Type["Uri"] = "URI";
        Type["Url"] = "URL";
        Type["Number"] = "NUMBER";
        Type["Boolean"] = "BOOLEAN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = ParameterDefinition.Type || (ParameterDefinition.Type = {}));
    let Direction;
    (function (Direction) {
        Direction["Input"] = "INPUT";
        Direction["Output"] = "OUTPUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Direction["UnknownValue"] = "UNKNOWN_VALUE";
    })(Direction = ParameterDefinition.Direction || (ParameterDefinition.Direction = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ParameterDefinition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ParameterDefinition.getDeserializedJsonObj = getDeserializedJsonObj;
})(ParameterDefinition = exports.ParameterDefinition || (exports.ParameterDefinition = {}));
