// oci-oda/lib/model/translation-service.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TranslationService = void 0;
/**
 * The valid set of Translation Services to use for translation.
 **/
var TranslationService;
(function (TranslationService) {
    TranslationService["Google"] = "GOOGLE";
    TranslationService["Microsoft"] = "MICROSOFT";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    TranslationService["UnknownValue"] = "UNKNOWN_VALUE";
})(TranslationService = exports.TranslationService || (exports.TranslationService = {}));
(function (TranslationService) {
    function getJsonObj(obj) {
        return obj;
    }
    TranslationService.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    TranslationService.getDeserializedJsonObj = getDeserializedJsonObj;
})(TranslationService = exports.TranslationService || (exports.TranslationService = {}));
