// oci-oda/lib/model/webhook-payload-version.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookPayloadVersion = void 0;
/**
 * The payload version defines the expected structure of the payloads sent and received by the Webhook channel.
 **/
var WebhookPayloadVersion;
(function (WebhookPayloadVersion) {
    WebhookPayloadVersion["_10"] = "1.0";
    WebhookPayloadVersion["_11"] = "1.1";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WebhookPayloadVersion["UnknownValue"] = "UNKNOWN_VALUE";
})(WebhookPayloadVersion = exports.WebhookPayloadVersion || (exports.WebhookPayloadVersion = {}));
(function (WebhookPayloadVersion) {
    function getJsonObj(obj) {
        return obj;
    }
    WebhookPayloadVersion.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WebhookPayloadVersion.getDeserializedJsonObj = getDeserializedJsonObj;
})(WebhookPayloadVersion = exports.WebhookPayloadVersion || (exports.WebhookPayloadVersion = {}));
