// oci-oda/lib/model/work-request-resource.js
"use strict";
/**
 * Digital Assistant Service Instance API
 * API to create and maintain Oracle Digital Assistant service instances.
 * OpenAPI spec version: 20190506
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = void 0;
var WorkRequestResource;
(function (WorkRequestResource) {
    let ResourceAction;
    (function (ResourceAction) {
        ResourceAction["Create"] = "CREATE";
        ResourceAction["Update"] = "UPDATE";
        ResourceAction["Delete"] = "DELETE";
        ResourceAction["Purge"] = "PURGE";
        ResourceAction["Recover"] = "RECOVER";
        ResourceAction["Stop"] = "STOP";
        ResourceAction["Start"] = "START";
        ResourceAction["ChangeCompartment"] = "CHANGE_COMPARTMENT";
        ResourceAction["ChangeCustEncKey"] = "CHANGE_CUST_ENC_KEY";
        ResourceAction["DeactCustEncKey"] = "DEACT_CUST_ENC_KEY";
        ResourceAction["CreateAssociation"] = "CREATE_ASSOCIATION";
        ResourceAction["DeleteAssociation"] = "DELETE_ASSOCIATION";
        ResourceAction["UpdateEntitlementsForCacct"] = "UPDATE_ENTITLEMENTS_FOR_CACCT";
        ResourceAction["CreateOdaInstanceAttachment"] = "CREATE_ODA_INSTANCE_ATTACHMENT";
        ResourceAction["UpdateOdaInstanceAttachment"] = "UPDATE_ODA_INSTANCE_ATTACHMENT";
        ResourceAction["DeleteOdaInstanceAttachment"] = "DELETE_ODA_INSTANCE_ATTACHMENT";
        ResourceAction["CreateImportedPackage"] = "CREATE_IMPORTED_PACKAGE";
        ResourceAction["UpdateImportedPackage"] = "UPDATE_IMPORTED_PACKAGE";
        ResourceAction["DeleteImportedPackage"] = "DELETE_IMPORTED_PACKAGE";
        ResourceAction["Export"] = "EXPORT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResourceAction["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResourceAction = WorkRequestResource.ResourceAction || (WorkRequestResource.ResourceAction = {}));
    let Status;
    (function (Status) {
        Status["Accepted"] = "ACCEPTED";
        Status["InProgress"] = "IN_PROGRESS";
        Status["Succeeded"] = "SUCCEEDED";
        Status["Failed"] = "FAILED";
        Status["Canceling"] = "CANCELING";
        Status["Canceled"] = "CANCELED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = WorkRequestResource.Status || (WorkRequestResource.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkRequestResource.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    WorkRequestResource.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestResource = exports.WorkRequestResource || (exports.WorkRequestResource = {}));
