// oci-lockbox/lib/model/index.js
"use strict";
/**
 * Managed Access API
 * Use the Managed Access API to approve access requests, create and manage templates, and manage resource approval settings. For more information, see [Managed Access Overview](https://docs.oracle.com/en-us/iaas/managed-access/overview.htm).

Use the table of contents and search tool to explore the Managed Access API.

 * OpenAPI spec version: 20220126
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateLockboxDetails = exports.UpdateApprovalTemplateDetails = exports.SortOrder = exports.PersonaLevel = exports.OperationType = exports.OperationStatus = exports.LockboxSummary = exports.LockboxPartner = exports.LockboxCollection = exports.LockboxAutoApprovalState = exports.Lockbox = exports.HandleAccessRequestDetails = exports.CreateLockboxDetails = exports.CreateApprovalTemplateDetails = exports.CreateAccessRequestDetails = exports.ChangeLockboxCompartmentDetails = exports.ChangeApprovalTemplateCompartmentDetails = exports.ApproverType = exports.ApproverLevels = exports.ApproverInfo = exports.ApprovalTemplateSummary = exports.ApprovalTemplateCollection = exports.ApprovalTemplate = exports.ActivityLog = exports.ActionType = exports.AccessRequestSummary = exports.AccessRequestCollection = exports.AccessRequestActionType = exports.AccessRequest = exports.AccessMaterials = exports.AccessContextAttributeCollection = exports.AccessContextAttribute = void 0;
const AccessContextAttribute = __importStar(require("./access-context-attribute"));
exports.AccessContextAttribute = AccessContextAttribute.AccessContextAttribute;
const AccessContextAttributeCollection = __importStar(require("./access-context-attribute-collection"));
exports.AccessContextAttributeCollection = AccessContextAttributeCollection.AccessContextAttributeCollection;
const AccessMaterials = __importStar(require("./access-materials"));
exports.AccessMaterials = AccessMaterials.AccessMaterials;
const AccessRequest = __importStar(require("./access-request"));
exports.AccessRequest = AccessRequest.AccessRequest;
const AccessRequestActionType = __importStar(require("./access-request-action-type"));
exports.AccessRequestActionType = AccessRequestActionType.AccessRequestActionType;
const AccessRequestCollection = __importStar(require("./access-request-collection"));
exports.AccessRequestCollection = AccessRequestCollection.AccessRequestCollection;
const AccessRequestSummary = __importStar(require("./access-request-summary"));
exports.AccessRequestSummary = AccessRequestSummary.AccessRequestSummary;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ActivityLog = __importStar(require("./activity-log"));
exports.ActivityLog = ActivityLog.ActivityLog;
const ApprovalTemplate = __importStar(require("./approval-template"));
exports.ApprovalTemplate = ApprovalTemplate.ApprovalTemplate;
const ApprovalTemplateCollection = __importStar(require("./approval-template-collection"));
exports.ApprovalTemplateCollection = ApprovalTemplateCollection.ApprovalTemplateCollection;
const ApprovalTemplateSummary = __importStar(require("./approval-template-summary"));
exports.ApprovalTemplateSummary = ApprovalTemplateSummary.ApprovalTemplateSummary;
const ApproverInfo = __importStar(require("./approver-info"));
exports.ApproverInfo = ApproverInfo.ApproverInfo;
const ApproverLevels = __importStar(require("./approver-levels"));
exports.ApproverLevels = ApproverLevels.ApproverLevels;
const ApproverType = __importStar(require("./approver-type"));
exports.ApproverType = ApproverType.ApproverType;
const ChangeApprovalTemplateCompartmentDetails = __importStar(require("./change-approval-template-compartment-details"));
exports.ChangeApprovalTemplateCompartmentDetails = ChangeApprovalTemplateCompartmentDetails.ChangeApprovalTemplateCompartmentDetails;
const ChangeLockboxCompartmentDetails = __importStar(require("./change-lockbox-compartment-details"));
exports.ChangeLockboxCompartmentDetails = ChangeLockboxCompartmentDetails.ChangeLockboxCompartmentDetails;
const CreateAccessRequestDetails = __importStar(require("./create-access-request-details"));
exports.CreateAccessRequestDetails = CreateAccessRequestDetails.CreateAccessRequestDetails;
const CreateApprovalTemplateDetails = __importStar(require("./create-approval-template-details"));
exports.CreateApprovalTemplateDetails = CreateApprovalTemplateDetails.CreateApprovalTemplateDetails;
const CreateLockboxDetails = __importStar(require("./create-lockbox-details"));
exports.CreateLockboxDetails = CreateLockboxDetails.CreateLockboxDetails;
const HandleAccessRequestDetails = __importStar(require("./handle-access-request-details"));
exports.HandleAccessRequestDetails = HandleAccessRequestDetails.HandleAccessRequestDetails;
const Lockbox = __importStar(require("./lockbox"));
exports.Lockbox = Lockbox.Lockbox;
const LockboxAutoApprovalState = __importStar(require("./lockbox-auto-approval-state"));
exports.LockboxAutoApprovalState = LockboxAutoApprovalState.LockboxAutoApprovalState;
const LockboxCollection = __importStar(require("./lockbox-collection"));
exports.LockboxCollection = LockboxCollection.LockboxCollection;
const LockboxPartner = __importStar(require("./lockbox-partner"));
exports.LockboxPartner = LockboxPartner.LockboxPartner;
const LockboxSummary = __importStar(require("./lockbox-summary"));
exports.LockboxSummary = LockboxSummary.LockboxSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PersonaLevel = __importStar(require("./persona-level"));
exports.PersonaLevel = PersonaLevel.PersonaLevel;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateApprovalTemplateDetails = __importStar(require("./update-approval-template-details"));
exports.UpdateApprovalTemplateDetails = UpdateApprovalTemplateDetails.UpdateApprovalTemplateDetails;
const UpdateLockboxDetails = __importStar(require("./update-lockbox-details"));
exports.UpdateLockboxDetails = UpdateLockboxDetails.UpdateLockboxDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
