// oci-lockbox/lib/model/operation-status.js
"use strict";
/**
 * Managed Access API
 * Use the Managed Access API to approve access requests, create and manage templates, and manage resource approval settings. For more information, see [Managed Access Overview](https://docs.oracle.com/en-us/iaas/managed-access/overview.htm).

Use the table of contents and search tool to explore the Managed Access API.

 * OpenAPI spec version: 20220126
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationStatus = void 0;
/**
 * Possible operation status.
 **/
var OperationStatus;
(function (OperationStatus) {
    OperationStatus["Accepted"] = "ACCEPTED";
    OperationStatus["InProgress"] = "IN_PROGRESS";
    OperationStatus["Waiting"] = "WAITING";
    OperationStatus["Failed"] = "FAILED";
    OperationStatus["Succeeded"] = "SUCCEEDED";
    OperationStatus["Canceling"] = "CANCELING";
    OperationStatus["Canceled"] = "CANCELED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
(function (OperationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
