// oci-lockbox/lib/model/operation-type.js
"use strict";
/**
 * Managed Access API
 * Use the Managed Access API to approve access requests, create and manage templates, and manage resource approval settings. For more information, see [Managed Access Overview](https://docs.oracle.com/en-us/iaas/managed-access/overview.htm).

Use the table of contents and search tool to explore the Managed Access API.

 * OpenAPI spec version: 20220126
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateLockbox"] = "CREATE_LOCKBOX";
    OperationType["UpdateLockbox"] = "UPDATE_LOCKBOX";
    OperationType["DeleteLockbox"] = "DELETE_LOCKBOX";
    OperationType["MoveLockbox"] = "MOVE_LOCKBOX";
    OperationType["CreateAccessRequest"] = "CREATE_ACCESS_REQUEST";
    OperationType["ApproveAccessRequest"] = "APPROVE_ACCESS_REQUEST";
    OperationType["RevokeAccessRequest"] = "REVOKE_ACCESS_REQUEST";
    OperationType["CreateApprovalTemplate"] = "CREATE_APPROVAL_TEMPLATE";
    OperationType["MoveApprovalTemplate"] = "MOVE_APPROVAL_TEMPLATE";
    OperationType["UpdateApprovalTemplate"] = "UPDATE_APPROVAL_TEMPLATE";
    OperationType["DeleteApprovalTemplate"] = "DELETE_APPROVAL_TEMPLATE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
