// oci-fusionapps/lib/request/index.js
"use strict";
/**
 * Fusion Applications Environment Management API
 * Use the Fusion Applications Environment Management API to manage the environments where your Fusion Applications run. For more information, see the [Fusion Applications Environment Management documentation](/iaas/Content/fusion-applications/home.htm).
 * OpenAPI spec version: 20211201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListTimeAvailableForRefreshesRequest = exports.ListServiceAttachmentsRequest = exports.ListScheduledActivitiesRequest = exports.ListRefreshActivitiesRequest = exports.ListFusionEnvironmentsRequest = exports.ListFusionEnvironmentFamiliesRequest = exports.ListDataMaskingActivitiesRequest = void 0;
const ListDataMaskingActivitiesRequest = __importStar(require("./list-data-masking-activities-request"));
exports.ListDataMaskingActivitiesRequest = ListDataMaskingActivitiesRequest.ListDataMaskingActivitiesRequest;
const ListFusionEnvironmentFamiliesRequest = __importStar(require("./list-fusion-environment-families-request"));
exports.ListFusionEnvironmentFamiliesRequest = ListFusionEnvironmentFamiliesRequest.ListFusionEnvironmentFamiliesRequest;
const ListFusionEnvironmentsRequest = __importStar(require("./list-fusion-environments-request"));
exports.ListFusionEnvironmentsRequest = ListFusionEnvironmentsRequest.ListFusionEnvironmentsRequest;
const ListRefreshActivitiesRequest = __importStar(require("./list-refresh-activities-request"));
exports.ListRefreshActivitiesRequest = ListRefreshActivitiesRequest.ListRefreshActivitiesRequest;
const ListScheduledActivitiesRequest = __importStar(require("./list-scheduled-activities-request"));
exports.ListScheduledActivitiesRequest = ListScheduledActivitiesRequest.ListScheduledActivitiesRequest;
const ListServiceAttachmentsRequest = __importStar(require("./list-service-attachments-request"));
exports.ListServiceAttachmentsRequest = ListServiceAttachmentsRequest.ListServiceAttachmentsRequest;
const ListTimeAvailableForRefreshesRequest = __importStar(require("./list-time-available-for-refreshes-request"));
exports.ListTimeAvailableForRefreshesRequest = ListTimeAvailableForRefreshesRequest.ListTimeAvailableForRefreshesRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
