// oci-usageapi/lib/model/dynamic-date-range.js
"use strict";
/**
 * Usage API
 * Use the Usage API to view your Oracle Cloud usage and costs. The API allows you to request data that meets the specified filter criteria, and to group that data by the dimension of your choosing. The Usage API is used by the Cost Analysis tool in the Console. Also see [Using the Usage API](/Content/Billing/Concepts/costanalysisoverview.htm#cost_analysis_using_the_api) for more information.
 * OpenAPI spec version: 20200107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DynamicDateRange = void 0;
const model = __importStar(require("../model"));
var DynamicDateRange;
(function (DynamicDateRange) {
    let DynamicDateRangeType;
    (function (DynamicDateRangeType) {
        DynamicDateRangeType["Last7Days"] = "LAST_7_DAYS";
        DynamicDateRangeType["Last10Days"] = "LAST_10_DAYS";
        DynamicDateRangeType["LastCalendarWeek"] = "LAST_CALENDAR_WEEK";
        DynamicDateRangeType["LastCalendarMonth"] = "LAST_CALENDAR_MONTH";
        DynamicDateRangeType["Last2CalendarMonths"] = "LAST_2_CALENDAR_MONTHS";
        DynamicDateRangeType["Last3CalendarMonths"] = "LAST_3_CALENDAR_MONTHS";
        DynamicDateRangeType["Last6CalendarMonths"] = "LAST_6_CALENDAR_MONTHS";
        DynamicDateRangeType["Last30Days"] = "LAST_30_DAYS";
        DynamicDateRangeType["MonthToDate"] = "MONTH_TO_DATE";
        DynamicDateRangeType["LastYear"] = "LAST_YEAR";
        DynamicDateRangeType["YearTodate"] = "YEAR_TODATE";
        DynamicDateRangeType["All"] = "ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        DynamicDateRangeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(DynamicDateRangeType = DynamicDateRange.DynamicDateRangeType || (DynamicDateRange.DynamicDateRangeType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.DateRange.getJsonObj(obj))), {});
        return jsonObj;
    }
    DynamicDateRange.getJsonObj = getJsonObj;
    DynamicDateRange.dateRangeType = "DYNAMIC";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.DateRange.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    DynamicDateRange.getDeserializedJsonObj = getDeserializedJsonObj;
})(DynamicDateRange = exports.DynamicDateRange || (exports.DynamicDateRange = {}));
