// oci-databasetools/lib/request/index.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkRequestsRequest = exports.ListWorkRequestLogsRequest = exports.ListWorkRequestErrorsRequest = exports.ListDatabaseToolsPrivateEndpointsRequest = exports.ListDatabaseToolsEndpointServicesRequest = exports.ListDatabaseToolsConnectionsRequest = void 0;
const ListDatabaseToolsConnectionsRequest = __importStar(require("./list-database-tools-connections-request"));
exports.ListDatabaseToolsConnectionsRequest = ListDatabaseToolsConnectionsRequest.ListDatabaseToolsConnectionsRequest;
const ListDatabaseToolsEndpointServicesRequest = __importStar(require("./list-database-tools-endpoint-services-request"));
exports.ListDatabaseToolsEndpointServicesRequest = ListDatabaseToolsEndpointServicesRequest.ListDatabaseToolsEndpointServicesRequest;
const ListDatabaseToolsPrivateEndpointsRequest = __importStar(require("./list-database-tools-private-endpoints-request"));
exports.ListDatabaseToolsPrivateEndpointsRequest = ListDatabaseToolsPrivateEndpointsRequest.ListDatabaseToolsPrivateEndpointsRequest;
const ListWorkRequestErrorsRequest = __importStar(require("./list-work-request-errors-request"));
exports.ListWorkRequestErrorsRequest = ListWorkRequestErrorsRequest.ListWorkRequestErrorsRequest;
const ListWorkRequestLogsRequest = __importStar(require("./list-work-request-logs-request"));
exports.ListWorkRequestLogsRequest = ListWorkRequestLogsRequest.ListWorkRequestLogsRequest;
const ListWorkRequestsRequest = __importStar(require("./list-work-requests-request"));
exports.ListWorkRequestsRequest = ListWorkRequestsRequest.ListWorkRequestsRequest;
