// oci-apmcontrolplane/lib/model/sort-orders.js
"use strict";
/**
 * Application Performance Monitoring Control Plane API
 * Use the Application Performance Monitoring Control Plane API to perform operations such as creating, updating,
deleting and listing APM domains and monitoring the progress of these operations using the work request APIs. For more information, see [Application Performance Monitoring](/iaas/application-performance-monitoring/index.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrders = void 0;
/**
 * Sort orders.
 **/
var SortOrders;
(function (SortOrders) {
    SortOrders["Asc"] = "ASC";
    SortOrders["Desc"] = "DESC";
})(SortOrders = exports.SortOrders || (exports.SortOrders = {}));
(function (SortOrders) {
    function getJsonObj(obj) {
        return obj;
    }
    SortOrders.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SortOrders.getDeserializedJsonObj = getDeserializedJsonObj;
})(SortOrders = exports.SortOrders || (exports.SortOrders = {}));
