// oci-oce/lib/model/lifecycle-details.js
"use strict";
/**
 * Oracle Content Management API
 * Oracle Content Management is a cloud-based content hub to drive omni-channel content management and accelerate experience delivery
 * OpenAPI spec version: 20190912
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleDetails = void 0;
/**
 * Instance lifecycle details
 **/
var LifecycleDetails;
(function (LifecycleDetails) {
    LifecycleDetails["Standby"] = "STANDBY";
    LifecycleDetails["Failover"] = "FAILOVER";
    LifecycleDetails["Down"] = "DOWN";
    LifecycleDetails["Primary"] = "PRIMARY";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LifecycleDetails["UnknownValue"] = "UNKNOWN_VALUE";
})(LifecycleDetails = exports.LifecycleDetails || (exports.LifecycleDetails = {}));
(function (LifecycleDetails) {
    function getJsonObj(obj) {
        return obj;
    }
    LifecycleDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LifecycleDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(LifecycleDetails = exports.LifecycleDetails || (exports.LifecycleDetails = {}));
