// oci-dashboardservice/lib/model/index.js
"use strict";
/**
 * Dashboards API
 * Use the Oracle Cloud Infrastructure Dashboards service API to manage dashboards in the Console.
Dashboards provide an organized and customizable view of resources and their metrics in the Console.
For more information, see [Dashboards](/Content/Dashboards/home.htm).

**Important:** Resources for the Dashboards service are created in the tenacy's home region.
Although it is possible to create dashboard and dashboard group resources in regions other than the home region,
you won't be able to view those resources in the Console.
Therefore, creating resources outside of the home region is not recommended.

 * OpenAPI spec version: 20210731
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.V1Dashboard = exports.UpdateV1DashboardDetails = exports.CreateV1DashboardDetails = exports.UpdateDashboardGroupDetails = exports.UpdateDashboardDetails = exports.SortOrder = exports.DashboardSummary = exports.DashboardGroupSummary = exports.DashboardGroupCollection = exports.DashboardGroup = exports.DashboardCollection = exports.Dashboard = exports.CreateDashboardGroupDetails = exports.CreateDashboardDetails = exports.ChangeDashboardGroupDetails = exports.ChangeDashboardGroupCompartmentDetails = void 0;
const ChangeDashboardGroupCompartmentDetails = __importStar(require("./change-dashboard-group-compartment-details"));
exports.ChangeDashboardGroupCompartmentDetails = ChangeDashboardGroupCompartmentDetails.ChangeDashboardGroupCompartmentDetails;
const ChangeDashboardGroupDetails = __importStar(require("./change-dashboard-group-details"));
exports.ChangeDashboardGroupDetails = ChangeDashboardGroupDetails.ChangeDashboardGroupDetails;
const CreateDashboardDetails = __importStar(require("./create-dashboard-details"));
exports.CreateDashboardDetails = CreateDashboardDetails.CreateDashboardDetails;
const CreateDashboardGroupDetails = __importStar(require("./create-dashboard-group-details"));
exports.CreateDashboardGroupDetails = CreateDashboardGroupDetails.CreateDashboardGroupDetails;
const Dashboard = __importStar(require("./dashboard"));
exports.Dashboard = Dashboard.Dashboard;
const DashboardCollection = __importStar(require("./dashboard-collection"));
exports.DashboardCollection = DashboardCollection.DashboardCollection;
const DashboardGroup = __importStar(require("./dashboard-group"));
exports.DashboardGroup = DashboardGroup.DashboardGroup;
const DashboardGroupCollection = __importStar(require("./dashboard-group-collection"));
exports.DashboardGroupCollection = DashboardGroupCollection.DashboardGroupCollection;
const DashboardGroupSummary = __importStar(require("./dashboard-group-summary"));
exports.DashboardGroupSummary = DashboardGroupSummary.DashboardGroupSummary;
const DashboardSummary = __importStar(require("./dashboard-summary"));
exports.DashboardSummary = DashboardSummary.DashboardSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateDashboardDetails = __importStar(require("./update-dashboard-details"));
exports.UpdateDashboardDetails = UpdateDashboardDetails.UpdateDashboardDetails;
const UpdateDashboardGroupDetails = __importStar(require("./update-dashboard-group-details"));
exports.UpdateDashboardGroupDetails = UpdateDashboardGroupDetails.UpdateDashboardGroupDetails;
const CreateV1DashboardDetails = __importStar(require("./create-v1-dashboard-details"));
exports.CreateV1DashboardDetails = CreateV1DashboardDetails.CreateV1DashboardDetails;
const UpdateV1DashboardDetails = __importStar(require("./update-v1-dashboard-details"));
exports.UpdateV1DashboardDetails = UpdateV1DashboardDetails.UpdateV1DashboardDetails;
const V1Dashboard = __importStar(require("./v1-dashboard"));
exports.V1Dashboard = V1Dashboard.V1Dashboard;
