// oci-identitydomains/lib/model/extension-mfa-user.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm). This REST API is SCIM compliant.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtensionMfaUser = void 0;
const model = __importStar(require("../model"));
var ExtensionMfaUser;
(function (ExtensionMfaUser) {
    let PreferredAuthenticationFactor;
    (function (PreferredAuthenticationFactor) {
        PreferredAuthenticationFactor["Email"] = "EMAIL";
        PreferredAuthenticationFactor["Sms"] = "SMS";
        PreferredAuthenticationFactor["Totp"] = "TOTP";
        PreferredAuthenticationFactor["Push"] = "PUSH";
        PreferredAuthenticationFactor["Offlinetotp"] = "OFFLINETOTP";
        PreferredAuthenticationFactor["UsernamePassword"] = "USERNAME_PASSWORD";
        PreferredAuthenticationFactor["SecurityQuestions"] = "SECURITY_QUESTIONS";
        PreferredAuthenticationFactor["Voice"] = "VOICE";
        PreferredAuthenticationFactor["PhoneCall"] = "PHONE_CALL";
        PreferredAuthenticationFactor["Thirdparty"] = "THIRDPARTY";
        PreferredAuthenticationFactor["FidoAuthenticator"] = "FIDO_AUTHENTICATOR";
        PreferredAuthenticationFactor["YubicoOtp"] = "YUBICO_OTP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PreferredAuthenticationFactor["UnknownValue"] = "UNKNOWN_VALUE";
    })(PreferredAuthenticationFactor = ExtensionMfaUser.PreferredAuthenticationFactor || (ExtensionMfaUser.PreferredAuthenticationFactor = {}));
    let MfaStatus;
    (function (MfaStatus) {
        MfaStatus["Enrolled"] = "ENROLLED";
        MfaStatus["Ignored"] = "IGNORED";
        MfaStatus["UnEnrolled"] = "UN_ENROLLED";
        MfaStatus["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MfaStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(MfaStatus = ExtensionMfaUser.MfaStatus || (ExtensionMfaUser.MfaStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "preferredDevice": obj.preferredDevice
                ? model.UserExtPreferredDevice.getJsonObj(obj.preferredDevice)
                : undefined,
            "devices": obj.devices
                ? obj.devices.map(item => {
                    return model.UserExtDevices.getJsonObj(item);
                })
                : undefined,
            "bypassCodes": obj.bypassCodes
                ? obj.bypassCodes.map(item => {
                    return model.UserExtBypassCodes.getJsonObj(item);
                })
                : undefined,
            "trustedUserAgents": obj.trustedUserAgents
                ? obj.trustedUserAgents.map(item => {
                    return model.UserExtTrustedUserAgents.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ExtensionMfaUser.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "preferredDevice": obj.preferredDevice
                ? model.UserExtPreferredDevice.getDeserializedJsonObj(obj.preferredDevice)
                : undefined,
            "devices": obj.devices
                ? obj.devices.map(item => {
                    return model.UserExtDevices.getDeserializedJsonObj(item);
                })
                : undefined,
            "bypassCodes": obj.bypassCodes
                ? obj.bypassCodes.map(item => {
                    return model.UserExtBypassCodes.getDeserializedJsonObj(item);
                })
                : undefined,
            "trustedUserAgents": obj.trustedUserAgents
                ? obj.trustedUserAgents.map(item => {
                    return model.UserExtTrustedUserAgents.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    ExtensionMfaUser.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExtensionMfaUser = exports.ExtensionMfaUser || (exports.ExtensionMfaUser = {}));
