// oci-identitydomains/lib/model/user.js
"use strict";
/**
 * Identity Domains API
 * Use the Identity Domains API to manage resources within an identity domain, for example, users, dynamic resource groups, groups, and identity providers. For information about managing resources within identity domains, see [Identity and Access Management (with identity domains)](https://docs.oracle.com/iaas/Content/Identity/home.htm). This REST API is SCIM compliant.
Use the table of contents and search tool to explore the Identity Domains API.
 * OpenAPI spec version: v1
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = void 0;
const model = __importStar(require("../model"));
var User;
(function (User) {
    let IdcsPreventedOperations;
    (function (IdcsPreventedOperations) {
        IdcsPreventedOperations["Replace"] = "replace";
        IdcsPreventedOperations["Update"] = "update";
        IdcsPreventedOperations["Delete"] = "delete";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        IdcsPreventedOperations["UnknownValue"] = "UNKNOWN_VALUE";
    })(IdcsPreventedOperations = User.IdcsPreventedOperations || (User.IdcsPreventedOperations = {}));
    let UserType;
    (function (UserType) {
        UserType["Contractor"] = "Contractor";
        UserType["Employee"] = "Employee";
        UserType["Intern"] = "Intern";
        UserType["Temp"] = "Temp";
        UserType["External"] = "External";
        UserType["Service"] = "Service";
        UserType["Generic"] = "Generic";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UserType["UnknownValue"] = "UNKNOWN_VALUE";
    })(UserType = User.UserType || (User.UserType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getJsonObj(item);
                })
                : undefined,
            "name": obj.name ? model.UserName.getJsonObj(obj.name) : undefined,
            "emails": obj.emails
                ? obj.emails.map(item => {
                    return model.UserEmails.getJsonObj(item);
                })
                : undefined,
            "phoneNumbers": obj.phoneNumbers
                ? obj.phoneNumbers.map(item => {
                    return model.UserPhoneNumbers.getJsonObj(item);
                })
                : undefined,
            "ims": obj.ims
                ? obj.ims.map(item => {
                    return model.UserIms.getJsonObj(item);
                })
                : undefined,
            "photos": obj.photos
                ? obj.photos.map(item => {
                    return model.UserPhotos.getJsonObj(item);
                })
                : undefined,
            "addresses": obj.addresses
                ? obj.addresses.map(item => {
                    return model.Addresses.getJsonObj(item);
                })
                : undefined,
            "groups": obj.groups
                ? obj.groups.map(item => {
                    return model.UserGroups.getJsonObj(item);
                })
                : undefined,
            "entitlements": obj.entitlements
                ? obj.entitlements.map(item => {
                    return model.UserEntitlements.getJsonObj(item);
                })
                : undefined,
            "roles": obj.roles
                ? obj.roles.map(item => {
                    return model.UserRoles.getJsonObj(item);
                })
                : undefined,
            "x509Certificates": obj.x509Certificates
                ? obj.x509Certificates.map(item => {
                    return model.UserX509Certificates.getJsonObj(item);
                })
                : undefined,
            "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User": obj.urnIetfParamsScimSchemasExtensionEnterprise2_0User
                ? model.ExtensionEnterprise20User.getJsonObj(obj.urnIetfParamsScimSchemasExtensionEnterprise2_0User)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:user:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserUser
                ? model.ExtensionUserUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordState:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordStateUser
                ? model.ExtensionPasswordStateUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordStateUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:userState:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserStateUser
                ? model.ExtensionUserStateUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserStateUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:posix:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionPosixUser
                ? model.ExtensionPosixUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionPosixUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosUser:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosUserUser
                ? model.ExtensionKerberosUserUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosUserUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:mfa:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionMfaUser
                ? model.ExtensionMfaUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionMfaUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:adaptive:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionAdaptiveUser
                ? model.ExtensionAdaptiveUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionAdaptiveUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:sff:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSffUser
                ? model.ExtensionSffUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSffUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:securityQuestions:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSecurityQuestionsUser
                ? model.ExtensionSecurityQuestionsUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSecurityQuestionsUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:selfRegistration:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSelfRegistrationUser
                ? model.ExtensionSelfRegistrationUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSelfRegistrationUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:socialAccount:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSocialAccountUser
                ? model.ExtensionSocialAccountUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSocialAccountUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:dbUser:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbUserUser
                ? model.ExtensionDbUserUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbUserUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:termsOfUse:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionTermsOfUseUser
                ? model.ExtensionTermsOfUseUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionTermsOfUseUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordless:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordlessUser
                ? model.ExtensionPasswordlessUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordlessUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags": obj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags
                ? model.ExtensionOCITags.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:userCredentials:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserCredentialsUser
                ? model.ExtensionUserCredentialsUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionUserCredentialsUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:capabilities:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionCapabilitiesUser
                ? model.ExtensionCapabilitiesUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionCapabilitiesUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:dbCredentials:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbCredentialsUser
                ? model.ExtensionDbCredentialsUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionDbCredentialsUser)
                : undefined,
            "urn:ietf:params:scim:schemas:oracle:idcs:extension:selfChange:User": obj.urnIetfParamsScimSchemasOracleIdcsExtensionSelfChangeUser
                ? model.ExtensionSelfChangeUser.getJsonObj(obj.urnIetfParamsScimSchemasOracleIdcsExtensionSelfChangeUser)
                : undefined
        });
        delete jsonObj.urnIetfParamsScimSchemasExtensionEnterprise2_0User;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionUserUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordStateUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionUserStateUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionPosixUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionKerberosUserUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionMfaUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionAdaptiveUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionSffUser;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionSecurityQuestionsUser;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionSelfRegistrationUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionSocialAccountUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionDbUserUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionTermsOfUseUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionPasswordlessUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionOciTags;
        delete jsonObj
            .urnIetfParamsScimSchemasOracleIdcsExtensionUserCredentialsUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionCapabilitiesUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionDbCredentialsUser;
        delete jsonObj.urnIetfParamsScimSchemasOracleIdcsExtensionSelfChangeUser;
        return jsonObj;
    }
    User.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "meta": obj.meta ? model.Meta.getDeserializedJsonObj(obj.meta) : undefined,
            "idcsCreatedBy": obj.idcsCreatedBy
                ? model.IdcsCreatedBy.getDeserializedJsonObj(obj.idcsCreatedBy)
                : undefined,
            "idcsLastModifiedBy": obj.idcsLastModifiedBy
                ? model.IdcsLastModifiedBy.getDeserializedJsonObj(obj.idcsLastModifiedBy)
                : undefined,
            "tags": obj.tags
                ? obj.tags.map(item => {
                    return model.Tags.getDeserializedJsonObj(item);
                })
                : undefined,
            "name": obj.name ? model.UserName.getDeserializedJsonObj(obj.name) : undefined,
            "emails": obj.emails
                ? obj.emails.map(item => {
                    return model.UserEmails.getDeserializedJsonObj(item);
                })
                : undefined,
            "phoneNumbers": obj.phoneNumbers
                ? obj.phoneNumbers.map(item => {
                    return model.UserPhoneNumbers.getDeserializedJsonObj(item);
                })
                : undefined,
            "ims": obj.ims
                ? obj.ims.map(item => {
                    return model.UserIms.getDeserializedJsonObj(item);
                })
                : undefined,
            "photos": obj.photos
                ? obj.photos.map(item => {
                    return model.UserPhotos.getDeserializedJsonObj(item);
                })
                : undefined,
            "addresses": obj.addresses
                ? obj.addresses.map(item => {
                    return model.Addresses.getDeserializedJsonObj(item);
                })
                : undefined,
            "groups": obj.groups
                ? obj.groups.map(item => {
                    return model.UserGroups.getDeserializedJsonObj(item);
                })
                : undefined,
            "entitlements": obj.entitlements
                ? obj.entitlements.map(item => {
                    return model.UserEntitlements.getDeserializedJsonObj(item);
                })
                : undefined,
            "roles": obj.roles
                ? obj.roles.map(item => {
                    return model.UserRoles.getDeserializedJsonObj(item);
                })
                : undefined,
            "x509Certificates": obj.x509Certificates
                ? obj.x509Certificates.map(item => {
                    return model.UserX509Certificates.getDeserializedJsonObj(item);
                })
                : undefined,
            "urnIetfParamsScimSchemasExtensionEnterprise2_0User": obj["urn:ietf:params:scim:schemas:extension:enterprise:2.0:User"]
                ? model.ExtensionEnterprise20User.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:extension:enterprise:2.0:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionUserUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:user:User"]
                ? model.ExtensionUserUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:user:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionPasswordStateUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordState:User"]
                ? model.ExtensionPasswordStateUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordState:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionUserStateUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userState:User"]
                ? model.ExtensionUserStateUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userState:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionPosixUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:posix:User"]
                ? model.ExtensionPosixUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:posix:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionKerberosUserUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosUser:User"]
                ? model.ExtensionKerberosUserUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosUser:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionMfaUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:mfa:User"]
                ? model.ExtensionMfaUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:mfa:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionAdaptiveUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:adaptive:User"]
                ? model.ExtensionAdaptiveUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:adaptive:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSffUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:sff:User"]
                ? model.ExtensionSffUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:sff:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSecurityQuestionsUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:securityQuestions:User"]
                ? model.ExtensionSecurityQuestionsUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:securityQuestions:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSelfRegistrationUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfRegistration:User"]
                ? model.ExtensionSelfRegistrationUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfRegistration:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSocialAccountUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:socialAccount:User"]
                ? model.ExtensionSocialAccountUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:socialAccount:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionDbUserUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbUser:User"]
                ? model.ExtensionDbUserUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbUser:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionTermsOfUseUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:termsOfUse:User"]
                ? model.ExtensionTermsOfUseUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:termsOfUse:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionPasswordlessUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordless:User"]
                ? model.ExtensionPasswordlessUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordless:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionOciTags": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"]
                ? model.ExtensionOCITags.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionUserCredentialsUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userCredentials:User"]
                ? model.ExtensionUserCredentialsUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userCredentials:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionCapabilitiesUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:capabilities:User"]
                ? model.ExtensionCapabilitiesUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:capabilities:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionDbCredentialsUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbCredentials:User"]
                ? model.ExtensionDbCredentialsUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbCredentials:User"])
                : undefined,
            "urnIetfParamsScimSchemasOracleIdcsExtensionSelfChangeUser": obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfChange:User"]
                ? model.ExtensionSelfChangeUser.getDeserializedJsonObj(obj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfChange:User"])
                : undefined
        });
        delete jsonObj["urn:ietf:params:scim:schemas:extension:enterprise:2.0:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:user:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordState:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userState:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:posix:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:kerberosUser:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:mfa:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:adaptive:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:sff:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:securityQuestions:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfRegistration:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:socialAccount:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbUser:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:termsOfUse:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:passwordless:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:OCITags"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:userCredentials:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:capabilities:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:dbCredentials:User"];
        delete jsonObj["urn:ietf:params:scim:schemas:oracle:idcs:extension:selfChange:User"];
        return jsonObj;
    }
    User.getDeserializedJsonObj = getDeserializedJsonObj;
})(User = exports.User || (exports.User = {}));
