// oci-emwarehouse/lib/model/index.js
"use strict";
/**
 * EM Warehouse API
 * Use the EM Warehouse API to manage EM Warehouse data collection.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateEmWarehouseDetails = exports.SortOrder = exports.ResourceUsage = exports.OperationType = exports.OperationStatus = exports.EtlRunSummary = exports.EtlRunCollection = exports.EmWarehouseSummary = exports.EmWarehouseCollection = exports.EmWarehouse = exports.EmInstancesDetails = exports.CreateEmWarehouseDetails = exports.ChangeEmWarehouseCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeEmWarehouseCompartmentDetails = __importStar(require("./change-em-warehouse-compartment-details"));
exports.ChangeEmWarehouseCompartmentDetails = ChangeEmWarehouseCompartmentDetails.ChangeEmWarehouseCompartmentDetails;
const CreateEmWarehouseDetails = __importStar(require("./create-em-warehouse-details"));
exports.CreateEmWarehouseDetails = CreateEmWarehouseDetails.CreateEmWarehouseDetails;
const EmInstancesDetails = __importStar(require("./em-instances-details"));
exports.EmInstancesDetails = EmInstancesDetails.EmInstancesDetails;
const EmWarehouse = __importStar(require("./em-warehouse"));
exports.EmWarehouse = EmWarehouse.EmWarehouse;
const EmWarehouseCollection = __importStar(require("./em-warehouse-collection"));
exports.EmWarehouseCollection = EmWarehouseCollection.EmWarehouseCollection;
const EmWarehouseSummary = __importStar(require("./em-warehouse-summary"));
exports.EmWarehouseSummary = EmWarehouseSummary.EmWarehouseSummary;
const EtlRunCollection = __importStar(require("./etl-run-collection"));
exports.EtlRunCollection = EtlRunCollection.EtlRunCollection;
const EtlRunSummary = __importStar(require("./etl-run-summary"));
exports.EtlRunSummary = EtlRunSummary.EtlRunSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ResourceUsage = __importStar(require("./resource-usage"));
exports.ResourceUsage = ResourceUsage.ResourceUsage;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateEmWarehouseDetails = __importStar(require("./update-em-warehouse-details"));
exports.UpdateEmWarehouseDetails = UpdateEmWarehouseDetails.UpdateEmWarehouseDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
