// oci-bastion/lib/model/index.js
"use strict";
/**
 * Bastion API
 * Use the Bastion API to provide restricted and time-limited access to target resources that don't have public endpoints. Bastions let authorized users connect from specific IP addresses to target resources using Secure Shell (SSH) sessions. For more information, see [the Bastion documentation](/iaas/Content/Bastion/home.htm).
 * OpenAPI spec version: 20210331
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PortForwardingSessionTargetResourceDetails = exports.ManagedSshSessionTargetResourceDetails = exports.DynamicPortForwardingSessionTargetResourceDetails = exports.CreatePortForwardingSessionTargetResourceDetails = exports.CreateManagedSshSessionTargetResourceDetails = exports.CreateDynamicPortForwardingSessionTargetResourceDetails = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntry = exports.WorkRequestError = exports.WorkRequest = exports.UpdateSessionDetails = exports.UpdateBastionDetails = exports.TargetResourceDetails = exports.SortOrder = exports.SessionType = exports.SessionSummary = exports.SessionLifecycleState = exports.Session = exports.PublicKeyDetails = exports.OperationType = exports.OperationStatus = exports.CreateSessionTargetResourceDetails = exports.CreateSessionDetails = exports.CreateBastionDetails = exports.ChangeBastionCompartmentDetails = exports.BastionSummary = exports.BastionLifecycleState = exports.BastionDnsProxyStatus = exports.Bastion = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const Bastion = __importStar(require("./bastion"));
exports.Bastion = Bastion.Bastion;
const BastionDnsProxyStatus = __importStar(require("./bastion-dns-proxy-status"));
exports.BastionDnsProxyStatus = BastionDnsProxyStatus.BastionDnsProxyStatus;
const BastionLifecycleState = __importStar(require("./bastion-lifecycle-state"));
exports.BastionLifecycleState = BastionLifecycleState.BastionLifecycleState;
const BastionSummary = __importStar(require("./bastion-summary"));
exports.BastionSummary = BastionSummary.BastionSummary;
const ChangeBastionCompartmentDetails = __importStar(require("./change-bastion-compartment-details"));
exports.ChangeBastionCompartmentDetails = ChangeBastionCompartmentDetails.ChangeBastionCompartmentDetails;
const CreateBastionDetails = __importStar(require("./create-bastion-details"));
exports.CreateBastionDetails = CreateBastionDetails.CreateBastionDetails;
const CreateSessionDetails = __importStar(require("./create-session-details"));
exports.CreateSessionDetails = CreateSessionDetails.CreateSessionDetails;
const CreateSessionTargetResourceDetails = __importStar(require("./create-session-target-resource-details"));
exports.CreateSessionTargetResourceDetails = CreateSessionTargetResourceDetails.CreateSessionTargetResourceDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PublicKeyDetails = __importStar(require("./public-key-details"));
exports.PublicKeyDetails = PublicKeyDetails.PublicKeyDetails;
const Session = __importStar(require("./session"));
exports.Session = Session.Session;
const SessionLifecycleState = __importStar(require("./session-lifecycle-state"));
exports.SessionLifecycleState = SessionLifecycleState.SessionLifecycleState;
const SessionSummary = __importStar(require("./session-summary"));
exports.SessionSummary = SessionSummary.SessionSummary;
const SessionType = __importStar(require("./session-type"));
exports.SessionType = SessionType.SessionType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TargetResourceDetails = __importStar(require("./target-resource-details"));
exports.TargetResourceDetails = TargetResourceDetails.TargetResourceDetails;
const UpdateBastionDetails = __importStar(require("./update-bastion-details"));
exports.UpdateBastionDetails = UpdateBastionDetails.UpdateBastionDetails;
const UpdateSessionDetails = __importStar(require("./update-session-details"));
exports.UpdateSessionDetails = UpdateSessionDetails.UpdateSessionDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const CreateDynamicPortForwardingSessionTargetResourceDetails = __importStar(require("./create-dynamic-port-forwarding-session-target-resource-details"));
exports.CreateDynamicPortForwardingSessionTargetResourceDetails = CreateDynamicPortForwardingSessionTargetResourceDetails.CreateDynamicPortForwardingSessionTargetResourceDetails;
const CreateManagedSshSessionTargetResourceDetails = __importStar(require("./create-managed-ssh-session-target-resource-details"));
exports.CreateManagedSshSessionTargetResourceDetails = CreateManagedSshSessionTargetResourceDetails.CreateManagedSshSessionTargetResourceDetails;
const CreatePortForwardingSessionTargetResourceDetails = __importStar(require("./create-port-forwarding-session-target-resource-details"));
exports.CreatePortForwardingSessionTargetResourceDetails = CreatePortForwardingSessionTargetResourceDetails.CreatePortForwardingSessionTargetResourceDetails;
const DynamicPortForwardingSessionTargetResourceDetails = __importStar(require("./dynamic-port-forwarding-session-target-resource-details"));
exports.DynamicPortForwardingSessionTargetResourceDetails = DynamicPortForwardingSessionTargetResourceDetails.DynamicPortForwardingSessionTargetResourceDetails;
const ManagedSshSessionTargetResourceDetails = __importStar(require("./managed-ssh-session-target-resource-details"));
exports.ManagedSshSessionTargetResourceDetails = ManagedSshSessionTargetResourceDetails.ManagedSshSessionTargetResourceDetails;
const PortForwardingSessionTargetResourceDetails = __importStar(require("./port-forwarding-session-target-resource-details"));
exports.PortForwardingSessionTargetResourceDetails = PortForwardingSessionTargetResourceDetails.PortForwardingSessionTargetResourceDetails;
