// oci-certificatesmanagement/lib/model/ca-bundle-lifecycle-state.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaBundleLifecycleState = void 0;
/**
 * The current lifecycle state of the CA bundle.
 **/
var CaBundleLifecycleState;
(function (CaBundleLifecycleState) {
    CaBundleLifecycleState["Creating"] = "CREATING";
    CaBundleLifecycleState["Active"] = "ACTIVE";
    CaBundleLifecycleState["Updating"] = "UPDATING";
    CaBundleLifecycleState["Deleting"] = "DELETING";
    CaBundleLifecycleState["Deleted"] = "DELETED";
    CaBundleLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CaBundleLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(CaBundleLifecycleState = exports.CaBundleLifecycleState || (exports.CaBundleLifecycleState = {}));
(function (CaBundleLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    CaBundleLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CaBundleLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(CaBundleLifecycleState = exports.CaBundleLifecycleState || (exports.CaBundleLifecycleState = {}));
