// oci-certificatesmanagement/lib/model/certificate-lifecycle-state.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateLifecycleState = void 0;
/**
 * The current lifecycle state of the certificate.
 **/
var CertificateLifecycleState;
(function (CertificateLifecycleState) {
    CertificateLifecycleState["Creating"] = "CREATING";
    CertificateLifecycleState["Active"] = "ACTIVE";
    CertificateLifecycleState["Updating"] = "UPDATING";
    CertificateLifecycleState["Deleting"] = "DELETING";
    CertificateLifecycleState["Deleted"] = "DELETED";
    CertificateLifecycleState["SchedulingDeletion"] = "SCHEDULING_DELETION";
    CertificateLifecycleState["PendingDeletion"] = "PENDING_DELETION";
    CertificateLifecycleState["CancellingDeletion"] = "CANCELLING_DELETION";
    CertificateLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertificateLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(CertificateLifecycleState = exports.CertificateLifecycleState || (exports.CertificateLifecycleState = {}));
(function (CertificateLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    CertificateLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertificateLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertificateLifecycleState = exports.CertificateLifecycleState || (exports.CertificateLifecycleState = {}));
