// oci-certificatesmanagement/lib/model/certificate-authority-config-type.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateAuthorityConfigType = void 0;
/**
 * The manner in which the certificate authority (CA) was created, whether it is a root CA generated internally by the service or a subordinate CA issued by another CA that was itself generated internally by the service.
 *
 **/
var CertificateAuthorityConfigType;
(function (CertificateAuthorityConfigType) {
    CertificateAuthorityConfigType["RootCaGeneratedInternally"] = "ROOT_CA_GENERATED_INTERNALLY";
    CertificateAuthorityConfigType["SubordinateCaIssuedByInternalCa"] = "SUBORDINATE_CA_ISSUED_BY_INTERNAL_CA";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertificateAuthorityConfigType["UnknownValue"] = "UNKNOWN_VALUE";
})(CertificateAuthorityConfigType = exports.CertificateAuthorityConfigType || (exports.CertificateAuthorityConfigType = {}));
(function (CertificateAuthorityConfigType) {
    function getJsonObj(obj) {
        return obj;
    }
    CertificateAuthorityConfigType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertificateAuthorityConfigType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertificateAuthorityConfigType = exports.CertificateAuthorityConfigType || (exports.CertificateAuthorityConfigType = {}));
