// oci-certificatesmanagement/lib/model/certificate-profile-type.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateProfileType = void 0;
/**
 * The name of the profile used to create the certificate, which depends on the type of certificate you need.
 **/
var CertificateProfileType;
(function (CertificateProfileType) {
    CertificateProfileType["TlsServerOrClient"] = "TLS_SERVER_OR_CLIENT";
    CertificateProfileType["TlsServer"] = "TLS_SERVER";
    CertificateProfileType["TlsClient"] = "TLS_CLIENT";
    CertificateProfileType["TlsCodeSign"] = "TLS_CODE_SIGN";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CertificateProfileType["UnknownValue"] = "UNKNOWN_VALUE";
})(CertificateProfileType = exports.CertificateProfileType || (exports.CertificateProfileType = {}));
(function (CertificateProfileType) {
    function getJsonObj(obj) {
        return obj;
    }
    CertificateProfileType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CertificateProfileType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertificateProfileType = exports.CertificateProfileType || (exports.CertificateProfileType = {}));
