// oci-certificatesmanagement/lib/model/certificate-subject-alternative-name.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateSubjectAlternativeName = void 0;
var CertificateSubjectAlternativeName;
(function (CertificateSubjectAlternativeName) {
    let Type;
    (function (Type) {
        Type["Dns"] = "DNS";
        Type["Ip"] = "IP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = CertificateSubjectAlternativeName.Type || (CertificateSubjectAlternativeName.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CertificateSubjectAlternativeName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CertificateSubjectAlternativeName.getDeserializedJsonObj = getDeserializedJsonObj;
})(CertificateSubjectAlternativeName = exports.CertificateSubjectAlternativeName || (exports.CertificateSubjectAlternativeName = {}));
