// oci-certificatesmanagement/lib/model/key-algorithm.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyAlgorithm = void 0;
/**
 * The algorithm used to create key pairs.
 **/
var KeyAlgorithm;
(function (KeyAlgorithm) {
    KeyAlgorithm["Rsa2048"] = "RSA2048";
    KeyAlgorithm["Rsa4096"] = "RSA4096";
    KeyAlgorithm["EcdsaP256"] = "ECDSA_P256";
    KeyAlgorithm["EcdsaP384"] = "ECDSA_P384";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    KeyAlgorithm["UnknownValue"] = "UNKNOWN_VALUE";
})(KeyAlgorithm = exports.KeyAlgorithm || (exports.KeyAlgorithm = {}));
(function (KeyAlgorithm) {
    function getJsonObj(obj) {
        return obj;
    }
    KeyAlgorithm.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    KeyAlgorithm.getDeserializedJsonObj = getDeserializedJsonObj;
})(KeyAlgorithm = exports.KeyAlgorithm || (exports.KeyAlgorithm = {}));
