// oci-certificatesmanagement/lib/model/revocation-reason.js
"use strict";
/**
 * Certificates Service Management API
 * API for managing certificates.
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevocationReason = void 0;
/**
 * The current reason for the certificate's revocation.
 **/
var RevocationReason;
(function (RevocationReason) {
    RevocationReason["Unspecified"] = "UNSPECIFIED";
    RevocationReason["KeyCompromise"] = "KEY_COMPROMISE";
    RevocationReason["CaCompromise"] = "CA_COMPROMISE";
    RevocationReason["AffiliationChanged"] = "AFFILIATION_CHANGED";
    RevocationReason["Superseded"] = "SUPERSEDED";
    RevocationReason["CessationOfOperation"] = "CESSATION_OF_OPERATION";
    RevocationReason["PrivilegeWithdrawn"] = "PRIVILEGE_WITHDRAWN";
    RevocationReason["AaCompromise"] = "AA_COMPROMISE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RevocationReason["UnknownValue"] = "UNKNOWN_VALUE";
})(RevocationReason = exports.RevocationReason || (exports.RevocationReason = {}));
(function (RevocationReason) {
    function getJsonObj(obj) {
        return obj;
    }
    RevocationReason.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RevocationReason.getDeserializedJsonObj = getDeserializedJsonObj;
})(RevocationReason = exports.RevocationReason || (exports.RevocationReason = {}));
