// oci-queue/lib/model/index.js
"use strict";
/**
 * Queue API
 * A description of the Queue API
 * OpenAPI spec version: 20210201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdatedMessage = exports.UpdateQueueDetails = exports.UpdateMessagesResultEntry = exports.UpdateMessagesResult = exports.UpdateMessagesDetailsEntry = exports.UpdateMessagesDetails = exports.UpdateMessageDetails = exports.Stats = exports.SortOrder = exports.QueueSummary = exports.QueueStats = exports.QueueCollection = exports.Queue = exports.PutMessagesDetailsEntry = exports.PutMessagesDetails = exports.PutMessages = exports.PutMessage = exports.PurgeQueueDetails = exports.OperationType = exports.OperationStatus = exports.GetMessages = exports.GetMessage = exports.DeleteMessagesResultEntry = exports.DeleteMessagesResult = exports.DeleteMessagesDetailsEntry = exports.DeleteMessagesDetails = exports.CreateQueueDetails = exports.ChangeQueueCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeQueueCompartmentDetails = __importStar(require("./change-queue-compartment-details"));
exports.ChangeQueueCompartmentDetails = ChangeQueueCompartmentDetails.ChangeQueueCompartmentDetails;
const CreateQueueDetails = __importStar(require("./create-queue-details"));
exports.CreateQueueDetails = CreateQueueDetails.CreateQueueDetails;
const DeleteMessagesDetails = __importStar(require("./delete-messages-details"));
exports.DeleteMessagesDetails = DeleteMessagesDetails.DeleteMessagesDetails;
const DeleteMessagesDetailsEntry = __importStar(require("./delete-messages-details-entry"));
exports.DeleteMessagesDetailsEntry = DeleteMessagesDetailsEntry.DeleteMessagesDetailsEntry;
const DeleteMessagesResult = __importStar(require("./delete-messages-result"));
exports.DeleteMessagesResult = DeleteMessagesResult.DeleteMessagesResult;
const DeleteMessagesResultEntry = __importStar(require("./delete-messages-result-entry"));
exports.DeleteMessagesResultEntry = DeleteMessagesResultEntry.DeleteMessagesResultEntry;
const GetMessage = __importStar(require("./get-message"));
exports.GetMessage = GetMessage.GetMessage;
const GetMessages = __importStar(require("./get-messages"));
exports.GetMessages = GetMessages.GetMessages;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PurgeQueueDetails = __importStar(require("./purge-queue-details"));
exports.PurgeQueueDetails = PurgeQueueDetails.PurgeQueueDetails;
const PutMessage = __importStar(require("./put-message"));
exports.PutMessage = PutMessage.PutMessage;
const PutMessages = __importStar(require("./put-messages"));
exports.PutMessages = PutMessages.PutMessages;
const PutMessagesDetails = __importStar(require("./put-messages-details"));
exports.PutMessagesDetails = PutMessagesDetails.PutMessagesDetails;
const PutMessagesDetailsEntry = __importStar(require("./put-messages-details-entry"));
exports.PutMessagesDetailsEntry = PutMessagesDetailsEntry.PutMessagesDetailsEntry;
const Queue = __importStar(require("./queue"));
exports.Queue = Queue.Queue;
const QueueCollection = __importStar(require("./queue-collection"));
exports.QueueCollection = QueueCollection.QueueCollection;
const QueueStats = __importStar(require("./queue-stats"));
exports.QueueStats = QueueStats.QueueStats;
const QueueSummary = __importStar(require("./queue-summary"));
exports.QueueSummary = QueueSummary.QueueSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Stats = __importStar(require("./stats"));
exports.Stats = Stats.Stats;
const UpdateMessageDetails = __importStar(require("./update-message-details"));
exports.UpdateMessageDetails = UpdateMessageDetails.UpdateMessageDetails;
const UpdateMessagesDetails = __importStar(require("./update-messages-details"));
exports.UpdateMessagesDetails = UpdateMessagesDetails.UpdateMessagesDetails;
const UpdateMessagesDetailsEntry = __importStar(require("./update-messages-details-entry"));
exports.UpdateMessagesDetailsEntry = UpdateMessagesDetailsEntry.UpdateMessagesDetailsEntry;
const UpdateMessagesResult = __importStar(require("./update-messages-result"));
exports.UpdateMessagesResult = UpdateMessagesResult.UpdateMessagesResult;
const UpdateMessagesResultEntry = __importStar(require("./update-messages-result-entry"));
exports.UpdateMessagesResultEntry = UpdateMessagesResultEntry.UpdateMessagesResultEntry;
const UpdateQueueDetails = __importStar(require("./update-queue-details"));
exports.UpdateQueueDetails = UpdateQueueDetails.UpdateQueueDetails;
const UpdatedMessage = __importStar(require("./updated-message"));
exports.UpdatedMessage = UpdatedMessage.UpdatedMessage;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
