// oci-datalabelingservicedataplane/lib/request/summarize-annotation-analytics-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeAnnotationAnalyticsRequest = void 0;
var SummarizeAnnotationAnalyticsRequest;
(function (SummarizeAnnotationAnalyticsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Count"] = "count";
        SortBy["Label"] = "label";
        SortBy["UpdatedBy"] = "updatedBy";
    })(SortBy = SummarizeAnnotationAnalyticsRequest.SortBy || (SummarizeAnnotationAnalyticsRequest.SortBy = {}));
    let AnnotationGroupBy;
    (function (AnnotationGroupBy) {
        AnnotationGroupBy["UpdatedBy"] = "updatedBy";
        AnnotationGroupBy["Label"] = "label";
    })(AnnotationGroupBy = SummarizeAnnotationAnalyticsRequest.AnnotationGroupBy || (SummarizeAnnotationAnalyticsRequest.AnnotationGroupBy = {}));
})(SummarizeAnnotationAnalyticsRequest = exports.SummarizeAnnotationAnalyticsRequest || (exports.SummarizeAnnotationAnalyticsRequest = {}));
