// oci-healthchecks/lib/client.js
"use strict";
/**
 * Health Checks API
 * API for the Health Checks service. Use this API to manage endpoint probes and monitors.
For more information, see
[Overview of the Health Checks Service](/iaas/Content/HealthChecks/Concepts/healthchecks.htm).

 * OpenAPI spec version: 20180501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthChecksClient = exports.HealthChecksApiKeys = void 0;
const common = require("oci-common");
const model = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
const oci_common_2 = require("oci-common");
// ===============================================
// This file is autogenerated - Please do not edit
// ===============================================
var HealthChecksApiKeys;
(function (HealthChecksApiKeys) {
})(HealthChecksApiKeys = exports.HealthChecksApiKeys || (exports.HealthChecksApiKeys = {}));
/**
 * This service client uses {@link common.CircuitBreaker.DefaultConfiguration} for all the operations by default if no circuit breaker configuration is defined by the user.
 */
class HealthChecksClient {
    constructor(params, clientConfiguration) {
        this["_realmSpecificEndpointTemplateEnabled"] = false;
        this["_endpoint"] = "";
        this["_defaultHeaders"] = {};
        this._circuitBreaker = null;
        this._httpOptions = undefined;
        this.targetService = "HealthChecks";
        this._regionId = "";
        this._lastSetRegionOrRegionId = "";
        const requestSigner = params.authenticationDetailsProvider
            ? new common.DefaultRequestSigner(params.authenticationDetailsProvider)
            : null;
        if (clientConfiguration) {
            this._clientConfiguration = clientConfiguration;
            this._circuitBreaker = clientConfiguration.circuitBreaker
                ? clientConfiguration.circuitBreaker.circuit
                : null;
            this._httpOptions = clientConfiguration.httpOptions
                ? clientConfiguration.httpOptions
                : undefined;
        }
        // if circuit breaker is not created, check if circuit breaker system is enabled to use default circuit breaker
        const specCircuitBreakerEnabled = true;
        if (!this._circuitBreaker &&
            common.utils.isCircuitBreakerSystemEnabled(clientConfiguration) &&
            (specCircuitBreakerEnabled || common.CircuitBreaker.DefaultCircuitBreakerOverriden)) {
            this._circuitBreaker = new common.CircuitBreaker().circuit;
        }
        this._httpClient =
            params.httpClient ||
                new common.FetchHttpClient(requestSigner, this._circuitBreaker, this._httpOptions);
        if (params.authenticationDetailsProvider &&
            common.isRegionProvider(params.authenticationDetailsProvider)) {
            const provider = params.authenticationDetailsProvider;
            if (provider.getRegion()) {
                this.region = provider.getRegion();
            }
        }
    }
    /**
     * Get the endpoint that is being used to call (ex, https://www.example.com).
     */
    get endpoint() {
        return this._endpoint;
    }
    /**
     * Sets the endpoint to call (ex, https://www.example.com).
     * @param endpoint The endpoint of the service.
     */
    set endpoint(endpoint) {
        this._endpoint = endpoint;
        this._endpoint = this._endpoint + "/20180501";
        if (this.logger)
            this.logger.info(`HealthChecksClient endpoint set to ${this._endpoint}`);
    }
    get logger() {
        return common.LOG.logger;
    }
    /**
     * Determines whether realm specific endpoint should be used or not.
     * Set realmSpecificEndpointTemplateEnabled to "true" if the user wants to enable use of realm specific endpoint template, otherwise set it to "false"
     * @param realmSpecificEndpointTemplateEnabled flag to enable the use of realm specific endpoint template
     */
    set useRealmSpecificEndpointTemplate(realmSpecificEndpointTemplateEnabled) {
        this._realmSpecificEndpointTemplateEnabled = realmSpecificEndpointTemplateEnabled;
        if (this.logger)
            this.logger.info(`realmSpecificEndpointTemplateEnabled set to ${this._realmSpecificEndpointTemplateEnabled}`);
        if (this._lastSetRegionOrRegionId === common.Region.REGION_STRING) {
            this.endpoint = common.EndpointBuilder.createEndpointFromRegion(HealthChecksClient.serviceEndpointTemplate, this._region, HealthChecksClient.endpointServiceName);
        }
        else if (this._lastSetRegionOrRegionId === common.Region.REGION_ID_STRING) {
            this.endpoint = common.EndpointBuilder.createEndpointFromRegionId(HealthChecksClient.serviceEndpointTemplate, this._regionId, HealthChecksClient.endpointServiceName);
        }
    }
    /**
     * Sets the region to call (ex, Region.US_PHOENIX_1).
     * Note, this will call {@link #endpoint(String) endpoint} after resolving the endpoint.
     * @param region The region of the service.
     */
    set region(region) {
        this._region = region;
        this.endpoint = common.EndpointBuilder.createEndpointFromRegion(HealthChecksClient.serviceEndpointTemplate, region, HealthChecksClient.endpointServiceName);
        this._lastSetRegionOrRegionId = common.Region.REGION_STRING;
    }
    /**
     * Sets the regionId to call (ex, 'us-phoenix-1').
     *
     * Note, this will first try to map the region ID to a known Region and call {@link #region(Region) region}.
     * If no known Region could be determined, it will create an endpoint assuming its in default Realm OC1
     * and then call {@link #endpoint(String) endpoint}.
     * @param regionId The public region ID.
     */
    set regionId(regionId) {
        this._regionId = regionId;
        this.endpoint = common.EndpointBuilder.createEndpointFromRegionId(HealthChecksClient.serviceEndpointTemplate, regionId, HealthChecksClient.endpointServiceName);
        this._lastSetRegionOrRegionId = common.Region.REGION_ID_STRING;
    }
    /**
     * Moves a monitor into a different compartment. When provided, `If-Match` is checked
     * against ETag values of the resource.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param ChangeHttpMonitorCompartmentRequest
     * @return ChangeHttpMonitorCompartmentResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ChangeHttpMonitorCompartment.ts.html |here} to see how to use ChangeHttpMonitorCompartment API.
     */
    changeHttpMonitorCompartment(changeHttpMonitorCompartmentRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#changeHttpMonitorCompartment.");
            const operationName = "changeHttpMonitorCompartment";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitor/ChangeHttpMonitorCompartment";
            const pathParams = {
                "{monitorId}": changeHttpMonitorCompartmentRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "if-match": changeHttpMonitorCompartmentRequest.ifMatch,
                "opc-request-id": changeHttpMonitorCompartmentRequest.opcRequestId,
                "opc-retry-token": changeHttpMonitorCompartmentRequest.opcRetryToken
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, changeHttpMonitorCompartmentRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors/{monitorId}/actions/changeCompartment",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(changeHttpMonitorCompartmentRequest.changeHttpMonitorCompartmentDetails, "ChangeHttpMonitorCompartmentDetails", model.ChangeHttpMonitorCompartmentDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Moves a monitor into a different compartment. When provided, `If-Match` is checked
     * against ETag values of the resource.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param ChangePingMonitorCompartmentRequest
     * @return ChangePingMonitorCompartmentResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ChangePingMonitorCompartment.ts.html |here} to see how to use ChangePingMonitorCompartment API.
     */
    changePingMonitorCompartment(changePingMonitorCompartmentRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#changePingMonitorCompartment.");
            const operationName = "changePingMonitorCompartment";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitor/ChangePingMonitorCompartment";
            const pathParams = {
                "{monitorId}": changePingMonitorCompartmentRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "if-match": changePingMonitorCompartmentRequest.ifMatch,
                "opc-request-id": changePingMonitorCompartmentRequest.opcRequestId,
                "opc-retry-token": changePingMonitorCompartmentRequest.opcRetryToken
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, changePingMonitorCompartmentRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors/{monitorId}/actions/changeCompartment",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(changePingMonitorCompartmentRequest.changePingMonitorCompartmentDetails, "ChangePingMonitorCompartmentDetails", model.ChangePingMonitorCompartmentDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Creates an HTTP monitor. Vantage points will be automatically selected if not specified,
     * and probes will be initiated from each vantage point to each of the targets at the frequency
     * specified by `intervalInSeconds`.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param CreateHttpMonitorRequest
     * @return CreateHttpMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/CreateHttpMonitor.ts.html |here} to see how to use CreateHttpMonitor API.
     */
    createHttpMonitor(createHttpMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#createHttpMonitor.");
            const operationName = "createHttpMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitor/CreateHttpMonitor";
            const pathParams = {};
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": createHttpMonitorRequest.opcRequestId,
                "opc-retry-token": createHttpMonitorRequest.opcRetryToken
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, createHttpMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(createHttpMonitorRequest.createHttpMonitorDetails, "CreateHttpMonitorDetails", model.CreateHttpMonitorDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "httpMonitor",
                    bodyModel: model.HttpMonitor,
                    type: "model.HttpMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("location"),
                            key: "location",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Creates an on-demand HTTP probe. The location response header contains the URL for
     * fetching the probe results.
     * <p>
     *Note:* On-demand probe configurations are not saved.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param CreateOnDemandHttpProbeRequest
     * @return CreateOnDemandHttpProbeResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/CreateOnDemandHttpProbe.ts.html |here} to see how to use CreateOnDemandHttpProbe API.
     */
    createOnDemandHttpProbe(createOnDemandHttpProbeRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#createOnDemandHttpProbe.");
            const operationName = "createOnDemandHttpProbe";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpProbe/CreateOnDemandHttpProbe";
            const pathParams = {};
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": createOnDemandHttpProbeRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, createOnDemandHttpProbeRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpProbeResults",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(createOnDemandHttpProbeRequest.createOnDemandHttpProbeDetails, "CreateOnDemandHttpProbeDetails", model.CreateOnDemandHttpProbeDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "httpProbe",
                    bodyModel: model.HttpProbe,
                    type: "model.HttpProbe",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("location"),
                            key: "location",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Creates an on-demand ping probe. The location response header contains the URL for
     * fetching probe results.
     * <p>
     *Note:* The on-demand probe configuration is not saved.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param CreateOnDemandPingProbeRequest
     * @return CreateOnDemandPingProbeResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/CreateOnDemandPingProbe.ts.html |here} to see how to use CreateOnDemandPingProbe API.
     */
    createOnDemandPingProbe(createOnDemandPingProbeRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#createOnDemandPingProbe.");
            const operationName = "createOnDemandPingProbe";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingProbe/CreateOnDemandPingProbe";
            const pathParams = {};
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": createOnDemandPingProbeRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, createOnDemandPingProbeRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingProbeResults",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(createOnDemandPingProbeRequest.createOnDemandPingProbeDetails, "CreateOnDemandPingProbeDetails", model.CreateOnDemandPingProbeDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "pingProbe",
                    bodyModel: model.PingProbe,
                    type: "model.PingProbe",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("location"),
                            key: "location",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Creates a ping monitor. Vantage points will be automatically selected if not specified,
     * and probes will be initiated from each vantage point to each of the targets at the frequency
     * specified by `intervalInSeconds`.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param CreatePingMonitorRequest
     * @return CreatePingMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/CreatePingMonitor.ts.html |here} to see how to use CreatePingMonitor API.
     */
    createPingMonitor(createPingMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#createPingMonitor.");
            const operationName = "createPingMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitor/CreatePingMonitor";
            const pathParams = {};
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": createPingMonitorRequest.opcRequestId,
                "opc-retry-token": createPingMonitorRequest.opcRetryToken
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, createPingMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors",
                method: "POST",
                bodyContent: common.ObjectSerializer.serialize(createPingMonitorRequest.createPingMonitorDetails, "CreatePingMonitorDetails", model.CreatePingMonitorDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "pingMonitor",
                    bodyModel: model.PingMonitor,
                    type: "model.PingMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("location"),
                            key: "location",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Deletes the HTTP monitor and its configuration. All future probes of this
     * monitor are stopped. Results associated with the monitor are not deleted.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param DeleteHttpMonitorRequest
     * @return DeleteHttpMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/DeleteHttpMonitor.ts.html |here} to see how to use DeleteHttpMonitor API.
     */
    deleteHttpMonitor(deleteHttpMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#deleteHttpMonitor.");
            const operationName = "deleteHttpMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitor/DeleteHttpMonitor";
            const pathParams = {
                "{monitorId}": deleteHttpMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": deleteHttpMonitorRequest.opcRequestId,
                "if-match": deleteHttpMonitorRequest.ifMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, deleteHttpMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors/{monitorId}",
                method: "DELETE",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Deletes the ping monitor and its configuration. All future probes of this
     * monitor are stopped. Results associated with the monitor are not deleted.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param DeletePingMonitorRequest
     * @return DeletePingMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/DeletePingMonitor.ts.html |here} to see how to use DeletePingMonitor API.
     */
    deletePingMonitor(deletePingMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#deletePingMonitor.");
            const operationName = "deletePingMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitor/DeletePingMonitor";
            const pathParams = {
                "{monitorId}": deletePingMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": deletePingMonitorRequest.opcRequestId,
                "if-match": deletePingMonitorRequest.ifMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, deletePingMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors/{monitorId}",
                method: "DELETE",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Gets the configuration for the specified monitor.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param GetHttpMonitorRequest
     * @return GetHttpMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/GetHttpMonitor.ts.html |here} to see how to use GetHttpMonitor API.
     */
    getHttpMonitor(getHttpMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#getHttpMonitor.");
            const operationName = "getHttpMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitor/GetHttpMonitor";
            const pathParams = {
                "{monitorId}": getHttpMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": getHttpMonitorRequest.opcRequestId,
                "if-none-match": getHttpMonitorRequest.ifNoneMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, getHttpMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors/{monitorId}",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "httpMonitor",
                    bodyModel: model.HttpMonitor,
                    type: "model.HttpMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Gets the configuration for the specified ping monitor.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param GetPingMonitorRequest
     * @return GetPingMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/GetPingMonitor.ts.html |here} to see how to use GetPingMonitor API.
     */
    getPingMonitor(getPingMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#getPingMonitor.");
            const operationName = "getPingMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitor/GetPingMonitor";
            const pathParams = {
                "{monitorId}": getPingMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": getPingMonitorRequest.opcRequestId,
                "if-none-match": getPingMonitorRequest.ifNoneMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, getPingMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors/{monitorId}",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "pingMonitor",
                    bodyModel: model.PingMonitor,
                    type: "model.PingMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Gets information about all vantage points available to the user.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param ListHealthChecksVantagePointsRequest
     * @return ListHealthChecksVantagePointsResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ListHealthChecksVantagePoints.ts.html |here} to see how to use ListHealthChecksVantagePoints API.
     */
    listHealthChecksVantagePoints(listHealthChecksVantagePointsRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#listHealthChecksVantagePoints.");
            const operationName = "listHealthChecksVantagePoints";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HealthChecksVantagePointSummary/ListHealthChecksVantagePoints";
            const pathParams = {};
            const queryParams = {
                "limit": listHealthChecksVantagePointsRequest.limit,
                "page": listHealthChecksVantagePointsRequest.page,
                "sortBy": listHealthChecksVantagePointsRequest.sortBy,
                "sortOrder": listHealthChecksVantagePointsRequest.sortOrder,
                "name": listHealthChecksVantagePointsRequest.name,
                "displayName": listHealthChecksVantagePointsRequest.displayName
            };
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": listHealthChecksVantagePointsRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, listHealthChecksVantagePointsRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/vantagePoints",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "items",
                    bodyModel: model.HealthChecksVantagePointSummary,
                    type: "Array<model.HealthChecksVantagePointSummary>",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("opc-next-page"),
                            key: "opcNextPage",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * NOTE: This function is deprecated in favor of listHealthChecksVantagePointsRecordIterator function.
     * Creates a new async iterator which will iterate over the models.HealthChecksVantagePointSummary objects
     * contained in responses from the listHealthChecksVantagePoints operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHealthChecksVantagePoints(request) {
        return oci_common_1.paginateRecords(request, req => this.listHealthChecksVantagePoints(req));
    }
    /**
     * NOTE: This function is deprecated in favor of listHealthChecksVantagePointsResponseIterator function.
     * Creates a new async iterator which will iterate over the responses received from the listHealthChecksVantagePoints operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHealthChecksVantagePointsResponses(request) {
        return oci_common_1.paginateResponses(request, req => this.listHealthChecksVantagePoints(req));
    }
    /**
     * Creates a new async iterator which will iterate over the models.HealthChecksVantagePointSummary objects
     * contained in responses from the listHealthChecksVantagePoints operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHealthChecksVantagePointsRecordIterator(request) {
        return oci_common_1.paginateRecords(request, req => this.listHealthChecksVantagePoints(req));
    }
    /**
     * Creates a new async iterator which will iterate over the responses received from the listHealthChecksVantagePoints operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHealthChecksVantagePointsResponseIterator(request) {
        return oci_common_1.paginateResponses(request, req => this.listHealthChecksVantagePoints(req));
    }
    /**
     * Gets a list of HTTP monitors.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param ListHttpMonitorsRequest
     * @return ListHttpMonitorsResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ListHttpMonitors.ts.html |here} to see how to use ListHttpMonitors API.
     */
    listHttpMonitors(listHttpMonitorsRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#listHttpMonitors.");
            const operationName = "listHttpMonitors";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitorSummary/ListHttpMonitors";
            const pathParams = {};
            const queryParams = {
                "limit": listHttpMonitorsRequest.limit,
                "page": listHttpMonitorsRequest.page,
                "compartmentId": listHttpMonitorsRequest.compartmentId,
                "sortBy": listHttpMonitorsRequest.sortBy,
                "sortOrder": listHttpMonitorsRequest.sortOrder,
                "displayName": listHttpMonitorsRequest.displayName,
                "homeRegion": listHttpMonitorsRequest.homeRegion
            };
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": listHttpMonitorsRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, listHttpMonitorsRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "items",
                    bodyModel: model.HttpMonitorSummary,
                    type: "Array<model.HttpMonitorSummary>",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("opc-next-page"),
                            key: "opcNextPage",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * NOTE: This function is deprecated in favor of listHttpMonitorsRecordIterator function.
     * Creates a new async iterator which will iterate over the models.HttpMonitorSummary objects
     * contained in responses from the listHttpMonitors operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHttpMonitors(request) {
        return oci_common_1.paginateRecords(request, req => this.listHttpMonitors(req));
    }
    /**
     * NOTE: This function is deprecated in favor of listHttpMonitorsResponseIterator function.
     * Creates a new async iterator which will iterate over the responses received from the listHttpMonitors operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHttpMonitorsResponses(request) {
        return oci_common_1.paginateResponses(request, req => this.listHttpMonitors(req));
    }
    /**
     * Creates a new async iterator which will iterate over the models.HttpMonitorSummary objects
     * contained in responses from the listHttpMonitors operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHttpMonitorsRecordIterator(request) {
        return oci_common_1.paginateRecords(request, req => this.listHttpMonitors(req));
    }
    /**
     * Creates a new async iterator which will iterate over the responses received from the listHttpMonitors operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHttpMonitorsResponseIterator(request) {
        return oci_common_1.paginateResponses(request, req => this.listHttpMonitors(req));
    }
    /**
     * Gets the HTTP probe results for the specified probe or monitor, where
     * the `probeConfigurationId` is the OCID of either a monitor or an
     * on-demand probe.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param ListHttpProbeResultsRequest
     * @return ListHttpProbeResultsResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ListHttpProbeResults.ts.html |here} to see how to use ListHttpProbeResults API.
     */
    listHttpProbeResults(listHttpProbeResultsRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#listHttpProbeResults.");
            const operationName = "listHttpProbeResults";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpProbeResultSummary/ListHttpProbeResults";
            const pathParams = {
                "{probeConfigurationId}": listHttpProbeResultsRequest.probeConfigurationId
            };
            const queryParams = {
                "limit": listHttpProbeResultsRequest.limit,
                "page": listHttpProbeResultsRequest.page,
                "startTimeGreaterThanOrEqualTo": listHttpProbeResultsRequest.startTimeGreaterThanOrEqualTo,
                "startTimeLessThanOrEqualTo": listHttpProbeResultsRequest.startTimeLessThanOrEqualTo,
                "sortOrder": listHttpProbeResultsRequest.sortOrder,
                "target": listHttpProbeResultsRequest.target
            };
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": listHttpProbeResultsRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, listHttpProbeResultsRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpProbeResults/{probeConfigurationId}",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "items",
                    bodyModel: model.HttpProbeResultSummary,
                    type: "Array<model.HttpProbeResultSummary>",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("opc-next-page"),
                            key: "opcNextPage",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * NOTE: This function is deprecated in favor of listHttpProbeResultsRecordIterator function.
     * Creates a new async iterator which will iterate over the models.HttpProbeResultSummary objects
     * contained in responses from the listHttpProbeResults operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHttpProbeResults(request) {
        return oci_common_1.paginateRecords(request, req => this.listHttpProbeResults(req));
    }
    /**
     * NOTE: This function is deprecated in favor of listHttpProbeResultsResponseIterator function.
     * Creates a new async iterator which will iterate over the responses received from the listHttpProbeResults operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllHttpProbeResultsResponses(request) {
        return oci_common_1.paginateResponses(request, req => this.listHttpProbeResults(req));
    }
    /**
     * Creates a new async iterator which will iterate over the models.HttpProbeResultSummary objects
     * contained in responses from the listHttpProbeResults operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHttpProbeResultsRecordIterator(request) {
        return oci_common_1.paginateRecords(request, req => this.listHttpProbeResults(req));
    }
    /**
     * Creates a new async iterator which will iterate over the responses received from the listHttpProbeResults operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listHttpProbeResultsResponseIterator(request) {
        return oci_common_1.paginateResponses(request, req => this.listHttpProbeResults(req));
    }
    /**
       * Gets a list of configured ping monitors.
  * <p>
  Results are paginated based on `page` and `limit`.  The `opc-next-page` header provides
  * a URL for fetching the next page.
  *
       * This operation does not retry by default if the user has not defined a retry configuration.
       * @param ListPingMonitorsRequest
       * @return ListPingMonitorsResponse
       * @throws OciError when an error occurs
       * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ListPingMonitors.ts.html |here} to see how to use ListPingMonitors API.
       */
    listPingMonitors(listPingMonitorsRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#listPingMonitors.");
            const operationName = "listPingMonitors";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitorSummary/ListPingMonitors";
            const pathParams = {};
            const queryParams = {
                "limit": listPingMonitorsRequest.limit,
                "page": listPingMonitorsRequest.page,
                "compartmentId": listPingMonitorsRequest.compartmentId,
                "sortBy": listPingMonitorsRequest.sortBy,
                "sortOrder": listPingMonitorsRequest.sortOrder,
                "displayName": listPingMonitorsRequest.displayName,
                "homeRegion": listPingMonitorsRequest.homeRegion
            };
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": listPingMonitorsRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, listPingMonitorsRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "items",
                    bodyModel: model.PingMonitorSummary,
                    type: "Array<model.PingMonitorSummary>",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("opc-next-page"),
                            key: "opcNextPage",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * NOTE: This function is deprecated in favor of listPingMonitorsRecordIterator function.
     * Creates a new async iterator which will iterate over the models.PingMonitorSummary objects
     * contained in responses from the listPingMonitors operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllPingMonitors(request) {
        return oci_common_1.paginateRecords(request, req => this.listPingMonitors(req));
    }
    /**
     * NOTE: This function is deprecated in favor of listPingMonitorsResponseIterator function.
     * Creates a new async iterator which will iterate over the responses received from the listPingMonitors operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllPingMonitorsResponses(request) {
        return oci_common_1.paginateResponses(request, req => this.listPingMonitors(req));
    }
    /**
     * Creates a new async iterator which will iterate over the models.PingMonitorSummary objects
     * contained in responses from the listPingMonitors operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listPingMonitorsRecordIterator(request) {
        return oci_common_1.paginateRecords(request, req => this.listPingMonitors(req));
    }
    /**
     * Creates a new async iterator which will iterate over the responses received from the listPingMonitors operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listPingMonitorsResponseIterator(request) {
        return oci_common_1.paginateResponses(request, req => this.listPingMonitors(req));
    }
    /**
       * Returns the results for the specified probe, where the `probeConfigurationId`
  * is the OCID of either a monitor or an on-demand probe.
  * <p>
  Results are paginated based on `page` and `limit`.  The `opc-next-page` header provides
  * a URL for fetching the next page.  Use `sortOrder` to set the order of the
  * results.  If `sortOrder` is unspecified, results are sorted in ascending order by
  * `startTime`.
  *
       * This operation does not retry by default if the user has not defined a retry configuration.
       * @param ListPingProbeResultsRequest
       * @return ListPingProbeResultsResponse
       * @throws OciError when an error occurs
       * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/ListPingProbeResults.ts.html |here} to see how to use ListPingProbeResults API.
       */
    listPingProbeResults(listPingProbeResultsRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#listPingProbeResults.");
            const operationName = "listPingProbeResults";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingProbeResultSummary/ListPingProbeResults";
            const pathParams = {
                "{probeConfigurationId}": listPingProbeResultsRequest.probeConfigurationId
            };
            const queryParams = {
                "limit": listPingProbeResultsRequest.limit,
                "page": listPingProbeResultsRequest.page,
                "startTimeGreaterThanOrEqualTo": listPingProbeResultsRequest.startTimeGreaterThanOrEqualTo,
                "startTimeLessThanOrEqualTo": listPingProbeResultsRequest.startTimeLessThanOrEqualTo,
                "sortOrder": listPingProbeResultsRequest.sortOrder,
                "target": listPingProbeResultsRequest.target
            };
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": listPingProbeResultsRequest.opcRequestId
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, listPingProbeResultsRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingProbeResults/{probeConfigurationId}",
                method: "GET",
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "items",
                    bodyModel: model.PingProbeResultSummary,
                    type: "Array<model.PingProbeResultSummary>",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("opc-next-page"),
                            key: "opcNextPage",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * NOTE: This function is deprecated in favor of listPingProbeResultsRecordIterator function.
     * Creates a new async iterator which will iterate over the models.PingProbeResultSummary objects
     * contained in responses from the listPingProbeResults operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllPingProbeResults(request) {
        return oci_common_1.paginateRecords(request, req => this.listPingProbeResults(req));
    }
    /**
     * NOTE: This function is deprecated in favor of listPingProbeResultsResponseIterator function.
     * Creates a new async iterator which will iterate over the responses received from the listPingProbeResults operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listAllPingProbeResultsResponses(request) {
        return oci_common_1.paginateResponses(request, req => this.listPingProbeResults(req));
    }
    /**
     * Creates a new async iterator which will iterate over the models.PingProbeResultSummary objects
     * contained in responses from the listPingProbeResults operation. This iterator will fetch more data from the
     * server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listPingProbeResultsRecordIterator(request) {
        return oci_common_1.paginateRecords(request, req => this.listPingProbeResults(req));
    }
    /**
     * Creates a new async iterator which will iterate over the responses received from the listPingProbeResults operation. This iterator
     * will fetch more data from the server as needed.
     *
     * @param request a request which can be sent to the service operation
     */
    listPingProbeResultsResponseIterator(request) {
        return oci_common_1.paginateResponses(request, req => this.listPingProbeResults(req));
    }
    /**
     * Updates the configuration of the specified HTTP monitor. Only the fields
     * specified in the request body will be updated; all other configuration
     * properties will remain unchanged.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param UpdateHttpMonitorRequest
     * @return UpdateHttpMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/UpdateHttpMonitor.ts.html |here} to see how to use UpdateHttpMonitor API.
     */
    updateHttpMonitor(updateHttpMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#updateHttpMonitor.");
            const operationName = "updateHttpMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/HttpMonitor/UpdateHttpMonitor";
            const pathParams = {
                "{monitorId}": updateHttpMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": updateHttpMonitorRequest.opcRequestId,
                "if-match": updateHttpMonitorRequest.ifMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, updateHttpMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/httpMonitors/{monitorId}",
                method: "PUT",
                bodyContent: common.ObjectSerializer.serialize(updateHttpMonitorRequest.updateHttpMonitorDetails, "UpdateHttpMonitorDetails", model.UpdateHttpMonitorDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "httpMonitor",
                    bodyModel: model.HttpMonitor,
                    type: "model.HttpMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
    /**
     * Updates the configuration of the specified ping monitor. Only the fields
     * specified in the request body will be updated; all other configuration properties
     * will remain unchanged.
     *
     * This operation does not retry by default if the user has not defined a retry configuration.
     * @param UpdatePingMonitorRequest
     * @return UpdatePingMonitorResponse
     * @throws OciError when an error occurs
     * @example Click {@link https://docs.cloud.oracle.com/en-us/iaas/tools/typescript-sdk-examples/2.62.0/healthchecks/UpdatePingMonitor.ts.html |here} to see how to use UpdatePingMonitor API.
     */
    updatePingMonitor(updatePingMonitorRequest) {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.logger)
                this.logger.debug("Calling operation HealthChecksClient#updatePingMonitor.");
            const operationName = "updatePingMonitor";
            const apiReferenceLink = "https://docs.oracle.com/iaas/api/#/en/healthchecks/20180501/PingMonitor/UpdatePingMonitor";
            const pathParams = {
                "{monitorId}": updatePingMonitorRequest.monitorId
            };
            const queryParams = {};
            let headerParams = {
                "Content-Type": common.Constants.APPLICATION_JSON,
                "opc-request-id": updatePingMonitorRequest.opcRequestId,
                "if-match": updatePingMonitorRequest.ifMatch
            };
            const specRetryConfiguration = common.NoRetryConfigurationDetails;
            const retrier = oci_common_2.GenericRetrier.createPreferredRetrier(this._clientConfiguration ? this._clientConfiguration.retryConfiguration : undefined, updatePingMonitorRequest.retryConfiguration, specRetryConfiguration);
            if (this.logger)
                retrier.logger = this.logger;
            const request = yield oci_common_2.composeRequest({
                baseEndpoint: this._endpoint,
                defaultHeaders: this._defaultHeaders,
                path: "/pingMonitors/{monitorId}",
                method: "PUT",
                bodyContent: common.ObjectSerializer.serialize(updatePingMonitorRequest.updatePingMonitorDetails, "UpdatePingMonitorDetails", model.UpdatePingMonitorDetails.getJsonObj),
                pathParams: pathParams,
                headerParams: headerParams,
                queryParams: queryParams
            });
            try {
                const response = yield retrier.makeServiceCall(this._httpClient, request, this.targetService, operationName, apiReferenceLink);
                const sdkResponse = oci_common_2.composeResponse({
                    responseObject: {},
                    body: yield response.json(),
                    bodyKey: "pingMonitor",
                    bodyModel: model.PingMonitor,
                    type: "model.PingMonitor",
                    responseHeaders: [
                        {
                            value: response.headers.get("opc-request-id"),
                            key: "opcRequestId",
                            dataType: "string"
                        },
                        {
                            value: response.headers.get("etag"),
                            key: "etag",
                            dataType: "string"
                        }
                    ]
                });
                return sdkResponse;
            }
            catch (err) {
                throw err;
            }
        });
    }
}
exports.HealthChecksClient = HealthChecksClient;
HealthChecksClient.serviceEndpointTemplate = "https://healthchecks.{region}.oci.{secondLevelDomain}";
HealthChecksClient.endpointServiceName = "";
