// oci-datalabelingservice/lib/model/export-format.js
"use strict";
/**
 * Data Labeling Management API
 * Use Data Labeling Management API to create, list, edit & delete datasets.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportFormat = void 0;
var ExportFormat;
(function (ExportFormat) {
    let Name;
    (function (Name) {
        Name["Jsonl"] = "JSONL";
        Name["JsonlConsolidated"] = "JSONL_CONSOLIDATED";
        Name["Conll"] = "CONLL";
        Name["Spacy"] = "SPACY";
        Name["Coco"] = "COCO";
        Name["Yolo"] = "YOLO";
        Name["PascalVoc"] = "PASCAL_VOC";
        Name["JsonlCompactPlusContent"] = "JSONL_COMPACT_PLUS_CONTENT";
    })(Name = ExportFormat.Name || (ExportFormat.Name = {}));
    let Version;
    (function (Version) {
        Version["V2003"] = "V2003";
        Version["V5"] = "V5";
    })(Version = ExportFormat.Version || (ExportFormat.Version = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportFormat.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ExportFormat.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExportFormat = exports.ExportFormat || (exports.ExportFormat = {}));
