// oci-datalabelingservice/lib/model/operation-type.js
"use strict";
/**
 * Data Labeling Management API
 * Use Data Labeling Management API to create, list, edit & delete datasets.
 * OpenAPI spec version: 20211001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateDataset"] = "CREATE_DATASET";
    OperationType["DeleteDataset"] = "DELETE_DATASET";
    OperationType["MoveDataset"] = "MOVE_DATASET";
    OperationType["GenerateDatasetRecords"] = "GENERATE_DATASET_RECORDS";
    OperationType["SnapshotDataset"] = "SNAPSHOT_DATASET";
    OperationType["AddDatasetLabels"] = "ADD_DATASET_LABELS";
    OperationType["RemoveDatasetLabels"] = "REMOVE_DATASET_LABELS";
    OperationType["RenameDatasetLabels"] = "RENAME_DATASET_LABELS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
