// oci-marketplace/lib/model/architecture-type.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArchitectureType = void 0;
/**
 * Possible values for a listings compatbile architecture
 **/
var ArchitectureType;
(function (ArchitectureType) {
    ArchitectureType["X86"] = "X86";
    ArchitectureType["Arm"] = "ARM";
})(ArchitectureType = exports.ArchitectureType || (exports.ArchitectureType = {}));
(function (ArchitectureType) {
    function getJsonObj(obj) {
        return obj;
    }
    ArchitectureType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ArchitectureType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ArchitectureType = exports.ArchitectureType || (exports.ArchitectureType = {}));
