// oci-marketplace/lib/model/eula-type-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EulaTypeEnum = void 0;
/**
 * Possible values for end user license agreement (EULA) formats.
 **/
var EulaTypeEnum;
(function (EulaTypeEnum) {
    EulaTypeEnum["Text"] = "TEXT";
})(EulaTypeEnum = exports.EulaTypeEnum || (exports.EulaTypeEnum = {}));
(function (EulaTypeEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    EulaTypeEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    EulaTypeEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(EulaTypeEnum = exports.EulaTypeEnum || (exports.EulaTypeEnum = {}));
