// oci-marketplace/lib/model/ineligibility-reason-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IneligibilityReasonEnum = void 0;
/**
 * Possible values of on why a tenant cannot launch a listing
 **/
var IneligibilityReasonEnum;
(function (IneligibilityReasonEnum) {
    IneligibilityReasonEnum["IneligibleAccountCountry"] = "INELIGIBLE_ACCOUNT_COUNTRY";
    IneligibilityReasonEnum["IneligibleRegion"] = "INELIGIBLE_REGION";
    IneligibilityReasonEnum["IneligibleAccountBlacklisted"] = "INELIGIBLE_ACCOUNT_BLACKLISTED";
    IneligibilityReasonEnum["IneligibleAccountFeatureDisabled"] = "INELIGIBLE_ACCOUNT_FEATURE_DISABLED";
    IneligibilityReasonEnum["IneligibleAccountCurrency"] = "INELIGIBLE_ACCOUNT_CURRENCY";
    IneligibilityReasonEnum["IneligibleAccountNotPaid"] = "INELIGIBLE_ACCOUNT_NOT_PAID";
    IneligibilityReasonEnum["IneligibleAccountInternal"] = "INELIGIBLE_ACCOUNT_INTERNAL";
    IneligibilityReasonEnum["IneligibleAccountGovSubscription"] = "INELIGIBLE_ACCOUNT_GOV_SUBSCRIPTION";
    IneligibilityReasonEnum["IneligiblePaidListingThrottled"] = "INELIGIBLE_PAID_LISTING_THROTTLED";
    IneligibilityReasonEnum["NotAuthorized"] = "NOT_AUTHORIZED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    IneligibilityReasonEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(IneligibilityReasonEnum = exports.IneligibilityReasonEnum || (exports.IneligibilityReasonEnum = {}));
(function (IneligibilityReasonEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    IneligibilityReasonEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    IneligibilityReasonEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(IneligibilityReasonEnum = exports.IneligibilityReasonEnum || (exports.IneligibilityReasonEnum = {}));
