// oci-marketplace/lib/model/link-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkEnum = void 0;
/**
 * Possible values for a link type.
 **/
var LinkEnum;
(function (LinkEnum) {
    LinkEnum["Self"] = "SELF";
    LinkEnum["Canonical"] = "CANONICAL";
    LinkEnum["Next"] = "NEXT";
    LinkEnum["Template"] = "TEMPLATE";
    LinkEnum["Prev"] = "PREV";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LinkEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(LinkEnum = exports.LinkEnum || (exports.LinkEnum = {}));
(function (LinkEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    LinkEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LinkEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(LinkEnum = exports.LinkEnum || (exports.LinkEnum = {}));
