// oci-marketplace/lib/model/matching-context-type-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MatchingContextTypeEnum = void 0;
/**
 * Possible values for matching context returned in the response.
 * If you specify HIGHLIGHTS, then the service will highlight fragments in its response. The default value is NONE.
 *
 **/
var MatchingContextTypeEnum;
(function (MatchingContextTypeEnum) {
    MatchingContextTypeEnum["None"] = "NONE";
    MatchingContextTypeEnum["Highlights"] = "HIGHLIGHTS";
})(MatchingContextTypeEnum = exports.MatchingContextTypeEnum || (exports.MatchingContextTypeEnum = {}));
(function (MatchingContextTypeEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    MatchingContextTypeEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MatchingContextTypeEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(MatchingContextTypeEnum = exports.MatchingContextTypeEnum || (exports.MatchingContextTypeEnum = {}));
