// oci-marketplace/lib/model/search-details-type-enum.js
"use strict";
/**
 * Marketplace Service API
 * Use the Marketplace API to manage applications in Oracle Cloud Infrastructure Marketplace. For more information, see [Overview of Marketplace](/Content/Marketplace/Concepts/marketoverview.htm)
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchDetailsTypeEnum = void 0;
/**
 * Possible values for the type of SearchDetails.
 * FreeText - A request containing arbitrary text that must be present in the resource.
 * Structured - A request that uses Search's structured query language to specify filter conditions to apply to search results.
 *
 **/
var SearchDetailsTypeEnum;
(function (SearchDetailsTypeEnum) {
    SearchDetailsTypeEnum["FreeText"] = "FreeText";
    SearchDetailsTypeEnum["Structured"] = "Structured";
})(SearchDetailsTypeEnum = exports.SearchDetailsTypeEnum || (exports.SearchDetailsTypeEnum = {}));
(function (SearchDetailsTypeEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    SearchDetailsTypeEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SearchDetailsTypeEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(SearchDetailsTypeEnum = exports.SearchDetailsTypeEnum || (exports.SearchDetailsTypeEnum = {}));
