// oci-secrets/lib/model/secret-bundle-version-summary.js
"use strict";
/**
 * Vault Secret Retrieval API
 * Use the Secret Retrieval API to retrieve secrets and secret versions from vaults. For more information, see [Managing Secrets](/Content/KeyManagement/Tasks/managingsecrets.htm).
 * OpenAPI spec version: 20190301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretBundleVersionSummary = void 0;
var SecretBundleVersionSummary;
(function (SecretBundleVersionSummary) {
    let Stages;
    (function (Stages) {
        Stages["Current"] = "CURRENT";
        Stages["Pending"] = "PENDING";
        Stages["Latest"] = "LATEST";
        Stages["Previous"] = "PREVIOUS";
        Stages["Deprecated"] = "DEPRECATED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Stages["UnknownValue"] = "UNKNOWN_VALUE";
    })(Stages = SecretBundleVersionSummary.Stages || (SecretBundleVersionSummary.Stages = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SecretBundleVersionSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SecretBundleVersionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecretBundleVersionSummary = exports.SecretBundleVersionSummary || (exports.SecretBundleVersionSummary = {}));
