// oci-networkloadbalancer/lib/model/hcs-infra-ip-version.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HcsInfraIpVersion = void 0;
/**
 * Mode specifying the IP version of HCS Infra.
 **/
var HcsInfraIpVersion;
(function (HcsInfraIpVersion) {
    HcsInfraIpVersion["Ipv4"] = "IPV4";
    HcsInfraIpVersion["Ipv4AndIpv6"] = "IPV4_AND_IPV6";
})(HcsInfraIpVersion = exports.HcsInfraIpVersion || (exports.HcsInfraIpVersion = {}));
(function (HcsInfraIpVersion) {
    function getJsonObj(obj) {
        return obj;
    }
    HcsInfraIpVersion.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    HcsInfraIpVersion.getDeserializedJsonObj = getDeserializedJsonObj;
})(HcsInfraIpVersion = exports.HcsInfraIpVersion || (exports.HcsInfraIpVersion = {}));
