// oci-networkloadbalancer/lib/model/ip-version.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpVersion = void 0;
/**
 * Mode specifying the version of IP.
 **/
var IpVersion;
(function (IpVersion) {
    IpVersion["Ipv4"] = "IPV4";
    IpVersion["Ipv6"] = "IPV6";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    IpVersion["UnknownValue"] = "UNKNOWN_VALUE";
})(IpVersion = exports.IpVersion || (exports.IpVersion = {}));
(function (IpVersion) {
    function getJsonObj(obj) {
        return obj;
    }
    IpVersion.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    IpVersion.getDeserializedJsonObj = getDeserializedJsonObj;
})(IpVersion = exports.IpVersion || (exports.IpVersion = {}));
