// oci-networkloadbalancer/lib/model/network-load-balancers-protocol-summary.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkLoadBalancersProtocolSummary = void 0;
/**
 * This object is deprecated.
 * Protocols supported for the listener of the network load balancer.
 *
 **/
var NetworkLoadBalancersProtocolSummary;
(function (NetworkLoadBalancersProtocolSummary) {
    NetworkLoadBalancersProtocolSummary["Any"] = "ANY";
    NetworkLoadBalancersProtocolSummary["Tcp"] = "TCP";
    NetworkLoadBalancersProtocolSummary["Udp"] = "UDP";
    NetworkLoadBalancersProtocolSummary["TcpAndUdp"] = "TCP_AND_UDP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NetworkLoadBalancersProtocolSummary["UnknownValue"] = "UNKNOWN_VALUE";
})(NetworkLoadBalancersProtocolSummary = exports.NetworkLoadBalancersProtocolSummary || (exports.NetworkLoadBalancersProtocolSummary = {}));
(function (NetworkLoadBalancersProtocolSummary) {
    function getJsonObj(obj) {
        return obj;
    }
    NetworkLoadBalancersProtocolSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NetworkLoadBalancersProtocolSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(NetworkLoadBalancersProtocolSummary = exports.NetworkLoadBalancersProtocolSummary || (exports.NetworkLoadBalancersProtocolSummary = {}));
