// oci-networkloadbalancer/lib/model/nlb-ip-version.js
"use strict";
/**
 * NetworkLoadBalancer API
 * This describes the network load balancer API.
 * OpenAPI spec version: 20200501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NlbIpVersion = void 0;
/**
 * Mode specifying the IP version of NLB.
 **/
var NlbIpVersion;
(function (NlbIpVersion) {
    NlbIpVersion["Ipv4"] = "IPV4";
    NlbIpVersion["Ipv4AndIpv6"] = "IPV4_AND_IPV6";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NlbIpVersion["UnknownValue"] = "UNKNOWN_VALUE";
})(NlbIpVersion = exports.NlbIpVersion || (exports.NlbIpVersion = {}));
(function (NlbIpVersion) {
    function getJsonObj(obj) {
        return obj;
    }
    NlbIpVersion.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NlbIpVersion.getDeserializedJsonObj = getDeserializedJsonObj;
})(NlbIpVersion = exports.NlbIpVersion || (exports.NlbIpVersion = {}));
