// oci-certificates/lib/request/get-certificate-bundle-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20210224
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCertificateBundleRequest = void 0;
var GetCertificateBundleRequest;
(function (GetCertificateBundleRequest) {
    let Stage;
    (function (Stage) {
        Stage["Current"] = "CURRENT";
        Stage["Pending"] = "PENDING";
        Stage["Latest"] = "LATEST";
        Stage["Previous"] = "PREVIOUS";
        Stage["Deprecated"] = "DEPRECATED";
    })(Stage = GetCertificateBundleRequest.Stage || (GetCertificateBundleRequest.Stage = {}));
    let CertificateBundleType;
    (function (CertificateBundleType) {
        CertificateBundleType["CertificateContentPublicOnly"] = "CERTIFICATE_CONTENT_PUBLIC_ONLY";
        CertificateBundleType["CertificateContentWithPrivateKey"] = "CERTIFICATE_CONTENT_WITH_PRIVATE_KEY";
    })(CertificateBundleType = GetCertificateBundleRequest.CertificateBundleType || (GetCertificateBundleRequest.CertificateBundleType = {}));
})(GetCertificateBundleRequest = exports.GetCertificateBundleRequest || (exports.GetCertificateBundleRequest = {}));
