// oci-operatoraccesscontrol/lib/model/access-request-lifecycle-states.js
"use strict";
/**
 * OperatorAccessControl API
 * Operator Access Control enables you to control the time duration and the actions an Oracle operator can perform on your Exadata Cloud@Customer infrastructure.
Using logging service, you can view a near real-time audit report of all actions performed by an Oracle operator.

Use the table of contents and search tool to explore the OperatorAccessControl API.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessRequestLifecycleStates = void 0;
/**
 * Possible lifecycle states for an access request.
 **/
var AccessRequestLifecycleStates;
(function (AccessRequestLifecycleStates) {
    AccessRequestLifecycleStates["Created"] = "CREATED";
    AccessRequestLifecycleStates["Approvalwaiting"] = "APPROVALWAITING";
    AccessRequestLifecycleStates["Preapproved"] = "PREAPPROVED";
    AccessRequestLifecycleStates["Approved"] = "APPROVED";
    AccessRequestLifecycleStates["Moreinfo"] = "MOREINFO";
    AccessRequestLifecycleStates["Rejected"] = "REJECTED";
    AccessRequestLifecycleStates["Deployed"] = "DEPLOYED";
    AccessRequestLifecycleStates["Deployfailed"] = "DEPLOYFAILED";
    AccessRequestLifecycleStates["Undeployed"] = "UNDEPLOYED";
    AccessRequestLifecycleStates["Undeployfailed"] = "UNDEPLOYFAILED";
    AccessRequestLifecycleStates["Closefailed"] = "CLOSEFAILED";
    AccessRequestLifecycleStates["Revokefailed"] = "REVOKEFAILED";
    AccessRequestLifecycleStates["Expiryfailed"] = "EXPIRYFAILED";
    AccessRequestLifecycleStates["Revoking"] = "REVOKING";
    AccessRequestLifecycleStates["Revoked"] = "REVOKED";
    AccessRequestLifecycleStates["Extending"] = "EXTENDING";
    AccessRequestLifecycleStates["Extended"] = "EXTENDED";
    AccessRequestLifecycleStates["Extensionrejected"] = "EXTENSIONREJECTED";
    AccessRequestLifecycleStates["Completing"] = "COMPLETING";
    AccessRequestLifecycleStates["Completed"] = "COMPLETED";
    AccessRequestLifecycleStates["Expired"] = "EXPIRED";
    AccessRequestLifecycleStates["Approvedforfuture"] = "APPROVEDFORFUTURE";
    AccessRequestLifecycleStates["Inreview"] = "INREVIEW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AccessRequestLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(AccessRequestLifecycleStates = exports.AccessRequestLifecycleStates || (exports.AccessRequestLifecycleStates = {}));
(function (AccessRequestLifecycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    AccessRequestLifecycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AccessRequestLifecycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(AccessRequestLifecycleStates = exports.AccessRequestLifecycleStates || (exports.AccessRequestLifecycleStates = {}));
