// oci-operatoraccesscontrol/lib/model/index.js
"use strict";
/**
 * OperatorAccessControl API
 * Operator Access Control enables you to control the time duration and the actions an Oracle operator can perform on your Exadata Cloud@Customer infrastructure.
Using logging service, you can view a near real-time audit report of all actions performed by an Oracle operator.

Use the table of contents and search tool to explore the OperatorAccessControl API.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOperatorControlDetails = exports.UpdateOperatorControlAssignmentDetails = exports.SortOrders = exports.RevokeAccessRequestDetails = exports.ReviewAccessRequestDetails = exports.ResourceTypes = exports.RejectAccessRequestDetails = exports.OperatorControlSummary = exports.OperatorControlLifecycleStates = exports.OperatorControlCollection = exports.OperatorControlAssignmentSummary = exports.OperatorControlAssignmentLifecycleStates = exports.OperatorControlAssignmentCollection = exports.OperatorControlAssignment = exports.OperatorControl = exports.OperatorActionSummary = exports.OperatorActionProperties = exports.OperatorActionLifecycleStates = exports.OperatorActionCollection = exports.OperatorActionCategories = exports.OperatorAction = exports.InteractionSummary = exports.InteractionRequestDetails = exports.InteractionCollection = exports.InfrastrcutureLayers = exports.CreateOperatorControlDetails = exports.CreateOperatorControlAssignmentDetails = exports.ChangeOperatorControlCompartmentDetails = exports.ChangeOperatorControlAssignmentCompartmentDetails = exports.ApproveAccessRequestDetails = exports.AccessRequestSummary = exports.AccessRequestSeverities = exports.AccessRequestLifecycleStates = exports.AccessRequestHistorySummary = exports.AccessRequestHistoryCollection = exports.AccessRequestCollection = exports.AccessRequest = void 0;
const AccessRequest = __importStar(require("./access-request"));
exports.AccessRequest = AccessRequest.AccessRequest;
const AccessRequestCollection = __importStar(require("./access-request-collection"));
exports.AccessRequestCollection = AccessRequestCollection.AccessRequestCollection;
const AccessRequestHistoryCollection = __importStar(require("./access-request-history-collection"));
exports.AccessRequestHistoryCollection = AccessRequestHistoryCollection.AccessRequestHistoryCollection;
const AccessRequestHistorySummary = __importStar(require("./access-request-history-summary"));
exports.AccessRequestHistorySummary = AccessRequestHistorySummary.AccessRequestHistorySummary;
const AccessRequestLifecycleStates = __importStar(require("./access-request-lifecycle-states"));
exports.AccessRequestLifecycleStates = AccessRequestLifecycleStates.AccessRequestLifecycleStates;
const AccessRequestSeverities = __importStar(require("./access-request-severities"));
exports.AccessRequestSeverities = AccessRequestSeverities.AccessRequestSeverities;
const AccessRequestSummary = __importStar(require("./access-request-summary"));
exports.AccessRequestSummary = AccessRequestSummary.AccessRequestSummary;
const ApproveAccessRequestDetails = __importStar(require("./approve-access-request-details"));
exports.ApproveAccessRequestDetails = ApproveAccessRequestDetails.ApproveAccessRequestDetails;
const ChangeOperatorControlAssignmentCompartmentDetails = __importStar(require("./change-operator-control-assignment-compartment-details"));
exports.ChangeOperatorControlAssignmentCompartmentDetails = ChangeOperatorControlAssignmentCompartmentDetails.ChangeOperatorControlAssignmentCompartmentDetails;
const ChangeOperatorControlCompartmentDetails = __importStar(require("./change-operator-control-compartment-details"));
exports.ChangeOperatorControlCompartmentDetails = ChangeOperatorControlCompartmentDetails.ChangeOperatorControlCompartmentDetails;
const CreateOperatorControlAssignmentDetails = __importStar(require("./create-operator-control-assignment-details"));
exports.CreateOperatorControlAssignmentDetails = CreateOperatorControlAssignmentDetails.CreateOperatorControlAssignmentDetails;
const CreateOperatorControlDetails = __importStar(require("./create-operator-control-details"));
exports.CreateOperatorControlDetails = CreateOperatorControlDetails.CreateOperatorControlDetails;
const InfrastrcutureLayers = __importStar(require("./infrastrcuture-layers"));
exports.InfrastrcutureLayers = InfrastrcutureLayers.InfrastrcutureLayers;
const InteractionCollection = __importStar(require("./interaction-collection"));
exports.InteractionCollection = InteractionCollection.InteractionCollection;
const InteractionRequestDetails = __importStar(require("./interaction-request-details"));
exports.InteractionRequestDetails = InteractionRequestDetails.InteractionRequestDetails;
const InteractionSummary = __importStar(require("./interaction-summary"));
exports.InteractionSummary = InteractionSummary.InteractionSummary;
const OperatorAction = __importStar(require("./operator-action"));
exports.OperatorAction = OperatorAction.OperatorAction;
const OperatorActionCategories = __importStar(require("./operator-action-categories"));
exports.OperatorActionCategories = OperatorActionCategories.OperatorActionCategories;
const OperatorActionCollection = __importStar(require("./operator-action-collection"));
exports.OperatorActionCollection = OperatorActionCollection.OperatorActionCollection;
const OperatorActionLifecycleStates = __importStar(require("./operator-action-lifecycle-states"));
exports.OperatorActionLifecycleStates = OperatorActionLifecycleStates.OperatorActionLifecycleStates;
const OperatorActionProperties = __importStar(require("./operator-action-properties"));
exports.OperatorActionProperties = OperatorActionProperties.OperatorActionProperties;
const OperatorActionSummary = __importStar(require("./operator-action-summary"));
exports.OperatorActionSummary = OperatorActionSummary.OperatorActionSummary;
const OperatorControl = __importStar(require("./operator-control"));
exports.OperatorControl = OperatorControl.OperatorControl;
const OperatorControlAssignment = __importStar(require("./operator-control-assignment"));
exports.OperatorControlAssignment = OperatorControlAssignment.OperatorControlAssignment;
const OperatorControlAssignmentCollection = __importStar(require("./operator-control-assignment-collection"));
exports.OperatorControlAssignmentCollection = OperatorControlAssignmentCollection.OperatorControlAssignmentCollection;
const OperatorControlAssignmentLifecycleStates = __importStar(require("./operator-control-assignment-lifecycle-states"));
exports.OperatorControlAssignmentLifecycleStates = OperatorControlAssignmentLifecycleStates.OperatorControlAssignmentLifecycleStates;
const OperatorControlAssignmentSummary = __importStar(require("./operator-control-assignment-summary"));
exports.OperatorControlAssignmentSummary = OperatorControlAssignmentSummary.OperatorControlAssignmentSummary;
const OperatorControlCollection = __importStar(require("./operator-control-collection"));
exports.OperatorControlCollection = OperatorControlCollection.OperatorControlCollection;
const OperatorControlLifecycleStates = __importStar(require("./operator-control-lifecycle-states"));
exports.OperatorControlLifecycleStates = OperatorControlLifecycleStates.OperatorControlLifecycleStates;
const OperatorControlSummary = __importStar(require("./operator-control-summary"));
exports.OperatorControlSummary = OperatorControlSummary.OperatorControlSummary;
const RejectAccessRequestDetails = __importStar(require("./reject-access-request-details"));
exports.RejectAccessRequestDetails = RejectAccessRequestDetails.RejectAccessRequestDetails;
const ResourceTypes = __importStar(require("./resource-types"));
exports.ResourceTypes = ResourceTypes.ResourceTypes;
const ReviewAccessRequestDetails = __importStar(require("./review-access-request-details"));
exports.ReviewAccessRequestDetails = ReviewAccessRequestDetails.ReviewAccessRequestDetails;
const RevokeAccessRequestDetails = __importStar(require("./revoke-access-request-details"));
exports.RevokeAccessRequestDetails = RevokeAccessRequestDetails.RevokeAccessRequestDetails;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateOperatorControlAssignmentDetails = __importStar(require("./update-operator-control-assignment-details"));
exports.UpdateOperatorControlAssignmentDetails = UpdateOperatorControlAssignmentDetails.UpdateOperatorControlAssignmentDetails;
const UpdateOperatorControlDetails = __importStar(require("./update-operator-control-details"));
exports.UpdateOperatorControlDetails = UpdateOperatorControlDetails.UpdateOperatorControlDetails;
