// oci-operatoraccesscontrol/lib/model/infrastrcuture-layers.js
"use strict";
/**
 * OperatorAccessControl API
 * Operator Access Control enables you to control the time duration and the actions an Oracle operator can perform on your Exadata Cloud@Customer infrastructure.
Using logging service, you can view a near real-time audit report of all actions performed by an Oracle operator.

Use the table of contents and search tool to explore the OperatorAccessControl API.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfrastrcutureLayers = void 0;
/**
 * Infrastructure layers available to be governed under operator control
 * - DOM0
 * - CellServer
 * - CPS (Control Plane Server)
 *
 **/
var InfrastrcutureLayers;
(function (InfrastrcutureLayers) {
    InfrastrcutureLayers["Dom0"] = "DOM0";
    InfrastrcutureLayers["Cellserver"] = "CELLSERVER";
    InfrastrcutureLayers["Cps"] = "CPS";
})(InfrastrcutureLayers = exports.InfrastrcutureLayers || (exports.InfrastrcutureLayers = {}));
(function (InfrastrcutureLayers) {
    function getJsonObj(obj) {
        return obj;
    }
    InfrastrcutureLayers.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InfrastrcutureLayers.getDeserializedJsonObj = getDeserializedJsonObj;
})(InfrastrcutureLayers = exports.InfrastrcutureLayers || (exports.InfrastrcutureLayers = {}));
