// oci-operatoraccesscontrol/lib/model/operator-control-lifecycle-states.js
"use strict";
/**
 * OperatorAccessControl API
 * Operator Access Control enables you to control the time duration and the actions an Oracle operator can perform on your Exadata Cloud@Customer infrastructure.
Using logging service, you can view a near real-time audit report of all actions performed by an Oracle operator.

Use the table of contents and search tool to explore the OperatorAccessControl API.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatorControlLifecycleStates = void 0;
/**
 * Possible lifecycle states for operator control.
 **/
var OperatorControlLifecycleStates;
(function (OperatorControlLifecycleStates) {
    OperatorControlLifecycleStates["Created"] = "CREATED";
    OperatorControlLifecycleStates["Assigned"] = "ASSIGNED";
    OperatorControlLifecycleStates["Unassigned"] = "UNASSIGNED";
    OperatorControlLifecycleStates["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperatorControlLifecycleStates["UnknownValue"] = "UNKNOWN_VALUE";
})(OperatorControlLifecycleStates = exports.OperatorControlLifecycleStates || (exports.OperatorControlLifecycleStates = {}));
(function (OperatorControlLifecycleStates) {
    function getJsonObj(obj) {
        return obj;
    }
    OperatorControlLifecycleStates.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperatorControlLifecycleStates.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperatorControlLifecycleStates = exports.OperatorControlLifecycleStates || (exports.OperatorControlLifecycleStates = {}));
