// oci-apmsynthetics/lib/model/aggregated-network-data.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedNetworkData = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var AggregatedNetworkData;
(function (AggregatedNetworkData) {
    let ResultState;
    (function (ResultState) {
        ResultState["Success"] = "SUCCESS";
        ResultState["Failure"] = "FAILURE";
        ResultState["Partial"] = "PARTIAL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResultState["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResultState = AggregatedNetworkData.ResultState || (AggregatedNetworkData.ResultState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "vantagePointNodes": obj.vantagePointNodes
                ? obj.vantagePointNodes.map(item => {
                    return model.VantagePointNode.getJsonObj(item);
                })
                : undefined,
            "links": obj.links ? common.mapContainer(obj.links, model.Link.getJsonObj) : undefined
        });
        return jsonObj;
    }
    AggregatedNetworkData.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "vantagePointNodes": obj.vantagePointNodes
                ? obj.vantagePointNodes.map(item => {
                    return model.VantagePointNode.getDeserializedJsonObj(item);
                })
                : undefined,
            "links": obj.links
                ? common.mapContainer(obj.links, model.Link.getDeserializedJsonObj)
                : undefined
        });
        return jsonObj;
    }
    AggregatedNetworkData.getDeserializedJsonObj = getDeserializedJsonObj;
})(AggregatedNetworkData = exports.AggregatedNetworkData || (exports.AggregatedNetworkData = {}));
