// oci-apmsynthetics/lib/model/protocol.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Protocol = void 0;
/**
 * Type of protocol for network configuration.
 **/
var Protocol;
(function (Protocol) {
    Protocol["Icmp"] = "ICMP";
    Protocol["Tcp"] = "TCP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Protocol["UnknownValue"] = "UNKNOWN_VALUE";
})(Protocol = exports.Protocol || (exports.Protocol = {}));
(function (Protocol) {
    function getJsonObj(obj) {
        return obj;
    }
    Protocol.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    Protocol.getDeserializedJsonObj = getDeserializedJsonObj;
})(Protocol = exports.Protocol || (exports.Protocol = {}));
