// oci-aispeech/lib/model/index.js
"use strict";
/**
 * Speech API
 * The OCI Speech Service harnesses the power of spoken language by allowing developers to easily convert file-based data containing human speech into highly accurate text transcriptions.
 * OpenAPI spec version: 20220101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfanityTranscriptionFilter = exports.ObjectListInlineInputLocation = exports.ObjectListFileInputLocation = exports.UpdateTranscriptionJobDetails = exports.TranscriptionTaskSummary = exports.TranscriptionTaskCollection = exports.TranscriptionTask = exports.TranscriptionNormalization = exports.TranscriptionModelDetails = exports.TranscriptionJobSummary = exports.TranscriptionJobCollection = exports.TranscriptionJob = exports.TranscriptionFilter = exports.SortOrder = exports.OutputLocation = exports.ObjectLocation = exports.InputLocation = exports.CreateTranscriptionJobDetails = exports.ChangeTranscriptionJobCompartmentDetails = exports.AudioFormatDetails = void 0;
const AudioFormatDetails = __importStar(require("./audio-format-details"));
exports.AudioFormatDetails = AudioFormatDetails.AudioFormatDetails;
const ChangeTranscriptionJobCompartmentDetails = __importStar(require("./change-transcription-job-compartment-details"));
exports.ChangeTranscriptionJobCompartmentDetails = ChangeTranscriptionJobCompartmentDetails.ChangeTranscriptionJobCompartmentDetails;
const CreateTranscriptionJobDetails = __importStar(require("./create-transcription-job-details"));
exports.CreateTranscriptionJobDetails = CreateTranscriptionJobDetails.CreateTranscriptionJobDetails;
const InputLocation = __importStar(require("./input-location"));
exports.InputLocation = InputLocation.InputLocation;
const ObjectLocation = __importStar(require("./object-location"));
exports.ObjectLocation = ObjectLocation.ObjectLocation;
const OutputLocation = __importStar(require("./output-location"));
exports.OutputLocation = OutputLocation.OutputLocation;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TranscriptionFilter = __importStar(require("./transcription-filter"));
exports.TranscriptionFilter = TranscriptionFilter.TranscriptionFilter;
const TranscriptionJob = __importStar(require("./transcription-job"));
exports.TranscriptionJob = TranscriptionJob.TranscriptionJob;
const TranscriptionJobCollection = __importStar(require("./transcription-job-collection"));
exports.TranscriptionJobCollection = TranscriptionJobCollection.TranscriptionJobCollection;
const TranscriptionJobSummary = __importStar(require("./transcription-job-summary"));
exports.TranscriptionJobSummary = TranscriptionJobSummary.TranscriptionJobSummary;
const TranscriptionModelDetails = __importStar(require("./transcription-model-details"));
exports.TranscriptionModelDetails = TranscriptionModelDetails.TranscriptionModelDetails;
const TranscriptionNormalization = __importStar(require("./transcription-normalization"));
exports.TranscriptionNormalization = TranscriptionNormalization.TranscriptionNormalization;
const TranscriptionTask = __importStar(require("./transcription-task"));
exports.TranscriptionTask = TranscriptionTask.TranscriptionTask;
const TranscriptionTaskCollection = __importStar(require("./transcription-task-collection"));
exports.TranscriptionTaskCollection = TranscriptionTaskCollection.TranscriptionTaskCollection;
const TranscriptionTaskSummary = __importStar(require("./transcription-task-summary"));
exports.TranscriptionTaskSummary = TranscriptionTaskSummary.TranscriptionTaskSummary;
const UpdateTranscriptionJobDetails = __importStar(require("./update-transcription-job-details"));
exports.UpdateTranscriptionJobDetails = UpdateTranscriptionJobDetails.UpdateTranscriptionJobDetails;
const ObjectListFileInputLocation = __importStar(require("./object-list-file-input-location"));
exports.ObjectListFileInputLocation = ObjectListFileInputLocation.ObjectListFileInputLocation;
const ObjectListInlineInputLocation = __importStar(require("./object-list-inline-input-location"));
exports.ObjectListInlineInputLocation = ObjectListInlineInputLocation.ObjectListInlineInputLocation;
const ProfanityTranscriptionFilter = __importStar(require("./profanity-transcription-filter"));
exports.ProfanityTranscriptionFilter = ProfanityTranscriptionFilter.ProfanityTranscriptionFilter;
