// oci-accessgovernancecp/lib/model/instance-lifecycle-state.js
"use strict";
/**
 * Access Governance APIs
 * Use the Oracle Access Governance API to create, view, and manage GovernanceInstances.
 * OpenAPI spec version: 20220518
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceLifecycleState = void 0;
/**
 * The lifecycle state of a GovernanceInstance.
 **/
var InstanceLifecycleState;
(function (InstanceLifecycleState) {
    InstanceLifecycleState["Creating"] = "CREATING";
    InstanceLifecycleState["Active"] = "ACTIVE";
    InstanceLifecycleState["Deleting"] = "DELETING";
    InstanceLifecycleState["Deleted"] = "DELETED";
    InstanceLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    InstanceLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(InstanceLifecycleState = exports.InstanceLifecycleState || (exports.InstanceLifecycleState = {}));
(function (InstanceLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    InstanceLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    InstanceLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(InstanceLifecycleState = exports.InstanceLifecycleState || (exports.InstanceLifecycleState = {}));
