// oci-cloudmigrations/lib/model/discovery-schedule-lifecycle-state.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryScheduleLifecycleState = void 0;
/**
 * Lifecycle state of a discovery schedule.
 **/
var DiscoveryScheduleLifecycleState;
(function (DiscoveryScheduleLifecycleState) {
    DiscoveryScheduleLifecycleState["Active"] = "ACTIVE";
    DiscoveryScheduleLifecycleState["Deleted"] = "DELETED";
})(DiscoveryScheduleLifecycleState = exports.DiscoveryScheduleLifecycleState || (exports.DiscoveryScheduleLifecycleState = {}));
(function (DiscoveryScheduleLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DiscoveryScheduleLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DiscoveryScheduleLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveryScheduleLifecycleState = exports.DiscoveryScheduleLifecycleState || (exports.DiscoveryScheduleLifecycleState = {}));
