// oci-cloudmigrations/lib/model/hydrated-volume.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HydratedVolume = void 0;
var HydratedVolume;
(function (HydratedVolume) {
    let VolumeType;
    (function (VolumeType) {
        VolumeType["Boot"] = "BOOT";
        VolumeType["Block"] = "BLOCK";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        VolumeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(VolumeType = HydratedVolume.VolumeType || (HydratedVolume.VolumeType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    HydratedVolume.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    HydratedVolume.getDeserializedJsonObj = getDeserializedJsonObj;
})(HydratedVolume = exports.HydratedVolume || (exports.HydratedVolume = {}));
