// oci-cloudmigrations/lib/model/index.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationPlan = exports.MigrationCollection = exports.MigrationAssetSummary = exports.MigrationAssetCollection = exports.MigrationAsset = exports.Migration = exports.MetricType = exports.MetricTimeWindow = exports.LaunchInstanceShapeConfigDetails = exports.LaunchInstanceDetails = exports.LaunchInstanceAgentConfigDetails = exports.InstanceSourceDetails = exports.InstanceOptions = exports.InstanceAgentPluginConfigDetails = exports.HydratedVolume = exports.DiscoveryScheduleSummary = exports.DiscoveryScheduleLifecycleState = exports.DiscoveryScheduleCollection = exports.DiscoverySchedule = exports.CreateVnicDetails = exports.CreateTargetAssetDetails = exports.CreateReplicationScheduleDetails = exports.CreateMigrationPlanDetails = exports.CreateMigrationDetails = exports.CreateMigrationAssetDetails = exports.CreateDiscoveryScheduleDetails = exports.CreateAssetSourceDetails = exports.CostEstimation = exports.ComputeCostEstimation = exports.CompatibilityMessage = exports.ChangeReplicationScheduleCompartmentDetails = exports.ChangeMigrationPlanCompartmentDetails = exports.ChangeMigrationCompartmentDetails = exports.ChangeDiscoveryScheduleCompartmentDetails = exports.ChangeAssetSourceCompartmentDetails = exports.AvailableShapesCollection = exports.AvailableShapeSummary = exports.AssetSourceType = exports.AssetSourceSummary = exports.AssetSourceLifecycleState = exports.AssetSourceCredentialsType = exports.AssetSourceCredentials = exports.AssetSourceConnectionType = exports.AssetSourceConnectionLifecycleState = exports.AssetSourceConnectionCollection = exports.AssetSourceConnection = exports.AssetSourceCollection = exports.AssetSource = exports.ActionType = exports.AccessLevel = void 0;
exports.VmTargetEnvironment = exports.VmTargetAssetSummary = exports.VmTargetAsset = exports.UpdateVmWareAssetSourceDetails = exports.UpdateVmTargetAssetDetails = exports.TerminatePreemptionAction = exports.PercentileResourceAssessmentStrategy = exports.PeakResourceAssessmentStrategy = exports.InstanceSourceViaImageDetails = exports.InstanceSourceViaBootVolumeDetails = exports.CreateVmWareAssetSourceDetails = exports.CreateVmTargetAssetDetails = exports.AverageResourceAssessmentStrategy = exports.AsIsResourceAssessmentStrategy = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VolumeCostEstimation = exports.UpdateTargetAssetDetails = exports.UpdateReplicationScheduleDetails = exports.UpdateMigrationPlanDetails = exports.UpdateMigrationDetails = exports.UpdateMigrationAssetDetails = exports.UpdateDiscoveryScheduleDetails = exports.UpdateAssetSourceDetails = exports.TargetEnvironment = exports.TargetAssetSummary = exports.TargetAssetCollection = exports.TargetAsset = exports.StorageCostEstimation = exports.SortOrder = exports.ResourceAssessmentStrategy = exports.ReplicationScheduleSummary = exports.ReplicationScheduleCollection = exports.ReplicationSchedule = exports.ReplicationProgress = exports.PreemptionAction = exports.PreemptibleInstanceConfigDetails = exports.OsImageEstimation = exports.OperationType = exports.OperationStatus = exports.MigrationSummary = exports.MigrationPlanSummary = exports.MigrationPlanStats = exports.MigrationPlanCollection = void 0;
exports.VmWareAssetSourceSummary = exports.VmWareAssetSource = void 0;
const AccessLevel = __importStar(require("./access-level"));
exports.AccessLevel = AccessLevel.AccessLevel;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AssetSource = __importStar(require("./asset-source"));
exports.AssetSource = AssetSource.AssetSource;
const AssetSourceCollection = __importStar(require("./asset-source-collection"));
exports.AssetSourceCollection = AssetSourceCollection.AssetSourceCollection;
const AssetSourceConnection = __importStar(require("./asset-source-connection"));
exports.AssetSourceConnection = AssetSourceConnection.AssetSourceConnection;
const AssetSourceConnectionCollection = __importStar(require("./asset-source-connection-collection"));
exports.AssetSourceConnectionCollection = AssetSourceConnectionCollection.AssetSourceConnectionCollection;
const AssetSourceConnectionLifecycleState = __importStar(require("./asset-source-connection-lifecycle-state"));
exports.AssetSourceConnectionLifecycleState = AssetSourceConnectionLifecycleState.AssetSourceConnectionLifecycleState;
const AssetSourceConnectionType = __importStar(require("./asset-source-connection-type"));
exports.AssetSourceConnectionType = AssetSourceConnectionType.AssetSourceConnectionType;
const AssetSourceCredentials = __importStar(require("./asset-source-credentials"));
exports.AssetSourceCredentials = AssetSourceCredentials.AssetSourceCredentials;
const AssetSourceCredentialsType = __importStar(require("./asset-source-credentials-type"));
exports.AssetSourceCredentialsType = AssetSourceCredentialsType.AssetSourceCredentialsType;
const AssetSourceLifecycleState = __importStar(require("./asset-source-lifecycle-state"));
exports.AssetSourceLifecycleState = AssetSourceLifecycleState.AssetSourceLifecycleState;
const AssetSourceSummary = __importStar(require("./asset-source-summary"));
exports.AssetSourceSummary = AssetSourceSummary.AssetSourceSummary;
const AssetSourceType = __importStar(require("./asset-source-type"));
exports.AssetSourceType = AssetSourceType.AssetSourceType;
const AvailableShapeSummary = __importStar(require("./available-shape-summary"));
exports.AvailableShapeSummary = AvailableShapeSummary.AvailableShapeSummary;
const AvailableShapesCollection = __importStar(require("./available-shapes-collection"));
exports.AvailableShapesCollection = AvailableShapesCollection.AvailableShapesCollection;
const ChangeAssetSourceCompartmentDetails = __importStar(require("./change-asset-source-compartment-details"));
exports.ChangeAssetSourceCompartmentDetails = ChangeAssetSourceCompartmentDetails.ChangeAssetSourceCompartmentDetails;
const ChangeDiscoveryScheduleCompartmentDetails = __importStar(require("./change-discovery-schedule-compartment-details"));
exports.ChangeDiscoveryScheduleCompartmentDetails = ChangeDiscoveryScheduleCompartmentDetails.ChangeDiscoveryScheduleCompartmentDetails;
const ChangeMigrationCompartmentDetails = __importStar(require("./change-migration-compartment-details"));
exports.ChangeMigrationCompartmentDetails = ChangeMigrationCompartmentDetails.ChangeMigrationCompartmentDetails;
const ChangeMigrationPlanCompartmentDetails = __importStar(require("./change-migration-plan-compartment-details"));
exports.ChangeMigrationPlanCompartmentDetails = ChangeMigrationPlanCompartmentDetails.ChangeMigrationPlanCompartmentDetails;
const ChangeReplicationScheduleCompartmentDetails = __importStar(require("./change-replication-schedule-compartment-details"));
exports.ChangeReplicationScheduleCompartmentDetails = ChangeReplicationScheduleCompartmentDetails.ChangeReplicationScheduleCompartmentDetails;
const CompatibilityMessage = __importStar(require("./compatibility-message"));
exports.CompatibilityMessage = CompatibilityMessage.CompatibilityMessage;
const ComputeCostEstimation = __importStar(require("./compute-cost-estimation"));
exports.ComputeCostEstimation = ComputeCostEstimation.ComputeCostEstimation;
const CostEstimation = __importStar(require("./cost-estimation"));
exports.CostEstimation = CostEstimation.CostEstimation;
const CreateAssetSourceDetails = __importStar(require("./create-asset-source-details"));
exports.CreateAssetSourceDetails = CreateAssetSourceDetails.CreateAssetSourceDetails;
const CreateDiscoveryScheduleDetails = __importStar(require("./create-discovery-schedule-details"));
exports.CreateDiscoveryScheduleDetails = CreateDiscoveryScheduleDetails.CreateDiscoveryScheduleDetails;
const CreateMigrationAssetDetails = __importStar(require("./create-migration-asset-details"));
exports.CreateMigrationAssetDetails = CreateMigrationAssetDetails.CreateMigrationAssetDetails;
const CreateMigrationDetails = __importStar(require("./create-migration-details"));
exports.CreateMigrationDetails = CreateMigrationDetails.CreateMigrationDetails;
const CreateMigrationPlanDetails = __importStar(require("./create-migration-plan-details"));
exports.CreateMigrationPlanDetails = CreateMigrationPlanDetails.CreateMigrationPlanDetails;
const CreateReplicationScheduleDetails = __importStar(require("./create-replication-schedule-details"));
exports.CreateReplicationScheduleDetails = CreateReplicationScheduleDetails.CreateReplicationScheduleDetails;
const CreateTargetAssetDetails = __importStar(require("./create-target-asset-details"));
exports.CreateTargetAssetDetails = CreateTargetAssetDetails.CreateTargetAssetDetails;
const CreateVnicDetails = __importStar(require("./create-vnic-details"));
exports.CreateVnicDetails = CreateVnicDetails.CreateVnicDetails;
const DiscoverySchedule = __importStar(require("./discovery-schedule"));
exports.DiscoverySchedule = DiscoverySchedule.DiscoverySchedule;
const DiscoveryScheduleCollection = __importStar(require("./discovery-schedule-collection"));
exports.DiscoveryScheduleCollection = DiscoveryScheduleCollection.DiscoveryScheduleCollection;
const DiscoveryScheduleLifecycleState = __importStar(require("./discovery-schedule-lifecycle-state"));
exports.DiscoveryScheduleLifecycleState = DiscoveryScheduleLifecycleState.DiscoveryScheduleLifecycleState;
const DiscoveryScheduleSummary = __importStar(require("./discovery-schedule-summary"));
exports.DiscoveryScheduleSummary = DiscoveryScheduleSummary.DiscoveryScheduleSummary;
const HydratedVolume = __importStar(require("./hydrated-volume"));
exports.HydratedVolume = HydratedVolume.HydratedVolume;
const InstanceAgentPluginConfigDetails = __importStar(require("./instance-agent-plugin-config-details"));
exports.InstanceAgentPluginConfigDetails = InstanceAgentPluginConfigDetails.InstanceAgentPluginConfigDetails;
const InstanceOptions = __importStar(require("./instance-options"));
exports.InstanceOptions = InstanceOptions.InstanceOptions;
const InstanceSourceDetails = __importStar(require("./instance-source-details"));
exports.InstanceSourceDetails = InstanceSourceDetails.InstanceSourceDetails;
const LaunchInstanceAgentConfigDetails = __importStar(require("./launch-instance-agent-config-details"));
exports.LaunchInstanceAgentConfigDetails = LaunchInstanceAgentConfigDetails.LaunchInstanceAgentConfigDetails;
const LaunchInstanceDetails = __importStar(require("./launch-instance-details"));
exports.LaunchInstanceDetails = LaunchInstanceDetails.LaunchInstanceDetails;
const LaunchInstanceShapeConfigDetails = __importStar(require("./launch-instance-shape-config-details"));
exports.LaunchInstanceShapeConfigDetails = LaunchInstanceShapeConfigDetails.LaunchInstanceShapeConfigDetails;
const MetricTimeWindow = __importStar(require("./metric-time-window"));
exports.MetricTimeWindow = MetricTimeWindow.MetricTimeWindow;
const MetricType = __importStar(require("./metric-type"));
exports.MetricType = MetricType.MetricType;
const Migration = __importStar(require("./migration"));
exports.Migration = Migration.Migration;
const MigrationAsset = __importStar(require("./migration-asset"));
exports.MigrationAsset = MigrationAsset.MigrationAsset;
const MigrationAssetCollection = __importStar(require("./migration-asset-collection"));
exports.MigrationAssetCollection = MigrationAssetCollection.MigrationAssetCollection;
const MigrationAssetSummary = __importStar(require("./migration-asset-summary"));
exports.MigrationAssetSummary = MigrationAssetSummary.MigrationAssetSummary;
const MigrationCollection = __importStar(require("./migration-collection"));
exports.MigrationCollection = MigrationCollection.MigrationCollection;
const MigrationPlan = __importStar(require("./migration-plan"));
exports.MigrationPlan = MigrationPlan.MigrationPlan;
const MigrationPlanCollection = __importStar(require("./migration-plan-collection"));
exports.MigrationPlanCollection = MigrationPlanCollection.MigrationPlanCollection;
const MigrationPlanStats = __importStar(require("./migration-plan-stats"));
exports.MigrationPlanStats = MigrationPlanStats.MigrationPlanStats;
const MigrationPlanSummary = __importStar(require("./migration-plan-summary"));
exports.MigrationPlanSummary = MigrationPlanSummary.MigrationPlanSummary;
const MigrationSummary = __importStar(require("./migration-summary"));
exports.MigrationSummary = MigrationSummary.MigrationSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OsImageEstimation = __importStar(require("./os-image-estimation"));
exports.OsImageEstimation = OsImageEstimation.OsImageEstimation;
const PreemptibleInstanceConfigDetails = __importStar(require("./preemptible-instance-config-details"));
exports.PreemptibleInstanceConfigDetails = PreemptibleInstanceConfigDetails.PreemptibleInstanceConfigDetails;
const PreemptionAction = __importStar(require("./preemption-action"));
exports.PreemptionAction = PreemptionAction.PreemptionAction;
const ReplicationProgress = __importStar(require("./replication-progress"));
exports.ReplicationProgress = ReplicationProgress.ReplicationProgress;
const ReplicationSchedule = __importStar(require("./replication-schedule"));
exports.ReplicationSchedule = ReplicationSchedule.ReplicationSchedule;
const ReplicationScheduleCollection = __importStar(require("./replication-schedule-collection"));
exports.ReplicationScheduleCollection = ReplicationScheduleCollection.ReplicationScheduleCollection;
const ReplicationScheduleSummary = __importStar(require("./replication-schedule-summary"));
exports.ReplicationScheduleSummary = ReplicationScheduleSummary.ReplicationScheduleSummary;
const ResourceAssessmentStrategy = __importStar(require("./resource-assessment-strategy"));
exports.ResourceAssessmentStrategy = ResourceAssessmentStrategy.ResourceAssessmentStrategy;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const StorageCostEstimation = __importStar(require("./storage-cost-estimation"));
exports.StorageCostEstimation = StorageCostEstimation.StorageCostEstimation;
const TargetAsset = __importStar(require("./target-asset"));
exports.TargetAsset = TargetAsset.TargetAsset;
const TargetAssetCollection = __importStar(require("./target-asset-collection"));
exports.TargetAssetCollection = TargetAssetCollection.TargetAssetCollection;
const TargetAssetSummary = __importStar(require("./target-asset-summary"));
exports.TargetAssetSummary = TargetAssetSummary.TargetAssetSummary;
const TargetEnvironment = __importStar(require("./target-environment"));
exports.TargetEnvironment = TargetEnvironment.TargetEnvironment;
const UpdateAssetSourceDetails = __importStar(require("./update-asset-source-details"));
exports.UpdateAssetSourceDetails = UpdateAssetSourceDetails.UpdateAssetSourceDetails;
const UpdateDiscoveryScheduleDetails = __importStar(require("./update-discovery-schedule-details"));
exports.UpdateDiscoveryScheduleDetails = UpdateDiscoveryScheduleDetails.UpdateDiscoveryScheduleDetails;
const UpdateMigrationAssetDetails = __importStar(require("./update-migration-asset-details"));
exports.UpdateMigrationAssetDetails = UpdateMigrationAssetDetails.UpdateMigrationAssetDetails;
const UpdateMigrationDetails = __importStar(require("./update-migration-details"));
exports.UpdateMigrationDetails = UpdateMigrationDetails.UpdateMigrationDetails;
const UpdateMigrationPlanDetails = __importStar(require("./update-migration-plan-details"));
exports.UpdateMigrationPlanDetails = UpdateMigrationPlanDetails.UpdateMigrationPlanDetails;
const UpdateReplicationScheduleDetails = __importStar(require("./update-replication-schedule-details"));
exports.UpdateReplicationScheduleDetails = UpdateReplicationScheduleDetails.UpdateReplicationScheduleDetails;
const UpdateTargetAssetDetails = __importStar(require("./update-target-asset-details"));
exports.UpdateTargetAssetDetails = UpdateTargetAssetDetails.UpdateTargetAssetDetails;
const VolumeCostEstimation = __importStar(require("./volume-cost-estimation"));
exports.VolumeCostEstimation = VolumeCostEstimation.VolumeCostEstimation;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const AsIsResourceAssessmentStrategy = __importStar(require("./as-is-resource-assessment-strategy"));
exports.AsIsResourceAssessmentStrategy = AsIsResourceAssessmentStrategy.AsIsResourceAssessmentStrategy;
const AverageResourceAssessmentStrategy = __importStar(require("./average-resource-assessment-strategy"));
exports.AverageResourceAssessmentStrategy = AverageResourceAssessmentStrategy.AverageResourceAssessmentStrategy;
const CreateVmTargetAssetDetails = __importStar(require("./create-vm-target-asset-details"));
exports.CreateVmTargetAssetDetails = CreateVmTargetAssetDetails.CreateVmTargetAssetDetails;
const CreateVmWareAssetSourceDetails = __importStar(require("./create-vm-ware-asset-source-details"));
exports.CreateVmWareAssetSourceDetails = CreateVmWareAssetSourceDetails.CreateVmWareAssetSourceDetails;
const InstanceSourceViaBootVolumeDetails = __importStar(require("./instance-source-via-boot-volume-details"));
exports.InstanceSourceViaBootVolumeDetails = InstanceSourceViaBootVolumeDetails.InstanceSourceViaBootVolumeDetails;
const InstanceSourceViaImageDetails = __importStar(require("./instance-source-via-image-details"));
exports.InstanceSourceViaImageDetails = InstanceSourceViaImageDetails.InstanceSourceViaImageDetails;
const PeakResourceAssessmentStrategy = __importStar(require("./peak-resource-assessment-strategy"));
exports.PeakResourceAssessmentStrategy = PeakResourceAssessmentStrategy.PeakResourceAssessmentStrategy;
const PercentileResourceAssessmentStrategy = __importStar(require("./percentile-resource-assessment-strategy"));
exports.PercentileResourceAssessmentStrategy = PercentileResourceAssessmentStrategy.PercentileResourceAssessmentStrategy;
const TerminatePreemptionAction = __importStar(require("./terminate-preemption-action"));
exports.TerminatePreemptionAction = TerminatePreemptionAction.TerminatePreemptionAction;
const UpdateVmTargetAssetDetails = __importStar(require("./update-vm-target-asset-details"));
exports.UpdateVmTargetAssetDetails = UpdateVmTargetAssetDetails.UpdateVmTargetAssetDetails;
const UpdateVmWareAssetSourceDetails = __importStar(require("./update-vm-ware-asset-source-details"));
exports.UpdateVmWareAssetSourceDetails = UpdateVmWareAssetSourceDetails.UpdateVmWareAssetSourceDetails;
const VmTargetAsset = __importStar(require("./vm-target-asset"));
exports.VmTargetAsset = VmTargetAsset.VmTargetAsset;
const VmTargetAssetSummary = __importStar(require("./vm-target-asset-summary"));
exports.VmTargetAssetSummary = VmTargetAssetSummary.VmTargetAssetSummary;
const VmTargetEnvironment = __importStar(require("./vm-target-environment"));
exports.VmTargetEnvironment = VmTargetEnvironment.VmTargetEnvironment;
const VmWareAssetSource = __importStar(require("./vm-ware-asset-source"));
exports.VmWareAssetSource = VmWareAssetSource.VmWareAssetSource;
const VmWareAssetSourceSummary = __importStar(require("./vm-ware-asset-source-summary"));
exports.VmWareAssetSourceSummary = VmWareAssetSourceSummary.VmWareAssetSourceSummary;
