// oci-cloudmigrations/lib/model/launch-instance-shape-config-details.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LaunchInstanceShapeConfigDetails = void 0;
var LaunchInstanceShapeConfigDetails;
(function (LaunchInstanceShapeConfigDetails) {
    let BaselineOcpuUtilization;
    (function (BaselineOcpuUtilization) {
        BaselineOcpuUtilization["Baseline18"] = "BASELINE_1_8";
        BaselineOcpuUtilization["Baseline12"] = "BASELINE_1_2";
        BaselineOcpuUtilization["Baseline11"] = "BASELINE_1_1";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BaselineOcpuUtilization["UnknownValue"] = "UNKNOWN_VALUE";
    })(BaselineOcpuUtilization = LaunchInstanceShapeConfigDetails.BaselineOcpuUtilization || (LaunchInstanceShapeConfigDetails.BaselineOcpuUtilization = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LaunchInstanceShapeConfigDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LaunchInstanceShapeConfigDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(LaunchInstanceShapeConfigDetails = exports.LaunchInstanceShapeConfigDetails || (exports.LaunchInstanceShapeConfigDetails = {}));
