// oci-cloudmigrations/lib/model/metric-time-window.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricTimeWindow = void 0;
/**
 * Time window for the metrics query
 **/
var MetricTimeWindow;
(function (MetricTimeWindow) {
    MetricTimeWindow["_1d"] = "1d";
    MetricTimeWindow["_7d"] = "7d";
    MetricTimeWindow["_30d"] = "30d";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MetricTimeWindow["UnknownValue"] = "UNKNOWN_VALUE";
})(MetricTimeWindow = exports.MetricTimeWindow || (exports.MetricTimeWindow = {}));
(function (MetricTimeWindow) {
    function getJsonObj(obj) {
        return obj;
    }
    MetricTimeWindow.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MetricTimeWindow.getDeserializedJsonObj = getDeserializedJsonObj;
})(MetricTimeWindow = exports.MetricTimeWindow || (exports.MetricTimeWindow = {}));
