// oci-cloudmigrations/lib/model/metric-type.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricType = void 0;
/**
 * Metric type for the recommendation process.
 **/
var MetricType;
(function (MetricType) {
    MetricType["Auto"] = "AUTO";
    MetricType["Historical"] = "HISTORICAL";
    MetricType["Runtime"] = "RUNTIME";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MetricType["UnknownValue"] = "UNKNOWN_VALUE";
})(MetricType = exports.MetricType || (exports.MetricType = {}));
(function (MetricType) {
    function getJsonObj(obj) {
        return obj;
    }
    MetricType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MetricType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MetricType = exports.MetricType || (exports.MetricType = {}));
