// oci-cloudmigrations/lib/model/migration-asset-summary.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationAssetSummary = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var MigrationAssetSummary;
(function (MigrationAssetSummary) {
    let Notifications;
    (function (Notifications) {
        Notifications["OutOfDate"] = "OUT_OF_DATE";
        Notifications["SourceRemoved"] = "SOURCE_REMOVED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Notifications["UnknownValue"] = "UNKNOWN_VALUE";
    })(Notifications = MigrationAssetSummary.Notifications || (MigrationAssetSummary.Notifications = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "snapshots": obj.snapshots
                ? common.mapContainer(obj.snapshots, model.HydratedVolume.getJsonObj)
                : undefined
        });
        return jsonObj;
    }
    MigrationAssetSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "snapshots": obj.snapshots
                ? common.mapContainer(obj.snapshots, model.HydratedVolume.getDeserializedJsonObj)
                : undefined
        });
        return jsonObj;
    }
    MigrationAssetSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(MigrationAssetSummary = exports.MigrationAssetSummary || (exports.MigrationAssetSummary = {}));
