// oci-cloudmigrations/lib/model/operation-type.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * Possible operation types.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateMigration"] = "CREATE_MIGRATION";
    OperationType["UpdateMigration"] = "UPDATE_MIGRATION";
    OperationType["RefreshMigration"] = "REFRESH_MIGRATION";
    OperationType["DeleteMigration"] = "DELETE_MIGRATION";
    OperationType["MoveMigration"] = "MOVE_MIGRATION";
    OperationType["StartAssetReplication"] = "START_ASSET_REPLICATION";
    OperationType["StartMigrationReplication"] = "START_MIGRATION_REPLICATION";
    OperationType["CreateReplicationSchedule"] = "CREATE_REPLICATION_SCHEDULE";
    OperationType["UpdateReplicationSchedule"] = "UPDATE_REPLICATION_SCHEDULE";
    OperationType["DeleteReplicationSchedule"] = "DELETE_REPLICATION_SCHEDULE";
    OperationType["MoveReplicationSchedule"] = "MOVE_REPLICATION_SCHEDULE";
    OperationType["CreateMigrationPlan"] = "CREATE_MIGRATION_PLAN";
    OperationType["UpdateMigrationPlan"] = "UPDATE_MIGRATION_PLAN";
    OperationType["DeleteMigrationPlan"] = "DELETE_MIGRATION_PLAN";
    OperationType["MoveMigrationPlan"] = "MOVE_MIGRATION_PLAN";
    OperationType["RefreshMigrationPlan"] = "REFRESH_MIGRATION_PLAN";
    OperationType["ExecuteMigrationPlan"] = "EXECUTE_MIGRATION_PLAN";
    OperationType["RefreshMigrationAsset"] = "REFRESH_MIGRATION_ASSET";
    OperationType["CreateMigrationAsset"] = "CREATE_MIGRATION_ASSET";
    OperationType["DeleteMigrationAsset"] = "DELETE_MIGRATION_ASSET";
    OperationType["CreateTargetAsset"] = "CREATE_TARGET_ASSET";
    OperationType["UpdateTargetAsset"] = "UPDATE_TARGET_ASSET";
    OperationType["DeleteTargetAsset"] = "DELETE_TARGET_ASSET";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
