// oci-cloudmigrations/lib/model/percentile-resource-assessment-strategy.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PercentileResourceAssessmentStrategy = void 0;
const model = __importStar(require("../model"));
var PercentileResourceAssessmentStrategy;
(function (PercentileResourceAssessmentStrategy) {
    let Percentile;
    (function (Percentile) {
        Percentile["P50"] = "P50";
        Percentile["P90"] = "P90";
        Percentile["P95"] = "P95";
        Percentile["P99"] = "P99";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Percentile["UnknownValue"] = "UNKNOWN_VALUE";
    })(Percentile = PercentileResourceAssessmentStrategy.Percentile || (PercentileResourceAssessmentStrategy.Percentile = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ResourceAssessmentStrategy.getJsonObj(obj))), {});
        return jsonObj;
    }
    PercentileResourceAssessmentStrategy.getJsonObj = getJsonObj;
    PercentileResourceAssessmentStrategy.strategyType = "PERCENTILE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.ResourceAssessmentStrategy.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    PercentileResourceAssessmentStrategy.getDeserializedJsonObj = getDeserializedJsonObj;
})(PercentileResourceAssessmentStrategy = exports.PercentileResourceAssessmentStrategy || (exports.PercentileResourceAssessmentStrategy = {}));
