// oci-cloudmigrations/lib/model/resource-assessment-strategy.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceAssessmentStrategy = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var ResourceAssessmentStrategy;
(function (ResourceAssessmentStrategy) {
    let ResourceType;
    (function (ResourceType) {
        ResourceType["Cpu"] = "CPU";
        ResourceType["Memory"] = "MEMORY";
        ResourceType["All"] = "ALL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResourceType = ResourceAssessmentStrategy.ResourceType || (ResourceAssessmentStrategy.ResourceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "strategyType" in obj && obj.strategyType) {
            switch (obj.strategyType) {
                case "PEAK":
                    return model.PeakResourceAssessmentStrategy.getJsonObj(jsonObj, true);
                case "PERCENTILE":
                    return model.PercentileResourceAssessmentStrategy.getJsonObj(jsonObj, true);
                case "AVERAGE":
                    return model.AverageResourceAssessmentStrategy.getJsonObj(jsonObj, true);
                case "AS_IS":
                    return model.AsIsResourceAssessmentStrategy.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.strategyType}`);
            }
        }
        return jsonObj;
    }
    ResourceAssessmentStrategy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "strategyType" in obj && obj.strategyType) {
            switch (obj.strategyType) {
                case "PEAK":
                    return model.PeakResourceAssessmentStrategy.getDeserializedJsonObj(jsonObj, true);
                case "PERCENTILE":
                    return model.PercentileResourceAssessmentStrategy.getDeserializedJsonObj(jsonObj, true);
                case "AVERAGE":
                    return model.AverageResourceAssessmentStrategy.getDeserializedJsonObj(jsonObj, true);
                case "AS_IS":
                    return model.AsIsResourceAssessmentStrategy.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.strategyType}`);
            }
        }
        return jsonObj;
    }
    ResourceAssessmentStrategy.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourceAssessmentStrategy = exports.ResourceAssessmentStrategy || (exports.ResourceAssessmentStrategy = {}));
