// oci-cloudmigrations/lib/model/target-asset.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetAsset = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var TargetAsset;
(function (TargetAsset) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = TargetAsset.LifecycleState || (TargetAsset.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "compatibilityMessages": obj.compatibilityMessages
                ? obj.compatibilityMessages.map(item => {
                    return model.CompatibilityMessage.getJsonObj(item);
                })
                : undefined,
            "estimatedCost": obj.estimatedCost
                ? model.CostEstimation.getJsonObj(obj.estimatedCost)
                : undefined,
            "migrationAsset": obj.migrationAsset
                ? model.MigrationAsset.getJsonObj(obj.migrationAsset)
                : undefined
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "INSTANCE":
                    return model.VmTargetAsset.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    TargetAsset.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "compatibilityMessages": obj.compatibilityMessages
                ? obj.compatibilityMessages.map(item => {
                    return model.CompatibilityMessage.getDeserializedJsonObj(item);
                })
                : undefined,
            "estimatedCost": obj.estimatedCost
                ? model.CostEstimation.getDeserializedJsonObj(obj.estimatedCost)
                : undefined,
            "migrationAsset": obj.migrationAsset
                ? model.MigrationAsset.getDeserializedJsonObj(obj.migrationAsset)
                : undefined
        });
        if (obj && "type" in obj && obj.type) {
            switch (obj.type) {
                case "INSTANCE":
                    return model.VmTargetAsset.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.type}`);
            }
        }
        return jsonObj;
    }
    TargetAsset.getDeserializedJsonObj = getDeserializedJsonObj;
})(TargetAsset = exports.TargetAsset || (exports.TargetAsset = {}));
