// oci-cloudmigrations/lib/model/vm-target-asset.js
"use strict";
/**
 * Oracle Cloud Migrations API
 * A description of the Oracle Cloud Migrations API.
 * OpenAPI spec version: 20220919
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmTargetAsset = void 0;
const model = __importStar(require("../model"));
var VmTargetAsset;
(function (VmTargetAsset) {
    let PreferredShapeType;
    (function (PreferredShapeType) {
        PreferredShapeType["Vm"] = "VM";
        PreferredShapeType["VmIntel"] = "VM_INTEL";
        PreferredShapeType["VmIntelStandard"] = "VM_INTEL_Standard";
        PreferredShapeType["VmIntelDensIo"] = "VM_INTEL_DensIO";
        PreferredShapeType["VmIntelGpu"] = "VM_INTEL_GPU";
        PreferredShapeType["VmIntelOptimized"] = "VM_INTEL_Optimized";
        PreferredShapeType["VmAmd"] = "VM_AMD";
        PreferredShapeType["VmAmdStandard"] = "VM_AMD_Standard";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PreferredShapeType["UnknownValue"] = "UNKNOWN_VALUE";
    })(PreferredShapeType = VmTargetAsset.PreferredShapeType || (VmTargetAsset.PreferredShapeType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.TargetAsset.getJsonObj(obj))), {
            "testSpec": obj.testSpec ? model.LaunchInstanceDetails.getJsonObj(obj.testSpec) : undefined,
            "userSpec": obj.userSpec ? model.LaunchInstanceDetails.getJsonObj(obj.userSpec) : undefined,
            "recommendedSpec": obj.recommendedSpec
                ? model.LaunchInstanceDetails.getJsonObj(obj.recommendedSpec)
                : undefined
        });
        return jsonObj;
    }
    VmTargetAsset.getJsonObj = getJsonObj;
    VmTargetAsset.type = "INSTANCE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.TargetAsset.getDeserializedJsonObj(obj))), {
            "testSpec": obj.testSpec
                ? model.LaunchInstanceDetails.getDeserializedJsonObj(obj.testSpec)
                : undefined,
            "userSpec": obj.userSpec
                ? model.LaunchInstanceDetails.getDeserializedJsonObj(obj.userSpec)
                : undefined,
            "recommendedSpec": obj.recommendedSpec
                ? model.LaunchInstanceDetails.getDeserializedJsonObj(obj.recommendedSpec)
                : undefined
        });
        return jsonObj;
    }
    VmTargetAsset.getDeserializedJsonObj = getDeserializedJsonObj;
})(VmTargetAsset = exports.VmTargetAsset || (exports.VmTargetAsset = {}));
