// oci-disasterrecovery/lib/model/dr-protection-group-role.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (FSDR) API to manage disaster recovery for business applications.
FSDR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster recovery
capabilities for all layers of an application stack, including infrastructure, middleware, database, and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrProtectionGroupRole = void 0;
/**
 * Enumeration of different roles of a DR Protection Group.
 * - PRIMARY - The DR Protection Group has a primary role in a DR confguration.
 * - STANDBY - The DR Protection Group has a standby role in a DR confguration.
 * - UNCONFIGURED - The DR Protection Group is not part of a DR confguration
 *
 **/
var DrProtectionGroupRole;
(function (DrProtectionGroupRole) {
    DrProtectionGroupRole["Primary"] = "PRIMARY";
    DrProtectionGroupRole["Standby"] = "STANDBY";
    DrProtectionGroupRole["Unconfigured"] = "UNCONFIGURED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrProtectionGroupRole["UnknownValue"] = "UNKNOWN_VALUE";
})(DrProtectionGroupRole = exports.DrProtectionGroupRole || (exports.DrProtectionGroupRole = {}));
(function (DrProtectionGroupRole) {
    function getJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupRole.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrProtectionGroupRole.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrProtectionGroupRole = exports.DrProtectionGroupRole || (exports.DrProtectionGroupRole = {}));
