// oci-disasterrecovery/lib/model/dr-plan-lifecycle-state.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (FSDR) API to manage disaster recovery for business applications.
FSDR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster recovery
capabilities for all layers of an application stack, including infrastructure, middleware, database, and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrPlanLifecycleState = void 0;
/**
 * Possible lifecycle states of a DR Plan.
 * - CREATING - DR Plan is being created.
 * - UPDATING - DR Plan is being updated.
 * - ACTIVE - DR Plan is active and available for operations.
 * - INACTIVE - DR Plan is inactive and is not available for operations.
 * - DELETING - DR Plan is being deleted.
 * - DELETED - DR Plan has been deleted.
 * - FAILED - DR Plan creation failed.
 * - NEEDS_ATTENTION - DR Plan has failed to update.
 *
 **/
var DrPlanLifecycleState;
(function (DrPlanLifecycleState) {
    DrPlanLifecycleState["Creating"] = "CREATING";
    DrPlanLifecycleState["Updating"] = "UPDATING";
    DrPlanLifecycleState["Active"] = "ACTIVE";
    DrPlanLifecycleState["Inactive"] = "INACTIVE";
    DrPlanLifecycleState["Deleting"] = "DELETING";
    DrPlanLifecycleState["Deleted"] = "DELETED";
    DrPlanLifecycleState["Failed"] = "FAILED";
    DrPlanLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DrPlanLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(DrPlanLifecycleState = exports.DrPlanLifecycleState || (exports.DrPlanLifecycleState = {}));
(function (DrPlanLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DrPlanLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrPlanLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrPlanLifecycleState = exports.DrPlanLifecycleState || (exports.DrPlanLifecycleState = {}));
