// oci-ailanguage/lib/model/index.js
"use strict";
/**
 * Language API
 * OCI Language Service solutions can help enterprise customers integrate AI into their products immediately using our proven,
pre-trained and custom models or containers, without a need to set up an house team of AI and ML experts.
This allows enterprises to focus on business drivers and development work rather than AI and ML operations, which shortens the time to market.

 * OpenAPI spec version: 20221001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EvaluationResults = exports.EvaluationResultSummary = exports.EvaluationResultCollection = exports.ErrorDetails = exports.EntityMetrics = exports.EntityLabelErrorAnalysis = exports.EntityDocumentResult = exports.Entity = exports.EndpointSummary = exports.EndpointCollection = exports.Endpoint = exports.DominantLanguageDocumentResult = exports.DominantLanguageDocument = exports.DocumentError = exports.DetectedLanguage = exports.DetectLanguageTextClassificationResult = exports.DetectLanguageTextClassificationDetails = exports.DetectLanguageSentimentsResult = exports.DetectLanguageSentimentsDetails = exports.DetectLanguageKeyPhrasesResult = exports.DetectLanguageKeyPhrasesDetails = exports.DetectLanguageEntitiesResult = exports.DetectLanguageEntitiesDetails = exports.DetectDominantLanguageResult = exports.DetectDominantLanguageDetails = exports.DatasetDetails = exports.CreateProjectDetails = exports.CreateModelDetails = exports.CreateEndpointDetails = exports.ConfusionMatrixDetails = exports.ClassificationType = exports.ClassMetrics = exports.ChangeProjectCompartmentDetails = exports.ChangeModelCompartmentDetails = exports.ChangeEndpointCompartmentDetails = exports.BatchLanguageTranslationResult = exports.BatchLanguageTranslationDetails = exports.BatchDetectLanguageTextClassificationResult = exports.BatchDetectLanguageTextClassificationDetails = exports.BatchDetectLanguageSentimentsResult = exports.BatchDetectLanguageSentimentsDetails = exports.BatchDetectLanguagePiiEntitiesResult = exports.BatchDetectLanguagePiiEntitiesDetails = exports.BatchDetectLanguageKeyPhrasesResult = exports.BatchDetectLanguageKeyPhrasesDetails = exports.BatchDetectLanguageEntitiesResult = exports.BatchDetectLanguageEntitiesDetails = exports.BatchDetectDominantLanguageResult = exports.BatchDetectDominantLanguageDetails = exports.ActionType = void 0;
exports.PiiEntityRemove = exports.PiiEntityMask = exports.ObjectStorageDataset = exports.ObjectListDataset = exports.NamedEntityRecognitionModelDetails = exports.NamedEntityRecognitionEvaluationResults = exports.NamedEntityRecognitionEvaluationResult = exports.DataScienceLabelingDataset = exports.ClassificationMultiLabelModeDetails = exports.ClassificationMultiClassModeDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogCollection = exports.WorkRequestLog = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateProjectDetails = exports.UpdateModelDetails = exports.UpdateEndpointDetails = exports.TranslationDocumentResult = exports.TextDocument = exports.TextClassificationModelMetrics = exports.TextClassificationDocumentResult = exports.TextClassification = exports.TestStrategy = exports.SortOrder = exports.SentimentSentence = exports.SentimentDocumentResult = exports.SentimentAspect = exports.ProjectSummary = exports.ProjectCollection = exports.Project = exports.PreDeployedLanguageModels = exports.PiiEntityMasking = exports.PiiEntityDocumentResult = exports.PiiEntity = exports.OperationType = exports.OperationStatus = exports.NerModelVersion = exports.NamedEntityRecognitionModelMetrics = exports.ModelSummary = exports.ModelDetails = exports.ModelCollection = exports.Model = exports.LocationDetails = exports.KeyPhraseDocumentResult = exports.KeyPhrase = exports.HierarchicalEntity = void 0;
exports.TextClassificationModelEvaluationResult = exports.TextClassificationModelDetails = exports.TextClassificationEvaluationResults = exports.TestAndValidationDatasetStrategy = exports.PiiEntityReplace = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const BatchDetectDominantLanguageDetails = __importStar(require("./batch-detect-dominant-language-details"));
exports.BatchDetectDominantLanguageDetails = BatchDetectDominantLanguageDetails.BatchDetectDominantLanguageDetails;
const BatchDetectDominantLanguageResult = __importStar(require("./batch-detect-dominant-language-result"));
exports.BatchDetectDominantLanguageResult = BatchDetectDominantLanguageResult.BatchDetectDominantLanguageResult;
const BatchDetectLanguageEntitiesDetails = __importStar(require("./batch-detect-language-entities-details"));
exports.BatchDetectLanguageEntitiesDetails = BatchDetectLanguageEntitiesDetails.BatchDetectLanguageEntitiesDetails;
const BatchDetectLanguageEntitiesResult = __importStar(require("./batch-detect-language-entities-result"));
exports.BatchDetectLanguageEntitiesResult = BatchDetectLanguageEntitiesResult.BatchDetectLanguageEntitiesResult;
const BatchDetectLanguageKeyPhrasesDetails = __importStar(require("./batch-detect-language-key-phrases-details"));
exports.BatchDetectLanguageKeyPhrasesDetails = BatchDetectLanguageKeyPhrasesDetails.BatchDetectLanguageKeyPhrasesDetails;
const BatchDetectLanguageKeyPhrasesResult = __importStar(require("./batch-detect-language-key-phrases-result"));
exports.BatchDetectLanguageKeyPhrasesResult = BatchDetectLanguageKeyPhrasesResult.BatchDetectLanguageKeyPhrasesResult;
const BatchDetectLanguagePiiEntitiesDetails = __importStar(require("./batch-detect-language-pii-entities-details"));
exports.BatchDetectLanguagePiiEntitiesDetails = BatchDetectLanguagePiiEntitiesDetails.BatchDetectLanguagePiiEntitiesDetails;
const BatchDetectLanguagePiiEntitiesResult = __importStar(require("./batch-detect-language-pii-entities-result"));
exports.BatchDetectLanguagePiiEntitiesResult = BatchDetectLanguagePiiEntitiesResult.BatchDetectLanguagePiiEntitiesResult;
const BatchDetectLanguageSentimentsDetails = __importStar(require("./batch-detect-language-sentiments-details"));
exports.BatchDetectLanguageSentimentsDetails = BatchDetectLanguageSentimentsDetails.BatchDetectLanguageSentimentsDetails;
const BatchDetectLanguageSentimentsResult = __importStar(require("./batch-detect-language-sentiments-result"));
exports.BatchDetectLanguageSentimentsResult = BatchDetectLanguageSentimentsResult.BatchDetectLanguageSentimentsResult;
const BatchDetectLanguageTextClassificationDetails = __importStar(require("./batch-detect-language-text-classification-details"));
exports.BatchDetectLanguageTextClassificationDetails = BatchDetectLanguageTextClassificationDetails.BatchDetectLanguageTextClassificationDetails;
const BatchDetectLanguageTextClassificationResult = __importStar(require("./batch-detect-language-text-classification-result"));
exports.BatchDetectLanguageTextClassificationResult = BatchDetectLanguageTextClassificationResult.BatchDetectLanguageTextClassificationResult;
const BatchLanguageTranslationDetails = __importStar(require("./batch-language-translation-details"));
exports.BatchLanguageTranslationDetails = BatchLanguageTranslationDetails.BatchLanguageTranslationDetails;
const BatchLanguageTranslationResult = __importStar(require("./batch-language-translation-result"));
exports.BatchLanguageTranslationResult = BatchLanguageTranslationResult.BatchLanguageTranslationResult;
const ChangeEndpointCompartmentDetails = __importStar(require("./change-endpoint-compartment-details"));
exports.ChangeEndpointCompartmentDetails = ChangeEndpointCompartmentDetails.ChangeEndpointCompartmentDetails;
const ChangeModelCompartmentDetails = __importStar(require("./change-model-compartment-details"));
exports.ChangeModelCompartmentDetails = ChangeModelCompartmentDetails.ChangeModelCompartmentDetails;
const ChangeProjectCompartmentDetails = __importStar(require("./change-project-compartment-details"));
exports.ChangeProjectCompartmentDetails = ChangeProjectCompartmentDetails.ChangeProjectCompartmentDetails;
const ClassMetrics = __importStar(require("./class-metrics"));
exports.ClassMetrics = ClassMetrics.ClassMetrics;
const ClassificationType = __importStar(require("./classification-type"));
exports.ClassificationType = ClassificationType.ClassificationType;
const ConfusionMatrixDetails = __importStar(require("./confusion-matrix-details"));
exports.ConfusionMatrixDetails = ConfusionMatrixDetails.ConfusionMatrixDetails;
const CreateEndpointDetails = __importStar(require("./create-endpoint-details"));
exports.CreateEndpointDetails = CreateEndpointDetails.CreateEndpointDetails;
const CreateModelDetails = __importStar(require("./create-model-details"));
exports.CreateModelDetails = CreateModelDetails.CreateModelDetails;
const CreateProjectDetails = __importStar(require("./create-project-details"));
exports.CreateProjectDetails = CreateProjectDetails.CreateProjectDetails;
const DatasetDetails = __importStar(require("./dataset-details"));
exports.DatasetDetails = DatasetDetails.DatasetDetails;
const DetectDominantLanguageDetails = __importStar(require("./detect-dominant-language-details"));
exports.DetectDominantLanguageDetails = DetectDominantLanguageDetails.DetectDominantLanguageDetails;
const DetectDominantLanguageResult = __importStar(require("./detect-dominant-language-result"));
exports.DetectDominantLanguageResult = DetectDominantLanguageResult.DetectDominantLanguageResult;
const DetectLanguageEntitiesDetails = __importStar(require("./detect-language-entities-details"));
exports.DetectLanguageEntitiesDetails = DetectLanguageEntitiesDetails.DetectLanguageEntitiesDetails;
const DetectLanguageEntitiesResult = __importStar(require("./detect-language-entities-result"));
exports.DetectLanguageEntitiesResult = DetectLanguageEntitiesResult.DetectLanguageEntitiesResult;
const DetectLanguageKeyPhrasesDetails = __importStar(require("./detect-language-key-phrases-details"));
exports.DetectLanguageKeyPhrasesDetails = DetectLanguageKeyPhrasesDetails.DetectLanguageKeyPhrasesDetails;
const DetectLanguageKeyPhrasesResult = __importStar(require("./detect-language-key-phrases-result"));
exports.DetectLanguageKeyPhrasesResult = DetectLanguageKeyPhrasesResult.DetectLanguageKeyPhrasesResult;
const DetectLanguageSentimentsDetails = __importStar(require("./detect-language-sentiments-details"));
exports.DetectLanguageSentimentsDetails = DetectLanguageSentimentsDetails.DetectLanguageSentimentsDetails;
const DetectLanguageSentimentsResult = __importStar(require("./detect-language-sentiments-result"));
exports.DetectLanguageSentimentsResult = DetectLanguageSentimentsResult.DetectLanguageSentimentsResult;
const DetectLanguageTextClassificationDetails = __importStar(require("./detect-language-text-classification-details"));
exports.DetectLanguageTextClassificationDetails = DetectLanguageTextClassificationDetails.DetectLanguageTextClassificationDetails;
const DetectLanguageTextClassificationResult = __importStar(require("./detect-language-text-classification-result"));
exports.DetectLanguageTextClassificationResult = DetectLanguageTextClassificationResult.DetectLanguageTextClassificationResult;
const DetectedLanguage = __importStar(require("./detected-language"));
exports.DetectedLanguage = DetectedLanguage.DetectedLanguage;
const DocumentError = __importStar(require("./document-error"));
exports.DocumentError = DocumentError.DocumentError;
const DominantLanguageDocument = __importStar(require("./dominant-language-document"));
exports.DominantLanguageDocument = DominantLanguageDocument.DominantLanguageDocument;
const DominantLanguageDocumentResult = __importStar(require("./dominant-language-document-result"));
exports.DominantLanguageDocumentResult = DominantLanguageDocumentResult.DominantLanguageDocumentResult;
const Endpoint = __importStar(require("./endpoint"));
exports.Endpoint = Endpoint.Endpoint;
const EndpointCollection = __importStar(require("./endpoint-collection"));
exports.EndpointCollection = EndpointCollection.EndpointCollection;
const EndpointSummary = __importStar(require("./endpoint-summary"));
exports.EndpointSummary = EndpointSummary.EndpointSummary;
const Entity = __importStar(require("./entity"));
exports.Entity = Entity.Entity;
const EntityDocumentResult = __importStar(require("./entity-document-result"));
exports.EntityDocumentResult = EntityDocumentResult.EntityDocumentResult;
const EntityLabelErrorAnalysis = __importStar(require("./entity-label-error-analysis"));
exports.EntityLabelErrorAnalysis = EntityLabelErrorAnalysis.EntityLabelErrorAnalysis;
const EntityMetrics = __importStar(require("./entity-metrics"));
exports.EntityMetrics = EntityMetrics.EntityMetrics;
const ErrorDetails = __importStar(require("./error-details"));
exports.ErrorDetails = ErrorDetails.ErrorDetails;
const EvaluationResultCollection = __importStar(require("./evaluation-result-collection"));
exports.EvaluationResultCollection = EvaluationResultCollection.EvaluationResultCollection;
const EvaluationResultSummary = __importStar(require("./evaluation-result-summary"));
exports.EvaluationResultSummary = EvaluationResultSummary.EvaluationResultSummary;
const EvaluationResults = __importStar(require("./evaluation-results"));
exports.EvaluationResults = EvaluationResults.EvaluationResults;
const HierarchicalEntity = __importStar(require("./hierarchical-entity"));
exports.HierarchicalEntity = HierarchicalEntity.HierarchicalEntity;
const KeyPhrase = __importStar(require("./key-phrase"));
exports.KeyPhrase = KeyPhrase.KeyPhrase;
const KeyPhraseDocumentResult = __importStar(require("./key-phrase-document-result"));
exports.KeyPhraseDocumentResult = KeyPhraseDocumentResult.KeyPhraseDocumentResult;
const LocationDetails = __importStar(require("./location-details"));
exports.LocationDetails = LocationDetails.LocationDetails;
const Model = __importStar(require("./model"));
exports.Model = Model.Model;
const ModelCollection = __importStar(require("./model-collection"));
exports.ModelCollection = ModelCollection.ModelCollection;
const ModelDetails = __importStar(require("./model-details"));
exports.ModelDetails = ModelDetails.ModelDetails;
const ModelSummary = __importStar(require("./model-summary"));
exports.ModelSummary = ModelSummary.ModelSummary;
const NamedEntityRecognitionModelMetrics = __importStar(require("./named-entity-recognition-model-metrics"));
exports.NamedEntityRecognitionModelMetrics = NamedEntityRecognitionModelMetrics.NamedEntityRecognitionModelMetrics;
const NerModelVersion = __importStar(require("./ner-model-version"));
exports.NerModelVersion = NerModelVersion.NerModelVersion;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PiiEntity = __importStar(require("./pii-entity"));
exports.PiiEntity = PiiEntity.PiiEntity;
const PiiEntityDocumentResult = __importStar(require("./pii-entity-document-result"));
exports.PiiEntityDocumentResult = PiiEntityDocumentResult.PiiEntityDocumentResult;
const PiiEntityMasking = __importStar(require("./pii-entity-masking"));
exports.PiiEntityMasking = PiiEntityMasking.PiiEntityMasking;
const PreDeployedLanguageModels = __importStar(require("./pre-deployed-language-models"));
exports.PreDeployedLanguageModels = PreDeployedLanguageModels.PreDeployedLanguageModels;
const Project = __importStar(require("./project"));
exports.Project = Project.Project;
const ProjectCollection = __importStar(require("./project-collection"));
exports.ProjectCollection = ProjectCollection.ProjectCollection;
const ProjectSummary = __importStar(require("./project-summary"));
exports.ProjectSummary = ProjectSummary.ProjectSummary;
const SentimentAspect = __importStar(require("./sentiment-aspect"));
exports.SentimentAspect = SentimentAspect.SentimentAspect;
const SentimentDocumentResult = __importStar(require("./sentiment-document-result"));
exports.SentimentDocumentResult = SentimentDocumentResult.SentimentDocumentResult;
const SentimentSentence = __importStar(require("./sentiment-sentence"));
exports.SentimentSentence = SentimentSentence.SentimentSentence;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TestStrategy = __importStar(require("./test-strategy"));
exports.TestStrategy = TestStrategy.TestStrategy;
const TextClassification = __importStar(require("./text-classification"));
exports.TextClassification = TextClassification.TextClassification;
const TextClassificationDocumentResult = __importStar(require("./text-classification-document-result"));
exports.TextClassificationDocumentResult = TextClassificationDocumentResult.TextClassificationDocumentResult;
const TextClassificationModelMetrics = __importStar(require("./text-classification-model-metrics"));
exports.TextClassificationModelMetrics = TextClassificationModelMetrics.TextClassificationModelMetrics;
const TextDocument = __importStar(require("./text-document"));
exports.TextDocument = TextDocument.TextDocument;
const TranslationDocumentResult = __importStar(require("./translation-document-result"));
exports.TranslationDocumentResult = TranslationDocumentResult.TranslationDocumentResult;
const UpdateEndpointDetails = __importStar(require("./update-endpoint-details"));
exports.UpdateEndpointDetails = UpdateEndpointDetails.UpdateEndpointDetails;
const UpdateModelDetails = __importStar(require("./update-model-details"));
exports.UpdateModelDetails = UpdateModelDetails.UpdateModelDetails;
const UpdateProjectDetails = __importStar(require("./update-project-details"));
exports.UpdateProjectDetails = UpdateProjectDetails.UpdateProjectDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLog = __importStar(require("./work-request-log"));
exports.WorkRequestLog = WorkRequestLog.WorkRequestLog;
const WorkRequestLogCollection = __importStar(require("./work-request-log-collection"));
exports.WorkRequestLogCollection = WorkRequestLogCollection.WorkRequestLogCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ClassificationMultiClassModeDetails = __importStar(require("./classification-multi-class-mode-details"));
exports.ClassificationMultiClassModeDetails = ClassificationMultiClassModeDetails.ClassificationMultiClassModeDetails;
const ClassificationMultiLabelModeDetails = __importStar(require("./classification-multi-label-mode-details"));
exports.ClassificationMultiLabelModeDetails = ClassificationMultiLabelModeDetails.ClassificationMultiLabelModeDetails;
const DataScienceLabelingDataset = __importStar(require("./data-science-labeling-dataset"));
exports.DataScienceLabelingDataset = DataScienceLabelingDataset.DataScienceLabelingDataset;
const NamedEntityRecognitionEvaluationResult = __importStar(require("./named-entity-recognition-evaluation-result"));
exports.NamedEntityRecognitionEvaluationResult = NamedEntityRecognitionEvaluationResult.NamedEntityRecognitionEvaluationResult;
const NamedEntityRecognitionEvaluationResults = __importStar(require("./named-entity-recognition-evaluation-results"));
exports.NamedEntityRecognitionEvaluationResults = NamedEntityRecognitionEvaluationResults.NamedEntityRecognitionEvaluationResults;
const NamedEntityRecognitionModelDetails = __importStar(require("./named-entity-recognition-model-details"));
exports.NamedEntityRecognitionModelDetails = NamedEntityRecognitionModelDetails.NamedEntityRecognitionModelDetails;
const ObjectListDataset = __importStar(require("./object-list-dataset"));
exports.ObjectListDataset = ObjectListDataset.ObjectListDataset;
const ObjectStorageDataset = __importStar(require("./object-storage-dataset"));
exports.ObjectStorageDataset = ObjectStorageDataset.ObjectStorageDataset;
const PiiEntityMask = __importStar(require("./pii-entity-mask"));
exports.PiiEntityMask = PiiEntityMask.PiiEntityMask;
const PiiEntityRemove = __importStar(require("./pii-entity-remove"));
exports.PiiEntityRemove = PiiEntityRemove.PiiEntityRemove;
const PiiEntityReplace = __importStar(require("./pii-entity-replace"));
exports.PiiEntityReplace = PiiEntityReplace.PiiEntityReplace;
const TestAndValidationDatasetStrategy = __importStar(require("./test-and-validation-dataset-strategy"));
exports.TestAndValidationDatasetStrategy = TestAndValidationDatasetStrategy.TestAndValidationDatasetStrategy;
const TextClassificationEvaluationResults = __importStar(require("./text-classification-evaluation-results"));
exports.TextClassificationEvaluationResults = TextClassificationEvaluationResults.TextClassificationEvaluationResults;
const TextClassificationModelDetails = __importStar(require("./text-classification-model-details"));
exports.TextClassificationModelDetails = TextClassificationModelDetails.TextClassificationModelDetails;
const TextClassificationModelEvaluationResult = __importStar(require("./text-classification-model-evaluation-result"));
exports.TextClassificationModelEvaluationResult = TextClassificationModelEvaluationResult.TextClassificationModelEvaluationResult;
