// oci-opsi/lib/model/awr-database-report.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwrDatabaseReport = void 0;
const model = __importStar(require("../model"));
var AwrDatabaseReport;
(function (AwrDatabaseReport) {
    let Format;
    (function (Format) {
        Format["Html"] = "HTML";
        Format["Text"] = "TEXT";
        Format["Xml"] = "XML";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Format["UnknownValue"] = "UNKNOWN_VALUE";
    })(Format = AwrDatabaseReport.Format || (AwrDatabaseReport.Format = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.AwrQueryResult.getJsonObj(obj))), {});
        return jsonObj;
    }
    AwrDatabaseReport.getJsonObj = getJsonObj;
    AwrDatabaseReport.awrResultType = "AWRDB_DB_REPORT";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.AwrQueryResult.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    AwrDatabaseReport.getDeserializedJsonObj = getDeserializedJsonObj;
})(AwrDatabaseReport = exports.AwrDatabaseReport || (exports.AwrDatabaseReport = {}));
