// oci-opsi/lib/model/configuration-item-type.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationItemType = void 0;
/**
 * Type of configuration item.
 **/
var ConfigurationItemType;
(function (ConfigurationItemType) {
    ConfigurationItemType["Basic"] = "BASIC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConfigurationItemType["UnknownValue"] = "UNKNOWN_VALUE";
})(ConfigurationItemType = exports.ConfigurationItemType || (exports.ConfigurationItemType = {}));
(function (ConfigurationItemType) {
    function getJsonObj(obj) {
        return obj;
    }
    ConfigurationItemType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ConfigurationItemType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConfigurationItemType = exports.ConfigurationItemType || (exports.ConfigurationItemType = {}));
