// oci-opsi/lib/model/data-object-time-column-unit.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataObjectTimeColumnUnit = void 0;
const model = __importStar(require("../model"));
var DataObjectTimeColumnUnit;
(function (DataObjectTimeColumnUnit) {
    let Unit;
    (function (Unit) {
        Unit["NanoSecond"] = "NANO_SECOND";
        Unit["MicroSecond"] = "MICRO_SECOND";
        Unit["MilliSecond"] = "MILLI_SECOND";
        Unit["CentiSecond"] = "CENTI_SECOND";
        Unit["Second"] = "SECOND";
        Unit["Hour"] = "HOUR";
        Unit["Day"] = "DAY";
        Unit["Week"] = "WEEK";
        Unit["Month"] = "MONTH";
        Unit["Year"] = "YEAR";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Unit["UnknownValue"] = "UNKNOWN_VALUE";
    })(Unit = DataObjectTimeColumnUnit.Unit || (DataObjectTimeColumnUnit.Unit = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.DataObjectColumnUnit.getJsonObj(obj))), {});
        return jsonObj;
    }
    DataObjectTimeColumnUnit.getJsonObj = getJsonObj;
    DataObjectTimeColumnUnit.unitCategory = "TIME";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.DataObjectColumnUnit.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    DataObjectTimeColumnUnit.getDeserializedJsonObj = getDeserializedJsonObj;
})(DataObjectTimeColumnUnit = exports.DataObjectTimeColumnUnit || (exports.DataObjectTimeColumnUnit = {}));
