// oci-opsi/lib/model/operations-insights-warehouse-user-lifecycle-state.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationsInsightsWarehouseUserLifecycleState = void 0;
/**
 * Possible lifecycle states.
 **/
var OperationsInsightsWarehouseUserLifecycleState;
(function (OperationsInsightsWarehouseUserLifecycleState) {
    OperationsInsightsWarehouseUserLifecycleState["Creating"] = "CREATING";
    OperationsInsightsWarehouseUserLifecycleState["Updating"] = "UPDATING";
    OperationsInsightsWarehouseUserLifecycleState["Active"] = "ACTIVE";
    OperationsInsightsWarehouseUserLifecycleState["Deleting"] = "DELETING";
    OperationsInsightsWarehouseUserLifecycleState["Deleted"] = "DELETED";
    OperationsInsightsWarehouseUserLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationsInsightsWarehouseUserLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationsInsightsWarehouseUserLifecycleState = exports.OperationsInsightsWarehouseUserLifecycleState || (exports.OperationsInsightsWarehouseUserLifecycleState = {}));
(function (OperationsInsightsWarehouseUserLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationsInsightsWarehouseUserLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationsInsightsWarehouseUserLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationsInsightsWarehouseUserLifecycleState = exports.OperationsInsightsWarehouseUserLifecycleState || (exports.OperationsInsightsWarehouseUserLifecycleState = {}));
