// oci-opsi/lib/model/summarize-exadata-insight-resource-usage-aggregation.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeExadataInsightResourceUsageAggregation = void 0;
var SummarizeExadataInsightResourceUsageAggregation;
(function (SummarizeExadataInsightResourceUsageAggregation) {
    let ExadataResourceMetric;
    (function (ExadataResourceMetric) {
        ExadataResourceMetric["Cpu"] = "CPU";
        ExadataResourceMetric["Storage"] = "STORAGE";
        ExadataResourceMetric["Io"] = "IO";
        ExadataResourceMetric["Memory"] = "MEMORY";
        ExadataResourceMetric["Iops"] = "IOPS";
        ExadataResourceMetric["Throughput"] = "THROUGHPUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExadataResourceMetric["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExadataResourceMetric = SummarizeExadataInsightResourceUsageAggregation.ExadataResourceMetric || (SummarizeExadataInsightResourceUsageAggregation.ExadataResourceMetric = {}));
    let ExadataResourceType;
    (function (ExadataResourceType) {
        ExadataResourceType["Database"] = "DATABASE";
        ExadataResourceType["Host"] = "HOST";
        ExadataResourceType["StorageServer"] = "STORAGE_SERVER";
        ExadataResourceType["Diskgroup"] = "DISKGROUP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExadataResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExadataResourceType = SummarizeExadataInsightResourceUsageAggregation.ExadataResourceType || (SummarizeExadataInsightResourceUsageAggregation.ExadataResourceType = {}));
    let UsageUnit;
    (function (UsageUnit) {
        UsageUnit["Cores"] = "CORES";
        UsageUnit["Gb"] = "GB";
        UsageUnit["Mbps"] = "MBPS";
        UsageUnit["Iops"] = "IOPS";
        UsageUnit["Percent"] = "PERCENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UsageUnit["UnknownValue"] = "UNKNOWN_VALUE";
    })(UsageUnit = SummarizeExadataInsightResourceUsageAggregation.UsageUnit || (SummarizeExadataInsightResourceUsageAggregation.UsageUnit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeExadataInsightResourceUsageAggregation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeExadataInsightResourceUsageAggregation.getDeserializedJsonObj = getDeserializedJsonObj;
})(SummarizeExadataInsightResourceUsageAggregation = exports.SummarizeExadataInsightResourceUsageAggregation || (exports.SummarizeExadataInsightResourceUsageAggregation = {}));
