// oci-opsi/lib/model/summarize-database-insight-resource-usage-aggregation.js
"use strict";
/**
 * Operations Insights API
 * Use the Operations Insights API to perform data extraction operations to obtain database
resource utilization, performance statistics, and reference information. For more information,
see [About Oracle Cloud Infrastructure Operations Insights](https://docs.cloud.oracle.com/en-us/iaas/operations-insights/doc/operations-insights.html).

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizeDatabaseInsightResourceUsageAggregation = void 0;
var SummarizeDatabaseInsightResourceUsageAggregation;
(function (SummarizeDatabaseInsightResourceUsageAggregation) {
    let ResourceMetric;
    (function (ResourceMetric) {
        ResourceMetric["Cpu"] = "CPU";
        ResourceMetric["Storage"] = "STORAGE";
        ResourceMetric["Io"] = "IO";
        ResourceMetric["Memory"] = "MEMORY";
        ResourceMetric["MemoryPga"] = "MEMORY_PGA";
        ResourceMetric["MemorySga"] = "MEMORY_SGA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResourceMetric["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResourceMetric = SummarizeDatabaseInsightResourceUsageAggregation.ResourceMetric || (SummarizeDatabaseInsightResourceUsageAggregation.ResourceMetric = {}));
    let UsageUnit;
    (function (UsageUnit) {
        UsageUnit["Cores"] = "CORES";
        UsageUnit["Gb"] = "GB";
        UsageUnit["Mbps"] = "MBPS";
        UsageUnit["Iops"] = "IOPS";
        UsageUnit["Percent"] = "PERCENT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UsageUnit["UnknownValue"] = "UNKNOWN_VALUE";
    })(UsageUnit = SummarizeDatabaseInsightResourceUsageAggregation.UsageUnit || (SummarizeDatabaseInsightResourceUsageAggregation.UsageUnit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeDatabaseInsightResourceUsageAggregation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SummarizeDatabaseInsightResourceUsageAggregation.getDeserializedJsonObj = getDeserializedJsonObj;
})(SummarizeDatabaseInsightResourceUsageAggregation = exports.SummarizeDatabaseInsightResourceUsageAggregation || (exports.SummarizeDatabaseInsightResourceUsageAggregation = {}));
