// oci-tenantmanagercontrolplane/lib/model/organization-tenancy-role.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20200801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationTenancyRole = void 0;
/**
 * Possible roles for the organization tenancy.
 **/
var OrganizationTenancyRole;
(function (OrganizationTenancyRole) {
    OrganizationTenancyRole["Parent"] = "PARENT";
    OrganizationTenancyRole["Child"] = "CHILD";
    OrganizationTenancyRole["None"] = "NONE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OrganizationTenancyRole["UnknownValue"] = "UNKNOWN_VALUE";
})(OrganizationTenancyRole = exports.OrganizationTenancyRole || (exports.OrganizationTenancyRole = {}));
(function (OrganizationTenancyRole) {
    function getJsonObj(obj) {
        return obj;
    }
    OrganizationTenancyRole.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OrganizationTenancyRole.getDeserializedJsonObj = getDeserializedJsonObj;
})(OrganizationTenancyRole = exports.OrganizationTenancyRole || (exports.OrganizationTenancyRole = {}));
