// oci-tenantmanagercontrolplane/lib/model/organization-tenancy-lifecycle-state.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20200801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationTenancyLifecycleState = void 0;
/**
 * Possible lifecycle states for the organization tenancy.
 **/
var OrganizationTenancyLifecycleState;
(function (OrganizationTenancyLifecycleState) {
    OrganizationTenancyLifecycleState["Creating"] = "CREATING";
    OrganizationTenancyLifecycleState["Active"] = "ACTIVE";
    OrganizationTenancyLifecycleState["Inactive"] = "INACTIVE";
    OrganizationTenancyLifecycleState["Deleted"] = "DELETED";
    OrganizationTenancyLifecycleState["Failed"] = "FAILED";
    OrganizationTenancyLifecycleState["Deleting"] = "DELETING";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OrganizationTenancyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(OrganizationTenancyLifecycleState = exports.OrganizationTenancyLifecycleState || (exports.OrganizationTenancyLifecycleState = {}));
(function (OrganizationTenancyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    OrganizationTenancyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OrganizationTenancyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(OrganizationTenancyLifecycleState = exports.OrganizationTenancyLifecycleState || (exports.OrganizationTenancyLifecycleState = {}));
