// oci-tenantmanagercontrolplane/lib/model/recipient-invitation-status.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20200801
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientInvitationStatus = void 0;
/**
 * Possible statuses for a recipient invitation.
 **/
var RecipientInvitationStatus;
(function (RecipientInvitationStatus) {
    RecipientInvitationStatus["Pending"] = "PENDING";
    RecipientInvitationStatus["Canceled"] = "CANCELED";
    RecipientInvitationStatus["Accepted"] = "ACCEPTED";
    RecipientInvitationStatus["Ignored"] = "IGNORED";
    RecipientInvitationStatus["Expired"] = "EXPIRED";
    RecipientInvitationStatus["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    RecipientInvitationStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(RecipientInvitationStatus = exports.RecipientInvitationStatus || (exports.RecipientInvitationStatus = {}));
(function (RecipientInvitationStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    RecipientInvitationStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    RecipientInvitationStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecipientInvitationStatus = exports.RecipientInvitationStatus || (exports.RecipientInvitationStatus = {}));
