// oci-mediaservices/lib/mediaservices-waiter.js
"use strict";
/**
 * Media Services API
 * Media Services (includes Media Flow and Media Streams) is a fully managed service for processing media (video) source content. Use Media Flow and Media Streams to transcode and package digital video using configurable workflows and stream video outputs.

Use the Media Services API to configure media workflows and run Media Flow jobs, create distribution channels, ingest assets, create Preview URLs and play assets. For more information, see [Media Flow](/iaas/Content/dms-mediaflow/home.htm) and [Media Streams](/iaas/Content/dms-mediastream/home.htm).

 * OpenAPI spec version: 20211101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaServicesWaiter = void 0;
const models = __importStar(require("./model"));
const oci_common_1 = require("oci-common");
class MediaServicesWaiter {
    constructor(client, config) {
        this.client = client;
        this.config = config;
    }
    /**
     * Waits forMediaAsset till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMediaAssetResponse | null (null in case of 404 response)
     */
    forMediaAsset(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getMediaAsset(request), response => targetStates.includes(response.mediaAsset.lifecycleState), targetStates.includes(models.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forMediaAssetDistributionChannelAttachment till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMediaAssetDistributionChannelAttachmentResponse
     */
    forMediaAssetDistributionChannelAttachment(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getMediaAssetDistributionChannelAttachment(request), response => targetStates.includes(response.mediaAssetDistributionChannelAttachment.lifecycleState));
        });
    }
    /**
     * Waits forMediaWorkflow till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMediaWorkflowResponse | null (null in case of 404 response)
     */
    forMediaWorkflow(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getMediaWorkflow(request), response => targetStates.includes(response.mediaWorkflow.lifecycleState), targetStates.includes(models.MediaWorkflow.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forMediaWorkflowConfiguration till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMediaWorkflowConfigurationResponse | null (null in case of 404 response)
     */
    forMediaWorkflowConfiguration(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getMediaWorkflowConfiguration(request), response => targetStates.includes(response.mediaWorkflowConfiguration.lifecycleState), targetStates.includes(models.MediaWorkflowConfiguration.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forMediaWorkflowJob till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetMediaWorkflowJobResponse
     */
    forMediaWorkflowJob(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericWaiter(this.config, () => this.client.getMediaWorkflowJob(request), response => targetStates.includes(response.mediaWorkflowJob.lifecycleState));
        });
    }
    /**
     * Waits forStreamCdnConfig till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetStreamCdnConfigResponse | null (null in case of 404 response)
     */
    forStreamCdnConfig(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getStreamCdnConfig(request), response => targetStates.includes(response.streamCdnConfig.lifecycleState), targetStates.includes(models.StreamCdnConfig.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forStreamDistributionChannel till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetStreamDistributionChannelResponse | null (null in case of 404 response)
     */
    forStreamDistributionChannel(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getStreamDistributionChannel(request), response => targetStates.includes(response.streamDistributionChannel.lifecycleState), targetStates.includes(models.StreamDistributionChannel.LifecycleState.Deleted));
        });
    }
    /**
     * Waits forStreamPackagingConfig till it reaches any of the provided states
     *
     * @param request the request to send
     * @param targetStates the desired states to wait for. The waiter will return once the resource reaches any of the provided states
     * @return response returns GetStreamPackagingConfigResponse | null (null in case of 404 response)
     */
    forStreamPackagingConfig(request, ...targetStates) {
        return __awaiter(this, void 0, void 0, function* () {
            return oci_common_1.genericTerminalConditionWaiter(this.config, () => this.client.getStreamPackagingConfig(request), response => targetStates.includes(response.streamPackagingConfig.lifecycleState), targetStates.includes(models.StreamPackagingConfig.LifecycleState.Deleted));
        });
    }
}
exports.MediaServicesWaiter = MediaServicesWaiter;
