// oci-filestorage/lib/request/delete-replication-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteReplicationRequest = void 0;
var DeleteReplicationRequest;
(function (DeleteReplicationRequest) {
    let DeleteMode;
    (function (DeleteMode) {
        DeleteMode["FinishCycleIfCapturingOrApplying"] = "FINISH_CYCLE_IF_CAPTURING_OR_APPLYING";
        DeleteMode["OneMoreCycle"] = "ONE_MORE_CYCLE";
        DeleteMode["FinishCycleIfApplying"] = "FINISH_CYCLE_IF_APPLYING";
    })(DeleteMode = DeleteReplicationRequest.DeleteMode || (DeleteReplicationRequest.DeleteMode = {}));
})(DeleteReplicationRequest = exports.DeleteReplicationRequest || (exports.DeleteReplicationRequest = {}));
