// oci-filestorage/lib/request/index.js
"use strict";
/**
 * File Storage API
 * Use the File Storage service API to manage file systems, mount targets, and snapshots.
For more information, see [Overview of File Storage](/iaas/Content/File/Concepts/filestorageoverview.htm).

 * OpenAPI spec version: 20171215
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSnapshotsRequest = exports.ListReplicationsRequest = exports.ListReplicationTargetsRequest = exports.ListMountTargetsRequest = exports.ListFilesystemSnapshotPoliciesRequest = exports.ListFileSystemsRequest = exports.ListExportsRequest = exports.ListExportSetsRequest = exports.DeleteReplicationRequest = void 0;
const DeleteReplicationRequest = __importStar(require("./delete-replication-request"));
exports.DeleteReplicationRequest = DeleteReplicationRequest.DeleteReplicationRequest;
const ListExportSetsRequest = __importStar(require("./list-export-sets-request"));
exports.ListExportSetsRequest = ListExportSetsRequest.ListExportSetsRequest;
const ListExportsRequest = __importStar(require("./list-exports-request"));
exports.ListExportsRequest = ListExportsRequest.ListExportsRequest;
const ListFileSystemsRequest = __importStar(require("./list-file-systems-request"));
exports.ListFileSystemsRequest = ListFileSystemsRequest.ListFileSystemsRequest;
const ListFilesystemSnapshotPoliciesRequest = __importStar(require("./list-filesystem-snapshot-policies-request"));
exports.ListFilesystemSnapshotPoliciesRequest = ListFilesystemSnapshotPoliciesRequest.ListFilesystemSnapshotPoliciesRequest;
const ListMountTargetsRequest = __importStar(require("./list-mount-targets-request"));
exports.ListMountTargetsRequest = ListMountTargetsRequest.ListMountTargetsRequest;
const ListReplicationTargetsRequest = __importStar(require("./list-replication-targets-request"));
exports.ListReplicationTargetsRequest = ListReplicationTargetsRequest.ListReplicationTargetsRequest;
const ListReplicationsRequest = __importStar(require("./list-replications-request"));
exports.ListReplicationsRequest = ListReplicationsRequest.ListReplicationsRequest;
const ListSnapshotsRequest = __importStar(require("./list-snapshots-request"));
exports.ListSnapshotsRequest = ListSnapshotsRequest.ListSnapshotsRequest;
