// oci-mysql/lib/model/channel-filter.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelFilter = void 0;
var ChannelFilter;
(function (ChannelFilter) {
    let Type;
    (function (Type) {
        Type["ReplicateDoDb"] = "REPLICATE_DO_DB";
        Type["ReplicateIgnoreDb"] = "REPLICATE_IGNORE_DB";
        Type["ReplicateDoTable"] = "REPLICATE_DO_TABLE";
        Type["ReplicateIgnoreTable"] = "REPLICATE_IGNORE_TABLE";
        Type["ReplicateWildDoTable"] = "REPLICATE_WILD_DO_TABLE";
        Type["ReplicateWildIgnoreTable"] = "REPLICATE_WILD_IGNORE_TABLE";
        Type["ReplicateRewriteDb"] = "REPLICATE_REWRITE_DB";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = ChannelFilter.Type || (ChannelFilter.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ChannelFilter.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ChannelFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(ChannelFilter = exports.ChannelFilter || (exports.ChannelFilter = {}));
