// oci-mysql/lib/model/crash-recovery-status.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CrashRecoveryStatus = void 0;
/**
 * Whether to run the DB System with InnoDB Redo Logs and the Double Write Buffer enabled or disabled,
 * and whether to enable or disable syncing of the Binary Logs.
 *
 **/
var CrashRecoveryStatus;
(function (CrashRecoveryStatus) {
    CrashRecoveryStatus["Enabled"] = "ENABLED";
    CrashRecoveryStatus["Disabled"] = "DISABLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CrashRecoveryStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(CrashRecoveryStatus = exports.CrashRecoveryStatus || (exports.CrashRecoveryStatus = {}));
(function (CrashRecoveryStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    CrashRecoveryStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CrashRecoveryStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(CrashRecoveryStatus = exports.CrashRecoveryStatus || (exports.CrashRecoveryStatus = {}));
