// oci-mysql/lib/model/create-deletion-policy-details.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDeletionPolicyDetails = void 0;
var CreateDeletionPolicyDetails;
(function (CreateDeletionPolicyDetails) {
    let AutomaticBackupRetention;
    (function (AutomaticBackupRetention) {
        AutomaticBackupRetention["Delete"] = "DELETE";
        AutomaticBackupRetention["Retain"] = "RETAIN";
    })(AutomaticBackupRetention = CreateDeletionPolicyDetails.AutomaticBackupRetention || (CreateDeletionPolicyDetails.AutomaticBackupRetention = {}));
    let FinalBackup;
    (function (FinalBackup) {
        FinalBackup["SkipFinalBackup"] = "SKIP_FINAL_BACKUP";
        FinalBackup["RequireFinalBackup"] = "REQUIRE_FINAL_BACKUP";
    })(FinalBackup = CreateDeletionPolicyDetails.FinalBackup || (CreateDeletionPolicyDetails.FinalBackup = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDeletionPolicyDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateDeletionPolicyDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateDeletionPolicyDetails = exports.CreateDeletionPolicyDetails || (exports.CreateDeletionPolicyDetails = {}));
