// oci-mysql/lib/model/deletion-policy-details.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeletionPolicyDetails = void 0;
var DeletionPolicyDetails;
(function (DeletionPolicyDetails) {
    let AutomaticBackupRetention;
    (function (AutomaticBackupRetention) {
        AutomaticBackupRetention["Delete"] = "DELETE";
        AutomaticBackupRetention["Retain"] = "RETAIN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutomaticBackupRetention["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutomaticBackupRetention = DeletionPolicyDetails.AutomaticBackupRetention || (DeletionPolicyDetails.AutomaticBackupRetention = {}));
    let FinalBackup;
    (function (FinalBackup) {
        FinalBackup["SkipFinalBackup"] = "SKIP_FINAL_BACKUP";
        FinalBackup["RequireFinalBackup"] = "REQUIRE_FINAL_BACKUP";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FinalBackup["UnknownValue"] = "UNKNOWN_VALUE";
    })(FinalBackup = DeletionPolicyDetails.FinalBackup || (DeletionPolicyDetails.FinalBackup = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeletionPolicyDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DeletionPolicyDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(DeletionPolicyDetails = exports.DeletionPolicyDetails || (exports.DeletionPolicyDetails = {}));
