// oci-aidocument/lib/model/dimensions.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Dimensions = void 0;
var Dimensions;
(function (Dimensions) {
    let Unit;
    (function (Unit) {
        Unit["Pixel"] = "PIXEL";
        Unit["Inch"] = "INCH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Unit["UnknownValue"] = "UNKNOWN_VALUE";
    })(Unit = Dimensions.Unit || (Dimensions.Unit = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Dimensions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    Dimensions.getDeserializedJsonObj = getDeserializedJsonObj;
})(Dimensions = exports.Dimensions || (exports.Dimensions = {}));
