// oci-aidocument/lib/model/document-type.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentType = void 0;
/**
 * The type of document.
 **/
var DocumentType;
(function (DocumentType) {
    DocumentType["Invoice"] = "INVOICE";
    DocumentType["Receipt"] = "RECEIPT";
    DocumentType["Resume"] = "RESUME";
    DocumentType["TaxForm"] = "TAX_FORM";
    DocumentType["DriverLicense"] = "DRIVER_LICENSE";
    DocumentType["Passport"] = "PASSPORT";
    DocumentType["BankStatement"] = "BANK_STATEMENT";
    DocumentType["Check"] = "CHECK";
    DocumentType["Payslip"] = "PAYSLIP";
    DocumentType["Others"] = "OTHERS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DocumentType["UnknownValue"] = "UNKNOWN_VALUE";
})(DocumentType = exports.DocumentType || (exports.DocumentType = {}));
(function (DocumentType) {
    function getJsonObj(obj) {
        return obj;
    }
    DocumentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DocumentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DocumentType = exports.DocumentType || (exports.DocumentType = {}));
