// oci-aidocument/lib/model/document-field.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentField = void 0;
const model = __importStar(require("../model"));
var DocumentField;
(function (DocumentField) {
    let FieldType;
    (function (FieldType) {
        FieldType["LineItemGroup"] = "LINE_ITEM_GROUP";
        FieldType["LineItem"] = "LINE_ITEM";
        FieldType["LineItemField"] = "LINE_ITEM_FIELD";
        FieldType["KeyValue"] = "KEY_VALUE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        FieldType["UnknownValue"] = "UNKNOWN_VALUE";
    })(FieldType = DocumentField.FieldType || (DocumentField.FieldType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "fieldLabel": obj.fieldLabel ? model.FieldLabel.getJsonObj(obj.fieldLabel) : undefined,
            "fieldName": obj.fieldName ? model.FieldName.getJsonObj(obj.fieldName) : undefined,
            "fieldValue": obj.fieldValue ? model.FieldValue.getJsonObj(obj.fieldValue) : undefined
        });
        return jsonObj;
    }
    DocumentField.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "fieldLabel": obj.fieldLabel
                ? model.FieldLabel.getDeserializedJsonObj(obj.fieldLabel)
                : undefined,
            "fieldName": obj.fieldName
                ? model.FieldName.getDeserializedJsonObj(obj.fieldName)
                : undefined,
            "fieldValue": obj.fieldValue
                ? model.FieldValue.getDeserializedJsonObj(obj.fieldValue)
                : undefined
        });
        return jsonObj;
    }
    DocumentField.getDeserializedJsonObj = getDeserializedJsonObj;
})(DocumentField = exports.DocumentField || (exports.DocumentField = {}));
