// oci-aidocument/lib/model/index.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueTime = exports.ValueString = exports.ValuePhoneNumber = exports.ValueNumber = exports.ValueInteger = exports.ValueDate = exports.ValueArray = exports.ObjectStorageLocations = exports.InlineDocumentContent = exports.GeneralProcessorConfig = exports.DocumentTextExtractionFeature = exports.DocumentTableExtractionFeature = exports.DocumentLanguageClassificationFeature = exports.DocumentKeyValueExtractionFeature = exports.DocumentClassificationFeature = exports.Word = exports.TableRow = exports.Table = exports.ProcessorType = exports.ProcessorJob = exports.ProcessorConfig = exports.ProcessingError = exports.Page = exports.OutputLocation = exports.ObjectLocation = exports.NormalizedVertex = exports.Line = exports.InputLocation = exports.FieldValue = exports.FieldName = exports.FieldLabel = exports.DocumentType = exports.DocumentMetadata = exports.DocumentField = exports.DocumentFeature = exports.Dimensions = exports.DetectedLanguage = exports.DetectedDocumentType = exports.CreateProcessorJobDetails = exports.Cell = exports.BoundingPolygon = exports.AnalyzeDocumentResult = void 0;
const AnalyzeDocumentResult = __importStar(require("./analyze-document-result"));
exports.AnalyzeDocumentResult = AnalyzeDocumentResult.AnalyzeDocumentResult;
const BoundingPolygon = __importStar(require("./bounding-polygon"));
exports.BoundingPolygon = BoundingPolygon.BoundingPolygon;
const Cell = __importStar(require("./cell"));
exports.Cell = Cell.Cell;
const CreateProcessorJobDetails = __importStar(require("./create-processor-job-details"));
exports.CreateProcessorJobDetails = CreateProcessorJobDetails.CreateProcessorJobDetails;
const DetectedDocumentType = __importStar(require("./detected-document-type"));
exports.DetectedDocumentType = DetectedDocumentType.DetectedDocumentType;
const DetectedLanguage = __importStar(require("./detected-language"));
exports.DetectedLanguage = DetectedLanguage.DetectedLanguage;
const Dimensions = __importStar(require("./dimensions"));
exports.Dimensions = Dimensions.Dimensions;
const DocumentFeature = __importStar(require("./document-feature"));
exports.DocumentFeature = DocumentFeature.DocumentFeature;
const DocumentField = __importStar(require("./document-field"));
exports.DocumentField = DocumentField.DocumentField;
const DocumentMetadata = __importStar(require("./document-metadata"));
exports.DocumentMetadata = DocumentMetadata.DocumentMetadata;
const DocumentType = __importStar(require("./document-type"));
exports.DocumentType = DocumentType.DocumentType;
const FieldLabel = __importStar(require("./field-label"));
exports.FieldLabel = FieldLabel.FieldLabel;
const FieldName = __importStar(require("./field-name"));
exports.FieldName = FieldName.FieldName;
const FieldValue = __importStar(require("./field-value"));
exports.FieldValue = FieldValue.FieldValue;
const InputLocation = __importStar(require("./input-location"));
exports.InputLocation = InputLocation.InputLocation;
const Line = __importStar(require("./line"));
exports.Line = Line.Line;
const NormalizedVertex = __importStar(require("./normalized-vertex"));
exports.NormalizedVertex = NormalizedVertex.NormalizedVertex;
const ObjectLocation = __importStar(require("./object-location"));
exports.ObjectLocation = ObjectLocation.ObjectLocation;
const OutputLocation = __importStar(require("./output-location"));
exports.OutputLocation = OutputLocation.OutputLocation;
const Page = __importStar(require("./page"));
exports.Page = Page.Page;
const ProcessingError = __importStar(require("./processing-error"));
exports.ProcessingError = ProcessingError.ProcessingError;
const ProcessorConfig = __importStar(require("./processor-config"));
exports.ProcessorConfig = ProcessorConfig.ProcessorConfig;
const ProcessorJob = __importStar(require("./processor-job"));
exports.ProcessorJob = ProcessorJob.ProcessorJob;
const ProcessorType = __importStar(require("./processor-type"));
exports.ProcessorType = ProcessorType.ProcessorType;
const Table = __importStar(require("./table"));
exports.Table = Table.Table;
const TableRow = __importStar(require("./table-row"));
exports.TableRow = TableRow.TableRow;
const Word = __importStar(require("./word"));
exports.Word = Word.Word;
const DocumentClassificationFeature = __importStar(require("./document-classification-feature"));
exports.DocumentClassificationFeature = DocumentClassificationFeature.DocumentClassificationFeature;
const DocumentKeyValueExtractionFeature = __importStar(require("./document-key-value-extraction-feature"));
exports.DocumentKeyValueExtractionFeature = DocumentKeyValueExtractionFeature.DocumentKeyValueExtractionFeature;
const DocumentLanguageClassificationFeature = __importStar(require("./document-language-classification-feature"));
exports.DocumentLanguageClassificationFeature = DocumentLanguageClassificationFeature.DocumentLanguageClassificationFeature;
const DocumentTableExtractionFeature = __importStar(require("./document-table-extraction-feature"));
exports.DocumentTableExtractionFeature = DocumentTableExtractionFeature.DocumentTableExtractionFeature;
const DocumentTextExtractionFeature = __importStar(require("./document-text-extraction-feature"));
exports.DocumentTextExtractionFeature = DocumentTextExtractionFeature.DocumentTextExtractionFeature;
const GeneralProcessorConfig = __importStar(require("./general-processor-config"));
exports.GeneralProcessorConfig = GeneralProcessorConfig.GeneralProcessorConfig;
const InlineDocumentContent = __importStar(require("./inline-document-content"));
exports.InlineDocumentContent = InlineDocumentContent.InlineDocumentContent;
const ObjectStorageLocations = __importStar(require("./object-storage-locations"));
exports.ObjectStorageLocations = ObjectStorageLocations.ObjectStorageLocations;
const ValueArray = __importStar(require("./value-array"));
exports.ValueArray = ValueArray.ValueArray;
const ValueDate = __importStar(require("./value-date"));
exports.ValueDate = ValueDate.ValueDate;
const ValueInteger = __importStar(require("./value-integer"));
exports.ValueInteger = ValueInteger.ValueInteger;
const ValueNumber = __importStar(require("./value-number"));
exports.ValueNumber = ValueNumber.ValueNumber;
const ValuePhoneNumber = __importStar(require("./value-phone-number"));
exports.ValuePhoneNumber = ValuePhoneNumber.ValuePhoneNumber;
const ValueString = __importStar(require("./value-string"));
exports.ValueString = ValueString.ValueString;
const ValueTime = __importStar(require("./value-time"));
exports.ValueTime = ValueTime.ValueTime;
