// oci-aidocument/lib/model/processor-job.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessorJob = void 0;
const model = __importStar(require("../model"));
var ProcessorJob;
(function (ProcessorJob) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Succeeded"] = "SUCCEEDED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Accepted"] = "ACCEPTED";
        LifecycleState["Canceled"] = "CANCELED";
        LifecycleState["InProgress"] = "IN_PROGRESS";
        LifecycleState["Canceling"] = "CANCELING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ProcessorJob.LifecycleState || (ProcessorJob.LifecycleState = {}));
    let LifecycleDetails;
    (function (LifecycleDetails) {
        LifecycleDetails["PartiallySucceeded"] = "PARTIALLY_SUCCEEDED";
        LifecycleDetails["CompletelyFailed"] = "COMPLETELY_FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleDetails["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleDetails = ProcessorJob.LifecycleDetails || (ProcessorJob.LifecycleDetails = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "processorConfig": obj.processorConfig
                ? model.ProcessorConfig.getJsonObj(obj.processorConfig)
                : undefined,
            "inputLocation": obj.inputLocation
                ? model.InputLocation.getJsonObj(obj.inputLocation)
                : undefined,
            "outputLocation": obj.outputLocation
                ? model.OutputLocation.getJsonObj(obj.outputLocation)
                : undefined
        });
        return jsonObj;
    }
    ProcessorJob.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "processorConfig": obj.processorConfig
                ? model.ProcessorConfig.getDeserializedJsonObj(obj.processorConfig)
                : undefined,
            "inputLocation": obj.inputLocation
                ? model.InputLocation.getDeserializedJsonObj(obj.inputLocation)
                : undefined,
            "outputLocation": obj.outputLocation
                ? model.OutputLocation.getDeserializedJsonObj(obj.outputLocation)
                : undefined
        });
        return jsonObj;
    }
    ProcessorJob.getDeserializedJsonObj = getDeserializedJsonObj;
})(ProcessorJob = exports.ProcessorJob || (exports.ProcessorJob = {}));
