// oci-dataintegration/lib/model/task-summary-from-rest-task.js
"use strict";
/**
 * Data Integration API
 * Use the Data Integration API to organize your data integration projects, create data flows, pipelines and tasks, and then publish, schedule, and run tasks that extract, transform, and load data. For more information, see [Data Integration](https://docs.oracle.com/iaas/data-integration/home.htm).
 * OpenAPI spec version: 20200430
 * Contact: di_dis_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskSummaryFromRestTask = void 0;
const model = __importStar(require("../model"));
var TaskSummaryFromRestTask;
(function (TaskSummaryFromRestTask) {
    let MethodType;
    (function (MethodType) {
        MethodType["Get"] = "GET";
        MethodType["Post"] = "POST";
        MethodType["Patch"] = "PATCH";
        MethodType["Delete"] = "DELETE";
        MethodType["Put"] = "PUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        MethodType["UnknownValue"] = "UNKNOWN_VALUE";
    })(MethodType = TaskSummaryFromRestTask.MethodType || (TaskSummaryFromRestTask.MethodType = {}));
    let ApiCallMode;
    (function (ApiCallMode) {
        ApiCallMode["Synchronous"] = "SYNCHRONOUS";
        ApiCallMode["AsyncOciWorkrequest"] = "ASYNC_OCI_WORKREQUEST";
        ApiCallMode["AsyncGeneric"] = "ASYNC_GENERIC";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ApiCallMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(ApiCallMode = TaskSummaryFromRestTask.ApiCallMode || (TaskSummaryFromRestTask.ApiCallMode = {}));
    let CancelMethodType;
    (function (CancelMethodType) {
        CancelMethodType["Get"] = "GET";
        CancelMethodType["Post"] = "POST";
        CancelMethodType["Patch"] = "PATCH";
        CancelMethodType["Delete"] = "DELETE";
        CancelMethodType["Put"] = "PUT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        CancelMethodType["UnknownValue"] = "UNKNOWN_VALUE";
    })(CancelMethodType = TaskSummaryFromRestTask.CancelMethodType || (TaskSummaryFromRestTask.CancelMethodType = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.TaskSummary.getJsonObj(obj))), {
            "authDetails": obj.authDetails ? model.AuthDetails.getJsonObj(obj.authDetails) : undefined,
            "authConfig": obj.authConfig ? model.AuthConfig.getJsonObj(obj.authConfig) : undefined,
            "endpoint": obj.endpoint ? model.Expression.getJsonObj(obj.endpoint) : undefined,
            "cancelEndpoint": obj.cancelEndpoint
                ? model.Expression.getJsonObj(obj.cancelEndpoint)
                : undefined,
            "executeRestCallConfig": obj.executeRestCallConfig
                ? model.ExecuteRestCallConfig.getJsonObj(obj.executeRestCallConfig)
                : undefined,
            "cancelRestCallConfig": obj.cancelRestCallConfig
                ? model.CancelRestCallConfig.getJsonObj(obj.cancelRestCallConfig)
                : undefined,
            "pollRestCallConfig": obj.pollRestCallConfig
                ? model.PollRestCallConfig.getJsonObj(obj.pollRestCallConfig)
                : undefined
        });
        return jsonObj;
    }
    TaskSummaryFromRestTask.getJsonObj = getJsonObj;
    TaskSummaryFromRestTask.modelType = "REST_TASK";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.TaskSummary.getDeserializedJsonObj(obj))), {
            "authDetails": obj.authDetails
                ? model.AuthDetails.getDeserializedJsonObj(obj.authDetails)
                : undefined,
            "authConfig": obj.authConfig
                ? model.AuthConfig.getDeserializedJsonObj(obj.authConfig)
                : undefined,
            "endpoint": obj.endpoint
                ? model.Expression.getDeserializedJsonObj(obj.endpoint)
                : undefined,
            "cancelEndpoint": obj.cancelEndpoint
                ? model.Expression.getDeserializedJsonObj(obj.cancelEndpoint)
                : undefined,
            "executeRestCallConfig": obj.executeRestCallConfig
                ? model.ExecuteRestCallConfig.getDeserializedJsonObj(obj.executeRestCallConfig)
                : undefined,
            "cancelRestCallConfig": obj.cancelRestCallConfig
                ? model.CancelRestCallConfig.getDeserializedJsonObj(obj.cancelRestCallConfig)
                : undefined,
            "pollRestCallConfig": obj.pollRestCallConfig
                ? model.PollRestCallConfig.getDeserializedJsonObj(obj.pollRestCallConfig)
                : undefined
        });
        return jsonObj;
    }
    TaskSummaryFromRestTask.getDeserializedJsonObj = getDeserializedJsonObj;
})(TaskSummaryFromRestTask = exports.TaskSummaryFromRestTask || (exports.TaskSummaryFromRestTask = {}));
