// oci-threatintelligence/lib/request/list-indicators-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListIndicatorsRequest = void 0;
var ListIndicatorsRequest;
(function (ListIndicatorsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["Confidence"] = "confidence";
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["TimeUpdated"] = "timeUpdated";
        SortBy["TimeLastSeen"] = "timeLastSeen";
    })(SortBy = ListIndicatorsRequest.SortBy || (ListIndicatorsRequest.SortBy = {}));
})(ListIndicatorsRequest = exports.ListIndicatorsRequest || (exports.ListIndicatorsRequest = {}));
